{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "2ce5ed0c",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import numpy as np\n",
    "import pydeseq2.default_inference\n",
    "import pydeseq2.ds\n",
    "import pydeseq2.dds"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "277a003e",
   "metadata": {},
   "source": [
    "# Jump to top"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "1f8bab3d",
   "metadata": {},
   "outputs": [],
   "source": [
    "###Define Thresholds\n",
    "\n",
    "rare_species_abundance_threshold = 6.5\n",
    "expressed_gene_reads_threshold = 2000\n",
    "number_of_samples_expressed_in_threshold = 21"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "1f56a29d",
   "metadata": {},
   "outputs": [],
   "source": [
    "transcriptome = pd.read_csv(\n",
    "    \"/Users/michael/Data/Luke_terrace_experiment/General_data/plate1_4_tpm_normalized.csv\",\n",
    "    index_col=0,\n",
    ")\n",
    "rows_to_drop_expression_data = [\n",
    "    \"A2450525897_n01_undetermined\",\n",
    "    \"A2449446903_n01_undetermined\",\n",
    "    \"B250508004_n01_undetermined\",\n",
    "    \"B2449500127_n01_undetermined\",\n",
    "]\n",
    "transcriptome = transcriptome.drop(index=rows_to_drop_expression_data)\n",
    "transcriptome = transcriptome.sort_index()\n",
    "\n",
    "metadata = pd.read_csv(\n",
    "    \"/Users/michael/Data/Luke_terrace_experiment/General_data/Plates_1_to_4_metadata_merged_luke.csv\",\n",
    "    index_col=0,\n",
    ")\n",
    "metadata = pd.read_csv(\n",
    "    \"/Users/michael/Data/Luke_terrace_experiment/General_data/Plates_1_to_4_metadata_merged_luke.csv\",\n",
    "    index_col=0,\n",
    ")\n",
    "metadata = metadata.drop(\n",
    "    columns=[\n",
    "        \"arb.sort\",\n",
    "        \"sample-id\",\n",
    "        \"Ambiguous Unstranded\",\n",
    "        \"Ambiguous Forward\",\n",
    "        \"Multimapping\",\n",
    "        \"Unmapped Over Mapped\",\n",
    "    ]\n",
    ")\n",
    "metadata[\"Date and Time\"] = metadata[\"date\"] + \" \" + metadata[\"time\"]\n",
    "luke_time_data_format = \"%-m/%-d/%y %-H:%-M\"\n",
    "metadata[\"Date and Time\"] = pd.to_datetime(\n",
    "    metadata[\"Date and Time\"], format=luke_time_data_format\n",
    ")\n",
    "\n",
    "unusable_samples = [\n",
    "    \"A2450525897_n01_LICRNA01_A06\",\n",
    "    \"A2450525897_n01_LICRNA01_A08\",\n",
    "    \"A2450525897_n01_LICRNA01_C11\",\n",
    "    \"A2450525897_n01_LICRNA01_D11\",\n",
    "    \"A2450525897_n01_LICRNA01_D12\",\n",
    "    \"A2449446903_n01_LICRNA02_F01\",\n",
    "    \"B2449500127_n01_LICRNA04_A04\",\n",
    "    \"B2449500127_n01_LICRNA04_A06\",\n",
    "    \"B2449500127_n01_LICRNA04_A07\",\n",
    "]\n",
    "borderline_unusable = [\n",
    "    \"A2450525897_n01_LICRNA01_F03\",\n",
    "    \"A2450525897_n01_LICRNA01_G07\",\n",
    "    \"A2449446903_n01_LICRNA02_A04\",\n",
    "    \"A2449446903_n01_LICRNA02_H10\",\n",
    "    \"B2449500127_n01_LICRNA04_H02\",\n",
    "    \"B2449500127_n01_LICRNA04_H11\",\n",
    "]\n",
    "all_unsable = borderline_unusable + unusable_samples\n",
    "\n",
    "raw_transcriptome = pd.read_csv(\n",
    "    \"/Users/michael/Data/Luke_terrace_experiment/General_data/plate1_4_for_norm.csv\",\n",
    "    index_col=0,\n",
    ")\n",
    "\n",
    "trimmed_transcriptome = transcriptome.drop(index=all_unsable)\n",
    "trimmed_metadata = metadata.drop(index=all_unsable)\n",
    "trimmed_raw_transcriptome = raw_transcriptome.drop(index=all_unsable)\n",
    "trimmed_metadata = trimmed_metadata.sort_index()\n",
    "trimmed_transcriptome = trimmed_transcriptome.dropna(how=\"all\", axis=0)\n",
    "trimmed_transcriptome = trimmed_transcriptome.dropna(how=\"all\", axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "20e8f132",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AT1G01010</th>\n",
       "      <th>AT1G01020</th>\n",
       "      <th>AT1G03987</th>\n",
       "      <th>AT1G01030</th>\n",
       "      <th>AT1G01040</th>\n",
       "      <th>AT1G03993</th>\n",
       "      <th>AT1G01046</th>\n",
       "      <th>AT1G01050</th>\n",
       "      <th>AT1G03997</th>\n",
       "      <th>AT1G01060</th>\n",
       "      <th>...</th>\n",
       "      <th>ArthCt124</th>\n",
       "      <th>ArthCr088</th>\n",
       "      <th>ArthCt112</th>\n",
       "      <th>ArthCp088</th>\n",
       "      <th>ArthCp086</th>\n",
       "      <th>ArthCt118</th>\n",
       "      <th>ArthCp083</th>\n",
       "      <th>ArthCt111</th>\n",
       "      <th>ArthCp084</th>\n",
       "      <th>ArthCp085</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A01</th>\n",
       "      <td>0.382839</td>\n",
       "      <td>11.929183</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.731527</td>\n",
       "      <td>3.602189</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>31.063543</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.966935</td>\n",
       "      <td>...</td>\n",
       "      <td>17.950907</td>\n",
       "      <td>44.209074</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>6.266498</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A02</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>5.826682</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.501594</td>\n",
       "      <td>3.439569</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>22.712417</td>\n",
       "      <td>0.0</td>\n",
       "      <td>18.186693</td>\n",
       "      <td>...</td>\n",
       "      <td>1.986484</td>\n",
       "      <td>9.688606</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.501032</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.519342</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.467317</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A03</th>\n",
       "      <td>1.021565</td>\n",
       "      <td>7.683520</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.301733</td>\n",
       "      <td>2.700528</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>42.431625</td>\n",
       "      <td>0.0</td>\n",
       "      <td>12.937632</td>\n",
       "      <td>...</td>\n",
       "      <td>7.983340</td>\n",
       "      <td>16.384320</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.983340</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.166972</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A05</th>\n",
       "      <td>0.732717</td>\n",
       "      <td>3.149144</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.947758</td>\n",
       "      <td>3.742586</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>26.541337</td>\n",
       "      <td>0.0</td>\n",
       "      <td>23.861601</td>\n",
       "      <td>...</td>\n",
       "      <td>17.178143</td>\n",
       "      <td>17.005325</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.628742</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A06</th>\n",
       "      <td>1.614233</td>\n",
       "      <td>1.238895</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.130092</td>\n",
       "      <td>4.215597</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>19.379509</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.006570</td>\n",
       "      <td>...</td>\n",
       "      <td>16.219201</td>\n",
       "      <td>43.599301</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.074213</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H08</th>\n",
       "      <td>0.074367</td>\n",
       "      <td>4.474710</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.383813</td>\n",
       "      <td>2.059206</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27.907884</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.118408</td>\n",
       "      <td>...</td>\n",
       "      <td>3.486989</td>\n",
       "      <td>4.209645</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.072920</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.948096</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.760798</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H09</th>\n",
       "      <td>2.232353</td>\n",
       "      <td>6.249002</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.873955</td>\n",
       "      <td>3.869254</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>40.686812</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.391606</td>\n",
       "      <td>...</td>\n",
       "      <td>9.640892</td>\n",
       "      <td>4.655562</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.085306</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.360071</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.758217</td>\n",
       "      <td>4.807925</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H10</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.953706</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>36.084398</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.975538</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H11</th>\n",
       "      <td>0.678594</td>\n",
       "      <td>7.655887</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.901942</td>\n",
       "      <td>4.195848</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>54.077818</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.334834</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.304762</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.166371</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.694222</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H12</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.436049</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.004263</td>\n",
       "      <td>3.900086</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28.479105</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.731854</td>\n",
       "      <td>...</td>\n",
       "      <td>1.828602</td>\n",
       "      <td>5.298163</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.657204</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.248594</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.553393</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>369 rows × 38287 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              AT1G01010  AT1G01020  AT1G03987  AT1G01030  \\\n",
       "A2449446903_n01_LICRNA02_A01   0.382839  11.929183        0.0   3.731527   \n",
       "A2449446903_n01_LICRNA02_A02   0.000000   5.826682        0.0   1.501594   \n",
       "A2449446903_n01_LICRNA02_A03   1.021565   7.683520        0.0   0.301733   \n",
       "A2449446903_n01_LICRNA02_A05   0.732717   3.149144        0.0   1.947758   \n",
       "A2449446903_n01_LICRNA02_A06   1.614233   1.238895        0.0   6.130092   \n",
       "...                                 ...        ...        ...        ...   \n",
       "B250508004_n01_LICRNA03_H08    0.074367   4.474710        0.0   1.383813   \n",
       "B250508004_n01_LICRNA03_H09    2.232353   6.249002        0.0   1.873955   \n",
       "B250508004_n01_LICRNA03_H10    0.000000   0.000000        0.0   0.000000   \n",
       "B250508004_n01_LICRNA03_H11    0.678594   7.655887        0.0   0.901942   \n",
       "B250508004_n01_LICRNA03_H12    0.000000   3.436049        0.0   2.004263   \n",
       "\n",
       "                              AT1G01040  AT1G03993  AT1G01046  AT1G01050  \\\n",
       "A2449446903_n01_LICRNA02_A01   3.602189        0.0        0.0  31.063543   \n",
       "A2449446903_n01_LICRNA02_A02   3.439569        0.0        0.0  22.712417   \n",
       "A2449446903_n01_LICRNA02_A03   2.700528        0.0        0.0  42.431625   \n",
       "A2449446903_n01_LICRNA02_A05   3.742586        0.0        0.0  26.541337   \n",
       "A2449446903_n01_LICRNA02_A06   4.215597        0.0        0.0  19.379509   \n",
       "...                                 ...        ...        ...        ...   \n",
       "B250508004_n01_LICRNA03_H08    2.059206        0.0        0.0  27.907884   \n",
       "B250508004_n01_LICRNA03_H09    3.869254        0.0        0.0  40.686812   \n",
       "B250508004_n01_LICRNA03_H10    2.953706        0.0        0.0  36.084398   \n",
       "B250508004_n01_LICRNA03_H11    4.195848        0.0        0.0  54.077818   \n",
       "B250508004_n01_LICRNA03_H12    3.900086        0.0        0.0  28.479105   \n",
       "\n",
       "                              AT1G03997  AT1G01060  ...  ArthCt124  ArthCr088  \\\n",
       "A2449446903_n01_LICRNA02_A01        0.0   3.966935  ...  17.950907  44.209074   \n",
       "A2449446903_n01_LICRNA02_A02        0.0  18.186693  ...   1.986484   9.688606   \n",
       "A2449446903_n01_LICRNA02_A03        0.0  12.937632  ...   7.983340  16.384320   \n",
       "A2449446903_n01_LICRNA02_A05        0.0  23.861601  ...  17.178143  17.005325   \n",
       "A2449446903_n01_LICRNA02_A06        0.0   5.006570  ...  16.219201  43.599301   \n",
       "...                                 ...        ...  ...        ...        ...   \n",
       "B250508004_n01_LICRNA03_H08         0.0  11.118408  ...   3.486989   4.209645   \n",
       "B250508004_n01_LICRNA03_H09         0.0   7.391606  ...   9.640892   4.655562   \n",
       "B250508004_n01_LICRNA03_H10         0.0   0.000000  ...   0.000000   4.975538   \n",
       "B250508004_n01_LICRNA03_H11         0.0   0.334834  ...   0.000000   2.304762   \n",
       "B250508004_n01_LICRNA03_H12         0.0   1.731854  ...   1.828602   5.298163   \n",
       "\n",
       "                              ArthCt112  ArthCp088  ArthCp086  ArthCt118  \\\n",
       "A2449446903_n01_LICRNA02_A01        0.0   0.000000        0.0        0.0   \n",
       "A2449446903_n01_LICRNA02_A02        0.0   5.501032        0.0        0.0   \n",
       "A2449446903_n01_LICRNA02_A03        0.0   7.983340        0.0        0.0   \n",
       "A2449446903_n01_LICRNA02_A05        0.0   0.000000        0.0        0.0   \n",
       "A2449446903_n01_LICRNA02_A06        0.0   0.000000        0.0        0.0   \n",
       "...                                 ...        ...        ...        ...   \n",
       "B250508004_n01_LICRNA03_H08         0.0   1.072920        0.0        0.0   \n",
       "B250508004_n01_LICRNA03_H09         0.0   5.085306        0.0        0.0   \n",
       "B250508004_n01_LICRNA03_H10         0.0   0.000000        0.0        0.0   \n",
       "B250508004_n01_LICRNA03_H11         0.0   0.000000        0.0        0.0   \n",
       "B250508004_n01_LICRNA03_H12         0.0   3.657204        0.0        0.0   \n",
       "\n",
       "                              ArthCp083  ArthCt111  ArthCp084  ArthCp085  \n",
       "A2449446903_n01_LICRNA02_A01   0.000000        0.0   0.000000   6.266498  \n",
       "A2449446903_n01_LICRNA02_A02   0.519342        0.0   0.000000   3.467317  \n",
       "A2449446903_n01_LICRNA02_A03   0.166972        0.0   0.000000   0.000000  \n",
       "A2449446903_n01_LICRNA02_A05   0.628742        0.0   0.000000   0.000000  \n",
       "A2449446903_n01_LICRNA02_A06   1.074213        0.0   0.000000   0.000000  \n",
       "...                                 ...        ...        ...        ...  \n",
       "B250508004_n01_LICRNA03_H08    0.948096        0.0   0.000000   0.760798  \n",
       "B250508004_n01_LICRNA03_H09    0.360071        0.0   1.758217   4.807925  \n",
       "B250508004_n01_LICRNA03_H10    0.000000        0.0   0.000000   0.000000  \n",
       "B250508004_n01_LICRNA03_H11    0.166371        0.0   0.000000   0.694222  \n",
       "B250508004_n01_LICRNA03_H12    0.248594        0.0   0.000000   2.553393  \n",
       "\n",
       "[369 rows x 38287 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "trimmed_transcriptome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "f0dd6b4f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AT1G01010</th>\n",
       "      <th>AT1G01020</th>\n",
       "      <th>AT1G03987</th>\n",
       "      <th>AT1G01030</th>\n",
       "      <th>AT1G01040</th>\n",
       "      <th>AT1G03993</th>\n",
       "      <th>AT1G01046</th>\n",
       "      <th>AT1G01050</th>\n",
       "      <th>AT1G03997</th>\n",
       "      <th>AT1G01060</th>\n",
       "      <th>...</th>\n",
       "      <th>ArthCt124</th>\n",
       "      <th>ArthCr088</th>\n",
       "      <th>ArthCt112</th>\n",
       "      <th>ArthCp088</th>\n",
       "      <th>ArthCp086</th>\n",
       "      <th>ArthCt118</th>\n",
       "      <th>ArthCp083</th>\n",
       "      <th>ArthCt111</th>\n",
       "      <th>ArthCp084</th>\n",
       "      <th>ArthCp085</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A2450525897_n01_LICRNA01_H06</th>\n",
       "      <td>0</td>\n",
       "      <td>48</td>\n",
       "      <td>0</td>\n",
       "      <td>34</td>\n",
       "      <td>111</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>343</td>\n",
       "      <td>0</td>\n",
       "      <td>198</td>\n",
       "      <td>...</td>\n",
       "      <td>5</td>\n",
       "      <td>53</td>\n",
       "      <td>1</td>\n",
       "      <td>29</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2450525897_n01_LICRNA01_F06</th>\n",
       "      <td>12</td>\n",
       "      <td>87</td>\n",
       "      <td>0</td>\n",
       "      <td>55</td>\n",
       "      <td>150</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>331</td>\n",
       "      <td>0</td>\n",
       "      <td>212</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>82</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2450525897_n01_LICRNA01_E01</th>\n",
       "      <td>19</td>\n",
       "      <td>27</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>41</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>135</td>\n",
       "      <td>0</td>\n",
       "      <td>89</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>44</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2450525897_n01_LICRNA01_H10</th>\n",
       "      <td>6</td>\n",
       "      <td>24</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>97</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>140</td>\n",
       "      <td>0</td>\n",
       "      <td>75</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>32</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2450525897_n01_LICRNA01_F10</th>\n",
       "      <td>6</td>\n",
       "      <td>42</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>85</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>215</td>\n",
       "      <td>0</td>\n",
       "      <td>69</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B2449500127_n01_LICRNA04_E07</th>\n",
       "      <td>6</td>\n",
       "      <td>59</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>103</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>206</td>\n",
       "      <td>0</td>\n",
       "      <td>53</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B2449500127_n01_LICRNA04_F10</th>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>74</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>178</td>\n",
       "      <td>0</td>\n",
       "      <td>73</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B2449500127_n01_LICRNA04_C01</th>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>64</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>99</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B2449500127_n01_LICRNA04_D12</th>\n",
       "      <td>15</td>\n",
       "      <td>82</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>192</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>347</td>\n",
       "      <td>0</td>\n",
       "      <td>116</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>47</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>33</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B2449500127_n01_LICRNA04_A03</th>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>29</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>97</td>\n",
       "      <td>0</td>\n",
       "      <td>134</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>373 rows × 38295 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              AT1G01010  AT1G01020  AT1G03987  AT1G01030  \\\n",
       "A2450525897_n01_LICRNA01_H06          0         48          0         34   \n",
       "A2450525897_n01_LICRNA01_F06         12         87          0         55   \n",
       "A2450525897_n01_LICRNA01_E01         19         27          0         22   \n",
       "A2450525897_n01_LICRNA01_H10          6         24          0          5   \n",
       "A2450525897_n01_LICRNA01_F10          6         42          0          6   \n",
       "...                                 ...        ...        ...        ...   \n",
       "B2449500127_n01_LICRNA04_E07          6         59          0          7   \n",
       "B2449500127_n01_LICRNA04_F10          0         25          0          0   \n",
       "B2449500127_n01_LICRNA04_C01          0         20          0         15   \n",
       "B2449500127_n01_LICRNA04_D12         15         82          0         11   \n",
       "B2449500127_n01_LICRNA04_A03          2          5          0          0   \n",
       "\n",
       "                              AT1G01040  AT1G03993  AT1G01046  AT1G01050  \\\n",
       "A2450525897_n01_LICRNA01_H06        111          0          0        343   \n",
       "A2450525897_n01_LICRNA01_F06        150          0          0        331   \n",
       "A2450525897_n01_LICRNA01_E01         41          0          0        135   \n",
       "A2450525897_n01_LICRNA01_H10         97          0          0        140   \n",
       "A2450525897_n01_LICRNA01_F10         85          0          0        215   \n",
       "...                                 ...        ...        ...        ...   \n",
       "B2449500127_n01_LICRNA04_E07        103          0          0        206   \n",
       "B2449500127_n01_LICRNA04_F10         74          0          0        178   \n",
       "B2449500127_n01_LICRNA04_C01         64          0          0         99   \n",
       "B2449500127_n01_LICRNA04_D12        192          0          0        347   \n",
       "B2449500127_n01_LICRNA04_A03         29          0          0         97   \n",
       "\n",
       "                              AT1G03997  AT1G01060  ...  ArthCt124  ArthCr088  \\\n",
       "A2450525897_n01_LICRNA01_H06          0        198  ...          5         53   \n",
       "A2450525897_n01_LICRNA01_F06          0        212  ...          3         82   \n",
       "A2450525897_n01_LICRNA01_E01          0         89  ...          0         44   \n",
       "A2450525897_n01_LICRNA01_H10          0         75  ...          0         32   \n",
       "A2450525897_n01_LICRNA01_F10          0         69  ...          0         15   \n",
       "...                                 ...        ...  ...        ...        ...   \n",
       "B2449500127_n01_LICRNA04_E07          0         53  ...          0         22   \n",
       "B2449500127_n01_LICRNA04_F10          0         73  ...          0         13   \n",
       "B2449500127_n01_LICRNA04_C01          0          4  ...          0          7   \n",
       "B2449500127_n01_LICRNA04_D12          0        116  ...          1         47   \n",
       "B2449500127_n01_LICRNA04_A03          0        134  ...          0         10   \n",
       "\n",
       "                              ArthCt112  ArthCp088  ArthCp086  ArthCt118  \\\n",
       "A2450525897_n01_LICRNA01_H06          1         29          0          0   \n",
       "A2450525897_n01_LICRNA01_F06          5         18          0          0   \n",
       "A2450525897_n01_LICRNA01_E01          0          0          0          0   \n",
       "A2450525897_n01_LICRNA01_H10          0          2          0          0   \n",
       "A2450525897_n01_LICRNA01_F10          0          9          0          0   \n",
       "...                                 ...        ...        ...        ...   \n",
       "B2449500127_n01_LICRNA04_E07          0         14          0          0   \n",
       "B2449500127_n01_LICRNA04_F10          0          3          0          0   \n",
       "B2449500127_n01_LICRNA04_C01          0          5          0          0   \n",
       "B2449500127_n01_LICRNA04_D12          0          4          0          0   \n",
       "B2449500127_n01_LICRNA04_A03          0          0          0          0   \n",
       "\n",
       "                              ArthCp083  ArthCt111  ArthCp084  ArthCp085  \n",
       "A2450525897_n01_LICRNA01_H06         30          0          0          6  \n",
       "A2450525897_n01_LICRNA01_F06          3          0          0         12  \n",
       "A2450525897_n01_LICRNA01_E01          3          0          0          5  \n",
       "A2450525897_n01_LICRNA01_H10          8          0          0          0  \n",
       "A2450525897_n01_LICRNA01_F10          4          0          0          1  \n",
       "...                                 ...        ...        ...        ...  \n",
       "B2449500127_n01_LICRNA04_E07          1          0          0          3  \n",
       "B2449500127_n01_LICRNA04_F10          0          0          0          0  \n",
       "B2449500127_n01_LICRNA04_C01          1          0          0         20  \n",
       "B2449500127_n01_LICRNA04_D12         33          0          0         25  \n",
       "B2449500127_n01_LICRNA04_A03          1          0          0          1  \n",
       "\n",
       "[373 rows x 38295 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "trimmed_raw_transcriptome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "bc1d6d88",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>plate.pos</th>\n",
       "      <th>sampID</th>\n",
       "      <th>sample.type</th>\n",
       "      <th>timepoint</th>\n",
       "      <th>date</th>\n",
       "      <th>time</th>\n",
       "      <th>extractplate</th>\n",
       "      <th>rnaprepplate</th>\n",
       "      <th>Unmapped</th>\n",
       "      <th>No Feature</th>\n",
       "      <th>plate_row</th>\n",
       "      <th>plate_col</th>\n",
       "      <th>Total Reads</th>\n",
       "      <th>Experiment Type</th>\n",
       "      <th>Sample Month</th>\n",
       "      <th>daysincestart</th>\n",
       "      <th>Date and Time</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>filename</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A01</th>\n",
       "      <td>A01</td>\n",
       "      <td>LIC157</td>\n",
       "      <td>plant</td>\n",
       "      <td>t14</td>\n",
       "      <td>1/23/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_02</td>\n",
       "      <td>LICRNA_02</td>\n",
       "      <td>758284.0</td>\n",
       "      <td>82176.0</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "      <td>1246970.0</td>\n",
       "      <td>Long Term</td>\n",
       "      <td>January</td>\n",
       "      <td>83</td>\n",
       "      <td>2024-01-23 08:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A02</th>\n",
       "      <td>A02</td>\n",
       "      <td>LIC169</td>\n",
       "      <td>plant</td>\n",
       "      <td>t15</td>\n",
       "      <td>1/27/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_02</td>\n",
       "      <td>LICRNA_02</td>\n",
       "      <td>3071877.0</td>\n",
       "      <td>353549.0</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>4985602.0</td>\n",
       "      <td>Long Term</td>\n",
       "      <td>January</td>\n",
       "      <td>87</td>\n",
       "      <td>2024-01-27 08:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A03</th>\n",
       "      <td>A03</td>\n",
       "      <td>LIC193</td>\n",
       "      <td>plant</td>\n",
       "      <td>t17</td>\n",
       "      <td>2/2/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_03</td>\n",
       "      <td>LICRNA_02</td>\n",
       "      <td>1213638.0</td>\n",
       "      <td>158864.0</td>\n",
       "      <td>A</td>\n",
       "      <td>3</td>\n",
       "      <td>2465662.0</td>\n",
       "      <td>Long Term</td>\n",
       "      <td>February</td>\n",
       "      <td>93</td>\n",
       "      <td>2024-02-02 08:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A05</th>\n",
       "      <td>A05</td>\n",
       "      <td>LIC217</td>\n",
       "      <td>plant</td>\n",
       "      <td>t19</td>\n",
       "      <td>2/9/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_03</td>\n",
       "      <td>LICRNA_02</td>\n",
       "      <td>1303436.0</td>\n",
       "      <td>90929.0</td>\n",
       "      <td>A</td>\n",
       "      <td>5</td>\n",
       "      <td>1515395.0</td>\n",
       "      <td>Long Term</td>\n",
       "      <td>February</td>\n",
       "      <td>100</td>\n",
       "      <td>2024-02-09 08:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A06</th>\n",
       "      <td>A06</td>\n",
       "      <td>LIC229</td>\n",
       "      <td>plant</td>\n",
       "      <td>t20</td>\n",
       "      <td>2/16/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_03</td>\n",
       "      <td>LICRNA_02</td>\n",
       "      <td>1591231.0</td>\n",
       "      <td>141665.0</td>\n",
       "      <td>A</td>\n",
       "      <td>6</td>\n",
       "      <td>1924391.0</td>\n",
       "      <td>Long Term</td>\n",
       "      <td>February</td>\n",
       "      <td>107</td>\n",
       "      <td>2024-02-16 08:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H08</th>\n",
       "      <td>H08</td>\n",
       "      <td>LIC428</td>\n",
       "      <td>plant</td>\n",
       "      <td>c1_t04</td>\n",
       "      <td>2/10/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LICCIRC_01</td>\n",
       "      <td>LICRNA_03</td>\n",
       "      <td>2552177.0</td>\n",
       "      <td>438552.0</td>\n",
       "      <td>H</td>\n",
       "      <td>8</td>\n",
       "      <td>5245242.0</td>\n",
       "      <td>Circadian Experiment 1</td>\n",
       "      <td>February</td>\n",
       "      <td>101</td>\n",
       "      <td>2024-02-10 08:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H09</th>\n",
       "      <td>H09</td>\n",
       "      <td>LIC440</td>\n",
       "      <td>plant</td>\n",
       "      <td>c1_t05</td>\n",
       "      <td>2/10/24</td>\n",
       "      <td>12:00</td>\n",
       "      <td>LICCIRC_01</td>\n",
       "      <td>LICRNA_03</td>\n",
       "      <td>4228242.0</td>\n",
       "      <td>493699.0</td>\n",
       "      <td>H</td>\n",
       "      <td>9</td>\n",
       "      <td>7715482.0</td>\n",
       "      <td>Circadian Experiment 1</td>\n",
       "      <td>February</td>\n",
       "      <td>101</td>\n",
       "      <td>2024-02-10 12:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H10</th>\n",
       "      <td>H10</td>\n",
       "      <td>LIC452</td>\n",
       "      <td>plant</td>\n",
       "      <td>c1_t06</td>\n",
       "      <td>2/10/24</td>\n",
       "      <td>16:00</td>\n",
       "      <td>LICCIRC_01</td>\n",
       "      <td>LICRNA_03</td>\n",
       "      <td>365095.0</td>\n",
       "      <td>39090.0</td>\n",
       "      <td>H</td>\n",
       "      <td>10</td>\n",
       "      <td>643786.0</td>\n",
       "      <td>Circadian Experiment 1</td>\n",
       "      <td>February</td>\n",
       "      <td>101</td>\n",
       "      <td>2024-02-10 16:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H11</th>\n",
       "      <td>H11</td>\n",
       "      <td>LIC464</td>\n",
       "      <td>plant</td>\n",
       "      <td>c1_t07</td>\n",
       "      <td>2/10/24</td>\n",
       "      <td>20:00</td>\n",
       "      <td>LICCIRC_01</td>\n",
       "      <td>LICRNA_03</td>\n",
       "      <td>326959.0</td>\n",
       "      <td>60346.0</td>\n",
       "      <td>H</td>\n",
       "      <td>11</td>\n",
       "      <td>1509679.0</td>\n",
       "      <td>Circadian Experiment 1</td>\n",
       "      <td>February</td>\n",
       "      <td>101</td>\n",
       "      <td>2024-02-10 20:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H12</th>\n",
       "      <td>H12</td>\n",
       "      <td>LIC488</td>\n",
       "      <td>plant</td>\n",
       "      <td>c1_t08</td>\n",
       "      <td>2/11/24</td>\n",
       "      <td>0:00</td>\n",
       "      <td>LICCIRC_02</td>\n",
       "      <td>LICRNA_03</td>\n",
       "      <td>2282102.0</td>\n",
       "      <td>402777.0</td>\n",
       "      <td>H</td>\n",
       "      <td>12</td>\n",
       "      <td>5444101.0</td>\n",
       "      <td>Circadian Experiment 1</td>\n",
       "      <td>February</td>\n",
       "      <td>102</td>\n",
       "      <td>2024-02-11 00:00:00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>369 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             plate.pos  sampID sample.type timepoint     date  \\\n",
       "filename                                                                        \n",
       "A2449446903_n01_LICRNA02_A01       A01  LIC157       plant       t14  1/23/24   \n",
       "A2449446903_n01_LICRNA02_A02       A02  LIC169       plant       t15  1/27/24   \n",
       "A2449446903_n01_LICRNA02_A03       A03  LIC193       plant       t17   2/2/24   \n",
       "A2449446903_n01_LICRNA02_A05       A05  LIC217       plant       t19   2/9/24   \n",
       "A2449446903_n01_LICRNA02_A06       A06  LIC229       plant       t20  2/16/24   \n",
       "...                                ...     ...         ...       ...      ...   \n",
       "B250508004_n01_LICRNA03_H08        H08  LIC428       plant    c1_t04  2/10/24   \n",
       "B250508004_n01_LICRNA03_H09        H09  LIC440       plant    c1_t05  2/10/24   \n",
       "B250508004_n01_LICRNA03_H10        H10  LIC452       plant    c1_t06  2/10/24   \n",
       "B250508004_n01_LICRNA03_H11        H11  LIC464       plant    c1_t07  2/10/24   \n",
       "B250508004_n01_LICRNA03_H12        H12  LIC488       plant    c1_t08  2/11/24   \n",
       "\n",
       "                               time extractplate rnaprepplate   Unmapped  \\\n",
       "filename                                                                   \n",
       "A2449446903_n01_LICRNA02_A01   8:00       LIC_02    LICRNA_02   758284.0   \n",
       "A2449446903_n01_LICRNA02_A02   8:00       LIC_02    LICRNA_02  3071877.0   \n",
       "A2449446903_n01_LICRNA02_A03   8:00       LIC_03    LICRNA_02  1213638.0   \n",
       "A2449446903_n01_LICRNA02_A05   8:00       LIC_03    LICRNA_02  1303436.0   \n",
       "A2449446903_n01_LICRNA02_A06   8:00       LIC_03    LICRNA_02  1591231.0   \n",
       "...                             ...          ...          ...        ...   \n",
       "B250508004_n01_LICRNA03_H08    8:00   LICCIRC_01    LICRNA_03  2552177.0   \n",
       "B250508004_n01_LICRNA03_H09   12:00   LICCIRC_01    LICRNA_03  4228242.0   \n",
       "B250508004_n01_LICRNA03_H10   16:00   LICCIRC_01    LICRNA_03   365095.0   \n",
       "B250508004_n01_LICRNA03_H11   20:00   LICCIRC_01    LICRNA_03   326959.0   \n",
       "B250508004_n01_LICRNA03_H12    0:00   LICCIRC_02    LICRNA_03  2282102.0   \n",
       "\n",
       "                              No Feature plate_row  plate_col  Total Reads  \\\n",
       "filename                                                                     \n",
       "A2449446903_n01_LICRNA02_A01     82176.0         A          1    1246970.0   \n",
       "A2449446903_n01_LICRNA02_A02    353549.0         A          2    4985602.0   \n",
       "A2449446903_n01_LICRNA02_A03    158864.0         A          3    2465662.0   \n",
       "A2449446903_n01_LICRNA02_A05     90929.0         A          5    1515395.0   \n",
       "A2449446903_n01_LICRNA02_A06    141665.0         A          6    1924391.0   \n",
       "...                                  ...       ...        ...          ...   \n",
       "B250508004_n01_LICRNA03_H08     438552.0         H          8    5245242.0   \n",
       "B250508004_n01_LICRNA03_H09     493699.0         H          9    7715482.0   \n",
       "B250508004_n01_LICRNA03_H10      39090.0         H         10     643786.0   \n",
       "B250508004_n01_LICRNA03_H11      60346.0         H         11    1509679.0   \n",
       "B250508004_n01_LICRNA03_H12     402777.0         H         12    5444101.0   \n",
       "\n",
       "                                     Experiment Type Sample Month  \\\n",
       "filename                                                            \n",
       "A2449446903_n01_LICRNA02_A01               Long Term      January   \n",
       "A2449446903_n01_LICRNA02_A02               Long Term      January   \n",
       "A2449446903_n01_LICRNA02_A03               Long Term     February   \n",
       "A2449446903_n01_LICRNA02_A05               Long Term     February   \n",
       "A2449446903_n01_LICRNA02_A06               Long Term     February   \n",
       "...                                              ...          ...   \n",
       "B250508004_n01_LICRNA03_H08   Circadian Experiment 1     February   \n",
       "B250508004_n01_LICRNA03_H09   Circadian Experiment 1     February   \n",
       "B250508004_n01_LICRNA03_H10   Circadian Experiment 1     February   \n",
       "B250508004_n01_LICRNA03_H11   Circadian Experiment 1     February   \n",
       "B250508004_n01_LICRNA03_H12   Circadian Experiment 1     February   \n",
       "\n",
       "                              daysincestart       Date and Time  \n",
       "filename                                                         \n",
       "A2449446903_n01_LICRNA02_A01             83 2024-01-23 08:00:00  \n",
       "A2449446903_n01_LICRNA02_A02             87 2024-01-27 08:00:00  \n",
       "A2449446903_n01_LICRNA02_A03             93 2024-02-02 08:00:00  \n",
       "A2449446903_n01_LICRNA02_A05            100 2024-02-09 08:00:00  \n",
       "A2449446903_n01_LICRNA02_A06            107 2024-02-16 08:00:00  \n",
       "...                                     ...                 ...  \n",
       "B250508004_n01_LICRNA03_H08             101 2024-02-10 08:00:00  \n",
       "B250508004_n01_LICRNA03_H09             101 2024-02-10 12:00:00  \n",
       "B250508004_n01_LICRNA03_H10             101 2024-02-10 16:00:00  \n",
       "B250508004_n01_LICRNA03_H11             101 2024-02-10 20:00:00  \n",
       "B250508004_n01_LICRNA03_H12             102 2024-02-11 00:00:00  \n",
       "\n",
       "[369 rows x 17 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "trimmed_metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "ac36d791",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample</th>\n",
       "      <th>sample.type</th>\n",
       "      <th>timepoint</th>\n",
       "      <th>date</th>\n",
       "      <th>time</th>\n",
       "      <th>platename</th>\n",
       "      <th>daysincestart</th>\n",
       "      <th>plantID</th>\n",
       "      <th>Kingdom</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "      <th>AbundR100</th>\n",
       "      <th>sampID</th>\n",
       "      <th>Experiment Type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LIC522</td>\n",
       "      <td>plant</td>\n",
       "      <td>c1_t11</td>\n",
       "      <td>2/11/24</td>\n",
       "      <td>12:00</td>\n",
       "      <td>LICCIRC_02</td>\n",
       "      <td>102</td>\n",
       "      <td>LIC522</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Pseudomonadales_A_650611</td>\n",
       "      <td>Pseudomonadaceae</td>\n",
       "      <td>Pseudomonas_E_647464</td>\n",
       "      <td>Pseudomonas_E_647464 viridiflava</td>\n",
       "      <td>79.1</td>\n",
       "      <td>LIC522</td>\n",
       "      <td>Circadian Experiment 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>LIC063</td>\n",
       "      <td>plant</td>\n",
       "      <td>t06</td>\n",
       "      <td>11/28/23</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_01</td>\n",
       "      <td>27</td>\n",
       "      <td>LIC063</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Sphingomonadales</td>\n",
       "      <td>Sphingomonadaceae_486827</td>\n",
       "      <td>Sphingomonas_L_486704</td>\n",
       "      <td>Sphingomonas_L_486704 sp000786205</td>\n",
       "      <td>66.1</td>\n",
       "      <td>LIC063</td>\n",
       "      <td>Long Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LIC159</td>\n",
       "      <td>plant</td>\n",
       "      <td>t14</td>\n",
       "      <td>1/23/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_02</td>\n",
       "      <td>83</td>\n",
       "      <td>LIC159</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>Cyanobacteriota</td>\n",
       "      <td>Cyanobacteriia</td>\n",
       "      <td>Cyanobacteriales</td>\n",
       "      <td>Coleofasciculaceae</td>\n",
       "      <td>Caldora</td>\n",
       "      <td>Caldora sp010672925</td>\n",
       "      <td>60.5</td>\n",
       "      <td>LIC159</td>\n",
       "      <td>Long Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LIC452</td>\n",
       "      <td>plant</td>\n",
       "      <td>c1_t06</td>\n",
       "      <td>2/10/24</td>\n",
       "      <td>16:00</td>\n",
       "      <td>LICCIRC_01</td>\n",
       "      <td>101</td>\n",
       "      <td>LIC452</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Burkholderiales</td>\n",
       "      <td>Burkholderiaceae_A_595421</td>\n",
       "      <td>Massilia_574544</td>\n",
       "      <td>Massilia atriviolacea</td>\n",
       "      <td>59.8</td>\n",
       "      <td>LIC452</td>\n",
       "      <td>Circadian Experiment 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LIC306</td>\n",
       "      <td>plant</td>\n",
       "      <td>t26</td>\n",
       "      <td>3/8/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_04</td>\n",
       "      <td>128</td>\n",
       "      <td>LIC306</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Sphingomonadales</td>\n",
       "      <td>Sphingomonadaceae_486827</td>\n",
       "      <td>Sphingomonas_L_486704</td>\n",
       "      <td>Sphingomonas_L_486704 sp000786205</td>\n",
       "      <td>57.2</td>\n",
       "      <td>LIC306</td>\n",
       "      <td>Long Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12281</th>\n",
       "      <td>LIC122</td>\n",
       "      <td>plant</td>\n",
       "      <td>t11</td>\n",
       "      <td>12/15/23</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_02</td>\n",
       "      <td>44</td>\n",
       "      <td>LIC122</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>Actinomycetota</td>\n",
       "      <td>Actinomycetes</td>\n",
       "      <td>Propionibacteriales</td>\n",
       "      <td>Nocardioidaceae</td>\n",
       "      <td>Nocardioides_A_392796</td>\n",
       "      <td>Nocardioides_A_392796 hankookensis</td>\n",
       "      <td>0.5</td>\n",
       "      <td>LIC122</td>\n",
       "      <td>Long Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12282</th>\n",
       "      <td>LIC028</td>\n",
       "      <td>plant</td>\n",
       "      <td>t03</td>\n",
       "      <td>11/14/23</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_01</td>\n",
       "      <td>13</td>\n",
       "      <td>LIC028</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>Actinomycetota</td>\n",
       "      <td>Actinomycetes</td>\n",
       "      <td>Mycobacteriales</td>\n",
       "      <td>Geodermatophilaceae</td>\n",
       "      <td>Modestobacter</td>\n",
       "      <td>Modestobacter muralis</td>\n",
       "      <td>0.5</td>\n",
       "      <td>LIC028</td>\n",
       "      <td>Long Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12283</th>\n",
       "      <td>LIC631</td>\n",
       "      <td>plant</td>\n",
       "      <td>c2_t05</td>\n",
       "      <td>3/2/24</td>\n",
       "      <td>12:00</td>\n",
       "      <td>LICCIRC_03</td>\n",
       "      <td>122</td>\n",
       "      <td>LIC631</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Caulobacterales</td>\n",
       "      <td>Caulobacteraceae</td>\n",
       "      <td>Brevundimonas</td>\n",
       "      <td>Brevundimonas subvibrioides</td>\n",
       "      <td>0.5</td>\n",
       "      <td>LIC631</td>\n",
       "      <td>Circadian Experiment 2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12284</th>\n",
       "      <td>LIC629</td>\n",
       "      <td>plant</td>\n",
       "      <td>c2_t05</td>\n",
       "      <td>3/2/24</td>\n",
       "      <td>12:00</td>\n",
       "      <td>LICCIRC_03</td>\n",
       "      <td>122</td>\n",
       "      <td>LIC629</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Rhizobiales_505101</td>\n",
       "      <td>Rhizobiaceae</td>\n",
       "      <td>Agrobacterium</td>\n",
       "      <td>Agrobacterium larrymoorei</td>\n",
       "      <td>0.5</td>\n",
       "      <td>LIC629</td>\n",
       "      <td>Circadian Experiment 2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12285</th>\n",
       "      <td>LIC230</td>\n",
       "      <td>plant</td>\n",
       "      <td>t20</td>\n",
       "      <td>2/16/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_03</td>\n",
       "      <td>107</td>\n",
       "      <td>LIC230</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Rhizobiales_505101</td>\n",
       "      <td>Beijerinckiaceae</td>\n",
       "      <td>Methylobacterium</td>\n",
       "      <td>Methylobacterium sp001424705</td>\n",
       "      <td>0.5</td>\n",
       "      <td>LIC230</td>\n",
       "      <td>Long Term</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12286 rows × 18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Sample sample.type timepoint      date   time   platename  \\\n",
       "0      LIC522       plant    c1_t11   2/11/24  12:00  LICCIRC_02   \n",
       "1      LIC063       plant       t06  11/28/23   8:00      LIC_01   \n",
       "2      LIC159       plant       t14   1/23/24   8:00      LIC_02   \n",
       "3      LIC452       plant    c1_t06   2/10/24  16:00  LICCIRC_01   \n",
       "4      LIC306       plant       t26    3/8/24   8:00      LIC_04   \n",
       "...       ...         ...       ...       ...    ...         ...   \n",
       "12281  LIC122       plant       t11  12/15/23   8:00      LIC_02   \n",
       "12282  LIC028       plant       t03  11/14/23   8:00      LIC_01   \n",
       "12283  LIC631       plant    c2_t05    3/2/24  12:00  LICCIRC_03   \n",
       "12284  LIC629       plant    c2_t05    3/2/24  12:00  LICCIRC_03   \n",
       "12285  LIC230       plant       t20   2/16/24   8:00      LIC_03   \n",
       "\n",
       "       daysincestart plantID      Kingdom           Phylum  \\\n",
       "0                102  LIC522  d__Bacteria   Pseudomonadota   \n",
       "1                 27  LIC063  d__Bacteria   Pseudomonadota   \n",
       "2                 83  LIC159  d__Bacteria  Cyanobacteriota   \n",
       "3                101  LIC452  d__Bacteria   Pseudomonadota   \n",
       "4                128  LIC306  d__Bacteria   Pseudomonadota   \n",
       "...              ...     ...          ...              ...   \n",
       "12281             44  LIC122  d__Bacteria   Actinomycetota   \n",
       "12282             13  LIC028  d__Bacteria   Actinomycetota   \n",
       "12283            122  LIC631  d__Bacteria   Pseudomonadota   \n",
       "12284            122  LIC629  d__Bacteria   Pseudomonadota   \n",
       "12285            107  LIC230  d__Bacteria   Pseudomonadota   \n",
       "\n",
       "                     Class                     Order  \\\n",
       "0      Gammaproteobacteria  Pseudomonadales_A_650611   \n",
       "1      Alphaproteobacteria          Sphingomonadales   \n",
       "2           Cyanobacteriia          Cyanobacteriales   \n",
       "3      Gammaproteobacteria           Burkholderiales   \n",
       "4      Alphaproteobacteria          Sphingomonadales   \n",
       "...                    ...                       ...   \n",
       "12281        Actinomycetes       Propionibacteriales   \n",
       "12282        Actinomycetes           Mycobacteriales   \n",
       "12283  Alphaproteobacteria           Caulobacterales   \n",
       "12284  Alphaproteobacteria        Rhizobiales_505101   \n",
       "12285  Alphaproteobacteria        Rhizobiales_505101   \n",
       "\n",
       "                          Family                  Genus  \\\n",
       "0               Pseudomonadaceae   Pseudomonas_E_647464   \n",
       "1       Sphingomonadaceae_486827  Sphingomonas_L_486704   \n",
       "2             Coleofasciculaceae                Caldora   \n",
       "3      Burkholderiaceae_A_595421        Massilia_574544   \n",
       "4       Sphingomonadaceae_486827  Sphingomonas_L_486704   \n",
       "...                          ...                    ...   \n",
       "12281            Nocardioidaceae  Nocardioides_A_392796   \n",
       "12282        Geodermatophilaceae          Modestobacter   \n",
       "12283           Caulobacteraceae          Brevundimonas   \n",
       "12284               Rhizobiaceae          Agrobacterium   \n",
       "12285           Beijerinckiaceae       Methylobacterium   \n",
       "\n",
       "                                  Species  AbundR100  sampID  \\\n",
       "0        Pseudomonas_E_647464 viridiflava       79.1  LIC522   \n",
       "1       Sphingomonas_L_486704 sp000786205       66.1  LIC063   \n",
       "2                     Caldora sp010672925       60.5  LIC159   \n",
       "3                   Massilia atriviolacea       59.8  LIC452   \n",
       "4       Sphingomonas_L_486704 sp000786205       57.2  LIC306   \n",
       "...                                   ...        ...     ...   \n",
       "12281  Nocardioides_A_392796 hankookensis        0.5  LIC122   \n",
       "12282               Modestobacter muralis        0.5  LIC028   \n",
       "12283         Brevundimonas subvibrioides        0.5  LIC631   \n",
       "12284           Agrobacterium larrymoorei        0.5  LIC629   \n",
       "12285        Methylobacterium sp001424705        0.5  LIC230   \n",
       "\n",
       "              Experiment Type  \n",
       "0      Circadian Experiment 1  \n",
       "1                   Long Term  \n",
       "2                   Long Term  \n",
       "3      Circadian Experiment 1  \n",
       "4                   Long Term  \n",
       "...                       ...  \n",
       "12281               Long Term  \n",
       "12282               Long Term  \n",
       "12283  Circadian Experiment 2  \n",
       "12284  Circadian Experiment 2  \n",
       "12285               Long Term  \n",
       "\n",
       "[12286 rows x 18 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "microbiome_abundance = pd.read_csv(\n",
    "    \"/Users/michael/Data/Luke_terrace_experiment/Microbiome/lic2024_16S_rab.csv\"\n",
    ")\n",
    "long_term_microbiome = microbiome_abundance.merge(\n",
    "    metadata[[\"sampID\", \"Experiment Type\"]], left_on=\"plantID\", right_on=\"sampID\"\n",
    ")\n",
    "long_term_microbiome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "33420220",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AT1G01010</th>\n",
       "      <th>AT1G01020</th>\n",
       "      <th>AT1G03987</th>\n",
       "      <th>AT1G01030</th>\n",
       "      <th>AT1G01040</th>\n",
       "      <th>AT1G03993</th>\n",
       "      <th>AT1G01046</th>\n",
       "      <th>AT1G01050</th>\n",
       "      <th>AT1G03997</th>\n",
       "      <th>AT1G01060</th>\n",
       "      <th>...</th>\n",
       "      <th>ArthCt124</th>\n",
       "      <th>ArthCr088</th>\n",
       "      <th>ArthCt112</th>\n",
       "      <th>ArthCp088</th>\n",
       "      <th>ArthCp086</th>\n",
       "      <th>ArthCt118</th>\n",
       "      <th>ArthCp083</th>\n",
       "      <th>ArthCt111</th>\n",
       "      <th>ArthCp084</th>\n",
       "      <th>ArthCp085</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A01</th>\n",
       "      <td>0.382839</td>\n",
       "      <td>11.929183</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.731527</td>\n",
       "      <td>3.602189</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>31.063543</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.966935</td>\n",
       "      <td>...</td>\n",
       "      <td>17.950907</td>\n",
       "      <td>44.209074</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>6.266498</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A02</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>5.826682</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.501594</td>\n",
       "      <td>3.439569</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>22.712417</td>\n",
       "      <td>0.0</td>\n",
       "      <td>18.186693</td>\n",
       "      <td>...</td>\n",
       "      <td>1.986484</td>\n",
       "      <td>9.688606</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.501032</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.519342</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.467317</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A03</th>\n",
       "      <td>1.021565</td>\n",
       "      <td>7.683520</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.301733</td>\n",
       "      <td>2.700528</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>42.431625</td>\n",
       "      <td>0.0</td>\n",
       "      <td>12.937632</td>\n",
       "      <td>...</td>\n",
       "      <td>7.983340</td>\n",
       "      <td>16.384320</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.983340</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.166972</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A05</th>\n",
       "      <td>0.732717</td>\n",
       "      <td>3.149144</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.947758</td>\n",
       "      <td>3.742586</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>26.541337</td>\n",
       "      <td>0.0</td>\n",
       "      <td>23.861601</td>\n",
       "      <td>...</td>\n",
       "      <td>17.178143</td>\n",
       "      <td>17.005325</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.628742</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A06</th>\n",
       "      <td>1.614233</td>\n",
       "      <td>1.238895</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.130092</td>\n",
       "      <td>4.215597</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>19.379509</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.006570</td>\n",
       "      <td>...</td>\n",
       "      <td>16.219201</td>\n",
       "      <td>43.599301</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.074213</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H08</th>\n",
       "      <td>0.074367</td>\n",
       "      <td>4.474710</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.383813</td>\n",
       "      <td>2.059206</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27.907884</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.118408</td>\n",
       "      <td>...</td>\n",
       "      <td>3.486989</td>\n",
       "      <td>4.209645</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.072920</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.948096</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.760798</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H09</th>\n",
       "      <td>2.232353</td>\n",
       "      <td>6.249002</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.873955</td>\n",
       "      <td>3.869254</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>40.686812</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.391606</td>\n",
       "      <td>...</td>\n",
       "      <td>9.640892</td>\n",
       "      <td>4.655562</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.085306</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.360071</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.758217</td>\n",
       "      <td>4.807925</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H10</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.953706</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>36.084398</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.975538</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H11</th>\n",
       "      <td>0.678594</td>\n",
       "      <td>7.655887</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.901942</td>\n",
       "      <td>4.195848</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>54.077818</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.334834</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.304762</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.166371</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.694222</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H12</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.436049</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.004263</td>\n",
       "      <td>3.900086</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28.479105</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.731854</td>\n",
       "      <td>...</td>\n",
       "      <td>1.828602</td>\n",
       "      <td>5.298163</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.657204</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.248594</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.553393</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>369 rows × 38287 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              AT1G01010  AT1G01020  AT1G03987  AT1G01030  \\\n",
       "A2449446903_n01_LICRNA02_A01   0.382839  11.929183        0.0   3.731527   \n",
       "A2449446903_n01_LICRNA02_A02   0.000000   5.826682        0.0   1.501594   \n",
       "A2449446903_n01_LICRNA02_A03   1.021565   7.683520        0.0   0.301733   \n",
       "A2449446903_n01_LICRNA02_A05   0.732717   3.149144        0.0   1.947758   \n",
       "A2449446903_n01_LICRNA02_A06   1.614233   1.238895        0.0   6.130092   \n",
       "...                                 ...        ...        ...        ...   \n",
       "B250508004_n01_LICRNA03_H08    0.074367   4.474710        0.0   1.383813   \n",
       "B250508004_n01_LICRNA03_H09    2.232353   6.249002        0.0   1.873955   \n",
       "B250508004_n01_LICRNA03_H10    0.000000   0.000000        0.0   0.000000   \n",
       "B250508004_n01_LICRNA03_H11    0.678594   7.655887        0.0   0.901942   \n",
       "B250508004_n01_LICRNA03_H12    0.000000   3.436049        0.0   2.004263   \n",
       "\n",
       "                              AT1G01040  AT1G03993  AT1G01046  AT1G01050  \\\n",
       "A2449446903_n01_LICRNA02_A01   3.602189        0.0        0.0  31.063543   \n",
       "A2449446903_n01_LICRNA02_A02   3.439569        0.0        0.0  22.712417   \n",
       "A2449446903_n01_LICRNA02_A03   2.700528        0.0        0.0  42.431625   \n",
       "A2449446903_n01_LICRNA02_A05   3.742586        0.0        0.0  26.541337   \n",
       "A2449446903_n01_LICRNA02_A06   4.215597        0.0        0.0  19.379509   \n",
       "...                                 ...        ...        ...        ...   \n",
       "B250508004_n01_LICRNA03_H08    2.059206        0.0        0.0  27.907884   \n",
       "B250508004_n01_LICRNA03_H09    3.869254        0.0        0.0  40.686812   \n",
       "B250508004_n01_LICRNA03_H10    2.953706        0.0        0.0  36.084398   \n",
       "B250508004_n01_LICRNA03_H11    4.195848        0.0        0.0  54.077818   \n",
       "B250508004_n01_LICRNA03_H12    3.900086        0.0        0.0  28.479105   \n",
       "\n",
       "                              AT1G03997  AT1G01060  ...  ArthCt124  ArthCr088  \\\n",
       "A2449446903_n01_LICRNA02_A01        0.0   3.966935  ...  17.950907  44.209074   \n",
       "A2449446903_n01_LICRNA02_A02        0.0  18.186693  ...   1.986484   9.688606   \n",
       "A2449446903_n01_LICRNA02_A03        0.0  12.937632  ...   7.983340  16.384320   \n",
       "A2449446903_n01_LICRNA02_A05        0.0  23.861601  ...  17.178143  17.005325   \n",
       "A2449446903_n01_LICRNA02_A06        0.0   5.006570  ...  16.219201  43.599301   \n",
       "...                                 ...        ...  ...        ...        ...   \n",
       "B250508004_n01_LICRNA03_H08         0.0  11.118408  ...   3.486989   4.209645   \n",
       "B250508004_n01_LICRNA03_H09         0.0   7.391606  ...   9.640892   4.655562   \n",
       "B250508004_n01_LICRNA03_H10         0.0   0.000000  ...   0.000000   4.975538   \n",
       "B250508004_n01_LICRNA03_H11         0.0   0.334834  ...   0.000000   2.304762   \n",
       "B250508004_n01_LICRNA03_H12         0.0   1.731854  ...   1.828602   5.298163   \n",
       "\n",
       "                              ArthCt112  ArthCp088  ArthCp086  ArthCt118  \\\n",
       "A2449446903_n01_LICRNA02_A01        0.0   0.000000        0.0        0.0   \n",
       "A2449446903_n01_LICRNA02_A02        0.0   5.501032        0.0        0.0   \n",
       "A2449446903_n01_LICRNA02_A03        0.0   7.983340        0.0        0.0   \n",
       "A2449446903_n01_LICRNA02_A05        0.0   0.000000        0.0        0.0   \n",
       "A2449446903_n01_LICRNA02_A06        0.0   0.000000        0.0        0.0   \n",
       "...                                 ...        ...        ...        ...   \n",
       "B250508004_n01_LICRNA03_H08         0.0   1.072920        0.0        0.0   \n",
       "B250508004_n01_LICRNA03_H09         0.0   5.085306        0.0        0.0   \n",
       "B250508004_n01_LICRNA03_H10         0.0   0.000000        0.0        0.0   \n",
       "B250508004_n01_LICRNA03_H11         0.0   0.000000        0.0        0.0   \n",
       "B250508004_n01_LICRNA03_H12         0.0   3.657204        0.0        0.0   \n",
       "\n",
       "                              ArthCp083  ArthCt111  ArthCp084  ArthCp085  \n",
       "A2449446903_n01_LICRNA02_A01   0.000000        0.0   0.000000   6.266498  \n",
       "A2449446903_n01_LICRNA02_A02   0.519342        0.0   0.000000   3.467317  \n",
       "A2449446903_n01_LICRNA02_A03   0.166972        0.0   0.000000   0.000000  \n",
       "A2449446903_n01_LICRNA02_A05   0.628742        0.0   0.000000   0.000000  \n",
       "A2449446903_n01_LICRNA02_A06   1.074213        0.0   0.000000   0.000000  \n",
       "...                                 ...        ...        ...        ...  \n",
       "B250508004_n01_LICRNA03_H08    0.948096        0.0   0.000000   0.760798  \n",
       "B250508004_n01_LICRNA03_H09    0.360071        0.0   1.758217   4.807925  \n",
       "B250508004_n01_LICRNA03_H10    0.000000        0.0   0.000000   0.000000  \n",
       "B250508004_n01_LICRNA03_H11    0.166371        0.0   0.000000   0.694222  \n",
       "B250508004_n01_LICRNA03_H12    0.248594        0.0   0.000000   2.553393  \n",
       "\n",
       "[369 rows x 38287 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "trimmed_transcriptome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4d158033",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "71f44110",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>plate.pos</th>\n",
       "      <th>sampID</th>\n",
       "      <th>sample.type</th>\n",
       "      <th>timepoint</th>\n",
       "      <th>date</th>\n",
       "      <th>time</th>\n",
       "      <th>extractplate</th>\n",
       "      <th>rnaprepplate</th>\n",
       "      <th>Unmapped</th>\n",
       "      <th>No Feature</th>\n",
       "      <th>plate_row</th>\n",
       "      <th>plate_col</th>\n",
       "      <th>Total Reads</th>\n",
       "      <th>Experiment Type</th>\n",
       "      <th>Sample Month</th>\n",
       "      <th>daysincestart</th>\n",
       "      <th>Date and Time</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>filename</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A01</th>\n",
       "      <td>A01</td>\n",
       "      <td>LIC157</td>\n",
       "      <td>plant</td>\n",
       "      <td>t14</td>\n",
       "      <td>1/23/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_02</td>\n",
       "      <td>LICRNA_02</td>\n",
       "      <td>758284.0</td>\n",
       "      <td>82176.0</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "      <td>1246970.0</td>\n",
       "      <td>Long Term</td>\n",
       "      <td>January</td>\n",
       "      <td>83</td>\n",
       "      <td>2024-01-23 08:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A02</th>\n",
       "      <td>A02</td>\n",
       "      <td>LIC169</td>\n",
       "      <td>plant</td>\n",
       "      <td>t15</td>\n",
       "      <td>1/27/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_02</td>\n",
       "      <td>LICRNA_02</td>\n",
       "      <td>3071877.0</td>\n",
       "      <td>353549.0</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>4985602.0</td>\n",
       "      <td>Long Term</td>\n",
       "      <td>January</td>\n",
       "      <td>87</td>\n",
       "      <td>2024-01-27 08:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A03</th>\n",
       "      <td>A03</td>\n",
       "      <td>LIC193</td>\n",
       "      <td>plant</td>\n",
       "      <td>t17</td>\n",
       "      <td>2/2/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_03</td>\n",
       "      <td>LICRNA_02</td>\n",
       "      <td>1213638.0</td>\n",
       "      <td>158864.0</td>\n",
       "      <td>A</td>\n",
       "      <td>3</td>\n",
       "      <td>2465662.0</td>\n",
       "      <td>Long Term</td>\n",
       "      <td>February</td>\n",
       "      <td>93</td>\n",
       "      <td>2024-02-02 08:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A05</th>\n",
       "      <td>A05</td>\n",
       "      <td>LIC217</td>\n",
       "      <td>plant</td>\n",
       "      <td>t19</td>\n",
       "      <td>2/9/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_03</td>\n",
       "      <td>LICRNA_02</td>\n",
       "      <td>1303436.0</td>\n",
       "      <td>90929.0</td>\n",
       "      <td>A</td>\n",
       "      <td>5</td>\n",
       "      <td>1515395.0</td>\n",
       "      <td>Long Term</td>\n",
       "      <td>February</td>\n",
       "      <td>100</td>\n",
       "      <td>2024-02-09 08:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A06</th>\n",
       "      <td>A06</td>\n",
       "      <td>LIC229</td>\n",
       "      <td>plant</td>\n",
       "      <td>t20</td>\n",
       "      <td>2/16/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_03</td>\n",
       "      <td>LICRNA_02</td>\n",
       "      <td>1591231.0</td>\n",
       "      <td>141665.0</td>\n",
       "      <td>A</td>\n",
       "      <td>6</td>\n",
       "      <td>1924391.0</td>\n",
       "      <td>Long Term</td>\n",
       "      <td>February</td>\n",
       "      <td>107</td>\n",
       "      <td>2024-02-16 08:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H08</th>\n",
       "      <td>H08</td>\n",
       "      <td>LIC428</td>\n",
       "      <td>plant</td>\n",
       "      <td>c1_t04</td>\n",
       "      <td>2/10/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LICCIRC_01</td>\n",
       "      <td>LICRNA_03</td>\n",
       "      <td>2552177.0</td>\n",
       "      <td>438552.0</td>\n",
       "      <td>H</td>\n",
       "      <td>8</td>\n",
       "      <td>5245242.0</td>\n",
       "      <td>Circadian Experiment 1</td>\n",
       "      <td>February</td>\n",
       "      <td>101</td>\n",
       "      <td>2024-02-10 08:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H09</th>\n",
       "      <td>H09</td>\n",
       "      <td>LIC440</td>\n",
       "      <td>plant</td>\n",
       "      <td>c1_t05</td>\n",
       "      <td>2/10/24</td>\n",
       "      <td>12:00</td>\n",
       "      <td>LICCIRC_01</td>\n",
       "      <td>LICRNA_03</td>\n",
       "      <td>4228242.0</td>\n",
       "      <td>493699.0</td>\n",
       "      <td>H</td>\n",
       "      <td>9</td>\n",
       "      <td>7715482.0</td>\n",
       "      <td>Circadian Experiment 1</td>\n",
       "      <td>February</td>\n",
       "      <td>101</td>\n",
       "      <td>2024-02-10 12:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H10</th>\n",
       "      <td>H10</td>\n",
       "      <td>LIC452</td>\n",
       "      <td>plant</td>\n",
       "      <td>c1_t06</td>\n",
       "      <td>2/10/24</td>\n",
       "      <td>16:00</td>\n",
       "      <td>LICCIRC_01</td>\n",
       "      <td>LICRNA_03</td>\n",
       "      <td>365095.0</td>\n",
       "      <td>39090.0</td>\n",
       "      <td>H</td>\n",
       "      <td>10</td>\n",
       "      <td>643786.0</td>\n",
       "      <td>Circadian Experiment 1</td>\n",
       "      <td>February</td>\n",
       "      <td>101</td>\n",
       "      <td>2024-02-10 16:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H11</th>\n",
       "      <td>H11</td>\n",
       "      <td>LIC464</td>\n",
       "      <td>plant</td>\n",
       "      <td>c1_t07</td>\n",
       "      <td>2/10/24</td>\n",
       "      <td>20:00</td>\n",
       "      <td>LICCIRC_01</td>\n",
       "      <td>LICRNA_03</td>\n",
       "      <td>326959.0</td>\n",
       "      <td>60346.0</td>\n",
       "      <td>H</td>\n",
       "      <td>11</td>\n",
       "      <td>1509679.0</td>\n",
       "      <td>Circadian Experiment 1</td>\n",
       "      <td>February</td>\n",
       "      <td>101</td>\n",
       "      <td>2024-02-10 20:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H12</th>\n",
       "      <td>H12</td>\n",
       "      <td>LIC488</td>\n",
       "      <td>plant</td>\n",
       "      <td>c1_t08</td>\n",
       "      <td>2/11/24</td>\n",
       "      <td>0:00</td>\n",
       "      <td>LICCIRC_02</td>\n",
       "      <td>LICRNA_03</td>\n",
       "      <td>2282102.0</td>\n",
       "      <td>402777.0</td>\n",
       "      <td>H</td>\n",
       "      <td>12</td>\n",
       "      <td>5444101.0</td>\n",
       "      <td>Circadian Experiment 1</td>\n",
       "      <td>February</td>\n",
       "      <td>102</td>\n",
       "      <td>2024-02-11 00:00:00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>369 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             plate.pos  sampID sample.type timepoint     date  \\\n",
       "filename                                                                        \n",
       "A2449446903_n01_LICRNA02_A01       A01  LIC157       plant       t14  1/23/24   \n",
       "A2449446903_n01_LICRNA02_A02       A02  LIC169       plant       t15  1/27/24   \n",
       "A2449446903_n01_LICRNA02_A03       A03  LIC193       plant       t17   2/2/24   \n",
       "A2449446903_n01_LICRNA02_A05       A05  LIC217       plant       t19   2/9/24   \n",
       "A2449446903_n01_LICRNA02_A06       A06  LIC229       plant       t20  2/16/24   \n",
       "...                                ...     ...         ...       ...      ...   \n",
       "B250508004_n01_LICRNA03_H08        H08  LIC428       plant    c1_t04  2/10/24   \n",
       "B250508004_n01_LICRNA03_H09        H09  LIC440       plant    c1_t05  2/10/24   \n",
       "B250508004_n01_LICRNA03_H10        H10  LIC452       plant    c1_t06  2/10/24   \n",
       "B250508004_n01_LICRNA03_H11        H11  LIC464       plant    c1_t07  2/10/24   \n",
       "B250508004_n01_LICRNA03_H12        H12  LIC488       plant    c1_t08  2/11/24   \n",
       "\n",
       "                               time extractplate rnaprepplate   Unmapped  \\\n",
       "filename                                                                   \n",
       "A2449446903_n01_LICRNA02_A01   8:00       LIC_02    LICRNA_02   758284.0   \n",
       "A2449446903_n01_LICRNA02_A02   8:00       LIC_02    LICRNA_02  3071877.0   \n",
       "A2449446903_n01_LICRNA02_A03   8:00       LIC_03    LICRNA_02  1213638.0   \n",
       "A2449446903_n01_LICRNA02_A05   8:00       LIC_03    LICRNA_02  1303436.0   \n",
       "A2449446903_n01_LICRNA02_A06   8:00       LIC_03    LICRNA_02  1591231.0   \n",
       "...                             ...          ...          ...        ...   \n",
       "B250508004_n01_LICRNA03_H08    8:00   LICCIRC_01    LICRNA_03  2552177.0   \n",
       "B250508004_n01_LICRNA03_H09   12:00   LICCIRC_01    LICRNA_03  4228242.0   \n",
       "B250508004_n01_LICRNA03_H10   16:00   LICCIRC_01    LICRNA_03   365095.0   \n",
       "B250508004_n01_LICRNA03_H11   20:00   LICCIRC_01    LICRNA_03   326959.0   \n",
       "B250508004_n01_LICRNA03_H12    0:00   LICCIRC_02    LICRNA_03  2282102.0   \n",
       "\n",
       "                              No Feature plate_row  plate_col  Total Reads  \\\n",
       "filename                                                                     \n",
       "A2449446903_n01_LICRNA02_A01     82176.0         A          1    1246970.0   \n",
       "A2449446903_n01_LICRNA02_A02    353549.0         A          2    4985602.0   \n",
       "A2449446903_n01_LICRNA02_A03    158864.0         A          3    2465662.0   \n",
       "A2449446903_n01_LICRNA02_A05     90929.0         A          5    1515395.0   \n",
       "A2449446903_n01_LICRNA02_A06    141665.0         A          6    1924391.0   \n",
       "...                                  ...       ...        ...          ...   \n",
       "B250508004_n01_LICRNA03_H08     438552.0         H          8    5245242.0   \n",
       "B250508004_n01_LICRNA03_H09     493699.0         H          9    7715482.0   \n",
       "B250508004_n01_LICRNA03_H10      39090.0         H         10     643786.0   \n",
       "B250508004_n01_LICRNA03_H11      60346.0         H         11    1509679.0   \n",
       "B250508004_n01_LICRNA03_H12     402777.0         H         12    5444101.0   \n",
       "\n",
       "                                     Experiment Type Sample Month  \\\n",
       "filename                                                            \n",
       "A2449446903_n01_LICRNA02_A01               Long Term      January   \n",
       "A2449446903_n01_LICRNA02_A02               Long Term      January   \n",
       "A2449446903_n01_LICRNA02_A03               Long Term     February   \n",
       "A2449446903_n01_LICRNA02_A05               Long Term     February   \n",
       "A2449446903_n01_LICRNA02_A06               Long Term     February   \n",
       "...                                              ...          ...   \n",
       "B250508004_n01_LICRNA03_H08   Circadian Experiment 1     February   \n",
       "B250508004_n01_LICRNA03_H09   Circadian Experiment 1     February   \n",
       "B250508004_n01_LICRNA03_H10   Circadian Experiment 1     February   \n",
       "B250508004_n01_LICRNA03_H11   Circadian Experiment 1     February   \n",
       "B250508004_n01_LICRNA03_H12   Circadian Experiment 1     February   \n",
       "\n",
       "                              daysincestart       Date and Time  \n",
       "filename                                                         \n",
       "A2449446903_n01_LICRNA02_A01             83 2024-01-23 08:00:00  \n",
       "A2449446903_n01_LICRNA02_A02             87 2024-01-27 08:00:00  \n",
       "A2449446903_n01_LICRNA02_A03             93 2024-02-02 08:00:00  \n",
       "A2449446903_n01_LICRNA02_A05            100 2024-02-09 08:00:00  \n",
       "A2449446903_n01_LICRNA02_A06            107 2024-02-16 08:00:00  \n",
       "...                                     ...                 ...  \n",
       "B250508004_n01_LICRNA03_H08             101 2024-02-10 08:00:00  \n",
       "B250508004_n01_LICRNA03_H09             101 2024-02-10 12:00:00  \n",
       "B250508004_n01_LICRNA03_H10             101 2024-02-10 16:00:00  \n",
       "B250508004_n01_LICRNA03_H11             101 2024-02-10 20:00:00  \n",
       "B250508004_n01_LICRNA03_H12             102 2024-02-11 00:00:00  \n",
       "\n",
       "[369 rows x 17 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "trimmed_metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "485cbe11",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Species</th>\n",
       "      <th>n_samples_present</th>\n",
       "      <th>mean_abundance_present</th>\n",
       "      <th>median_abundance_present</th>\n",
       "      <th>prevalence</th>\n",
       "      <th>core</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Kordiimonas sp002401685</td>\n",
       "      <td>372</td>\n",
       "      <td>3.661053</td>\n",
       "      <td>3.30</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Sphingomonas_L_486704 sp000786205</td>\n",
       "      <td>372</td>\n",
       "      <td>17.954211</td>\n",
       "      <td>15.90</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Rhodoferax_C aquaticus</td>\n",
       "      <td>370</td>\n",
       "      <td>4.407143</td>\n",
       "      <td>3.70</td>\n",
       "      <td>0.994624</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Neorhizobium soli</td>\n",
       "      <td>366</td>\n",
       "      <td>3.855080</td>\n",
       "      <td>2.25</td>\n",
       "      <td>0.983871</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Klenkia sp001424455</td>\n",
       "      <td>363</td>\n",
       "      <td>3.581402</td>\n",
       "      <td>2.70</td>\n",
       "      <td>0.975806</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>274</th>\n",
       "      <td>Tardiphaga robiniae</td>\n",
       "      <td>1</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>0.50</td>\n",
       "      <td>0.002688</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>275</th>\n",
       "      <td>Telluria timonae</td>\n",
       "      <td>1</td>\n",
       "      <td>10.800000</td>\n",
       "      <td>10.80</td>\n",
       "      <td>0.002688</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>276</th>\n",
       "      <td>UBA4416 sp002420145</td>\n",
       "      <td>1</td>\n",
       "      <td>0.700000</td>\n",
       "      <td>0.70</td>\n",
       "      <td>0.002688</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>277</th>\n",
       "      <td>Umezawaea tangerina</td>\n",
       "      <td>1</td>\n",
       "      <td>1.100000</td>\n",
       "      <td>1.10</td>\n",
       "      <td>0.002688</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>278</th>\n",
       "      <td>Xenophilus aerolatus</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.002688</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>279 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                               Species  n_samples_present  \\\n",
       "0              Kordiimonas sp002401685                372   \n",
       "1    Sphingomonas_L_486704 sp000786205                372   \n",
       "2               Rhodoferax_C aquaticus                370   \n",
       "3                    Neorhizobium soli                366   \n",
       "4                  Klenkia sp001424455                363   \n",
       "..                                 ...                ...   \n",
       "274                Tardiphaga robiniae                  1   \n",
       "275                   Telluria timonae                  1   \n",
       "276                UBA4416 sp002420145                  1   \n",
       "277                Umezawaea tangerina                  1   \n",
       "278               Xenophilus aerolatus                  1   \n",
       "\n",
       "     mean_abundance_present  median_abundance_present  prevalence   core  \n",
       "0                  3.661053                      3.30    1.000000   True  \n",
       "1                 17.954211                     15.90    1.000000   True  \n",
       "2                  4.407143                      3.70    0.994624   True  \n",
       "3                  3.855080                      2.25    0.983871   True  \n",
       "4                  3.581402                      2.70    0.975806   True  \n",
       "..                      ...                       ...         ...    ...  \n",
       "274                0.500000                      0.50    0.002688  False  \n",
       "275               10.800000                     10.80    0.002688  False  \n",
       "276                0.700000                      0.70    0.002688  False  \n",
       "277                1.100000                      1.10    0.002688  False  \n",
       "278                1.000000                      1.00    0.002688  False  \n",
       "\n",
       "[279 rows x 6 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "species_abundance_table = pd.read_csv(\n",
    "    \"/Users/michael/Data/Luke_terrace_experiment/General_data/Generated_data/species_core_results.csv\"\n",
    ")\n",
    "species_abundance_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "bf7e66ae",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Note 372 total samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "2a132e97",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Species</th>\n",
       "      <th>n_samples_present</th>\n",
       "      <th>mean_abundance_present</th>\n",
       "      <th>median_abundance_present</th>\n",
       "      <th>prevalence</th>\n",
       "      <th>core</th>\n",
       "      <th>rare</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Kordiimonas sp002401685</td>\n",
       "      <td>372</td>\n",
       "      <td>3.661053</td>\n",
       "      <td>3.30</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Sphingomonas_L_486704 sp000786205</td>\n",
       "      <td>372</td>\n",
       "      <td>17.954211</td>\n",
       "      <td>15.90</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Rhodoferax_C aquaticus</td>\n",
       "      <td>370</td>\n",
       "      <td>4.407143</td>\n",
       "      <td>3.70</td>\n",
       "      <td>0.994624</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Neorhizobium soli</td>\n",
       "      <td>366</td>\n",
       "      <td>3.855080</td>\n",
       "      <td>2.25</td>\n",
       "      <td>0.983871</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Klenkia sp001424455</td>\n",
       "      <td>363</td>\n",
       "      <td>3.581402</td>\n",
       "      <td>2.70</td>\n",
       "      <td>0.975806</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>274</th>\n",
       "      <td>Tardiphaga robiniae</td>\n",
       "      <td>1</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>0.50</td>\n",
       "      <td>0.002688</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>275</th>\n",
       "      <td>Telluria timonae</td>\n",
       "      <td>1</td>\n",
       "      <td>10.800000</td>\n",
       "      <td>10.80</td>\n",
       "      <td>0.002688</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>276</th>\n",
       "      <td>UBA4416 sp002420145</td>\n",
       "      <td>1</td>\n",
       "      <td>0.700000</td>\n",
       "      <td>0.70</td>\n",
       "      <td>0.002688</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>277</th>\n",
       "      <td>Umezawaea tangerina</td>\n",
       "      <td>1</td>\n",
       "      <td>1.100000</td>\n",
       "      <td>1.10</td>\n",
       "      <td>0.002688</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>278</th>\n",
       "      <td>Xenophilus aerolatus</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.002688</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>279 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                               Species  n_samples_present  \\\n",
       "0              Kordiimonas sp002401685                372   \n",
       "1    Sphingomonas_L_486704 sp000786205                372   \n",
       "2               Rhodoferax_C aquaticus                370   \n",
       "3                    Neorhizobium soli                366   \n",
       "4                  Klenkia sp001424455                363   \n",
       "..                                 ...                ...   \n",
       "274                Tardiphaga robiniae                  1   \n",
       "275                   Telluria timonae                  1   \n",
       "276                UBA4416 sp002420145                  1   \n",
       "277                Umezawaea tangerina                  1   \n",
       "278               Xenophilus aerolatus                  1   \n",
       "\n",
       "     mean_abundance_present  median_abundance_present  prevalence   core  \\\n",
       "0                  3.661053                      3.30    1.000000   True   \n",
       "1                 17.954211                     15.90    1.000000   True   \n",
       "2                  4.407143                      3.70    0.994624   True   \n",
       "3                  3.855080                      2.25    0.983871   True   \n",
       "4                  3.581402                      2.70    0.975806   True   \n",
       "..                      ...                       ...         ...    ...   \n",
       "274                0.500000                      0.50    0.002688  False   \n",
       "275               10.800000                     10.80    0.002688  False   \n",
       "276                0.700000                      0.70    0.002688  False   \n",
       "277                1.100000                      1.10    0.002688  False   \n",
       "278                1.000000                      1.00    0.002688  False   \n",
       "\n",
       "      rare  \n",
       "0    False  \n",
       "1    False  \n",
       "2    False  \n",
       "3    False  \n",
       "4    False  \n",
       "..     ...  \n",
       "274   True  \n",
       "275   True  \n",
       "276   True  \n",
       "277   True  \n",
       "278   True  \n",
       "\n",
       "[279 rows x 7 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "species_abundance_table[\"rare\"] = False\n",
    "species_abundance_table.loc[\n",
    "    species_abundance_table[\"n_samples_present\"] <= 37, \"rare\"\n",
    "] = True\n",
    "species_abundance_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "dbf0d1e6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Bradyrhizobium sp000938255',\n",
       " 'Noviherbaspirillum_A_568104 psychrotolerans',\n",
       " 'Brevundimonas huaxiensis',\n",
       " 'Mycobacterium paragordonae',\n",
       " 'Aeromicrobium sp001426485',\n",
       " 'Marmoricola_A_392027 sp001424755',\n",
       " 'Flavobacterium sangjuense',\n",
       " 'Herbaspirillum robiniae',\n",
       " 'Hylemonella sp001432305',\n",
       " 'Telluria mixta',\n",
       " 'Undibacterium arcticum',\n",
       " 'Brevundimonas sp001424025',\n",
       " 'Herminiimonas arsenicoxydans',\n",
       " 'Amnibacterium kyonggiense',\n",
       " 'Patulibacter minatonensis',\n",
       " 'Aquabacterium_A_592457 parvum',\n",
       " 'Chioneia frigida',\n",
       " 'Curtobacterium sp001422205',\n",
       " 'Kineosporia rhamnosa',\n",
       " 'Pseudonocardia hierapolitana',\n",
       " 'Nakamurella panacisegetis',\n",
       " 'Curvibacter gracilis',\n",
       " 'Agrobacterium rubi',\n",
       " 'Nocardioides_A_392796 ginsengisegetis',\n",
       " 'Pseudoduganella namucuonensis',\n",
       " 'Pseudomonas_E_647464 rhizosphaerae',\n",
       " 'Actinoplanes rectilineatus',\n",
       " 'Microbacterium_A_383321 testaceum_F',\n",
       " 'Actinoplanes globisporus',\n",
       " 'Herbaspirillum seropedicae',\n",
       " 'Methylophilus methylotrophus',\n",
       " 'Pseudomonas_E_647626 sp900573885',\n",
       " 'Aeromicrobium sp001428805',\n",
       " 'Microbacterium_A_383321 lemovicicum',\n",
       " 'Nakamurella lactea',\n",
       " 'Sphingomonas_L_486704 elodea',\n",
       " 'Pelomonas saccharophila',\n",
       " 'Sphingomonas_L_486704 lenta',\n",
       " 'Sphingomonas_L_486704 sp000382485',\n",
       " 'Cryptosporangium arvum',\n",
       " 'Herbiconiux flava',\n",
       " 'Homoserinimonas sp008001035',\n",
       " 'Methylotenera_A_557637 sp000799735',\n",
       " 'Microbacterium_A_383321 profundi',\n",
       " 'Nocardioides_A_392796 alkalitolerans',\n",
       " 'Stutzerimonas urumqiensis',\n",
       " 'Brevundimonas bacteroides',\n",
       " 'Nocardioides marmoriterrae',\n",
       " 'Nocardioides_A_392796 hankookensis',\n",
       " 'Nocardioides_A_392796 terrae',\n",
       " 'Polaromonas sp002001015',\n",
       " 'Pseudomonas_E_647464 pisciculturae',\n",
       " 'Rhizobium_C_501058 metallidurans',\n",
       " 'Pseudonocardia seranimata',\n",
       " 'Blastococcus capsensis',\n",
       " 'DSM-44207 sp900101335',\n",
       " 'Duganella aquatica',\n",
       " 'Rhizobium_C_501058 grahamii',\n",
       " 'Sphingomonas_L_486704 mali',\n",
       " 'UBA1952 sp004144185',\n",
       " 'Agreia bicolorata',\n",
       " 'Aureimonas_A_501548 sp001423245',\n",
       " 'Microbacterium_A_383321 luteum',\n",
       " 'Nakamurella multipartita',\n",
       " 'Nocardioides_A_392796 psychrotolerans',\n",
       " 'Rugamonas rubra',\n",
       " 'Sphingomonas_L_486704 insulae',\n",
       " 'Sphingomonas_R_483406 montana',\n",
       " 'Telluria brevitalea',\n",
       " 'Devosia_A_502124 sp001426345',\n",
       " 'Labilithrix luteola',\n",
       " 'Modestobacter muralis',\n",
       " 'Neorhizobium sp900469445',\n",
       " 'Noviherbaspirillum_A_568106 aurantiacum',\n",
       " 'Sphingomonas_L_486704 olei',\n",
       " 'Sphingomonas_L_486704 sp001421355',\n",
       " 'Brevundimonas denitrificans',\n",
       " 'Caulobacter soli',\n",
       " 'Methylobacterium radiotolerans',\n",
       " 'Motilibacter_A rhizosphaerae',\n",
       " 'Telluria sp000745265',\n",
       " 'Aldersonia kunmingensis',\n",
       " 'Aquabacterium_A_591266 sp002198735',\n",
       " 'Flavobacterium gilvum',\n",
       " 'Noviherbaspirillum_A_569121 massiliense',\n",
       " 'Sphingomonas_L_486704 aracearum',\n",
       " 'Uliginosibacterium sediminicola',\n",
       " 'Aeromicrobium sp001423335',\n",
       " 'Amycolatopsis_D_380400 oliviviridis',\n",
       " 'Baekduia woesei_A',\n",
       " 'Bosea sp011764485',\n",
       " 'Buchnera aphidicola_E',\n",
       " 'Flavobacterium akiainvivens',\n",
       " 'Microlunatus_B parietis',\n",
       " 'Naasia flava',\n",
       " 'Nakamurella albus',\n",
       " 'Nocardioides_A_392796 cynanchi',\n",
       " 'Pseudorhodoferax sp001422445',\n",
       " 'Rhizobacter sp001425825',\n",
       " 'Rhodoferax_C lacus',\n",
       " 'UBA5216 sp902825845',\n",
       " 'Virgisporangium myanmarense',\n",
       " 'Abditibacterium utsteinense',\n",
       " 'Arthrobacter ureafaciens',\n",
       " 'Arthrobacter_E_385777 globiformis',\n",
       " 'Caballeronia pedi',\n",
       " 'Chryseobacterium_A_795111 sp001507335',\n",
       " 'Chryseobacterium_A_796612 daeguense',\n",
       " 'Chryseobacterium_A_796612 sp000333355',\n",
       " 'Conyzicola nivalis',\n",
       " 'Flavobacterium antarcticum',\n",
       " 'Flavobacterium cellulosilyticum',\n",
       " 'Inhella proteolytica',\n",
       " 'Labedella gwakjiensis',\n",
       " 'Methylobacterium organophilum',\n",
       " 'Methylobacterium thiocyanatum',\n",
       " 'Motilibacter_A deserti',\n",
       " 'Phenylobacterium sp002256425',\n",
       " 'Pseudoduganella rivuli',\n",
       " 'Pseudomonas_O kuykendallii',\n",
       " 'Pseudonocardia asaccharolytica',\n",
       " 'Pseudonocardia thermophila',\n",
       " 'Rhizorhabdus argentea',\n",
       " 'AVCC01 sp000622895',\n",
       " 'Actinocorallia herbida',\n",
       " 'Aureimonas_A_501549 ferruginea',\n",
       " 'Bosea caraganae',\n",
       " 'Bosea massiliensis',\n",
       " 'Bradyrhizobium sp000244915',\n",
       " 'Caballeronia fortuita',\n",
       " 'Caulobacter henricii',\n",
       " 'Caulobacter vibrioides_C_487720',\n",
       " 'Dankookia rubra',\n",
       " 'Ga0074137 sp001464715',\n",
       " 'Geodermatophilus sp000701365',\n",
       " 'Janthinobacterium sp000344615',\n",
       " 'Jatrophihabitans_A_372606 huperziae',\n",
       " 'Kineosporia aurantiaca',\n",
       " 'Methylobacterium brachiatum',\n",
       " 'Methylobacterium sp000372825',\n",
       " 'Micromonospora_H_372070 lutea',\n",
       " 'Mycobacterium aromaticivorans',\n",
       " 'Nocardioides_A massiliensis',\n",
       " 'Nocardioides_A_392796 albus',\n",
       " 'Patulibacter ginsengiterrae',\n",
       " 'Pedococcus sp001429605',\n",
       " 'Pseudokineococcus basanitobsidens',\n",
       " 'Pseudomonas_E_647464 lutea',\n",
       " 'Pseudomonas_E_647464 siliginis',\n",
       " 'Rathayibacter sp001423885',\n",
       " 'Terracoccus luteus',\n",
       " 'Williamsia_A deligens',\n",
       " 'Actinoallomurus bryophytorum',\n",
       " 'Actinomycetospora succinea',\n",
       " 'Agrococcus jejuensis',\n",
       " 'Allorhizobium sp900176345',\n",
       " 'Alsobacter soli',\n",
       " 'Amnibacterium endophyticum',\n",
       " 'Aquabacterium_A_592457 soli',\n",
       " 'Arsenicicoccus dermatophilus',\n",
       " 'Arthrobacter chlorophenolicus',\n",
       " 'Asticcacaulis solisilvae',\n",
       " 'Aureimonas_A_501548 galii',\n",
       " 'Brevundimonas humi',\n",
       " 'Brevundimonas sp001425945',\n",
       " 'Catenuloplanes crispus',\n",
       " 'Cellvibrio zantedeschiae',\n",
       " 'Chryseobacterium_A_796612 arachidis',\n",
       " 'Clavibacter sepedonicus',\n",
       " 'Croceibacterium mercuriale',\n",
       " 'Flavobacterium hauense',\n",
       " 'Granulicoccus phenolivorans',\n",
       " 'Gryllotalpicola ginsengisoli',\n",
       " 'Herbiconiux moechotypicola',\n",
       " 'Herbiconiux solani',\n",
       " 'Kocuria polaris',\n",
       " 'Kutzneria kofuensis',\n",
       " 'Limnohabitans_A planktonicus',\n",
       " 'Marihabitans melonis',\n",
       " 'Marmoricola_A_392027 ginsengisoli',\n",
       " 'Massilia polaris',\n",
       " 'Mesorhizobium_I ciceri',\n",
       " 'Methylobacterium durans',\n",
       " 'Micromonospora_H_372070 siamensis',\n",
       " 'Nakamurella silvestris',\n",
       " 'Nocardioides_A_392796 allogilvus',\n",
       " 'Nocardioides_A_392796 jishulii',\n",
       " 'Nocardioides_A_392796 sp000015265',\n",
       " 'Nocardioides_A_392796 sp000364605',\n",
       " 'Nocardioides_A_392796 sp001429625',\n",
       " 'Paraburkholderia sprentiae',\n",
       " 'Paracidovorax avenae',\n",
       " 'Paraherbaspirillum soli',\n",
       " 'Pararhizobium_B_499893 sp001424985',\n",
       " 'Phenylobacterium soli',\n",
       " 'Phytohabitans suffuscus',\n",
       " 'Pseudoclavibacter_C sp002931055',\n",
       " 'Pseudoduganella sp001191005',\n",
       " 'Rhizobium_C_501058 mesoamericanum',\n",
       " 'Roseateles depolymerans',\n",
       " 'Roseomonas rosea',\n",
       " 'Roseomonas sp001941945',\n",
       " 'Solirubrobacter sp000425945',\n",
       " 'Sphingomonas_L_486704 hengshuiensis',\n",
       " 'Sphingomonas_L_486704 soli',\n",
       " 'Sphingomonas_L_486704 sp001421995',\n",
       " 'Stenotrophomonas_A_615274 rhizophila',\n",
       " 'Streptomyces acidiscabies',\n",
       " 'Streptomyces sp001298575',\n",
       " 'Tardiphaga robiniae',\n",
       " 'Telluria timonae',\n",
       " 'UBA4416 sp002420145',\n",
       " 'Umezawaea tangerina',\n",
       " 'Xenophilus aerolatus']"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list_of_rare_species = species_abundance_table.loc[\n",
    "    species_abundance_table[\"rare\"] == True, \"Species\"\n",
    "].tolist()\n",
    "list_of_core_species = species_abundance_table.loc[\n",
    "    species_abundance_table[\"core\"] == True, \"Species\"\n",
    "].tolist()\n",
    "list_of_rare_species"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "1390a171",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Kordiimonas sp002401685',\n",
       " 'Sphingomonas_L_486704 sp000786205',\n",
       " 'Rhodoferax_C aquaticus',\n",
       " 'Neorhizobium soli',\n",
       " 'Klenkia sp001424455',\n",
       " 'Massilia atriviolacea',\n",
       " 'Variovorax paradoxus_C',\n",
       " 'Flavobacterium chilense',\n",
       " 'Sphingomonas_L_486704 taxi',\n",
       " 'Frigoribacterium sp001424645']"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list_of_core_species"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "f0de5efd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample</th>\n",
       "      <th>sample.type</th>\n",
       "      <th>timepoint</th>\n",
       "      <th>date</th>\n",
       "      <th>time</th>\n",
       "      <th>platename</th>\n",
       "      <th>daysincestart</th>\n",
       "      <th>plantID</th>\n",
       "      <th>Kingdom</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "      <th>AbundR100</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LIC522</td>\n",
       "      <td>plant</td>\n",
       "      <td>c1_t11</td>\n",
       "      <td>2/11/24</td>\n",
       "      <td>12:00</td>\n",
       "      <td>LICCIRC_02</td>\n",
       "      <td>102</td>\n",
       "      <td>LIC522</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Pseudomonadales_A_650611</td>\n",
       "      <td>Pseudomonadaceae</td>\n",
       "      <td>Pseudomonas_E_647464</td>\n",
       "      <td>Pseudomonas_E_647464 viridiflava</td>\n",
       "      <td>79.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>LIC063</td>\n",
       "      <td>plant</td>\n",
       "      <td>t06</td>\n",
       "      <td>11/28/23</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_01</td>\n",
       "      <td>27</td>\n",
       "      <td>LIC063</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Sphingomonadales</td>\n",
       "      <td>Sphingomonadaceae_486827</td>\n",
       "      <td>Sphingomonas_L_486704</td>\n",
       "      <td>Sphingomonas_L_486704 sp000786205</td>\n",
       "      <td>66.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LIC159</td>\n",
       "      <td>plant</td>\n",
       "      <td>t14</td>\n",
       "      <td>1/23/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_02</td>\n",
       "      <td>83</td>\n",
       "      <td>LIC159</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>Cyanobacteriota</td>\n",
       "      <td>Cyanobacteriia</td>\n",
       "      <td>Cyanobacteriales</td>\n",
       "      <td>Coleofasciculaceae</td>\n",
       "      <td>Caldora</td>\n",
       "      <td>Caldora sp010672925</td>\n",
       "      <td>60.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LIC452</td>\n",
       "      <td>plant</td>\n",
       "      <td>c1_t06</td>\n",
       "      <td>2/10/24</td>\n",
       "      <td>16:00</td>\n",
       "      <td>LICCIRC_01</td>\n",
       "      <td>101</td>\n",
       "      <td>LIC452</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Burkholderiales</td>\n",
       "      <td>Burkholderiaceae_A_595421</td>\n",
       "      <td>Massilia_574544</td>\n",
       "      <td>Massilia atriviolacea</td>\n",
       "      <td>59.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LIC306</td>\n",
       "      <td>plant</td>\n",
       "      <td>t26</td>\n",
       "      <td>3/8/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_04</td>\n",
       "      <td>128</td>\n",
       "      <td>LIC306</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Sphingomonadales</td>\n",
       "      <td>Sphingomonadaceae_486827</td>\n",
       "      <td>Sphingomonas_L_486704</td>\n",
       "      <td>Sphingomonas_L_486704 sp000786205</td>\n",
       "      <td>57.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16888</th>\n",
       "      <td>t03_soil</td>\n",
       "      <td>soil</td>\n",
       "      <td>t03</td>\n",
       "      <td>11/14/23</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_01</td>\n",
       "      <td>13</td>\n",
       "      <td>t03_soil</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>Bacillota_I</td>\n",
       "      <td>Bacilli_A</td>\n",
       "      <td>Bacillales_B_306089</td>\n",
       "      <td>Bacillaceae_H_289398</td>\n",
       "      <td>Priestia_289346</td>\n",
       "      <td>Priestia megaterium</td>\n",
       "      <td>0.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16889</th>\n",
       "      <td>LIC631</td>\n",
       "      <td>plant</td>\n",
       "      <td>c2_t05</td>\n",
       "      <td>3/2/24</td>\n",
       "      <td>12:00</td>\n",
       "      <td>LICCIRC_03</td>\n",
       "      <td>122</td>\n",
       "      <td>LIC631</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Caulobacterales</td>\n",
       "      <td>Caulobacteraceae</td>\n",
       "      <td>Brevundimonas</td>\n",
       "      <td>Brevundimonas subvibrioides</td>\n",
       "      <td>0.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16890</th>\n",
       "      <td>LIC712</td>\n",
       "      <td>plant</td>\n",
       "      <td>c2_t10</td>\n",
       "      <td>3/3/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LICCIRC_04</td>\n",
       "      <td>123</td>\n",
       "      <td>LIC712</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>Actinomycetota</td>\n",
       "      <td>Actinomycetes</td>\n",
       "      <td>Mycobacteriales</td>\n",
       "      <td>Geodermatophilaceae</td>\n",
       "      <td>Klenkia</td>\n",
       "      <td>Klenkia sp001424455</td>\n",
       "      <td>0.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16891</th>\n",
       "      <td>LIC629</td>\n",
       "      <td>plant</td>\n",
       "      <td>c2_t05</td>\n",
       "      <td>3/2/24</td>\n",
       "      <td>12:00</td>\n",
       "      <td>LICCIRC_03</td>\n",
       "      <td>122</td>\n",
       "      <td>LIC629</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Rhizobiales_505101</td>\n",
       "      <td>Rhizobiaceae</td>\n",
       "      <td>Agrobacterium</td>\n",
       "      <td>Agrobacterium larrymoorei</td>\n",
       "      <td>0.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16892</th>\n",
       "      <td>LIC230</td>\n",
       "      <td>plant</td>\n",
       "      <td>t20</td>\n",
       "      <td>2/16/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_03</td>\n",
       "      <td>107</td>\n",
       "      <td>LIC230</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Rhizobiales_505101</td>\n",
       "      <td>Beijerinckiaceae</td>\n",
       "      <td>Methylobacterium</td>\n",
       "      <td>Methylobacterium sp001424705</td>\n",
       "      <td>0.5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>16893 rows × 16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Sample sample.type timepoint      date   time   platename  \\\n",
       "0        LIC522       plant    c1_t11   2/11/24  12:00  LICCIRC_02   \n",
       "1        LIC063       plant       t06  11/28/23   8:00      LIC_01   \n",
       "2        LIC159       plant       t14   1/23/24   8:00      LIC_02   \n",
       "3        LIC452       plant    c1_t06   2/10/24  16:00  LICCIRC_01   \n",
       "4        LIC306       plant       t26    3/8/24   8:00      LIC_04   \n",
       "...         ...         ...       ...       ...    ...         ...   \n",
       "16888  t03_soil        soil       t03  11/14/23   8:00      LIC_01   \n",
       "16889    LIC631       plant    c2_t05    3/2/24  12:00  LICCIRC_03   \n",
       "16890    LIC712       plant    c2_t10    3/3/24   8:00  LICCIRC_04   \n",
       "16891    LIC629       plant    c2_t05    3/2/24  12:00  LICCIRC_03   \n",
       "16892    LIC230       plant       t20   2/16/24   8:00      LIC_03   \n",
       "\n",
       "       daysincestart   plantID      Kingdom           Phylum  \\\n",
       "0                102    LIC522  d__Bacteria   Pseudomonadota   \n",
       "1                 27    LIC063  d__Bacteria   Pseudomonadota   \n",
       "2                 83    LIC159  d__Bacteria  Cyanobacteriota   \n",
       "3                101    LIC452  d__Bacteria   Pseudomonadota   \n",
       "4                128    LIC306  d__Bacteria   Pseudomonadota   \n",
       "...              ...       ...          ...              ...   \n",
       "16888             13  t03_soil  d__Bacteria      Bacillota_I   \n",
       "16889            122    LIC631  d__Bacteria   Pseudomonadota   \n",
       "16890            123    LIC712  d__Bacteria   Actinomycetota   \n",
       "16891            122    LIC629  d__Bacteria   Pseudomonadota   \n",
       "16892            107    LIC230  d__Bacteria   Pseudomonadota   \n",
       "\n",
       "                     Class                     Order  \\\n",
       "0      Gammaproteobacteria  Pseudomonadales_A_650611   \n",
       "1      Alphaproteobacteria          Sphingomonadales   \n",
       "2           Cyanobacteriia          Cyanobacteriales   \n",
       "3      Gammaproteobacteria           Burkholderiales   \n",
       "4      Alphaproteobacteria          Sphingomonadales   \n",
       "...                    ...                       ...   \n",
       "16888            Bacilli_A       Bacillales_B_306089   \n",
       "16889  Alphaproteobacteria           Caulobacterales   \n",
       "16890        Actinomycetes           Mycobacteriales   \n",
       "16891  Alphaproteobacteria        Rhizobiales_505101   \n",
       "16892  Alphaproteobacteria        Rhizobiales_505101   \n",
       "\n",
       "                          Family                  Genus  \\\n",
       "0               Pseudomonadaceae   Pseudomonas_E_647464   \n",
       "1       Sphingomonadaceae_486827  Sphingomonas_L_486704   \n",
       "2             Coleofasciculaceae                Caldora   \n",
       "3      Burkholderiaceae_A_595421        Massilia_574544   \n",
       "4       Sphingomonadaceae_486827  Sphingomonas_L_486704   \n",
       "...                          ...                    ...   \n",
       "16888       Bacillaceae_H_289398        Priestia_289346   \n",
       "16889           Caulobacteraceae          Brevundimonas   \n",
       "16890        Geodermatophilaceae                Klenkia   \n",
       "16891               Rhizobiaceae          Agrobacterium   \n",
       "16892           Beijerinckiaceae       Methylobacterium   \n",
       "\n",
       "                                 Species  AbundR100  \n",
       "0       Pseudomonas_E_647464 viridiflava       79.1  \n",
       "1      Sphingomonas_L_486704 sp000786205       66.1  \n",
       "2                    Caldora sp010672925       60.5  \n",
       "3                  Massilia atriviolacea       59.8  \n",
       "4      Sphingomonas_L_486704 sp000786205       57.2  \n",
       "...                                  ...        ...  \n",
       "16888                Priestia megaterium        0.5  \n",
       "16889        Brevundimonas subvibrioides        0.5  \n",
       "16890                Klenkia sp001424455        0.5  \n",
       "16891          Agrobacterium larrymoorei        0.5  \n",
       "16892       Methylobacterium sp001424705        0.5  \n",
       "\n",
       "[16893 rows x 16 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "microbiome_abundance"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "bb8d0657",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>plate.pos</th>\n",
       "      <th>sampID</th>\n",
       "      <th>sample.type</th>\n",
       "      <th>timepoint</th>\n",
       "      <th>date</th>\n",
       "      <th>time</th>\n",
       "      <th>extractplate</th>\n",
       "      <th>rnaprepplate</th>\n",
       "      <th>Unmapped</th>\n",
       "      <th>No Feature</th>\n",
       "      <th>plate_row</th>\n",
       "      <th>plate_col</th>\n",
       "      <th>Total Reads</th>\n",
       "      <th>Experiment Type</th>\n",
       "      <th>Sample Month</th>\n",
       "      <th>daysincestart</th>\n",
       "      <th>Date and Time</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>filename</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A01</th>\n",
       "      <td>A01</td>\n",
       "      <td>LIC157</td>\n",
       "      <td>plant</td>\n",
       "      <td>t14</td>\n",
       "      <td>1/23/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_02</td>\n",
       "      <td>LICRNA_02</td>\n",
       "      <td>758284.0</td>\n",
       "      <td>82176.0</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "      <td>1246970.0</td>\n",
       "      <td>Long Term</td>\n",
       "      <td>January</td>\n",
       "      <td>83</td>\n",
       "      <td>2024-01-23 08:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A02</th>\n",
       "      <td>A02</td>\n",
       "      <td>LIC169</td>\n",
       "      <td>plant</td>\n",
       "      <td>t15</td>\n",
       "      <td>1/27/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_02</td>\n",
       "      <td>LICRNA_02</td>\n",
       "      <td>3071877.0</td>\n",
       "      <td>353549.0</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>4985602.0</td>\n",
       "      <td>Long Term</td>\n",
       "      <td>January</td>\n",
       "      <td>87</td>\n",
       "      <td>2024-01-27 08:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A03</th>\n",
       "      <td>A03</td>\n",
       "      <td>LIC193</td>\n",
       "      <td>plant</td>\n",
       "      <td>t17</td>\n",
       "      <td>2/2/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_03</td>\n",
       "      <td>LICRNA_02</td>\n",
       "      <td>1213638.0</td>\n",
       "      <td>158864.0</td>\n",
       "      <td>A</td>\n",
       "      <td>3</td>\n",
       "      <td>2465662.0</td>\n",
       "      <td>Long Term</td>\n",
       "      <td>February</td>\n",
       "      <td>93</td>\n",
       "      <td>2024-02-02 08:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A05</th>\n",
       "      <td>A05</td>\n",
       "      <td>LIC217</td>\n",
       "      <td>plant</td>\n",
       "      <td>t19</td>\n",
       "      <td>2/9/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_03</td>\n",
       "      <td>LICRNA_02</td>\n",
       "      <td>1303436.0</td>\n",
       "      <td>90929.0</td>\n",
       "      <td>A</td>\n",
       "      <td>5</td>\n",
       "      <td>1515395.0</td>\n",
       "      <td>Long Term</td>\n",
       "      <td>February</td>\n",
       "      <td>100</td>\n",
       "      <td>2024-02-09 08:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A06</th>\n",
       "      <td>A06</td>\n",
       "      <td>LIC229</td>\n",
       "      <td>plant</td>\n",
       "      <td>t20</td>\n",
       "      <td>2/16/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_03</td>\n",
       "      <td>LICRNA_02</td>\n",
       "      <td>1591231.0</td>\n",
       "      <td>141665.0</td>\n",
       "      <td>A</td>\n",
       "      <td>6</td>\n",
       "      <td>1924391.0</td>\n",
       "      <td>Long Term</td>\n",
       "      <td>February</td>\n",
       "      <td>107</td>\n",
       "      <td>2024-02-16 08:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H08</th>\n",
       "      <td>H08</td>\n",
       "      <td>LIC428</td>\n",
       "      <td>plant</td>\n",
       "      <td>c1_t04</td>\n",
       "      <td>2/10/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LICCIRC_01</td>\n",
       "      <td>LICRNA_03</td>\n",
       "      <td>2552177.0</td>\n",
       "      <td>438552.0</td>\n",
       "      <td>H</td>\n",
       "      <td>8</td>\n",
       "      <td>5245242.0</td>\n",
       "      <td>Circadian Experiment 1</td>\n",
       "      <td>February</td>\n",
       "      <td>101</td>\n",
       "      <td>2024-02-10 08:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H09</th>\n",
       "      <td>H09</td>\n",
       "      <td>LIC440</td>\n",
       "      <td>plant</td>\n",
       "      <td>c1_t05</td>\n",
       "      <td>2/10/24</td>\n",
       "      <td>12:00</td>\n",
       "      <td>LICCIRC_01</td>\n",
       "      <td>LICRNA_03</td>\n",
       "      <td>4228242.0</td>\n",
       "      <td>493699.0</td>\n",
       "      <td>H</td>\n",
       "      <td>9</td>\n",
       "      <td>7715482.0</td>\n",
       "      <td>Circadian Experiment 1</td>\n",
       "      <td>February</td>\n",
       "      <td>101</td>\n",
       "      <td>2024-02-10 12:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H10</th>\n",
       "      <td>H10</td>\n",
       "      <td>LIC452</td>\n",
       "      <td>plant</td>\n",
       "      <td>c1_t06</td>\n",
       "      <td>2/10/24</td>\n",
       "      <td>16:00</td>\n",
       "      <td>LICCIRC_01</td>\n",
       "      <td>LICRNA_03</td>\n",
       "      <td>365095.0</td>\n",
       "      <td>39090.0</td>\n",
       "      <td>H</td>\n",
       "      <td>10</td>\n",
       "      <td>643786.0</td>\n",
       "      <td>Circadian Experiment 1</td>\n",
       "      <td>February</td>\n",
       "      <td>101</td>\n",
       "      <td>2024-02-10 16:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H11</th>\n",
       "      <td>H11</td>\n",
       "      <td>LIC464</td>\n",
       "      <td>plant</td>\n",
       "      <td>c1_t07</td>\n",
       "      <td>2/10/24</td>\n",
       "      <td>20:00</td>\n",
       "      <td>LICCIRC_01</td>\n",
       "      <td>LICRNA_03</td>\n",
       "      <td>326959.0</td>\n",
       "      <td>60346.0</td>\n",
       "      <td>H</td>\n",
       "      <td>11</td>\n",
       "      <td>1509679.0</td>\n",
       "      <td>Circadian Experiment 1</td>\n",
       "      <td>February</td>\n",
       "      <td>101</td>\n",
       "      <td>2024-02-10 20:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H12</th>\n",
       "      <td>H12</td>\n",
       "      <td>LIC488</td>\n",
       "      <td>plant</td>\n",
       "      <td>c1_t08</td>\n",
       "      <td>2/11/24</td>\n",
       "      <td>0:00</td>\n",
       "      <td>LICCIRC_02</td>\n",
       "      <td>LICRNA_03</td>\n",
       "      <td>2282102.0</td>\n",
       "      <td>402777.0</td>\n",
       "      <td>H</td>\n",
       "      <td>12</td>\n",
       "      <td>5444101.0</td>\n",
       "      <td>Circadian Experiment 1</td>\n",
       "      <td>February</td>\n",
       "      <td>102</td>\n",
       "      <td>2024-02-11 00:00:00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>369 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             plate.pos  sampID sample.type timepoint     date  \\\n",
       "filename                                                                        \n",
       "A2449446903_n01_LICRNA02_A01       A01  LIC157       plant       t14  1/23/24   \n",
       "A2449446903_n01_LICRNA02_A02       A02  LIC169       plant       t15  1/27/24   \n",
       "A2449446903_n01_LICRNA02_A03       A03  LIC193       plant       t17   2/2/24   \n",
       "A2449446903_n01_LICRNA02_A05       A05  LIC217       plant       t19   2/9/24   \n",
       "A2449446903_n01_LICRNA02_A06       A06  LIC229       plant       t20  2/16/24   \n",
       "...                                ...     ...         ...       ...      ...   \n",
       "B250508004_n01_LICRNA03_H08        H08  LIC428       plant    c1_t04  2/10/24   \n",
       "B250508004_n01_LICRNA03_H09        H09  LIC440       plant    c1_t05  2/10/24   \n",
       "B250508004_n01_LICRNA03_H10        H10  LIC452       plant    c1_t06  2/10/24   \n",
       "B250508004_n01_LICRNA03_H11        H11  LIC464       plant    c1_t07  2/10/24   \n",
       "B250508004_n01_LICRNA03_H12        H12  LIC488       plant    c1_t08  2/11/24   \n",
       "\n",
       "                               time extractplate rnaprepplate   Unmapped  \\\n",
       "filename                                                                   \n",
       "A2449446903_n01_LICRNA02_A01   8:00       LIC_02    LICRNA_02   758284.0   \n",
       "A2449446903_n01_LICRNA02_A02   8:00       LIC_02    LICRNA_02  3071877.0   \n",
       "A2449446903_n01_LICRNA02_A03   8:00       LIC_03    LICRNA_02  1213638.0   \n",
       "A2449446903_n01_LICRNA02_A05   8:00       LIC_03    LICRNA_02  1303436.0   \n",
       "A2449446903_n01_LICRNA02_A06   8:00       LIC_03    LICRNA_02  1591231.0   \n",
       "...                             ...          ...          ...        ...   \n",
       "B250508004_n01_LICRNA03_H08    8:00   LICCIRC_01    LICRNA_03  2552177.0   \n",
       "B250508004_n01_LICRNA03_H09   12:00   LICCIRC_01    LICRNA_03  4228242.0   \n",
       "B250508004_n01_LICRNA03_H10   16:00   LICCIRC_01    LICRNA_03   365095.0   \n",
       "B250508004_n01_LICRNA03_H11   20:00   LICCIRC_01    LICRNA_03   326959.0   \n",
       "B250508004_n01_LICRNA03_H12    0:00   LICCIRC_02    LICRNA_03  2282102.0   \n",
       "\n",
       "                              No Feature plate_row  plate_col  Total Reads  \\\n",
       "filename                                                                     \n",
       "A2449446903_n01_LICRNA02_A01     82176.0         A          1    1246970.0   \n",
       "A2449446903_n01_LICRNA02_A02    353549.0         A          2    4985602.0   \n",
       "A2449446903_n01_LICRNA02_A03    158864.0         A          3    2465662.0   \n",
       "A2449446903_n01_LICRNA02_A05     90929.0         A          5    1515395.0   \n",
       "A2449446903_n01_LICRNA02_A06    141665.0         A          6    1924391.0   \n",
       "...                                  ...       ...        ...          ...   \n",
       "B250508004_n01_LICRNA03_H08     438552.0         H          8    5245242.0   \n",
       "B250508004_n01_LICRNA03_H09     493699.0         H          9    7715482.0   \n",
       "B250508004_n01_LICRNA03_H10      39090.0         H         10     643786.0   \n",
       "B250508004_n01_LICRNA03_H11      60346.0         H         11    1509679.0   \n",
       "B250508004_n01_LICRNA03_H12     402777.0         H         12    5444101.0   \n",
       "\n",
       "                                     Experiment Type Sample Month  \\\n",
       "filename                                                            \n",
       "A2449446903_n01_LICRNA02_A01               Long Term      January   \n",
       "A2449446903_n01_LICRNA02_A02               Long Term      January   \n",
       "A2449446903_n01_LICRNA02_A03               Long Term     February   \n",
       "A2449446903_n01_LICRNA02_A05               Long Term     February   \n",
       "A2449446903_n01_LICRNA02_A06               Long Term     February   \n",
       "...                                              ...          ...   \n",
       "B250508004_n01_LICRNA03_H08   Circadian Experiment 1     February   \n",
       "B250508004_n01_LICRNA03_H09   Circadian Experiment 1     February   \n",
       "B250508004_n01_LICRNA03_H10   Circadian Experiment 1     February   \n",
       "B250508004_n01_LICRNA03_H11   Circadian Experiment 1     February   \n",
       "B250508004_n01_LICRNA03_H12   Circadian Experiment 1     February   \n",
       "\n",
       "                              daysincestart       Date and Time  \n",
       "filename                                                         \n",
       "A2449446903_n01_LICRNA02_A01             83 2024-01-23 08:00:00  \n",
       "A2449446903_n01_LICRNA02_A02             87 2024-01-27 08:00:00  \n",
       "A2449446903_n01_LICRNA02_A03             93 2024-02-02 08:00:00  \n",
       "A2449446903_n01_LICRNA02_A05            100 2024-02-09 08:00:00  \n",
       "A2449446903_n01_LICRNA02_A06            107 2024-02-16 08:00:00  \n",
       "...                                     ...                 ...  \n",
       "B250508004_n01_LICRNA03_H08             101 2024-02-10 08:00:00  \n",
       "B250508004_n01_LICRNA03_H09             101 2024-02-10 12:00:00  \n",
       "B250508004_n01_LICRNA03_H10             101 2024-02-10 16:00:00  \n",
       "B250508004_n01_LICRNA03_H11             101 2024-02-10 20:00:00  \n",
       "B250508004_n01_LICRNA03_H12             102 2024-02-11 00:00:00  \n",
       "\n",
       "[369 rows x 17 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "trimmed_metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "b4cb9d1e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sampID</th>\n",
       "      <th>has_rare_species</th>\n",
       "      <th>lacking_core_species</th>\n",
       "      <th>rare_species_total_abundR100</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>filename</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A01</th>\n",
       "      <td>LIC157</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>13.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A02</th>\n",
       "      <td>LIC169</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>0.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A03</th>\n",
       "      <td>LIC193</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>1.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A05</th>\n",
       "      <td>LIC217</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A06</th>\n",
       "      <td>LIC229</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H08</th>\n",
       "      <td>LIC428</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H09</th>\n",
       "      <td>LIC440</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H10</th>\n",
       "      <td>LIC452</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H11</th>\n",
       "      <td>LIC464</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>2.9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H12</th>\n",
       "      <td>LIC488</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>369 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              sampID  has_rare_species  lacking_core_species  \\\n",
       "filename                                                                       \n",
       "A2449446903_n01_LICRNA02_A01  LIC157              True                 False   \n",
       "A2449446903_n01_LICRNA02_A02  LIC169              True                  True   \n",
       "A2449446903_n01_LICRNA02_A03  LIC193              True                 False   \n",
       "A2449446903_n01_LICRNA02_A05  LIC217              True                 False   \n",
       "A2449446903_n01_LICRNA02_A06  LIC229              True                  True   \n",
       "...                              ...               ...                   ...   \n",
       "B250508004_n01_LICRNA03_H08   LIC428             False                  True   \n",
       "B250508004_n01_LICRNA03_H09   LIC440             False                 False   \n",
       "B250508004_n01_LICRNA03_H10   LIC452              True                  True   \n",
       "B250508004_n01_LICRNA03_H11   LIC464              True                  True   \n",
       "B250508004_n01_LICRNA03_H12   LIC488              True                  True   \n",
       "\n",
       "                              rare_species_total_abundR100  \n",
       "filename                                                    \n",
       "A2449446903_n01_LICRNA02_A01                          13.5  \n",
       "A2449446903_n01_LICRNA02_A02                           0.7  \n",
       "A2449446903_n01_LICRNA02_A03                           1.7  \n",
       "A2449446903_n01_LICRNA02_A05                           1.0  \n",
       "A2449446903_n01_LICRNA02_A06                           5.0  \n",
       "...                                                    ...  \n",
       "B250508004_n01_LICRNA03_H08                            0.0  \n",
       "B250508004_n01_LICRNA03_H09                            0.0  \n",
       "B250508004_n01_LICRNA03_H10                            1.0  \n",
       "B250508004_n01_LICRNA03_H11                            2.9  \n",
       "B250508004_n01_LICRNA03_H12                            1.0  \n",
       "\n",
       "[369 rows x 4 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# For each sample, check if it has any rare species or is lacking any core species\n",
    "rare_species_set = set(list_of_rare_species)\n",
    "core_species_set = set(list_of_core_species)\n",
    "\n",
    "# Get species present for each sample (using sampID from long_term_microbiome)\n",
    "sample_species = long_term_microbiome.groupby(\"sampID\")[\"Species\"].apply(set)\n",
    "\n",
    "# Check if sample has any rare species\n",
    "has_rare_species = sample_species.apply(lambda x: len(x & rare_species_set) > 0)\n",
    "\n",
    "# Check if sample is lacking any core species (missing at least one core species)\n",
    "lacking_core_species = sample_species.apply(lambda x: not core_species_set.issubset(x))\n",
    "\n",
    "# Total relative abundance of rare species per sample (sum of AbundR100 over rare species)\n",
    "rare_abund_by_sample = (\n",
    "    long_term_microbiome.loc[\n",
    "        long_term_microbiome[\"Species\"].isin(rare_species_set),\n",
    "        [\"sampID\", \"AbundR100\"],\n",
    "    ]\n",
    "    .assign(\n",
    "        AbundR100=lambda d: pd.to_numeric(d[\"AbundR100\"], errors=\"coerce\").fillna(\n",
    "            0.0\n",
    "        )  ##Junky column that just makes sure that its numeric\n",
    "    )\n",
    "    .groupby(\"sampID\", sort=False)[\"AbundR100\"]\n",
    "    .sum()\n",
    "    .rename(\"rare_species_total_abundR100\")\n",
    ")\n",
    "\n",
    "# Add columns to metadata using the sampID column (not the index)\n",
    "trimmed_metadata[\"has_rare_species\"] = (\n",
    "    trimmed_metadata[\"sampID\"].map(has_rare_species).fillna(False)\n",
    ")\n",
    "trimmed_metadata[\"lacking_core_species\"] = (\n",
    "    trimmed_metadata[\"sampID\"].map(lacking_core_species).fillna(True)\n",
    ")\n",
    "trimmed_metadata[\"rare_species_total_abundR100\"] = (\n",
    "    trimmed_metadata[\"sampID\"].map(rare_abund_by_sample).fillna(0.0)\n",
    ")\n",
    "\n",
    "trimmed_metadata[\n",
    "    [\n",
    "        \"sampID\",\n",
    "        \"has_rare_species\",\n",
    "        \"lacking_core_species\",\n",
    "        \"rare_species_total_abundR100\",\n",
    "    ]\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "03fbf5a8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Axes: xlabel='rare_species_total_abundR100', ylabel='Count'>"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAjsAAAGxCAYAAACEFXd4AAAAOnRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjEwLjYsIGh0dHBzOi8vbWF0cGxvdGxpYi5vcmcvq6yFwwAAAAlwSFlzAAAPYQAAD2EBqD+naQAAMlpJREFUeJzt3XlY1WX+//HXUeEICrhzIEXI3VwyNdOmsEzKss2+TY1pmuXXcinUaTEryasgnTLLdWwxm75mXd+0aZoWyQUrqkHUXELLQnEMhi8OASqCwv37o59nOoILcPAcbp6P6/pcl+f+fO7Ped/clC8/q8MYYwQAAGCpBr4uAAAAoDYRdgAAgNUIOwAAwGqEHQAAYDXCDgAAsBphBwAAWI2wAwAArEbYAQAAVmvk6wL8QXl5uX7++WeFhITI4XD4uhwAAHAOjDEqKipSZGSkGjQ4/fEbwo6kn3/+We3atfN1GQAAoBoOHDigtm3bnnY9YUdSSEiIpF9/WKGhoT6uBgAAnIvCwkK1a9fO/ff46RB2JPepq9DQUMIOAAB1zNkuQeECZQAAYDXCDgAAsBphBwAAWI2wAwAArEbYAQAAViPsAAAAqxF2AACA1Qg7AADAaoQdAABgNcIOAACwGmEHAABYjbADAACsRtgBAABWI+wAAACrNfJ1AbbLyspSXl5etfq2atVKUVFRXq4IAID6hbBTi7KystS1azcVFx+tVv+goGDt3p1B4AEAoAYIO7UoLy9PxcVHNWDcLIVGRFepb2H2Pn3z+tPKy8sj7AAAUAOEnfMgNCJaLaK6+LoMAADqJS5QBgAAViPsAAAAqxF2AACA1Qg7AADAaoQdAABgNcIOAACwmk/DzqZNm3TjjTcqMjJSDodD77//vsd6Y4wSEhIUGRmpoKAgDR48WLt27fLYpqSkRFOmTFGrVq3UpEkT3XTTTfrnP/95HkcBAAD8mU/DzpEjR9S7d28tXLiw0vVz587VvHnztHDhQqWlpcnlcmno0KEqKipybxMfH681a9Zo1apV+uKLL3T48GENHz5cZWVl52sYAADAj/n0oYLDhg3TsGHDKl1njNH8+fM1c+ZMjRgxQpK0YsUKhYeHa+XKlZowYYIKCgr02muv6S9/+YuuueYaSdJbb72ldu3a6bPPPtO111573sYCAAD8k99es5OZmamcnBzFxcW525xOp2JjY5WamipJSk9P1/Hjxz22iYyMVI8ePdzbVKakpESFhYUeCwAAsJPfhp2cnBxJUnh4uEd7eHi4e11OTo4CAwPVvHnz025TmaSkJIWFhbmXdu3aebl6AADgL/w27JzkcDg8PhtjKrSd6mzbzJgxQwUFBe7lwIEDXqkVAAD4H78NOy6XS5IqHKHJzc11H+1xuVwqLS1Vfn7+abepjNPpVGhoqMcCAADs5LdhJyYmRi6XS8nJye620tJSpaSkaNCgQZKkvn37KiAgwGOb7Oxs7dy5070NAACo33x6N9bhw4e1d+9e9+fMzExt27ZNLVq0UFRUlOLj45WYmKhOnTqpU6dOSkxMVHBwsEaOHClJCgsL07333qvp06erZcuWatGihf74xz+qZ8+e7ruzAABA/ebTsLN582ZdddVV7s/Tpk2TJI0ZM0ZvvPGGHnnkERUXF2vixInKz8/XgAEDtHbtWoWEhLj7vPjii2rUqJF+//vfq7i4WEOGDNEbb7yhhg0bnvfxAAAA/+PTsDN48GAZY0673uFwKCEhQQkJCafdpnHjxlqwYIEWLFhQCxUCAIC6zm+v2QEAAPAGwg4AALAaYQcAAFiNsAMAAKxG2AEAAFYj7AAAAKsRdgAAgNUIOwAAwGqEHQAAYDXCDgAAsBphBwAAWI2wAwAArEbYAQAAViPsAAAAqxF2AACA1Qg7AADAaoQdAABgNcIOAACwGmEHAABYjbADAACsRtgBAABWI+wAAACrEXYAAIDVCDsAAMBqhB0AAGA1wg4AALAaYQcAAFiNsAMAAKxG2AEAAFYj7AAAAKsRdgAAgNUIOwAAwGqEHQAAYDXCDgAAsBphBwAAWI2wAwAArEbYAQAAViPsAAAAqxF2AACA1Qg7AADAaoQdAABgNcIOAACwGmEHAABYjbADAACsRtgBAABWI+wAAACrEXYAAIDVCDsAAMBqhB0AAGA1wg4AALAaYQcAAFiNsAMAAKxG2AEAAFYj7AAAAKsRdgAAgNUIOwAAwGp+HXZOnDihJ554QjExMQoKCtKFF16o2bNnq7y83L2NMUYJCQmKjIxUUFCQBg8erF27dvmwagAA4E/8OuzMmTNHS5cu1cKFC5WRkaG5c+fqT3/6kxYsWODeZu7cuZo3b54WLlyotLQ0uVwuDR06VEVFRT6sHAAA+Au/DjtfffWVbr75Zt1www2Kjo7Wf/3XfykuLk6bN2+W9OtRnfnz52vmzJkaMWKEevTooRUrVujo0aNauXKlj6sHAAD+wK/Dzu9+9zutW7dO33//vSTp22+/1RdffKHrr79ekpSZmamcnBzFxcW5+zidTsXGxio1NdUnNQMAAP/SyNcFnMmjjz6qgoICde3aVQ0bNlRZWZmeffZZ/eEPf5Ak5eTkSJLCw8M9+oWHh2v//v2n3W9JSYlKSkrcnwsLC2uhegAA4A/8+sjOO++8o7feeksrV67Uli1btGLFCj3//PNasWKFx3YOh8PjszGmQttvJSUlKSwszL20a9euVuoHAAC+59dh5+GHH9Zjjz2mO++8Uz179tTo0aM1depUJSUlSZJcLpek/xzhOSk3N7fC0Z7fmjFjhgoKCtzLgQMHam8QAADAp/z6NNbRo0fVoIFnHmvYsKH71vOYmBi5XC4lJyerT58+kqTS0lKlpKRozpw5p92v0+mU0+msvcK9KCMjo1r9WrVqpaioKC9XAwBA3ePXYefGG2/Us88+q6ioKF100UXaunWr5s2bp3Hjxkn69fRVfHy8EhMT1alTJ3Xq1EmJiYkKDg7WyJEjfVx9zRQXHJLk0KhRo6rVPygoWLt3ZxB4AAD1nl+HnQULFujJJ5/UxIkTlZubq8jISE2YMEFPPfWUe5tHHnlExcXFmjhxovLz8zVgwACtXbtWISEhPqy85o4fLZJkdPHIR9U6pmuV+hZm79M3rz+tvLw8wg4AoN7z67ATEhKi+fPna/78+afdxuFwKCEhQQkJCeetrvOpaZsotYjq4usyAACos/z6AmUAAICaIuwAAACrEXYAAIDVCDsAAMBqhB0AAGA1wg4AALAaYQcAAFiNsAMAAKxG2AEAAFYj7AAAAKsRdgAAgNUIOwAAwGqEHQAAYDXCDgAAsBphBwAAWI2wAwAArEbYAQAAViPsAAAAqxF2AACA1Qg7AADAaoQdAABgNcIOAACwGmEHAABYjbADAACsRtgBAABWI+wAAACrEXYAAIDVCDsAAMBqhB0AAGA1wg4AALAaYQcAAFiNsAMAAKxG2AEAAFYj7AAAAKsRdgAAgNUIOwAAwGqEHQAAYDXCDgAAsBphBwAAWI2wAwAArEbYAQAAViPsAAAAqxF2AACA1Qg7AADAaoQdAABgNcIOAACwGmEHAABYjbADAACsRtgBAABWI+wAAACrEXYAAIDVCDsAAMBqhB0AAGA1wg4AALAaYQcAAFiNsAMAAKxG2AEAAFbz+7Bz8OBBjRo1Si1btlRwcLAuvvhipaenu9cbY5SQkKDIyEgFBQVp8ODB2rVrlw8rBgAA/sSvw05+fr4uv/xyBQQE6OOPP9Z3332nF154Qc2aNXNvM3fuXM2bN08LFy5UWlqaXC6Xhg4dqqKiIt8VDgAA/EYjXxdwJnPmzFG7du20fPlyd1t0dLT7z8YYzZ8/XzNnztSIESMkSStWrFB4eLhWrlypCRMmnO+SAQCAn/HrIzsffPCB+vXrp9tvv11t2rRRnz599Morr7jXZ2ZmKicnR3Fxce42p9Op2NhYpaam+qJkAADgZ6oVdi688EIdOnSoQvsvv/yiCy+8sMZFnfTTTz9pyZIl6tSpkz799FPdf//9evDBB/Xmm29KknJyciRJ4eHhHv3Cw8Pd6ypTUlKiwsJCjwUAANipWqex9u3bp7KysgrtJSUlOnjwYI2LOqm8vFz9+vVTYmKiJKlPnz7atWuXlixZorvvvtu9ncPh8OhnjKnQ9ltJSUl6+umnvVYnAADwX1UKOx988IH7z59++qnCwsLcn8vKyrRu3TqPa2pqKiIiQt27d/do69atm9577z1JksvlkvTrEZ6IiAj3Nrm5uRWO9vzWjBkzNG3aNPfnwsJCtWvXzmt1AwAA/1GlsHPLLbdI+vVIypgxYzzWBQQEKDo6Wi+88ILXirv88su1Z88ej7bvv/9e7du3lyTFxMTI5XIpOTlZffr0kSSVlpYqJSVFc+bMOe1+nU6nnE6n1+oEAAD+q0php7y8XNKvISMtLU2tWrWqlaJOmjp1qgYNGqTExET9/ve/1z/+8Q8tW7ZMy5Ytk/Rr6IqPj1diYqI6deqkTp06KTExUcHBwRo5cmSt1gYAAOqGal2zk5mZ6e06KtW/f3+tWbNGM2bM0OzZsxUTE6P58+frrrvucm/zyCOPqLi4WBMnTlR+fr4GDBigtWvXKiQk5LzUCAAA/Fu1n7Ozbt06rVu3Trm5ue4jPie9/vrrNS7spOHDh2v48OGnXe9wOJSQkKCEhASvfScAALBHtcLO008/rdmzZ6tfv36KiIg4451PAAAAvlStsLN06VK98cYbGj16tLfrAQAA8KpqPVSwtLRUgwYN8nYtAAAAXletsHPfffdp5cqV3q4FAADA66p1GuvYsWNatmyZPvvsM/Xq1UsBAQEe6+fNm+eV4gAAAGqqWmFn+/btuvjiiyVJO3fu9FjHxcoAAMCfVCvsbNiwwdt1AAAA1IpqXbMDAABQV1TryM5VV111xtNV69evr3ZBAAAA3lStsHPyep2Tjh8/rm3btmnnzp0VXhAKAADgS9UKOy+++GKl7QkJCTp8+HCNCgIAAPAmr16zM2rUKK++FwsAAKCmvBp2vvrqKzVu3NibuwQAAKiRap3GGjFihMdnY4yys7O1efNmPfnkk14pDAAAwBuqFXbCwsI8Pjdo0EBdunTR7NmzFRcX55XCAAAAvKFaYWf58uXergMAAKBWVCvsnJSenq6MjAw5HA51795dffr08VZdAAAAXlGtsJObm6s777xTGzduVLNmzWSMUUFBga666iqtWrVKrVu39nadAAAA1VKtu7GmTJmiwsJC7dq1S//+97+Vn5+vnTt3qrCwUA8++KC3awQAAKi2ah3Z+eSTT/TZZ5+pW7du7rbu3btr0aJFXKAMAAD8SrWO7JSXlysgIKBCe0BAgMrLy2tcFAAAgLdUK+xcffXVeuihh/Tzzz+72w4ePKipU6dqyJAhXisOAACgpqoVdhYuXKiioiJFR0erQ4cO6tixo2JiYlRUVKQFCxZ4u0YAAIBqq9Y1O+3atdOWLVuUnJys3bt3yxij7t2765prrvF2fQAAADVSpSM769evV/fu3VVYWChJGjp0qKZMmaIHH3xQ/fv310UXXaTPP/+8VgoFAACojiqFnfnz52v8+PEKDQ2tsC4sLEwTJkzQvHnzvFYcAABATVUp7Hz77be67rrrTrs+Li5O6enpNS4KAADAW6oUdv71r39Vesv5SY0aNdL//d//1bgoAAAAb6lS2Lngggu0Y8eO067fvn27IiIialwUAACAt1Qp7Fx//fV66qmndOzYsQrriouLNWvWLA0fPtxrxQEAANRUlW49f+KJJ7R69Wp17txZkydPVpcuXeRwOJSRkaFFixaprKxMM2fOrK1aAQAAqqxKYSc8PFypqal64IEHNGPGDBljJEkOh0PXXnutFi9erPDw8FopFAAAoDqq/FDB9u3b66OPPlJ+fr727t0rY4w6deqk5s2b10Z9AAAANVKtJyhLUvPmzdW/f39v1gIAAOB11Xo3FgAAQF1B2AEAAFYj7AAAAKsRdgAAgNUIOwAAwGqEHQAAYDXCDgAAsBphBwAAWI2wAwAArEbYAQAAViPsAAAAqxF2AACA1Qg7AADAaoQdAABgNcIOAACwGmEHAABYjbADAACsRtgBAABWI+wAAACrEXYAAIDVCDsAAMBqhB0AAGA1wg4AALBanQo7SUlJcjgcio+Pd7cZY5SQkKDIyEgFBQVp8ODB2rVrl++KBAAAfqXOhJ20tDQtW7ZMvXr18mifO3eu5s2bp4ULFyotLU0ul0tDhw5VUVGRjyoFAAD+pE6EncOHD+uuu+7SK6+8oubNm7vbjTGaP3++Zs6cqREjRqhHjx5asWKFjh49qpUrV/qwYgAA4C/qRNiZNGmSbrjhBl1zzTUe7ZmZmcrJyVFcXJy7zel0KjY2VqmpqafdX0lJiQoLCz0WAABgp0a+LuBsVq1apS1btigtLa3CupycHElSeHi4R3t4eLj2799/2n0mJSXp6aef9m6hAADAL/n1kZ0DBw7ooYce0ltvvaXGjRufdjuHw+Hx2RhToe23ZsyYoYKCAvdy4MABr9UMAAD8i18f2UlPT1dubq769u3rbisrK9OmTZu0cOFC7dmzR9KvR3giIiLc2+Tm5lY42vNbTqdTTqez9goHAAB+w6/DzpAhQ7Rjxw6PtnvuuUddu3bVo48+qgsvvFAul0vJycnq06ePJKm0tFQpKSmaM2eOL0r2KxkZGVXu06pVK0VFRdVCNQAA+IZfh52QkBD16NHDo61JkyZq2bKluz0+Pl6JiYnq1KmTOnXqpMTERAUHB2vkyJG+KNkvFBcckuTQqFGjqtw3KChYu3dnEHgAANbw67BzLh555BEVFxdr4sSJys/P14ABA7R27VqFhIT4ujSfOX60SJLRxSMfVeuYrufcrzB7n755/Wnl5eURdgAA1qhzYWfjxo0enx0OhxISEpSQkOCTevxZ0zZRahHVxddlAADgU359NxYAAEBNEXYAAIDV6txpLNS+6tzFJXEnFwDAPxF24FaTu7gk7uQCAPgnwg7cqnsXl8SdXAAA/0XYQQXcxQUAsAkXKAMAAKsRdgAAgNUIOwAAwGqEHQAAYDXCDgAAsBphBwAAWI2wAwAArEbYAQAAViPsAAAAqxF2AACA1Qg7AADAaoQdAABgNcIOAACwGmEHAABYjbADAACsRtgBAABWI+wAAACrEXYAAIDVCDsAAMBqhB0AAGA1wg4AALAaYQcAAFiNsAMAAKxG2AEAAFYj7AAAAKsRdgAAgNUIOwAAwGqNfF0A7JKRkVHlPq1atVJUVFQtVAMAAGEHXlJccEiSQ6NGjapy36CgYO3enUHgAQDUCsIOvOL40SJJRhePfFStY7qec7/C7H365vWnlZeXR9gBANQKwg68qmmbKLWI6uLrMgAAcOMCZQAAYDXCDgAAsBphBwAAWI2wAwAArEbYAQAAViPsAAAAqxF2AACA1Qg7AADAaoQdAABgNcIOAACwGmEHAABYjbADAACsRtgBAABWI+wAAACrEXYAAIDVCDsAAMBqhB0AAGA1wg4AALAaYQcAAFjNr8NOUlKS+vfvr5CQELVp00a33HKL9uzZ47GNMUYJCQmKjIxUUFCQBg8erF27dvmoYgAA4G/8OuykpKRo0qRJ+vrrr5WcnKwTJ04oLi5OR44ccW8zd+5czZs3TwsXLlRaWppcLpeGDh2qoqIiH1YOAAD8RSNfF3Amn3zyicfn5cuXq02bNkpPT9eVV14pY4zmz5+vmTNnasSIEZKkFStWKDw8XCtXrtSECRN8UTYAAPAjfn1k51QFBQWSpBYtWkiSMjMzlZOTo7i4OPc2TqdTsbGxSk1N9UmNAADAv/j1kZ3fMsZo2rRp+t3vfqcePXpIknJyciRJ4eHhHtuGh4dr//79p91XSUmJSkpK3J8LCwtroWIAAOAP6syRncmTJ2v79u16++23K6xzOBwen40xFdp+KykpSWFhYe6lXbt2Xq8XAAD4hzoRdqZMmaIPPvhAGzZsUNu2bd3tLpdL0n+O8JyUm5tb4WjPb82YMUMFBQXu5cCBA7VTOAAA8Dm/DjvGGE2ePFmrV6/W+vXrFRMT47E+JiZGLpdLycnJ7rbS0lKlpKRo0KBBp92v0+lUaGioxwIAAOzk19fsTJo0SStXrtRf//pXhYSEuI/ghIWFKSgoSA6HQ/Hx8UpMTFSnTp3UqVMnJSYmKjg4WCNHjvRx9QAAwB/4ddhZsmSJJGnw4MEe7cuXL9fYsWMlSY888oiKi4s1ceJE5efna8CAAVq7dq1CQkLOc7UAAMAf+XXYMcacdRuHw6GEhAQlJCTUfkEAAKDO8etrdgAAAGqKsAMAAKxG2AEAAFYj7AAAAKsRdgAAgNUIOwAAwGqEHQAAYDXCDgAAsBphBwAAWI2wAwAArObXr4sAziYrK0t5eXlV7teqVStFRUXVQkUAAH9D2EGdlZWVpa5du6m4+GiV+wYFBWv37gwCDwDUA4Qd1Fl5eXkqLj6qAeNmKTQi+pz7FWbv0zevP628vDzCDgDUA4Qd+IWMjIxq9wmNiFaLqC7eLgkAYAnCDnyquOCQJIdGjRpV7X0cLyn1XkEAAOsQduBTx48WSTK6eOSjah3TtUp9s3d8pZ0fLNOJEydqpzgAgBUIO/ALTdtEVflUVGH2vtopBgBgFZ6zAwAArEbYAQAAViPsAAAAqxF2AACA1Qg7AADAaoQdAABgNcIOAACwGmEHAABYjbADAACsRtgBAABW43URQB2QlZWlvLy8Kvdr1aqVoqKiaqEiAKg7CDuAn8vKylLXrt1UXHy0yn2DgoK1e3cGgQdAvUbYAfxcXl6eiouPasC4WQqNiD7nfoXZ+/TN608rLy+PsAOgXiPsAHVEaER0ld8MDwDgAmUAAGA5wg4AALAaYQcAAFiNsAMAAKxG2AEAAFYj7AAAAKsRdgAAgNUIOwAAwGqEHQAAYDWeoIx6KyMjo1r9SkpK5HQ6q9yPl3ICgG8QdlDvFBcckuTQqFGjqrcDh0MypsrdeCknAPgGYQf1zvGjRZKMLh75qFrHdK1S3+wdX2nnB8uq3JeXcgKA7xB2UG81bRNV5RdrFmbvq3ZfAIBvEHaA86g61wlV99oiAMCvCDvAeVDj64QkHS8p9V5BAFCPEHaA88Ab1wmdOHGidooDAMsRdoDzqCbXCVXX+b7FviZ9uT0fQG0g7ACW8tUt9jXpy+35AGoDYQewlC9usa9JX27PB1BbCDuA5c73Lfa+uj0/KytLeXl5Ve7ni1Nn1a1V4lQfUB2EHQB1XlZWlrp27abi4qNV7nu+T53VpFaJU31AdRB2ANR5eXl5Ki4+qgHjZik0Ivqc+/ni1Fl1a5U41QdUF2EHgDVCI6LrzJOt61KtQF1H2AHgV+rSU6arc+0NT8QGzj/CDgC/UNeeMl3Ta294IjZw/hB2APiFuvaU6epee8MTsYHzz5qws3jxYv3pT39Sdna2LrroIs2fP19XXHGFr8sCUEW+eMp0TVT12htf1lpddem2fl+p7s+oJk8qr0uPTfD174IVYeedd95RfHy8Fi9erMsvv1x//vOfNWzYMH333Xf15j80AKgNdem2fl+p0SnNGjypvC49NsHXvwtWhJ158+bp3nvv1X333SdJmj9/vj799FMtWbJESUlJPq4OAOquunRbv6/U9JRmdU7d1qXHJvjD70KdDzulpaVKT0/XY4895tEeFxen1NRUH1UFAHbhVvmzq+4pzfP9tPGaqou/C3U+7OTl5amsrEzh4eEe7eHh4crJyam0T0lJiUpKStyfCwoKJEmFhYVere3w4cOSpH/v36MTJcVV6luYvf/X2g7+oIBGjvPSl+/kO33Z1yffmZMlSUpPT3f/93qu9uzZI6nq/33XaJw1qLdBgwYqLy+vUh+pBuOsQa1S9eutSd/z/jOqZ78Lhw8f9vrfsyf3Z852KtDUcQcPHjSSTGpqqkf7M888Y7p06VJpn1mzZhlJLCwsLCwsLBYsBw4cOGNWqPNHdlq1aqWGDRtWOIqTm5tb4WjPSTNmzNC0adPcn8vLy/Xvf/9bLVu2lMNRtXR9JoWFhWrXrp0OHDig0NBQr+3XX9Wn8dansUr1a7z1aaxS/RpvfRqrVD/Ga4xRUVGRIiMjz7hdnQ87gYGB6tu3r5KTk3Xrrbe625OTk3XzzTdX2sfpdFa41a9Zs2a1VmNoaKi1v2iVqU/jrU9jlerXeOvTWKX6Nd76NFbJ/vGGhYWddZs6H3Ykadq0aRo9erT69eungQMHatmyZcrKytL999/v69IAAICPWRF27rjjDh06dEizZ89Wdna2evTooY8++kjt27f3dWkAAMDHrAg7kjRx4kRNnDjR12V4cDqdmjVrVrWfjlnX1Kfx1qexSvVrvPVprFL9Gm99GqtU/8Z7Jg5jqvnoRgAAgDqgga8LAAAAqE2EHQAAYDXCDgAAsBphpxYtXrxYMTExaty4sfr27avPP//c1yXVioSEBDkcDo/F5XL5uiyv2LRpk2688UZFRkbK4XDo/fff91hvjFFCQoIiIyMVFBSkwYMHa9euXb4p1gvONt6xY8dWmOvLLrvMN8XWUFJSkvr376+QkBC1adNGt9xyi/tx+CfZMr/nMlab5nbJkiXq1auX+/kyAwcO1Mcff+xeb8u8Smcfq03zWhOEnVryzjvvKD4+XjNnztTWrVt1xRVXaNiwYcrKyvJ1abXioosuUnZ2tnvZsWOHr0vyiiNHjqh3795auHBhpevnzp2refPmaeHChUpLS5PL5dLQoUNVVFR0niv1jrONV5Kuu+46j7n+6KOPzmOF3pOSkqJJkybp66+/VnJysk6cOKG4uDgdOXLEvY0t83suY5Xsmdu2bdvqueee0+bNm7V582ZdffXVuvnmm92BxpZ5lc4+Vsmeea0RL7yeCpW49NJLzf333+/R1rVrV/PYY4/5qKLaM2vWLNO7d29fl1HrJJk1a9a4P5eXlxuXy2Wee+45d9uxY8dMWFiYWbp0qQ8q9K5Tx2uMMWPGjDE333yzT+qpbbm5uUaSSUlJMcbYPb+njtUYu+fWGGOaN29uXn31Vavn9aSTYzXG/nk9VxzZqQWlpaVKT09XXFycR3tcXJxSU1N9VFXt+uGHHxQZGamYmBjdeeed+umnn3xdUq3LzMxUTk6Oxzw7nU7FxsZaO8+StHHjRrVp00adO3fW+PHjlZub6+uSvKKgoECS1KJFC0l2z++pYz3JxrktKyvTqlWrdOTIEQ0cONDqeT11rCfZOK9VZc1DBf1JXl6eysrKKryINDw8vMILS20wYMAAvfnmm+rcubP+9a9/6ZlnntGgQYO0a9cutWzZ0tfl1ZqTc1nZPO/fv98XJdW6YcOG6fbbb1f79u2VmZmpJ598UldffbXS09Pr9IPLjDGaNm2afve736lHjx6S7J3fysYq2Te3O3bs0MCBA3Xs2DE1bdpUa9asUffu3d2BxqZ5Pd1YJfvmtboIO7Xo1DeoG2O8+lZ1fzFs2DD3n3v27KmBAweqQ4cOWrFihcfb5W1VX+ZZ+vXVLCf16NFD/fr1U/v27fX3v/9dI0aM8GFlNTN58mRt375dX3zxRYV1ts3v6cZq29x26dJF27Zt0y+//KL33ntPY8aMUUpKinu9TfN6urF2797dunmtLk5j1YJWrVqpYcOGFY7i5ObmVvjXhI2aNGminj176ocffvB1KbXq5B1n9XWeJSkiIkLt27ev03M9ZcoUffDBB9qwYYPatm3rbrdxfk831srU9bkNDAxUx44d1a9fPyUlJal379566aWXrJzX0421MnV9XquLsFMLAgMD1bdvXyUnJ3u0Jycna9CgQT6q6vwpKSlRRkaGIiIifF1KrYqJiZHL5fKY59LSUqWkpNSLeZakQ4cO6cCBA3Vyro0xmjx5slavXq3169crJibGY71N83u2sVamLs9tZYwxKikpsWpeT+fkWCtj27yeM19dGW27VatWmYCAAPPaa6+Z7777zsTHx5smTZqYffv2+bo0r5s+fbrZuHGj+emnn8zXX39thg8fbkJCQqwYa1FRkdm6davZunWrkWTmzZtntm7davbv32+MMea5554zYWFhZvXq1WbHjh3mD3/4g4mIiDCFhYU+rrx6zjTeoqIiM336dJOammoyMzPNhg0bzMCBA80FF1xQJ8f7wAMPmLCwMLNx40aTnZ3tXo4ePerexpb5PdtYbZvbGTNmmE2bNpnMzEyzfft28/jjj5sGDRqYtWvXGmPsmVdjzjxW2+a1Jgg7tWjRokWmffv2JjAw0FxyySUet3na5I477jAREREmICDAREZGmhEjRphdu3b5uiyv2LBhg5FUYRkzZowx5tfbk2fNmmVcLpdxOp3myiuvNDt27PBt0TVwpvEePXrUxMXFmdatW5uAgAATFRVlxowZY7KysnxddrVUNk5JZvny5e5tbJnfs43VtrkdN26c+/+9rVu3NkOGDHEHHWPsmVdjzjxW2+a1JnjrOQAAsBrX7AAAAKsRdgAAgNUIOwAAwGqEHQAAYDXCDgAAsBphBwAAWI2wAwAArEbYAQAAViPsAPCKjRs3yuFw6JdffvF1KV43duxY3XLLLV7Z1759++RwOLRt2zav7K+qBg8erPj4eJ98N+ArhB0AXjFo0CBlZ2crLCys1r+run9h8xd9RYMHD5bD4ZDD4VBgYKA6dOigGTNmVHiR5LPPPqtBgwYpODhYzZo1q3RfWVlZuvHGG9WkSRO1atVKDz74oEpLSz222bFjh2JjYxUUFKQLLrhAs2fPFg/yR21r5OsCANuUlpYqMDCwSn3KysrkcDjUoEHd/fdHYGCgXC6Xr8tANYwfP16zZ89WaWmp0tLSdM8990iSkpKS3NuUlpbq9ttv18CBA/Xaa69V2EdZWZluuOEGtW7dWl988YUOHTqkMWPGyBijBQsWSJIKCws1dOhQXXXVVUpLS9P333+vsWPHqkmTJpo+ffr5GSzqJ9++mguo+2JjY82kSZPM1KlTTcuWLc2VV15pXnjhBdOjRw8THBxs2rZtax544AFTVFTk7rN8+XITFhZm/va3v5lu3bqZhg0bmp9++smUlJSYhx9+2ERGRprg4GBz6aWXmg0bNpxTHfv27TPDhw83zZo1M8HBwaZ79+7m73//uzHmPy/4/PDDD02vXr2M0+k0l156qdm+fbvHPr788ktzxRVXmMaNG5u2bduaKVOmmMOHD7vXHzt2zDz88MOmbdu2JjAw0HTs2NG8+uqrHt+Rn59/zvtbtGiR6dixo3E6naZNmzbmtttuO+s4x4wZU+GFlpmZmcYYYzZu3Gj69+9vAgMDjcvlMo8++qg5fvz4GfudOHHCjBs3zkRHR5vGjRubzp07m/nz51f4zptvvvmc5uHjjz82l19+uQkLCzMtWrQwN9xwg9m7d697fWZmppFk3n77bTNw4EDjdDpN9+7dPeb55O/Hb61Zs8b89n/Zs2bNMr179zZvvvmmad++vQkNDTV33HGHx9usDx8+bEaPHm2aNGliXC6Xef75501sbKx56KGH3Nuc+tkYY0aMGGEuueSSSsdXWW3GGPPRRx+ZBg0amIMHD7rb3n77beN0Ok1BQYExxpjFixebsLAwc+zYMfc2SUlJJjIy0pSXl1f6fYA31N1/RgJ+ZMWKFWrUqJG+/PJL/fnPf1aDBg308ssva+fOnVqxYoXWr1+vRx55xKPP0aNHlZSUpFdffVW7du1SmzZtdM899+jLL7/UqlWrtH37dt1+++267rrr9MMPP5y1hkmTJqmkpESbNm3Sjh07NGfOHDVt2tRjm4cffljPP/+80tLS1KZNG9100006fvy4pF9PL1x77bUaMWKEtm/frnfeeUdffPGFJk+e7O5/9913a9WqVXr55ZeVkZGhpUuXVviOk862v82bN+vBBx/U7NmztWfPHn3yySe68sorzzrOl156SQMHDtT48eOVnZ2t7OxstWvXTgcPHtT111+v/v3769tvv9WSJUv02muv6Zlnnjljv/LycrVt21bvvvuuvvvuOz311FN6/PHH9e677561lsocOXJE06ZNU1pamtatW6cGDRro1ltvVXl5eYW5mD59urZu3apBgwbppptu0qFDh6r0XT/++KPef/99ffjhh/rwww+VkpKi5557zuM7NmzYoDVr1mjt2rXauHGj0tPTz7jPb7/9Vl9++aUCAgKqVMtXX32lHj16KDIy0t127bXXqqSkxP2dX331lWJjY+V0Oj22+fnnn7Vv374qfR9QJb5OW0BdFxsbay6++OIzbvPuu++ali1buj8vX77cSDLbtm1zt+3du9c4HA6PfxkbY8yQIUPMjBkzzlpHz549TUJCQqXrTh51WbVqlbvt0KFDJigoyLzzzjvGGGNGjx5t/vu//9uj3+eff24aNGhgiouLzZ49e4wkk5ycfMbvOHlk52z7e++990xoaKjHkYhzVdnRiMcff9x06dLF4wjBokWLTNOmTU1ZWdlp+1Vm4sSJHkeZqnJk51S5ublGktmxY4cx5j9Hdp577jn3NsePHzdt27Y1c+bMMcac+5Gd4OBgj5/fww8/bAYMGGCMMaaoqMgEBgZWOuenHtkJCAgwTZo0MYGBgUaSadCggfnf//3fSsdzuiM748ePN0OHDq3QHhgYaFauXGmMMWbo0KFm/PjxHusPHjxoJJnU1NRKvw/wBq7ZAbygX79+Hp83bNigxMREfffddyosLNSJEyd07NgxHTlyRE2aNJH06zUuvXr1cvfZsmWLjDHq3Lmzx75KSkrUsmXLs9bw4IMP6oEHHtDatWt1zTXX6LbbbvPYvyQNHDjQ/ecWLVqoS5cuysjIkCSlp6dr7969+p//+R/3NsYYlZeXKzMzUzt27FDDhg0VGxt7Tj+Ts+1v6NChat++vS688EJdd911uu6663TrrbcqODj4nPZ/qoyMDA0cOFAOh8Pddvnll+vw4cP65z//qaioqNP2Xbp0qV599VXt379fxcXFKi0t1cUXX1ytOn788Uc9+eST+vrrr5WXl+c+opOVlaUePXq4t/vtXDRq1Ej9+vVzz8W5io6OVkhIiPtzRESEcnNz3XWUlpZWOuenuuuuuzRz5kwVFhZqzpw5Cg0N1W233ValWiR5/OxPMsZ4tJ+6jfn/FydX1hfwFk5jAV5wMsBI0v79+3X99derR48eeu+995Senq5FixZJkvuUkSQFBQV5/A++vLxcDRs2VHp6urZt2+ZeMjIy9NJLL521hvvuu08//fSTRo8erR07dqhfv37uC0PP5GQN5eXlmjBhgsd3f/vtt/rhhx/UoUMHBQUFnfPP41z2FxISoi1btujtt99WRESEnnrqKfXu3bvat66f+pfqybbfjrEy7777rqZOnapx48Zp7dq12rZtm+65554KdxGdqxtvvFGHDh3SK6+8om+++UbffPONJJ3T/k7W2aBBgwp3KP32d+ekU081ORwOd7g6tf+ZhIWFqWPHjrrkkkv01ltvKSUlpdKLkM/E5XIpJyfHoy0/P1/Hjx9XeHj4abc5Gc5ObgPUBsIO4GWbN2/WiRMn9MILL+iyyy5T586d9fPPP5+1X58+fVRWVqbc3Fx17NjRYznXu5zatWun+++/X6tXr9b06dP1yiuveKz/+uuv3X/Oz8/X999/r65du0qSLrnkEu3atavCd3fs2FGBgYHq2bOnysvLlZKSck61nG1/0q9HNK655hrNnTtX27dv1759+7R+/fqz7jswMFBlZWUebd27d1dqaqrHX/KpqakKCQnRBRdccNp+n3/+uQYNGqSJEyeqT58+6tixo3788cdzGuOpDh06pIyMDD3xxBMaMmSIunXrpvz8/Eq3/e1cnDhxQunp6e65aN26tYqKinTkyBH3NlV9Lk/Hjh0VEBBQ6ZyfSUBAgB5//HE98cQTOnr06Dl/38CBA7Vz505lZ2e729auXSun06m+ffu6t9m0aZNH8Fu7dq0iIyMVHR19zt8FVBVhB/CyDh066MSJE1qwYIF++ukn/eUvf9HSpUvP2q9z58666667dPfdd2v16tXKzMxUWlqa5syZo48++uis/ePj4/Xpp58qMzNTW7Zs0fr169WtWzePbWbPnq1169Zp586dGjt2rFq1auV+WN6jjz6qr776SpMmTdK2bdv0ww8/6IMPPtCUKVMk/XrKZMyYMRo3bpzef/99ZWZmauPGjae9kPds+/vwww/18ssva9u2bdq/f7/efPNNlZeXV3qa5VTR0dH65ptvtG/fPvepookTJ+rAgQOaMmWKdu/erb/+9a+aNWuWpk2b5r6lv7J+HTt21ObNm/Xpp5/q+++/15NPPqm0tLSz1lCZ5s2bq2XLllq2bJn27t2r9evXa9q0aZVuu2jRIq1Zs0a7d+/WpEmTlJ+fr3HjxkmSBgwYoODgYD3++OPau3evVq5cqTfeeKNKtTRt2lT33nuvHn74YY85P5fHG4wcOVIOh0OLFy92t2VlZWnbtm3KyspSWVmZ+2jd4cOHJUlxcXHq3r27Ro8era1bt2rdunX64x//qPHjxys0NNS9X6fTqbFjx2rnzp1as2aNEhMTNW3aNE5joXb56mIhwBaVXfQ6b948ExERYYKCgsy1115r3nzzTY+Ld093kWdpaal56qmnTHR0tAkICDAul8vceuutFW4Rr8zkyZNNhw4djNPpNK1btzajR482eXl5xpj/XDz8t7/9zVx00UUmMDDQ9O/f3+MCaWOM+cc//mGGDh1qmjZtapo0aWJ69eplnn32Wff64uJiM3XqVBMREeG+9fz111/3+I7f3np+pv19/vnnJjY21jRv3twEBQWZXr16uS+WPps9e/aYyy67zAQFBZ3zreen63fs2DEzduxYExYWZpo1a2YeeOAB89hjj5nevXu7+1XlAuXk5GTTrVs343Q6Ta9evczGjRuNJLNmzRpjzH8uUF65cqUZMGCACQwMNN26dTPr1q3z2M+aNWtMx44dTePGjc3w4cPNsmXLKr31/LdefPFF0759e/fnoqIiM2rUKBMcHGzCw8PN3Llzz+nWc2OMefbZZ03r1q3dj0yo7NZ9SR63zO/fv9/ccMMNJigoyLRo0cJMnjzZ4zZzY4zZvn27ueKKK4zT6TQul8skJCRw2zlqncMYHl0J2G7jxo266qqrlJ+ff9qn3wKArTiNBQAArEbYAeqIYcOGqWnTppUuiYmJvi7Pa7Kysk47zqZNmyorK4vaAFQJp7GAOuLgwYMqLi6udF2LFi3UokWL81xR7Thx4sQZn6YbHR2tRo1884gwf64NwOkRdgAAgNU4jQUAAKxG2AEAAFYj7AAAAKsRdgAAgNUIOwAAwGqEHQAAYDXCDgAAsBphBwAAWO3/ARreebA3ntCVAAAAAElFTkSuQmCC",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sns.histplot(trimmed_metadata[\"rare_species_total_abundR100\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "55924da0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AT1G01010</th>\n",
       "      <th>AT1G03987</th>\n",
       "      <th>AT1G01030</th>\n",
       "      <th>AT1G03993</th>\n",
       "      <th>AT1G01046</th>\n",
       "      <th>AT1G03997</th>\n",
       "      <th>AT1G01070</th>\n",
       "      <th>AT1G04003</th>\n",
       "      <th>AT1G01110</th>\n",
       "      <th>AT1G01150</th>\n",
       "      <th>...</th>\n",
       "      <th>ArthCt114</th>\n",
       "      <th>ArthCt124</th>\n",
       "      <th>ArthCt112</th>\n",
       "      <th>ArthCp088</th>\n",
       "      <th>ArthCp086</th>\n",
       "      <th>ArthCt118</th>\n",
       "      <th>ArthCp083</th>\n",
       "      <th>ArthCt111</th>\n",
       "      <th>ArthCp084</th>\n",
       "      <th>ArthCp085</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A2450525897_n01_LICRNA01_H06</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>34</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>10</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>29</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2450525897_n01_LICRNA01_F06</th>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>55</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>19</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2450525897_n01_LICRNA01_E01</th>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2450525897_n01_LICRNA01_H10</th>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2450525897_n01_LICRNA01_F10</th>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B2449500127_n01_LICRNA04_E07</th>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>...</td>\n",
       "      <td>21</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B2449500127_n01_LICRNA04_F10</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B2449500127_n01_LICRNA04_C01</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B2449500127_n01_LICRNA04_D12</th>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>16</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>33</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B2449500127_n01_LICRNA04_A03</th>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>373 rows × 28860 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              AT1G01010  AT1G03987  AT1G01030  AT1G03993  \\\n",
       "A2450525897_n01_LICRNA01_H06          0          0         34          0   \n",
       "A2450525897_n01_LICRNA01_F06         12          0         55          0   \n",
       "A2450525897_n01_LICRNA01_E01         19          0         22          0   \n",
       "A2450525897_n01_LICRNA01_H10          6          0          5          0   \n",
       "A2450525897_n01_LICRNA01_F10          6          0          6          0   \n",
       "...                                 ...        ...        ...        ...   \n",
       "B2449500127_n01_LICRNA04_E07          6          0          7          0   \n",
       "B2449500127_n01_LICRNA04_F10          0          0          0          0   \n",
       "B2449500127_n01_LICRNA04_C01          0          0         15          0   \n",
       "B2449500127_n01_LICRNA04_D12         15          0         11          0   \n",
       "B2449500127_n01_LICRNA04_A03          2          0          0          0   \n",
       "\n",
       "                              AT1G01046  AT1G03997  AT1G01070  AT1G04003  \\\n",
       "A2450525897_n01_LICRNA01_H06          0          0         10          0   \n",
       "A2450525897_n01_LICRNA01_F06          0          0         28          0   \n",
       "A2450525897_n01_LICRNA01_E01          0          0          3          0   \n",
       "A2450525897_n01_LICRNA01_H10          0          0         13          0   \n",
       "A2450525897_n01_LICRNA01_F10          0          0         12          0   \n",
       "...                                 ...        ...        ...        ...   \n",
       "B2449500127_n01_LICRNA04_E07          0          0          1          0   \n",
       "B2449500127_n01_LICRNA04_F10          0          0          0          0   \n",
       "B2449500127_n01_LICRNA04_C01          0          0          7          0   \n",
       "B2449500127_n01_LICRNA04_D12          0          0         15          0   \n",
       "B2449500127_n01_LICRNA04_A03          0          0          0          0   \n",
       "\n",
       "                              AT1G01110  AT1G01150  ...  ArthCt114  ArthCt124  \\\n",
       "A2450525897_n01_LICRNA01_H06          7          0  ...         10          5   \n",
       "A2450525897_n01_LICRNA01_F06         19          2  ...          0          3   \n",
       "A2450525897_n01_LICRNA01_E01          5          0  ...          0          0   \n",
       "A2450525897_n01_LICRNA01_H10          1          2  ...          2          0   \n",
       "A2450525897_n01_LICRNA01_F10          2          1  ...          6          0   \n",
       "...                                 ...        ...  ...        ...        ...   \n",
       "B2449500127_n01_LICRNA04_E07          7          8  ...         21          0   \n",
       "B2449500127_n01_LICRNA04_F10          0          0  ...          0          0   \n",
       "B2449500127_n01_LICRNA04_C01          0          0  ...          0          0   \n",
       "B2449500127_n01_LICRNA04_D12         10          0  ...         16          1   \n",
       "B2449500127_n01_LICRNA04_A03         11          0  ...          0          0   \n",
       "\n",
       "                              ArthCt112  ArthCp088  ArthCp086  ArthCt118  \\\n",
       "A2450525897_n01_LICRNA01_H06          1         29          0          0   \n",
       "A2450525897_n01_LICRNA01_F06          5         18          0          0   \n",
       "A2450525897_n01_LICRNA01_E01          0          0          0          0   \n",
       "A2450525897_n01_LICRNA01_H10          0          2          0          0   \n",
       "A2450525897_n01_LICRNA01_F10          0          9          0          0   \n",
       "...                                 ...        ...        ...        ...   \n",
       "B2449500127_n01_LICRNA04_E07          0         14          0          0   \n",
       "B2449500127_n01_LICRNA04_F10          0          3          0          0   \n",
       "B2449500127_n01_LICRNA04_C01          0          5          0          0   \n",
       "B2449500127_n01_LICRNA04_D12          0          4          0          0   \n",
       "B2449500127_n01_LICRNA04_A03          0          0          0          0   \n",
       "\n",
       "                              ArthCp083  ArthCt111  ArthCp084  ArthCp085  \n",
       "A2450525897_n01_LICRNA01_H06         30          0          0          6  \n",
       "A2450525897_n01_LICRNA01_F06          3          0          0         12  \n",
       "A2450525897_n01_LICRNA01_E01          3          0          0          5  \n",
       "A2450525897_n01_LICRNA01_H10          8          0          0          0  \n",
       "A2450525897_n01_LICRNA01_F10          4          0          0          1  \n",
       "...                                 ...        ...        ...        ...  \n",
       "B2449500127_n01_LICRNA04_E07          1          0          0          3  \n",
       "B2449500127_n01_LICRNA04_F10          0          0          0          0  \n",
       "B2449500127_n01_LICRNA04_C01          1          0          0         20  \n",
       "B2449500127_n01_LICRNA04_D12         33          0          0         25  \n",
       "B2449500127_n01_LICRNA04_A03          1          0          0          1  \n",
       "\n",
       "[373 rows x 28860 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "trimmed_raw_transcriptome.loc[:, trimmed_raw_transcriptome.sum() < 200_00]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "92d3bc5a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAeoAAAHqCAYAAADLbQ06AAAAOnRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjEwLjYsIGh0dHBzOi8vbWF0cGxvdGxpYi5vcmcvq6yFwwAAAAlwSFlzAAAPYQAAD2EBqD+naQAAPGdJREFUeJzt3Xl0FGX6/v+rISuYNGs2AiFCZDEBFSQEIvCRXSHMuADiRHAQFwSMgLh9hcyICIzGDbdRBBzAOAooM2oERlYJq0R2hCGgxARQQ9gJy/P7w196aLIA6YQ8mvfrnD6HfuquqrsqdK5Ud1WXwxhjBAAArFSlohsAAADFI6gBALAYQQ0AgMUIagAALEZQAwBgMYIaAACLEdQAAFiMoAYAwGIENQAAFiOoAQCwmHVBnZycLIfD4fYICQlxTTfGKDk5WWFhYfL391enTp20ZcsWt2WcOnVKw4cPV506dVS9enUlJCRo3759bjW5ublKTEyU0+mU0+lUYmKiDh06dCU2EQCAS2ZdUEvStddeq+zsbNdj06ZNrmmTJ09WSkqKpkyZorVr1yokJERdu3bVkSNHXDVJSUmaN2+eUlNTtWLFCh09elS9evXS2bNnXTUDBgxQRkaG0tLSlJaWpoyMDCUmJl7R7QQA4KKMZcaNG2datmxZ5LRz586ZkJAQM3HiRNfYyZMnjdPpNG+99ZYxxphDhw4Zb29vk5qa6qrJysoyVapUMWlpacYYY7Zu3WokmVWrVrlq0tPTjSSzffv2ctgqAABKx8oj6p07dyosLEyRkZHq37+/du/eLUnKzMxUTk6OunXr5qr19fVVx44dtXLlSknS+vXrdfr0abeasLAwRUdHu2rS09PldDoVGxvrqmnbtq2cTqerpigpKSkKDw93PXx8fFSjRg21bt26TLcfAIACXhXdwIViY2P1/vvv65prrtH+/fs1fvx4tWvXTlu2bFFOTo4kKTg42G2e4OBg7d27V5KUk5MjHx8f1axZs1BNwfw5OTkKCgoqtO6goCBXTVEOHz6srKwst7G8vLwS5wEAwBPWBXXPnj1d/46JiVFcXJwaNWqkGTNmqG3btpIkh8PhNo8xptDYhS6sKar+YssJDAxUvXr1XM+zs7N17tw5/XIsX/3/ni5JOrjrW/0lIVrx8fEl9gMAwKWw8q3v81WvXl0xMTHauXOn6+zvC49gDxw44DrKDgkJUX5+vnJzc0us2b9/f6F1HTx4sNDR+vlGjhypffv2uR6hoaEebRsAABdjfVCfOnVK27ZtU2hoqCIjIxUSEqKFCxe6pufn52vp0qVq166dJKlVq1by9vZ2q8nOztbmzZtdNXFxccrLy9OaNWtcNatXr1ZeXp6rBgAAG1j31vfo0aPVu3dvNWjQQAcOHND48eN1+PBhDRw4UA6HQ0lJSZowYYKioqIUFRWlCRMmqFq1ahowYIAkyel0avDgwRo1apRq166tWrVqafTo0YqJiVGXLl0kSc2aNVOPHj00ZMgQvf3225Kk+++/X7169VKTJk0qbNsBALiQdUG9b98+3XXXXfrpp59Ut25dtW3bVqtWrVJERIQkacyYMTpx4oSGDh2q3NxcxcbGasGCBQoICHAt46WXXpKXl5f69u2rEydOqHPnzpo+fbqqVq3qqpk1a5ZGjBjhOjs8ISFBU6ZMubIbCwDARTiMMaaim/itCg8PV1ZWlvxr1FXCpE8lcTIZAKBsWf8ZNQAAlRlBDQCAxQhqAAAsRlADAGAxghoAAIsR1AAAWIygBgDAYgQ1AAAWI6gBALAYQQ0AgMUIagAALEZQAwBgMYIaAACLEdQAAFiMoAYAwGIENQAAFiOoAQCwGEENAIDFCGoAACxGUAMAYDGCGgAAixHUAABYjKAGAMBiBDUAABYjqAEAsBhBDQCAxQhqAAAsRlADAGAxghoAAIsR1AAAWIygBgDAYgQ1AAAWI6gBALAYQQ0AgMUIagAALEZQAwBgMYIaAACLEdQAAFiMoAYAwGIENQAAFiOoAQCwGEENAIDFCGoAACxGUAMAYDGCGgAAixHUAABYjKAGAMBiBDUAABYjqAEAsBhBDQCAxQhqAAAsRlADAGAxghoAAIsR1AAAWIygBgDAYgQ1AAAWI6gBALAYQQ0AgMUIagAALEZQAwBgMYIaAACLEdQAAFiMoAYAwGIENQAAFiOoAQCwGEENAIDFCGoAACxGUAMAYDGCGgAAixHUAABYjKAGAMBiBDUAABazOqiff/55ORwOJSUlucaMMUpOTlZYWJj8/f3VqVMnbdmyxW2+U6dOafjw4apTp46qV6+uhIQE7du3z60mNzdXiYmJcjqdcjqdSkxM1KFDh67AVgEAcOmsDeq1a9fq73//u1q0aOE2PnnyZKWkpGjKlClau3atQkJC1LVrVx05csRVk5SUpHnz5ik1NVUrVqzQ0aNH1atXL509e9ZVM2DAAGVkZCgtLU1paWnKyMhQYmLiFds+AAAuhZVBffToUd1999165513VLNmTde4MUYvv/yynn76ad12222Kjo7WjBkzdPz4cc2ePVuSlJeXp6lTp+rFF19Uly5ddP3112vmzJnatGmTFi1aJEnatm2b0tLS9O677youLk5xcXF655139O9//1s7duyokG0GAKAoVgb1ww8/rFtvvVVdunRxG8/MzFROTo66devmGvP19VXHjh21cuVKSdL69et1+vRpt5qwsDBFR0e7atLT0+V0OhUbG+uqadu2rZxOp6sGAAAbeFV0AxdKTU3VN998o7Vr1xaalpOTI0kKDg52Gw8ODtbevXtdNT4+Pm5H4gU1BfPn5OQoKCio0PKDgoJcNUVJSUlRSkqK63l2dvYlbhUAAKVjVVD/8MMPeuSRR7RgwQL5+fkVW+dwONyeG2MKjV3owpqi6i+2nMOHDysrK6vE9QAAUJaseut7/fr1OnDggFq1aiUvLy95eXlp6dKlevXVV+Xl5eU6kr7wqPfAgQOuaSEhIcrPz1dubm6JNfv37y+0/oMHDxY6Wj9fYGCg6tWr53pUqWLV7gMA/A5ZlTSdO3fWpk2blJGR4Xq0bt1ad999tzIyMnT11VcrJCRECxcudM2Tn5+vpUuXql27dpKkVq1aydvb260mOztbmzdvdtXExcUpLy9Pa9ascdWsXr1aeXl5rpqijBw5Uvv27XM9QkNDy3oXAADgxqq3vgMCAhQdHe02Vr16ddWuXds1npSUpAkTJigqKkpRUVGaMGGCqlWrpgEDBkiSnE6nBg8erFGjRql27dqqVauWRo8erZiYGNfJac2aNVOPHj00ZMgQvf3225Kk+++/X7169VKTJk2u4BYDAFAyq4L6UowZM0YnTpzQ0KFDlZubq9jYWC1YsEABAQGumpdeekleXl7q27evTpw4oc6dO2v69OmqWrWqq2bWrFkaMWKE6+zwhIQETZky5YpvDwAAJXEYY0xFN/FbFR4erqysLPnXqKuESZ9Kkg7u+lZ/SYhWfHx8BXcHAPg9sOozagAA4I6gBgDAYgQ1AAAWI6gBALAYQQ0AgMUIagAALEZQAwBgMYIaAACLEdQAAFiMoAYAwGIENQAAFiOoAQCwGEENAIDFCGoAACxGUAMAYDGCGgAAixHUAABYjKAGAMBiBDUAABYjqAEAsBhBDQCAxQhqAAAsRlADAGAxghoAAIsR1AAAWIygBgDAYgQ1AAAWI6gBALAYQQ0AgMUIagAALEZQAwBgMYIaAACLEdQAAFiMoAYAwGIENQAAFiOoAQCwGEENAIDFCGoAACxGUAMAYDGCGgAAixHUAABYjKAGAMBiBDUAABYjqAEAsBhBDQCAxQhqAAAsRlADAGAxghoAAIsR1AAAWIygBgDAYgQ1AAAWI6gBALAYQQ0AgMUIagAALEZQAwBgMYIaAACLEdQAAFiMoAYAwGIENQAAFiOoAQCwGEENAIDFCGoAACxGUAMAYDGCGgAAixHUAABYjKAGAMBiBDUAABYjqAEAsBhBDQCAxQhqAAAsRlADAGAxghoAAIt5FNTXX3+93nzzTR0+fLis+tGbb76pFi1aKDAwUIGBgYqLi9MXX3zhmm6MUXJyssLCwuTv769OnTppy5Ytbss4deqUhg8frjp16qh69epKSEjQvn373Gpyc3OVmJgop9Mpp9OpxMREHTp0qMy2AwCAsuBRUG/btk3Dhg1TaGioBg0apBUrVnjcUHh4uCZOnKh169Zp3bp1uvnmm9WnTx9XGE+ePFkpKSmaMmWK1q5dq5CQEHXt2lVHjhxxLSMpKUnz5s1TamqqVqxYoaNHj6pXr146e/asq2bAgAHKyMhQWlqa0tLSlJGRocTERI/7BwCgTBkP5ObmmldeecW0aNHCOBwOU6VKFdO0aVPz4osvmoMHD3qyaDc1a9Y07777rjl37pwJCQkxEydOdE07efKkcTqd5q233jLGGHPo0CHj7e1tUlNTXTVZWVmmSpUqJi0tzRhjzNatW40ks2rVKldNenq6kWS2b99+yX3Vq1fPSDL+Neqafm+vNP3eXmlufuxNs3z5ck83GQAAY4wxHh1R16hRQyNGjNC3336rNWvWaMiQIcrOztbo0aMVHh6ufv36acGCBaVe/tmzZ5Wamqpjx44pLi5OmZmZysnJUbdu3Vw1vr6+6tixo1auXClJWr9+vU6fPu1WExYWpujoaFdNenq6nE6nYmNjXTVt27aV0+l01QAAYIMyO5msdevWeuutt5Sdna333ntPbdq00UcffaSePXsqMjJSzz33nLKzsy9pWZs2bdJVV10lX19fPfjgg5o3b56aN2+unJwcSVJwcLBbfXBwsGtaTk6OfHx8VLNmzRJrgoKCCq03KCjIVVOUlJQUhYeHux6Xuj0AAJRWmZ/17e/vr4SEBP3xj39UWFiYjDHau3evnnnmGTVs2FDDhg3T8ePHS1xGkyZNlJGRoVWrVumhhx7SwIEDtXXrVtd0h8PhVm+MKTR2oQtriqq/2HIOHz6srKws1+PcuXMlrhMAAE+VaVAvWrRI/fv3V7169TR69GidO3dOTz31lHbs2KHU1FTXWeLDhg0rcTk+Pj5q3LixWrdureeff14tW7bUK6+8opCQEEkqdNR74MAB11F2SEiI8vPzlZubW2LN/v37C6334MGDhY7WzxcYGKh69eq5HlWqcHUbAKB8eZw0P/74o8aPH69GjRqpe/fu+uijj9SxY0d99NFH+uGHHzR+/HhFRUWpb9++WrVqlW655RZ9+umnl7UOY4xOnTqlyMhIhYSEaOHCha5p+fn5Wrp0qdq1aydJatWqlby9vd1qsrOztXnzZldNXFyc8vLytGbNGlfN6tWrlZeX56opysiRI7Vv3z7XIzQ09LK2AwCAy+Xlycy9e/dWWlqazp49q+DgYD3xxBMaMmSIGjZsWOw87dq10+eff17s9Keeeko9e/ZU/fr1deTIEaWmpmrJkiVKS0uTw+FQUlKSJkyYoKioKEVFRWnChAmqVq2aBgwYIElyOp0aPHiwRo0apdq1a6tWrVoaPXq0YmJi1KVLF0lSs2bN1KNHDw0ZMkRvv/22JOn+++9Xr1691KRJE092CQAAZcqjoP7888/VpUsX3X///erTp4+8vC6+uN69eyssLKzY6fv371diYqKys7PldDrVokULpaWlqWvXrpKkMWPG6MSJExo6dKhyc3MVGxurBQsWKCAgwLWMl156SV5eXurbt69OnDihzp07a/r06apataqrZtasWRoxYoTr7PCEhARNmTKltLsCAIBy4TDGmNLOnJmZqcjIyLLs5zclPDxcWVlZ8q9RVwmTfn07/+Cub/WXhGjFx8dXcHcAgN8Djz6jrswhDQDAleBRUKekpKhOnTr68ccfi5z+448/qm7dunr11Vc9WQ0AAJWWR0H90UcfqUWLFsV+5hwWFqbrrrtOqampnqwGAIBKy6Og/u677xQdHV1izbXXXqudO3d6shoAACotj4L6+PHjql69eok1fn5+Onr0qCerAQCg0vIoqCMiIi56E4v09HSFh4d7shoAACotj4K6V69eWrFihd57770ip7/77rtasWKFevfu7clqAACotDz6wpPHH39cqampGjJkiGbOnKmuXbuqXr16ysrK0oIFC7Rs2TKFhYXpySefLKt+AQCoVDwK6rp162rx4sX605/+pCVLlmjJkiVyOBwq+A6VNm3aaObMmapbt26ZNAsAQGXjUVBLUlRUlFavXq1169ZpzZo1OnTokGrUqKE2bdqodevWZdEjAACVlsdBXaB169YEMwAAZYwbKgMAYDGPj6gPHjyoadOmae3atTp06JDOnj1bqMbhcOg///mPp6sCAKDS8SioN27cqJtvvlm5ubkq6SZcDofDk9UAAFBpefTW96hRo/TLL7/o6aefVmZmpk6fPq1z584VehR1lA0AAC7OoyPq9PR0/eEPf9Bf//rXsuoHAACcx6Mjah8fHzVq1KisegEAABfwKKhvvvlmrVu3rqx6AQAAF/AoqP/2t79py5YteuGFF8qqHwAAcB6PPqN+9tlnde211+rxxx/XW2+9pZYtW8rpdBaqczgcmjp1qierAgCgUvIoqKdPn+769+7du7V79+4i6whqAABKx6OgzszMLKs+AABAETwK6oiIiLLqAwAAFKFMv+v7l19+0Q8//FCWiwQAoFLzOKjz8vL0yCOPKDg4WHXr1lVkZKRr2urVq3XLLbdo/fr1nq4GAIBKyaOg/uWXXxQbG6vXXntN9evXV7Nmzdy+87tFixb6+uuvNWvWLI8bBQCgMvIoqJOTk/Xdd9/pgw8+0Lp163TnnXe6Tff391fHjh311VdfedQkAACVlUdBPX/+fPXq1Uv9+vUrtiYiIkL79u3zZDUAAFRaHgV1dna2mjdvXmKNn5+fjh075slqAACotDwK6tq1a1/0LO/t27crNDTUk9UAAFBpeRTUHTp00Pz585WVlVXk9K1btyotLU1dunTxZDUAAFRaHgX1008/rTNnzqh9+/aaPXu2fvrpJ0nStm3bNHXqVN18883y9fXVY489VibNAgBQ2Xj0zWQxMTH68MMPdc899ygxMVGSZIxRdHS0jDEKCAjQP//5T0VFRZVJswAAVDYeBbUkJSQkaPfu3ZoxY4ZWr16tX375RYGBgYqNjdW9996rOnXqlEWfAABUSh4HtSTVqlVLjz76aFksCgAAnKdMv+sbAACULY+OqN9///1Lrr3nnns8WRUAAJWSR0E9aNAgORyOEmuMMXI4HAQ1AACl4FFQT5s2rcjxvLw8ffPNN5o9e7YSEhLUu3dvT1YDAECl5VFQDxw4sMTpDzzwgDp37qyHHnrIk9UAAFBplevJZHFxcerdu7fGjh1bnqsBAOB3q9zP+o6IiNC3335b3qsBAOB3qVyD2hijZcuWyd/fvzxXAwDA75ZHn1EvW7asyPEzZ84oKytL77//vtauXev6elEAAHB5PArqTp06lXh5ljFGcXFxSklJ8WQ1AABUWh4F9dixY4sM6ipVqqhmzZpq3bq12rZt68kqAACo1DwK6uTk5DJqAwAAFIXv+gYAwGIeHVF///33pZ63QYMGnqwaAIBKwaOgbtiw4UW/67soDodDZ86c8WTVAABUCh4F9T333KPMzEwtX75cNWrU0HXXXafg4GDt379fGRkZOnTokDp06KDIyMiy6hcAgErFo6B+7LHH1L59ez311FN68sknVb16dde0Y8eO6bnnntObb76pN954Q82bN/e4WQAAKhuPTiYbM2aM2rRpo/Hjx7uFtCRVr15dEyZM0I033qjHH3/coyYBAKisPArqr7/+Wm3atCmx5sYbb9Ty5cs9WQ0AAJWWR0F97tw57dq1q8SanTt3yhjjyWoAAKi0PArqDh06aM6cOUpNTS1y+gcffKC5c+eqQ4cOnqwGAIBKy6OTySZPnqzly5fr7rvv1qRJkxQfH6+goCAdOHBAK1as0MaNGxUQEKBJkyaVVb8AAFQqHgV18+bN9fXXX2vYsGFatmxZoftOd+jQQa+//jpnfAMAUEoeBbUkRUdHa8mSJfrhhx/07bffKi8vT06nUy1btlT9+vXLokcAACotj4O6QP369QlmAADKWJkEdX5+vhYtWqTt27fr2LFjeuaZZyRJJ0+e1OHDh1WnTh1VqcL9PwAAuFwep+f8+fPVoEED9e7dW6NHj3a79eXGjRsVGhpa7FnhAACgZB5/4ckdd9whX19fvfLKKxowYIDb9DZt2qhx48aaM2eOR00CAFBZefTW9/jx41WjRg2tW7dOdevW1c8//1yoplWrVlqzZo0nqwEAoNLy6Ih61apV6tOnj+rWrVtsTf369ZWTk+PJagAAqLQ8CupTp07J6XSWWJOXl8eJZAAAlJJHCXr11Vdr3bp1Jdakp6eradOmnqwGAIBKy6Ogvv3227V8+XK9//77RU5/4YUXtHnzZvXr18+T1QAAUGl5dDLZY489pjlz5ujee+/VzJkzdfLkSUm/3qc6PT1dK1eu1HXXXadhw4aVSbMAAFQ2HgX1VVddpeXLl2vYsGH65z//qbNnz0r69Uja4XCob9++euONN+Tr61smzQIAUNl4/M1kNWvW1KxZs/Tqq69q7dq1+uWXXxQYGKgbb7xRwcHBZdEjAACVlkdBffPNNys+Pl5//etfVbt2bfXo0aOs+gIAAPLwZLLVq1frzJkzZdULAAC4gEdB3axZM+3Zs6eMWvnV888/rxtvvFEBAQEKCgrSH/7wB+3YscOtxhij5ORkhYWFyd/fX506ddKWLVvcak6dOqXhw4erTp06ql69uhISErRv3z63mtzcXCUmJsrpdMrpdCoxMVGHDh0q0+0BAMATHgX18OHDNX/+fG3durWs+tHSpUv18MMPa9WqVVq4cKHOnDmjbt266dixY66ayZMnKyUlRVOmTNHatWsVEhKirl276siRI66apKQkzZs3T6mpqVqxYoWOHj2qXr16uU54k6QBAwYoIyNDaWlpSktLU0ZGhhITE8tsWwAA8JTDGGNKO/OyZcs0efJkLVu2TA888IDrBDKHw1GotkOHDqVax8GDBxUUFKSlS5eqQ4cOMsYoLCxMSUlJevzxxyX9evQcHBysSZMm6YEHHlBeXp7q1q2rf/zjH65ruH/88UfVr19fn3/+ubp3765t27apefPmWrVqlWJjYyX9+pWocXFx2r59u5o0aXLR3sLDw5WVlSX/GnWVMOnTX/vd9a3+khCt+Pj4Um0vAADn8+hksk6dOsnhcMgYoxdffLHIgC5w/pHs5cjLy5Mk1apVS5KUmZmpnJwcdevWzVXj6+urjh07auXKlXrggQe0fv16nT592q0mLCxM0dHRWrlypbp376709HQ5nU5XSEtS27Zt5XQ6tXLlyiKDOiUlRSkpKa7n2dnZpdomAAAulUdBPXbs2BLD2VPGGI0cOVLx8fGKjo6WJNcNPi689Cs4OFh79+511fj4+KhmzZqFagrmz8nJUVBQUKF1BgUFFXsTkcOHDysrK8uzjQIA4DJcdlBXrVpVycnJeuaZZ5ScnCzp17O/V69erREjRpRpc8OGDdPGjRu1YsWKQtMu/APBGHPRPxourCmqvqTlBAYGql69eq7n2dnZOnfuXInrBADAE5d9MpkxRhd+rJ2WlqZHH320zJqS/nei2uLFixUeHu4aDwkJkaRCR70HDhxwHWWHhIQoPz9fubm5Jdbs37+/0HoPHjxY7Be1jBw5Uvv27XM9QkNDS7+BAABcAuvuP2mM0bBhwzR37lx99dVXioyMdJseGRmpkJAQLVy40DWWn5+vpUuXql27dpKkVq1aydvb260mOztbmzdvdtXExcUpLy9Pa9ascdWsXr1aeXl5rhoAACqax18hWtYefvhhzZ49W59++qkCAgJcR85Op1P+/v5yOBxKSkrShAkTFBUVpaioKE2YMEHVqlXTgAEDXLWDBw/WqFGjVLt2bdWqVUujR49WTEyMunTpIunXa8B79OihIUOG6O2335Yk3X///erVq9clnfENAMCVYF1Qv/nmm5J+PaP8fNOmTdOgQYMk/Xp3rhMnTmjo0KHKzc1VbGysFixYoICAAFf9Sy+9JC8vL/Xt21cnTpxQ586dNX36dFWtWtVVM2vWLI0YMcJ1dnhCQoKmTJlSvhsIAMBlsC6oL+WybofDoeTkZNfJbEXx8/PTa6+9ptdee63Ymlq1amnmzJmlaRMAgCuiVEE9c+ZMrVq1yvV8165dkqRbbrmlyHqHw6HPPvusNKsCAKBSK1VQ79q1yxXO50tLSyuyvjyvtQYA4PfssoM6MzOzPPoAAABFuOygjoiIKI8+AABAEay7jhoAAPwPQQ0AgMUIagAALEZQAwBgMYIaAACLEdQAAFiMoAYAwGIENQAAFiOoAQCwGEENAIDFCGoAACxGUAMAYDGCGgAAixHUAABYjKAGAMBiBDUAABYjqAEAsBhBDQCAxQhqAAAsRlADAGAxghoAAIsR1AAAWIygBgDAYgQ1AAAWI6gBALAYQQ0AgMUIagAALEZQAwBgMYIaAACLEdQAAFiMoAYAwGIENQAAFiOoAQCwGEENAIDFCGoAACxGUAMAYDGCGgAAixHUAABYjKAGAMBiBDUAABYjqAEAsBhBDQCAxQhqAAAsRlADAGAxghoAAIsR1AAAWIygBgDAYgQ1AAAWI6gBALAYQQ0AgMUIagAALEZQAwBgMYIaAACLEdQAAFiMoAYAwGIENQAAFiOoAQCwGEENAIDFvCq6gcro5MmTWrduXaHx1q1by8/PrwI6AgDYiqCuAOvWrdMjb3yiGuGNXWOH9u3SK0Ol+Pj4CuwMAGAbgrqC1AhvrLqNW1Z0GwAAy/EZNQAAFiOoAQCwGEENAIDFCGoAACxGUAMAYDGCGgAAixHUAABYzLqgXrZsmXr37q2wsDA5HA598sknbtONMUpOTlZYWJj8/f3VqVMnbdmyxa3m1KlTGj58uOrUqaPq1asrISFB+/btc6vJzc1VYmKinE6nnE6nEhMTdejQoXLeOgAALo91QX3s2DG1bNlSU6ZMKXL65MmTlZKSoilTpmjt2rUKCQlR165ddeTIEVdNUlKS5s2bp9TUVK1YsUJHjx5Vr169dPbsWVfNgAEDlJGRobS0NKWlpSkjI0OJiYnlvn0AAFwO676ZrGfPnurZs2eR04wxevnll/X000/rtttukyTNmDFDwcHBmj17th544AHl5eVp6tSp+sc//qEuXbpIkmbOnKn69etr0aJF6t69u7Zt26a0tDStWrVKsbGxkqR33nlHcXFx2rFjh5o0aXJlNhYAgIuw7oi6JJmZmcrJyVG3bt1cY76+vurYsaNWrlwpSVq/fr1Onz7tVhMWFqbo6GhXTXp6upxOpyukJalt27ZyOp2umqKkpKQoPDzc9cjOzi7rTQQAwI11R9QlycnJkSQFBwe7jQcHB2vv3r2uGh8fH9WsWbNQTcH8OTk5CgoKKrT8oKAgV01RDh8+rKysLI+2AQCAy/GbCuoCDofD7bkxptDYhS6sKar+YssJDAxUvXr1XM+zs7N17ty5S20bAIDL9pt66zskJESSCh31HjhwwHWUHRISovz8fOXm5pZYs3///kLLP3jwYKGj9fONHDlS+/btcz1CQ0M92h4AAC7mNxXUkZGRCgkJ0cKFC11j+fn5Wrp0qdq1aydJatWqlby9vd1qsrOztXnzZldNXFyc8vLytGbNGlfN6tWrlZeX56oBAMAG1r31ffToUe3atcv1PDMzUxkZGapVq5YaNGigpKQkTZgwQVFRUYqKitKECRNUrVo1DRgwQJLkdDo1ePBgjRo1SrVr11atWrU0evRoxcTEuM4Cb9asmXr06KEhQ4bo7bffliTdf//96tWrF2d8AwCsYl1Qr1u3Tv/3f//nej5y5EhJ0sCBAzV9+nSNGTNGJ06c0NChQ5Wbm6vY2FgtWLBAAQEBrnleeukleXl5qW/fvjpx4oQ6d+6s6dOnq2rVqq6aWbNmacSIEa6zwxMSEoq9dhsAgIpiXVB36tRJxphipzscDiUnJys5ObnYGj8/P7322mt67bXXiq2pVauWZs6c6UmrAACUu9/UZ9QAAFQ2BDUAABYjqAEAsBhBDQCAxQhqAAAsRlADAGAxghoAAIsR1AAAWIygBgDAYgQ1AAAWI6gBALAYQQ0AgMUIagAALEZQAwBgMYIaAACLEdQAAFiMoAYAwGIENQAAFiOoAQCwGEENAIDFCGoAACxGUAMAYDGCGgAAixHUAABYjKAGAMBiBDUAABYjqAEAsBhBDQCAxQhqAAAsRlADAGAxghoAAIsR1AAAWIygBgDAYgQ1AAAWI6gBALAYQQ0AgMW8KroBeO7kyZNat25dofHWrVvLz8+vAjoCAJQVgvp3YN26dXrkjU9UI7yxa+zQvl16ZagUHx9fgZ0BADxFUP9O1AhvrLqNW1Z0GwCAMsZn1AAAWIygBgDAYgQ1AAAWI6gBALAYJ5PBDZd6AYBdCGq44VIvALALQY1CuNQLAOzBZ9QAAFiMoAYAwGIENQAAFiOoAQCwGCeToVyV5eVeXDoGoDIiqFGuyvJyLy4dA1AZEdQod2V5uReXjgGobPiMGgAAi3FEjUqLz7wB/BYQ1Ki0yvIzb0IfQHkhqFGpldVn3pzoBqC8ENRAGSnLE904QgdQgKAGLMQROoACBDVgqbI6QufoHPhtI6iB37myPjon+IEri6AGKoGy/Pycs+WBK4ugBnDZOFseuHIIagAVirPlgZIR1AB+NzhCx+8RQQ3gd8XWG7fYfMtXm3sDQQ0ARSrrwLH5lq829waCGgCKVB6BY/MtX23tjSN0ghoAimXr2+iVCUfoBDUAwHKl+YPp93QkTlADAH53fk9H4gQ1AOB36ffy0UWVim6gor3xxhuKjIyUn5+fWrVqpeXLl1d0SwAAuFTqoP7www+VlJSkp59+Whs2bNBNN92knj176vvvv6/o1gAAkFTJgzolJUWDBw/Wfffdp2bNmunll19W/fr19eabb1Z0awAASKrEQZ2fn6/169erW7dubuPdunXTypUrK6grAADcOYwxpqKbqAg//vij6tWrp6+//lrt2rVzjU+YMEEzZszQjh07Cs2TkpKilJQU1/OsrKxf/+FwyLe6U5J07uwZVff1kre3d7HrPn36tI6dOqMqVf93Lt+lzHcllkdv9EZv9PZ76K24+Wpc5S8fH5/L7qMkISEhRV4KVlYq/VnfDofD7bkxptBYgcOHD/8vnN1n0qmjh1xPD50oXS+lne9KLI/eKn5ZZb08eqv4ZZX18ujt4g6eOFq2jUg6erTsl3m+ShvUderUUdWqVZWTk+M2fuDAAQUHBxc5T2BgoOrVq+d6fn5onz/+W5Gdna1z586pSpUqCg0Nreh2Lhv9Vyz6r1j0X/EKtuH48ePluyJTibVp08Y89NBDbmPNmjUzTzzxxCXNX69ePSPJ1KtXrzzaK3f0X7Hov2LRf8X6rfdvzJXbhkp7RC1JI0eOVGJiolq3bq24uDj9/e9/1/fff68HH3ywolsDAEBSJX7rW5L69eunn3/+WX/961+VnZ2t6Ohoff7554qIiKjo1gAAkFTJg1qShg4dqqFDh5Zq3pEjR+rw4cMKDAws466uDPqvWPRfsei/Yv3W+5eu3DZU2suzAAD4Lai0X3gCAMBvAUENAIDFCGoAACxGUF/E5d4Gc+nSpWrVqpX8/Px09dVX66233rpCnbp7/vnndeONNyogIEBBQUH6wx/+UOTXop5vyZIlcjgchR7bt2+/Ql3/T3JycqE+QkJCSpzHln0vSQ0bNixyXz788MNF1lf0vl+2bJl69+6tsLAwORwOffLJJ27TjTFKTk5WWFiY/P391alTJ23ZsuWiy50zZ46aN28uX19fNW/eXPPmzbvi/Z8+fVqPP/64YmJiVL16dYWFhemee+7Rjz/+WOIyp0+fXuTP5OTJk1e0f0kaNGhQoT7atm170eXasP8lFbkfHQ6H/va3vxW7zCu5/y/l92VFvgYI6hJc7m0wMzMzdcstt+imm27Shg0b9NRTT2nEiBGaM2fOFe7819B6+OGHtWrVKi1cuFBnzpxRt27ddOzYsYvOu2PHDmVnZ7seUVFRV6Djwq699lq3PjZt2lRsrU37XpLWrl3r1vvChQslSXfeeWeJ81XUvj927JhatmypKVOmFDl98uTJSklJ0ZQpU7R27VqFhISoa9euOnLkSLHLTE9PV79+/ZSYmKhvv/1WiYmJ6tu3r1avXn1F+z9+/Li++eYbPfPMM/rmm280d+5cfffdd0pISLjocgMDA91+HtnZ2fLz87ui/Rfo0aOHWx+ff/55icu0Zf9LKrQP33vvPTkcDt1+++0lLvdK7f9L+X1Zoa+Bcv06ld+4Nm3amAcffNBtrGnTpsV+c9mYMWNM06ZN3cYeeOAB07Zt23Lr8VIdOHDASDJLly4ttmbx4sVGksnNzb1yjRVj3LhxpmXLlpdcb/O+N8aYRx55xDRq1MicO3euyOk27XtJZt68ea7n586dMyEhIWbixImusZMnTxqn02neeuutYpfTt29f06NHD7ex7t27m/79+5d5z+e7sP+irFmzxkgye/fuLbZm2rRpxul0lm1zl6Co/gcOHGj69OlzWcuxef/36dPH3HzzzSXWVNT+N6bw78uKfg1wRF2M0twGMz09vVB99+7dtW7dOp0+fbrcer0UeXl5kqRatWpdtPb6669XaGioOnfurMWLF5d3a8XauXOnwsLCFBkZqf79+2v37t3F1tq87/Pz8zVz5kz9+c9/LvaGLwVs2ffny8zMVE5Ojtv+9fX1VceOHUu8JWxxPxMbbiObl5cnh8OhGjVqlFh39OhRRUREKDw8XL169dKGDRuuTINFWLJkiYKCgnTNNddoyJAhOnDgQIn1tu7//fv367PPPtPgwYMvWltR+//C35cV/RogqIvx008/6ezZs4Vu0BEcHFzoRh4FcnJyiqw/c+aMfvrpp3Lr9WKMMRo5cqTi4+MVHR1dbF1oaKj+/ve/a86cOZo7d66aNGmizp07a9myZVew21/Fxsbq/fff15dffql33nlHOTk5ateunX7++eci623d95L0ySef6NChQxo0aFCxNTbt+wsV/H+/nNdCwXyXO8+VcPLkST3xxBMaMGBAiV9U0bRpU02fPl3z58/XBx98ID8/P7Vv3147d+68gt3+qmfPnpo1a5a++uorvfjii1q7dq1uvvlmnTp1qth5bN3/M2bMUEBAgG677bYS6ypq/xf1+7KiXwOV/pvJLuZyboNZXH1R41fSsGHDtHHjRq1YsaLEuiZNmqhJkyau53Fxcfrhhx/0wgsvqEOHDuXdppuePXu6/h0TE6O4uDg1atRIM2bM0MiRI4ucx8Z9L0lTp05Vz549FRYWVmyNTfu+OJf7WijtPOXp9OnT6t+/v86dO6c33nijxNq2bdu6nbDVvn173XDDDXrttdf06quvlnerbvr16+f6d3R0tFq3bq2IiAh99tlnJQaebftfkt577z3dfffdF/2suaL2f0m/LyvqNcARdTFKcxvMkJCQIuu9vLxUu3btcuu1JMOHD9f8+fO1ePFihYeHX/b8bdu2rZAjiAtVr15dMTExxfZi476XpL1792rRokW67777LnteW/Z9wdn2l/NaKJjvcucpT6dPn1bfvn2VmZmphQsXXvbXPlapUkU33nijFT+T0NBQRURElNiLbftfkpYvX64dO3aU6vVwJfZ/cb8vK/o1QFAXw8fHR61atXKdrVtg4cKFateuXZHzxMXFFapfsGCBWrduLW9v73LrtSjGGA0bNkxz587VV199pcjIyFItZ8OGDVbcK/bUqVPatm1bsb3YtO/PN23aNAUFBenWW2+97Hlt2feRkZEKCQlx27/5+flaunRpsa8FqfifSUnzlJeCkN65c6cWLVpUqj/ejDHKyMiw4mfy888/64cffiixF5v2f4GpU6eqVatWatmy5WXPW577/2K/Lyv8NXBZp55VMqmpqcbb29tMnTrVbN261SQlJZnq1aubPXv2GGOMeeKJJ0xiYqKrfvfu3aZatWrm0UcfNVu3bjVTp0413t7e5uOPP77ivT/00EPG6XSaJUuWmOzsbNfj+PHjrpoL+3/ppZfMvHnzzHfffWc2b95snnjiCSPJzJkz54r3P2rUKLNkyRKze/dus2rVKtOrVy8TEBDwm9j3Bc6ePWsaNGhgHn/88ULTbNv3R44cMRs2bDAbNmwwkkxKSorZsGGD66zoiRMnGqfTaebOnWs2bdpk7rrrLhMaGmoOHz7sWkZiYqLbFRFff/21qVq1qpk4caLZtm2bmThxovHy8jKrVq26ov2fPn3aJCQkmPDwcJORkeH2ejh16lSx/ScnJ5u0tDTz3//+12zYsMHce++9xsvLy6xevfqK9n/kyBEzatQos3LlSpOZmWkWL15s4uLiTL169X4T+79AXl6eqVatmnnzzTeLXEZF7v9L+X1Zka8BgvoiXn/9dRMREWF8fHzMDTfc4HZ508CBA03Hjh3d6pcsWWKuv/564+PjYxo2bFjsf8ryJqnIx7Rp01w1F/Y/adIk06hRI+Pn52dq1qxp4uPjzWeffXblmzfG9OvXz4SGhhpvb28TFhZmbrvtNrNlyxbXdJv3fYEvv/zSSDI7duwoNM22fV9wediFj4EDBxpjfr08Zdy4cSYkJMT4+vqaDh06mE2bNrkto2PHjq76Ah999JFp0qSJ8fb2Nk2bNi23PzxK6j8zM7PY18PixYuL7T8pKck0aNDA+Pj4mLp165pu3bqZlStXXvH+jx8/brp162bq1q1rvL29TYMGDczAgQPN999/77YMW/d/gbffftv4+/ubQ4cOFbmMitz/l/L7siJfA9w9CwAAi/EZNQAAFiOoAQCwGEENAIDFCGoAACxGUAMAYDGCGgAAixHUAABYjKAGAMBiBDWAy7Jjxw716dNHwcHBcjgcatiwYUW3VKYcDoc6depU0W0ALgQ1UErHjx/XhAkTdMMNN+iqq66Sn5+fwsPDddNNN+nJJ5/Uf//734puscydPXtWf/zjH/Xll18qISFB48aNU1JSUonzJCcny+FwuD2qVaum6OhoPf300zp8+PCVaR74jeJ+1EApHDlyRPHx8dq4caMaN26sP/3pT6pRo4Z++OEHbdmyRRMnTlSjRo3UqFGjim61TGVmZmrbtm164IEH9NZbb13WvLfffruio6Ml/Xq7wC+++EITJkzQv//9b61Zs0a+vr7l0TLwm0dQA6Xw8ssva+PGjRo8eLDeeeedQjeCz8zM1KlTpyqou/Lz448/Svrf/Xkvxx133KH+/fu7np88eVJt27bVt99+q9mzZ+vee+8tsz6B3xPe+gZKIT09XZI0bNiwQiEt/Xr/2qZNm7qNlfTZZ8OGDQt91jto0CA5HA7t3r1bL7zwgq655hr5+/urefPmSk1NlfTrfZbHjh2ryMhI+fn5qUWLFvryyy8va1t+/vlnPfroo4qMjJSvr6+CgoLUr18/bd26tVCPHTt2lCT95S9/cb2NPX369MtaXwE/Pz/dfffdkqT169cXmp6Zman77rtPDRo0kK+vr0JDQzVo0CDt3bu3UO28efN01113qXHjxqpWrZqcTqduuukmzZkzp9j1v/vuu4qOjpafn5/q16+vMWPG6OTJk0XWZmdn65FHHlFUVJT8/f1Vq1YtxcTEaOjQobx1j3LHETVQCrVq1ZIk7dq1S9ddd125rmvkyJFavXq1evfurapVqyo1NVUDBgxQzZo19frrr2vz5s265ZZbdPLkSc2ePVsJCQnavn27IiMjL7rsn3/+WW3bttWuXbvUqVMn9e/fX3v27NHHH3+szz77TAsXLlRcXJwkKSkpSRkZGZoxY4Y6duzo+qPDk+0vuHmfl5f7r6LVq1ere/fuOnbsmHr37q3GjRtrz549mjVrlr744gulp6fr6quvdtU/+eST8vHxUXx8vEJDQ3Xw4EHNnz9fd9xxh1599VUNHz7cbfnPPvusxo4dq+DgYA0ZMkTe3t768MMPtW3btkI9Hj9+XO3bt9eePXvUrVs3/fGPf1R+fr52796t6dOna8yYMQoMDCz1PgAu6rJvjAnAfPLJJ0aSCQwMNI8//rj5z3/+Y3755ZcS55FU6B7aBSIiIkxERITb2MCBA40kExUVZQ4cOOAaX7VqlZFkatSoYeLj483Ro0dd0z788EMjyYwYMeKStuPPf/6zkWSefPJJt/G0tDTXus+ePesaL7jv8Lhx4y5p+cYYM27cOCPJfPDBB27jx48fNzExMUaS+eijj1zj+fn5pmHDhiYgIMBkZGS4zbN8+XJTtWpV06tXL7fx//73v4XWe+TIERMTE2OcTqc5duyYa3znzp3Gy8vL1KtXz+zfv981npeXZ5o0aVLo5zR//nwjyTz66KOF1nH48GFz6tSpS9sRQCnx1jdQCn369NHkyZN17tw5TZo0SZ07d1atWrXUuHFjDRs2TDt37iyzdT399NOqW7eu63lsbKyuvvpqHTp0SM8995yqV6/umnb77bfL29tb33777UWXm5+frw8++EC1a9fW//t//89tWvfu3dW9e3ft3LlTK1euLJPt+Pjjj5WcnKzk5GQ99NBDuuaaa7Rp0yb16dNHt912m6vu3//+t/bs2aMxY8aoZcuWbsuIj49Xnz599Pnnn7u95Xz+0XWBq666SoMGDVJeXp7Wrl3rGp89e7bOnDmjkSNHKigoyDUeGBhYaD+cz9/fv9BYQECAfHx8Lm0HAKXEW99AKT322GN68MEHlZaWppUrV2rdunVavXq1Xn/9dU2dOlUffvihEhISPF7P9ddfX2gsNDRUu3fvLvS2c9WqVRUUFKSsrKyLLnf79u06ceKEOnXqpGrVqhWa3qlTJ3355ZfKyMhQfHx8qfsvMGfOnEKfGd922236+OOP3T7nX7Vqlau/5OTkQsvJycnRuXPn9N1336l169aSpAMHDmjixIn64osvtHfvXp04ccJtnoKT4CS5/oi56aabCi27qLEOHTooJCREzz//vDIyMnTrrbcqPj5eMTExRZ6fAJQ1ghrwQEBAgO68807deeedkqS8vDw99dRTeuONNzR48GBlZWV5fMRV1OefBZ/pFjft9OnTF11uwRFpcHBwkdMLzuzOy8u75F5L8sEHH6h///46c+aMduzYodGjR2vu3LkaO3asnn32WVfdL7/8IkmaNWtWics7duyYq/7GG2/U999/r/bt26tLly6qUaOGqlatqoyMDH366aduZ+AXbM/5R9MFitoXTqdT6enpGjdunP71r3/p888/lySFh4frySef1NChQy9zTwCXh7e+gTLkdDo1ZcoURURE6KefftKmTZtc0xwOh86cOVPkfGUVhpejIOT3799f5PSC8bI+UcrLy0vXXnut5s2bp8aNG+u5557TN998U6ivf/3rXzLGFPsoOAN96tSp+v777zV+/HitWLFCr732mp599lklJyerbdu2hdbvdDol/XoUXtw2X6hhw4aaMWOGDh48qA0bNmjSpEkyxujhhx/WBx984PE+AUpCUANlrOCbty5Us2bNIt+S3rNnjw4dOnQFOnPXtGlT+fn5ae3atTp+/Hih6UuXLpXk2VndJfHz89MLL7wgY4yeeOIJ13hsbKyk/10CdzEF3wBX1McMy5cvLzRW8Ll3UdOKGjtf1apVdd1112nMmDGugJ4/f/4l9QmUFkENlMLbb7/tdoLS+ebOnavt27erRo0arm/ikqTWrVtrz549WrJkiWssPz9fI0eOLO92i+Tj46O77rpLP/30k55//nm3aYsWLdIXX3yhxo0bq3379uXWQ58+fXTDDTdo4cKFrpDs06ePGjRooJSUFC1btqzQPKdPn9aKFStczyMiIiTJbUz69aSxgrepzzdgwABVrVpVKSkpbkfVhw8f1vjx4wvVb968uchrtwuOvos6yQwoS3xGDZTCF198oQcffNAVZGFhYTp69KgyMjK0fPlyValSRW+88Ybb12I++uijWrBggW699VbdddddqlatmhYuXKgaNWooNDS0QrZj0qRJWrp0qcaPH6+VK1cqNjbWdR11tWrVNG3aNFWpUr5/zycnJyshIUFjx47V4sWL5evrq48//lg9e/ZUx44d1blzZ9cfPN9//72WL1+u2rVra/v27ZKkxMRETZo0ScOHD9fixYsVERGhjRs3atGiRbrttts0d+5ct/U1btxYY8eO1bhx49SiRQv17dtXXl5emjNnjmJiYrRjxw63+kWLFmnUqFFq3769mjZtqtq1a2v37t2aP3++/P39NWzYsHLdPwDXUQOlsH37djN58mTTtWtXExkZafz8/Iyfn59p1KiRGThwoFm3bl2R83344YcmJibG+Pj4mJCQEDN8+HBz5MiREq+jzszMLLScjh07muJevkUtqyQHDx40I0aMMBEREcbb29vUqVPH3HHHHWbTpk2FasvyOurztW7d2kgy//nPf1xj+/btM4888oiJiooyvr6+JjAw0DRr1szcd999bnXGGJORkWG6detmatasaQICAkzHjh3NokWLzLRp04wkM23atELrfOedd0zz5s2Nj4+PCQ8PN6NHjzbHjx8vdB311q1bzSOPPGKuv/56U7t2bePr62uuvvpqM2jQILN169ZL3g9AaTmM+f+/GggAAFiHz6gBALAYQQ0AgMUIagAALEZQAwBgMYIaAACLEdQAAFiMoAYAwGIENQAAFiOoAQCwGEENAIDFCGoAACxGUAMAYDGCGgAAi/1/y8jtHB89s5IAAAAASUVORK5CYII=",
      "text/plain": [
       "<Figure size 500x500 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "fig, ax = plt.subplots(figsize=(5, 5))\n",
    "fig.patch.set_facecolor(\"white\")\n",
    "ax = sns.histplot(\n",
    "    trimmed_raw_transcriptome.loc[:, trimmed_raw_transcriptome.sum() < 20].sum()\n",
    ")\n",
    "plt.xlabel(\"Sum of Reads\", fontsize=14)\n",
    "plt.ylabel(\"Frequency\", fontsize=14)\n",
    "sns.despine()\n",
    "# ax.grid(False)\n",
    "# plt.axhline(1, color = 'red', linestyle = 'dashed')\n",
    "ax.spines[\"bottom\"].set_color(\"black\")\n",
    "ax.spines[\"bottom\"].set_linewidth(2)\n",
    "ax.spines[\"left\"].set_color(\"black\")\n",
    "ax.spines[\"left\"].set_linewidth(2)\n",
    "ax.tick_params(axis=\"both\", width=2)\n",
    "plt.xlim((0, 20))\n",
    "plt.ylim((0, 5000))\n",
    "# handles, labels  =  ax.get_legend_handles_labels()\n",
    "# ax.legend(loc=\"upper right\")\n",
    "# plt.title(\"Current Threshold is 31\", fontsize = 20)\n",
    "plt.xticks(\n",
    "    fontsize=10,\n",
    ")  # rotation=90\n",
    "plt.yticks(fontsize=10)\n",
    "# plt.ylim(-.02,1)\n",
    "plt.tight_layout()\n",
    "# ax.plot([0,1],[0,1], transform=ax.transAxes, linestyle = 'dashed', color = 'k', linewidth = 1.5)\n",
    "# for line in range(0,full_meta_data.shape[0]):\n",
    "#      ax.text(pca[:,0][line]+0.01, pca[:,1][line],\n",
    "#      full_meta_data['plate.pos'][line], horizontalalignment='left',\n",
    "#      size='medium', color='black', weight='semibold')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "69ead239",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>plate.pos</th>\n",
       "      <th>sampID</th>\n",
       "      <th>sample.type</th>\n",
       "      <th>timepoint</th>\n",
       "      <th>date</th>\n",
       "      <th>time</th>\n",
       "      <th>extractplate</th>\n",
       "      <th>rnaprepplate</th>\n",
       "      <th>Unmapped</th>\n",
       "      <th>No Feature</th>\n",
       "      <th>plate_row</th>\n",
       "      <th>plate_col</th>\n",
       "      <th>Total Reads</th>\n",
       "      <th>Experiment Type</th>\n",
       "      <th>Sample Month</th>\n",
       "      <th>daysincestart</th>\n",
       "      <th>Date and Time</th>\n",
       "      <th>has_rare_species</th>\n",
       "      <th>lacking_core_species</th>\n",
       "      <th>rare_species_total_abundR100</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>filename</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A01</th>\n",
       "      <td>A01</td>\n",
       "      <td>LIC157</td>\n",
       "      <td>plant</td>\n",
       "      <td>t14</td>\n",
       "      <td>1/23/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_02</td>\n",
       "      <td>LICRNA_02</td>\n",
       "      <td>758284.0</td>\n",
       "      <td>82176.0</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "      <td>1246970.0</td>\n",
       "      <td>Long Term</td>\n",
       "      <td>January</td>\n",
       "      <td>83</td>\n",
       "      <td>2024-01-23 08:00:00</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>13.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_B02</th>\n",
       "      <td>B02</td>\n",
       "      <td>LIC170</td>\n",
       "      <td>plant</td>\n",
       "      <td>t15</td>\n",
       "      <td>1/27/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_02</td>\n",
       "      <td>LICRNA_02</td>\n",
       "      <td>1839268.0</td>\n",
       "      <td>243038.0</td>\n",
       "      <td>B</td>\n",
       "      <td>2</td>\n",
       "      <td>2982080.0</td>\n",
       "      <td>Long Term</td>\n",
       "      <td>January</td>\n",
       "      <td>87</td>\n",
       "      <td>2024-01-27 08:00:00</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_C07</th>\n",
       "      <td>C07</td>\n",
       "      <td>LIC243</td>\n",
       "      <td>plant</td>\n",
       "      <td>t21</td>\n",
       "      <td>2/20/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_03</td>\n",
       "      <td>LICRNA_02</td>\n",
       "      <td>5931387.0</td>\n",
       "      <td>372994.0</td>\n",
       "      <td>C</td>\n",
       "      <td>7</td>\n",
       "      <td>5220104.0</td>\n",
       "      <td>Long Term</td>\n",
       "      <td>February</td>\n",
       "      <td>111</td>\n",
       "      <td>2024-02-20 08:00:00</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>11.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_C12</th>\n",
       "      <td>C12</td>\n",
       "      <td>LIC315</td>\n",
       "      <td>plant</td>\n",
       "      <td>t27</td>\n",
       "      <td>3/12/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_04</td>\n",
       "      <td>LICRNA_02</td>\n",
       "      <td>10581103.0</td>\n",
       "      <td>770403.0</td>\n",
       "      <td>C</td>\n",
       "      <td>12</td>\n",
       "      <td>8128106.0</td>\n",
       "      <td>Long Term</td>\n",
       "      <td>March</td>\n",
       "      <td>132</td>\n",
       "      <td>2024-03-12 08:00:00</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_D07</th>\n",
       "      <td>D07</td>\n",
       "      <td>LIC244</td>\n",
       "      <td>plant</td>\n",
       "      <td>t21</td>\n",
       "      <td>2/20/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_03</td>\n",
       "      <td>LICRNA_02</td>\n",
       "      <td>5559066.0</td>\n",
       "      <td>408088.0</td>\n",
       "      <td>D</td>\n",
       "      <td>7</td>\n",
       "      <td>4167468.0</td>\n",
       "      <td>Long Term</td>\n",
       "      <td>February</td>\n",
       "      <td>111</td>\n",
       "      <td>2024-02-20 08:00:00</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>8.9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_F03</th>\n",
       "      <td>F03</td>\n",
       "      <td>LIC354</td>\n",
       "      <td>plant</td>\n",
       "      <td>t16</td>\n",
       "      <td>1/30/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_04</td>\n",
       "      <td>LICRNA_03</td>\n",
       "      <td>6334749.0</td>\n",
       "      <td>510584.0</td>\n",
       "      <td>F</td>\n",
       "      <td>3</td>\n",
       "      <td>8399773.0</td>\n",
       "      <td>Long Term</td>\n",
       "      <td>January</td>\n",
       "      <td>90</td>\n",
       "      <td>2024-01-30 08:00:00</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_F12</th>\n",
       "      <td>F12</td>\n",
       "      <td>LIC486</td>\n",
       "      <td>plant</td>\n",
       "      <td>c1_t08</td>\n",
       "      <td>2/11/24</td>\n",
       "      <td>0:00</td>\n",
       "      <td>LICCIRC_02</td>\n",
       "      <td>LICRNA_03</td>\n",
       "      <td>1281256.0</td>\n",
       "      <td>107554.0</td>\n",
       "      <td>F</td>\n",
       "      <td>12</td>\n",
       "      <td>1278970.0</td>\n",
       "      <td>Circadian Experiment 1</td>\n",
       "      <td>February</td>\n",
       "      <td>102</td>\n",
       "      <td>2024-02-11 00:00:00</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>12.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_G01</th>\n",
       "      <td>G01</td>\n",
       "      <td>LIC331</td>\n",
       "      <td>plant</td>\n",
       "      <td>t28</td>\n",
       "      <td>3/14/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_04</td>\n",
       "      <td>LICRNA_03</td>\n",
       "      <td>7669950.0</td>\n",
       "      <td>769306.0</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>8957746.0</td>\n",
       "      <td>Long Term</td>\n",
       "      <td>March</td>\n",
       "      <td>134</td>\n",
       "      <td>2024-03-14 08:00:00</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>13.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_G04</th>\n",
       "      <td>G04</td>\n",
       "      <td>LIC367</td>\n",
       "      <td>plant</td>\n",
       "      <td>t08</td>\n",
       "      <td>12/5/23</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_04</td>\n",
       "      <td>LICRNA_03</td>\n",
       "      <td>1477644.0</td>\n",
       "      <td>189073.0</td>\n",
       "      <td>G</td>\n",
       "      <td>4</td>\n",
       "      <td>2732871.0</td>\n",
       "      <td>Long Term</td>\n",
       "      <td>December</td>\n",
       "      <td>34</td>\n",
       "      <td>2023-12-05 08:00:00</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>15.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_G05</th>\n",
       "      <td>G05</td>\n",
       "      <td>LIC391</td>\n",
       "      <td>plant</td>\n",
       "      <td>c1_t01</td>\n",
       "      <td>2/9/24</td>\n",
       "      <td>20:00</td>\n",
       "      <td>LICCIRC_01</td>\n",
       "      <td>LICRNA_03</td>\n",
       "      <td>1459556.0</td>\n",
       "      <td>323179.0</td>\n",
       "      <td>G</td>\n",
       "      <td>5</td>\n",
       "      <td>4860915.0</td>\n",
       "      <td>Circadian Experiment 1</td>\n",
       "      <td>February</td>\n",
       "      <td>100</td>\n",
       "      <td>2024-02-09 20:00:00</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>7.5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>79 rows × 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             plate.pos  sampID sample.type timepoint     date  \\\n",
       "filename                                                                        \n",
       "A2449446903_n01_LICRNA02_A01       A01  LIC157       plant       t14  1/23/24   \n",
       "A2449446903_n01_LICRNA02_B02       B02  LIC170       plant       t15  1/27/24   \n",
       "A2449446903_n01_LICRNA02_C07       C07  LIC243       plant       t21  2/20/24   \n",
       "A2449446903_n01_LICRNA02_C12       C12  LIC315       plant       t27  3/12/24   \n",
       "A2449446903_n01_LICRNA02_D07       D07  LIC244       plant       t21  2/20/24   \n",
       "...                                ...     ...         ...       ...      ...   \n",
       "B250508004_n01_LICRNA03_F03        F03  LIC354       plant       t16  1/30/24   \n",
       "B250508004_n01_LICRNA03_F12        F12  LIC486       plant    c1_t08  2/11/24   \n",
       "B250508004_n01_LICRNA03_G01        G01  LIC331       plant       t28  3/14/24   \n",
       "B250508004_n01_LICRNA03_G04        G04  LIC367       plant       t08  12/5/23   \n",
       "B250508004_n01_LICRNA03_G05        G05  LIC391       plant    c1_t01   2/9/24   \n",
       "\n",
       "                               time extractplate rnaprepplate    Unmapped  \\\n",
       "filename                                                                    \n",
       "A2449446903_n01_LICRNA02_A01   8:00       LIC_02    LICRNA_02    758284.0   \n",
       "A2449446903_n01_LICRNA02_B02   8:00       LIC_02    LICRNA_02   1839268.0   \n",
       "A2449446903_n01_LICRNA02_C07   8:00       LIC_03    LICRNA_02   5931387.0   \n",
       "A2449446903_n01_LICRNA02_C12   8:00       LIC_04    LICRNA_02  10581103.0   \n",
       "A2449446903_n01_LICRNA02_D07   8:00       LIC_03    LICRNA_02   5559066.0   \n",
       "...                             ...          ...          ...         ...   \n",
       "B250508004_n01_LICRNA03_F03    8:00       LIC_04    LICRNA_03   6334749.0   \n",
       "B250508004_n01_LICRNA03_F12    0:00   LICCIRC_02    LICRNA_03   1281256.0   \n",
       "B250508004_n01_LICRNA03_G01    8:00       LIC_04    LICRNA_03   7669950.0   \n",
       "B250508004_n01_LICRNA03_G04    8:00       LIC_04    LICRNA_03   1477644.0   \n",
       "B250508004_n01_LICRNA03_G05   20:00   LICCIRC_01    LICRNA_03   1459556.0   \n",
       "\n",
       "                              No Feature plate_row  plate_col  Total Reads  \\\n",
       "filename                                                                     \n",
       "A2449446903_n01_LICRNA02_A01     82176.0         A          1    1246970.0   \n",
       "A2449446903_n01_LICRNA02_B02    243038.0         B          2    2982080.0   \n",
       "A2449446903_n01_LICRNA02_C07    372994.0         C          7    5220104.0   \n",
       "A2449446903_n01_LICRNA02_C12    770403.0         C         12    8128106.0   \n",
       "A2449446903_n01_LICRNA02_D07    408088.0         D          7    4167468.0   \n",
       "...                                  ...       ...        ...          ...   \n",
       "B250508004_n01_LICRNA03_F03     510584.0         F          3    8399773.0   \n",
       "B250508004_n01_LICRNA03_F12     107554.0         F         12    1278970.0   \n",
       "B250508004_n01_LICRNA03_G01     769306.0         G          1    8957746.0   \n",
       "B250508004_n01_LICRNA03_G04     189073.0         G          4    2732871.0   \n",
       "B250508004_n01_LICRNA03_G05     323179.0         G          5    4860915.0   \n",
       "\n",
       "                                     Experiment Type Sample Month  \\\n",
       "filename                                                            \n",
       "A2449446903_n01_LICRNA02_A01               Long Term      January   \n",
       "A2449446903_n01_LICRNA02_B02               Long Term      January   \n",
       "A2449446903_n01_LICRNA02_C07               Long Term     February   \n",
       "A2449446903_n01_LICRNA02_C12               Long Term        March   \n",
       "A2449446903_n01_LICRNA02_D07               Long Term     February   \n",
       "...                                              ...          ...   \n",
       "B250508004_n01_LICRNA03_F03                Long Term      January   \n",
       "B250508004_n01_LICRNA03_F12   Circadian Experiment 1     February   \n",
       "B250508004_n01_LICRNA03_G01                Long Term        March   \n",
       "B250508004_n01_LICRNA03_G04                Long Term     December   \n",
       "B250508004_n01_LICRNA03_G05   Circadian Experiment 1     February   \n",
       "\n",
       "                              daysincestart       Date and Time  \\\n",
       "filename                                                          \n",
       "A2449446903_n01_LICRNA02_A01             83 2024-01-23 08:00:00   \n",
       "A2449446903_n01_LICRNA02_B02             87 2024-01-27 08:00:00   \n",
       "A2449446903_n01_LICRNA02_C07            111 2024-02-20 08:00:00   \n",
       "A2449446903_n01_LICRNA02_C12            132 2024-03-12 08:00:00   \n",
       "A2449446903_n01_LICRNA02_D07            111 2024-02-20 08:00:00   \n",
       "...                                     ...                 ...   \n",
       "B250508004_n01_LICRNA03_F03              90 2024-01-30 08:00:00   \n",
       "B250508004_n01_LICRNA03_F12             102 2024-02-11 00:00:00   \n",
       "B250508004_n01_LICRNA03_G01             134 2024-03-14 08:00:00   \n",
       "B250508004_n01_LICRNA03_G04              34 2023-12-05 08:00:00   \n",
       "B250508004_n01_LICRNA03_G05             100 2024-02-09 20:00:00   \n",
       "\n",
       "                              has_rare_species  lacking_core_species  \\\n",
       "filename                                                               \n",
       "A2449446903_n01_LICRNA02_A01              True                 False   \n",
       "A2449446903_n01_LICRNA02_B02              True                 False   \n",
       "A2449446903_n01_LICRNA02_C07              True                 False   \n",
       "A2449446903_n01_LICRNA02_C12              True                 False   \n",
       "A2449446903_n01_LICRNA02_D07              True                 False   \n",
       "...                                        ...                   ...   \n",
       "B250508004_n01_LICRNA03_F03               True                  True   \n",
       "B250508004_n01_LICRNA03_F12               True                 False   \n",
       "B250508004_n01_LICRNA03_G01               True                  True   \n",
       "B250508004_n01_LICRNA03_G04               True                 False   \n",
       "B250508004_n01_LICRNA03_G05               True                  True   \n",
       "\n",
       "                              rare_species_total_abundR100  \n",
       "filename                                                    \n",
       "A2449446903_n01_LICRNA02_A01                          13.5  \n",
       "A2449446903_n01_LICRNA02_B02                           9.0  \n",
       "A2449446903_n01_LICRNA02_C07                          11.5  \n",
       "A2449446903_n01_LICRNA02_C12                           7.0  \n",
       "A2449446903_n01_LICRNA02_D07                           8.9  \n",
       "...                                                    ...  \n",
       "B250508004_n01_LICRNA03_F03                            7.0  \n",
       "B250508004_n01_LICRNA03_F12                           12.8  \n",
       "B250508004_n01_LICRNA03_G01                           13.3  \n",
       "B250508004_n01_LICRNA03_G04                           15.1  \n",
       "B250508004_n01_LICRNA03_G05                            7.5  \n",
       "\n",
       "[79 rows x 20 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "trimmed_metadata.loc[\n",
    "    trimmed_metadata[\"rare_species_total_abundR100\"] > rare_species_abundance_threshold\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "254b4691",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AT1G01010</th>\n",
       "      <th>AT1G01020</th>\n",
       "      <th>AT1G01030</th>\n",
       "      <th>AT1G01040</th>\n",
       "      <th>AT1G01050</th>\n",
       "      <th>AT1G01060</th>\n",
       "      <th>AT1G01070</th>\n",
       "      <th>AT1G01080</th>\n",
       "      <th>AT1G01090</th>\n",
       "      <th>AT1G01100</th>\n",
       "      <th>...</th>\n",
       "      <th>ArthCp080</th>\n",
       "      <th>ArthCp081</th>\n",
       "      <th>Arthcp087</th>\n",
       "      <th>ArthCr091</th>\n",
       "      <th>ArthCr090</th>\n",
       "      <th>ArthCr089</th>\n",
       "      <th>ArthCr088</th>\n",
       "      <th>ArthCp088</th>\n",
       "      <th>ArthCp083</th>\n",
       "      <th>ArthCp085</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A2450525897_n01_LICRNA01_H06</th>\n",
       "      <td>0</td>\n",
       "      <td>48</td>\n",
       "      <td>34</td>\n",
       "      <td>111</td>\n",
       "      <td>343</td>\n",
       "      <td>198</td>\n",
       "      <td>10</td>\n",
       "      <td>181</td>\n",
       "      <td>673</td>\n",
       "      <td>5893</td>\n",
       "      <td>...</td>\n",
       "      <td>70</td>\n",
       "      <td>934</td>\n",
       "      <td>10005</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>451</td>\n",
       "      <td>53</td>\n",
       "      <td>29</td>\n",
       "      <td>30</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2450525897_n01_LICRNA01_F06</th>\n",
       "      <td>12</td>\n",
       "      <td>87</td>\n",
       "      <td>55</td>\n",
       "      <td>150</td>\n",
       "      <td>331</td>\n",
       "      <td>212</td>\n",
       "      <td>28</td>\n",
       "      <td>256</td>\n",
       "      <td>1024</td>\n",
       "      <td>15009</td>\n",
       "      <td>...</td>\n",
       "      <td>64</td>\n",
       "      <td>792</td>\n",
       "      <td>14093</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>866</td>\n",
       "      <td>82</td>\n",
       "      <td>18</td>\n",
       "      <td>3</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2450525897_n01_LICRNA01_E01</th>\n",
       "      <td>19</td>\n",
       "      <td>27</td>\n",
       "      <td>22</td>\n",
       "      <td>41</td>\n",
       "      <td>135</td>\n",
       "      <td>89</td>\n",
       "      <td>3</td>\n",
       "      <td>56</td>\n",
       "      <td>299</td>\n",
       "      <td>3131</td>\n",
       "      <td>...</td>\n",
       "      <td>24</td>\n",
       "      <td>395</td>\n",
       "      <td>4594</td>\n",
       "      <td>6</td>\n",
       "      <td>10</td>\n",
       "      <td>542</td>\n",
       "      <td>44</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2450525897_n01_LICRNA01_H10</th>\n",
       "      <td>6</td>\n",
       "      <td>24</td>\n",
       "      <td>5</td>\n",
       "      <td>97</td>\n",
       "      <td>140</td>\n",
       "      <td>75</td>\n",
       "      <td>13</td>\n",
       "      <td>53</td>\n",
       "      <td>373</td>\n",
       "      <td>3347</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>319</td>\n",
       "      <td>1723</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>549</td>\n",
       "      <td>32</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2450525897_n01_LICRNA01_F10</th>\n",
       "      <td>6</td>\n",
       "      <td>42</td>\n",
       "      <td>6</td>\n",
       "      <td>85</td>\n",
       "      <td>215</td>\n",
       "      <td>69</td>\n",
       "      <td>12</td>\n",
       "      <td>50</td>\n",
       "      <td>368</td>\n",
       "      <td>3535</td>\n",
       "      <td>...</td>\n",
       "      <td>29</td>\n",
       "      <td>397</td>\n",
       "      <td>2477</td>\n",
       "      <td>6</td>\n",
       "      <td>4</td>\n",
       "      <td>261</td>\n",
       "      <td>15</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B2449500127_n01_LICRNA04_E07</th>\n",
       "      <td>6</td>\n",
       "      <td>59</td>\n",
       "      <td>7</td>\n",
       "      <td>103</td>\n",
       "      <td>206</td>\n",
       "      <td>53</td>\n",
       "      <td>1</td>\n",
       "      <td>144</td>\n",
       "      <td>529</td>\n",
       "      <td>3073</td>\n",
       "      <td>...</td>\n",
       "      <td>58</td>\n",
       "      <td>911</td>\n",
       "      <td>4851</td>\n",
       "      <td>36</td>\n",
       "      <td>2</td>\n",
       "      <td>360</td>\n",
       "      <td>22</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B2449500127_n01_LICRNA04_F10</th>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>0</td>\n",
       "      <td>74</td>\n",
       "      <td>178</td>\n",
       "      <td>73</td>\n",
       "      <td>0</td>\n",
       "      <td>55</td>\n",
       "      <td>252</td>\n",
       "      <td>1357</td>\n",
       "      <td>...</td>\n",
       "      <td>21</td>\n",
       "      <td>497</td>\n",
       "      <td>2300</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>590</td>\n",
       "      <td>13</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B2449500127_n01_LICRNA04_C01</th>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>15</td>\n",
       "      <td>64</td>\n",
       "      <td>99</td>\n",
       "      <td>4</td>\n",
       "      <td>7</td>\n",
       "      <td>78</td>\n",
       "      <td>210</td>\n",
       "      <td>1061</td>\n",
       "      <td>...</td>\n",
       "      <td>12</td>\n",
       "      <td>586</td>\n",
       "      <td>2356</td>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "      <td>463</td>\n",
       "      <td>7</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B2449500127_n01_LICRNA04_D12</th>\n",
       "      <td>15</td>\n",
       "      <td>82</td>\n",
       "      <td>11</td>\n",
       "      <td>192</td>\n",
       "      <td>347</td>\n",
       "      <td>116</td>\n",
       "      <td>15</td>\n",
       "      <td>228</td>\n",
       "      <td>1108</td>\n",
       "      <td>6467</td>\n",
       "      <td>...</td>\n",
       "      <td>183</td>\n",
       "      <td>2575</td>\n",
       "      <td>9627</td>\n",
       "      <td>32</td>\n",
       "      <td>2</td>\n",
       "      <td>379</td>\n",
       "      <td>47</td>\n",
       "      <td>4</td>\n",
       "      <td>33</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B2449500127_n01_LICRNA04_A03</th>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>29</td>\n",
       "      <td>97</td>\n",
       "      <td>134</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>187</td>\n",
       "      <td>898</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>369</td>\n",
       "      <td>1862</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>224</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>373 rows × 17682 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              AT1G01010  AT1G01020  AT1G01030  AT1G01040  \\\n",
       "A2450525897_n01_LICRNA01_H06          0         48         34        111   \n",
       "A2450525897_n01_LICRNA01_F06         12         87         55        150   \n",
       "A2450525897_n01_LICRNA01_E01         19         27         22         41   \n",
       "A2450525897_n01_LICRNA01_H10          6         24          5         97   \n",
       "A2450525897_n01_LICRNA01_F10          6         42          6         85   \n",
       "...                                 ...        ...        ...        ...   \n",
       "B2449500127_n01_LICRNA04_E07          6         59          7        103   \n",
       "B2449500127_n01_LICRNA04_F10          0         25          0         74   \n",
       "B2449500127_n01_LICRNA04_C01          0         20         15         64   \n",
       "B2449500127_n01_LICRNA04_D12         15         82         11        192   \n",
       "B2449500127_n01_LICRNA04_A03          2          5          0         29   \n",
       "\n",
       "                              AT1G01050  AT1G01060  AT1G01070  AT1G01080  \\\n",
       "A2450525897_n01_LICRNA01_H06        343        198         10        181   \n",
       "A2450525897_n01_LICRNA01_F06        331        212         28        256   \n",
       "A2450525897_n01_LICRNA01_E01        135         89          3         56   \n",
       "A2450525897_n01_LICRNA01_H10        140         75         13         53   \n",
       "A2450525897_n01_LICRNA01_F10        215         69         12         50   \n",
       "...                                 ...        ...        ...        ...   \n",
       "B2449500127_n01_LICRNA04_E07        206         53          1        144   \n",
       "B2449500127_n01_LICRNA04_F10        178         73          0         55   \n",
       "B2449500127_n01_LICRNA04_C01         99          4          7         78   \n",
       "B2449500127_n01_LICRNA04_D12        347        116         15        228   \n",
       "B2449500127_n01_LICRNA04_A03         97        134          0         22   \n",
       "\n",
       "                              AT1G01090  AT1G01100  ...  ArthCp080  ArthCp081  \\\n",
       "A2450525897_n01_LICRNA01_H06        673       5893  ...         70        934   \n",
       "A2450525897_n01_LICRNA01_F06       1024      15009  ...         64        792   \n",
       "A2450525897_n01_LICRNA01_E01        299       3131  ...         24        395   \n",
       "A2450525897_n01_LICRNA01_H10        373       3347  ...         22        319   \n",
       "A2450525897_n01_LICRNA01_F10        368       3535  ...         29        397   \n",
       "...                                 ...        ...  ...        ...        ...   \n",
       "B2449500127_n01_LICRNA04_E07        529       3073  ...         58        911   \n",
       "B2449500127_n01_LICRNA04_F10        252       1357  ...         21        497   \n",
       "B2449500127_n01_LICRNA04_C01        210       1061  ...         12        586   \n",
       "B2449500127_n01_LICRNA04_D12       1108       6467  ...        183       2575   \n",
       "B2449500127_n01_LICRNA04_A03        187        898  ...          1        369   \n",
       "\n",
       "                              Arthcp087  ArthCr091  ArthCr090  ArthCr089  \\\n",
       "A2450525897_n01_LICRNA01_H06      10005          0          5        451   \n",
       "A2450525897_n01_LICRNA01_F06      14093          0          7        866   \n",
       "A2450525897_n01_LICRNA01_E01       4594          6         10        542   \n",
       "A2450525897_n01_LICRNA01_H10       1723          1          1        549   \n",
       "A2450525897_n01_LICRNA01_F10       2477          6          4        261   \n",
       "...                                 ...        ...        ...        ...   \n",
       "B2449500127_n01_LICRNA04_E07       4851         36          2        360   \n",
       "B2449500127_n01_LICRNA04_F10       2300          0          0        590   \n",
       "B2449500127_n01_LICRNA04_C01       2356          9          2        463   \n",
       "B2449500127_n01_LICRNA04_D12       9627         32          2        379   \n",
       "B2449500127_n01_LICRNA04_A03       1862          0          5        224   \n",
       "\n",
       "                              ArthCr088  ArthCp088  ArthCp083  ArthCp085  \n",
       "A2450525897_n01_LICRNA01_H06         53         29         30          6  \n",
       "A2450525897_n01_LICRNA01_F06         82         18          3         12  \n",
       "A2450525897_n01_LICRNA01_E01         44          0          3          5  \n",
       "A2450525897_n01_LICRNA01_H10         32          2          8          0  \n",
       "A2450525897_n01_LICRNA01_F10         15          9          4          1  \n",
       "...                                 ...        ...        ...        ...  \n",
       "B2449500127_n01_LICRNA04_E07         22         14          1          3  \n",
       "B2449500127_n01_LICRNA04_F10         13          3          0          0  \n",
       "B2449500127_n01_LICRNA04_C01          7          5          1         20  \n",
       "B2449500127_n01_LICRNA04_D12         47          4         33         25  \n",
       "B2449500127_n01_LICRNA04_A03         10          0          1          1  \n",
       "\n",
       "[373 rows x 17682 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cleaned_raw_transcriptome = trimmed_raw_transcriptome.loc[\n",
    "    :, trimmed_raw_transcriptome.sum() > expressed_gene_reads_threshold\n",
    "]\n",
    "\n",
    "cleaned_raw_transcriptome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "c1240000",
   "metadata": {},
   "outputs": [],
   "source": [
    "cleaned_raw_transcriptome = cleaned_raw_transcriptome.loc[\n",
    "    :, (cleaned_raw_transcriptome > 1).sum() > number_of_samples_expressed_in_threshold\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "ad722625",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AT1G01010</th>\n",
       "      <th>AT1G01020</th>\n",
       "      <th>AT1G01030</th>\n",
       "      <th>AT1G01040</th>\n",
       "      <th>AT1G01050</th>\n",
       "      <th>AT1G01060</th>\n",
       "      <th>AT1G01070</th>\n",
       "      <th>AT1G01080</th>\n",
       "      <th>AT1G01090</th>\n",
       "      <th>AT1G01100</th>\n",
       "      <th>...</th>\n",
       "      <th>ArthCp080</th>\n",
       "      <th>ArthCp081</th>\n",
       "      <th>Arthcp087</th>\n",
       "      <th>ArthCr091</th>\n",
       "      <th>ArthCr090</th>\n",
       "      <th>ArthCr089</th>\n",
       "      <th>ArthCr088</th>\n",
       "      <th>ArthCp088</th>\n",
       "      <th>ArthCp083</th>\n",
       "      <th>ArthCp085</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A2450525897_n01_LICRNA01_H06</th>\n",
       "      <td>0</td>\n",
       "      <td>48</td>\n",
       "      <td>34</td>\n",
       "      <td>111</td>\n",
       "      <td>343</td>\n",
       "      <td>198</td>\n",
       "      <td>10</td>\n",
       "      <td>181</td>\n",
       "      <td>673</td>\n",
       "      <td>5893</td>\n",
       "      <td>...</td>\n",
       "      <td>70</td>\n",
       "      <td>934</td>\n",
       "      <td>10005</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>451</td>\n",
       "      <td>53</td>\n",
       "      <td>29</td>\n",
       "      <td>30</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2450525897_n01_LICRNA01_F06</th>\n",
       "      <td>12</td>\n",
       "      <td>87</td>\n",
       "      <td>55</td>\n",
       "      <td>150</td>\n",
       "      <td>331</td>\n",
       "      <td>212</td>\n",
       "      <td>28</td>\n",
       "      <td>256</td>\n",
       "      <td>1024</td>\n",
       "      <td>15009</td>\n",
       "      <td>...</td>\n",
       "      <td>64</td>\n",
       "      <td>792</td>\n",
       "      <td>14093</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>866</td>\n",
       "      <td>82</td>\n",
       "      <td>18</td>\n",
       "      <td>3</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2450525897_n01_LICRNA01_E01</th>\n",
       "      <td>19</td>\n",
       "      <td>27</td>\n",
       "      <td>22</td>\n",
       "      <td>41</td>\n",
       "      <td>135</td>\n",
       "      <td>89</td>\n",
       "      <td>3</td>\n",
       "      <td>56</td>\n",
       "      <td>299</td>\n",
       "      <td>3131</td>\n",
       "      <td>...</td>\n",
       "      <td>24</td>\n",
       "      <td>395</td>\n",
       "      <td>4594</td>\n",
       "      <td>6</td>\n",
       "      <td>10</td>\n",
       "      <td>542</td>\n",
       "      <td>44</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2450525897_n01_LICRNA01_H10</th>\n",
       "      <td>6</td>\n",
       "      <td>24</td>\n",
       "      <td>5</td>\n",
       "      <td>97</td>\n",
       "      <td>140</td>\n",
       "      <td>75</td>\n",
       "      <td>13</td>\n",
       "      <td>53</td>\n",
       "      <td>373</td>\n",
       "      <td>3347</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>319</td>\n",
       "      <td>1723</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>549</td>\n",
       "      <td>32</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2450525897_n01_LICRNA01_F10</th>\n",
       "      <td>6</td>\n",
       "      <td>42</td>\n",
       "      <td>6</td>\n",
       "      <td>85</td>\n",
       "      <td>215</td>\n",
       "      <td>69</td>\n",
       "      <td>12</td>\n",
       "      <td>50</td>\n",
       "      <td>368</td>\n",
       "      <td>3535</td>\n",
       "      <td>...</td>\n",
       "      <td>29</td>\n",
       "      <td>397</td>\n",
       "      <td>2477</td>\n",
       "      <td>6</td>\n",
       "      <td>4</td>\n",
       "      <td>261</td>\n",
       "      <td>15</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B2449500127_n01_LICRNA04_E07</th>\n",
       "      <td>6</td>\n",
       "      <td>59</td>\n",
       "      <td>7</td>\n",
       "      <td>103</td>\n",
       "      <td>206</td>\n",
       "      <td>53</td>\n",
       "      <td>1</td>\n",
       "      <td>144</td>\n",
       "      <td>529</td>\n",
       "      <td>3073</td>\n",
       "      <td>...</td>\n",
       "      <td>58</td>\n",
       "      <td>911</td>\n",
       "      <td>4851</td>\n",
       "      <td>36</td>\n",
       "      <td>2</td>\n",
       "      <td>360</td>\n",
       "      <td>22</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B2449500127_n01_LICRNA04_F10</th>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>0</td>\n",
       "      <td>74</td>\n",
       "      <td>178</td>\n",
       "      <td>73</td>\n",
       "      <td>0</td>\n",
       "      <td>55</td>\n",
       "      <td>252</td>\n",
       "      <td>1357</td>\n",
       "      <td>...</td>\n",
       "      <td>21</td>\n",
       "      <td>497</td>\n",
       "      <td>2300</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>590</td>\n",
       "      <td>13</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B2449500127_n01_LICRNA04_C01</th>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>15</td>\n",
       "      <td>64</td>\n",
       "      <td>99</td>\n",
       "      <td>4</td>\n",
       "      <td>7</td>\n",
       "      <td>78</td>\n",
       "      <td>210</td>\n",
       "      <td>1061</td>\n",
       "      <td>...</td>\n",
       "      <td>12</td>\n",
       "      <td>586</td>\n",
       "      <td>2356</td>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "      <td>463</td>\n",
       "      <td>7</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B2449500127_n01_LICRNA04_D12</th>\n",
       "      <td>15</td>\n",
       "      <td>82</td>\n",
       "      <td>11</td>\n",
       "      <td>192</td>\n",
       "      <td>347</td>\n",
       "      <td>116</td>\n",
       "      <td>15</td>\n",
       "      <td>228</td>\n",
       "      <td>1108</td>\n",
       "      <td>6467</td>\n",
       "      <td>...</td>\n",
       "      <td>183</td>\n",
       "      <td>2575</td>\n",
       "      <td>9627</td>\n",
       "      <td>32</td>\n",
       "      <td>2</td>\n",
       "      <td>379</td>\n",
       "      <td>47</td>\n",
       "      <td>4</td>\n",
       "      <td>33</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B2449500127_n01_LICRNA04_A03</th>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>29</td>\n",
       "      <td>97</td>\n",
       "      <td>134</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>187</td>\n",
       "      <td>898</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>369</td>\n",
       "      <td>1862</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>224</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>373 rows × 17681 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              AT1G01010  AT1G01020  AT1G01030  AT1G01040  \\\n",
       "A2450525897_n01_LICRNA01_H06          0         48         34        111   \n",
       "A2450525897_n01_LICRNA01_F06         12         87         55        150   \n",
       "A2450525897_n01_LICRNA01_E01         19         27         22         41   \n",
       "A2450525897_n01_LICRNA01_H10          6         24          5         97   \n",
       "A2450525897_n01_LICRNA01_F10          6         42          6         85   \n",
       "...                                 ...        ...        ...        ...   \n",
       "B2449500127_n01_LICRNA04_E07          6         59          7        103   \n",
       "B2449500127_n01_LICRNA04_F10          0         25          0         74   \n",
       "B2449500127_n01_LICRNA04_C01          0         20         15         64   \n",
       "B2449500127_n01_LICRNA04_D12         15         82         11        192   \n",
       "B2449500127_n01_LICRNA04_A03          2          5          0         29   \n",
       "\n",
       "                              AT1G01050  AT1G01060  AT1G01070  AT1G01080  \\\n",
       "A2450525897_n01_LICRNA01_H06        343        198         10        181   \n",
       "A2450525897_n01_LICRNA01_F06        331        212         28        256   \n",
       "A2450525897_n01_LICRNA01_E01        135         89          3         56   \n",
       "A2450525897_n01_LICRNA01_H10        140         75         13         53   \n",
       "A2450525897_n01_LICRNA01_F10        215         69         12         50   \n",
       "...                                 ...        ...        ...        ...   \n",
       "B2449500127_n01_LICRNA04_E07        206         53          1        144   \n",
       "B2449500127_n01_LICRNA04_F10        178         73          0         55   \n",
       "B2449500127_n01_LICRNA04_C01         99          4          7         78   \n",
       "B2449500127_n01_LICRNA04_D12        347        116         15        228   \n",
       "B2449500127_n01_LICRNA04_A03         97        134          0         22   \n",
       "\n",
       "                              AT1G01090  AT1G01100  ...  ArthCp080  ArthCp081  \\\n",
       "A2450525897_n01_LICRNA01_H06        673       5893  ...         70        934   \n",
       "A2450525897_n01_LICRNA01_F06       1024      15009  ...         64        792   \n",
       "A2450525897_n01_LICRNA01_E01        299       3131  ...         24        395   \n",
       "A2450525897_n01_LICRNA01_H10        373       3347  ...         22        319   \n",
       "A2450525897_n01_LICRNA01_F10        368       3535  ...         29        397   \n",
       "...                                 ...        ...  ...        ...        ...   \n",
       "B2449500127_n01_LICRNA04_E07        529       3073  ...         58        911   \n",
       "B2449500127_n01_LICRNA04_F10        252       1357  ...         21        497   \n",
       "B2449500127_n01_LICRNA04_C01        210       1061  ...         12        586   \n",
       "B2449500127_n01_LICRNA04_D12       1108       6467  ...        183       2575   \n",
       "B2449500127_n01_LICRNA04_A03        187        898  ...          1        369   \n",
       "\n",
       "                              Arthcp087  ArthCr091  ArthCr090  ArthCr089  \\\n",
       "A2450525897_n01_LICRNA01_H06      10005          0          5        451   \n",
       "A2450525897_n01_LICRNA01_F06      14093          0          7        866   \n",
       "A2450525897_n01_LICRNA01_E01       4594          6         10        542   \n",
       "A2450525897_n01_LICRNA01_H10       1723          1          1        549   \n",
       "A2450525897_n01_LICRNA01_F10       2477          6          4        261   \n",
       "...                                 ...        ...        ...        ...   \n",
       "B2449500127_n01_LICRNA04_E07       4851         36          2        360   \n",
       "B2449500127_n01_LICRNA04_F10       2300          0          0        590   \n",
       "B2449500127_n01_LICRNA04_C01       2356          9          2        463   \n",
       "B2449500127_n01_LICRNA04_D12       9627         32          2        379   \n",
       "B2449500127_n01_LICRNA04_A03       1862          0          5        224   \n",
       "\n",
       "                              ArthCr088  ArthCp088  ArthCp083  ArthCp085  \n",
       "A2450525897_n01_LICRNA01_H06         53         29         30          6  \n",
       "A2450525897_n01_LICRNA01_F06         82         18          3         12  \n",
       "A2450525897_n01_LICRNA01_E01         44          0          3          5  \n",
       "A2450525897_n01_LICRNA01_H10         32          2          8          0  \n",
       "A2450525897_n01_LICRNA01_F10         15          9          4          1  \n",
       "...                                 ...        ...        ...        ...  \n",
       "B2449500127_n01_LICRNA04_E07         22         14          1          3  \n",
       "B2449500127_n01_LICRNA04_F10         13          3          0          0  \n",
       "B2449500127_n01_LICRNA04_C01          7          5          1         20  \n",
       "B2449500127_n01_LICRNA04_D12         47          4         33         25  \n",
       "B2449500127_n01_LICRNA04_A03         10          0          1          1  \n",
       "\n",
       "[373 rows x 17681 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cleaned_raw_transcriptome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "9336f67c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Axes: ylabel='Count'>"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAkQAAAGdCAYAAADzOWwgAAAAOnRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjEwLjYsIGh0dHBzOi8vbWF0cGxvdGxpYi5vcmcvq6yFwwAAAAlwSFlzAAAPYQAAD2EBqD+naQAAMglJREFUeJzt3X9UVPed//HXKEiVwI1KYJxKDCaUVTA/1mQR260mKtEtpT3uiWlpOXZrNKmJlka/dm22G5vTA4lt1e6SJsZmYxrNYU/Pxm5Om07EJiHr8TcpGyGGjV9t4g9GbDoMYBAUPt8/8uWuwy8RkQE+z8c59xzm3vfc+/nMjGdefu7n3vEYY4wAAAAsNiLSDQAAAIg0AhEAALAegQgAAFiPQAQAAKxHIAIAANYjEAEAAOsRiAAAgPUIRAAAwHpRkW7AUNHW1qbTp08rLi5OHo8n0s0BAAC9YIxRQ0ODfD6fRozofhyIQNRLp0+fVnJycqSbAQAA+uDEiROaOHFit9sJRL0UFxcn6dMXND4+PsKtAQAAvVFfX6/k5GT3e7w7BKJeaj9NFh8fTyACAGCIudx0FyZVAwAA6xGIAACA9QhEAADAegQiAABgPQIRAACwXkQD0U033SSPx9NpefjhhyV9ejOldevWyefzafTo0Zo9e7aqqqrC9tHc3KwVK1YoISFBsbGxys3N1cmTJ8NqgsGg8vPz5TiOHMdRfn6+6urqBqqbAABgkItoIDp48KBqamrcpbS0VJJ03333SZLWr1+vDRs2qLi4WAcPHpTX69W8efPU0NDg7qOgoEA7duxQSUmJdu/ercbGRuXk5Ki1tdWtycvLU0VFhfx+v/x+vyoqKpSfnz+wnQUAAIOXGUS++93vmptvvtm0tbWZtrY24/V6zZNPPuluP3/+vHEcxzz77LPGGGPq6upMdHS0KSkpcWtOnTplRowYYfx+vzHGmPfee89IMvv27XNr9u7daySZ999/v9dtC4VCRpIJhUJX200AADBAevv9PWjmELW0tGjbtm369re/LY/Ho+PHjysQCCg7O9utiYmJ0axZs7Rnzx5JUnl5uS5cuBBW4/P5lJGR4dbs3btXjuMoMzPTrZkxY4Ycx3FrutLc3Kz6+vqwBQAADE+DJhD95je/UV1dnb71rW9JkgKBgCQpKSkprC4pKcndFggENGrUKI0dO7bHmsTExE7HS0xMdGu6UlRU5M45chyH3zEDAGAYGzSB6Pnnn9eCBQvk8/nC1ne81bYx5rK33+5Y01X95fazdu1ahUIhdzlx4kRvugEAAIagQRGIPvzwQ+3atUsPPPCAu87r9UpSp1Gc2tpad9TI6/WqpaVFwWCwx5ozZ850OubZs2c7jT5dKiYmxv3dMn6/DACA4W1QBKIXXnhBiYmJ+tKXvuSuS0lJkdfrda88kz6dZ1RWVqaZM2dKkqZPn67o6OiwmpqaGlVWVro1WVlZCoVCOnDggFuzf/9+hUIhtwYAANgt4r9239bWphdeeEGLFy9WVNT/Nsfj8aigoECFhYVKTU1VamqqCgsLNWbMGOXl5UmSHMfRkiVLtGrVKo0fP17jxo3T6tWrNW3aNM2dO1eSNGXKFM2fP19Lly7V5s2bJUnLli1TTk6O0tLSBr7DAABAFy9eVHV1tfs4LS0tLAcMtIgHol27dumjjz7St7/97U7b1qxZo6amJi1fvlzBYFCZmZnauXOn4uLi3JqNGzcqKipKixYtUlNTk+bMmaOtW7dq5MiRbs327du1cuVK92q03NxcFRcXX/vOAQCALlVXV+uhp3+r6xInqrH2pJ59WEpPT49YezzGGBOxow8h9fX1chxHoVCI+UQAAFylqqoqrf51hRzfZIVOH9NP77v9mgSi3n5/D4o5RAAAAJFEIAIAANYjEAEAAOsRiAAAgPUIRAAAwHoEIgAAYD0CEQAAsB6BCAAAWI9ABAAArEcgAgAA1iMQAQAA6xGIAACA9QhEAADAegQiAABgPQIRAACwHoEIAABYj0AEAACsRyACAADWIxABAADrEYgAAID1CEQAAMB6BCIAAGA9AhEAALAegQgAAFiPQAQAAKxHIAIAANYjEAEAAOsRiAAAgPUIRAAAwHoEIgAAYD0CEQAAsB6BCAAAWI9ABAAArEcgAgAA1iMQAQAA6xGIAACA9QhEAADAegQiAABgPQIRAACwHoEIAABYL+KB6NSpU/rmN7+p8ePHa8yYMbr99ttVXl7ubjfGaN26dfL5fBo9erRmz56tqqqqsH00NzdrxYoVSkhIUGxsrHJzc3Xy5MmwmmAwqPz8fDmOI8dxlJ+fr7q6uoHoIgAAGOQiGoiCwaA+//nPKzo6Wr///e/13nvv6Wc/+5muv/56t2b9+vXasGGDiouLdfDgQXm9Xs2bN08NDQ1uTUFBgXbs2KGSkhLt3r1bjY2NysnJUWtrq1uTl5eniooK+f1++f1+VVRUKD8/fyC7CwAABqmoSB78qaeeUnJysl544QV33U033eT+bYzRpk2b9Nhjj2nhwoWSpBdffFFJSUl6+eWX9eCDDyoUCun555/XSy+9pLlz50qStm3bpuTkZO3atUv33nuvjhw5Ir/fr3379ikzM1OStGXLFmVlZam6ulppaWkD12kAADDoRHSE6NVXX9Wdd96p++67T4mJibrjjju0ZcsWd/vx48cVCASUnZ3trouJidGsWbO0Z88eSVJ5ebkuXLgQVuPz+ZSRkeHW7N27V47juGFIkmbMmCHHcdyajpqbm1VfXx+2AACA4SmigejYsWN65plnlJqaqtdff10PPfSQVq5cqV/96leSpEAgIElKSkoKe15SUpK7LRAIaNSoURo7dmyPNYmJiZ2On5iY6NZ0VFRU5M43chxHycnJV9dZAAAwaEU0ELW1temv//qvVVhYqDvuuEMPPvigli5dqmeeeSaszuPxhD02xnRa11HHmq7qe9rP2rVrFQqF3OXEiRO97RYAABhiIhqIJkyYoKlTp4atmzJlij766CNJktfrlaROozi1tbXuqJHX61VLS4uCwWCPNWfOnOl0/LNnz3YafWoXExOj+Pj4sAUAAAxPEQ1En//851VdXR227n/+5380adIkSVJKSoq8Xq9KS0vd7S0tLSorK9PMmTMlSdOnT1d0dHRYTU1NjSorK92arKwshUIhHThwwK3Zv3+/QqGQWwMAAOwV0avMvve972nmzJkqLCzUokWLdODAAT333HN67rnnJH16mqugoECFhYVKTU1VamqqCgsLNWbMGOXl5UmSHMfRkiVLtGrVKo0fP17jxo3T6tWrNW3aNPeqsylTpmj+/PlaunSpNm/eLElatmyZcnJyuMIMAABENhDddddd2rFjh9auXasnnnhCKSkp2rRpk77xjW+4NWvWrFFTU5OWL1+uYDCozMxM7dy5U3FxcW7Nxo0bFRUVpUWLFqmpqUlz5szR1q1bNXLkSLdm+/btWrlypXs1Wm5uroqLiweuswAAYNDyGGNMpBsxFNTX18txHIVCIeYTAQBwlaqqqrT61xVyfJMVOn1MP73vdqWnp/f7cXr7/R3xn+4AAACINAIRAACwHoEIAABYj0AEAACsRyACAADWIxABAADrEYgAAID1CEQAAMB6BCIAAGA9AhEAALAegQgAAFiPQAQAAKxHIAIAANYjEAEAAOsRiAAAgPUIRAAAwHoEIgAAYD0CEQAAsB6BCAAAWI9ABAAArEcgAgAA1iMQAQAA6xGIAACA9QhEAADAegQiAABgPQIRAACwHoEIAABYj0AEAACsRyACAADWIxABAADrEYgAAID1CEQAAMB6BCIAAGA9AhEAALAegQgAAFiPQAQAAKxHIAIAANYjEAEAAOsRiAAAgPUIRAAAwHoRDUTr1q2Tx+MJW7xer7vdGKN169bJ5/Np9OjRmj17tqqqqsL20dzcrBUrVighIUGxsbHKzc3VyZMnw2qCwaDy8/PlOI4cx1F+fr7q6uoGoosAAGAIiPgIUXp6umpqatzl8OHD7rb169drw4YNKi4u1sGDB+X1ejVv3jw1NDS4NQUFBdqxY4dKSkq0e/duNTY2KicnR62trW5NXl6eKioq5Pf75ff7VVFRofz8/AHtJwAAGLyiIt6AqKiwUaF2xhht2rRJjz32mBYuXChJevHFF5WUlKSXX35ZDz74oEKhkJ5//nm99NJLmjt3riRp27ZtSk5O1q5du3TvvffqyJEj8vv92rdvnzIzMyVJW7ZsUVZWlqqrq5WWljZwnQUAAINSxEeIPvjgA/l8PqWkpOhrX/uajh07Jkk6fvy4AoGAsrOz3dqYmBjNmjVLe/bskSSVl5frwoULYTU+n08ZGRluzd69e+U4jhuGJGnGjBlyHMet6Upzc7Pq6+vDFgAAMDxFNBBlZmbqV7/6lV5//XVt2bJFgUBAM2fO1Mcff6xAICBJSkpKCntOUlKSuy0QCGjUqFEaO3ZsjzWJiYmdjp2YmOjWdKWoqMidc+Q4jpKTk6+qrwAAYPCKaCBasGCB/v7v/17Tpk3T3Llz9bvf/U7Sp6fG2nk8nrDnGGM6reuoY01X9Zfbz9q1axUKhdzlxIkTveoTAAAYeiJ+yuxSsbGxmjZtmj744AN3XlHHUZza2lp31Mjr9aqlpUXBYLDHmjNnznQ61tmzZzuNPl0qJiZG8fHxYQsAABieBlUgam5u1pEjRzRhwgSlpKTI6/WqtLTU3d7S0qKysjLNnDlTkjR9+nRFR0eH1dTU1KiystKtycrKUigU0oEDB9ya/fv3KxQKuTUAAMBuEb3KbPXq1fryl7+sG2+8UbW1tfrxj3+s+vp6LV68WB6PRwUFBSosLFRqaqpSU1NVWFioMWPGKC8vT5LkOI6WLFmiVatWafz48Ro3bpxWr17tnoKTpClTpmj+/PlaunSpNm/eLElatmyZcnJyuMIMAABIinAgOnnypL7+9a/rz3/+s2644QbNmDFD+/bt06RJkyRJa9asUVNTk5YvX65gMKjMzEzt3LlTcXFx7j42btyoqKgoLVq0SE1NTZozZ462bt2qkSNHujXbt2/XypUr3avRcnNzVVxcPLCdBQAAg5bHGGMi3YihoL6+Xo7jKBQKMZ8IAICrVFVVpdW/rpDjm6zQ6WP66X23Kz09vd+P09vv70E1hwgAACASCEQAAMB6BCIAAGA9AhEAALAegQgAAFiPQAQAAKxHIAIAANYjEAEAAOsRiAAAgPUIRAAAwHoEIgAAYD0CEQAAsB6BCAAAWI9ABAAArEcgAgAA1iMQAQAA6xGIAACA9QhEAADAegQiAABgPQIRAACwHoEIAABYj0AEAACsRyACAADWIxABAADrEYgAAID1CEQAAMB6BCIAAGA9AhEAALAegQgAAFiPQAQAAKxHIAIAANYjEAEAAOsRiAAAgPUIRAAAwHoEIgAAYD0CEQAAsB6BCAAAWI9ABAAArEcgAgAA1iMQAQAA6w2aQFRUVCSPx6OCggJ3nTFG69atk8/n0+jRozV79mxVVVWFPa+5uVkrVqxQQkKCYmNjlZubq5MnT4bVBINB5efny3EcOY6j/Px81dXVDUCvAADAUDAoAtHBgwf13HPP6dZbbw1bv379em3YsEHFxcU6ePCgvF6v5s2bp4aGBremoKBAO3bsUElJiXbv3q3Gxkbl5OSotbXVrcnLy1NFRYX8fr/8fr8qKiqUn58/YP0DAACDW8QDUWNjo77xjW9oy5YtGjt2rLveGKNNmzbpscce08KFC5WRkaEXX3xRn3zyiV5++WVJUigU0vPPP6+f/exnmjt3ru644w5t27ZNhw8f1q5duyRJR44ckd/v1y9/+UtlZWUpKytLW7Zs0W9/+1tVV1dHpM8AAGBwiXggevjhh/WlL31Jc+fODVt//PhxBQIBZWdnu+tiYmI0a9Ys7dmzR5JUXl6uCxcuhNX4fD5lZGS4NXv37pXjOMrMzHRrZsyYIcdx3JquNDc3q76+PmwBAADDU1QkD15SUqJ33nlHBw8e7LQtEAhIkpKSksLWJyUl6cMPP3RrRo0aFTay1F7T/vxAIKDExMRO+09MTHRrulJUVKQf/ehHV9YhAAAwJPVphGjy5Mn6+OOPO62vq6vT5MmTe7WPEydO6Lvf/a62bdumz3zmM93WeTyesMfGmE7rOupY01X95fazdu1ahUIhdzlx4kSPxwQAAENXnwLRn/70p7BJy+2am5t16tSpXu2jvLxctbW1mj59uqKiohQVFaWysjL9y7/8i6KiotyRoY6jOLW1te42r9erlpYWBYPBHmvOnDnT6fhnz57tNPp0qZiYGMXHx4ctAABgeLqiU2avvvqq+/frr78ux3Hcx62trfrDH/6gm266qVf7mjNnjg4fPhy27h/+4R/0V3/1V/r+97+vyZMny+v1qrS0VHfccYckqaWlRWVlZXrqqackSdOnT1d0dLRKS0u1aNEiSVJNTY0qKyu1fv16SVJWVpZCoZAOHDigv/mbv5Ek7d+/X6FQSDNnzryS7gMAgGHqigLRV7/6VUmfnoJavHhx2Lbo6GjddNNN+tnPftarfcXFxSkjIyNsXWxsrMaPH++uLygoUGFhoVJTU5WamqrCwkKNGTNGeXl5kiTHcbRkyRKtWrVK48eP17hx47R69WpNmzbNnaQ9ZcoUzZ8/X0uXLtXmzZslScuWLVNOTo7S0tKupPsAAGCYuqJA1NbWJklKSUnRwYMHlZCQcE0a1W7NmjVqamrS8uXLFQwGlZmZqZ07dyouLs6t2bhxo6KiorRo0SI1NTVpzpw52rp1q0aOHOnWbN++XStXrnSvRsvNzVVxcfE1bTsAABg6PMYYE+lGDAX19fVyHEehUIj5RAAAXKWqqiqt/nWFHN9khU4f00/vu13p6en9fpzefn/3+bL7P/zhD/rDH/6g2tpad+So3b/927/1dbcAAAADrk+B6Ec/+pGeeOIJ3XnnnZowYcJlL4MHAAAYzPoUiJ599llt3bqV3wMDAADDQp/uQ9TS0sIl6wAAYNjoUyB64IEH3B9YBQAAGOr6dMrs/Pnzeu6557Rr1y7deuutio6ODtu+YcOGfmkcAADAQOhTIHr33Xd1++23S5IqKyvDtjHBGgAADDV9CkRvvvlmf7cDAAAgYvo0hwgAAGA46dMI0d13393jqbE33nijzw0CAAAYaH0KRO3zh9pduHBBFRUVqqys7PSjrwAAAINdnwLRxo0bu1y/bt06NTY2XlWDAAAABlq/ziH65je/ye+YAQCAIadfA9HevXv1mc98pj93CQAAcM316ZTZwoULwx4bY1RTU6NDhw7phz/8Yb80DAAAYKD0KRA5jhP2eMSIEUpLS9MTTzyh7OzsfmkYAADAQOlTIHrhhRf6ux0AAAAR06dA1K68vFxHjhyRx+PR1KlTdccdd/RXuwAAAAZMnwJRbW2tvva1r+mtt97S9ddfL2OMQqGQ7r77bpWUlOiGG27o73YCAABcM326ymzFihWqr69XVVWV/vKXvygYDKqyslL19fVauXJlf7cRAADgmurTCJHf79euXbs0ZcoUd93UqVP19NNPM6kaAAAMOX0aIWpra1N0dHSn9dHR0Wpra7vqRgEAAAykPgWie+65R9/97nd1+vRpd92pU6f0ve99T3PmzOm3xgEAAAyEPgWi4uJiNTQ06KabbtLNN9+sW265RSkpKWpoaNC//uu/9ncbAQAArqk+zSFKTk7WO++8o9LSUr3//vsyxmjq1KmaO3duf7cPAADgmruiEaI33nhDU6dOVX19vSRp3rx5WrFihVauXKm77rpL6enp+q//+q9r0lAAAIBr5YoC0aZNm7R06VLFx8d32uY4jh588EFt2LCh3xoHAAAwEK4oEP33f/+35s+f3+327OxslZeXX3WjAAAABtIVBaIzZ850ebl9u6ioKJ09e/aqGwUAADCQrigQffazn9Xhw4e73f7uu+9qwoQJV90oAACAgXRFgejv/u7v9M///M86f/58p21NTU16/PHHlZOT02+NAwAAGAhXdNn9P/3TP+mVV17R5z73OT3yyCNKS0uTx+PRkSNH9PTTT6u1tVWPPfbYtWorAADANXFFgSgpKUl79uzRd77zHa1du1bGGEmSx+PRvffeq1/84hdKSkq6Jg0FAAC4Vq74xoyTJk3Sa6+9pmAwqKNHj8oYo9TUVI0dO/ZatA8AAOCa69OdqiVp7Nixuuuuu/qzLQAAABHRp98yAwAAGE4IRAAAwHoEIgAAYD0CEQAAsB6BCAAAWC+igeiZZ57Rrbfeqvj4eMXHxysrK0u///3v3e3GGK1bt04+n0+jR4/W7NmzVVVVFbaP5uZmrVixQgkJCYqNjVVubq5OnjwZVhMMBpWfny/HceQ4jvLz81VXVzcQXQQAAENARAPRxIkT9eSTT+rQoUM6dOiQ7rnnHn3lK19xQ8/69eu1YcMGFRcX6+DBg/J6vZo3b54aGhrcfRQUFGjHjh0qKSnR7t271djYqJycHLW2tro1eXl5qqiokN/vl9/vV0VFhfLz8we8vwAAYJAyg8zYsWPNL3/5S9PW1ma8Xq958skn3W3nz583juOYZ5991hhjTF1dnYmOjjYlJSVuzalTp8yIESOM3+83xhjz3nvvGUlm3759bs3evXuNJPP+++/3ul2hUMhIMqFQ6Gq7CACA9SorK838x7eZ+zfvMfMf32YqKyuvyXF6+/09aOYQtba2qqSkROfOnVNWVpaOHz+uQCCg7OxstyYmJkazZs3Snj17JEnl5eW6cOFCWI3P51NGRoZbs3fvXjmOo8zMTLdmxowZchzHrelKc3Oz6uvrwxYAADA8RTwQHT58WNddd51iYmL00EMPaceOHZo6daoCgYAkdfpttKSkJHdbIBDQqFGjOv1sSMeaxMTETsdNTEx0a7pSVFTkzjlyHEfJyclX1U8AADB4RTwQpaWlqaKiQvv27dN3vvMdLV68WO+995673ePxhNUbYzqt66hjTVf1l9vP2rVrFQqF3OXEiRO97RIAABhiIh6IRo0apVtuuUV33nmnioqKdNttt+nnP/+5vF6vJHUaxamtrXVHjbxer1paWhQMBnusOXPmTKfjnj17ttPo06ViYmLcq9/aFwAAMDxFPBB1ZIxRc3OzUlJS5PV6VVpa6m5raWlRWVmZZs6cKUmaPn26oqOjw2pqampUWVnp1mRlZSkUCunAgQNuzf79+xUKhdwaAABgtz7/2n1/+MEPfqAFCxYoOTlZDQ0NKikp0VtvvSW/3y+Px6OCggIVFhYqNTVVqampKiws1JgxY5SXlydJchxHS5Ys0apVqzR+/HiNGzdOq1ev1rRp0zR37lxJ0pQpUzR//nwtXbpUmzdvliQtW7ZMOTk5SktLi1jfAQDA4BHRQHTmzBnl5+erpqZGjuPo1ltvld/v17x58yRJa9asUVNTk5YvX65gMKjMzEzt3LlTcXFx7j42btyoqKgoLVq0SE1NTZozZ462bt2qkSNHujXbt2/XypUr3avRcnNzVVxcPLCdBQAAg5bHGGMi3YihoL6+Xo7jKBQKMZ8IAICrVFVVpdW/rpDjm6zQ6WP66X23Kz09vd+P09vv70E3hwgAAGCgEYgAAID1CEQAAMB6BCIAAGA9AhEAALAegQgAAFiPQAQAAKxHIAIAANYjEAEAAOsRiAAAgPUIRAAAwHoEIgAAYD0CEQAAsB6BCAAAWI9ABAAArEcgAgAA1iMQAQAA6xGIAACA9QhEAADAegQiAABgPQIRAACwHoEIAABYj0AEAACsRyACAADWIxABAADrEYgAAID1CEQAAMB6BCIAAGA9AhEAALAegQgAAFiPQAQAAKxHIAIAANYjEAEAAOsRiAAAgPUIRAAAwHoEIgAAYD0CEQAAsB6BCAAAWI9ABAAArEcgAgAA1otoICoqKtJdd92luLg4JSYm6qtf/aqqq6vDaowxWrdunXw+n0aPHq3Zs2erqqoqrKa5uVkrVqxQQkKCYmNjlZubq5MnT4bVBINB5efny3EcOY6j/Px81dXVXesuAgCAISCigaisrEwPP/yw9u3bp9LSUl28eFHZ2dk6d+6cW7N+/Xpt2LBBxcXFOnjwoLxer+bNm6eGhga3pqCgQDt27FBJSYl2796txsZG5eTkqLW11a3Jy8tTRUWF/H6//H6/KioqlJ+fP6D9BQAAg5PHGGMi3Yh2Z8+eVWJiosrKyvTFL35Rxhj5fD4VFBTo+9//vqRPR4OSkpL01FNP6cEHH1QoFNINN9ygl156Sffff78k6fTp00pOTtZrr72me++9V0eOHNHUqVO1b98+ZWZmSpL27dunrKwsvf/++0pLS7ts2+rr6+U4jkKhkOLj46/diwAAgAWqqqq0+tcVcnyTFTp9TD+973alp6f3+3F6+/09qOYQhUIhSdK4ceMkScePH1cgEFB2drZbExMTo1mzZmnPnj2SpPLycl24cCGsxufzKSMjw63Zu3evHMdxw5AkzZgxQ47juDUdNTc3q76+PmwBAADD06AJRMYYPfroo/rCF76gjIwMSVIgEJAkJSUlhdUmJSW52wKBgEaNGqWxY8f2WJOYmNjpmImJiW5NR0VFRe58I8dxlJycfHUdBADAchcvXlRVVZWqqqp09OhRDaKTVIqKdAPaPfLII3r33Xe1e/fuTts8Hk/YY2NMp3Uddazpqr6n/axdu1aPPvqo+7i+vp5QBADAVaiurtZDT/9W1yVO1Jn3yxU/qf9PkfXVoBghWrFihV599VW9+eabmjhxorve6/VKUqdRnNraWnfUyOv1qqWlRcFgsMeaM2fOdDru2bNnO40+tYuJiVF8fHzYAgAArs51iRPl+CZrzLiuv38jJaKByBijRx55RK+88oreeOMNpaSkhG1PSUmR1+tVaWmpu66lpUVlZWWaOXOmJGn69OmKjo4Oq6mpqVFlZaVbk5WVpVAopAMHDrg1+/fvVygUcmsAAIC9InrK7OGHH9bLL7+s//zP/1RcXJw7EuQ4jkaPHi2Px6OCggIVFhYqNTVVqampKiws1JgxY5SXl+fWLlmyRKtWrdL48eM1btw4rV69WtOmTdPcuXMlSVOmTNH8+fO1dOlSbd68WZK0bNky5eTk9OoKMwAAMLxFNBA988wzkqTZs2eHrX/hhRf0rW99S5K0Zs0aNTU1afny5QoGg8rMzNTOnTsVFxfn1m/cuFFRUVFatGiRmpqaNGfOHG3dulUjR450a7Zv366VK1e6V6Pl5uaquLj42nYQAAAMCYPqPkSDGfchAgDg6lx676GTFW8rKi5B3punch8iAACAwYBABAAArEcgAgAA1iMQAQAA6xGIAACA9QhEAADAegQiAABgPQIRAACwHoEIAABYj0AEAACsRyACAADWIxABAADrEYgAAID1CEQAAMB6BCIAAGA9AhEAALAegQgAAFiPQAQAAKxHIAIAANYjEAEAAOsRiAAAgPUIRAAAwHoEIgAAYD0CEQAAsB6BCAAAWI9ABAAArEcgAgAA1iMQAQAA6xGIAACA9QhEAADAegQiAABgPQIRAACwHoEIAABYj0AEAACsRyACAADWIxABAADrEYgAAID1CEQAAMB6BCIAAGA9AhEAALBeRAPR22+/rS9/+cvy+XzyeDz6zW9+E7bdGKN169bJ5/Np9OjRmj17tqqqqsJqmpubtWLFCiUkJCg2Nla5ubk6efJkWE0wGFR+fr4cx5HjOMrPz1ddXd017h0AABgqIhqIzp07p9tuu03FxcVdbl+/fr02bNig4uJiHTx4UF6vV/PmzVNDQ4NbU1BQoB07dqikpES7d+9WY2OjcnJy1Nra6tbk5eWpoqJCfr9ffr9fFRUVys/Pv+b9AwAAQ0NUJA++YMECLViwoMttxhht2rRJjz32mBYuXChJevHFF5WUlKSXX35ZDz74oEKhkJ5//nm99NJLmjt3riRp27ZtSk5O1q5du3TvvffqyJEj8vv92rdvnzIzMyVJW7ZsUVZWlqqrq5WWljYwnQUAAIPWoJ1DdPz4cQUCAWVnZ7vrYmJiNGvWLO3Zs0eSVF5ergsXLoTV+Hw+ZWRkuDV79+6V4zhuGJKkGTNmyHEct6Yrzc3Nqq+vD1sAAMDwNGgDUSAQkCQlJSWFrU9KSnK3BQIBjRo1SmPHju2xJjExsdP+ExMT3ZquFBUVuXOOHMdRcnLyVfUHAAAMXoM2ELXzeDxhj40xndZ11LGmq/rL7Wft2rUKhULucuLEiStsOQAAGCoGbSDyer2S1GkUp7a21h018nq9amlpUTAY7LHmzJkznfZ/9uzZTqNPl4qJiVF8fHzYAgAAhqdBG4hSUlLk9XpVWlrqrmtpaVFZWZlmzpwpSZo+fbqio6PDampqalRZWenWZGVlKRQK6cCBA27N/v37FQqF3BoAAGC3iF5l1tjYqKNHj7qPjx8/roqKCo0bN0433nijCgoKVFhYqNTUVKWmpqqwsFBjxoxRXl6eJMlxHC1ZskSrVq3S+PHjNW7cOK1evVrTpk1zrzqbMmWK5s+fr6VLl2rz5s2SpGXLliknJ4crzAAAgKQIB6JDhw7p7rvvdh8/+uijkqTFixdr69atWrNmjZqamrR8+XIFg0FlZmZq586diouLc5+zceNGRUVFadGiRWpqatKcOXO0detWjRw50q3Zvn27Vq5c6V6Nlpub2+29jwAAgH08xhgT6UYMBfX19XIcR6FQiPlEAAD0QVVVlVb/ukKOb7JOVrytqLgEeW+eqtDpY/rpfbcrPT2934/Z2+/vQTuHCAAAYKAQiAAAgPUIRAAAwHoRnVQNAACGl4sXL6q6utr9W5Kioj6NG0ePHtVgnbpMIAIAAP2murpaDz39W12XOFFn3i/XyNjrlZB8syTpzPvlip/U/xOn+wOBCAAA9KvrEifK8U1WQ+1JRcUlyPFNliQ11J6McMu6xxwiAABgPQIRAACwHoEIAABYj0AEAACsRyACAADWIxABAADrEYgAAID1CEQAAMB6BCIAAGA9AhEAALAegQgAAFiPQAQAAKxHIAIAANbj1+4BAMAVuXjxoqqrq8PWpaWlKSpq6MaKodtyAAAQEdXV1Xro6d/qusSJkqTG2pN69mEpPT09wi3rOwIRAAC4YtclTpTjmxzpZvQbAhEAALisS0+THT16VMaYCLeofxGIAADAZV16muzM++WKn/S/p8fa2lp19OhRSUM3LBGIAABAr7SfJmuoPRm2/tyfa/TjV/+khOTGTmFpqOCyewAAcNViEz4rxzdZY8YlRbopfUIgAgAA1uOUGQAAkDQ87y/UW8O/hwAAoFeG4/2FeotABACARS43CjTc7i/UWwQiAAAsYvMoUE8IRAAAWKa3o0DD4f5CvUUgAgBgmOvrXaaHw/2FeotABADAMNDT3KCruct0+/2FOt6McbghEAEAMAxcbm7QcL7LdH8gEAEAMIh1HPm5ePGiJLlXhfXHFWK2jAL1hEAEAEA3OoaRvtyk8NJ99BRmujtWx5GfM++Xa2Ts9UpIvlkNgQ/1fxak65Zbbul0usumCdH9gUAEAEA3Lg0jl4aPdu2hpafg1HH+TnuY6XhK69K67k53SVJD7UlFxSW4Izo/fvW/uzzdxamwK0MgAgAMab0dxemurrc3Krw0fEjhc3R6CjMd99EeZrrSXnclozs9ne7iVFjvEYgAAP2iL6eXejqd1Nt99DSK091ITXdhRur5RoXtAUPqfEoq9obPdgoz7du6CjQ91TG6M/CsCkS/+MUv9JOf/EQ1NTVKT0/Xpk2b9Ld/+7eRbhYAy/Vl5KKnIHHp477MWenrcY8ePaqf+N9XXFJ44OhpUvClz7n0dJKksHDTU58uDSOXjuJ0DEfdhZZL10u9n3vTXWi5dL2kbgPN5eoY3RlY1gSif//3f1dBQYF+8Ytf6POf/7w2b96sBQsW6L333tONN94Y6eYBGGT6MhG2py/tnvbR25GLjhNouwsSlz7ubgKuFB5gutv3lR43flJ6l4Hj0v11bF/7czqeTuo4P6arPl163HaXBolLT3F1F1quZu5Nd6Hl0pGkngJNb+tw7VkTiDZs2KAlS5bogQcekCRt2rRJr7/+up555hkVFRVFuHUArlZvg4l05aMdvQ0Vl/vS7mkfvRm56GoCbXdB4tJJt11NwG1v06X76GrfV3rcdl2Fiu7a15NLA0dXfep43O6e31Udc29wKSsCUUtLi8rLy/WP//iPYeuzs7O1Z8+eLp/T3Nys5uZm93EoFJIk1dfX93v7jhw50u/7BGxz7NgxFf77Wxp9/Q0KflStEZ+Jk5PoC/tbkprqzuoH98/W5MmTu32OJAU/qlbcZ1M1+vpxar3QLLWc18XmT9Rw9pTWbD562bpL/5bUq320tjSp9n8qtOZQk9v29vWS1HDmI40c47j7rz/9fxU9ou3T9Q0hRY9o+9+6//+409////ntbQrbRxf7vqrj9rS/7trX235Qd83rBvJYjWdPq7HxlmvyHdu+z8vddsCKQPTnP/9Zra2tSkpKCluflJSkQCDQ5XOKior0ox/9qNP65OTka9JGAAMn7z+fi3QTAHQw4/l113T/DQ0Nchyn2+1WBKJ2Ho8n7LExptO6dmvXrtWjjz7qPm5ra9Nf/vIXjR8/vtvnDHX19fVKTk7WiRMnFB8fH+nmRITtrwH9t7v/Eq+B7f2Xht9rYIxRQ0ODfD5fj3VWBKKEhASNHDmy02hQbW1tp1GjdjExMYqJiQlbd/3111+rJg4q8fHxw+IfwdWw/TWg/3b3X+I1sL3/0vB6DXoaGWo3YgDaEXGjRo3S9OnTVVpaGra+tLRUM2fOjFCrAADAYGHFCJEkPfroo8rPz9edd96prKwsPffcc/roo4/00EMPRbppAAAgwqwJRPfff78+/vhjPfHEE6qpqVFGRoZee+01TZo0KdJNGzRiYmL0+OOPdzpVaBPbXwP6b3f/JV4D2/sv2fsaeAw/fwsAACxnxRwiAACAnhCIAACA9QhEAADAegQiAABgPQKRZdatWyePxxO2eL1ed7sxRuvWrZPP59Po0aM1e/ZsVVVVRbDFV+/tt9/Wl7/8Zfl8Pnk8Hv3mN78J296bPjc3N2vFihVKSEhQbGyscnNzdfLk0PjRx8v1/1vf+lanz8SMGTPCaoZy/4uKinTXXXcpLi5OiYmJ+upXv+r+oGu74fwZ6E3/h/tn4JlnntGtt97q3mgwKytLv//9793tw/n9ly7f/+H+/vcWgchC6enpqqmpcZfDhw+729avX68NGzaouLhYBw8elNfr1bx589TQ0BDBFl+dc+fO6bbbblNxcXGX23vT54KCAu3YsUMlJSXavXu3GhsblZOTo9bW1oHqRp9drv+SNH/+/LDPxGuvvRa2fSj3v6ysTA8//LD27dun0tJSXbx4UdnZ2Tp37pxbM5w/A73pvzS8PwMTJ07Uk08+qUOHDunQoUO655579JWvfMUNPcP5/Zcu339peL//vWZglccff9zcdtttXW5ra2szXq/XPPnkk+668+fPG8dxzLPPPjtALby2JJkdO3a4j3vT57q6OhMdHW1KSkrcmlOnTpkRI0YYv98/YG3vDx37b4wxixcvNl/5yle6fc5w6r8xxtTW1hpJpqyszBhj32egY/+Nse8zYIwxY8eONb/85S+te//btfffGDvf/64wQmShDz74QD6fTykpKfra176mY8eOSZKOHz+uQCCg7OxstzYmJkazZs3Snj17ItXca6o3fS4vL9eFCxfCanw+nzIyMobN6/LWW28pMTFRn/vc57R06VLV1ta624Zb/0OhkCRp3Lhxkuz7DHTsfztbPgOtra0qKSnRuXPnlJWVZd3737H/7Wx5/3tizZ2q8anMzEz96le/0uc+9zmdOXNGP/7xjzVz5kxVVVW5P37b8Qdvk5KS9OGHH0aiuddcb/ocCAQ0atQojR07tlNNxx8MHooWLFig++67T5MmTdLx48f1wx/+UPfcc4/Ky8sVExMzrPpvjNGjjz6qL3zhC8rIyJBk12egq/5LdnwGDh8+rKysLJ0/f17XXXedduzYoalTp7pf6MP9/e+u/5Id739vEIgss2DBAvfvadOmKSsrSzfffLNefPFFdxKdx+MJe44xptO64aYvfR4ur8v999/v/p2RkaE777xTkyZN0u9+9zstXLiw2+cNxf4/8sgjevfdd7V79+5O22z4DHTXfxs+A2lpaaqoqFBdXZ3+4z/+Q4sXL1ZZWZm7fbi//931f+rUqVa8/73BKTPLxcbGatq0afrggw/cq806Jv7a2tpO/3saLnrTZ6/Xq5aWFgWDwW5rhpMJEyZo0qRJ+uCDDyQNn/6vWLFCr776qt58801NnDjRXW/LZ6C7/ndlOH4GRo0apVtuuUV33nmnioqKdNttt+nnP/+5Ne9/d/3vynB8/3uDQGS55uZmHTlyRBMmTFBKSoq8Xq9KS0vd7S0tLSorK9PMmTMj2Mprpzd9nj59uqKjo8NqampqVFlZOSxfl48//lgnTpzQhAkTJA39/htj9Mgjj+iVV17RG2+8oZSUlLDtw/0zcLn+d2W4fQa6YoxRc3PzsH//u9Pe/67Y8P53aeDncSOSVq1aZd566y1z7Ngxs2/fPpOTk2Pi4uLMn/70J2OMMU8++aRxHMe88sor5vDhw+brX/+6mTBhgqmvr49wy/uuoaHB/PGPfzR//OMfjSSzYcMG88c//tF8+OGHxpje9fmhhx4yEydONLt27TLvvPOOueeee8xtt91mLl68GKlu9VpP/W9oaDCrVq0ye/bsMcePHzdvvvmmycrKMp/97GeHTf+/853vGMdxzFtvvWVqamrc5ZNPPnFrhvNn4HL9t+EzsHbtWvP222+b48ePm3fffdf84Ac/MCNGjDA7d+40xgzv99+Ynvtvw/vfWwQiy9x///1mwoQJJjo62vh8PrNw4UJTVVXlbm9razOPP/648Xq9JiYmxnzxi180hw8fjmCLr96bb75pJHVaFi9ebIzpXZ+bmprMI488YsaNG2dGjx5tcnJyzEcffRSB3ly5nvr/ySefmOzsbHPDDTeY6Ohoc+ONN5rFixd36ttQ7n9XfZdkXnjhBbdmOH8GLtd/Gz4D3/72t82kSZPMqFGjzA033GDmzJnjhiFjhvf7b0zP/bfh/e8tjzHGDNx4FAAAwODDHCIAAGA9AhEAALAegQgAAFiPQAQAAKxHIAIAANYjEAEAAOsRiAAAgPUIRAAAwHoEIgAAYD0CEQAAsB6BCAAAWI9ABAAArPf/ALzakZ/4elL+AAAAAElFTkSuQmCC",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sns.histplot((cleaned_raw_transcriptome > 1).sum())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "64ea0f40",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "sampID\n",
       "LIC030    33.6\n",
       "LIC125    22.3\n",
       "LIC603    18.5\n",
       "LIC078    22.4\n",
       "LIC007    25.9\n",
       "          ... \n",
       "LIC626     0.5\n",
       "LIC387     0.5\n",
       "LIC449     0.5\n",
       "LIC056     0.5\n",
       "LIC055     1.0\n",
       "Name: rare_species_total_abundR100, Length: 333, dtype: float64"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rare_abund_by_sample"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "0a4c0a71",
   "metadata": {},
   "outputs": [],
   "source": [
    "def calculate_differential_expression_btwn_2_groups(\n",
    "    list_of_normal_samples: list,\n",
    "    list_of_perturbed_samples: list,\n",
    "    raw_transcriptome_data: pd.DataFrame = cleaned_raw_transcriptome,\n",
    "    og_metadata: pd.DataFrame = trimmed_metadata,\n",
    "    name_of_perturbation_column: str = \"perturbation\",\n",
    "    use_plate_for_design: bool = False,\n",
    "):\n",
    "\n",
    "    ##Print out info for diagnosing any issues:\n",
    "\n",
    "    print(f\"Number of normal samples: {len(list_of_normal_samples)}\")\n",
    "    print(f\"Number of perturbed samples: {len(list_of_perturbed_samples)}\")\n",
    "\n",
    "    # Subset the transcriptome data\n",
    "    normal_samples = raw_transcriptome_data.loc[list_of_normal_samples]\n",
    "    perturbed_samples = raw_transcriptome_data.loc[list_of_perturbed_samples]\n",
    "    full_transcriptome = pd.concat([normal_samples, perturbed_samples])\n",
    "\n",
    "    ##generate the metadata object\n",
    "    normal_meta_data = pd.DataFrame(index=normal_samples.index)\n",
    "    perturbed_meta_data = pd.DataFrame(index=perturbed_samples.index)\n",
    "    normal_meta_data[name_of_perturbation_column] = \"normal\"\n",
    "    perturbed_meta_data[name_of_perturbation_column] = \"perturbed\"\n",
    "\n",
    "    ### Get plate info from the metadata\n",
    "    normal_meta_data[\"plate\"] = og_metadata.loc[normal_samples.index, \"rnaprepplate\"]\n",
    "    perturbed_meta_data[\"plate\"] = og_metadata.loc[\n",
    "        perturbed_samples.index, \"rnaprepplate\"\n",
    "    ]\n",
    "\n",
    "    # Assemble metadata\n",
    "    full_metadata = pd.concat([normal_meta_data, perturbed_meta_data])\n",
    "\n",
    "    inference = pydeseq2.default_inference.DefaultInference(n_cpus=6)\n",
    "    dds = pydeseq2.dds.DeseqDataSet(\n",
    "        counts=full_transcriptome,\n",
    "        metadata=full_metadata,\n",
    "        design=f\"~{name_of_perturbation_column}\",\n",
    "        refit_cooks=True,\n",
    "        inference=inference,\n",
    "    )\n",
    "    if use_plate_for_design:\n",
    "        dds = pydeseq2.dds.DeseqDataSet(\n",
    "            counts=full_transcriptome,\n",
    "            metadata=full_metadata,\n",
    "            design=f\"~{name_of_perturbation_column}+plate\",\n",
    "            refit_cooks=True,\n",
    "            inference=inference,\n",
    "        )\n",
    "    dds.deseq2()\n",
    "    ds = pydeseq2.ds.DeseqStats(\n",
    "        dds,\n",
    "        contrast=[name_of_perturbation_column, \"perturbed\", \"normal\"],\n",
    "        inference=inference,\n",
    "    )\n",
    "    ds.summary()\n",
    "    return ds"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "d7db1076",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "==================================================\n",
      "Processing timepoint: t14\n",
      "==================================================\n",
      "Number of normal samples: 5\n",
      "Number of perturbed samples: 2\n",
      "Using None as control genes, passed at DeseqDataSet initialization\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Fitting size factors...\n",
      "... done in 0.00 seconds.\n",
      "\n",
      "Fitting dispersions...\n",
      "... done in 3.31 seconds.\n",
      "\n",
      "Fitting dispersion trend curve...\n",
      "... done in 0.42 seconds.\n",
      "\n",
      "Fitting MAP dispersions...\n",
      "... done in 3.64 seconds.\n",
      "\n",
      "Fitting LFCs...\n",
      "... done in 2.88 seconds.\n",
      "\n",
      "Calculating cook's distance...\n",
      "... done in 0.01 seconds.\n",
      "\n",
      "Replacing 0 outlier genes.\n",
      "\n",
      "Running Wald tests...\n",
      "... done in 0.95 seconds.\n",
      "\n",
      "Fitting size factors...\n",
      "... done in 0.01 seconds.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log2 fold change & Wald test p-value: rare_species_greater_5 perturbed vs normal\n",
      "              baseMean  log2FoldChange     lfcSE      stat    pvalue      padj\n",
      "AT1G01010    23.042526       -0.438901  0.783892 -0.559900  0.575548  0.906801\n",
      "AT1G01020    84.253266        0.225365  0.436973  0.515742  0.606035  0.915891\n",
      "AT1G01030    44.017698       -0.068202  0.493065 -0.138322  0.889986  0.980893\n",
      "AT1G01040   170.361128        0.132467  0.312117  0.424414  0.671264  0.934509\n",
      "AT1G01050   255.120935       -0.146770  0.305053 -0.481129  0.630425  0.923509\n",
      "...                ...             ...       ...       ...       ...       ...\n",
      "ArthCr089  2316.079433       -0.131242  0.447349 -0.293377  0.769234  0.956143\n",
      "ArthCr088   210.774282        0.364102  0.491273  0.741141  0.458608  0.860778\n",
      "ArthCp088     9.797099       -0.503911  1.254607 -0.401648  0.687943       NaN\n",
      "ArthCp083    18.783767       -1.119463  0.836596 -1.338116  0.180859  0.696347\n",
      "ArthCp085    10.738686        2.215972  0.981429  2.257903  0.023952  0.382626\n",
      "\n",
      "[17681 rows x 6 columns]\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: t15\n",
      "==================================================\n",
      "Number of normal samples: 6\n",
      "Number of perturbed samples: 2\n",
      "Using None as control genes, passed at DeseqDataSet initialization\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Fitting dispersions...\n",
      "... done in 3.30 seconds.\n",
      "\n",
      "Fitting dispersion trend curve...\n",
      "... done in 0.42 seconds.\n",
      "\n",
      "Fitting MAP dispersions...\n",
      "... done in 3.73 seconds.\n",
      "\n",
      "Fitting LFCs...\n",
      "... done in 2.41 seconds.\n",
      "\n",
      "Calculating cook's distance...\n",
      "... done in 0.01 seconds.\n",
      "\n",
      "Replacing 0 outlier genes.\n",
      "\n",
      "Running Wald tests...\n",
      "... done in 0.93 seconds.\n",
      "\n",
      "Fitting size factors...\n",
      "... done in 0.01 seconds.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log2 fold change & Wald test p-value: rare_species_greater_5 perturbed vs normal\n",
      "              baseMean  log2FoldChange     lfcSE      stat    pvalue      padj\n",
      "AT1G01010    11.080991        1.815010  1.143715  1.586943  0.112525  0.998868\n",
      "AT1G01020    63.765101        0.009388  0.418337  0.022441  0.982096  0.998868\n",
      "AT1G01030    18.677859        0.364702  0.676238  0.539311  0.589672  0.998868\n",
      "AT1G01040   149.392352        0.148930  0.261896  0.568661  0.569586  0.998868\n",
      "AT1G01050   194.570799       -0.258307  0.301717 -0.856123  0.391930  0.998868\n",
      "...                ...             ...       ...       ...       ...       ...\n",
      "ArthCr089  2355.193560        0.548500  0.260441  2.106040  0.035201  0.998868\n",
      "ArthCr088   166.575863        0.659517  0.319527  2.064040  0.039014  0.998868\n",
      "ArthCp088    15.402725       -0.080626  0.833250 -0.096761  0.922916  0.998868\n",
      "ArthCp083    37.411412       -0.376246  0.623310 -0.603627  0.546092  0.998868\n",
      "ArthCp085     7.482260       -1.165388  1.218454 -0.956448  0.338846  0.998868\n",
      "\n",
      "[17681 rows x 6 columns]\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: t17\n",
      "==================================================\n",
      "Skipping timepoint t17: insufficient samples (normal=8, perturbed=0)\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: t19\n",
      "==================================================\n",
      "Skipping timepoint t19: insufficient samples (normal=7, perturbed=1)\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: t20\n",
      "==================================================\n",
      "Skipping timepoint t20: insufficient samples (normal=7, perturbed=1)\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: t21\n",
      "==================================================\n",
      "Number of normal samples: 6\n",
      "Number of perturbed samples: 2\n",
      "Using None as control genes, passed at DeseqDataSet initialization\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Fitting dispersions...\n",
      "... done in 3.51 seconds.\n",
      "\n",
      "Fitting dispersion trend curve...\n",
      "... done in 0.42 seconds.\n",
      "\n",
      "Fitting MAP dispersions...\n",
      "... done in 3.53 seconds.\n",
      "\n",
      "Fitting LFCs...\n",
      "... done in 2.62 seconds.\n",
      "\n",
      "Calculating cook's distance...\n",
      "... done in 0.01 seconds.\n",
      "\n",
      "Replacing 0 outlier genes.\n",
      "\n",
      "Running Wald tests...\n",
      "... done in 0.98 seconds.\n",
      "\n",
      "Fitting size factors...\n",
      "... done in 0.01 seconds.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log2 fold change & Wald test p-value: rare_species_greater_5 perturbed vs normal\n",
      "              baseMean  log2FoldChange     lfcSE      stat    pvalue      padj\n",
      "AT1G01010    14.734969       -0.556409  1.446766 -0.384588  0.700542  0.999991\n",
      "AT1G01020    48.166910        0.293579  0.614729  0.477574  0.632953  0.999991\n",
      "AT1G01030    10.790900       -1.207695  1.039226 -1.162110  0.245191  0.999991\n",
      "AT1G01040   144.750024        0.082728  0.408906  0.202315  0.839671  0.999991\n",
      "AT1G01050   254.732610        0.452441  0.395704  1.143383  0.252880  0.999991\n",
      "...                ...             ...       ...       ...       ...       ...\n",
      "ArthCr089  3583.848026        0.166027  0.385337  0.430861  0.666569  0.999991\n",
      "ArthCr088   269.523568       -0.076732  0.352802 -0.217494  0.827823  0.999991\n",
      "ArthCp088     8.504604        0.233773  1.177554  0.198524  0.842635  0.999991\n",
      "ArthCp083    49.194937        0.157983  0.527765  0.299343  0.764679  0.999991\n",
      "ArthCp085    11.383310        0.545875  1.153965  0.473043  0.636183  0.999991\n",
      "\n",
      "[17681 rows x 6 columns]\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: t22\n",
      "==================================================\n",
      "Skipping timepoint t22: insufficient samples (normal=8, perturbed=0)\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: t23\n",
      "==================================================\n",
      "Skipping timepoint t23: insufficient samples (normal=8, perturbed=0)\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: t25\n",
      "==================================================\n",
      "Skipping timepoint t25: insufficient samples (normal=6, perturbed=1)\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: t26\n",
      "==================================================\n",
      "Skipping timepoint t26: insufficient samples (normal=8, perturbed=0)\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: t27\n",
      "==================================================\n",
      "Number of normal samples: 6\n",
      "Number of perturbed samples: 2\n",
      "Using None as control genes, passed at DeseqDataSet initialization\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Fitting dispersions...\n",
      "... done in 3.27 seconds.\n",
      "\n",
      "Fitting dispersion trend curve...\n",
      "... done in 0.44 seconds.\n",
      "\n",
      "Fitting MAP dispersions...\n",
      "... done in 3.55 seconds.\n",
      "\n",
      "Fitting LFCs...\n",
      "... done in 2.36 seconds.\n",
      "\n",
      "Calculating cook's distance...\n",
      "... done in 0.01 seconds.\n",
      "\n",
      "Replacing 0 outlier genes.\n",
      "\n",
      "Running Wald tests...\n",
      "... done in 0.97 seconds.\n",
      "\n",
      "Fitting size factors...\n",
      "... done in 0.01 seconds.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log2 fold change & Wald test p-value: rare_species_greater_5 perturbed vs normal\n",
      "              baseMean  log2FoldChange     lfcSE      stat    pvalue      padj\n",
      "AT1G01010    22.701126       -1.149790  0.685947 -1.676209  0.093697  0.999936\n",
      "AT1G01020    78.400483       -0.321935  0.474305 -0.678751  0.497295  0.999936\n",
      "AT1G01030    25.868840       -0.296070  0.708009 -0.418173  0.675820  0.999936\n",
      "AT1G01040   206.690808       -0.254629  0.292291 -0.871148  0.383673  0.999936\n",
      "AT1G01050   368.018417        0.062710  0.292357  0.214498  0.830159  0.999936\n",
      "...                ...             ...       ...       ...       ...       ...\n",
      "ArthCr089  3462.477424       -0.848056  0.342807 -2.473858  0.013366  0.999936\n",
      "ArthCr088   353.771856       -0.778216  0.383437 -2.029579  0.042399  0.999936\n",
      "ArthCp088    12.499028       -0.231173  0.849215 -0.272220  0.785453  0.999936\n",
      "ArthCp083    56.446051       -0.861654  0.478505 -1.800721  0.071747  0.999936\n",
      "ArthCp085    12.084169        0.568269  1.038807  0.547040  0.584351  0.999936\n",
      "\n",
      "[17681 rows x 6 columns]\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: t18\n",
      "==================================================\n",
      "Skipping timepoint t18: insufficient samples (normal=6, perturbed=1)\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: t01\n",
      "==================================================\n",
      "Number of normal samples: 2\n",
      "Number of perturbed samples: 6\n",
      "Using None as control genes, passed at DeseqDataSet initialization\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Fitting dispersions...\n",
      "... done in 3.27 seconds.\n",
      "\n",
      "Fitting dispersion trend curve...\n",
      "... done in 0.44 seconds.\n",
      "\n",
      "Fitting MAP dispersions...\n",
      "... done in 3.45 seconds.\n",
      "\n",
      "Fitting LFCs...\n",
      "... done in 2.62 seconds.\n",
      "\n",
      "Calculating cook's distance...\n",
      "... done in 0.01 seconds.\n",
      "\n",
      "Replacing 0 outlier genes.\n",
      "\n",
      "Running Wald tests...\n",
      "... done in 1.04 seconds.\n",
      "\n",
      "Fitting size factors...\n",
      "... done in 0.00 seconds.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log2 fold change & Wald test p-value: rare_species_greater_5 perturbed vs normal\n",
      "             baseMean  log2FoldChange     lfcSE      stat    pvalue      padj\n",
      "AT1G01010   16.732559        0.144779  0.864966  0.167381  0.867070  0.999734\n",
      "AT1G01020   20.742664        0.658963  0.954868  0.690109  0.490126  0.996330\n",
      "AT1G01030   12.787656        0.038921  1.205511  0.032286  0.974244  0.999734\n",
      "AT1G01040   64.732276        0.494419  0.583417  0.847454  0.396742  0.989591\n",
      "AT1G01050  143.779824        0.167261  0.345012  0.484799  0.627819  0.996330\n",
      "...               ...             ...       ...       ...       ...       ...\n",
      "ArthCr089  811.218826        0.538275  0.329625  1.632991  0.102471  0.972765\n",
      "ArthCr088   74.673964        0.972477  0.620151  1.568130  0.116851  0.972765\n",
      "ArthCp088    0.187393        0.290871  4.544790  0.064001  0.948969  0.999734\n",
      "ArthCp083    4.532873        1.718074  1.771467  0.969859  0.332117  0.989591\n",
      "ArthCp085    2.133608       -0.396790  2.184929 -0.181603  0.855894  0.999327\n",
      "\n",
      "[17681 rows x 6 columns]\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: t02\n",
      "==================================================\n",
      "Number of normal samples: 2\n",
      "Number of perturbed samples: 6\n",
      "Using None as control genes, passed at DeseqDataSet initialization\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Fitting dispersions...\n",
      "... done in 3.55 seconds.\n",
      "\n",
      "Fitting dispersion trend curve...\n",
      "... done in 0.44 seconds.\n",
      "\n",
      "Fitting MAP dispersions...\n",
      "... done in 3.52 seconds.\n",
      "\n",
      "Fitting LFCs...\n",
      "... done in 2.77 seconds.\n",
      "\n",
      "Calculating cook's distance...\n",
      "... done in 0.01 seconds.\n",
      "\n",
      "Replacing 0 outlier genes.\n",
      "\n",
      "Running Wald tests...\n",
      "... done in 0.95 seconds.\n",
      "\n",
      "Fitting size factors...\n",
      "... done in 0.00 seconds.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log2 fold change & Wald test p-value: rare_species_greater_5 perturbed vs normal\n",
      "              baseMean  log2FoldChange     lfcSE      stat    pvalue      padj\n",
      "AT1G01010     5.828940        1.259751  2.666378  0.472458  0.636600       NaN\n",
      "AT1G01020    41.755678        0.195693  0.874354  0.223815  0.822901  0.983773\n",
      "AT1G01030    27.669855       -0.399910  1.145150 -0.349220  0.726924       NaN\n",
      "AT1G01040   101.171370       -0.161655  0.633214 -0.255293  0.798497  0.980585\n",
      "AT1G01050   112.721100        0.494518  0.635182  0.778545  0.436248  0.926175\n",
      "...                ...             ...       ...       ...       ...       ...\n",
      "ArthCr089  1269.615638        0.088598  0.559733  0.158286  0.874232  0.985954\n",
      "ArthCr088   107.427765        0.785661  0.887943  0.884811  0.376259  0.907967\n",
      "ArthCp088     0.102665       -0.446844  4.581138 -0.097540  0.922298       NaN\n",
      "ArthCp083     7.631815       -0.668419  2.497711 -0.267613  0.788998       NaN\n",
      "ArthCp085     4.798149        0.427054  2.196727  0.194404  0.845859       NaN\n",
      "\n",
      "[17681 rows x 6 columns]\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: t03\n",
      "==================================================\n",
      "Number of normal samples: 4\n",
      "Number of perturbed samples: 3\n",
      "Using None as control genes, passed at DeseqDataSet initialization\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Fitting dispersions...\n",
      "... done in 3.59 seconds.\n",
      "\n",
      "Fitting dispersion trend curve...\n",
      "... done in 0.50 seconds.\n",
      "\n",
      "Fitting MAP dispersions...\n",
      "... done in 3.97 seconds.\n",
      "\n",
      "Fitting LFCs...\n",
      "... done in 2.49 seconds.\n",
      "\n",
      "Calculating cook's distance...\n",
      "... done in 0.01 seconds.\n",
      "\n",
      "Replacing 0 outlier genes.\n",
      "\n",
      "Running Wald tests...\n",
      "... done in 0.95 seconds.\n",
      "\n",
      "Fitting size factors...\n",
      "... done in 0.01 seconds.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log2 fold change & Wald test p-value: rare_species_greater_5 perturbed vs normal\n",
      "             baseMean  log2FoldChange     lfcSE      stat    pvalue      padj\n",
      "AT1G01010   12.673318       -2.193585  1.227035 -1.787712  0.073823  0.999826\n",
      "AT1G01020   51.535167        0.398766  0.558594  0.713875  0.475304  0.999826\n",
      "AT1G01030   33.619525        0.337937  0.568506  0.594431  0.552224  0.999826\n",
      "AT1G01040  111.160668       -0.393967  0.398107 -0.989600  0.322370  0.999826\n",
      "AT1G01050  229.082971        0.070597  0.297352  0.237420  0.812331  0.999826\n",
      "...               ...             ...       ...       ...       ...       ...\n",
      "ArthCr089  623.470960        0.089571  0.267282  0.335116  0.737537  0.999826\n",
      "ArthCr088   41.439495        0.368392  0.639607  0.575967  0.564638  0.999826\n",
      "ArthCp088    4.623443        0.027485  1.982870  0.013861  0.988941  0.999826\n",
      "ArthCp083    6.067498        1.061115  1.641200  0.646548  0.517924  0.999826\n",
      "ArthCp085    3.441357       -0.652307  1.920804 -0.339601  0.734157  0.999826\n",
      "\n",
      "[17681 rows x 6 columns]\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: t04\n",
      "==================================================\n",
      "Skipping timepoint t04: insufficient samples (normal=7, perturbed=1)\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: t05\n",
      "==================================================\n",
      "Number of normal samples: 4\n",
      "Number of perturbed samples: 4\n",
      "Using None as control genes, passed at DeseqDataSet initialization\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Fitting dispersions...\n",
      "... done in 3.28 seconds.\n",
      "\n",
      "Fitting dispersion trend curve...\n",
      "... done in 0.46 seconds.\n",
      "\n",
      "Fitting MAP dispersions...\n",
      "... done in 3.42 seconds.\n",
      "\n",
      "Fitting LFCs...\n",
      "... done in 2.48 seconds.\n",
      "\n",
      "Calculating cook's distance...\n",
      "... done in 0.01 seconds.\n",
      "\n",
      "Replacing 0 outlier genes.\n",
      "\n",
      "Running Wald tests...\n",
      "... done in 0.98 seconds.\n",
      "\n",
      "Fitting size factors...\n",
      "... done in 0.01 seconds.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log2 fold change & Wald test p-value: rare_species_greater_5 perturbed vs normal\n",
      "             baseMean  log2FoldChange     lfcSE      stat    pvalue      padj\n",
      "AT1G01010    6.932322       -0.802089  0.940649 -0.852698  0.393827  0.998538\n",
      "AT1G01020   23.651654       -0.628872  0.565666 -1.111737  0.266251  0.998538\n",
      "AT1G01030   14.073083        0.159133  0.626822  0.253873  0.799594  0.998538\n",
      "AT1G01040   52.306085       -0.118103  0.421158 -0.280424  0.779153  0.998538\n",
      "AT1G01050  119.268157       -0.133876  0.347603 -0.385142  0.700133  0.998538\n",
      "...               ...             ...       ...       ...       ...       ...\n",
      "ArthCr089  352.901763        0.508934  0.344646  1.476685  0.139760  0.998538\n",
      "ArthCr088   39.357860        0.471790  0.546412  0.863433  0.387900  0.998538\n",
      "ArthCp088    1.352288        1.245881  1.942264  0.641458  0.521225  0.998538\n",
      "ArthCp083    3.105030        0.649018  1.475486  0.439868  0.660033  0.998538\n",
      "ArthCp085    4.022016        1.874606  1.432230  1.308872  0.190578  0.998538\n",
      "\n",
      "[17681 rows x 6 columns]\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: t07\n",
      "==================================================\n",
      "Number of normal samples: 3\n",
      "Number of perturbed samples: 4\n",
      "Using None as control genes, passed at DeseqDataSet initialization\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Fitting dispersions...\n",
      "... done in 3.21 seconds.\n",
      "\n",
      "Fitting dispersion trend curve...\n",
      "... done in 0.44 seconds.\n",
      "\n",
      "Fitting MAP dispersions...\n",
      "... done in 3.56 seconds.\n",
      "\n",
      "Fitting LFCs...\n",
      "... done in 2.28 seconds.\n",
      "\n",
      "Calculating cook's distance...\n",
      "... done in 0.01 seconds.\n",
      "\n",
      "Replacing 0 outlier genes.\n",
      "\n",
      "Running Wald tests...\n",
      "... done in 0.96 seconds.\n",
      "\n",
      "Fitting size factors...\n",
      "... done in 0.00 seconds.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log2 fold change & Wald test p-value: rare_species_greater_5 perturbed vs normal\n",
      "             baseMean  log2FoldChange     lfcSE      stat    pvalue      padj\n",
      "AT1G01010    9.075198       -0.069277  0.866482 -0.079952  0.936276  0.998965\n",
      "AT1G01020   49.354920        0.012714  0.477526  0.026626  0.978758  0.998965\n",
      "AT1G01030   17.667445       -0.217179  0.565901 -0.383776  0.701144  0.998965\n",
      "AT1G01040  117.384550        0.025057  0.339129  0.073887  0.941101  0.998965\n",
      "AT1G01050  287.272888        0.166384  0.235625  0.706137  0.480103  0.998965\n",
      "...               ...             ...       ...       ...       ...       ...\n",
      "ArthCr089  562.888905       -0.000126  0.405815 -0.000312  0.999751  0.999948\n",
      "ArthCr088   55.967579       -0.201847  0.426828 -0.472901  0.636284  0.998965\n",
      "ArthCp088    8.208301       -1.549820  1.019692 -1.519890  0.128539  0.998965\n",
      "ArthCp083   12.867363        0.043666  0.810250  0.053893  0.957021  0.998965\n",
      "ArthCp085    9.425587       -0.307243  0.773147 -0.397393  0.691078  0.998965\n",
      "\n",
      "[17681 rows x 6 columns]\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: t10\n",
      "==================================================\n",
      "Skipping timepoint t10: insufficient samples (normal=7, perturbed=1)\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: t11\n",
      "==================================================\n",
      "Skipping timepoint t11: insufficient samples (normal=7, perturbed=1)\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: t12\n",
      "==================================================\n",
      "Number of normal samples: 3\n",
      "Number of perturbed samples: 3\n",
      "Using None as control genes, passed at DeseqDataSet initialization\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Fitting dispersions...\n",
      "... done in 3.27 seconds.\n",
      "\n",
      "Fitting dispersion trend curve...\n",
      "... done in 0.45 seconds.\n",
      "\n",
      "Fitting MAP dispersions...\n",
      "... done in 3.43 seconds.\n",
      "\n",
      "Fitting LFCs...\n",
      "... done in 2.34 seconds.\n",
      "\n",
      "Calculating cook's distance...\n",
      "... done in 0.01 seconds.\n",
      "\n",
      "Replacing 0 outlier genes.\n",
      "\n",
      "Running Wald tests...\n",
      "... done in 0.94 seconds.\n",
      "\n",
      "Fitting size factors...\n",
      "... done in 0.00 seconds.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log2 fold change & Wald test p-value: rare_species_greater_5 perturbed vs normal\n",
      "             baseMean  log2FoldChange     lfcSE      stat    pvalue      padj\n",
      "AT1G01010   17.131550       -1.638760  0.781312 -2.097446  0.035954       NaN\n",
      "AT1G01020   42.763589        0.258651  0.527911  0.489953  0.624167  0.973041\n",
      "AT1G01030    9.323262        0.136428  0.861678  0.158328  0.874198       NaN\n",
      "AT1G01040   80.984725        1.030227  0.376956  2.733018  0.006276  0.304084\n",
      "AT1G01050  205.670125        0.182673  0.286670  0.637225  0.523978  0.950352\n",
      "...               ...             ...       ...       ...       ...       ...\n",
      "ArthCr089  723.254919       -0.439716  0.461039 -0.953750  0.340210  0.899083\n",
      "ArthCr088   48.604026       -0.287700  0.569237 -0.505414  0.613268  0.972305\n",
      "ArthCp088    6.236000        2.128524  1.521665  1.398813  0.161869       NaN\n",
      "ArthCp083    6.901236        0.629817  1.169768  0.538412  0.590292       NaN\n",
      "ArthCp085    7.028430        0.164456  1.307245  0.125804  0.899887       NaN\n",
      "\n",
      "[17681 rows x 6 columns]\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: t13\n",
      "==================================================\n",
      "Number of normal samples: 4\n",
      "Number of perturbed samples: 3\n",
      "Using None as control genes, passed at DeseqDataSet initialization\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Fitting dispersions...\n",
      "... done in 3.49 seconds.\n",
      "\n",
      "Fitting dispersion trend curve...\n",
      "... done in 0.47 seconds.\n",
      "\n",
      "Fitting MAP dispersions...\n",
      "... done in 3.52 seconds.\n",
      "\n",
      "Fitting LFCs...\n",
      "... done in 2.33 seconds.\n",
      "\n",
      "Calculating cook's distance...\n",
      "... done in 0.01 seconds.\n",
      "\n",
      "Replacing 0 outlier genes.\n",
      "\n",
      "Running Wald tests...\n",
      "... done in 1.01 seconds.\n",
      "\n",
      "Fitting size factors...\n",
      "... done in 0.00 seconds.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log2 fold change & Wald test p-value: rare_species_greater_5 perturbed vs normal\n",
      "             baseMean  log2FoldChange     lfcSE      stat    pvalue      padj\n",
      "AT1G01010   10.573842        0.903503  0.828066  1.091100  0.275229       NaN\n",
      "AT1G01020   45.383900       -0.306708  0.432196 -0.709651  0.477921  0.999167\n",
      "AT1G01030   18.919309       -0.026417  0.617688 -0.042768  0.965887  0.999167\n",
      "AT1G01040  104.869243        0.109121  0.412837  0.264320  0.791533  0.999167\n",
      "AT1G01050  240.106429        0.040425  0.301612  0.134030  0.893379  0.999167\n",
      "...               ...             ...       ...       ...       ...       ...\n",
      "ArthCr089  223.172557       -0.772469  0.475441 -1.624743  0.104217  0.999167\n",
      "ArthCr088   26.706730       -0.870310  0.574952 -1.513709  0.130100  0.999167\n",
      "ArthCp088    9.188339       -0.901103  0.907173 -0.993309  0.320559       NaN\n",
      "ArthCp083    8.470385       -0.731932  0.929137 -0.787755  0.430840       NaN\n",
      "ArthCp085    7.445681       -0.114704  0.934077 -0.122799  0.902266       NaN\n",
      "\n",
      "[17681 rows x 6 columns]\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: t06\n",
      "==================================================\n",
      "Number of normal samples: 3\n",
      "Number of perturbed samples: 4\n",
      "Using None as control genes, passed at DeseqDataSet initialization\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Fitting dispersions...\n",
      "... done in 3.31 seconds.\n",
      "\n",
      "Fitting dispersion trend curve...\n",
      "... done in 0.42 seconds.\n",
      "\n",
      "Fitting MAP dispersions...\n",
      "... done in 3.58 seconds.\n",
      "\n",
      "Fitting LFCs...\n",
      "... done in 2.25 seconds.\n",
      "\n",
      "Calculating cook's distance...\n",
      "... done in 0.01 seconds.\n",
      "\n",
      "Replacing 0 outlier genes.\n",
      "\n",
      "Running Wald tests...\n",
      "... done in 0.97 seconds.\n",
      "\n",
      "Fitting size factors...\n",
      "... done in 0.00 seconds.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log2 fold change & Wald test p-value: rare_species_greater_5 perturbed vs normal\n",
      "             baseMean  log2FoldChange     lfcSE      stat    pvalue      padj\n",
      "AT1G01010    8.327423        0.629310  1.120493  0.561636  0.574364  0.998874\n",
      "AT1G01020   57.483968        0.261749  0.370510  0.706458  0.479903  0.998874\n",
      "AT1G01030   33.288089        1.127381  0.513225  2.196662  0.028045  0.998874\n",
      "AT1G01040   94.194888        0.065185  0.318144  0.204893  0.837656  0.998874\n",
      "AT1G01050  249.921278        0.213253  0.295621  0.721372  0.470681  0.998874\n",
      "...               ...             ...       ...       ...       ...       ...\n",
      "ArthCr089  506.040298        0.152887  0.315864  0.484028  0.628366  0.998874\n",
      "ArthCr088   51.157487       -0.024741  0.419530 -0.058974  0.952973  0.998874\n",
      "ArthCp088   10.959722        0.453296  0.898380  0.504570  0.613861  0.998874\n",
      "ArthCp083   14.438688       -0.376090  1.022617 -0.367773  0.713043  0.998874\n",
      "ArthCp085    8.047360       -0.837036  0.861227 -0.971910  0.331095  0.998874\n",
      "\n",
      "[17681 rows x 6 columns]\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: t09\n",
      "==================================================\n",
      "Number of normal samples: 2\n",
      "Number of perturbed samples: 5\n",
      "Using None as control genes, passed at DeseqDataSet initialization\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Fitting dispersions...\n",
      "... done in 3.59 seconds.\n",
      "\n",
      "Fitting dispersion trend curve...\n",
      "... done in 0.45 seconds.\n",
      "\n",
      "Fitting MAP dispersions...\n",
      "... done in 3.42 seconds.\n",
      "\n",
      "Fitting LFCs...\n",
      "... done in 2.27 seconds.\n",
      "\n",
      "Calculating cook's distance...\n",
      "... done in 0.01 seconds.\n",
      "\n",
      "Replacing 0 outlier genes.\n",
      "\n",
      "Running Wald tests...\n",
      "... done in 0.96 seconds.\n",
      "\n",
      "Fitting size factors...\n",
      "... done in 0.01 seconds.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log2 fold change & Wald test p-value: rare_species_greater_5 perturbed vs normal\n",
      "             baseMean  log2FoldChange     lfcSE      stat    pvalue      padj\n",
      "AT1G01010    9.360369        0.679901  1.074170  0.632955  0.526763       NaN\n",
      "AT1G01020   67.102513       -0.151185  0.362234 -0.417367  0.676410  0.999694\n",
      "AT1G01030   22.755263       -0.621428  0.551410 -1.126981  0.259751  0.999694\n",
      "AT1G01040  111.470100        0.189984  0.353041  0.538136  0.590483  0.999694\n",
      "AT1G01050  290.369334       -0.067788  0.307730 -0.220285  0.825649  0.999694\n",
      "...               ...             ...       ...       ...       ...       ...\n",
      "ArthCr089  674.780966       -0.554155  0.313819 -1.765841  0.077423  0.999694\n",
      "ArthCr088   61.928726        0.005096  0.429269  0.011872  0.990528  0.999694\n",
      "ArthCp088    9.035784        0.236899  0.815283  0.290572  0.771378       NaN\n",
      "ArthCp083   14.608172        0.042033  0.746118  0.056336  0.955074  0.999694\n",
      "ArthCp085   10.522192       -0.307905  0.767531 -0.401163  0.688300       NaN\n",
      "\n",
      "[17681 rows x 6 columns]\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: c1_t08\n",
      "==================================================\n",
      "Number of normal samples: 12\n",
      "Number of perturbed samples: 4\n",
      "Using None as control genes, passed at DeseqDataSet initialization\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Fitting dispersions...\n",
      "... done in 3.44 seconds.\n",
      "\n",
      "Fitting dispersion trend curve...\n",
      "... done in 0.48 seconds.\n",
      "\n",
      "Fitting MAP dispersions...\n",
      "... done in 3.86 seconds.\n",
      "\n",
      "Fitting LFCs...\n",
      "... done in 2.69 seconds.\n",
      "\n",
      "Calculating cook's distance...\n",
      "... done in 0.02 seconds.\n",
      "\n",
      "Replacing 73 outlier genes.\n",
      "\n",
      "Fitting dispersions...\n",
      "... done in 0.03 seconds.\n",
      "\n",
      "Fitting MAP dispersions...\n",
      "... done in 0.03 seconds.\n",
      "\n",
      "Fitting LFCs...\n",
      "... done in 0.05 seconds.\n",
      "\n",
      "Running Wald tests...\n",
      "... done in 1.00 seconds.\n",
      "\n",
      "Fitting size factors...\n",
      "... done in 0.00 seconds.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log2 fold change & Wald test p-value: rare_species_greater_5 perturbed vs normal\n",
      "             baseMean  log2FoldChange     lfcSE      stat    pvalue      padj\n",
      "AT1G01010    5.462072       -0.871802  1.244159 -0.700716  0.483480       NaN\n",
      "AT1G01020   53.215496        0.046077  0.370893  0.124234  0.901130  0.983892\n",
      "AT1G01030   14.328351       -2.213171  0.728341 -3.038649  0.002376  0.151302\n",
      "AT1G01040  111.308722        0.045876  0.288281  0.159136  0.873562  0.980523\n",
      "AT1G01050  222.646625        0.025464  0.217214  0.117231  0.906677  0.985269\n",
      "...               ...             ...       ...       ...       ...       ...\n",
      "ArthCr089  808.504968        0.704372  0.572716  1.229879  0.218742  0.784157\n",
      "ArthCr088   29.639512        0.259014  0.422450  0.613124  0.539795  0.907850\n",
      "ArthCp088   10.022036       -0.289541  0.892605 -0.324377  0.745653       NaN\n",
      "ArthCp083   16.646939        0.128222  0.683192  0.187681  0.851127  0.976541\n",
      "ArthCp085   17.456079       -0.932648  0.813521 -1.146435  0.251615  0.806393\n",
      "\n",
      "[17681 rows x 6 columns]\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: c1_t09\n",
      "==================================================\n",
      "Number of normal samples: 5\n",
      "Number of perturbed samples: 2\n",
      "Using None as control genes, passed at DeseqDataSet initialization\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Fitting dispersions...\n",
      "... done in 3.34 seconds.\n",
      "\n",
      "Fitting dispersion trend curve...\n",
      "... done in 0.46 seconds.\n",
      "\n",
      "Fitting MAP dispersions...\n",
      "... done in 3.49 seconds.\n",
      "\n",
      "Fitting LFCs...\n",
      "... done in 2.65 seconds.\n",
      "\n",
      "Calculating cook's distance...\n",
      "... done in 0.01 seconds.\n",
      "\n",
      "Replacing 0 outlier genes.\n",
      "\n",
      "Running Wald tests...\n",
      "... done in 0.97 seconds.\n",
      "\n",
      "Fitting size factors...\n",
      "... done in 0.00 seconds.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log2 fold change & Wald test p-value: rare_species_greater_5 perturbed vs normal\n",
      "             baseMean  log2FoldChange     lfcSE      stat    pvalue      padj\n",
      "AT1G01010   17.137315        0.108087  1.283542  0.084210  0.932890  0.998001\n",
      "AT1G01020   59.802847        0.452086  0.865967  0.522059  0.601629  0.996350\n",
      "AT1G01030   13.732965        1.761999  1.221169  1.442879  0.149055  0.964830\n",
      "AT1G01040  131.260076        0.295253  0.495166  0.596270  0.550995  0.996350\n",
      "AT1G01050  280.135775        0.107018  0.324086  0.330214  0.741238  0.996350\n",
      "...               ...             ...       ...       ...       ...       ...\n",
      "ArthCr089  625.415151       -0.010304  0.264244 -0.038996  0.968894  0.999523\n",
      "ArthCr088   48.420611       -0.189673  0.681469 -0.278329  0.780760  0.996350\n",
      "ArthCp088    4.190219       -2.301021  2.267348 -1.014852  0.310177  0.996350\n",
      "ArthCp083   14.071409        0.922632  1.482461  0.622365  0.533702  0.996350\n",
      "ArthCp085   10.497329        0.286316  1.405752  0.203675  0.838607  0.996350\n",
      "\n",
      "[17681 rows x 6 columns]\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: c1_t10\n",
      "==================================================\n",
      "Skipping timepoint c1_t10: insufficient samples (normal=8, perturbed=0)\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: c1_t12\n",
      "==================================================\n",
      "Skipping timepoint c1_t12: insufficient samples (normal=8, perturbed=0)\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: c2_t02\n",
      "==================================================\n",
      "Skipping timepoint c2_t02: insufficient samples (normal=7, perturbed=1)\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: c2_t03\n",
      "==================================================\n",
      "Skipping timepoint c2_t03: insufficient samples (normal=7, perturbed=1)\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: c2_t04\n",
      "==================================================\n",
      "Skipping timepoint c2_t04: insufficient samples (normal=7, perturbed=1)\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: c2_t05\n",
      "==================================================\n",
      "Skipping timepoint c2_t05: insufficient samples (normal=6, perturbed=1)\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: c2_t06\n",
      "==================================================\n",
      "Skipping timepoint c2_t06: insufficient samples (normal=7, perturbed=1)\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: c1_t11\n",
      "==================================================\n",
      "Skipping timepoint c1_t11: insufficient samples (normal=7, perturbed=0)\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: c1_t13\n",
      "==================================================\n",
      "Number of normal samples: 4\n",
      "Number of perturbed samples: 3\n",
      "Using None as control genes, passed at DeseqDataSet initialization\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Fitting dispersions...\n",
      "... done in 3.29 seconds.\n",
      "\n",
      "Fitting dispersion trend curve...\n",
      "... done in 0.46 seconds.\n",
      "\n",
      "Fitting MAP dispersions...\n",
      "... done in 3.43 seconds.\n",
      "\n",
      "Fitting LFCs...\n",
      "... done in 2.58 seconds.\n",
      "\n",
      "Calculating cook's distance...\n",
      "... done in 0.01 seconds.\n",
      "\n",
      "Replacing 0 outlier genes.\n",
      "\n",
      "Running Wald tests...\n",
      "... done in 0.98 seconds.\n",
      "\n",
      "Fitting size factors...\n",
      "... done in 0.01 seconds.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log2 fold change & Wald test p-value: rare_species_greater_5 perturbed vs normal\n",
      "             baseMean  log2FoldChange     lfcSE      stat    pvalue      padj\n",
      "AT1G01010    5.043403       -5.300503  2.541733 -2.085389  0.037034       NaN\n",
      "AT1G01020   50.136346        0.168356  0.501489  0.335713  0.737088       NaN\n",
      "AT1G01030   21.925333       -0.175491  0.929139 -0.188875  0.850191       NaN\n",
      "AT1G01040  149.302430       -0.006924  0.535489 -0.012930  0.989683  0.998195\n",
      "AT1G01050  189.922085        0.299026  0.351935  0.849662  0.395513  0.763234\n",
      "...               ...             ...       ...       ...       ...       ...\n",
      "ArthCr089  389.520596       -0.253676  0.419211 -0.605126  0.545095  0.843528\n",
      "ArthCr088   14.661313        0.029614  0.809996  0.036561  0.970835       NaN\n",
      "ArthCp088    3.343409        2.078143  2.549914  0.814986  0.415081       NaN\n",
      "ArthCp083   13.098330        0.058721  1.131660  0.051889  0.958617       NaN\n",
      "ArthCp085   20.559756       -0.124393  0.738641 -0.168408  0.866262       NaN\n",
      "\n",
      "[17681 rows x 6 columns]\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: c2_t01\n",
      "==================================================\n",
      "Number of normal samples: 5\n",
      "Number of perturbed samples: 2\n",
      "Using None as control genes, passed at DeseqDataSet initialization\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Fitting dispersions...\n",
      "... done in 3.44 seconds.\n",
      "\n",
      "Fitting dispersion trend curve...\n",
      "... done in 0.43 seconds.\n",
      "\n",
      "Fitting MAP dispersions...\n",
      "... done in 3.42 seconds.\n",
      "\n",
      "Fitting LFCs...\n",
      "... done in 2.87 seconds.\n",
      "\n",
      "Calculating cook's distance...\n",
      "... done in 0.01 seconds.\n",
      "\n",
      "Replacing 0 outlier genes.\n",
      "\n",
      "Running Wald tests...\n",
      "... done in 1.04 seconds.\n",
      "\n",
      "Fitting size factors...\n",
      "... done in 0.00 seconds.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log2 fold change & Wald test p-value: rare_species_greater_5 perturbed vs normal\n",
      "             baseMean  log2FoldChange     lfcSE      stat    pvalue      padj\n",
      "AT1G01010    3.586636       -2.910709  2.263665 -1.285839  0.198499  0.999573\n",
      "AT1G01020   42.399036       -0.231824  0.682162 -0.339838  0.733979  0.999573\n",
      "AT1G01030   22.503768        1.081662  0.842342  1.284112  0.199103  0.999573\n",
      "AT1G01040   85.020311       -0.845684  0.590135 -1.433035  0.151848  0.999573\n",
      "AT1G01050  212.001365       -0.205143  0.413295 -0.496360  0.619640  0.999573\n",
      "...               ...             ...       ...       ...       ...       ...\n",
      "ArthCr089  346.683127        0.566411  0.387945  1.460031  0.144282  0.999573\n",
      "ArthCr088   23.799195        0.494250  0.972373  0.508292  0.611248  0.999573\n",
      "ArthCp088    8.848439       -0.588265  1.375068 -0.427808  0.668791  0.999573\n",
      "ArthCp083    8.653670        0.252605  1.672612  0.151024  0.879956  0.999573\n",
      "ArthCp085    7.149250       -1.018922  1.652322 -0.616661  0.537459  0.999573\n",
      "\n",
      "[17681 rows x 6 columns]\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: t28\n",
      "==================================================\n",
      "Skipping timepoint t28: insufficient samples (normal=7, perturbed=1)\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: t24\n",
      "==================================================\n",
      "Skipping timepoint t24: insufficient samples (normal=8, perturbed=0)\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: t16\n",
      "==================================================\n",
      "Skipping timepoint t16: insufficient samples (normal=7, perturbed=1)\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: t08\n",
      "==================================================\n",
      "Number of normal samples: 5\n",
      "Number of perturbed samples: 3\n",
      "Using None as control genes, passed at DeseqDataSet initialization\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Fitting dispersions...\n",
      "... done in 3.33 seconds.\n",
      "\n",
      "Fitting dispersion trend curve...\n",
      "... done in 0.48 seconds.\n",
      "\n",
      "Fitting MAP dispersions...\n",
      "... done in 3.54 seconds.\n",
      "\n",
      "Fitting LFCs...\n",
      "... done in 2.47 seconds.\n",
      "\n",
      "Calculating cook's distance...\n",
      "... done in 0.01 seconds.\n",
      "\n",
      "Replacing 0 outlier genes.\n",
      "\n",
      "Running Wald tests...\n",
      "... done in 0.98 seconds.\n",
      "\n",
      "Fitting size factors...\n",
      "... done in 0.00 seconds.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log2 fold change & Wald test p-value: rare_species_greater_5 perturbed vs normal\n",
      "              baseMean  log2FoldChange     lfcSE      stat    pvalue     padj\n",
      "AT1G01010     6.340787        1.133709  1.428117  0.793849  0.427283  0.99993\n",
      "AT1G01020    51.564805        0.845124  0.489194  1.727584  0.084063  0.99993\n",
      "AT1G01030    20.237407        0.609256  0.806532  0.755403  0.450007  0.99993\n",
      "AT1G01040   126.621703        0.355287  0.359201  0.989106  0.322612  0.99993\n",
      "AT1G01050   359.624581        0.631476  0.271266  2.327887  0.019918  0.99993\n",
      "...                ...             ...       ...       ...       ...      ...\n",
      "ArthCr089  1811.955207        0.328602  0.672666  0.488507  0.625190  0.99993\n",
      "ArthCr088    78.999450        0.675054  0.438975  1.537796  0.124099  0.99993\n",
      "ArthCp088    16.299308        0.035769  0.840878  0.042537  0.966070  0.99993\n",
      "ArthCp083    24.429109       -0.199099  0.758671 -0.262431  0.792989  0.99993\n",
      "ArthCp085    20.051421        0.091396  0.854808  0.106920  0.914853  0.99993\n",
      "\n",
      "[17681 rows x 6 columns]\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: c1_t01\n",
      "==================================================\n",
      "Number of normal samples: 6\n",
      "Number of perturbed samples: 2\n",
      "Using None as control genes, passed at DeseqDataSet initialization\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Fitting dispersions...\n",
      "... done in 3.38 seconds.\n",
      "\n",
      "Fitting dispersion trend curve...\n",
      "... done in 0.44 seconds.\n",
      "\n",
      "Fitting MAP dispersions...\n",
      "... done in 3.56 seconds.\n",
      "\n",
      "Fitting LFCs...\n",
      "... done in 2.48 seconds.\n",
      "\n",
      "Calculating cook's distance...\n",
      "... done in 0.01 seconds.\n",
      "\n",
      "Replacing 0 outlier genes.\n",
      "\n",
      "Running Wald tests...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log2 fold change & Wald test p-value: rare_species_greater_5 perturbed vs normal\n",
      "              baseMean  log2FoldChange     lfcSE      stat    pvalue      padj\n",
      "AT1G01010     4.831789       -1.323158  1.724141 -0.767430  0.442826  0.999599\n",
      "AT1G01020    37.396573       -0.538198  0.863539 -0.623247  0.533122  0.999599\n",
      "AT1G01030     5.417486       -1.949862  1.727815 -1.128513  0.259103  0.999599\n",
      "AT1G01040   103.969284       -0.292908  0.433339 -0.675933  0.499083  0.999599\n",
      "AT1G01050   224.359910        0.240858  0.400848  0.600870  0.547926  0.999599\n",
      "...                ...             ...       ...       ...       ...       ...\n",
      "ArthCr089  1026.867062       -0.967052  0.831006 -1.163713  0.244541  0.999599\n",
      "ArthCr088    42.833990        0.401681  0.717900  0.559522  0.575806  0.999599\n",
      "ArthCp088     7.695965       -0.649868  1.643325 -0.395459  0.692504  0.999599\n",
      "ArthCp083    19.478713        0.002324  0.980631  0.002370  0.998109  0.999599\n",
      "ArthCp085    11.272994        1.857692  1.346784  1.379354  0.167786  0.999599\n",
      "\n",
      "[17681 rows x 6 columns]\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: c1_t02\n",
      "==================================================\n",
      "Skipping timepoint c1_t02: insufficient samples (normal=7, perturbed=1)\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: c1_t03\n",
      "==================================================\n",
      "Skipping timepoint c1_t03: insufficient samples (normal=8, perturbed=0)\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: c1_t04\n",
      "==================================================\n",
      "Skipping timepoint c1_t04: insufficient samples (normal=7, perturbed=1)\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: c1_t05\n",
      "==================================================\n",
      "Skipping timepoint c1_t05: insufficient samples (normal=7, perturbed=1)\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: c1_t06\n",
      "==================================================\n",
      "Skipping timepoint c1_t06: insufficient samples (normal=8, perturbed=0)\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: c1_t07\n",
      "==================================================\n",
      "Skipping timepoint c1_t07: insufficient samples (normal=8, perturbed=0)\n",
      "\n",
      "Completed! Total results: 19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "... done in 0.95 seconds.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "list_of_differential_expression_results = []\n",
    "\n",
    "\n",
    "# Get unique timepoints\n",
    "timepoints = trimmed_metadata[\"timepoint\"].unique()\n",
    "\n",
    "for tp in timepoints:\n",
    "    print(f\"\\n{'='*50}\")\n",
    "    print(f\"Processing timepoint: {tp}\")\n",
    "    print(f\"{'='*50}\")\n",
    "\n",
    "    # Get samples for this timepoint\n",
    "    tp_metadata = trimmed_metadata[trimmed_metadata[\"timepoint\"] == tp]\n",
    "\n",
    "    # Split into normal (<=5) and perturbed (>5) based on rare species abundance\n",
    "    normal_samples = tp_metadata[\n",
    "        tp_metadata[\"rare_species_total_abundR100\"] <= rare_species_abundance_threshold\n",
    "    ].index.tolist()\n",
    "    perturbed_samples = tp_metadata[\n",
    "        tp_metadata[\"rare_species_total_abundR100\"] > rare_species_abundance_threshold\n",
    "    ].index.tolist()\n",
    "\n",
    "    # Filter to only include samples present in cleaned_raw_transcriptome\n",
    "    normal_samples = [s for s in normal_samples if s in cleaned_raw_transcriptome.index]\n",
    "    perturbed_samples = [\n",
    "        s for s in perturbed_samples if s in cleaned_raw_transcriptome.index\n",
    "    ]\n",
    "\n",
    "    # Skip if either group has too few samples\n",
    "    if len(normal_samples) < 2 or len(perturbed_samples) < 2:\n",
    "        print(\n",
    "            f\"Skipping timepoint {tp}: insufficient samples (normal={len(normal_samples)}, perturbed={len(perturbed_samples)})\"\n",
    "        )\n",
    "        continue\n",
    "\n",
    "    try:\n",
    "        ds_result = calculate_differential_expression_btwn_2_groups(\n",
    "            list_of_normal_samples=normal_samples,\n",
    "            list_of_perturbed_samples=perturbed_samples,\n",
    "            raw_transcriptome_data=cleaned_raw_transcriptome,\n",
    "            og_metadata=trimmed_metadata,\n",
    "            name_of_perturbation_column=\"rare_species_greater_5\",\n",
    "            use_plate_for_design=False,\n",
    "        )\n",
    "        list_of_differential_expression_results.append(\n",
    "            {\"timepoint\": tp, \"result\": ds_result}\n",
    "        )\n",
    "    except Exception as e:\n",
    "        print(f\"Error processing timepoint {tp}: {e}\")\n",
    "\n",
    "print(f\"\\nCompleted! Total results: {len(list_of_differential_expression_results)}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "8b88a127",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>plate.pos</th>\n",
       "      <th>sampID</th>\n",
       "      <th>sample.type</th>\n",
       "      <th>timepoint</th>\n",
       "      <th>date</th>\n",
       "      <th>time</th>\n",
       "      <th>extractplate</th>\n",
       "      <th>rnaprepplate</th>\n",
       "      <th>Unmapped</th>\n",
       "      <th>No Feature</th>\n",
       "      <th>plate_row</th>\n",
       "      <th>plate_col</th>\n",
       "      <th>Total Reads</th>\n",
       "      <th>Experiment Type</th>\n",
       "      <th>Sample Month</th>\n",
       "      <th>daysincestart</th>\n",
       "      <th>Date and Time</th>\n",
       "      <th>has_rare_species</th>\n",
       "      <th>lacking_core_species</th>\n",
       "      <th>rare_species_total_abundR100</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>filename</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_A11</th>\n",
       "      <td>A11</td>\n",
       "      <td>LIC457</td>\n",
       "      <td>plant</td>\n",
       "      <td>c1_t07</td>\n",
       "      <td>2/10/24</td>\n",
       "      <td>20:00</td>\n",
       "      <td>LICCIRC_01</td>\n",
       "      <td>LICRNA_03</td>\n",
       "      <td>2674528.0</td>\n",
       "      <td>376117.0</td>\n",
       "      <td>A</td>\n",
       "      <td>11</td>\n",
       "      <td>7080205.0</td>\n",
       "      <td>Circadian Experiment 1</td>\n",
       "      <td>February</td>\n",
       "      <td>101</td>\n",
       "      <td>2024-02-10 20:00:00</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>1.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_B11</th>\n",
       "      <td>B11</td>\n",
       "      <td>LIC458</td>\n",
       "      <td>plant</td>\n",
       "      <td>c1_t07</td>\n",
       "      <td>2/10/24</td>\n",
       "      <td>20:00</td>\n",
       "      <td>LICCIRC_01</td>\n",
       "      <td>LICRNA_03</td>\n",
       "      <td>2708899.0</td>\n",
       "      <td>279807.0</td>\n",
       "      <td>B</td>\n",
       "      <td>11</td>\n",
       "      <td>5975492.0</td>\n",
       "      <td>Circadian Experiment 1</td>\n",
       "      <td>February</td>\n",
       "      <td>101</td>\n",
       "      <td>2024-02-10 20:00:00</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>5.9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_C11</th>\n",
       "      <td>C11</td>\n",
       "      <td>LIC459</td>\n",
       "      <td>plant</td>\n",
       "      <td>c1_t07</td>\n",
       "      <td>2/10/24</td>\n",
       "      <td>20:00</td>\n",
       "      <td>LICCIRC_01</td>\n",
       "      <td>LICRNA_03</td>\n",
       "      <td>9536959.0</td>\n",
       "      <td>1915565.0</td>\n",
       "      <td>C</td>\n",
       "      <td>11</td>\n",
       "      <td>22087076.0</td>\n",
       "      <td>Circadian Experiment 1</td>\n",
       "      <td>February</td>\n",
       "      <td>101</td>\n",
       "      <td>2024-02-10 20:00:00</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_D11</th>\n",
       "      <td>D11</td>\n",
       "      <td>LIC460</td>\n",
       "      <td>plant</td>\n",
       "      <td>c1_t07</td>\n",
       "      <td>2/10/24</td>\n",
       "      <td>20:00</td>\n",
       "      <td>LICCIRC_01</td>\n",
       "      <td>LICRNA_03</td>\n",
       "      <td>4996759.0</td>\n",
       "      <td>1265822.0</td>\n",
       "      <td>D</td>\n",
       "      <td>11</td>\n",
       "      <td>14457644.0</td>\n",
       "      <td>Circadian Experiment 1</td>\n",
       "      <td>February</td>\n",
       "      <td>101</td>\n",
       "      <td>2024-02-10 20:00:00</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_E11</th>\n",
       "      <td>E11</td>\n",
       "      <td>LIC461</td>\n",
       "      <td>plant</td>\n",
       "      <td>c1_t07</td>\n",
       "      <td>2/10/24</td>\n",
       "      <td>20:00</td>\n",
       "      <td>LICCIRC_01</td>\n",
       "      <td>LICRNA_03</td>\n",
       "      <td>874618.0</td>\n",
       "      <td>149646.0</td>\n",
       "      <td>E</td>\n",
       "      <td>11</td>\n",
       "      <td>2408154.0</td>\n",
       "      <td>Circadian Experiment 1</td>\n",
       "      <td>February</td>\n",
       "      <td>101</td>\n",
       "      <td>2024-02-10 20:00:00</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>0.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_F11</th>\n",
       "      <td>F11</td>\n",
       "      <td>LIC462</td>\n",
       "      <td>plant</td>\n",
       "      <td>c1_t07</td>\n",
       "      <td>2/10/24</td>\n",
       "      <td>20:00</td>\n",
       "      <td>LICCIRC_01</td>\n",
       "      <td>LICRNA_03</td>\n",
       "      <td>1218515.0</td>\n",
       "      <td>110644.0</td>\n",
       "      <td>F</td>\n",
       "      <td>11</td>\n",
       "      <td>2197530.0</td>\n",
       "      <td>Circadian Experiment 1</td>\n",
       "      <td>February</td>\n",
       "      <td>101</td>\n",
       "      <td>2024-02-10 20:00:00</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>2.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_G11</th>\n",
       "      <td>G11</td>\n",
       "      <td>LIC463</td>\n",
       "      <td>plant</td>\n",
       "      <td>c1_t07</td>\n",
       "      <td>2/10/24</td>\n",
       "      <td>20:00</td>\n",
       "      <td>LICCIRC_01</td>\n",
       "      <td>LICRNA_03</td>\n",
       "      <td>2261397.0</td>\n",
       "      <td>492243.0</td>\n",
       "      <td>G</td>\n",
       "      <td>11</td>\n",
       "      <td>6315886.0</td>\n",
       "      <td>Circadian Experiment 1</td>\n",
       "      <td>February</td>\n",
       "      <td>101</td>\n",
       "      <td>2024-02-10 20:00:00</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H11</th>\n",
       "      <td>H11</td>\n",
       "      <td>LIC464</td>\n",
       "      <td>plant</td>\n",
       "      <td>c1_t07</td>\n",
       "      <td>2/10/24</td>\n",
       "      <td>20:00</td>\n",
       "      <td>LICCIRC_01</td>\n",
       "      <td>LICRNA_03</td>\n",
       "      <td>326959.0</td>\n",
       "      <td>60346.0</td>\n",
       "      <td>H</td>\n",
       "      <td>11</td>\n",
       "      <td>1509679.0</td>\n",
       "      <td>Circadian Experiment 1</td>\n",
       "      <td>February</td>\n",
       "      <td>101</td>\n",
       "      <td>2024-02-10 20:00:00</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>2.9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            plate.pos  sampID sample.type timepoint     date  \\\n",
       "filename                                                                       \n",
       "B250508004_n01_LICRNA03_A11       A11  LIC457       plant    c1_t07  2/10/24   \n",
       "B250508004_n01_LICRNA03_B11       B11  LIC458       plant    c1_t07  2/10/24   \n",
       "B250508004_n01_LICRNA03_C11       C11  LIC459       plant    c1_t07  2/10/24   \n",
       "B250508004_n01_LICRNA03_D11       D11  LIC460       plant    c1_t07  2/10/24   \n",
       "B250508004_n01_LICRNA03_E11       E11  LIC461       plant    c1_t07  2/10/24   \n",
       "B250508004_n01_LICRNA03_F11       F11  LIC462       plant    c1_t07  2/10/24   \n",
       "B250508004_n01_LICRNA03_G11       G11  LIC463       plant    c1_t07  2/10/24   \n",
       "B250508004_n01_LICRNA03_H11       H11  LIC464       plant    c1_t07  2/10/24   \n",
       "\n",
       "                              time extractplate rnaprepplate   Unmapped  \\\n",
       "filename                                                                  \n",
       "B250508004_n01_LICRNA03_A11  20:00   LICCIRC_01    LICRNA_03  2674528.0   \n",
       "B250508004_n01_LICRNA03_B11  20:00   LICCIRC_01    LICRNA_03  2708899.0   \n",
       "B250508004_n01_LICRNA03_C11  20:00   LICCIRC_01    LICRNA_03  9536959.0   \n",
       "B250508004_n01_LICRNA03_D11  20:00   LICCIRC_01    LICRNA_03  4996759.0   \n",
       "B250508004_n01_LICRNA03_E11  20:00   LICCIRC_01    LICRNA_03   874618.0   \n",
       "B250508004_n01_LICRNA03_F11  20:00   LICCIRC_01    LICRNA_03  1218515.0   \n",
       "B250508004_n01_LICRNA03_G11  20:00   LICCIRC_01    LICRNA_03  2261397.0   \n",
       "B250508004_n01_LICRNA03_H11  20:00   LICCIRC_01    LICRNA_03   326959.0   \n",
       "\n",
       "                             No Feature plate_row  plate_col  Total Reads  \\\n",
       "filename                                                                    \n",
       "B250508004_n01_LICRNA03_A11    376117.0         A         11    7080205.0   \n",
       "B250508004_n01_LICRNA03_B11    279807.0         B         11    5975492.0   \n",
       "B250508004_n01_LICRNA03_C11   1915565.0         C         11   22087076.0   \n",
       "B250508004_n01_LICRNA03_D11   1265822.0         D         11   14457644.0   \n",
       "B250508004_n01_LICRNA03_E11    149646.0         E         11    2408154.0   \n",
       "B250508004_n01_LICRNA03_F11    110644.0         F         11    2197530.0   \n",
       "B250508004_n01_LICRNA03_G11    492243.0         G         11    6315886.0   \n",
       "B250508004_n01_LICRNA03_H11     60346.0         H         11    1509679.0   \n",
       "\n",
       "                                    Experiment Type Sample Month  \\\n",
       "filename                                                           \n",
       "B250508004_n01_LICRNA03_A11  Circadian Experiment 1     February   \n",
       "B250508004_n01_LICRNA03_B11  Circadian Experiment 1     February   \n",
       "B250508004_n01_LICRNA03_C11  Circadian Experiment 1     February   \n",
       "B250508004_n01_LICRNA03_D11  Circadian Experiment 1     February   \n",
       "B250508004_n01_LICRNA03_E11  Circadian Experiment 1     February   \n",
       "B250508004_n01_LICRNA03_F11  Circadian Experiment 1     February   \n",
       "B250508004_n01_LICRNA03_G11  Circadian Experiment 1     February   \n",
       "B250508004_n01_LICRNA03_H11  Circadian Experiment 1     February   \n",
       "\n",
       "                             daysincestart       Date and Time  \\\n",
       "filename                                                         \n",
       "B250508004_n01_LICRNA03_A11            101 2024-02-10 20:00:00   \n",
       "B250508004_n01_LICRNA03_B11            101 2024-02-10 20:00:00   \n",
       "B250508004_n01_LICRNA03_C11            101 2024-02-10 20:00:00   \n",
       "B250508004_n01_LICRNA03_D11            101 2024-02-10 20:00:00   \n",
       "B250508004_n01_LICRNA03_E11            101 2024-02-10 20:00:00   \n",
       "B250508004_n01_LICRNA03_F11            101 2024-02-10 20:00:00   \n",
       "B250508004_n01_LICRNA03_G11            101 2024-02-10 20:00:00   \n",
       "B250508004_n01_LICRNA03_H11            101 2024-02-10 20:00:00   \n",
       "\n",
       "                             has_rare_species  lacking_core_species  \\\n",
       "filename                                                              \n",
       "B250508004_n01_LICRNA03_A11              True                 False   \n",
       "B250508004_n01_LICRNA03_B11              True                  True   \n",
       "B250508004_n01_LICRNA03_C11             False                 False   \n",
       "B250508004_n01_LICRNA03_D11              True                 False   \n",
       "B250508004_n01_LICRNA03_E11              True                  True   \n",
       "B250508004_n01_LICRNA03_F11              True                  True   \n",
       "B250508004_n01_LICRNA03_G11              True                  True   \n",
       "B250508004_n01_LICRNA03_H11              True                  True   \n",
       "\n",
       "                             rare_species_total_abundR100  \n",
       "filename                                                   \n",
       "B250508004_n01_LICRNA03_A11                           1.7  \n",
       "B250508004_n01_LICRNA03_B11                           5.9  \n",
       "B250508004_n01_LICRNA03_C11                           0.0  \n",
       "B250508004_n01_LICRNA03_D11                           4.0  \n",
       "B250508004_n01_LICRNA03_E11                           0.5  \n",
       "B250508004_n01_LICRNA03_F11                           2.3  \n",
       "B250508004_n01_LICRNA03_G11                           4.0  \n",
       "B250508004_n01_LICRNA03_H11                           2.9  "
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tp_metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "3fa39c20",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'timepoint': 't14', 'result': <pydeseq2.ds.DeseqStats at 0x18bcc7a10>},\n",
       " {'timepoint': 't15', 'result': <pydeseq2.ds.DeseqStats at 0x1981ce210>},\n",
       " {'timepoint': 't21', 'result': <pydeseq2.ds.DeseqStats at 0x1981cf4d0>},\n",
       " {'timepoint': 't27', 'result': <pydeseq2.ds.DeseqStats at 0x197282520>},\n",
       " {'timepoint': 't01', 'result': <pydeseq2.ds.DeseqStats at 0x197282780>},\n",
       " {'timepoint': 't02', 'result': <pydeseq2.ds.DeseqStats at 0x1972d5490>},\n",
       " {'timepoint': 't03', 'result': <pydeseq2.ds.DeseqStats at 0x193cb7bd0>},\n",
       " {'timepoint': 't05', 'result': <pydeseq2.ds.DeseqStats at 0x1972f4270>},\n",
       " {'timepoint': 't07', 'result': <pydeseq2.ds.DeseqStats at 0x1972f0650>},\n",
       " {'timepoint': 't12', 'result': <pydeseq2.ds.DeseqStats at 0x1972f0d50>},\n",
       " {'timepoint': 't13', 'result': <pydeseq2.ds.DeseqStats at 0x1972f1050>},\n",
       " {'timepoint': 't06', 'result': <pydeseq2.ds.DeseqStats at 0x1972f1250>},\n",
       " {'timepoint': 't09', 'result': <pydeseq2.ds.DeseqStats at 0x1972f1450>},\n",
       " {'timepoint': 'c1_t08', 'result': <pydeseq2.ds.DeseqStats at 0x1972f1650>},\n",
       " {'timepoint': 'c1_t09', 'result': <pydeseq2.ds.DeseqStats at 0x1972f1750>},\n",
       " {'timepoint': 'c1_t13', 'result': <pydeseq2.ds.DeseqStats at 0x1972f1950>},\n",
       " {'timepoint': 'c2_t01', 'result': <pydeseq2.ds.DeseqStats at 0x1972f1b50>},\n",
       " {'timepoint': 't08', 'result': <pydeseq2.ds.DeseqStats at 0x1972f1d50>},\n",
       " {'timepoint': 'c1_t01', 'result': <pydeseq2.ds.DeseqStats at 0x1972f1f50>}]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list_of_differential_expression_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "c2c9b447",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Fitting size factors...\n",
      "... done in 0.00 seconds.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "==================================================\n",
      "Processing timepoint: t14 (random split)\n",
      "==================================================\n",
      "Group 1: 3 samples, Group 2: 4 samples\n",
      "Number of normal samples: 3\n",
      "Number of perturbed samples: 4\n",
      "Using None as control genes, passed at DeseqDataSet initialization\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Fitting dispersions...\n",
      "... done in 3.41 seconds.\n",
      "\n",
      "Fitting dispersion trend curve...\n",
      "... done in 0.45 seconds.\n",
      "\n",
      "Fitting MAP dispersions...\n",
      "... done in 3.50 seconds.\n",
      "\n",
      "Fitting LFCs...\n",
      "... done in 2.57 seconds.\n",
      "\n",
      "Calculating cook's distance...\n",
      "... done in 0.01 seconds.\n",
      "\n",
      "Replacing 0 outlier genes.\n",
      "\n",
      "Running Wald tests...\n",
      "... done in 1.00 seconds.\n",
      "\n",
      "Fitting size factors...\n",
      "... done in 0.01 seconds.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log2 fold change & Wald test p-value: random_split perturbed vs normal\n",
      "              baseMean  log2FoldChange     lfcSE      stat    pvalue      padj\n",
      "AT1G01010    23.042526        0.162051  0.701418  0.231033  0.817289  0.999581\n",
      "AT1G01020    84.253266       -0.373938  0.385710 -0.969479  0.332306  0.999581\n",
      "AT1G01030    44.017698        0.095325  0.438248  0.217514  0.827808  0.999581\n",
      "AT1G01040   170.361128       -0.037911  0.289040 -0.131161  0.895648  0.999581\n",
      "AT1G01050   255.120935        0.125416  0.281603  0.445365  0.656056  0.999581\n",
      "...                ...             ...       ...       ...       ...       ...\n",
      "ArthCr089  2316.079433        0.078219  0.426205  0.183524  0.854387  0.999581\n",
      "ArthCr088   210.774282       -0.273548  0.463582 -0.590074  0.555141  0.999581\n",
      "ArthCp088     9.797099       -1.078325  1.055991 -1.021149  0.307184  0.999581\n",
      "ArthCp083    18.783767        0.447837  0.748754  0.598109  0.549767  0.999581\n",
      "ArthCp085    10.738686       -2.192592  0.925471 -2.369165  0.017828  0.999581\n",
      "\n",
      "[17681 rows x 6 columns]\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: t15 (random split)\n",
      "==================================================\n",
      "Group 1: 4 samples, Group 2: 4 samples\n",
      "Number of normal samples: 4\n",
      "Number of perturbed samples: 4\n",
      "Using None as control genes, passed at DeseqDataSet initialization\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Fitting dispersions...\n",
      "... done in 3.20 seconds.\n",
      "\n",
      "Fitting dispersion trend curve...\n",
      "... done in 0.47 seconds.\n",
      "\n",
      "Fitting MAP dispersions...\n",
      "... done in 3.48 seconds.\n",
      "\n",
      "Fitting LFCs...\n",
      "... done in 2.40 seconds.\n",
      "\n",
      "Calculating cook's distance...\n",
      "... done in 0.01 seconds.\n",
      "\n",
      "Replacing 0 outlier genes.\n",
      "\n",
      "Running Wald tests...\n",
      "... done in 0.93 seconds.\n",
      "\n",
      "Fitting size factors...\n",
      "... done in 0.01 seconds.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log2 fold change & Wald test p-value: random_split perturbed vs normal\n",
      "              baseMean  log2FoldChange     lfcSE      stat    pvalue      padj\n",
      "AT1G01010    11.080991        0.555187  1.090324  0.509194  0.610616  0.999729\n",
      "AT1G01020    63.765101        0.633406  0.322200  1.965879  0.049313  0.999729\n",
      "AT1G01030    18.677859       -0.181133  0.593108 -0.305396  0.760065  0.999729\n",
      "AT1G01040   149.392352        0.120520  0.226846  0.531283  0.595223  0.999729\n",
      "AT1G01050   194.570799        0.003157  0.267135  0.011819  0.990570  0.999729\n",
      "...                ...             ...       ...       ...       ...       ...\n",
      "ArthCr089  2355.193560        0.330655  0.244097  1.354601  0.175545  0.999729\n",
      "ArthCr088   166.575863        0.499448  0.291469  1.713553  0.086611  0.999729\n",
      "ArthCp088    15.402725        1.032431  0.677678  1.523484  0.127638  0.999729\n",
      "ArthCp083    37.411412        0.330951  0.536595  0.616763  0.537391  0.999729\n",
      "ArthCp085     7.482260        0.358379  1.029981  0.347948  0.727880  0.999729\n",
      "\n",
      "[17681 rows x 6 columns]\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: t21 (random split)\n",
      "==================================================\n",
      "Group 1: 4 samples, Group 2: 4 samples\n",
      "Number of normal samples: 4\n",
      "Number of perturbed samples: 4\n",
      "Using None as control genes, passed at DeseqDataSet initialization\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Fitting dispersions...\n",
      "... done in 3.25 seconds.\n",
      "\n",
      "Fitting dispersion trend curve...\n",
      "... done in 0.47 seconds.\n",
      "\n",
      "Fitting MAP dispersions...\n",
      "... done in 3.65 seconds.\n",
      "\n",
      "Fitting LFCs...\n",
      "... done in 2.49 seconds.\n",
      "\n",
      "Calculating cook's distance...\n",
      "... done in 0.01 seconds.\n",
      "\n",
      "Replacing 0 outlier genes.\n",
      "\n",
      "Running Wald tests...\n",
      "... done in 0.93 seconds.\n",
      "\n",
      "Fitting size factors...\n",
      "... done in 0.01 seconds.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log2 fold change & Wald test p-value: random_split perturbed vs normal\n",
      "              baseMean  log2FoldChange     lfcSE      stat    pvalue      padj\n",
      "AT1G01010    14.734969        2.208043  1.123597  1.965155  0.049396  0.792238\n",
      "AT1G01020    48.166910       -0.477646  0.521391 -0.916098  0.359615  0.966033\n",
      "AT1G01030    10.790900       -0.607121  0.887598 -0.684005  0.493972       NaN\n",
      "AT1G01040   144.750024        0.206963  0.343400  0.602687  0.546717  0.983647\n",
      "AT1G01050   254.732610        0.462874  0.330529  1.400402  0.161393  0.935077\n",
      "...                ...             ...       ...       ...       ...       ...\n",
      "ArthCr089  3583.848026        0.753927  0.260441  2.894811  0.003794  0.412401\n",
      "ArthCr088   269.523568        0.594458  0.237268  2.505434  0.012230  0.565003\n",
      "ArthCp088     8.504604        0.426282  1.017605  0.418907  0.675284       NaN\n",
      "ArthCp083    49.194937        0.139602  0.454391  0.307229  0.758669  0.995193\n",
      "ArthCp085    11.383310        1.445786  0.937907  1.541503  0.123194       NaN\n",
      "\n",
      "[17681 rows x 6 columns]\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: t27 (random split)\n",
      "==================================================\n",
      "Group 1: 4 samples, Group 2: 4 samples\n",
      "Number of normal samples: 4\n",
      "Number of perturbed samples: 4\n",
      "Using None as control genes, passed at DeseqDataSet initialization\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Fitting dispersions...\n",
      "... done in 3.19 seconds.\n",
      "\n",
      "Fitting dispersion trend curve...\n",
      "... done in 0.46 seconds.\n",
      "\n",
      "Fitting MAP dispersions...\n",
      "... done in 3.56 seconds.\n",
      "\n",
      "Fitting LFCs...\n",
      "... done in 2.24 seconds.\n",
      "\n",
      "Calculating cook's distance...\n",
      "... done in 0.01 seconds.\n",
      "\n",
      "Replacing 0 outlier genes.\n",
      "\n",
      "Running Wald tests...\n",
      "... done in 1.01 seconds.\n",
      "\n",
      "Fitting size factors...\n",
      "... done in 0.00 seconds.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log2 fold change & Wald test p-value: random_split perturbed vs normal\n",
      "              baseMean  log2FoldChange     lfcSE      stat    pvalue      padj\n",
      "AT1G01010    22.701126        0.207773  0.633617  0.327915  0.742976  0.999975\n",
      "AT1G01020    78.400483       -0.734302  0.375750 -1.954230  0.050674  0.999975\n",
      "AT1G01030    25.868840       -0.856776  0.585855 -1.462437  0.143622  0.999975\n",
      "AT1G01040   206.690808       -0.113277  0.258723 -0.437833  0.661507  0.999975\n",
      "AT1G01050   368.018417        0.234727  0.243864  0.962535  0.335781  0.999975\n",
      "...                ...             ...       ...       ...       ...       ...\n",
      "ArthCr089  3462.477424       -0.029957  0.477967 -0.062675  0.950025  0.999975\n",
      "ArthCr088   353.771856       -0.147371  0.360245 -0.409086  0.682476  0.999975\n",
      "ArthCp088    12.499028       -0.461355  0.738480 -0.624736  0.532144  0.999975\n",
      "ArthCp083    56.446051        0.261823  0.447001  0.585734  0.558055  0.999975\n",
      "ArthCp085    12.084169       -0.396020  0.918758 -0.431038  0.666441  0.999975\n",
      "\n",
      "[17681 rows x 6 columns]\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: t01 (random split)\n",
      "==================================================\n",
      "Group 1: 4 samples, Group 2: 4 samples\n",
      "Number of normal samples: 4\n",
      "Number of perturbed samples: 4\n",
      "Using None as control genes, passed at DeseqDataSet initialization\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Fitting dispersions...\n",
      "... done in 3.38 seconds.\n",
      "\n",
      "Fitting dispersion trend curve...\n",
      "... done in 0.45 seconds.\n",
      "\n",
      "Fitting MAP dispersions...\n",
      "... done in 3.44 seconds.\n",
      "\n",
      "Fitting LFCs...\n",
      "... done in 2.52 seconds.\n",
      "\n",
      "Calculating cook's distance...\n",
      "... done in 0.01 seconds.\n",
      "\n",
      "Replacing 0 outlier genes.\n",
      "\n",
      "Running Wald tests...\n",
      "... done in 0.98 seconds.\n",
      "\n",
      "Fitting size factors...\n",
      "... done in 0.00 seconds.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log2 fold change & Wald test p-value: random_split perturbed vs normal\n",
      "             baseMean  log2FoldChange     lfcSE      stat    pvalue      padj\n",
      "AT1G01010   16.732559       -0.751263  0.716064 -1.049156  0.294106  0.999343\n",
      "AT1G01020   20.742664        1.051663  0.790537  1.330314  0.183415  0.999343\n",
      "AT1G01030   12.787656       -0.616610  1.031987 -0.597498  0.550175  0.999343\n",
      "AT1G01040   64.732276       -0.200861  0.510853 -0.393187  0.694181  0.999343\n",
      "AT1G01050  143.779824        0.034156  0.303044  0.112710  0.910260  0.999343\n",
      "...               ...             ...       ...       ...       ...       ...\n",
      "ArthCr089  811.218826        0.114720  0.311446  0.368347  0.712614  0.999343\n",
      "ArthCr088   74.673964        1.012281  0.514225  1.968557  0.049004  0.999343\n",
      "ArthCp088    0.187393       -1.151127  3.730461 -0.308575  0.757645  0.999343\n",
      "ArthCp083    4.532873       -0.249408  1.447569 -0.172294  0.863206  0.999343\n",
      "ArthCp085    2.133608       -0.025914  1.908794 -0.013576  0.989168  0.999343\n",
      "\n",
      "[17681 rows x 6 columns]\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: t02 (random split)\n",
      "==================================================\n",
      "Group 1: 4 samples, Group 2: 4 samples\n",
      "Number of normal samples: 4\n",
      "Number of perturbed samples: 4\n",
      "Using None as control genes, passed at DeseqDataSet initialization\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Fitting dispersions...\n",
      "... done in 3.47 seconds.\n",
      "\n",
      "Fitting dispersion trend curve...\n",
      "... done in 0.47 seconds.\n",
      "\n",
      "Fitting MAP dispersions...\n",
      "... done in 3.57 seconds.\n",
      "\n",
      "Fitting LFCs...\n",
      "... done in 2.86 seconds.\n",
      "\n",
      "Calculating cook's distance...\n",
      "... done in 0.01 seconds.\n",
      "\n",
      "Replacing 0 outlier genes.\n",
      "\n",
      "Running Wald tests...\n",
      "... done in 0.97 seconds.\n",
      "\n",
      "Fitting size factors...\n",
      "... done in 0.01 seconds.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log2 fold change & Wald test p-value: random_split perturbed vs normal\n",
      "              baseMean  log2FoldChange     lfcSE      stat    pvalue      padj\n",
      "AT1G01010     5.828940       -3.010080  2.210608 -1.361652  0.173308  0.999899\n",
      "AT1G01020    41.755678        0.811378  0.731672  1.108937  0.267457  0.999899\n",
      "AT1G01030    27.669855        1.117928  0.959363  1.165282  0.243905  0.999899\n",
      "AT1G01040   101.171370       -0.169621  0.552446 -0.307037  0.758816  0.999899\n",
      "AT1G01050   112.721100        0.431688  0.550672  0.783929  0.433082  0.999899\n",
      "...                ...             ...       ...       ...       ...       ...\n",
      "ArthCr089  1269.615638        0.441114  0.477622  0.923563  0.355714  0.999899\n",
      "ArthCr088   107.427765        0.828749  0.756994  1.094789  0.273609  0.999899\n",
      "ArthCp088     0.102665        0.768431  3.876498  0.198228  0.842867  0.999899\n",
      "ArthCp083     7.631815        0.000105  2.174334  0.000048  0.999962  0.999968\n",
      "ArthCp085     4.798149        0.305286  1.894098  0.161178  0.871953  0.999899\n",
      "\n",
      "[17681 rows x 6 columns]\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: t03 (random split)\n",
      "==================================================\n",
      "Group 1: 3 samples, Group 2: 4 samples\n",
      "Number of normal samples: 3\n",
      "Number of perturbed samples: 4\n",
      "Using None as control genes, passed at DeseqDataSet initialization\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Fitting dispersions...\n",
      "... done in 3.33 seconds.\n",
      "\n",
      "Fitting dispersion trend curve...\n",
      "... done in 0.44 seconds.\n",
      "\n",
      "Fitting MAP dispersions...\n",
      "... done in 3.72 seconds.\n",
      "\n",
      "Fitting LFCs...\n",
      "... done in 2.47 seconds.\n",
      "\n",
      "Calculating cook's distance...\n",
      "... done in 0.01 seconds.\n",
      "\n",
      "Replacing 0 outlier genes.\n",
      "\n",
      "Running Wald tests...\n",
      "... done in 0.94 seconds.\n",
      "\n",
      "Fitting size factors...\n",
      "... done in 0.00 seconds.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log2 fold change & Wald test p-value: random_split perturbed vs normal\n",
      "             baseMean  log2FoldChange     lfcSE      stat    pvalue      padj\n",
      "AT1G01010   12.673318        2.193587  1.227035  1.787714  0.073822  0.999826\n",
      "AT1G01020   51.535167       -0.398765  0.558594 -0.713874  0.475305  0.999826\n",
      "AT1G01030   33.619525       -0.337936  0.568506 -0.594429  0.552225  0.999826\n",
      "AT1G01040  111.160668        0.393967  0.398107  0.989601  0.322369  0.999826\n",
      "AT1G01050  229.082971       -0.070597  0.297352 -0.237419  0.812332  0.999826\n",
      "...               ...             ...       ...       ...       ...       ...\n",
      "ArthCr089  623.470960       -0.089570  0.267282 -0.335115  0.737538  0.999826\n",
      "ArthCr088   41.439495       -0.368391  0.639607 -0.575965  0.564639  0.999826\n",
      "ArthCp088    4.623443       -0.027481  1.982870 -0.013859  0.988942  0.999826\n",
      "ArthCp083    6.067498       -1.061111  1.641199 -0.646546  0.517926  0.999826\n",
      "ArthCp085    3.441357        0.652310  1.920803  0.339603  0.734156  0.999826\n",
      "\n",
      "[17681 rows x 6 columns]\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: t05 (random split)\n",
      "==================================================\n",
      "Group 1: 4 samples, Group 2: 4 samples\n",
      "Number of normal samples: 4\n",
      "Number of perturbed samples: 4\n",
      "Using None as control genes, passed at DeseqDataSet initialization\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Fitting dispersions...\n",
      "... done in 3.40 seconds.\n",
      "\n",
      "Fitting dispersion trend curve...\n",
      "... done in 0.43 seconds.\n",
      "\n",
      "Fitting MAP dispersions...\n",
      "... done in 3.53 seconds.\n",
      "\n",
      "Fitting LFCs...\n",
      "... done in 2.46 seconds.\n",
      "\n",
      "Calculating cook's distance...\n",
      "... done in 0.01 seconds.\n",
      "\n",
      "Replacing 0 outlier genes.\n",
      "\n",
      "Running Wald tests...\n",
      "... done in 0.97 seconds.\n",
      "\n",
      "Fitting size factors...\n",
      "... done in 0.01 seconds.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log2 fold change & Wald test p-value: random_split perturbed vs normal\n",
      "             baseMean  log2FoldChange     lfcSE      stat    pvalue      padj\n",
      "AT1G01010    6.932322        0.376454  0.956737  0.393476  0.693968  0.999762\n",
      "AT1G01020   23.651654       -0.494539  0.580006 -0.852643  0.393857  0.999762\n",
      "AT1G01030   14.073083        0.293119  0.624654  0.469250  0.638891  0.999762\n",
      "AT1G01040   52.306085        0.616285  0.387634  1.589863  0.111866  0.999762\n",
      "AT1G01050  119.268157       -0.035473  0.350002 -0.101349  0.919273  0.999762\n",
      "...               ...             ...       ...       ...       ...       ...\n",
      "ArthCr089  352.901763        0.476410  0.348318  1.367745  0.171392  0.999762\n",
      "ArthCr088   39.357860        0.984949  0.504496  1.952344  0.050897  0.999762\n",
      "ArthCp088    1.352288       -0.015433  1.941258 -0.007950  0.993657  0.999762\n",
      "ArthCp083    3.105030        0.890318  1.468141  0.606425  0.544232  0.999762\n",
      "ArthCp085    4.022016       -0.700569  1.465497 -0.478042  0.632621  0.999762\n",
      "\n",
      "[17681 rows x 6 columns]\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: t07 (random split)\n",
      "==================================================\n",
      "Group 1: 3 samples, Group 2: 4 samples\n",
      "Number of normal samples: 3\n",
      "Number of perturbed samples: 4\n",
      "Using None as control genes, passed at DeseqDataSet initialization\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Fitting dispersions...\n",
      "... done in 3.35 seconds.\n",
      "\n",
      "Fitting dispersion trend curve...\n",
      "... done in 0.43 seconds.\n",
      "\n",
      "Fitting MAP dispersions...\n",
      "... done in 3.61 seconds.\n",
      "\n",
      "Fitting LFCs...\n",
      "... done in 2.39 seconds.\n",
      "\n",
      "Calculating cook's distance...\n",
      "... done in 0.01 seconds.\n",
      "\n",
      "Replacing 0 outlier genes.\n",
      "\n",
      "Running Wald tests...\n",
      "... done in 0.94 seconds.\n",
      "\n",
      "Fitting size factors...\n",
      "... done in 0.01 seconds.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log2 fold change & Wald test p-value: random_split perturbed vs normal\n",
      "             baseMean  log2FoldChange     lfcSE      stat    pvalue      padj\n",
      "AT1G01010    9.075198       -0.223374  0.861280 -0.259352  0.795364  0.999845\n",
      "AT1G01020   49.354920        0.368166  0.464707  0.792254  0.428213  0.999845\n",
      "AT1G01030   17.667445        0.302837  0.564455  0.536512  0.591605  0.999845\n",
      "AT1G01040  117.384550        0.367903  0.317832  1.157540  0.247052  0.999845\n",
      "AT1G01050  287.272888       -0.143429  0.242739 -0.590875  0.554604  0.999845\n",
      "...               ...             ...       ...       ...       ...       ...\n",
      "ArthCr089  562.888905       -0.740342  0.338649 -2.186163  0.028804  0.999845\n",
      "ArthCr088   55.967579       -0.500624  0.404120 -1.238800  0.215420  0.999845\n",
      "ArthCp088    8.208301        0.886851  1.096645  0.808695  0.418691  0.999845\n",
      "ArthCp083   12.867363        0.377439  0.801259  0.471058  0.637599  0.999845\n",
      "ArthCp085    9.425587       -0.281776  0.771434 -0.365263  0.714915  0.999845\n",
      "\n",
      "[17681 rows x 6 columns]\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: t12 (random split)\n",
      "==================================================\n",
      "Group 1: 3 samples, Group 2: 3 samples\n",
      "Number of normal samples: 3\n",
      "Number of perturbed samples: 3\n",
      "Using None as control genes, passed at DeseqDataSet initialization\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Fitting dispersions...\n",
      "... done in 3.33 seconds.\n",
      "\n",
      "Fitting dispersion trend curve...\n",
      "... done in 0.46 seconds.\n",
      "\n",
      "Fitting MAP dispersions...\n",
      "... done in 3.44 seconds.\n",
      "\n",
      "Fitting LFCs...\n",
      "... done in 2.40 seconds.\n",
      "\n",
      "Calculating cook's distance...\n",
      "... done in 0.01 seconds.\n",
      "\n",
      "Replacing 0 outlier genes.\n",
      "\n",
      "Running Wald tests...\n",
      "... done in 1.05 seconds.\n",
      "\n",
      "Fitting size factors...\n",
      "... done in 0.00 seconds.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log2 fold change & Wald test p-value: random_split perturbed vs normal\n",
      "             baseMean  log2FoldChange     lfcSE      stat    pvalue      padj\n",
      "AT1G01010   17.131550        0.173144  0.902278  0.191897  0.847823  0.999928\n",
      "AT1G01020   42.763589       -0.821563  0.487400 -1.685603  0.091872  0.999928\n",
      "AT1G01030    9.323262        0.244718  0.848103  0.288548  0.772928  0.999928\n",
      "AT1G01040   80.984725       -0.048125  0.486376 -0.098946  0.921181  0.999928\n",
      "AT1G01050  205.670125       -0.170488  0.291146 -0.585577  0.558160  0.999928\n",
      "...               ...             ...       ...       ...       ...       ...\n",
      "ArthCr089  723.254919       -0.108282  0.497468 -0.217667  0.827689  0.999928\n",
      "ArthCr088   48.604026        0.247194  0.584859  0.422655  0.672547  0.999928\n",
      "ArthCp088    6.236000        1.295598  1.544982  0.838584  0.401703  0.999928\n",
      "ArthCp083    6.901236        0.192128  1.177072  0.163225  0.870341  0.999928\n",
      "ArthCp085    7.028430        1.031603  1.295388  0.796366  0.425819  0.999928\n",
      "\n",
      "[17681 rows x 6 columns]\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: t13 (random split)\n",
      "==================================================\n",
      "Group 1: 3 samples, Group 2: 4 samples\n",
      "Number of normal samples: 3\n",
      "Number of perturbed samples: 4\n",
      "Using None as control genes, passed at DeseqDataSet initialization\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Fitting dispersions...\n",
      "... done in 3.30 seconds.\n",
      "\n",
      "Fitting dispersion trend curve...\n",
      "... done in 0.44 seconds.\n",
      "\n",
      "Fitting MAP dispersions...\n",
      "... done in 3.62 seconds.\n",
      "\n",
      "Fitting LFCs...\n",
      "... done in 2.33 seconds.\n",
      "\n",
      "Calculating cook's distance...\n",
      "... done in 0.01 seconds.\n",
      "\n",
      "Replacing 0 outlier genes.\n",
      "\n",
      "Running Wald tests...\n",
      "... done in 0.96 seconds.\n",
      "\n",
      "Fitting size factors...\n",
      "... done in 0.00 seconds.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log2 fold change & Wald test p-value: random_split perturbed vs normal\n",
      "             baseMean  log2FoldChange     lfcSE      stat    pvalue      padj\n",
      "AT1G01010   10.573842       -1.193577  0.793129 -1.504897  0.132351  0.999416\n",
      "AT1G01020   45.383900        0.101629  0.434712  0.233785  0.815152  0.999416\n",
      "AT1G01030   18.919309        0.594678  0.602228  0.987463  0.323416  0.999416\n",
      "AT1G01040  104.869243       -0.500943  0.377900 -1.325596  0.184973  0.999416\n",
      "AT1G01050  240.106429        0.102635  0.291433  0.352172  0.724709  0.999416\n",
      "...               ...             ...       ...       ...       ...       ...\n",
      "ArthCr089  223.172557        0.065988  0.521571  0.126519  0.899321  0.999416\n",
      "ArthCr088   26.706730        0.431067  0.607159  0.709975  0.477720  0.999416\n",
      "ArthCp088    9.188339        0.944250  0.911063  1.036426  0.300003  0.999416\n",
      "ArthCp083    8.470385        0.297932  0.933599  0.319122  0.749634  0.999416\n",
      "ArthCp085    7.445681        0.442350  0.939745  0.470712  0.637846  0.999416\n",
      "\n",
      "[17681 rows x 6 columns]\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: t06 (random split)\n",
      "==================================================\n",
      "Group 1: 3 samples, Group 2: 4 samples\n",
      "Number of normal samples: 3\n",
      "Number of perturbed samples: 4\n",
      "Using None as control genes, passed at DeseqDataSet initialization\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Fitting dispersions...\n",
      "... done in 3.51 seconds.\n",
      "\n",
      "Fitting dispersion trend curve...\n",
      "... done in 0.48 seconds.\n",
      "\n",
      "Fitting MAP dispersions...\n",
      "... done in 3.56 seconds.\n",
      "\n",
      "Fitting LFCs...\n",
      "... done in 2.31 seconds.\n",
      "\n",
      "Calculating cook's distance...\n",
      "... done in 0.01 seconds.\n",
      "\n",
      "Replacing 0 outlier genes.\n",
      "\n",
      "Running Wald tests...\n",
      "... done in 0.98 seconds.\n",
      "\n",
      "Fitting size factors...\n",
      "... done in 0.01 seconds.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log2 fold change & Wald test p-value: random_split perturbed vs normal\n",
      "             baseMean  log2FoldChange     lfcSE      stat    pvalue      padj\n",
      "AT1G01010    8.327423        0.422466  1.130395  0.373733  0.708603  0.999877\n",
      "AT1G01020   57.483968        0.213982  0.376482  0.568371  0.569783  0.999877\n",
      "AT1G01030   33.288089        0.760163  0.566401  1.342093  0.179566  0.999877\n",
      "AT1G01040   94.194888       -0.045738  0.322908 -0.141644  0.887361  0.999877\n",
      "AT1G01050  249.921278        0.121515  0.306106  0.396972  0.691388  0.999877\n",
      "...               ...             ...       ...       ...       ...       ...\n",
      "ArthCr089  506.040298       -0.183199  0.320959 -0.570788  0.568144  0.999877\n",
      "ArthCr088   51.157487       -0.401168  0.404730 -0.991200  0.321588  0.999877\n",
      "ArthCp088   10.959722        0.684904  0.900015  0.760992  0.446662  0.999877\n",
      "ArthCp083   14.438688        0.141320  1.030031  0.137200  0.890873  0.999877\n",
      "ArthCp085    8.047360        0.309932  0.900520  0.344170  0.730718  0.999877\n",
      "\n",
      "[17681 rows x 6 columns]\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: t09 (random split)\n",
      "==================================================\n",
      "Group 1: 3 samples, Group 2: 4 samples\n",
      "Number of normal samples: 3\n",
      "Number of perturbed samples: 4\n",
      "Using None as control genes, passed at DeseqDataSet initialization\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Fitting dispersions...\n",
      "... done in 3.38 seconds.\n",
      "\n",
      "Fitting dispersion trend curve...\n",
      "... done in 0.44 seconds.\n",
      "\n",
      "Fitting MAP dispersions...\n",
      "... done in 3.46 seconds.\n",
      "\n",
      "Fitting LFCs...\n",
      "... done in 2.30 seconds.\n",
      "\n",
      "Calculating cook's distance...\n",
      "... done in 0.01 seconds.\n",
      "\n",
      "Replacing 0 outlier genes.\n",
      "\n",
      "Running Wald tests...\n",
      "... done in 0.97 seconds.\n",
      "\n",
      "Fitting size factors...\n",
      "... done in 0.01 seconds.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log2 fold change & Wald test p-value: random_split perturbed vs normal\n",
      "             baseMean  log2FoldChange     lfcSE      stat    pvalue      padj\n",
      "AT1G01010    9.360369       -0.421186  0.970707 -0.433896  0.664364  0.999816\n",
      "AT1G01020   67.102513       -0.026360  0.335442 -0.078582  0.937365  0.999816\n",
      "AT1G01030   22.755263       -0.019959  0.537199 -0.037154  0.970362  0.999816\n",
      "AT1G01040  111.470100       -0.220659  0.317599 -0.694771  0.487199  0.999816\n",
      "AT1G01050  290.369334       -0.038933  0.280613 -0.138741  0.889655  0.999816\n",
      "...               ...             ...       ...       ...       ...       ...\n",
      "ArthCr089  674.780966        0.199600  0.315043  0.633563  0.526366  0.999816\n",
      "ArthCr088   61.928726       -0.231366  0.386895 -0.598006  0.549836  0.999816\n",
      "ArthCp088    9.035784       -0.409233  0.734923 -0.556837  0.577639  0.999816\n",
      "ArthCp083   14.608172        0.650300  0.665917  0.976548  0.328793  0.999816\n",
      "ArthCp085   10.522192        0.170579  0.716633  0.238028  0.811859  0.999816\n",
      "\n",
      "[17681 rows x 6 columns]\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: c1_t08 (random split)\n",
      "==================================================\n",
      "Group 1: 8 samples, Group 2: 8 samples\n",
      "Number of normal samples: 8\n",
      "Number of perturbed samples: 8\n",
      "Using None as control genes, passed at DeseqDataSet initialization\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Fitting dispersions...\n",
      "... done in 3.52 seconds.\n",
      "\n",
      "Fitting dispersion trend curve...\n",
      "... done in 0.43 seconds.\n",
      "\n",
      "Fitting MAP dispersions...\n",
      "... done in 3.71 seconds.\n",
      "\n",
      "Fitting LFCs...\n",
      "... done in 2.63 seconds.\n",
      "\n",
      "Calculating cook's distance...\n",
      "... done in 0.02 seconds.\n",
      "\n",
      "Replacing 87 outlier genes.\n",
      "\n",
      "Fitting dispersions...\n",
      "... done in 0.05 seconds.\n",
      "\n",
      "Fitting MAP dispersions...\n",
      "... done in 0.04 seconds.\n",
      "\n",
      "Fitting LFCs...\n",
      "... done in 0.05 seconds.\n",
      "\n",
      "Running Wald tests...\n",
      "... done in 1.01 seconds.\n",
      "\n",
      "Fitting size factors...\n",
      "... done in 0.00 seconds.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log2 fold change & Wald test p-value: random_split perturbed vs normal\n",
      "             baseMean  log2FoldChange     lfcSE      stat    pvalue     padj\n",
      "AT1G01010    5.462072       -0.461066  1.060510 -0.434759  0.663738  0.99935\n",
      "AT1G01020   53.215496       -0.016622  0.321936 -0.051630  0.958823  0.99935\n",
      "AT1G01030   14.328351       -0.057356  0.664086 -0.086369  0.931173  0.99935\n",
      "AT1G01040  111.308722        0.288188  0.242738  1.187239  0.235133  0.99935\n",
      "AT1G01050  222.646625       -0.036620  0.189437 -0.193308  0.846718  0.99935\n",
      "...               ...             ...       ...       ...       ...      ...\n",
      "ArthCr089  808.504968        0.665305  0.493632  1.347776  0.177730  0.99935\n",
      "ArthCr088   29.639512        0.274734  0.364808  0.753090  0.451396  0.99935\n",
      "ArthCp088   10.022036       -1.138676  0.740745 -1.537205  0.124243  0.99935\n",
      "ArthCp083   16.646939        0.400669  0.589203  0.680019  0.496493  0.99935\n",
      "ArthCp085   17.456079       -0.119266  0.708035 -0.168446  0.866232  0.99935\n",
      "\n",
      "[17681 rows x 6 columns]\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: c1_t09 (random split)\n",
      "==================================================\n",
      "Group 1: 3 samples, Group 2: 4 samples\n",
      "Number of normal samples: 3\n",
      "Number of perturbed samples: 4\n",
      "Using None as control genes, passed at DeseqDataSet initialization\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Fitting dispersions...\n",
      "... done in 3.31 seconds.\n",
      "\n",
      "Fitting dispersion trend curve...\n",
      "... done in 0.46 seconds.\n",
      "\n",
      "Fitting MAP dispersions...\n",
      "... done in 3.61 seconds.\n",
      "\n",
      "Fitting LFCs...\n",
      "... done in 2.59 seconds.\n",
      "\n",
      "Calculating cook's distance...\n",
      "... done in 0.01 seconds.\n",
      "\n",
      "Replacing 0 outlier genes.\n",
      "\n",
      "Running Wald tests...\n",
      "... done in 0.95 seconds.\n",
      "\n",
      "Fitting size factors...\n",
      "... done in 0.01 seconds.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log2 fold change & Wald test p-value: random_split perturbed vs normal\n",
      "             baseMean  log2FoldChange     lfcSE      stat    pvalue      padj\n",
      "AT1G01010   17.137315       -1.037161  1.146364 -0.904739  0.365603  0.999956\n",
      "AT1G01020   59.802847        0.861294  0.776303  1.109481  0.267223  0.999956\n",
      "AT1G01030   13.732965       -0.049554  1.228888 -0.040324  0.967835  0.999956\n",
      "AT1G01040  131.260076       -0.062324  0.461356 -0.135089  0.892542  0.999956\n",
      "AT1G01050  280.135775        0.203812  0.295473  0.689782  0.490331  0.999956\n",
      "...               ...             ...       ...       ...       ...       ...\n",
      "ArthCr089  625.415151       -0.092032  0.242451 -0.379589  0.704251  0.999956\n",
      "ArthCr088   48.420611       -1.181616  0.526995 -2.242176  0.024950  0.999956\n",
      "ArthCp088    4.190219       -0.100852  1.977412 -0.051002  0.959324  0.999956\n",
      "ArthCp083   14.071409       -1.037072  1.358594 -0.763342  0.445259  0.999956\n",
      "ArthCp085   10.497329       -0.362398  1.292852 -0.280309  0.779240  0.999956\n",
      "\n",
      "[17681 rows x 6 columns]\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: c1_t13 (random split)\n",
      "==================================================\n",
      "Group 1: 3 samples, Group 2: 4 samples\n",
      "Number of normal samples: 3\n",
      "Number of perturbed samples: 4\n",
      "Using None as control genes, passed at DeseqDataSet initialization\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Fitting dispersions...\n",
      "... done in 3.30 seconds.\n",
      "\n",
      "Fitting dispersion trend curve...\n",
      "... done in 0.44 seconds.\n",
      "\n",
      "Fitting MAP dispersions...\n",
      "... done in 3.62 seconds.\n",
      "\n",
      "Fitting LFCs...\n",
      "... done in 2.61 seconds.\n",
      "\n",
      "Calculating cook's distance...\n",
      "... done in 0.01 seconds.\n",
      "\n",
      "Replacing 0 outlier genes.\n",
      "\n",
      "Running Wald tests...\n",
      "... done in 0.94 seconds.\n",
      "\n",
      "Fitting size factors...\n",
      "... done in 0.00 seconds.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log2 fold change & Wald test p-value: random_split perturbed vs normal\n",
      "             baseMean  log2FoldChange     lfcSE      stat    pvalue      padj\n",
      "AT1G01010    5.043403       -0.452274  2.097996 -0.215574  0.829320  0.999965\n",
      "AT1G01020   50.136346       -0.578587  0.492984 -1.173642  0.240538  0.999965\n",
      "AT1G01030   21.925333        0.625633  0.935318  0.668898  0.503560  0.999965\n",
      "AT1G01040  149.302430        0.827473  0.529099  1.563928  0.117835  0.999965\n",
      "AT1G01050  189.922085        0.165516  0.384979  0.429935  0.667243  0.999965\n",
      "...               ...             ...       ...       ...       ...       ...\n",
      "ArthCr089  389.520596       -0.282434  0.448464 -0.629780  0.528838  0.999965\n",
      "ArthCr088   14.661313       -0.335735  0.813792 -0.412557  0.679931  0.999965\n",
      "ArthCp088    3.343409       -2.023961  2.603559 -0.777382  0.436933  0.999965\n",
      "ArthCp083   13.098330        0.194566  1.147813  0.169510  0.865395  0.999965\n",
      "ArthCp085   20.559756        0.002682  0.750783  0.003572  0.997150  0.999965\n",
      "\n",
      "[17681 rows x 6 columns]\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: c2_t01 (random split)\n",
      "==================================================\n",
      "Group 1: 3 samples, Group 2: 4 samples\n",
      "Number of normal samples: 3\n",
      "Number of perturbed samples: 4\n",
      "Using None as control genes, passed at DeseqDataSet initialization\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Fitting dispersions...\n",
      "... done in 3.44 seconds.\n",
      "\n",
      "Fitting dispersion trend curve...\n",
      "... done in 0.45 seconds.\n",
      "\n",
      "Fitting MAP dispersions...\n",
      "... done in 3.40 seconds.\n",
      "\n",
      "Fitting LFCs...\n",
      "... done in 2.63 seconds.\n",
      "\n",
      "Calculating cook's distance...\n",
      "... done in 0.01 seconds.\n",
      "\n",
      "Replacing 0 outlier genes.\n",
      "\n",
      "Running Wald tests...\n",
      "... done in 0.95 seconds.\n",
      "\n",
      "Fitting size factors...\n",
      "... done in 0.00 seconds.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log2 fold change & Wald test p-value: random_split perturbed vs normal\n",
      "             baseMean  log2FoldChange     lfcSE      stat    pvalue      padj\n",
      "AT1G01010    3.586636       -0.970208  1.857781 -0.522240  0.601503  0.999861\n",
      "AT1G01020   42.399036       -0.029945  0.628945 -0.047612  0.962025  0.999861\n",
      "AT1G01030   22.503768        0.027450  0.834170  0.032906  0.973749  0.999861\n",
      "AT1G01040   85.020311        0.282795  0.566517  0.499181  0.617652  0.999861\n",
      "AT1G01050  212.001365        0.018379  0.383890  0.047875  0.961816  0.999861\n",
      "...               ...             ...       ...       ...       ...       ...\n",
      "ArthCr089  346.683127        0.427677  0.367950  1.162325  0.245103  0.999861\n",
      "ArthCr088   23.799195        0.834850  0.884558  0.943805  0.345269  0.999861\n",
      "ArthCp088    8.848439       -0.103240  1.262146 -0.081797  0.934808  0.999861\n",
      "ArthCp083    8.653670       -1.384656  1.506222 -0.919291  0.357943  0.999861\n",
      "ArthCp085    7.149250       -1.801146  1.447837 -1.244025  0.213490  0.999861\n",
      "\n",
      "[17681 rows x 6 columns]\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: t08 (random split)\n",
      "==================================================\n",
      "Group 1: 4 samples, Group 2: 4 samples\n",
      "Number of normal samples: 4\n",
      "Number of perturbed samples: 4\n",
      "Using None as control genes, passed at DeseqDataSet initialization\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Fitting dispersions...\n",
      "... done in 3.34 seconds.\n",
      "\n",
      "Fitting dispersion trend curve...\n",
      "... done in 0.47 seconds.\n",
      "\n",
      "Fitting MAP dispersions...\n",
      "... done in 3.48 seconds.\n",
      "\n",
      "Fitting LFCs...\n",
      "... done in 2.61 seconds.\n",
      "\n",
      "Calculating cook's distance...\n",
      "... done in 0.01 seconds.\n",
      "\n",
      "Replacing 0 outlier genes.\n",
      "\n",
      "Running Wald tests...\n",
      "... done in 0.97 seconds.\n",
      "\n",
      "Fitting size factors...\n",
      "... done in 0.00 seconds.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log2 fold change & Wald test p-value: random_split perturbed vs normal\n",
      "              baseMean  log2FoldChange     lfcSE      stat    pvalue      padj\n",
      "AT1G01010     6.340787        0.475326  1.414211  0.336107  0.736790       NaN\n",
      "AT1G01020    51.564805        0.455820  0.507279  0.898560  0.368887  0.916675\n",
      "AT1G01030    20.237407       -0.215282  0.792901 -0.271512  0.785997  0.983097\n",
      "AT1G01040   126.621703        0.380279  0.339883  1.118854  0.263202  0.883907\n",
      "AT1G01050   359.624581        0.108254  0.301170  0.359444  0.719263  0.974953\n",
      "...                ...             ...       ...       ...       ...       ...\n",
      "ArthCr089  1811.955207        0.834208  0.583199  1.430400  0.152602  0.828120\n",
      "ArthCr088    78.999450        0.191770  0.453543  0.422826  0.672422  0.969520\n",
      "ArthCp088    16.299308        0.958436  0.774088  1.238148  0.215661  0.857263\n",
      "ArthCp083    24.429109        0.877558  0.697165  1.258753  0.208120  0.855089\n",
      "ArthCp085    20.051421        1.198682  0.771684  1.553334  0.120343  0.808173\n",
      "\n",
      "[17681 rows x 6 columns]\n",
      "\n",
      "==================================================\n",
      "Processing timepoint: c1_t01 (random split)\n",
      "==================================================\n",
      "Group 1: 4 samples, Group 2: 4 samples\n",
      "Number of normal samples: 4\n",
      "Number of perturbed samples: 4\n",
      "Using None as control genes, passed at DeseqDataSet initialization\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Fitting dispersions...\n",
      "... done in 3.34 seconds.\n",
      "\n",
      "Fitting dispersion trend curve...\n",
      "... done in 0.43 seconds.\n",
      "\n",
      "Fitting MAP dispersions...\n",
      "... done in 3.52 seconds.\n",
      "\n",
      "Fitting LFCs...\n",
      "... done in 2.63 seconds.\n",
      "\n",
      "Calculating cook's distance...\n",
      "... done in 0.01 seconds.\n",
      "\n",
      "Replacing 0 outlier genes.\n",
      "\n",
      "Running Wald tests...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log2 fold change & Wald test p-value: random_split perturbed vs normal\n",
      "              baseMean  log2FoldChange     lfcSE      stat    pvalue      padj\n",
      "AT1G01010     4.831789        0.953009  1.467269  0.649512  0.516007  0.999784\n",
      "AT1G01020    37.396573        0.514588  0.744268  0.691402  0.489313  0.999784\n",
      "AT1G01030     5.417486        0.218336  1.470446  0.148483  0.881961  0.999784\n",
      "AT1G01040   103.969284       -0.288537  0.375021 -0.769391  0.441661  0.999784\n",
      "AT1G01050   224.359910        0.454001  0.337137  1.346639  0.178096  0.999784\n",
      "...                ...             ...       ...       ...       ...       ...\n",
      "ArthCr089  1026.867062       -0.765731  0.723567 -1.058273  0.289931  0.999784\n",
      "ArthCr088    42.833990       -0.263326  0.627309 -0.419772  0.674652  0.999784\n",
      "ArthCp088     7.695965        0.217887  1.417004  0.153766  0.877794  0.999784\n",
      "ArthCp083    19.478713       -0.550780  0.836779 -0.658214  0.510401  0.999784\n",
      "ArthCp085    11.272994        1.374931  1.220516  1.126516  0.259947  0.999784\n",
      "\n",
      "[17681 rows x 6 columns]\n",
      "\n",
      "Completed! Total results: 19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "... done in 0.98 seconds.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "import random\n",
    "\n",
    "list_of_random_differential_expression_results = []\n",
    "\n",
    "# Get only timepoints that had sufficient samples in the previous analysis\n",
    "timepoints = [result[\"timepoint\"] for result in list_of_differential_expression_results]\n",
    "\n",
    "# Set seed for reproducibility\n",
    "random.seed(11106)\n",
    "\n",
    "for tp in timepoints:\n",
    "    print(f\"\\n{'='*50}\")\n",
    "    print(f\"Processing timepoint: {tp} (random split)\")\n",
    "    print(f\"{'='*50}\")\n",
    "\n",
    "    # Get samples for this timepoint\n",
    "    tp_metadata = trimmed_metadata[trimmed_metadata[\"timepoint\"] == tp]\n",
    "\n",
    "    # Get all samples present in cleaned_raw_transcriptome\n",
    "    all_samples = [\n",
    "        s for s in tp_metadata.index.tolist() if s in cleaned_raw_transcriptome.index\n",
    "    ]\n",
    "\n",
    "    # Skip if too few samples\n",
    "    if len(all_samples) < 4:\n",
    "        print(f\"Skipping timepoint {tp}: insufficient samples ({len(all_samples)})\")\n",
    "        continue\n",
    "\n",
    "    # Randomly shuffle and split into two groups\n",
    "    random.shuffle(all_samples)\n",
    "    midpoint = len(all_samples) // 2\n",
    "    group1_samples = all_samples[:midpoint]\n",
    "    group2_samples = all_samples[midpoint:]\n",
    "\n",
    "    print(\n",
    "        f\"Group 1: {len(group1_samples)} samples, Group 2: {len(group2_samples)} samples\"\n",
    "    )\n",
    "\n",
    "    try:\n",
    "        ds_result = calculate_differential_expression_btwn_2_groups(\n",
    "            list_of_normal_samples=group1_samples,\n",
    "            list_of_perturbed_samples=group2_samples,\n",
    "            raw_transcriptome_data=cleaned_raw_transcriptome,\n",
    "            og_metadata=trimmed_metadata,\n",
    "            name_of_perturbation_column=\"random_split\",\n",
    "            use_plate_for_design=False,\n",
    "        )\n",
    "        list_of_random_differential_expression_results.append(\n",
    "            {\"timepoint\": tp, \"result\": ds_result}\n",
    "        )\n",
    "    except Exception as e:\n",
    "        print(f\"Error processing timepoint {tp}: {e}\")\n",
    "\n",
    "print(\n",
    "    f\"\\nCompleted! Total results: {len(list_of_random_differential_expression_results)}\"\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "a0a5d4dc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>padj</th>\n",
       "      <th>timepoint</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AT1G01010</th>\n",
       "      <td>0.999581</td>\n",
       "      <td>t14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT1G01020</th>\n",
       "      <td>0.999581</td>\n",
       "      <td>t14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT1G01030</th>\n",
       "      <td>0.999581</td>\n",
       "      <td>t14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT1G01040</th>\n",
       "      <td>0.999581</td>\n",
       "      <td>t14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT1G01050</th>\n",
       "      <td>0.999581</td>\n",
       "      <td>t14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ArthCr089</th>\n",
       "      <td>0.999784</td>\n",
       "      <td>c1_t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ArthCr088</th>\n",
       "      <td>0.999784</td>\n",
       "      <td>c1_t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ArthCp088</th>\n",
       "      <td>0.999784</td>\n",
       "      <td>c1_t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ArthCp083</th>\n",
       "      <td>0.999784</td>\n",
       "      <td>c1_t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ArthCp085</th>\n",
       "      <td>0.999784</td>\n",
       "      <td>c1_t01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>335939 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               padj timepoint\n",
       "AT1G01010  0.999581       t14\n",
       "AT1G01020  0.999581       t14\n",
       "AT1G01030  0.999581       t14\n",
       "AT1G01040  0.999581       t14\n",
       "AT1G01050  0.999581       t14\n",
       "...             ...       ...\n",
       "ArthCr089  0.999784    c1_t01\n",
       "ArthCr088  0.999784    c1_t01\n",
       "ArthCp088  0.999784    c1_t01\n",
       "ArthCp083  0.999784    c1_t01\n",
       "ArthCp085  0.999784    c1_t01\n",
       "\n",
       "[335939 rows x 2 columns]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Extract padj values from rare species differential expression results\n",
    "list_of_rare_species_padj = []\n",
    "\n",
    "for result_dict in list_of_differential_expression_results:\n",
    "    tp = result_dict[\"timepoint\"]\n",
    "    ds = result_dict[\"result\"]\n",
    "\n",
    "    # Get the results dataframe\n",
    "    results_df = ds.results_df\n",
    "\n",
    "    # Add timepoint info and append padj values\n",
    "    padj_df = results_df[[\"padj\"]].copy()\n",
    "    padj_df[\"timepoint\"] = tp\n",
    "    list_of_rare_species_padj.append(padj_df)\n",
    "\n",
    "# Combine into single dataframe\n",
    "rare_species_padj_df = pd.concat(list_of_rare_species_padj)\n",
    "rare_species_padj_df\n",
    "\n",
    "\n",
    "list_of_random_padj = []\n",
    "\n",
    "for result_dict in list_of_random_differential_expression_results:\n",
    "    tp = result_dict[\"timepoint\"]\n",
    "    ds = result_dict[\"result\"]\n",
    "\n",
    "    # Get the results dataframe\n",
    "    results_df = ds.results_df\n",
    "\n",
    "    # Add timepoint info and append padj values\n",
    "    padj_df = results_df[[\"padj\"]].copy()\n",
    "    padj_df[\"timepoint\"] = tp\n",
    "    list_of_random_padj.append(padj_df)\n",
    "\n",
    "# Combine into single dataframe\n",
    "random_padj_df = pd.concat(list_of_random_padj)\n",
    "random_padj_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "9360da38",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>padj</th>\n",
       "      <th>timepoint</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AT1G01470</th>\n",
       "      <td>0.039277</td>\n",
       "      <td>t14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT1G09350</th>\n",
       "      <td>0.021703</td>\n",
       "      <td>t14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT1G11580</th>\n",
       "      <td>0.001244</td>\n",
       "      <td>t14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT1G14880</th>\n",
       "      <td>0.000880</td>\n",
       "      <td>t14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT1G05273</th>\n",
       "      <td>0.021703</td>\n",
       "      <td>t14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT4G25433</th>\n",
       "      <td>0.005863</td>\n",
       "      <td>t08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT5G55450</th>\n",
       "      <td>0.046397</td>\n",
       "      <td>t08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT3G49580</th>\n",
       "      <td>0.008294</td>\n",
       "      <td>c1_t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT5G18290</th>\n",
       "      <td>0.048262</td>\n",
       "      <td>c1_t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT5G24780</th>\n",
       "      <td>0.008294</td>\n",
       "      <td>c1_t01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>625 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               padj timepoint\n",
       "AT1G01470  0.039277       t14\n",
       "AT1G09350  0.021703       t14\n",
       "AT1G11580  0.001244       t14\n",
       "AT1G14880  0.000880       t14\n",
       "AT1G05273  0.021703       t14\n",
       "...             ...       ...\n",
       "AT4G25433  0.005863       t08\n",
       "AT5G55450  0.046397       t08\n",
       "AT3G49580  0.008294    c1_t01\n",
       "AT5G18290  0.048262    c1_t01\n",
       "AT5G24780  0.008294    c1_t01\n",
       "\n",
       "[625 rows x 2 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rare_species_padj_df.loc[rare_species_padj_df[\"padj\"] < 0.05]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "dfc708e4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>padj</th>\n",
       "      <th>timepoint</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AT1G12280</th>\n",
       "      <td>4.616799e-02</td>\n",
       "      <td>t14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT5G54610</th>\n",
       "      <td>3.906845e-05</td>\n",
       "      <td>t14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT1G19380</th>\n",
       "      <td>1.292459e-02</td>\n",
       "      <td>t21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT1G27470</th>\n",
       "      <td>4.884885e-02</td>\n",
       "      <td>t21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT1G49870</th>\n",
       "      <td>2.377736e-02</td>\n",
       "      <td>t21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT1G69490</th>\n",
       "      <td>2.441194e-02</td>\n",
       "      <td>t21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT2G15020</th>\n",
       "      <td>2.377736e-02</td>\n",
       "      <td>t21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT2G38465</th>\n",
       "      <td>4.884885e-02</td>\n",
       "      <td>t21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT3G22600</th>\n",
       "      <td>3.458275e-02</td>\n",
       "      <td>t21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT3G45970</th>\n",
       "      <td>3.458275e-02</td>\n",
       "      <td>t21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT3G50480</th>\n",
       "      <td>1.236369e-04</td>\n",
       "      <td>t21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT4G18422</th>\n",
       "      <td>4.884885e-02</td>\n",
       "      <td>t21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT4G22530</th>\n",
       "      <td>4.884885e-02</td>\n",
       "      <td>t21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT4G37410</th>\n",
       "      <td>4.884885e-02</td>\n",
       "      <td>t21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT5G03210</th>\n",
       "      <td>2.377736e-02</td>\n",
       "      <td>t21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT5G10550</th>\n",
       "      <td>4.884885e-02</td>\n",
       "      <td>t21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT1G14880</th>\n",
       "      <td>3.872983e-03</td>\n",
       "      <td>t02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT1G19680</th>\n",
       "      <td>3.185560e-02</td>\n",
       "      <td>t02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT1G49870</th>\n",
       "      <td>3.126927e-02</td>\n",
       "      <td>t02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT2G14610</th>\n",
       "      <td>2.594245e-02</td>\n",
       "      <td>t02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT3G26830</th>\n",
       "      <td>3.872983e-03</td>\n",
       "      <td>t02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT3G47295</th>\n",
       "      <td>4.941419e-02</td>\n",
       "      <td>t02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT5G27420</th>\n",
       "      <td>3.185560e-02</td>\n",
       "      <td>t02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT2G05440</th>\n",
       "      <td>7.653120e-03</td>\n",
       "      <td>t03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT1G07430</th>\n",
       "      <td>3.430289e-02</td>\n",
       "      <td>t12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT1G52400</th>\n",
       "      <td>2.981564e-03</td>\n",
       "      <td>t12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT1G54010</th>\n",
       "      <td>3.655764e-02</td>\n",
       "      <td>t12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT2G26010</th>\n",
       "      <td>1.465273e-02</td>\n",
       "      <td>t12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT2G26020</th>\n",
       "      <td>4.253278e-02</td>\n",
       "      <td>t12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT2G43150</th>\n",
       "      <td>3.584629e-02</td>\n",
       "      <td>t12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT3G23840</th>\n",
       "      <td>1.465273e-02</td>\n",
       "      <td>t12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT3G26200</th>\n",
       "      <td>2.728771e-03</td>\n",
       "      <td>t12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT3G28220</th>\n",
       "      <td>2.981564e-03</td>\n",
       "      <td>t12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT3G44300</th>\n",
       "      <td>3.430289e-02</td>\n",
       "      <td>t12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT3G45140</th>\n",
       "      <td>3.584629e-02</td>\n",
       "      <td>t12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT4G02520</th>\n",
       "      <td>3.430289e-02</td>\n",
       "      <td>t12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT4G08870</th>\n",
       "      <td>3.430289e-02</td>\n",
       "      <td>t12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT4G23600</th>\n",
       "      <td>3.430289e-02</td>\n",
       "      <td>t12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT5G44420</th>\n",
       "      <td>1.903211e-05</td>\n",
       "      <td>t12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT5G44430</th>\n",
       "      <td>1.052318e-02</td>\n",
       "      <td>t12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT3G05727</th>\n",
       "      <td>3.822226e-02</td>\n",
       "      <td>t13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT4G02520</th>\n",
       "      <td>3.822226e-02</td>\n",
       "      <td>t13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT4G22485</th>\n",
       "      <td>6.796984e-07</td>\n",
       "      <td>t13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT4G22505</th>\n",
       "      <td>1.018808e-05</td>\n",
       "      <td>t13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT4G22513</th>\n",
       "      <td>6.317742e-05</td>\n",
       "      <td>t13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT4G22517</th>\n",
       "      <td>1.170733e-02</td>\n",
       "      <td>t13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT5G01870</th>\n",
       "      <td>1.528776e-02</td>\n",
       "      <td>t13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT1G15550</th>\n",
       "      <td>4.903525e-02</td>\n",
       "      <td>t08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT1G21120</th>\n",
       "      <td>8.620748e-03</td>\n",
       "      <td>t08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT1G21130</th>\n",
       "      <td>3.729306e-02</td>\n",
       "      <td>t08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT1G27730</th>\n",
       "      <td>2.213441e-02</td>\n",
       "      <td>t08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT1G09755</th>\n",
       "      <td>3.729306e-02</td>\n",
       "      <td>t08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT1G70570</th>\n",
       "      <td>2.213441e-02</td>\n",
       "      <td>t08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT2G23810</th>\n",
       "      <td>4.903525e-02</td>\n",
       "      <td>t08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT3G16220</th>\n",
       "      <td>4.903525e-02</td>\n",
       "      <td>t08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT3G49940</th>\n",
       "      <td>8.620748e-03</td>\n",
       "      <td>t08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT5G05640</th>\n",
       "      <td>3.436422e-02</td>\n",
       "      <td>t08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT5G44570</th>\n",
       "      <td>4.903525e-02</td>\n",
       "      <td>t08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT3G49580</th>\n",
       "      <td>2.259543e-06</td>\n",
       "      <td>c1_t01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   padj timepoint\n",
       "AT1G12280  4.616799e-02       t14\n",
       "AT5G54610  3.906845e-05       t14\n",
       "AT1G19380  1.292459e-02       t21\n",
       "AT1G27470  4.884885e-02       t21\n",
       "AT1G49870  2.377736e-02       t21\n",
       "AT1G69490  2.441194e-02       t21\n",
       "AT2G15020  2.377736e-02       t21\n",
       "AT2G38465  4.884885e-02       t21\n",
       "AT3G22600  3.458275e-02       t21\n",
       "AT3G45970  3.458275e-02       t21\n",
       "AT3G50480  1.236369e-04       t21\n",
       "AT4G18422  4.884885e-02       t21\n",
       "AT4G22530  4.884885e-02       t21\n",
       "AT4G37410  4.884885e-02       t21\n",
       "AT5G03210  2.377736e-02       t21\n",
       "AT5G10550  4.884885e-02       t21\n",
       "AT1G14880  3.872983e-03       t02\n",
       "AT1G19680  3.185560e-02       t02\n",
       "AT1G49870  3.126927e-02       t02\n",
       "AT2G14610  2.594245e-02       t02\n",
       "AT3G26830  3.872983e-03       t02\n",
       "AT3G47295  4.941419e-02       t02\n",
       "AT5G27420  3.185560e-02       t02\n",
       "AT2G05440  7.653120e-03       t03\n",
       "AT1G07430  3.430289e-02       t12\n",
       "AT1G52400  2.981564e-03       t12\n",
       "AT1G54010  3.655764e-02       t12\n",
       "AT2G26010  1.465273e-02       t12\n",
       "AT2G26020  4.253278e-02       t12\n",
       "AT2G43150  3.584629e-02       t12\n",
       "AT3G23840  1.465273e-02       t12\n",
       "AT3G26200  2.728771e-03       t12\n",
       "AT3G28220  2.981564e-03       t12\n",
       "AT3G44300  3.430289e-02       t12\n",
       "AT3G45140  3.584629e-02       t12\n",
       "AT4G02520  3.430289e-02       t12\n",
       "AT4G08870  3.430289e-02       t12\n",
       "AT4G23600  3.430289e-02       t12\n",
       "AT5G44420  1.903211e-05       t12\n",
       "AT5G44430  1.052318e-02       t12\n",
       "AT3G05727  3.822226e-02       t13\n",
       "AT4G02520  3.822226e-02       t13\n",
       "AT4G22485  6.796984e-07       t13\n",
       "AT4G22505  1.018808e-05       t13\n",
       "AT4G22513  6.317742e-05       t13\n",
       "AT4G22517  1.170733e-02       t13\n",
       "AT5G01870  1.528776e-02       t13\n",
       "AT1G15550  4.903525e-02       t08\n",
       "AT1G21120  8.620748e-03       t08\n",
       "AT1G21130  3.729306e-02       t08\n",
       "AT1G27730  2.213441e-02       t08\n",
       "AT1G09755  3.729306e-02       t08\n",
       "AT1G70570  2.213441e-02       t08\n",
       "AT2G23810  4.903525e-02       t08\n",
       "AT3G16220  4.903525e-02       t08\n",
       "AT3G49940  8.620748e-03       t08\n",
       "AT5G05640  3.436422e-02       t08\n",
       "AT5G44570  4.903525e-02       t08\n",
       "AT3G49580  2.259543e-06    c1_t01"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "random_padj_df.loc[random_padj_df[\"padj\"] < 0.05]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "b2300471",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(\n",
    "    random_padj_df.loc[\n",
    "        (random_padj_df[\"padj\"] < 0.05) & (random_padj_df[\"timepoint\"] == \"t15\")\n",
    "    ]\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "b518b115",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['t14', 't15', 't17', 't19', 't20', 't21', 't22', 't23', 't25',\n",
       "       't26', 't27', 't18', 't01', 't02', 't03', 't04', 't05', 't07',\n",
       "       't10', 't11', 't12', 't13', 't06', 't09', 'c1_t08', 'c1_t09',\n",
       "       'c1_t10', 'c1_t12', 'c2_t02', 'c2_t03', 'c2_t04', 'c2_t05',\n",
       "       'c2_t06', 'c1_t11', 'c1_t13', 'c2_t01', 't28', 't24', 't16', 't08',\n",
       "       'c1_t01', 'c1_t02', 'c1_t03', 'c1_t04', 'c1_t05', 'c1_t06',\n",
       "       'c1_t07'], dtype=object)"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "trimmed_metadata[\"timepoint\"].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "561ed797",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2\n",
      "t14\n",
      "0\n",
      "t15\n",
      "14\n",
      "t21\n",
      "0\n",
      "t27\n",
      "0\n",
      "t01\n",
      "7\n",
      "t02\n",
      "1\n",
      "t03\n",
      "0\n",
      "t05\n",
      "0\n",
      "t07\n",
      "16\n",
      "t12\n",
      "7\n",
      "t13\n",
      "0\n",
      "t06\n",
      "0\n",
      "t09\n",
      "0\n",
      "c1_t08\n",
      "0\n",
      "c1_t09\n",
      "0\n",
      "c1_t13\n",
      "0\n",
      "c2_t01\n",
      "11\n",
      "t08\n",
      "1\n",
      "c1_t01\n"
     ]
    }
   ],
   "source": [
    "list_of_diff = []\n",
    "list_of_tp = []\n",
    "for timepoint in trimmed_metadata[\"timepoint\"].unique():\n",
    "    if timepoint in rare_species_padj_df[\"timepoint\"].unique():\n",
    "        random_len = len(\n",
    "            random_padj_df.loc[\n",
    "                (random_padj_df[\"padj\"] < 0.05)\n",
    "                & (random_padj_df[\"timepoint\"] == timepoint)\n",
    "            ]\n",
    "        )\n",
    "        print(random_len)\n",
    "        print(timepoint)\n",
    "        rare_species = len(\n",
    "            rare_species_padj_df.loc[\n",
    "                (rare_species_padj_df[\"padj\"] < 0.05)\n",
    "                & (rare_species_padj_df[\"timepoint\"] == timepoint)\n",
    "            ]\n",
    "        )\n",
    "        diff_in_len = rare_species - random_len\n",
    "        list_of_diff.append(diff_in_len)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f565dda8",
   "metadata": {},
   "source": [
    "# Jump to bottom"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "ae693834",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "566"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sum(list_of_diff)  ## current 492\n",
    "# 6.5 = 539\n",
    "# 5 = 493"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "17848076",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[102, 1, -12, 4, 6, 3, 0, 1, 2, 18, 9, 0, 30, 78, 8, 315, 6, -7, 2]"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list_of_diff"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1501c637",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Luke_terrace",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
