{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "8912758a",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "78e03a8a",
   "metadata": {},
   "outputs": [],
   "source": [
    "expressed_gene_tpm_threshold = 1\n",
    "minimum_samples_threshold_long = 1\n",
    "minimum_samples_threshold_circadian = 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "a7e4dd14",
   "metadata": {},
   "outputs": [],
   "source": [
    "transcriptome = pd.read_csv(\n",
    "    \"/Users/michael/Data/Luke_terrace_experiment/General_data/plate1_4_tpm_normalized.csv\",\n",
    "    index_col=0,\n",
    ")\n",
    "rows_to_drop_expression_data = [\n",
    "    \"A2450525897_n01_undetermined\",\n",
    "    \"A2449446903_n01_undetermined\",\n",
    "    \"B250508004_n01_undetermined\",\n",
    "    \"B2449500127_n01_undetermined\",\n",
    "]\n",
    "transcriptome = transcriptome.drop(index=rows_to_drop_expression_data)\n",
    "transcriptome = transcriptome.sort_index()\n",
    "\n",
    "metadata = pd.read_csv(\n",
    "    \"/Users/michael/Data/Luke_terrace_experiment/General_data/Plates_1_to_4_metadata_merged_luke.csv\",\n",
    "    index_col=0,\n",
    ")\n",
    "metadata = metadata.drop(\n",
    "    columns=[\n",
    "        \"arb.sort\",\n",
    "        \"sample-id\",\n",
    "        \"Ambiguous Unstranded\",\n",
    "        \"Ambiguous Forward\",\n",
    "        \"Multimapping\",\n",
    "        \"Unmapped Over Mapped\",\n",
    "    ]\n",
    ")\n",
    "metadata[\"Date and Time\"] = metadata[\"date\"] + \" \" + metadata[\"time\"]\n",
    "luke_time_data_format = \"%-m/%-d/%y %-H:%-M\"\n",
    "metadata[\"Date and Time\"] = pd.to_datetime(\n",
    "    metadata[\"Date and Time\"], format=luke_time_data_format\n",
    ")\n",
    "\n",
    "unusable_samples = [\n",
    "    \"A2450525897_n01_LICRNA01_A06\",\n",
    "    \"A2450525897_n01_LICRNA01_A08\",\n",
    "    \"A2450525897_n01_LICRNA01_C11\",\n",
    "    \"A2450525897_n01_LICRNA01_D11\",\n",
    "    \"A2450525897_n01_LICRNA01_D12\",\n",
    "    \"A2449446903_n01_LICRNA02_F01\",\n",
    "    \"B2449500127_n01_LICRNA04_A04\",\n",
    "    \"B2449500127_n01_LICRNA04_A06\",\n",
    "    \"B2449500127_n01_LICRNA04_A07\",\n",
    "]\n",
    "borderline_unusable = [\n",
    "    \"A2450525897_n01_LICRNA01_F03\",\n",
    "    \"A2450525897_n01_LICRNA01_G07\",\n",
    "    \"A2449446903_n01_LICRNA02_A04\",\n",
    "    \"A2449446903_n01_LICRNA02_H10\",\n",
    "    \"B2449500127_n01_LICRNA04_H02\",\n",
    "    \"B2449500127_n01_LICRNA04_H11\",\n",
    "]\n",
    "all_unsable = borderline_unusable + unusable_samples\n",
    "\n",
    "trimmed_transcriptome = transcriptome.drop(index=all_unsable)\n",
    "trimmed_metadata = metadata.drop(index=all_unsable)\n",
    "trimmed_metadata = trimmed_metadata.sort_index()\n",
    "trimmed_transcriptome = trimmed_transcriptome.dropna(how=\"all\", axis=0)\n",
    "trimmed_transcriptome = trimmed_transcriptome.dropna(how=\"all\", axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "b1cfdbcb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AT1G01010</th>\n",
       "      <th>AT1G01020</th>\n",
       "      <th>AT1G03987</th>\n",
       "      <th>AT1G01030</th>\n",
       "      <th>AT1G01040</th>\n",
       "      <th>AT1G01046</th>\n",
       "      <th>AT1G01050</th>\n",
       "      <th>AT1G01060</th>\n",
       "      <th>AT1G01070</th>\n",
       "      <th>AT1G04003</th>\n",
       "      <th>...</th>\n",
       "      <th>ArthCr090</th>\n",
       "      <th>ArthCr089</th>\n",
       "      <th>ArthCt114</th>\n",
       "      <th>ArthCt124</th>\n",
       "      <th>ArthCr088</th>\n",
       "      <th>ArthCt112</th>\n",
       "      <th>ArthCp088</th>\n",
       "      <th>ArthCp083</th>\n",
       "      <th>ArthCp084</th>\n",
       "      <th>ArthCp085</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A01</th>\n",
       "      <td>0.382839</td>\n",
       "      <td>11.929183</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.731527</td>\n",
       "      <td>3.602189</td>\n",
       "      <td>0.0</td>\n",
       "      <td>31.063543</td>\n",
       "      <td>3.966935</td>\n",
       "      <td>1.411759</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>62.741033</td>\n",
       "      <td>199.389218</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>17.950907</td>\n",
       "      <td>44.209074</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>6.266498</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A02</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>5.826682</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.501594</td>\n",
       "      <td>3.439569</td>\n",
       "      <td>0.0</td>\n",
       "      <td>22.712417</td>\n",
       "      <td>18.186693</td>\n",
       "      <td>1.249825</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>4.165830</td>\n",
       "      <td>69.172057</td>\n",
       "      <td>19.592718</td>\n",
       "      <td>1.986484</td>\n",
       "      <td>9.688606</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.501032</td>\n",
       "      <td>0.519342</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.467317</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A03</th>\n",
       "      <td>1.021565</td>\n",
       "      <td>7.683520</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.301733</td>\n",
       "      <td>2.700528</td>\n",
       "      <td>0.0</td>\n",
       "      <td>42.431625</td>\n",
       "      <td>12.937632</td>\n",
       "      <td>1.412672</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>50.225286</td>\n",
       "      <td>122.221814</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>7.983340</td>\n",
       "      <td>16.384320</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.983340</td>\n",
       "      <td>0.166972</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A05</th>\n",
       "      <td>0.732717</td>\n",
       "      <td>3.149144</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.947758</td>\n",
       "      <td>3.742586</td>\n",
       "      <td>0.0</td>\n",
       "      <td>26.541337</td>\n",
       "      <td>23.861601</td>\n",
       "      <td>6.417176</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>6.004011</td>\n",
       "      <td>73.945487</td>\n",
       "      <td>42.357065</td>\n",
       "      <td>17.178143</td>\n",
       "      <td>17.005325</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.628742</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A06</th>\n",
       "      <td>1.614233</td>\n",
       "      <td>1.238895</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.130092</td>\n",
       "      <td>4.215597</td>\n",
       "      <td>0.0</td>\n",
       "      <td>19.379509</td>\n",
       "      <td>5.006570</td>\n",
       "      <td>1.275568</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>41.571544</td>\n",
       "      <td>302.404404</td>\n",
       "      <td>85.317440</td>\n",
       "      <td>16.219201</td>\n",
       "      <td>43.599301</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.074213</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H08</th>\n",
       "      <td>0.074367</td>\n",
       "      <td>4.474710</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.383813</td>\n",
       "      <td>2.059206</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27.907884</td>\n",
       "      <td>11.118408</td>\n",
       "      <td>0.959827</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>19.500054</td>\n",
       "      <td>33.460202</td>\n",
       "      <td>8.598055</td>\n",
       "      <td>3.486989</td>\n",
       "      <td>4.209645</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.072920</td>\n",
       "      <td>0.948096</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.760798</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H09</th>\n",
       "      <td>2.232353</td>\n",
       "      <td>6.249002</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.873955</td>\n",
       "      <td>3.869254</td>\n",
       "      <td>0.0</td>\n",
       "      <td>40.686812</td>\n",
       "      <td>7.391606</td>\n",
       "      <td>2.003849</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.925504</td>\n",
       "      <td>58.827578</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>9.640892</td>\n",
       "      <td>4.655562</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.085306</td>\n",
       "      <td>0.360071</td>\n",
       "      <td>1.758217</td>\n",
       "      <td>4.807925</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H10</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.953706</td>\n",
       "      <td>0.0</td>\n",
       "      <td>36.084398</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>133.762303</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.975538</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H11</th>\n",
       "      <td>0.678594</td>\n",
       "      <td>7.655887</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.901942</td>\n",
       "      <td>4.195848</td>\n",
       "      <td>0.0</td>\n",
       "      <td>54.077818</td>\n",
       "      <td>0.334834</td>\n",
       "      <td>4.691971</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>26.292737</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.304762</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.166371</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.694222</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H12</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.436049</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.004263</td>\n",
       "      <td>3.900086</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28.479105</td>\n",
       "      <td>1.731854</td>\n",
       "      <td>1.150491</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>15.338954</td>\n",
       "      <td>45.682515</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.828602</td>\n",
       "      <td>5.298163</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.657204</td>\n",
       "      <td>0.248594</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.553393</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>369 rows × 27418 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              AT1G01010  AT1G01020  AT1G03987  AT1G01030  \\\n",
       "A2449446903_n01_LICRNA02_A01   0.382839  11.929183        0.0   3.731527   \n",
       "A2449446903_n01_LICRNA02_A02   0.000000   5.826682        0.0   1.501594   \n",
       "A2449446903_n01_LICRNA02_A03   1.021565   7.683520        0.0   0.301733   \n",
       "A2449446903_n01_LICRNA02_A05   0.732717   3.149144        0.0   1.947758   \n",
       "A2449446903_n01_LICRNA02_A06   1.614233   1.238895        0.0   6.130092   \n",
       "...                                 ...        ...        ...        ...   \n",
       "B250508004_n01_LICRNA03_H08    0.074367   4.474710        0.0   1.383813   \n",
       "B250508004_n01_LICRNA03_H09    2.232353   6.249002        0.0   1.873955   \n",
       "B250508004_n01_LICRNA03_H10    0.000000   0.000000        0.0   0.000000   \n",
       "B250508004_n01_LICRNA03_H11    0.678594   7.655887        0.0   0.901942   \n",
       "B250508004_n01_LICRNA03_H12    0.000000   3.436049        0.0   2.004263   \n",
       "\n",
       "                              AT1G01040  AT1G01046  AT1G01050  AT1G01060  \\\n",
       "A2449446903_n01_LICRNA02_A01   3.602189        0.0  31.063543   3.966935   \n",
       "A2449446903_n01_LICRNA02_A02   3.439569        0.0  22.712417  18.186693   \n",
       "A2449446903_n01_LICRNA02_A03   2.700528        0.0  42.431625  12.937632   \n",
       "A2449446903_n01_LICRNA02_A05   3.742586        0.0  26.541337  23.861601   \n",
       "A2449446903_n01_LICRNA02_A06   4.215597        0.0  19.379509   5.006570   \n",
       "...                                 ...        ...        ...        ...   \n",
       "B250508004_n01_LICRNA03_H08    2.059206        0.0  27.907884  11.118408   \n",
       "B250508004_n01_LICRNA03_H09    3.869254        0.0  40.686812   7.391606   \n",
       "B250508004_n01_LICRNA03_H10    2.953706        0.0  36.084398   0.000000   \n",
       "B250508004_n01_LICRNA03_H11    4.195848        0.0  54.077818   0.334834   \n",
       "B250508004_n01_LICRNA03_H12    3.900086        0.0  28.479105   1.731854   \n",
       "\n",
       "                              AT1G01070  AT1G04003  ...  ArthCr090  \\\n",
       "A2449446903_n01_LICRNA02_A01   1.411759        0.0  ...  62.741033   \n",
       "A2449446903_n01_LICRNA02_A02   1.249825        0.0  ...   4.165830   \n",
       "A2449446903_n01_LICRNA02_A03   1.412672        0.0  ...  50.225286   \n",
       "A2449446903_n01_LICRNA02_A05   6.417176        0.0  ...   6.004011   \n",
       "A2449446903_n01_LICRNA02_A06   1.275568        0.0  ...  41.571544   \n",
       "...                                 ...        ...  ...        ...   \n",
       "B250508004_n01_LICRNA03_H08    0.959827        0.0  ...  19.500054   \n",
       "B250508004_n01_LICRNA03_H09    2.003849        0.0  ...   1.925504   \n",
       "B250508004_n01_LICRNA03_H10    0.000000        0.0  ...   0.000000   \n",
       "B250508004_n01_LICRNA03_H11    4.691971        0.0  ...   0.000000   \n",
       "B250508004_n01_LICRNA03_H12    1.150491        0.0  ...  15.338954   \n",
       "\n",
       "                               ArthCr089  ArthCt114  ArthCt124  ArthCr088  \\\n",
       "A2449446903_n01_LICRNA02_A01  199.389218   0.000000  17.950907  44.209074   \n",
       "A2449446903_n01_LICRNA02_A02   69.172057  19.592718   1.986484   9.688606   \n",
       "A2449446903_n01_LICRNA02_A03  122.221814   0.000000   7.983340  16.384320   \n",
       "A2449446903_n01_LICRNA02_A05   73.945487  42.357065  17.178143  17.005325   \n",
       "A2449446903_n01_LICRNA02_A06  302.404404  85.317440  16.219201  43.599301   \n",
       "...                                  ...        ...        ...        ...   \n",
       "B250508004_n01_LICRNA03_H08    33.460202   8.598055   3.486989   4.209645   \n",
       "B250508004_n01_LICRNA03_H09    58.827578   0.000000   9.640892   4.655562   \n",
       "B250508004_n01_LICRNA03_H10   133.762303   0.000000   0.000000   4.975538   \n",
       "B250508004_n01_LICRNA03_H11    26.292737   0.000000   0.000000   2.304762   \n",
       "B250508004_n01_LICRNA03_H12    45.682515   0.000000   1.828602   5.298163   \n",
       "\n",
       "                              ArthCt112  ArthCp088  ArthCp083  ArthCp084  \\\n",
       "A2449446903_n01_LICRNA02_A01        0.0   0.000000   0.000000   0.000000   \n",
       "A2449446903_n01_LICRNA02_A02        0.0   5.501032   0.519342   0.000000   \n",
       "A2449446903_n01_LICRNA02_A03        0.0   7.983340   0.166972   0.000000   \n",
       "A2449446903_n01_LICRNA02_A05        0.0   0.000000   0.628742   0.000000   \n",
       "A2449446903_n01_LICRNA02_A06        0.0   0.000000   1.074213   0.000000   \n",
       "...                                 ...        ...        ...        ...   \n",
       "B250508004_n01_LICRNA03_H08         0.0   1.072920   0.948096   0.000000   \n",
       "B250508004_n01_LICRNA03_H09         0.0   5.085306   0.360071   1.758217   \n",
       "B250508004_n01_LICRNA03_H10         0.0   0.000000   0.000000   0.000000   \n",
       "B250508004_n01_LICRNA03_H11         0.0   0.000000   0.166371   0.000000   \n",
       "B250508004_n01_LICRNA03_H12         0.0   3.657204   0.248594   0.000000   \n",
       "\n",
       "                              ArthCp085  \n",
       "A2449446903_n01_LICRNA02_A01   6.266498  \n",
       "A2449446903_n01_LICRNA02_A02   3.467317  \n",
       "A2449446903_n01_LICRNA02_A03   0.000000  \n",
       "A2449446903_n01_LICRNA02_A05   0.000000  \n",
       "A2449446903_n01_LICRNA02_A06   0.000000  \n",
       "...                                 ...  \n",
       "B250508004_n01_LICRNA03_H08    0.760798  \n",
       "B250508004_n01_LICRNA03_H09    4.807925  \n",
       "B250508004_n01_LICRNA03_H10    0.000000  \n",
       "B250508004_n01_LICRNA03_H11    0.694222  \n",
       "B250508004_n01_LICRNA03_H12    2.553393  \n",
       "\n",
       "[369 rows x 27418 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "trimmed_transcriptome = trimmed_transcriptome.loc[\n",
    "    :, trimmed_transcriptome.sum() > expressed_gene_tpm_threshold\n",
    "]\n",
    "\n",
    "trimmed_transcriptome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "5390d3af",
   "metadata": {},
   "outputs": [],
   "source": [
    "long_term_transcriptome = trimmed_transcriptome.loc[\n",
    "    trimmed_metadata[\"Experiment Type\"] == \"Long Term\"\n",
    "]\n",
    "long_term_metadata = trimmed_metadata.loc[\n",
    "    trimmed_metadata[\"Experiment Type\"] == \"Long Term\"\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "9532a1bd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>plate.pos</th>\n",
       "      <th>sampID</th>\n",
       "      <th>sample.type</th>\n",
       "      <th>timepoint</th>\n",
       "      <th>date</th>\n",
       "      <th>time</th>\n",
       "      <th>extractplate</th>\n",
       "      <th>rnaprepplate</th>\n",
       "      <th>Unmapped</th>\n",
       "      <th>No Feature</th>\n",
       "      <th>plate_row</th>\n",
       "      <th>plate_col</th>\n",
       "      <th>Total Reads</th>\n",
       "      <th>Experiment Type</th>\n",
       "      <th>Sample Month</th>\n",
       "      <th>daysincestart</th>\n",
       "      <th>Date and Time</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>filename</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A01</th>\n",
       "      <td>A01</td>\n",
       "      <td>LIC157</td>\n",
       "      <td>plant</td>\n",
       "      <td>t14</td>\n",
       "      <td>1/23/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_02</td>\n",
       "      <td>LICRNA_02</td>\n",
       "      <td>758284.0</td>\n",
       "      <td>82176.0</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "      <td>1246970.0</td>\n",
       "      <td>Long Term</td>\n",
       "      <td>January</td>\n",
       "      <td>83</td>\n",
       "      <td>2024-01-23 08:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A02</th>\n",
       "      <td>A02</td>\n",
       "      <td>LIC169</td>\n",
       "      <td>plant</td>\n",
       "      <td>t15</td>\n",
       "      <td>1/27/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_02</td>\n",
       "      <td>LICRNA_02</td>\n",
       "      <td>3071877.0</td>\n",
       "      <td>353549.0</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>4985602.0</td>\n",
       "      <td>Long Term</td>\n",
       "      <td>January</td>\n",
       "      <td>87</td>\n",
       "      <td>2024-01-27 08:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A03</th>\n",
       "      <td>A03</td>\n",
       "      <td>LIC193</td>\n",
       "      <td>plant</td>\n",
       "      <td>t17</td>\n",
       "      <td>2/2/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_03</td>\n",
       "      <td>LICRNA_02</td>\n",
       "      <td>1213638.0</td>\n",
       "      <td>158864.0</td>\n",
       "      <td>A</td>\n",
       "      <td>3</td>\n",
       "      <td>2465662.0</td>\n",
       "      <td>Long Term</td>\n",
       "      <td>February</td>\n",
       "      <td>93</td>\n",
       "      <td>2024-02-02 08:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A05</th>\n",
       "      <td>A05</td>\n",
       "      <td>LIC217</td>\n",
       "      <td>plant</td>\n",
       "      <td>t19</td>\n",
       "      <td>2/9/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_03</td>\n",
       "      <td>LICRNA_02</td>\n",
       "      <td>1303436.0</td>\n",
       "      <td>90929.0</td>\n",
       "      <td>A</td>\n",
       "      <td>5</td>\n",
       "      <td>1515395.0</td>\n",
       "      <td>Long Term</td>\n",
       "      <td>February</td>\n",
       "      <td>100</td>\n",
       "      <td>2024-02-09 08:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A06</th>\n",
       "      <td>A06</td>\n",
       "      <td>LIC229</td>\n",
       "      <td>plant</td>\n",
       "      <td>t20</td>\n",
       "      <td>2/16/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_03</td>\n",
       "      <td>LICRNA_02</td>\n",
       "      <td>1591231.0</td>\n",
       "      <td>141665.0</td>\n",
       "      <td>A</td>\n",
       "      <td>6</td>\n",
       "      <td>1924391.0</td>\n",
       "      <td>Long Term</td>\n",
       "      <td>February</td>\n",
       "      <td>107</td>\n",
       "      <td>2024-02-16 08:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H08</th>\n",
       "      <td>H08</td>\n",
       "      <td>LIC428</td>\n",
       "      <td>plant</td>\n",
       "      <td>c1_t04</td>\n",
       "      <td>2/10/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LICCIRC_01</td>\n",
       "      <td>LICRNA_03</td>\n",
       "      <td>2552177.0</td>\n",
       "      <td>438552.0</td>\n",
       "      <td>H</td>\n",
       "      <td>8</td>\n",
       "      <td>5245242.0</td>\n",
       "      <td>Circadian Experiment 1</td>\n",
       "      <td>February</td>\n",
       "      <td>101</td>\n",
       "      <td>2024-02-10 08:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H09</th>\n",
       "      <td>H09</td>\n",
       "      <td>LIC440</td>\n",
       "      <td>plant</td>\n",
       "      <td>c1_t05</td>\n",
       "      <td>2/10/24</td>\n",
       "      <td>12:00</td>\n",
       "      <td>LICCIRC_01</td>\n",
       "      <td>LICRNA_03</td>\n",
       "      <td>4228242.0</td>\n",
       "      <td>493699.0</td>\n",
       "      <td>H</td>\n",
       "      <td>9</td>\n",
       "      <td>7715482.0</td>\n",
       "      <td>Circadian Experiment 1</td>\n",
       "      <td>February</td>\n",
       "      <td>101</td>\n",
       "      <td>2024-02-10 12:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H10</th>\n",
       "      <td>H10</td>\n",
       "      <td>LIC452</td>\n",
       "      <td>plant</td>\n",
       "      <td>c1_t06</td>\n",
       "      <td>2/10/24</td>\n",
       "      <td>16:00</td>\n",
       "      <td>LICCIRC_01</td>\n",
       "      <td>LICRNA_03</td>\n",
       "      <td>365095.0</td>\n",
       "      <td>39090.0</td>\n",
       "      <td>H</td>\n",
       "      <td>10</td>\n",
       "      <td>643786.0</td>\n",
       "      <td>Circadian Experiment 1</td>\n",
       "      <td>February</td>\n",
       "      <td>101</td>\n",
       "      <td>2024-02-10 16:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H11</th>\n",
       "      <td>H11</td>\n",
       "      <td>LIC464</td>\n",
       "      <td>plant</td>\n",
       "      <td>c1_t07</td>\n",
       "      <td>2/10/24</td>\n",
       "      <td>20:00</td>\n",
       "      <td>LICCIRC_01</td>\n",
       "      <td>LICRNA_03</td>\n",
       "      <td>326959.0</td>\n",
       "      <td>60346.0</td>\n",
       "      <td>H</td>\n",
       "      <td>11</td>\n",
       "      <td>1509679.0</td>\n",
       "      <td>Circadian Experiment 1</td>\n",
       "      <td>February</td>\n",
       "      <td>101</td>\n",
       "      <td>2024-02-10 20:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H12</th>\n",
       "      <td>H12</td>\n",
       "      <td>LIC488</td>\n",
       "      <td>plant</td>\n",
       "      <td>c1_t08</td>\n",
       "      <td>2/11/24</td>\n",
       "      <td>0:00</td>\n",
       "      <td>LICCIRC_02</td>\n",
       "      <td>LICRNA_03</td>\n",
       "      <td>2282102.0</td>\n",
       "      <td>402777.0</td>\n",
       "      <td>H</td>\n",
       "      <td>12</td>\n",
       "      <td>5444101.0</td>\n",
       "      <td>Circadian Experiment 1</td>\n",
       "      <td>February</td>\n",
       "      <td>102</td>\n",
       "      <td>2024-02-11 00:00:00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>369 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             plate.pos  sampID sample.type timepoint     date  \\\n",
       "filename                                                                        \n",
       "A2449446903_n01_LICRNA02_A01       A01  LIC157       plant       t14  1/23/24   \n",
       "A2449446903_n01_LICRNA02_A02       A02  LIC169       plant       t15  1/27/24   \n",
       "A2449446903_n01_LICRNA02_A03       A03  LIC193       plant       t17   2/2/24   \n",
       "A2449446903_n01_LICRNA02_A05       A05  LIC217       plant       t19   2/9/24   \n",
       "A2449446903_n01_LICRNA02_A06       A06  LIC229       plant       t20  2/16/24   \n",
       "...                                ...     ...         ...       ...      ...   \n",
       "B250508004_n01_LICRNA03_H08        H08  LIC428       plant    c1_t04  2/10/24   \n",
       "B250508004_n01_LICRNA03_H09        H09  LIC440       plant    c1_t05  2/10/24   \n",
       "B250508004_n01_LICRNA03_H10        H10  LIC452       plant    c1_t06  2/10/24   \n",
       "B250508004_n01_LICRNA03_H11        H11  LIC464       plant    c1_t07  2/10/24   \n",
       "B250508004_n01_LICRNA03_H12        H12  LIC488       plant    c1_t08  2/11/24   \n",
       "\n",
       "                               time extractplate rnaprepplate   Unmapped  \\\n",
       "filename                                                                   \n",
       "A2449446903_n01_LICRNA02_A01   8:00       LIC_02    LICRNA_02   758284.0   \n",
       "A2449446903_n01_LICRNA02_A02   8:00       LIC_02    LICRNA_02  3071877.0   \n",
       "A2449446903_n01_LICRNA02_A03   8:00       LIC_03    LICRNA_02  1213638.0   \n",
       "A2449446903_n01_LICRNA02_A05   8:00       LIC_03    LICRNA_02  1303436.0   \n",
       "A2449446903_n01_LICRNA02_A06   8:00       LIC_03    LICRNA_02  1591231.0   \n",
       "...                             ...          ...          ...        ...   \n",
       "B250508004_n01_LICRNA03_H08    8:00   LICCIRC_01    LICRNA_03  2552177.0   \n",
       "B250508004_n01_LICRNA03_H09   12:00   LICCIRC_01    LICRNA_03  4228242.0   \n",
       "B250508004_n01_LICRNA03_H10   16:00   LICCIRC_01    LICRNA_03   365095.0   \n",
       "B250508004_n01_LICRNA03_H11   20:00   LICCIRC_01    LICRNA_03   326959.0   \n",
       "B250508004_n01_LICRNA03_H12    0:00   LICCIRC_02    LICRNA_03  2282102.0   \n",
       "\n",
       "                              No Feature plate_row  plate_col  Total Reads  \\\n",
       "filename                                                                     \n",
       "A2449446903_n01_LICRNA02_A01     82176.0         A          1    1246970.0   \n",
       "A2449446903_n01_LICRNA02_A02    353549.0         A          2    4985602.0   \n",
       "A2449446903_n01_LICRNA02_A03    158864.0         A          3    2465662.0   \n",
       "A2449446903_n01_LICRNA02_A05     90929.0         A          5    1515395.0   \n",
       "A2449446903_n01_LICRNA02_A06    141665.0         A          6    1924391.0   \n",
       "...                                  ...       ...        ...          ...   \n",
       "B250508004_n01_LICRNA03_H08     438552.0         H          8    5245242.0   \n",
       "B250508004_n01_LICRNA03_H09     493699.0         H          9    7715482.0   \n",
       "B250508004_n01_LICRNA03_H10      39090.0         H         10     643786.0   \n",
       "B250508004_n01_LICRNA03_H11      60346.0         H         11    1509679.0   \n",
       "B250508004_n01_LICRNA03_H12     402777.0         H         12    5444101.0   \n",
       "\n",
       "                                     Experiment Type Sample Month  \\\n",
       "filename                                                            \n",
       "A2449446903_n01_LICRNA02_A01               Long Term      January   \n",
       "A2449446903_n01_LICRNA02_A02               Long Term      January   \n",
       "A2449446903_n01_LICRNA02_A03               Long Term     February   \n",
       "A2449446903_n01_LICRNA02_A05               Long Term     February   \n",
       "A2449446903_n01_LICRNA02_A06               Long Term     February   \n",
       "...                                              ...          ...   \n",
       "B250508004_n01_LICRNA03_H08   Circadian Experiment 1     February   \n",
       "B250508004_n01_LICRNA03_H09   Circadian Experiment 1     February   \n",
       "B250508004_n01_LICRNA03_H10   Circadian Experiment 1     February   \n",
       "B250508004_n01_LICRNA03_H11   Circadian Experiment 1     February   \n",
       "B250508004_n01_LICRNA03_H12   Circadian Experiment 1     February   \n",
       "\n",
       "                              daysincestart       Date and Time  \n",
       "filename                                                         \n",
       "A2449446903_n01_LICRNA02_A01             83 2024-01-23 08:00:00  \n",
       "A2449446903_n01_LICRNA02_A02             87 2024-01-27 08:00:00  \n",
       "A2449446903_n01_LICRNA02_A03             93 2024-02-02 08:00:00  \n",
       "A2449446903_n01_LICRNA02_A05            100 2024-02-09 08:00:00  \n",
       "A2449446903_n01_LICRNA02_A06            107 2024-02-16 08:00:00  \n",
       "...                                     ...                 ...  \n",
       "B250508004_n01_LICRNA03_H08             101 2024-02-10 08:00:00  \n",
       "B250508004_n01_LICRNA03_H09             101 2024-02-10 12:00:00  \n",
       "B250508004_n01_LICRNA03_H10             101 2024-02-10 16:00:00  \n",
       "B250508004_n01_LICRNA03_H11             101 2024-02-10 20:00:00  \n",
       "B250508004_n01_LICRNA03_H12             102 2024-02-11 00:00:00  \n",
       "\n",
       "[369 rows x 17 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "trimmed_metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "9275c94e",
   "metadata": {},
   "outputs": [],
   "source": [
    "circadian_metadata_1 = trimmed_metadata.loc[\n",
    "    trimmed_metadata[\"Experiment Type\"] == (\"Circadian Experiment 1\")\n",
    "]\n",
    "circadian_metadata_2 = trimmed_metadata.loc[\n",
    "    trimmed_metadata[\"Experiment Type\"] == (\"Circadian Experiment 2\")\n",
    "]\n",
    "\n",
    "circadian_transcriptome_1 = trimmed_transcriptome.loc[\n",
    "    trimmed_metadata[\"Experiment Type\"] == (\"Circadian Experiment 1\")\n",
    "]\n",
    "circadian_transcriptome_2 = trimmed_transcriptome.loc[\n",
    "    trimmed_metadata[\"Experiment Type\"] == (\"Circadian Experiment 2\")\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "c9fc055e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AT1G01010</th>\n",
       "      <th>AT1G01020</th>\n",
       "      <th>AT1G03987</th>\n",
       "      <th>AT1G01030</th>\n",
       "      <th>AT1G01040</th>\n",
       "      <th>AT1G01046</th>\n",
       "      <th>AT1G01050</th>\n",
       "      <th>AT1G01060</th>\n",
       "      <th>AT1G01070</th>\n",
       "      <th>AT1G04003</th>\n",
       "      <th>...</th>\n",
       "      <th>ArthCr090</th>\n",
       "      <th>ArthCr089</th>\n",
       "      <th>ArthCt114</th>\n",
       "      <th>ArthCt124</th>\n",
       "      <th>ArthCr088</th>\n",
       "      <th>ArthCt112</th>\n",
       "      <th>ArthCp088</th>\n",
       "      <th>ArthCp083</th>\n",
       "      <th>ArthCp084</th>\n",
       "      <th>ArthCp085</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A01</th>\n",
       "      <td>0.382839</td>\n",
       "      <td>11.929183</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.731527</td>\n",
       "      <td>3.602189</td>\n",
       "      <td>0.0</td>\n",
       "      <td>31.063543</td>\n",
       "      <td>3.966935</td>\n",
       "      <td>1.411759</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>62.741033</td>\n",
       "      <td>199.389218</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>17.950907</td>\n",
       "      <td>44.209074</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>6.266498</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A02</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>5.826682</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.501594</td>\n",
       "      <td>3.439569</td>\n",
       "      <td>0.0</td>\n",
       "      <td>22.712417</td>\n",
       "      <td>18.186693</td>\n",
       "      <td>1.249825</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>4.165830</td>\n",
       "      <td>69.172057</td>\n",
       "      <td>19.592718</td>\n",
       "      <td>1.986484</td>\n",
       "      <td>9.688606</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>5.501032</td>\n",
       "      <td>0.519342</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.467317</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A03</th>\n",
       "      <td>1.021565</td>\n",
       "      <td>7.683520</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.301733</td>\n",
       "      <td>2.700528</td>\n",
       "      <td>0.0</td>\n",
       "      <td>42.431625</td>\n",
       "      <td>12.937632</td>\n",
       "      <td>1.412672</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>50.225286</td>\n",
       "      <td>122.221814</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>7.983340</td>\n",
       "      <td>16.384320</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>7.983340</td>\n",
       "      <td>0.166972</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A05</th>\n",
       "      <td>0.732717</td>\n",
       "      <td>3.149144</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.947758</td>\n",
       "      <td>3.742586</td>\n",
       "      <td>0.0</td>\n",
       "      <td>26.541337</td>\n",
       "      <td>23.861601</td>\n",
       "      <td>6.417176</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>6.004011</td>\n",
       "      <td>73.945487</td>\n",
       "      <td>42.357065</td>\n",
       "      <td>17.178143</td>\n",
       "      <td>17.005325</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.628742</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A06</th>\n",
       "      <td>1.614233</td>\n",
       "      <td>1.238895</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.130092</td>\n",
       "      <td>4.215597</td>\n",
       "      <td>0.0</td>\n",
       "      <td>19.379509</td>\n",
       "      <td>5.006570</td>\n",
       "      <td>1.275568</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>41.571544</td>\n",
       "      <td>302.404404</td>\n",
       "      <td>85.317440</td>\n",
       "      <td>16.219201</td>\n",
       "      <td>43.599301</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.074213</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_G04</th>\n",
       "      <td>0.496406</td>\n",
       "      <td>8.356214</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.372257</td>\n",
       "      <td>3.069350</td>\n",
       "      <td>0.0</td>\n",
       "      <td>60.177712</td>\n",
       "      <td>23.758980</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>18.982294</td>\n",
       "      <td>141.544025</td>\n",
       "      <td>15.304706</td>\n",
       "      <td>3.879318</td>\n",
       "      <td>10.115887</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>10.145909</td>\n",
       "      <td>0.365112</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.401263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H01</th>\n",
       "      <td>1.365172</td>\n",
       "      <td>7.171230</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.926502</td>\n",
       "      <td>3.574878</td>\n",
       "      <td>0.0</td>\n",
       "      <td>53.113875</td>\n",
       "      <td>40.291678</td>\n",
       "      <td>5.174049</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>6.629010</td>\n",
       "      <td>44.223281</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.185396</td>\n",
       "      <td>6.525640</td>\n",
       "      <td>16.595542</td>\n",
       "      <td>4.923952</td>\n",
       "      <td>0.384285</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.448467</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H02</th>\n",
       "      <td>3.903593</td>\n",
       "      <td>1.198375</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.482399</td>\n",
       "      <td>3.897820</td>\n",
       "      <td>0.0</td>\n",
       "      <td>24.240094</td>\n",
       "      <td>8.805132</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>27.417193</td>\n",
       "      <td>166.825328</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>91.517576</td>\n",
       "      <td>14.520754</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.011375</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>46.732379</td>\n",
       "      <td>2.281997</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H03</th>\n",
       "      <td>0.365827</td>\n",
       "      <td>4.913399</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.972466</td>\n",
       "      <td>3.761743</td>\n",
       "      <td>0.0</td>\n",
       "      <td>14.311528</td>\n",
       "      <td>10.334051</td>\n",
       "      <td>1.096082</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>14.988251</td>\n",
       "      <td>73.563509</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.288305</td>\n",
       "      <td>7.558461</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.329870</td>\n",
       "      <td>0.022423</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.865277</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H04</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.986115</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.736795</td>\n",
       "      <td>2.716844</td>\n",
       "      <td>0.0</td>\n",
       "      <td>20.203592</td>\n",
       "      <td>11.551172</td>\n",
       "      <td>1.356243</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>13.627122</td>\n",
       "      <td>87.950335</td>\n",
       "      <td>5.916095</td>\n",
       "      <td>8.997395</td>\n",
       "      <td>4.417232</td>\n",
       "      <td>8.997395</td>\n",
       "      <td>5.767561</td>\n",
       "      <td>0.297953</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>7.852272</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>214 rows × 27418 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              AT1G01010  AT1G01020  AT1G03987  AT1G01030  \\\n",
       "A2449446903_n01_LICRNA02_A01   0.382839  11.929183        0.0   3.731527   \n",
       "A2449446903_n01_LICRNA02_A02   0.000000   5.826682        0.0   1.501594   \n",
       "A2449446903_n01_LICRNA02_A03   1.021565   7.683520        0.0   0.301733   \n",
       "A2449446903_n01_LICRNA02_A05   0.732717   3.149144        0.0   1.947758   \n",
       "A2449446903_n01_LICRNA02_A06   1.614233   1.238895        0.0   6.130092   \n",
       "...                                 ...        ...        ...        ...   \n",
       "B250508004_n01_LICRNA03_G04    0.496406   8.356214        0.0   3.372257   \n",
       "B250508004_n01_LICRNA03_H01    1.365172   7.171230        0.0   1.926502   \n",
       "B250508004_n01_LICRNA03_H02    3.903593   1.198375        0.0   1.482399   \n",
       "B250508004_n01_LICRNA03_H03    0.365827   4.913399        0.0   0.972466   \n",
       "B250508004_n01_LICRNA03_H04    0.000000   3.986115        0.0   0.736795   \n",
       "\n",
       "                              AT1G01040  AT1G01046  AT1G01050  AT1G01060  \\\n",
       "A2449446903_n01_LICRNA02_A01   3.602189        0.0  31.063543   3.966935   \n",
       "A2449446903_n01_LICRNA02_A02   3.439569        0.0  22.712417  18.186693   \n",
       "A2449446903_n01_LICRNA02_A03   2.700528        0.0  42.431625  12.937632   \n",
       "A2449446903_n01_LICRNA02_A05   3.742586        0.0  26.541337  23.861601   \n",
       "A2449446903_n01_LICRNA02_A06   4.215597        0.0  19.379509   5.006570   \n",
       "...                                 ...        ...        ...        ...   \n",
       "B250508004_n01_LICRNA03_G04    3.069350        0.0  60.177712  23.758980   \n",
       "B250508004_n01_LICRNA03_H01    3.574878        0.0  53.113875  40.291678   \n",
       "B250508004_n01_LICRNA03_H02    3.897820        0.0  24.240094   8.805132   \n",
       "B250508004_n01_LICRNA03_H03    3.761743        0.0  14.311528  10.334051   \n",
       "B250508004_n01_LICRNA03_H04    2.716844        0.0  20.203592  11.551172   \n",
       "\n",
       "                              AT1G01070  AT1G04003  ...  ArthCr090  \\\n",
       "A2449446903_n01_LICRNA02_A01   1.411759        0.0  ...  62.741033   \n",
       "A2449446903_n01_LICRNA02_A02   1.249825        0.0  ...   4.165830   \n",
       "A2449446903_n01_LICRNA02_A03   1.412672        0.0  ...  50.225286   \n",
       "A2449446903_n01_LICRNA02_A05   6.417176        0.0  ...   6.004011   \n",
       "A2449446903_n01_LICRNA02_A06   1.275568        0.0  ...  41.571544   \n",
       "...                                 ...        ...  ...        ...   \n",
       "B250508004_n01_LICRNA03_G04    0.000000        0.0  ...  18.982294   \n",
       "B250508004_n01_LICRNA03_H01    5.174049        0.0  ...   6.629010   \n",
       "B250508004_n01_LICRNA03_H02    0.000000        0.0  ...  27.417193   \n",
       "B250508004_n01_LICRNA03_H03    1.096082        0.0  ...  14.988251   \n",
       "B250508004_n01_LICRNA03_H04    1.356243        0.0  ...  13.627122   \n",
       "\n",
       "                               ArthCr089  ArthCt114  ArthCt124  ArthCr088  \\\n",
       "A2449446903_n01_LICRNA02_A01  199.389218   0.000000  17.950907  44.209074   \n",
       "A2449446903_n01_LICRNA02_A02   69.172057  19.592718   1.986484   9.688606   \n",
       "A2449446903_n01_LICRNA02_A03  122.221814   0.000000   7.983340  16.384320   \n",
       "A2449446903_n01_LICRNA02_A05   73.945487  42.357065  17.178143  17.005325   \n",
       "A2449446903_n01_LICRNA02_A06  302.404404  85.317440  16.219201  43.599301   \n",
       "...                                  ...        ...        ...        ...   \n",
       "B250508004_n01_LICRNA03_G04   141.544025  15.304706   3.879318  10.115887   \n",
       "B250508004_n01_LICRNA03_H01    44.223281   0.000000   1.185396   6.525640   \n",
       "B250508004_n01_LICRNA03_H02   166.825328   0.000000  91.517576  14.520754   \n",
       "B250508004_n01_LICRNA03_H03    73.563509   0.000000   4.288305   7.558461   \n",
       "B250508004_n01_LICRNA03_H04    87.950335   5.916095   8.997395   4.417232   \n",
       "\n",
       "                              ArthCt112  ArthCp088  ArthCp083  ArthCp084  \\\n",
       "A2449446903_n01_LICRNA02_A01   0.000000   0.000000   0.000000   0.000000   \n",
       "A2449446903_n01_LICRNA02_A02   0.000000   5.501032   0.519342   0.000000   \n",
       "A2449446903_n01_LICRNA02_A03   0.000000   7.983340   0.166972   0.000000   \n",
       "A2449446903_n01_LICRNA02_A05   0.000000   0.000000   0.628742   0.000000   \n",
       "A2449446903_n01_LICRNA02_A06   0.000000   0.000000   1.074213   0.000000   \n",
       "...                                 ...        ...        ...        ...   \n",
       "B250508004_n01_LICRNA03_G04    0.000000  10.145909   0.365112   0.000000   \n",
       "B250508004_n01_LICRNA03_H01   16.595542   4.923952   0.384285   0.000000   \n",
       "B250508004_n01_LICRNA03_H02    0.000000   2.011375   0.000000  46.732379   \n",
       "B250508004_n01_LICRNA03_H03    0.000000   0.329870   0.022423   0.000000   \n",
       "B250508004_n01_LICRNA03_H04    8.997395   5.767561   0.297953   0.000000   \n",
       "\n",
       "                              ArthCp085  \n",
       "A2449446903_n01_LICRNA02_A01   6.266498  \n",
       "A2449446903_n01_LICRNA02_A02   3.467317  \n",
       "A2449446903_n01_LICRNA02_A03   0.000000  \n",
       "A2449446903_n01_LICRNA02_A05   0.000000  \n",
       "A2449446903_n01_LICRNA02_A06   0.000000  \n",
       "...                                 ...  \n",
       "B250508004_n01_LICRNA03_G04    4.401263  \n",
       "B250508004_n01_LICRNA03_H01    4.448467  \n",
       "B250508004_n01_LICRNA03_H02    2.281997  \n",
       "B250508004_n01_LICRNA03_H03    4.865277  \n",
       "B250508004_n01_LICRNA03_H04    7.852272  \n",
       "\n",
       "[214 rows x 27418 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "long_term_transcriptome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "042cda60",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>plate.pos</th>\n",
       "      <th>sampID</th>\n",
       "      <th>sample.type</th>\n",
       "      <th>timepoint</th>\n",
       "      <th>date</th>\n",
       "      <th>time</th>\n",
       "      <th>extractplate</th>\n",
       "      <th>rnaprepplate</th>\n",
       "      <th>Unmapped</th>\n",
       "      <th>No Feature</th>\n",
       "      <th>plate_row</th>\n",
       "      <th>plate_col</th>\n",
       "      <th>Total Reads</th>\n",
       "      <th>Experiment Type</th>\n",
       "      <th>Sample Month</th>\n",
       "      <th>daysincestart</th>\n",
       "      <th>Date and Time</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>filename</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A01</th>\n",
       "      <td>A01</td>\n",
       "      <td>LIC157</td>\n",
       "      <td>plant</td>\n",
       "      <td>t14</td>\n",
       "      <td>1/23/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_02</td>\n",
       "      <td>LICRNA_02</td>\n",
       "      <td>758284.0</td>\n",
       "      <td>82176.0</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "      <td>1246970.0</td>\n",
       "      <td>Long Term</td>\n",
       "      <td>January</td>\n",
       "      <td>83</td>\n",
       "      <td>2024-01-23 08:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A02</th>\n",
       "      <td>A02</td>\n",
       "      <td>LIC169</td>\n",
       "      <td>plant</td>\n",
       "      <td>t15</td>\n",
       "      <td>1/27/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_02</td>\n",
       "      <td>LICRNA_02</td>\n",
       "      <td>3071877.0</td>\n",
       "      <td>353549.0</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>4985602.0</td>\n",
       "      <td>Long Term</td>\n",
       "      <td>January</td>\n",
       "      <td>87</td>\n",
       "      <td>2024-01-27 08:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A03</th>\n",
       "      <td>A03</td>\n",
       "      <td>LIC193</td>\n",
       "      <td>plant</td>\n",
       "      <td>t17</td>\n",
       "      <td>2/2/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_03</td>\n",
       "      <td>LICRNA_02</td>\n",
       "      <td>1213638.0</td>\n",
       "      <td>158864.0</td>\n",
       "      <td>A</td>\n",
       "      <td>3</td>\n",
       "      <td>2465662.0</td>\n",
       "      <td>Long Term</td>\n",
       "      <td>February</td>\n",
       "      <td>93</td>\n",
       "      <td>2024-02-02 08:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A05</th>\n",
       "      <td>A05</td>\n",
       "      <td>LIC217</td>\n",
       "      <td>plant</td>\n",
       "      <td>t19</td>\n",
       "      <td>2/9/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_03</td>\n",
       "      <td>LICRNA_02</td>\n",
       "      <td>1303436.0</td>\n",
       "      <td>90929.0</td>\n",
       "      <td>A</td>\n",
       "      <td>5</td>\n",
       "      <td>1515395.0</td>\n",
       "      <td>Long Term</td>\n",
       "      <td>February</td>\n",
       "      <td>100</td>\n",
       "      <td>2024-02-09 08:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A06</th>\n",
       "      <td>A06</td>\n",
       "      <td>LIC229</td>\n",
       "      <td>plant</td>\n",
       "      <td>t20</td>\n",
       "      <td>2/16/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_03</td>\n",
       "      <td>LICRNA_02</td>\n",
       "      <td>1591231.0</td>\n",
       "      <td>141665.0</td>\n",
       "      <td>A</td>\n",
       "      <td>6</td>\n",
       "      <td>1924391.0</td>\n",
       "      <td>Long Term</td>\n",
       "      <td>February</td>\n",
       "      <td>107</td>\n",
       "      <td>2024-02-16 08:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_G04</th>\n",
       "      <td>G04</td>\n",
       "      <td>LIC367</td>\n",
       "      <td>plant</td>\n",
       "      <td>t08</td>\n",
       "      <td>12/5/23</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_04</td>\n",
       "      <td>LICRNA_03</td>\n",
       "      <td>1477644.0</td>\n",
       "      <td>189073.0</td>\n",
       "      <td>G</td>\n",
       "      <td>4</td>\n",
       "      <td>2732871.0</td>\n",
       "      <td>Long Term</td>\n",
       "      <td>December</td>\n",
       "      <td>34</td>\n",
       "      <td>2023-12-05 08:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H01</th>\n",
       "      <td>H01</td>\n",
       "      <td>LIC332</td>\n",
       "      <td>plant</td>\n",
       "      <td>t28</td>\n",
       "      <td>3/14/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_04</td>\n",
       "      <td>LICRNA_03</td>\n",
       "      <td>5963067.0</td>\n",
       "      <td>589822.0</td>\n",
       "      <td>H</td>\n",
       "      <td>1</td>\n",
       "      <td>9338754.0</td>\n",
       "      <td>Long Term</td>\n",
       "      <td>March</td>\n",
       "      <td>134</td>\n",
       "      <td>2024-03-14 08:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H02</th>\n",
       "      <td>H02</td>\n",
       "      <td>LIC344</td>\n",
       "      <td>plant</td>\n",
       "      <td>t24</td>\n",
       "      <td>3/1/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_04</td>\n",
       "      <td>LICRNA_03</td>\n",
       "      <td>275002.0</td>\n",
       "      <td>47458.0</td>\n",
       "      <td>H</td>\n",
       "      <td>2</td>\n",
       "      <td>717003.0</td>\n",
       "      <td>Long Term</td>\n",
       "      <td>March</td>\n",
       "      <td>121</td>\n",
       "      <td>2024-03-01 08:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H03</th>\n",
       "      <td>H03</td>\n",
       "      <td>LIC356</td>\n",
       "      <td>plant</td>\n",
       "      <td>t16</td>\n",
       "      <td>1/30/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_04</td>\n",
       "      <td>LICRNA_03</td>\n",
       "      <td>1718519.0</td>\n",
       "      <td>230389.0</td>\n",
       "      <td>H</td>\n",
       "      <td>3</td>\n",
       "      <td>4705779.0</td>\n",
       "      <td>Long Term</td>\n",
       "      <td>January</td>\n",
       "      <td>90</td>\n",
       "      <td>2024-01-30 08:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H04</th>\n",
       "      <td>H04</td>\n",
       "      <td>LIC368</td>\n",
       "      <td>plant</td>\n",
       "      <td>t08</td>\n",
       "      <td>12/5/23</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_04</td>\n",
       "      <td>LICRNA_03</td>\n",
       "      <td>3555625.0</td>\n",
       "      <td>471419.0</td>\n",
       "      <td>H</td>\n",
       "      <td>4</td>\n",
       "      <td>6250771.0</td>\n",
       "      <td>Long Term</td>\n",
       "      <td>December</td>\n",
       "      <td>34</td>\n",
       "      <td>2023-12-05 08:00:00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>214 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             plate.pos  sampID sample.type timepoint     date  \\\n",
       "filename                                                                        \n",
       "A2449446903_n01_LICRNA02_A01       A01  LIC157       plant       t14  1/23/24   \n",
       "A2449446903_n01_LICRNA02_A02       A02  LIC169       plant       t15  1/27/24   \n",
       "A2449446903_n01_LICRNA02_A03       A03  LIC193       plant       t17   2/2/24   \n",
       "A2449446903_n01_LICRNA02_A05       A05  LIC217       plant       t19   2/9/24   \n",
       "A2449446903_n01_LICRNA02_A06       A06  LIC229       plant       t20  2/16/24   \n",
       "...                                ...     ...         ...       ...      ...   \n",
       "B250508004_n01_LICRNA03_G04        G04  LIC367       plant       t08  12/5/23   \n",
       "B250508004_n01_LICRNA03_H01        H01  LIC332       plant       t28  3/14/24   \n",
       "B250508004_n01_LICRNA03_H02        H02  LIC344       plant       t24   3/1/24   \n",
       "B250508004_n01_LICRNA03_H03        H03  LIC356       plant       t16  1/30/24   \n",
       "B250508004_n01_LICRNA03_H04        H04  LIC368       plant       t08  12/5/23   \n",
       "\n",
       "                              time extractplate rnaprepplate   Unmapped  \\\n",
       "filename                                                                  \n",
       "A2449446903_n01_LICRNA02_A01  8:00       LIC_02    LICRNA_02   758284.0   \n",
       "A2449446903_n01_LICRNA02_A02  8:00       LIC_02    LICRNA_02  3071877.0   \n",
       "A2449446903_n01_LICRNA02_A03  8:00       LIC_03    LICRNA_02  1213638.0   \n",
       "A2449446903_n01_LICRNA02_A05  8:00       LIC_03    LICRNA_02  1303436.0   \n",
       "A2449446903_n01_LICRNA02_A06  8:00       LIC_03    LICRNA_02  1591231.0   \n",
       "...                            ...          ...          ...        ...   \n",
       "B250508004_n01_LICRNA03_G04   8:00       LIC_04    LICRNA_03  1477644.0   \n",
       "B250508004_n01_LICRNA03_H01   8:00       LIC_04    LICRNA_03  5963067.0   \n",
       "B250508004_n01_LICRNA03_H02   8:00       LIC_04    LICRNA_03   275002.0   \n",
       "B250508004_n01_LICRNA03_H03   8:00       LIC_04    LICRNA_03  1718519.0   \n",
       "B250508004_n01_LICRNA03_H04   8:00       LIC_04    LICRNA_03  3555625.0   \n",
       "\n",
       "                              No Feature plate_row  plate_col  Total Reads  \\\n",
       "filename                                                                     \n",
       "A2449446903_n01_LICRNA02_A01     82176.0         A          1    1246970.0   \n",
       "A2449446903_n01_LICRNA02_A02    353549.0         A          2    4985602.0   \n",
       "A2449446903_n01_LICRNA02_A03    158864.0         A          3    2465662.0   \n",
       "A2449446903_n01_LICRNA02_A05     90929.0         A          5    1515395.0   \n",
       "A2449446903_n01_LICRNA02_A06    141665.0         A          6    1924391.0   \n",
       "...                                  ...       ...        ...          ...   \n",
       "B250508004_n01_LICRNA03_G04     189073.0         G          4    2732871.0   \n",
       "B250508004_n01_LICRNA03_H01     589822.0         H          1    9338754.0   \n",
       "B250508004_n01_LICRNA03_H02      47458.0         H          2     717003.0   \n",
       "B250508004_n01_LICRNA03_H03     230389.0         H          3    4705779.0   \n",
       "B250508004_n01_LICRNA03_H04     471419.0         H          4    6250771.0   \n",
       "\n",
       "                             Experiment Type Sample Month  daysincestart  \\\n",
       "filename                                                                   \n",
       "A2449446903_n01_LICRNA02_A01       Long Term      January             83   \n",
       "A2449446903_n01_LICRNA02_A02       Long Term      January             87   \n",
       "A2449446903_n01_LICRNA02_A03       Long Term     February             93   \n",
       "A2449446903_n01_LICRNA02_A05       Long Term     February            100   \n",
       "A2449446903_n01_LICRNA02_A06       Long Term     February            107   \n",
       "...                                      ...          ...            ...   \n",
       "B250508004_n01_LICRNA03_G04        Long Term     December             34   \n",
       "B250508004_n01_LICRNA03_H01        Long Term        March            134   \n",
       "B250508004_n01_LICRNA03_H02        Long Term        March            121   \n",
       "B250508004_n01_LICRNA03_H03        Long Term      January             90   \n",
       "B250508004_n01_LICRNA03_H04        Long Term     December             34   \n",
       "\n",
       "                                   Date and Time  \n",
       "filename                                          \n",
       "A2449446903_n01_LICRNA02_A01 2024-01-23 08:00:00  \n",
       "A2449446903_n01_LICRNA02_A02 2024-01-27 08:00:00  \n",
       "A2449446903_n01_LICRNA02_A03 2024-02-02 08:00:00  \n",
       "A2449446903_n01_LICRNA02_A05 2024-02-09 08:00:00  \n",
       "A2449446903_n01_LICRNA02_A06 2024-02-16 08:00:00  \n",
       "...                                          ...  \n",
       "B250508004_n01_LICRNA03_G04  2023-12-05 08:00:00  \n",
       "B250508004_n01_LICRNA03_H01  2024-03-14 08:00:00  \n",
       "B250508004_n01_LICRNA03_H02  2024-03-01 08:00:00  \n",
       "B250508004_n01_LICRNA03_H03  2024-01-30 08:00:00  \n",
       "B250508004_n01_LICRNA03_H04  2023-12-05 08:00:00  \n",
       "\n",
       "[214 rows x 17 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "long_term_metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "f4883ef2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AT1G01010</th>\n",
       "      <th>AT1G01020</th>\n",
       "      <th>AT1G03987</th>\n",
       "      <th>AT1G01030</th>\n",
       "      <th>AT1G01040</th>\n",
       "      <th>AT1G01046</th>\n",
       "      <th>AT1G01050</th>\n",
       "      <th>AT1G01060</th>\n",
       "      <th>AT1G01070</th>\n",
       "      <th>AT1G04003</th>\n",
       "      <th>...</th>\n",
       "      <th>ArthCr090</th>\n",
       "      <th>ArthCr089</th>\n",
       "      <th>ArthCt114</th>\n",
       "      <th>ArthCt124</th>\n",
       "      <th>ArthCr088</th>\n",
       "      <th>ArthCt112</th>\n",
       "      <th>ArthCp088</th>\n",
       "      <th>ArthCp083</th>\n",
       "      <th>ArthCp084</th>\n",
       "      <th>ArthCp085</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A01</th>\n",
       "      <td>0.382839</td>\n",
       "      <td>11.929183</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.731527</td>\n",
       "      <td>3.602189</td>\n",
       "      <td>0.0</td>\n",
       "      <td>31.063543</td>\n",
       "      <td>3.966935</td>\n",
       "      <td>1.411759</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>62.741033</td>\n",
       "      <td>199.389218</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>17.950907</td>\n",
       "      <td>44.209074</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>6.266498</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A02</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>5.826682</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.501594</td>\n",
       "      <td>3.439569</td>\n",
       "      <td>0.0</td>\n",
       "      <td>22.712417</td>\n",
       "      <td>18.186693</td>\n",
       "      <td>1.249825</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>4.165830</td>\n",
       "      <td>69.172057</td>\n",
       "      <td>19.592718</td>\n",
       "      <td>1.986484</td>\n",
       "      <td>9.688606</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>5.501032</td>\n",
       "      <td>0.519342</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.467317</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A03</th>\n",
       "      <td>1.021565</td>\n",
       "      <td>7.683520</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.301733</td>\n",
       "      <td>2.700528</td>\n",
       "      <td>0.0</td>\n",
       "      <td>42.431625</td>\n",
       "      <td>12.937632</td>\n",
       "      <td>1.412672</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>50.225286</td>\n",
       "      <td>122.221814</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>7.983340</td>\n",
       "      <td>16.384320</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>7.983340</td>\n",
       "      <td>0.166972</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A05</th>\n",
       "      <td>0.732717</td>\n",
       "      <td>3.149144</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.947758</td>\n",
       "      <td>3.742586</td>\n",
       "      <td>0.0</td>\n",
       "      <td>26.541337</td>\n",
       "      <td>23.861601</td>\n",
       "      <td>6.417176</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>6.004011</td>\n",
       "      <td>73.945487</td>\n",
       "      <td>42.357065</td>\n",
       "      <td>17.178143</td>\n",
       "      <td>17.005325</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.628742</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2449446903_n01_LICRNA02_A06</th>\n",
       "      <td>1.614233</td>\n",
       "      <td>1.238895</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.130092</td>\n",
       "      <td>4.215597</td>\n",
       "      <td>0.0</td>\n",
       "      <td>19.379509</td>\n",
       "      <td>5.006570</td>\n",
       "      <td>1.275568</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>41.571544</td>\n",
       "      <td>302.404404</td>\n",
       "      <td>85.317440</td>\n",
       "      <td>16.219201</td>\n",
       "      <td>43.599301</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.074213</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_G04</th>\n",
       "      <td>0.496406</td>\n",
       "      <td>8.356214</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.372257</td>\n",
       "      <td>3.069350</td>\n",
       "      <td>0.0</td>\n",
       "      <td>60.177712</td>\n",
       "      <td>23.758980</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>18.982294</td>\n",
       "      <td>141.544025</td>\n",
       "      <td>15.304706</td>\n",
       "      <td>3.879318</td>\n",
       "      <td>10.115887</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>10.145909</td>\n",
       "      <td>0.365112</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.401263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H01</th>\n",
       "      <td>1.365172</td>\n",
       "      <td>7.171230</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.926502</td>\n",
       "      <td>3.574878</td>\n",
       "      <td>0.0</td>\n",
       "      <td>53.113875</td>\n",
       "      <td>40.291678</td>\n",
       "      <td>5.174049</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>6.629010</td>\n",
       "      <td>44.223281</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.185396</td>\n",
       "      <td>6.525640</td>\n",
       "      <td>16.595542</td>\n",
       "      <td>4.923952</td>\n",
       "      <td>0.384285</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.448467</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H02</th>\n",
       "      <td>3.903593</td>\n",
       "      <td>1.198375</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.482399</td>\n",
       "      <td>3.897820</td>\n",
       "      <td>0.0</td>\n",
       "      <td>24.240094</td>\n",
       "      <td>8.805132</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>27.417193</td>\n",
       "      <td>166.825328</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>91.517576</td>\n",
       "      <td>14.520754</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.011375</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>46.732379</td>\n",
       "      <td>2.281997</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H03</th>\n",
       "      <td>0.365827</td>\n",
       "      <td>4.913399</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.972466</td>\n",
       "      <td>3.761743</td>\n",
       "      <td>0.0</td>\n",
       "      <td>14.311528</td>\n",
       "      <td>10.334051</td>\n",
       "      <td>1.096082</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>14.988251</td>\n",
       "      <td>73.563509</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.288305</td>\n",
       "      <td>7.558461</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.329870</td>\n",
       "      <td>0.022423</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.865277</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B250508004_n01_LICRNA03_H04</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.986115</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.736795</td>\n",
       "      <td>2.716844</td>\n",
       "      <td>0.0</td>\n",
       "      <td>20.203592</td>\n",
       "      <td>11.551172</td>\n",
       "      <td>1.356243</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>13.627122</td>\n",
       "      <td>87.950335</td>\n",
       "      <td>5.916095</td>\n",
       "      <td>8.997395</td>\n",
       "      <td>4.417232</td>\n",
       "      <td>8.997395</td>\n",
       "      <td>5.767561</td>\n",
       "      <td>0.297953</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>7.852272</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>214 rows × 27418 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              AT1G01010  AT1G01020  AT1G03987  AT1G01030  \\\n",
       "A2449446903_n01_LICRNA02_A01   0.382839  11.929183        0.0   3.731527   \n",
       "A2449446903_n01_LICRNA02_A02   0.000000   5.826682        0.0   1.501594   \n",
       "A2449446903_n01_LICRNA02_A03   1.021565   7.683520        0.0   0.301733   \n",
       "A2449446903_n01_LICRNA02_A05   0.732717   3.149144        0.0   1.947758   \n",
       "A2449446903_n01_LICRNA02_A06   1.614233   1.238895        0.0   6.130092   \n",
       "...                                 ...        ...        ...        ...   \n",
       "B250508004_n01_LICRNA03_G04    0.496406   8.356214        0.0   3.372257   \n",
       "B250508004_n01_LICRNA03_H01    1.365172   7.171230        0.0   1.926502   \n",
       "B250508004_n01_LICRNA03_H02    3.903593   1.198375        0.0   1.482399   \n",
       "B250508004_n01_LICRNA03_H03    0.365827   4.913399        0.0   0.972466   \n",
       "B250508004_n01_LICRNA03_H04    0.000000   3.986115        0.0   0.736795   \n",
       "\n",
       "                              AT1G01040  AT1G01046  AT1G01050  AT1G01060  \\\n",
       "A2449446903_n01_LICRNA02_A01   3.602189        0.0  31.063543   3.966935   \n",
       "A2449446903_n01_LICRNA02_A02   3.439569        0.0  22.712417  18.186693   \n",
       "A2449446903_n01_LICRNA02_A03   2.700528        0.0  42.431625  12.937632   \n",
       "A2449446903_n01_LICRNA02_A05   3.742586        0.0  26.541337  23.861601   \n",
       "A2449446903_n01_LICRNA02_A06   4.215597        0.0  19.379509   5.006570   \n",
       "...                                 ...        ...        ...        ...   \n",
       "B250508004_n01_LICRNA03_G04    3.069350        0.0  60.177712  23.758980   \n",
       "B250508004_n01_LICRNA03_H01    3.574878        0.0  53.113875  40.291678   \n",
       "B250508004_n01_LICRNA03_H02    3.897820        0.0  24.240094   8.805132   \n",
       "B250508004_n01_LICRNA03_H03    3.761743        0.0  14.311528  10.334051   \n",
       "B250508004_n01_LICRNA03_H04    2.716844        0.0  20.203592  11.551172   \n",
       "\n",
       "                              AT1G01070  AT1G04003  ...  ArthCr090  \\\n",
       "A2449446903_n01_LICRNA02_A01   1.411759        0.0  ...  62.741033   \n",
       "A2449446903_n01_LICRNA02_A02   1.249825        0.0  ...   4.165830   \n",
       "A2449446903_n01_LICRNA02_A03   1.412672        0.0  ...  50.225286   \n",
       "A2449446903_n01_LICRNA02_A05   6.417176        0.0  ...   6.004011   \n",
       "A2449446903_n01_LICRNA02_A06   1.275568        0.0  ...  41.571544   \n",
       "...                                 ...        ...  ...        ...   \n",
       "B250508004_n01_LICRNA03_G04    0.000000        0.0  ...  18.982294   \n",
       "B250508004_n01_LICRNA03_H01    5.174049        0.0  ...   6.629010   \n",
       "B250508004_n01_LICRNA03_H02    0.000000        0.0  ...  27.417193   \n",
       "B250508004_n01_LICRNA03_H03    1.096082        0.0  ...  14.988251   \n",
       "B250508004_n01_LICRNA03_H04    1.356243        0.0  ...  13.627122   \n",
       "\n",
       "                               ArthCr089  ArthCt114  ArthCt124  ArthCr088  \\\n",
       "A2449446903_n01_LICRNA02_A01  199.389218   0.000000  17.950907  44.209074   \n",
       "A2449446903_n01_LICRNA02_A02   69.172057  19.592718   1.986484   9.688606   \n",
       "A2449446903_n01_LICRNA02_A03  122.221814   0.000000   7.983340  16.384320   \n",
       "A2449446903_n01_LICRNA02_A05   73.945487  42.357065  17.178143  17.005325   \n",
       "A2449446903_n01_LICRNA02_A06  302.404404  85.317440  16.219201  43.599301   \n",
       "...                                  ...        ...        ...        ...   \n",
       "B250508004_n01_LICRNA03_G04   141.544025  15.304706   3.879318  10.115887   \n",
       "B250508004_n01_LICRNA03_H01    44.223281   0.000000   1.185396   6.525640   \n",
       "B250508004_n01_LICRNA03_H02   166.825328   0.000000  91.517576  14.520754   \n",
       "B250508004_n01_LICRNA03_H03    73.563509   0.000000   4.288305   7.558461   \n",
       "B250508004_n01_LICRNA03_H04    87.950335   5.916095   8.997395   4.417232   \n",
       "\n",
       "                              ArthCt112  ArthCp088  ArthCp083  ArthCp084  \\\n",
       "A2449446903_n01_LICRNA02_A01   0.000000   0.000000   0.000000   0.000000   \n",
       "A2449446903_n01_LICRNA02_A02   0.000000   5.501032   0.519342   0.000000   \n",
       "A2449446903_n01_LICRNA02_A03   0.000000   7.983340   0.166972   0.000000   \n",
       "A2449446903_n01_LICRNA02_A05   0.000000   0.000000   0.628742   0.000000   \n",
       "A2449446903_n01_LICRNA02_A06   0.000000   0.000000   1.074213   0.000000   \n",
       "...                                 ...        ...        ...        ...   \n",
       "B250508004_n01_LICRNA03_G04    0.000000  10.145909   0.365112   0.000000   \n",
       "B250508004_n01_LICRNA03_H01   16.595542   4.923952   0.384285   0.000000   \n",
       "B250508004_n01_LICRNA03_H02    0.000000   2.011375   0.000000  46.732379   \n",
       "B250508004_n01_LICRNA03_H03    0.000000   0.329870   0.022423   0.000000   \n",
       "B250508004_n01_LICRNA03_H04    8.997395   5.767561   0.297953   0.000000   \n",
       "\n",
       "                              ArthCp085  \n",
       "A2449446903_n01_LICRNA02_A01   6.266498  \n",
       "A2449446903_n01_LICRNA02_A02   3.467317  \n",
       "A2449446903_n01_LICRNA02_A03   0.000000  \n",
       "A2449446903_n01_LICRNA02_A05   0.000000  \n",
       "A2449446903_n01_LICRNA02_A06   0.000000  \n",
       "...                                 ...  \n",
       "B250508004_n01_LICRNA03_G04    4.401263  \n",
       "B250508004_n01_LICRNA03_H01    4.448467  \n",
       "B250508004_n01_LICRNA03_H02    2.281997  \n",
       "B250508004_n01_LICRNA03_H03    4.865277  \n",
       "B250508004_n01_LICRNA03_H04    7.852272  \n",
       "\n",
       "[214 rows x 27418 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "long_term_transcriptome = long_term_transcriptome.sort_index()\n",
    "long_term_metadata = long_term_metadata.sort_index()\n",
    "\n",
    "long_term_transcriptome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "2d961b85",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample</th>\n",
       "      <th>sample.type</th>\n",
       "      <th>timepoint</th>\n",
       "      <th>date</th>\n",
       "      <th>time</th>\n",
       "      <th>platename</th>\n",
       "      <th>daysincestart</th>\n",
       "      <th>plantID</th>\n",
       "      <th>Kingdom</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "      <th>AbundR100</th>\n",
       "      <th>sampID</th>\n",
       "      <th>Experiment Type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>LIC063</td>\n",
       "      <td>plant</td>\n",
       "      <td>t06</td>\n",
       "      <td>11/28/23</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_01</td>\n",
       "      <td>27</td>\n",
       "      <td>LIC063</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Sphingomonadales</td>\n",
       "      <td>Sphingomonadaceae_486827</td>\n",
       "      <td>Sphingomonas_L_486704</td>\n",
       "      <td>Sphingomonas_L_486704 sp000786205</td>\n",
       "      <td>66.1</td>\n",
       "      <td>LIC063</td>\n",
       "      <td>Long Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LIC159</td>\n",
       "      <td>plant</td>\n",
       "      <td>t14</td>\n",
       "      <td>1/23/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_02</td>\n",
       "      <td>83</td>\n",
       "      <td>LIC159</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>Cyanobacteriota</td>\n",
       "      <td>Cyanobacteriia</td>\n",
       "      <td>Cyanobacteriales</td>\n",
       "      <td>Coleofasciculaceae</td>\n",
       "      <td>Caldora</td>\n",
       "      <td>Caldora sp010672925</td>\n",
       "      <td>60.5</td>\n",
       "      <td>LIC159</td>\n",
       "      <td>Long Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LIC306</td>\n",
       "      <td>plant</td>\n",
       "      <td>t26</td>\n",
       "      <td>3/8/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_04</td>\n",
       "      <td>128</td>\n",
       "      <td>LIC306</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Sphingomonadales</td>\n",
       "      <td>Sphingomonadaceae_486827</td>\n",
       "      <td>Sphingomonas_L_486704</td>\n",
       "      <td>Sphingomonas_L_486704 sp000786205</td>\n",
       "      <td>57.2</td>\n",
       "      <td>LIC306</td>\n",
       "      <td>Long Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>LIC016</td>\n",
       "      <td>plant</td>\n",
       "      <td>t02</td>\n",
       "      <td>11/7/23</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_01</td>\n",
       "      <td>6</td>\n",
       "      <td>LIC016</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Sphingomonadales</td>\n",
       "      <td>Sphingomonadaceae_486827</td>\n",
       "      <td>Sphingomonas_L_486704</td>\n",
       "      <td>Sphingomonas_L_486704 sp000786205</td>\n",
       "      <td>54.4</td>\n",
       "      <td>LIC016</td>\n",
       "      <td>Long Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>LIC051</td>\n",
       "      <td>plant</td>\n",
       "      <td>t05</td>\n",
       "      <td>11/21/23</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_01</td>\n",
       "      <td>20</td>\n",
       "      <td>LIC051</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Rhizobiales_505101</td>\n",
       "      <td>Rhizobiaceae</td>\n",
       "      <td>Neorhizobium_500197</td>\n",
       "      <td>Neorhizobium soli</td>\n",
       "      <td>49.7</td>\n",
       "      <td>LIC051</td>\n",
       "      <td>Long Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12278</th>\n",
       "      <td>LIC157</td>\n",
       "      <td>plant</td>\n",
       "      <td>t14</td>\n",
       "      <td>1/23/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_02</td>\n",
       "      <td>83</td>\n",
       "      <td>LIC157</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>Actinomycetota</td>\n",
       "      <td>Actinomycetes</td>\n",
       "      <td>Propionibacteriales</td>\n",
       "      <td>Propionibacteriaceae</td>\n",
       "      <td>Friedmanniella</td>\n",
       "      <td>Friedmanniella luteola</td>\n",
       "      <td>0.5</td>\n",
       "      <td>LIC157</td>\n",
       "      <td>Long Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12280</th>\n",
       "      <td>LIC195</td>\n",
       "      <td>plant</td>\n",
       "      <td>t17</td>\n",
       "      <td>2/2/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_03</td>\n",
       "      <td>93</td>\n",
       "      <td>LIC195</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>Actinomycetota</td>\n",
       "      <td>Actinomycetes</td>\n",
       "      <td>Propionibacteriales</td>\n",
       "      <td>Propionibacteriaceae</td>\n",
       "      <td>Friedmanniella</td>\n",
       "      <td>Friedmanniella flava</td>\n",
       "      <td>0.5</td>\n",
       "      <td>LIC195</td>\n",
       "      <td>Long Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12281</th>\n",
       "      <td>LIC122</td>\n",
       "      <td>plant</td>\n",
       "      <td>t11</td>\n",
       "      <td>12/15/23</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_02</td>\n",
       "      <td>44</td>\n",
       "      <td>LIC122</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>Actinomycetota</td>\n",
       "      <td>Actinomycetes</td>\n",
       "      <td>Propionibacteriales</td>\n",
       "      <td>Nocardioidaceae</td>\n",
       "      <td>Nocardioides_A_392796</td>\n",
       "      <td>Nocardioides_A_392796 hankookensis</td>\n",
       "      <td>0.5</td>\n",
       "      <td>LIC122</td>\n",
       "      <td>Long Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12282</th>\n",
       "      <td>LIC028</td>\n",
       "      <td>plant</td>\n",
       "      <td>t03</td>\n",
       "      <td>11/14/23</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_01</td>\n",
       "      <td>13</td>\n",
       "      <td>LIC028</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>Actinomycetota</td>\n",
       "      <td>Actinomycetes</td>\n",
       "      <td>Mycobacteriales</td>\n",
       "      <td>Geodermatophilaceae</td>\n",
       "      <td>Modestobacter</td>\n",
       "      <td>Modestobacter muralis</td>\n",
       "      <td>0.5</td>\n",
       "      <td>LIC028</td>\n",
       "      <td>Long Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12285</th>\n",
       "      <td>LIC230</td>\n",
       "      <td>plant</td>\n",
       "      <td>t20</td>\n",
       "      <td>2/16/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_03</td>\n",
       "      <td>107</td>\n",
       "      <td>LIC230</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Rhizobiales_505101</td>\n",
       "      <td>Beijerinckiaceae</td>\n",
       "      <td>Methylobacterium</td>\n",
       "      <td>Methylobacterium sp001424705</td>\n",
       "      <td>0.5</td>\n",
       "      <td>LIC230</td>\n",
       "      <td>Long Term</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7379 rows × 18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Sample sample.type timepoint      date  time platename  daysincestart  \\\n",
       "1      LIC063       plant       t06  11/28/23  8:00    LIC_01             27   \n",
       "2      LIC159       plant       t14   1/23/24  8:00    LIC_02             83   \n",
       "4      LIC306       plant       t26    3/8/24  8:00    LIC_04            128   \n",
       "5      LIC016       plant       t02   11/7/23  8:00    LIC_01              6   \n",
       "9      LIC051       plant       t05  11/21/23  8:00    LIC_01             20   \n",
       "...       ...         ...       ...       ...   ...       ...            ...   \n",
       "12278  LIC157       plant       t14   1/23/24  8:00    LIC_02             83   \n",
       "12280  LIC195       plant       t17    2/2/24  8:00    LIC_03             93   \n",
       "12281  LIC122       plant       t11  12/15/23  8:00    LIC_02             44   \n",
       "12282  LIC028       plant       t03  11/14/23  8:00    LIC_01             13   \n",
       "12285  LIC230       plant       t20   2/16/24  8:00    LIC_03            107   \n",
       "\n",
       "      plantID      Kingdom           Phylum                Class  \\\n",
       "1      LIC063  d__Bacteria   Pseudomonadota  Alphaproteobacteria   \n",
       "2      LIC159  d__Bacteria  Cyanobacteriota       Cyanobacteriia   \n",
       "4      LIC306  d__Bacteria   Pseudomonadota  Alphaproteobacteria   \n",
       "5      LIC016  d__Bacteria   Pseudomonadota  Alphaproteobacteria   \n",
       "9      LIC051  d__Bacteria   Pseudomonadota  Alphaproteobacteria   \n",
       "...       ...          ...              ...                  ...   \n",
       "12278  LIC157  d__Bacteria   Actinomycetota        Actinomycetes   \n",
       "12280  LIC195  d__Bacteria   Actinomycetota        Actinomycetes   \n",
       "12281  LIC122  d__Bacteria   Actinomycetota        Actinomycetes   \n",
       "12282  LIC028  d__Bacteria   Actinomycetota        Actinomycetes   \n",
       "12285  LIC230  d__Bacteria   Pseudomonadota  Alphaproteobacteria   \n",
       "\n",
       "                     Order                    Family                  Genus  \\\n",
       "1         Sphingomonadales  Sphingomonadaceae_486827  Sphingomonas_L_486704   \n",
       "2         Cyanobacteriales        Coleofasciculaceae                Caldora   \n",
       "4         Sphingomonadales  Sphingomonadaceae_486827  Sphingomonas_L_486704   \n",
       "5         Sphingomonadales  Sphingomonadaceae_486827  Sphingomonas_L_486704   \n",
       "9       Rhizobiales_505101              Rhizobiaceae    Neorhizobium_500197   \n",
       "...                    ...                       ...                    ...   \n",
       "12278  Propionibacteriales      Propionibacteriaceae         Friedmanniella   \n",
       "12280  Propionibacteriales      Propionibacteriaceae         Friedmanniella   \n",
       "12281  Propionibacteriales           Nocardioidaceae  Nocardioides_A_392796   \n",
       "12282      Mycobacteriales       Geodermatophilaceae          Modestobacter   \n",
       "12285   Rhizobiales_505101          Beijerinckiaceae       Methylobacterium   \n",
       "\n",
       "                                  Species  AbundR100  sampID Experiment Type  \n",
       "1       Sphingomonas_L_486704 sp000786205       66.1  LIC063       Long Term  \n",
       "2                     Caldora sp010672925       60.5  LIC159       Long Term  \n",
       "4       Sphingomonas_L_486704 sp000786205       57.2  LIC306       Long Term  \n",
       "5       Sphingomonas_L_486704 sp000786205       54.4  LIC016       Long Term  \n",
       "9                       Neorhizobium soli       49.7  LIC051       Long Term  \n",
       "...                                   ...        ...     ...             ...  \n",
       "12278              Friedmanniella luteola        0.5  LIC157       Long Term  \n",
       "12280                Friedmanniella flava        0.5  LIC195       Long Term  \n",
       "12281  Nocardioides_A_392796 hankookensis        0.5  LIC122       Long Term  \n",
       "12282               Modestobacter muralis        0.5  LIC028       Long Term  \n",
       "12285        Methylobacterium sp001424705        0.5  LIC230       Long Term  \n",
       "\n",
       "[7379 rows x 18 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "microbiome_abundance = pd.read_csv(\n",
    "    \"/Users/michael/Data/Luke_terrace_experiment/Microbiome/lic2024_16S_rab.csv\"\n",
    ")\n",
    "merged_microbiome = microbiome_abundance.merge(\n",
    "    metadata[[\"sampID\", \"Experiment Type\"]], left_on=\"plantID\", right_on=\"sampID\"\n",
    ")\n",
    "long_term_microbiome = merged_microbiome.loc[\n",
    "    merged_microbiome[\"Experiment Type\"] == \"Long Term\"\n",
    "]\n",
    "long_term_microbiome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "25cdfeb2",
   "metadata": {},
   "outputs": [],
   "source": [
    "circadian_microbiome_1 = merged_microbiome.loc[\n",
    "    merged_microbiome[\"Experiment Type\"] == \"Circadian Experiment 1\"\n",
    "]\n",
    "\n",
    "circadian_microbiome_1 = merged_microbiome.loc[\n",
    "    merged_microbiome[\"Experiment Type\"] == \"Circadian Experiment 1\"\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "944d67e1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample</th>\n",
       "      <th>sample.type</th>\n",
       "      <th>timepoint</th>\n",
       "      <th>date</th>\n",
       "      <th>time</th>\n",
       "      <th>platename</th>\n",
       "      <th>daysincestart</th>\n",
       "      <th>plantID</th>\n",
       "      <th>Kingdom</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "      <th>AbundR100</th>\n",
       "      <th>sampID</th>\n",
       "      <th>Experiment Type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LIC522</td>\n",
       "      <td>plant</td>\n",
       "      <td>c1_t11</td>\n",
       "      <td>2/11/24</td>\n",
       "      <td>12:00</td>\n",
       "      <td>LICCIRC_02</td>\n",
       "      <td>102</td>\n",
       "      <td>LIC522</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Pseudomonadales_A_650611</td>\n",
       "      <td>Pseudomonadaceae</td>\n",
       "      <td>Pseudomonas_E_647464</td>\n",
       "      <td>Pseudomonas_E_647464 viridiflava</td>\n",
       "      <td>79.1</td>\n",
       "      <td>LIC522</td>\n",
       "      <td>Circadian Experiment 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>LIC063</td>\n",
       "      <td>plant</td>\n",
       "      <td>t06</td>\n",
       "      <td>11/28/23</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_01</td>\n",
       "      <td>27</td>\n",
       "      <td>LIC063</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Sphingomonadales</td>\n",
       "      <td>Sphingomonadaceae_486827</td>\n",
       "      <td>Sphingomonas_L_486704</td>\n",
       "      <td>Sphingomonas_L_486704 sp000786205</td>\n",
       "      <td>66.1</td>\n",
       "      <td>LIC063</td>\n",
       "      <td>Long Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LIC159</td>\n",
       "      <td>plant</td>\n",
       "      <td>t14</td>\n",
       "      <td>1/23/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_02</td>\n",
       "      <td>83</td>\n",
       "      <td>LIC159</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>Cyanobacteriota</td>\n",
       "      <td>Cyanobacteriia</td>\n",
       "      <td>Cyanobacteriales</td>\n",
       "      <td>Coleofasciculaceae</td>\n",
       "      <td>Caldora</td>\n",
       "      <td>Caldora sp010672925</td>\n",
       "      <td>60.5</td>\n",
       "      <td>LIC159</td>\n",
       "      <td>Long Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LIC452</td>\n",
       "      <td>plant</td>\n",
       "      <td>c1_t06</td>\n",
       "      <td>2/10/24</td>\n",
       "      <td>16:00</td>\n",
       "      <td>LICCIRC_01</td>\n",
       "      <td>101</td>\n",
       "      <td>LIC452</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Burkholderiales</td>\n",
       "      <td>Burkholderiaceae_A_595421</td>\n",
       "      <td>Massilia_574544</td>\n",
       "      <td>Massilia atriviolacea</td>\n",
       "      <td>59.8</td>\n",
       "      <td>LIC452</td>\n",
       "      <td>Circadian Experiment 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LIC306</td>\n",
       "      <td>plant</td>\n",
       "      <td>t26</td>\n",
       "      <td>3/8/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_04</td>\n",
       "      <td>128</td>\n",
       "      <td>LIC306</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Sphingomonadales</td>\n",
       "      <td>Sphingomonadaceae_486827</td>\n",
       "      <td>Sphingomonas_L_486704</td>\n",
       "      <td>Sphingomonas_L_486704 sp000786205</td>\n",
       "      <td>57.2</td>\n",
       "      <td>LIC306</td>\n",
       "      <td>Long Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12281</th>\n",
       "      <td>LIC122</td>\n",
       "      <td>plant</td>\n",
       "      <td>t11</td>\n",
       "      <td>12/15/23</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_02</td>\n",
       "      <td>44</td>\n",
       "      <td>LIC122</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>Actinomycetota</td>\n",
       "      <td>Actinomycetes</td>\n",
       "      <td>Propionibacteriales</td>\n",
       "      <td>Nocardioidaceae</td>\n",
       "      <td>Nocardioides_A_392796</td>\n",
       "      <td>Nocardioides_A_392796 hankookensis</td>\n",
       "      <td>0.5</td>\n",
       "      <td>LIC122</td>\n",
       "      <td>Long Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12282</th>\n",
       "      <td>LIC028</td>\n",
       "      <td>plant</td>\n",
       "      <td>t03</td>\n",
       "      <td>11/14/23</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_01</td>\n",
       "      <td>13</td>\n",
       "      <td>LIC028</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>Actinomycetota</td>\n",
       "      <td>Actinomycetes</td>\n",
       "      <td>Mycobacteriales</td>\n",
       "      <td>Geodermatophilaceae</td>\n",
       "      <td>Modestobacter</td>\n",
       "      <td>Modestobacter muralis</td>\n",
       "      <td>0.5</td>\n",
       "      <td>LIC028</td>\n",
       "      <td>Long Term</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12283</th>\n",
       "      <td>LIC631</td>\n",
       "      <td>plant</td>\n",
       "      <td>c2_t05</td>\n",
       "      <td>3/2/24</td>\n",
       "      <td>12:00</td>\n",
       "      <td>LICCIRC_03</td>\n",
       "      <td>122</td>\n",
       "      <td>LIC631</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Caulobacterales</td>\n",
       "      <td>Caulobacteraceae</td>\n",
       "      <td>Brevundimonas</td>\n",
       "      <td>Brevundimonas subvibrioides</td>\n",
       "      <td>0.5</td>\n",
       "      <td>LIC631</td>\n",
       "      <td>Circadian Experiment 2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12284</th>\n",
       "      <td>LIC629</td>\n",
       "      <td>plant</td>\n",
       "      <td>c2_t05</td>\n",
       "      <td>3/2/24</td>\n",
       "      <td>12:00</td>\n",
       "      <td>LICCIRC_03</td>\n",
       "      <td>122</td>\n",
       "      <td>LIC629</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Rhizobiales_505101</td>\n",
       "      <td>Rhizobiaceae</td>\n",
       "      <td>Agrobacterium</td>\n",
       "      <td>Agrobacterium larrymoorei</td>\n",
       "      <td>0.5</td>\n",
       "      <td>LIC629</td>\n",
       "      <td>Circadian Experiment 2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12285</th>\n",
       "      <td>LIC230</td>\n",
       "      <td>plant</td>\n",
       "      <td>t20</td>\n",
       "      <td>2/16/24</td>\n",
       "      <td>8:00</td>\n",
       "      <td>LIC_03</td>\n",
       "      <td>107</td>\n",
       "      <td>LIC230</td>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Rhizobiales_505101</td>\n",
       "      <td>Beijerinckiaceae</td>\n",
       "      <td>Methylobacterium</td>\n",
       "      <td>Methylobacterium sp001424705</td>\n",
       "      <td>0.5</td>\n",
       "      <td>LIC230</td>\n",
       "      <td>Long Term</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12286 rows × 18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Sample sample.type timepoint      date   time   platename  \\\n",
       "0      LIC522       plant    c1_t11   2/11/24  12:00  LICCIRC_02   \n",
       "1      LIC063       plant       t06  11/28/23   8:00      LIC_01   \n",
       "2      LIC159       plant       t14   1/23/24   8:00      LIC_02   \n",
       "3      LIC452       plant    c1_t06   2/10/24  16:00  LICCIRC_01   \n",
       "4      LIC306       plant       t26    3/8/24   8:00      LIC_04   \n",
       "...       ...         ...       ...       ...    ...         ...   \n",
       "12281  LIC122       plant       t11  12/15/23   8:00      LIC_02   \n",
       "12282  LIC028       plant       t03  11/14/23   8:00      LIC_01   \n",
       "12283  LIC631       plant    c2_t05    3/2/24  12:00  LICCIRC_03   \n",
       "12284  LIC629       plant    c2_t05    3/2/24  12:00  LICCIRC_03   \n",
       "12285  LIC230       plant       t20   2/16/24   8:00      LIC_03   \n",
       "\n",
       "       daysincestart plantID      Kingdom           Phylum  \\\n",
       "0                102  LIC522  d__Bacteria   Pseudomonadota   \n",
       "1                 27  LIC063  d__Bacteria   Pseudomonadota   \n",
       "2                 83  LIC159  d__Bacteria  Cyanobacteriota   \n",
       "3                101  LIC452  d__Bacteria   Pseudomonadota   \n",
       "4                128  LIC306  d__Bacteria   Pseudomonadota   \n",
       "...              ...     ...          ...              ...   \n",
       "12281             44  LIC122  d__Bacteria   Actinomycetota   \n",
       "12282             13  LIC028  d__Bacteria   Actinomycetota   \n",
       "12283            122  LIC631  d__Bacteria   Pseudomonadota   \n",
       "12284            122  LIC629  d__Bacteria   Pseudomonadota   \n",
       "12285            107  LIC230  d__Bacteria   Pseudomonadota   \n",
       "\n",
       "                     Class                     Order  \\\n",
       "0      Gammaproteobacteria  Pseudomonadales_A_650611   \n",
       "1      Alphaproteobacteria          Sphingomonadales   \n",
       "2           Cyanobacteriia          Cyanobacteriales   \n",
       "3      Gammaproteobacteria           Burkholderiales   \n",
       "4      Alphaproteobacteria          Sphingomonadales   \n",
       "...                    ...                       ...   \n",
       "12281        Actinomycetes       Propionibacteriales   \n",
       "12282        Actinomycetes           Mycobacteriales   \n",
       "12283  Alphaproteobacteria           Caulobacterales   \n",
       "12284  Alphaproteobacteria        Rhizobiales_505101   \n",
       "12285  Alphaproteobacteria        Rhizobiales_505101   \n",
       "\n",
       "                          Family                  Genus  \\\n",
       "0               Pseudomonadaceae   Pseudomonas_E_647464   \n",
       "1       Sphingomonadaceae_486827  Sphingomonas_L_486704   \n",
       "2             Coleofasciculaceae                Caldora   \n",
       "3      Burkholderiaceae_A_595421        Massilia_574544   \n",
       "4       Sphingomonadaceae_486827  Sphingomonas_L_486704   \n",
       "...                          ...                    ...   \n",
       "12281            Nocardioidaceae  Nocardioides_A_392796   \n",
       "12282        Geodermatophilaceae          Modestobacter   \n",
       "12283           Caulobacteraceae          Brevundimonas   \n",
       "12284               Rhizobiaceae          Agrobacterium   \n",
       "12285           Beijerinckiaceae       Methylobacterium   \n",
       "\n",
       "                                  Species  AbundR100  sampID  \\\n",
       "0        Pseudomonas_E_647464 viridiflava       79.1  LIC522   \n",
       "1       Sphingomonas_L_486704 sp000786205       66.1  LIC063   \n",
       "2                     Caldora sp010672925       60.5  LIC159   \n",
       "3                   Massilia atriviolacea       59.8  LIC452   \n",
       "4       Sphingomonas_L_486704 sp000786205       57.2  LIC306   \n",
       "...                                   ...        ...     ...   \n",
       "12281  Nocardioides_A_392796 hankookensis        0.5  LIC122   \n",
       "12282               Modestobacter muralis        0.5  LIC028   \n",
       "12283         Brevundimonas subvibrioides        0.5  LIC631   \n",
       "12284           Agrobacterium larrymoorei        0.5  LIC629   \n",
       "12285        Methylobacterium sp001424705        0.5  LIC230   \n",
       "\n",
       "              Experiment Type  \n",
       "0      Circadian Experiment 1  \n",
       "1                   Long Term  \n",
       "2                   Long Term  \n",
       "3      Circadian Experiment 1  \n",
       "4                   Long Term  \n",
       "...                       ...  \n",
       "12281               Long Term  \n",
       "12282               Long Term  \n",
       "12283  Circadian Experiment 2  \n",
       "12284  Circadian Experiment 2  \n",
       "12285               Long Term  \n",
       "\n",
       "[12286 rows x 18 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_microbiome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "cb786bf2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample</th>\n",
       "      <th>Genus</th>\n",
       "      <th>AbundR100</th>\n",
       "      <th>sampID</th>\n",
       "      <th>timepoint</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LIC001</td>\n",
       "      <td>Actinomycetospora</td>\n",
       "      <td>0.6</td>\n",
       "      <td>LIC001</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>LIC001</td>\n",
       "      <td>Actinoplanes</td>\n",
       "      <td>1.1</td>\n",
       "      <td>LIC001</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LIC001</td>\n",
       "      <td>Aeromicrobium</td>\n",
       "      <td>0.8</td>\n",
       "      <td>LIC001</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LIC001</td>\n",
       "      <td>Blastococcus</td>\n",
       "      <td>1.7</td>\n",
       "      <td>LIC001</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LIC001</td>\n",
       "      <td>Caulobacter_487784</td>\n",
       "      <td>0.5</td>\n",
       "      <td>LIC001</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5098</th>\n",
       "      <td>LIC368</td>\n",
       "      <td>Pseudoduganella_570666</td>\n",
       "      <td>0.5</td>\n",
       "      <td>LIC368</td>\n",
       "      <td>t08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5099</th>\n",
       "      <td>LIC368</td>\n",
       "      <td>Pseudomonas_E_647464</td>\n",
       "      <td>2.6</td>\n",
       "      <td>LIC368</td>\n",
       "      <td>t08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5100</th>\n",
       "      <td>LIC368</td>\n",
       "      <td>Rhodoferax_C</td>\n",
       "      <td>6.6</td>\n",
       "      <td>LIC368</td>\n",
       "      <td>t08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5101</th>\n",
       "      <td>LIC368</td>\n",
       "      <td>Sphingomonas_L_486704</td>\n",
       "      <td>19.7</td>\n",
       "      <td>LIC368</td>\n",
       "      <td>t08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5102</th>\n",
       "      <td>LIC368</td>\n",
       "      <td>Variovorax</td>\n",
       "      <td>1.3</td>\n",
       "      <td>LIC368</td>\n",
       "      <td>t08</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5103 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Sample                   Genus  AbundR100  sampID timepoint\n",
       "0     LIC001       Actinomycetospora        0.6  LIC001       t01\n",
       "1     LIC001            Actinoplanes        1.1  LIC001       t01\n",
       "2     LIC001           Aeromicrobium        0.8  LIC001       t01\n",
       "3     LIC001            Blastococcus        1.7  LIC001       t01\n",
       "4     LIC001      Caulobacter_487784        0.5  LIC001       t01\n",
       "...      ...                     ...        ...     ...       ...\n",
       "5098  LIC368  Pseudoduganella_570666        0.5  LIC368       t08\n",
       "5099  LIC368    Pseudomonas_E_647464        2.6  LIC368       t08\n",
       "5100  LIC368            Rhodoferax_C        6.6  LIC368       t08\n",
       "5101  LIC368   Sphingomonas_L_486704       19.7  LIC368       t08\n",
       "5102  LIC368              Variovorax        1.3  LIC368       t08\n",
       "\n",
       "[5103 rows x 5 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genus_sum = (\n",
    "    long_term_microbiome[[\"Sample\", \"Genus\", \"AbundR100\"]]\n",
    "    .groupby([\"Sample\", \"Genus\"])\n",
    "    .sum()\n",
    ")\n",
    "genus_sum = genus_sum.reset_index()\n",
    "\n",
    "genus_sum_w_tp = genus_sum.merge(\n",
    "    metadata[[\"sampID\", \"timepoint\"]], left_on=\"Sample\", right_on=\"sampID\"\n",
    ")\n",
    "genus_sum_w_tp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "6c6227a7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index([], dtype='object', name='Sample')"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genus_sum.set_index(\n",
    "    \"Sample\",\n",
    "    drop=True,\n",
    ").index.difference(\n",
    "    genus_sum_w_tp.set_index(\"Sample\", drop=True).index\n",
    ")  ## Difference is from the dropped samples (due to low quality RNA) which we dont' wanna drop here cause its DNA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "cc7df394",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample</th>\n",
       "      <th>Genus</th>\n",
       "      <th>AbundR100</th>\n",
       "      <th>sampID</th>\n",
       "      <th>timepoint</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LIC001</td>\n",
       "      <td>Actinomycetospora</td>\n",
       "      <td>0.6</td>\n",
       "      <td>LIC001</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>LIC001</td>\n",
       "      <td>Actinoplanes</td>\n",
       "      <td>1.1</td>\n",
       "      <td>LIC001</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LIC001</td>\n",
       "      <td>Aeromicrobium</td>\n",
       "      <td>0.8</td>\n",
       "      <td>LIC001</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LIC001</td>\n",
       "      <td>Blastococcus</td>\n",
       "      <td>1.7</td>\n",
       "      <td>LIC001</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LIC001</td>\n",
       "      <td>Caulobacter_487784</td>\n",
       "      <td>0.5</td>\n",
       "      <td>LIC001</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>LIC001</td>\n",
       "      <td>Chryseobacterium_A_796612</td>\n",
       "      <td>0.6</td>\n",
       "      <td>LIC001</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>LIC001</td>\n",
       "      <td>Curtobacterium</td>\n",
       "      <td>1.0</td>\n",
       "      <td>LIC001</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>LIC001</td>\n",
       "      <td>Dactylopiibacterium</td>\n",
       "      <td>0.7</td>\n",
       "      <td>LIC001</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>LIC001</td>\n",
       "      <td>Flavobacterium</td>\n",
       "      <td>0.8</td>\n",
       "      <td>LIC001</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>LIC001</td>\n",
       "      <td>Friedmanniella</td>\n",
       "      <td>2.7</td>\n",
       "      <td>LIC001</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>LIC001</td>\n",
       "      <td>Frigoribacterium</td>\n",
       "      <td>0.5</td>\n",
       "      <td>LIC001</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>LIC001</td>\n",
       "      <td>Geodermatophilus</td>\n",
       "      <td>0.6</td>\n",
       "      <td>LIC001</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>LIC001</td>\n",
       "      <td>Herbaspirillum</td>\n",
       "      <td>0.6</td>\n",
       "      <td>LIC001</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>LIC001</td>\n",
       "      <td>Hylemonella</td>\n",
       "      <td>0.8</td>\n",
       "      <td>LIC001</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>LIC001</td>\n",
       "      <td>Kineococcus</td>\n",
       "      <td>1.9</td>\n",
       "      <td>LIC001</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>LIC001</td>\n",
       "      <td>Kineosporia</td>\n",
       "      <td>0.6</td>\n",
       "      <td>LIC001</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>LIC001</td>\n",
       "      <td>Klenkia</td>\n",
       "      <td>3.3</td>\n",
       "      <td>LIC001</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>LIC001</td>\n",
       "      <td>Kordiimonas</td>\n",
       "      <td>3.5</td>\n",
       "      <td>LIC001</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>LIC001</td>\n",
       "      <td>Massilia_574544</td>\n",
       "      <td>1.1</td>\n",
       "      <td>LIC001</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>LIC001</td>\n",
       "      <td>Methylobacterium</td>\n",
       "      <td>3.2</td>\n",
       "      <td>LIC001</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>LIC001</td>\n",
       "      <td>Methylophilus</td>\n",
       "      <td>0.6</td>\n",
       "      <td>LIC001</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>LIC001</td>\n",
       "      <td>Microbacterium_A_383321</td>\n",
       "      <td>1.1</td>\n",
       "      <td>LIC001</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>LIC001</td>\n",
       "      <td>Nakamurella</td>\n",
       "      <td>1.3</td>\n",
       "      <td>LIC001</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>LIC001</td>\n",
       "      <td>Neorhizobium_500197</td>\n",
       "      <td>2.1</td>\n",
       "      <td>LIC001</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>LIC001</td>\n",
       "      <td>Nocardioides_A_392796</td>\n",
       "      <td>4.3</td>\n",
       "      <td>LIC001</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>LIC001</td>\n",
       "      <td>Pseudoduganella_570666</td>\n",
       "      <td>2.4</td>\n",
       "      <td>LIC001</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>LIC001</td>\n",
       "      <td>Pseudomonas_B_650453</td>\n",
       "      <td>1.1</td>\n",
       "      <td>LIC001</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>LIC001</td>\n",
       "      <td>Pseudomonas_E_647464</td>\n",
       "      <td>1.6</td>\n",
       "      <td>LIC001</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>LIC001</td>\n",
       "      <td>Pseudomonas_E_647626</td>\n",
       "      <td>0.9</td>\n",
       "      <td>LIC001</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>LIC001</td>\n",
       "      <td>Rhodoferax_C</td>\n",
       "      <td>5.0</td>\n",
       "      <td>LIC001</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>LIC001</td>\n",
       "      <td>Solirubrobacter</td>\n",
       "      <td>0.7</td>\n",
       "      <td>LIC001</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>LIC001</td>\n",
       "      <td>Sphingomonas_L_486704</td>\n",
       "      <td>33.0</td>\n",
       "      <td>LIC001</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>LIC001</td>\n",
       "      <td>Telluria_573210</td>\n",
       "      <td>0.9</td>\n",
       "      <td>LIC001</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>LIC001</td>\n",
       "      <td>Variovorax</td>\n",
       "      <td>0.7</td>\n",
       "      <td>LIC001</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>LIC002</td>\n",
       "      <td>Arthrobacter_E_385446</td>\n",
       "      <td>0.8</td>\n",
       "      <td>LIC002</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>LIC002</td>\n",
       "      <td>Arthrobacter_E_385777</td>\n",
       "      <td>2.3</td>\n",
       "      <td>LIC002</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>LIC002</td>\n",
       "      <td>Blastococcus</td>\n",
       "      <td>3.1</td>\n",
       "      <td>LIC002</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>LIC002</td>\n",
       "      <td>Caballeronia</td>\n",
       "      <td>1.6</td>\n",
       "      <td>LIC002</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>LIC002</td>\n",
       "      <td>Caulobacter_487784</td>\n",
       "      <td>0.5</td>\n",
       "      <td>LIC002</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>LIC002</td>\n",
       "      <td>Curtobacterium</td>\n",
       "      <td>0.6</td>\n",
       "      <td>LIC002</td>\n",
       "      <td>t01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Sample                      Genus  AbundR100  sampID timepoint\n",
       "0   LIC001          Actinomycetospora        0.6  LIC001       t01\n",
       "1   LIC001               Actinoplanes        1.1  LIC001       t01\n",
       "2   LIC001              Aeromicrobium        0.8  LIC001       t01\n",
       "3   LIC001               Blastococcus        1.7  LIC001       t01\n",
       "4   LIC001         Caulobacter_487784        0.5  LIC001       t01\n",
       "5   LIC001  Chryseobacterium_A_796612        0.6  LIC001       t01\n",
       "6   LIC001             Curtobacterium        1.0  LIC001       t01\n",
       "7   LIC001        Dactylopiibacterium        0.7  LIC001       t01\n",
       "8   LIC001             Flavobacterium        0.8  LIC001       t01\n",
       "9   LIC001             Friedmanniella        2.7  LIC001       t01\n",
       "10  LIC001           Frigoribacterium        0.5  LIC001       t01\n",
       "11  LIC001           Geodermatophilus        0.6  LIC001       t01\n",
       "12  LIC001             Herbaspirillum        0.6  LIC001       t01\n",
       "13  LIC001                Hylemonella        0.8  LIC001       t01\n",
       "14  LIC001                Kineococcus        1.9  LIC001       t01\n",
       "15  LIC001                Kineosporia        0.6  LIC001       t01\n",
       "16  LIC001                    Klenkia        3.3  LIC001       t01\n",
       "17  LIC001                Kordiimonas        3.5  LIC001       t01\n",
       "18  LIC001            Massilia_574544        1.1  LIC001       t01\n",
       "19  LIC001           Methylobacterium        3.2  LIC001       t01\n",
       "20  LIC001              Methylophilus        0.6  LIC001       t01\n",
       "21  LIC001    Microbacterium_A_383321        1.1  LIC001       t01\n",
       "22  LIC001                Nakamurella        1.3  LIC001       t01\n",
       "23  LIC001        Neorhizobium_500197        2.1  LIC001       t01\n",
       "24  LIC001      Nocardioides_A_392796        4.3  LIC001       t01\n",
       "25  LIC001     Pseudoduganella_570666        2.4  LIC001       t01\n",
       "26  LIC001       Pseudomonas_B_650453        1.1  LIC001       t01\n",
       "27  LIC001       Pseudomonas_E_647464        1.6  LIC001       t01\n",
       "28  LIC001       Pseudomonas_E_647626        0.9  LIC001       t01\n",
       "29  LIC001               Rhodoferax_C        5.0  LIC001       t01\n",
       "30  LIC001            Solirubrobacter        0.7  LIC001       t01\n",
       "31  LIC001      Sphingomonas_L_486704       33.0  LIC001       t01\n",
       "32  LIC001            Telluria_573210        0.9  LIC001       t01\n",
       "33  LIC001                 Variovorax        0.7  LIC001       t01\n",
       "34  LIC002      Arthrobacter_E_385446        0.8  LIC002       t01\n",
       "35  LIC002      Arthrobacter_E_385777        2.3  LIC002       t01\n",
       "36  LIC002               Blastococcus        3.1  LIC002       t01\n",
       "37  LIC002               Caballeronia        1.6  LIC002       t01\n",
       "38  LIC002         Caulobacter_487784        0.5  LIC002       t01\n",
       "39  LIC002             Curtobacterium        0.6  LIC002       t01"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genus_sum_w_tp.head(40)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "28fcd035",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Actinomycetospora', 'Actinoplanes', 'Aeromicrobium',\n",
       "       'Blastococcus', 'Caulobacter_487784', 'Chryseobacterium_A_796612',\n",
       "       'Curtobacterium', 'Dactylopiibacterium', 'Flavobacterium',\n",
       "       'Friedmanniella', 'Frigoribacterium', 'Geodermatophilus',\n",
       "       'Herbaspirillum', 'Hylemonella', 'Kineococcus', 'Kineosporia',\n",
       "       'Klenkia', 'Kordiimonas', 'Massilia_574544', 'Methylobacterium',\n",
       "       'Methylophilus', 'Microbacterium_A_383321', 'Nakamurella',\n",
       "       'Neorhizobium_500197', 'Nocardioides_A_392796',\n",
       "       'Pseudoduganella_570666', 'Pseudomonas_B_650453',\n",
       "       'Pseudomonas_E_647464', 'Pseudomonas_E_647626', 'Rhodoferax_C',\n",
       "       'Solirubrobacter', 'Sphingomonas_L_486704', 'Telluria_573210',\n",
       "       'Variovorax', 'Arthrobacter_E_385446', 'Arthrobacter_E_385777',\n",
       "       'Caballeronia', 'Marmoricola_A_392027', 'Massilia_574424',\n",
       "       'Microlunatus_B', 'Nocardioides_A_392805',\n",
       "       'Noviherbaspirillum_A_568106', 'Paucimonas', 'Pedococcus',\n",
       "       'Pseudoduganella_570662', 'Pseudoduganella_570691',\n",
       "       'Rhizobium_C_501058', 'Rhodoglobus', 'Rugamonas_570800',\n",
       "       'Aquabacterium_A_592457', 'Duganella_570917',\n",
       "       'Janthinobacterium_571130', 'Janthinobacterium_571526',\n",
       "       'Pelomonas', 'Aureimonas_A_501549', 'Devosia_A_502124',\n",
       "       'Duganella_570715', 'Neorhizobium_500218', 'Pigmentiphaga',\n",
       "       'Amnibacterium_382409', 'Buchnera', 'Ramlibacter_588642',\n",
       "       'Brevundimonas', 'Alsobacter', 'Rathayibacter',\n",
       "       'Herbiconiux_382276', 'Actinocorallia', 'Agrobacterium',\n",
       "       'Amycolatopsis_D_380379', 'Asticcacaulis', 'Cellvibrio',\n",
       "       'Herminiimonas_567389', 'Massilia_573596', 'Curvibacter_587305',\n",
       "       'Modestobacter', 'Motilibacter_A', 'Pseudonocardia',\n",
       "       'Methylotenera_A_557637', 'Mycobacterium', 'Cryptosporangium',\n",
       "       'Caldora', 'Williamsia_A', 'Aureimonas_A_501548',\n",
       "       'Marihabitans_390501', 'Noviherbaspirillum_A_568104',\n",
       "       'Terracoccus', 'DSM-44207', 'UBA5216', 'Homoserinimonas_381256',\n",
       "       'Bosea', 'Actinoallomurus', 'Baekduia', 'Belnapia',\n",
       "       'Micromonospora_H_372070', 'Phenylobacterium', 'Gryllotalpicola',\n",
       "       'Agrococcus', 'Pseudorhodoferax', 'Patulibacter', 'Aetherobacter',\n",
       "       'Tardiphaga', 'Virgisporangium', 'Noviherbaspirillum_A_569121',\n",
       "       'Pseudomonas_M', 'Labedella', 'Roseateles_592282', 'UBA1952',\n",
       "       'Bradyrhizobium_503372', 'Aquabacterium_A_591266',\n",
       "       'Roseomonas_507234', 'Inhella', 'Agreia', 'Abditibacterium',\n",
       "       'Granulicoccus', 'Rhizobacter_590950', 'Aldersonia',\n",
       "       'Xenophilus_587940', 'Paraburkholderia_580243',\n",
       "       'Streptomyces_400150', 'Pseudokineococcus', 'Naasia_381491',\n",
       "       'Arsenicicoccus', 'Ga0074137', 'Labilithrix',\n",
       "       'Chryseobacterium_A_795111', 'Chioneia', 'Sphingomonas_R_483406',\n",
       "       'Massilia_573446', 'Mesorhizobium_I', 'Umezawaea',\n",
       "       'Croceibacterium_484881', 'Rhizorhabdus', 'Polaromonas',\n",
       "       'Jatrophihabitans_A_372606', 'UBA4416', 'Allorhizobium_500053',\n",
       "       'Herbiconiux_382278'], dtype=object)"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_present_genus = genus_sum_w_tp[\"Genus\"].unique()\n",
    "all_present_genus"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "c7606cb2",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Need to fill in missing genus in every sample first\n",
    "for sampID in genus_sum_w_tp[\"sampID\"].unique():\n",
    "    sampid_only_df = genus_sum_w_tp.loc[genus_sum_w_tp[\"sampID\"] == sampID]\n",
    "    present_genus_in_samp = sampid_only_df[\"Genus\"].unique()\n",
    "    difference_in_present_bacteria = list(\n",
    "        set(all_present_genus) - set(present_genus_in_samp)\n",
    "    )\n",
    "    current_time_point = sampid_only_df[\"timepoint\"].iloc[0]\n",
    "    empty_df_to_add = pd.DataFrame(\n",
    "        columns=[\"Sample\", \"Genus\", \"AbundR100\", \"sampID\", \"timepoint\"]\n",
    "    )\n",
    "    empty_df_to_add[\"Genus\"] = difference_in_present_bacteria\n",
    "    empty_df_to_add[\"Sample\"] = sampID\n",
    "    empty_df_to_add[\"AbundR100\"] = 0\n",
    "    empty_df_to_add[\"sampID\"] = sampID\n",
    "    empty_df_to_add[\"timepoint\"] = current_time_point\n",
    "    genus_sum_w_tp = pd.concat([genus_sum_w_tp, empty_df_to_add])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "7e0e31eb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Genus</th>\n",
       "      <th>timepoint</th>\n",
       "      <th>AbundR100</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Abditibacterium</td>\n",
       "      <td>t01</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Abditibacterium</td>\n",
       "      <td>t02</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Abditibacterium</td>\n",
       "      <td>t03</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Abditibacterium</td>\n",
       "      <td>t04</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Abditibacterium</td>\n",
       "      <td>t05</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3831</th>\n",
       "      <td>Xenophilus_587940</td>\n",
       "      <td>t24</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3832</th>\n",
       "      <td>Xenophilus_587940</td>\n",
       "      <td>t25</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3833</th>\n",
       "      <td>Xenophilus_587940</td>\n",
       "      <td>t26</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3834</th>\n",
       "      <td>Xenophilus_587940</td>\n",
       "      <td>t27</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3835</th>\n",
       "      <td>Xenophilus_587940</td>\n",
       "      <td>t28</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3836 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Genus timepoint  AbundR100\n",
       "0       Abditibacterium       t01        0.0\n",
       "1       Abditibacterium       t02        0.0\n",
       "2       Abditibacterium       t03        0.0\n",
       "3       Abditibacterium       t04        0.0\n",
       "4       Abditibacterium       t05        0.0\n",
       "...                 ...       ...        ...\n",
       "3831  Xenophilus_587940       t24        0.0\n",
       "3832  Xenophilus_587940       t25        0.0\n",
       "3833  Xenophilus_587940       t26        0.0\n",
       "3834  Xenophilus_587940       t27        0.0\n",
       "3835  Xenophilus_587940       t28        0.0\n",
       "\n",
       "[3836 rows x 3 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "timepoint_average_abundance = (\n",
    "    genus_sum_w_tp[[\"Genus\", \"timepoint\", \"AbundR100\"]]\n",
    "    .groupby([\"Genus\", \"timepoint\"])\n",
    "    .mean()\n",
    ")\n",
    "timepoint_average_abundance = timepoint_average_abundance.reset_index()\n",
    "timepoint_average_abundance"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "e1e253cc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AT1G01010</th>\n",
       "      <th>AT1G01020</th>\n",
       "      <th>AT1G03987</th>\n",
       "      <th>AT1G01030</th>\n",
       "      <th>AT1G01040</th>\n",
       "      <th>AT1G01046</th>\n",
       "      <th>AT1G01050</th>\n",
       "      <th>AT1G01060</th>\n",
       "      <th>AT1G01070</th>\n",
       "      <th>AT1G04003</th>\n",
       "      <th>...</th>\n",
       "      <th>ArthCr090</th>\n",
       "      <th>ArthCr089</th>\n",
       "      <th>ArthCt114</th>\n",
       "      <th>ArthCt124</th>\n",
       "      <th>ArthCr088</th>\n",
       "      <th>ArthCt112</th>\n",
       "      <th>ArthCp088</th>\n",
       "      <th>ArthCp083</th>\n",
       "      <th>ArthCp084</th>\n",
       "      <th>ArthCp085</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>timepoint</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>t01</th>\n",
       "      <td>4.178007</td>\n",
       "      <td>5.437430</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.605825</td>\n",
       "      <td>4.439901</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>50.977363</td>\n",
       "      <td>13.004802</td>\n",
       "      <td>1.589169</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>43.520549</td>\n",
       "      <td>122.823875</td>\n",
       "      <td>8.980496</td>\n",
       "      <td>6.651848</td>\n",
       "      <td>21.939469</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.161073</td>\n",
       "      <td>0.267902</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.965547</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t02</th>\n",
       "      <td>0.940355</td>\n",
       "      <td>7.298013</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.842908</td>\n",
       "      <td>4.463269</td>\n",
       "      <td>5.166150</td>\n",
       "      <td>25.168823</td>\n",
       "      <td>47.655766</td>\n",
       "      <td>0.571242</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>6.751200</td>\n",
       "      <td>130.786877</td>\n",
       "      <td>4.160837</td>\n",
       "      <td>0.244885</td>\n",
       "      <td>21.553952</td>\n",
       "      <td>3.428396</td>\n",
       "      <td>0.037675</td>\n",
       "      <td>0.238550</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.325997</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t03</th>\n",
       "      <td>1.783955</td>\n",
       "      <td>7.880381</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.144576</td>\n",
       "      <td>4.227300</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>46.778572</td>\n",
       "      <td>28.141387</td>\n",
       "      <td>3.471598</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>10.900785</td>\n",
       "      <td>52.858647</td>\n",
       "      <td>5.848172</td>\n",
       "      <td>1.035596</td>\n",
       "      <td>6.500587</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.156090</td>\n",
       "      <td>0.197246</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.972086</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t04</th>\n",
       "      <td>3.294595</td>\n",
       "      <td>7.674816</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>5.922081</td>\n",
       "      <td>5.006461</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>28.959007</td>\n",
       "      <td>39.972919</td>\n",
       "      <td>1.555315</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>6.978923</td>\n",
       "      <td>80.358859</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.078758</td>\n",
       "      <td>19.860159</td>\n",
       "      <td>4.717444</td>\n",
       "      <td>4.191593</td>\n",
       "      <td>0.176193</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.531536</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t05</th>\n",
       "      <td>1.695050</td>\n",
       "      <td>6.069584</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.895225</td>\n",
       "      <td>3.362750</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>40.021234</td>\n",
       "      <td>10.213291</td>\n",
       "      <td>0.867596</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>19.211318</td>\n",
       "      <td>49.267266</td>\n",
       "      <td>2.607296</td>\n",
       "      <td>3.691827</td>\n",
       "      <td>10.404960</td>\n",
       "      <td>8.562370</td>\n",
       "      <td>1.142433</td>\n",
       "      <td>0.176383</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.769217</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t06</th>\n",
       "      <td>0.817885</td>\n",
       "      <td>5.963264</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.836390</td>\n",
       "      <td>2.420270</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>34.425932</td>\n",
       "      <td>8.355582</td>\n",
       "      <td>2.112976</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>13.454872</td>\n",
       "      <td>29.321538</td>\n",
       "      <td>13.028191</td>\n",
       "      <td>7.357152</td>\n",
       "      <td>5.552243</td>\n",
       "      <td>1.294320</td>\n",
       "      <td>3.633794</td>\n",
       "      <td>0.327666</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.578309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t07</th>\n",
       "      <td>1.108656</td>\n",
       "      <td>6.329721</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.828056</td>\n",
       "      <td>3.699847</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>47.950309</td>\n",
       "      <td>6.811390</td>\n",
       "      <td>1.480485</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>12.268870</td>\n",
       "      <td>39.640506</td>\n",
       "      <td>4.836168</td>\n",
       "      <td>4.843796</td>\n",
       "      <td>7.371363</td>\n",
       "      <td>1.893213</td>\n",
       "      <td>3.312303</td>\n",
       "      <td>0.351533</td>\n",
       "      <td>0.471103</td>\n",
       "      <td>2.210424</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t08</th>\n",
       "      <td>0.432387</td>\n",
       "      <td>4.034252</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.315234</td>\n",
       "      <td>2.412136</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>37.309378</td>\n",
       "      <td>12.520038</td>\n",
       "      <td>0.617279</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>18.927096</td>\n",
       "      <td>76.691200</td>\n",
       "      <td>10.530133</td>\n",
       "      <td>7.631265</td>\n",
       "      <td>6.334308</td>\n",
       "      <td>3.134487</td>\n",
       "      <td>4.265254</td>\n",
       "      <td>0.406276</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.934250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t09</th>\n",
       "      <td>0.892427</td>\n",
       "      <td>6.731354</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.876925</td>\n",
       "      <td>2.804225</td>\n",
       "      <td>0.660741</td>\n",
       "      <td>38.215024</td>\n",
       "      <td>4.794368</td>\n",
       "      <td>1.310240</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>8.966486</td>\n",
       "      <td>36.647809</td>\n",
       "      <td>11.555561</td>\n",
       "      <td>10.159008</td>\n",
       "      <td>6.338279</td>\n",
       "      <td>2.003623</td>\n",
       "      <td>2.928722</td>\n",
       "      <td>0.324891</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.937635</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t10</th>\n",
       "      <td>0.580167</td>\n",
       "      <td>4.477800</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.854897</td>\n",
       "      <td>2.295173</td>\n",
       "      <td>1.497334</td>\n",
       "      <td>26.936016</td>\n",
       "      <td>9.849358</td>\n",
       "      <td>0.645853</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>8.092580</td>\n",
       "      <td>51.960621</td>\n",
       "      <td>10.901817</td>\n",
       "      <td>18.273275</td>\n",
       "      <td>9.370636</td>\n",
       "      <td>0.543560</td>\n",
       "      <td>2.641585</td>\n",
       "      <td>0.214544</td>\n",
       "      <td>0.506434</td>\n",
       "      <td>2.206808</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t11</th>\n",
       "      <td>1.405294</td>\n",
       "      <td>6.647035</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.888008</td>\n",
       "      <td>3.363988</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>40.135106</td>\n",
       "      <td>6.137138</td>\n",
       "      <td>1.316598</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>7.347965</td>\n",
       "      <td>42.647855</td>\n",
       "      <td>10.125324</td>\n",
       "      <td>7.903202</td>\n",
       "      <td>6.330487</td>\n",
       "      <td>1.751637</td>\n",
       "      <td>3.533760</td>\n",
       "      <td>0.216323</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.640193</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t12</th>\n",
       "      <td>2.093983</td>\n",
       "      <td>5.309525</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.952212</td>\n",
       "      <td>2.458480</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>34.240911</td>\n",
       "      <td>6.331188</td>\n",
       "      <td>0.950306</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>10.026191</td>\n",
       "      <td>52.389935</td>\n",
       "      <td>4.925738</td>\n",
       "      <td>4.856406</td>\n",
       "      <td>6.577449</td>\n",
       "      <td>1.839493</td>\n",
       "      <td>2.473787</td>\n",
       "      <td>0.187691</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.629081</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t13</th>\n",
       "      <td>1.320447</td>\n",
       "      <td>5.672182</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.853176</td>\n",
       "      <td>3.259841</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>39.840651</td>\n",
       "      <td>7.970370</td>\n",
       "      <td>1.643706</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>2.197317</td>\n",
       "      <td>14.739778</td>\n",
       "      <td>5.188332</td>\n",
       "      <td>3.418423</td>\n",
       "      <td>3.251258</td>\n",
       "      <td>1.176304</td>\n",
       "      <td>3.502721</td>\n",
       "      <td>0.232899</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.762472</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t14</th>\n",
       "      <td>1.952456</td>\n",
       "      <td>7.580967</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.260930</td>\n",
       "      <td>3.884446</td>\n",
       "      <td>0.285149</td>\n",
       "      <td>31.003940</td>\n",
       "      <td>4.279266</td>\n",
       "      <td>0.777145</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>35.956481</td>\n",
       "      <td>115.132271</td>\n",
       "      <td>8.577471</td>\n",
       "      <td>18.219991</td>\n",
       "      <td>20.001145</td>\n",
       "      <td>5.330660</td>\n",
       "      <td>2.905501</td>\n",
       "      <td>0.375767</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.965142</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t15</th>\n",
       "      <td>1.026444</td>\n",
       "      <td>6.282623</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.500479</td>\n",
       "      <td>3.724847</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>25.525418</td>\n",
       "      <td>17.855742</td>\n",
       "      <td>1.073219</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>21.167245</td>\n",
       "      <td>129.949292</td>\n",
       "      <td>11.787012</td>\n",
       "      <td>18.141093</td>\n",
       "      <td>17.139890</td>\n",
       "      <td>8.665054</td>\n",
       "      <td>4.959690</td>\n",
       "      <td>0.781063</td>\n",
       "      <td>0.069186</td>\n",
       "      <td>1.396571</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t16</th>\n",
       "      <td>0.368887</td>\n",
       "      <td>5.646004</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.166233</td>\n",
       "      <td>4.197370</td>\n",
       "      <td>0.535615</td>\n",
       "      <td>27.194511</td>\n",
       "      <td>9.875509</td>\n",
       "      <td>1.339578</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>17.773178</td>\n",
       "      <td>140.333699</td>\n",
       "      <td>3.548782</td>\n",
       "      <td>1.720370</td>\n",
       "      <td>11.651617</td>\n",
       "      <td>1.471990</td>\n",
       "      <td>3.184776</td>\n",
       "      <td>0.361575</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.089132</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t17</th>\n",
       "      <td>0.636169</td>\n",
       "      <td>5.583331</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.254393</td>\n",
       "      <td>2.576151</td>\n",
       "      <td>0.413929</td>\n",
       "      <td>31.241836</td>\n",
       "      <td>13.860416</td>\n",
       "      <td>1.764411</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>29.483188</td>\n",
       "      <td>107.060234</td>\n",
       "      <td>6.964001</td>\n",
       "      <td>18.367253</td>\n",
       "      <td>16.866326</td>\n",
       "      <td>13.836056</td>\n",
       "      <td>4.532371</td>\n",
       "      <td>0.776638</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.136534</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t18</th>\n",
       "      <td>0.816282</td>\n",
       "      <td>5.547303</td>\n",
       "      <td>0.327006</td>\n",
       "      <td>1.706347</td>\n",
       "      <td>3.814218</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>31.384922</td>\n",
       "      <td>5.054087</td>\n",
       "      <td>0.933390</td>\n",
       "      <td>0.514762</td>\n",
       "      <td>...</td>\n",
       "      <td>20.804795</td>\n",
       "      <td>82.273498</td>\n",
       "      <td>19.894355</td>\n",
       "      <td>26.296068</td>\n",
       "      <td>13.502918</td>\n",
       "      <td>4.309940</td>\n",
       "      <td>2.315590</td>\n",
       "      <td>0.647061</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.472653</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t19</th>\n",
       "      <td>1.268809</td>\n",
       "      <td>5.775510</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.559373</td>\n",
       "      <td>2.853177</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>30.101271</td>\n",
       "      <td>13.147662</td>\n",
       "      <td>2.027122</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>33.823267</td>\n",
       "      <td>145.509469</td>\n",
       "      <td>26.657459</td>\n",
       "      <td>22.378787</td>\n",
       "      <td>26.349864</td>\n",
       "      <td>12.905727</td>\n",
       "      <td>3.168036</td>\n",
       "      <td>0.928841</td>\n",
       "      <td>0.584784</td>\n",
       "      <td>1.767190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t20</th>\n",
       "      <td>0.965510</td>\n",
       "      <td>6.532259</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.008219</td>\n",
       "      <td>3.848316</td>\n",
       "      <td>3.015759</td>\n",
       "      <td>22.170258</td>\n",
       "      <td>4.634278</td>\n",
       "      <td>0.415657</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>33.164471</td>\n",
       "      <td>127.163926</td>\n",
       "      <td>27.462534</td>\n",
       "      <td>29.092474</td>\n",
       "      <td>17.349198</td>\n",
       "      <td>8.045448</td>\n",
       "      <td>2.935488</td>\n",
       "      <td>0.567233</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.487746</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t21</th>\n",
       "      <td>1.084696</td>\n",
       "      <td>3.911506</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.699122</td>\n",
       "      <td>2.891112</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>25.822231</td>\n",
       "      <td>3.572692</td>\n",
       "      <td>0.639334</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>57.091646</td>\n",
       "      <td>154.344280</td>\n",
       "      <td>45.056356</td>\n",
       "      <td>40.132013</td>\n",
       "      <td>22.015792</td>\n",
       "      <td>6.301920</td>\n",
       "      <td>2.044977</td>\n",
       "      <td>0.848478</td>\n",
       "      <td>0.024524</td>\n",
       "      <td>1.624038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t22</th>\n",
       "      <td>2.074586</td>\n",
       "      <td>4.915989</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.134428</td>\n",
       "      <td>3.469577</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>19.541986</td>\n",
       "      <td>7.230762</td>\n",
       "      <td>1.176890</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>32.662945</td>\n",
       "      <td>143.946929</td>\n",
       "      <td>12.212214</td>\n",
       "      <td>18.661344</td>\n",
       "      <td>21.263880</td>\n",
       "      <td>3.229432</td>\n",
       "      <td>2.292980</td>\n",
       "      <td>1.036278</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.224413</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t23</th>\n",
       "      <td>0.759788</td>\n",
       "      <td>5.700726</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.045190</td>\n",
       "      <td>3.068678</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>26.869983</td>\n",
       "      <td>5.934330</td>\n",
       "      <td>0.621263</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>19.644578</td>\n",
       "      <td>85.396672</td>\n",
       "      <td>12.445205</td>\n",
       "      <td>18.654071</td>\n",
       "      <td>12.588228</td>\n",
       "      <td>8.295015</td>\n",
       "      <td>2.934011</td>\n",
       "      <td>0.855945</td>\n",
       "      <td>0.146556</td>\n",
       "      <td>1.518470</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t24</th>\n",
       "      <td>1.061980</td>\n",
       "      <td>3.180492</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.134040</td>\n",
       "      <td>3.243189</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>26.702798</td>\n",
       "      <td>8.194852</td>\n",
       "      <td>0.670002</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>16.896719</td>\n",
       "      <td>100.952403</td>\n",
       "      <td>4.064760</td>\n",
       "      <td>23.047104</td>\n",
       "      <td>8.572817</td>\n",
       "      <td>0.647449</td>\n",
       "      <td>4.415334</td>\n",
       "      <td>0.291652</td>\n",
       "      <td>6.277642</td>\n",
       "      <td>3.292149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t25</th>\n",
       "      <td>0.841613</td>\n",
       "      <td>4.586502</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.436027</td>\n",
       "      <td>2.919767</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>32.147860</td>\n",
       "      <td>23.751213</td>\n",
       "      <td>0.855850</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>27.493800</td>\n",
       "      <td>88.629797</td>\n",
       "      <td>2.070804</td>\n",
       "      <td>4.461611</td>\n",
       "      <td>23.691364</td>\n",
       "      <td>5.826791</td>\n",
       "      <td>1.913455</td>\n",
       "      <td>0.567557</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.206079</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t26</th>\n",
       "      <td>0.756034</td>\n",
       "      <td>4.714830</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.658975</td>\n",
       "      <td>3.301417</td>\n",
       "      <td>1.562652</td>\n",
       "      <td>38.277536</td>\n",
       "      <td>31.354376</td>\n",
       "      <td>1.833060</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>21.608976</td>\n",
       "      <td>117.951206</td>\n",
       "      <td>8.304278</td>\n",
       "      <td>8.022113</td>\n",
       "      <td>22.501514</td>\n",
       "      <td>9.979777</td>\n",
       "      <td>3.903073</td>\n",
       "      <td>0.706675</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.846763</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t27</th>\n",
       "      <td>1.493336</td>\n",
       "      <td>5.823960</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.483665</td>\n",
       "      <td>3.625887</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>34.135599</td>\n",
       "      <td>17.104593</td>\n",
       "      <td>1.701320</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>30.278081</td>\n",
       "      <td>140.606941</td>\n",
       "      <td>18.876090</td>\n",
       "      <td>16.233629</td>\n",
       "      <td>27.525178</td>\n",
       "      <td>3.004944</td>\n",
       "      <td>3.044700</td>\n",
       "      <td>0.888145</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.641984</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t28</th>\n",
       "      <td>1.128406</td>\n",
       "      <td>6.908700</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.035635</td>\n",
       "      <td>4.057351</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>43.540276</td>\n",
       "      <td>35.252888</td>\n",
       "      <td>2.155311</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>8.455318</td>\n",
       "      <td>68.205609</td>\n",
       "      <td>8.850996</td>\n",
       "      <td>6.219817</td>\n",
       "      <td>5.079004</td>\n",
       "      <td>5.674576</td>\n",
       "      <td>3.650114</td>\n",
       "      <td>0.330707</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.669698</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>28 rows × 27418 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           AT1G01010  AT1G01020  AT1G03987  AT1G01030  AT1G01040  AT1G01046  \\\n",
       "timepoint                                                                     \n",
       "t01         4.178007   5.437430   0.000000   2.605825   4.439901   0.000000   \n",
       "t02         0.940355   7.298013   0.000000   3.842908   4.463269   5.166150   \n",
       "t03         1.783955   7.880381   0.000000   4.144576   4.227300   0.000000   \n",
       "t04         3.294595   7.674816   0.000000   5.922081   5.006461   0.000000   \n",
       "t05         1.695050   6.069584   0.000000   2.895225   3.362750   0.000000   \n",
       "t06         0.817885   5.963264   0.000000   2.836390   2.420270   0.000000   \n",
       "t07         1.108656   6.329721   0.000000   1.828056   3.699847   0.000000   \n",
       "t08         0.432387   4.034252   0.000000   1.315234   2.412136   0.000000   \n",
       "t09         0.892427   6.731354   0.000000   1.876925   2.804225   0.660741   \n",
       "t10         0.580167   4.477800   0.000000   2.854897   2.295173   1.497334   \n",
       "t11         1.405294   6.647035   0.000000   2.888008   3.363988   0.000000   \n",
       "t12         2.093983   5.309525   0.000000   0.952212   2.458480   0.000000   \n",
       "t13         1.320447   5.672182   0.000000   1.853176   3.259841   0.000000   \n",
       "t14         1.952456   7.580967   0.000000   3.260930   3.884446   0.285149   \n",
       "t15         1.026444   6.282623   0.000000   1.500479   3.724847   0.000000   \n",
       "t16         0.368887   5.646004   0.000000   1.166233   4.197370   0.535615   \n",
       "t17         0.636169   5.583331   0.000000   1.254393   2.576151   0.413929   \n",
       "t18         0.816282   5.547303   0.327006   1.706347   3.814218   0.000000   \n",
       "t19         1.268809   5.775510   0.000000   1.559373   2.853177   0.000000   \n",
       "t20         0.965510   6.532259   0.000000   2.008219   3.848316   3.015759   \n",
       "t21         1.084696   3.911506   0.000000   0.699122   2.891112   0.000000   \n",
       "t22         2.074586   4.915989   0.000000   1.134428   3.469577   0.000000   \n",
       "t23         0.759788   5.700726   0.000000   3.045190   3.068678   0.000000   \n",
       "t24         1.061980   3.180492   0.000000   2.134040   3.243189   0.000000   \n",
       "t25         0.841613   4.586502   0.000000   1.436027   2.919767   0.000000   \n",
       "t26         0.756034   4.714830   0.000000   1.658975   3.301417   1.562652   \n",
       "t27         1.493336   5.823960   0.000000   1.483665   3.625887   0.000000   \n",
       "t28         1.128406   6.908700   0.000000   2.035635   4.057351   0.000000   \n",
       "\n",
       "           AT1G01050  AT1G01060  AT1G01070  AT1G04003  ...  ArthCr090  \\\n",
       "timepoint                                              ...              \n",
       "t01        50.977363  13.004802   1.589169   0.000000  ...  43.520549   \n",
       "t02        25.168823  47.655766   0.571242   0.000000  ...   6.751200   \n",
       "t03        46.778572  28.141387   3.471598   0.000000  ...  10.900785   \n",
       "t04        28.959007  39.972919   1.555315   0.000000  ...   6.978923   \n",
       "t05        40.021234  10.213291   0.867596   0.000000  ...  19.211318   \n",
       "t06        34.425932   8.355582   2.112976   0.000000  ...  13.454872   \n",
       "t07        47.950309   6.811390   1.480485   0.000000  ...  12.268870   \n",
       "t08        37.309378  12.520038   0.617279   0.000000  ...  18.927096   \n",
       "t09        38.215024   4.794368   1.310240   0.000000  ...   8.966486   \n",
       "t10        26.936016   9.849358   0.645853   0.000000  ...   8.092580   \n",
       "t11        40.135106   6.137138   1.316598   0.000000  ...   7.347965   \n",
       "t12        34.240911   6.331188   0.950306   0.000000  ...  10.026191   \n",
       "t13        39.840651   7.970370   1.643706   0.000000  ...   2.197317   \n",
       "t14        31.003940   4.279266   0.777145   0.000000  ...  35.956481   \n",
       "t15        25.525418  17.855742   1.073219   0.000000  ...  21.167245   \n",
       "t16        27.194511   9.875509   1.339578   0.000000  ...  17.773178   \n",
       "t17        31.241836  13.860416   1.764411   0.000000  ...  29.483188   \n",
       "t18        31.384922   5.054087   0.933390   0.514762  ...  20.804795   \n",
       "t19        30.101271  13.147662   2.027122   0.000000  ...  33.823267   \n",
       "t20        22.170258   4.634278   0.415657   0.000000  ...  33.164471   \n",
       "t21        25.822231   3.572692   0.639334   0.000000  ...  57.091646   \n",
       "t22        19.541986   7.230762   1.176890   0.000000  ...  32.662945   \n",
       "t23        26.869983   5.934330   0.621263   0.000000  ...  19.644578   \n",
       "t24        26.702798   8.194852   0.670002   0.000000  ...  16.896719   \n",
       "t25        32.147860  23.751213   0.855850   0.000000  ...  27.493800   \n",
       "t26        38.277536  31.354376   1.833060   0.000000  ...  21.608976   \n",
       "t27        34.135599  17.104593   1.701320   0.000000  ...  30.278081   \n",
       "t28        43.540276  35.252888   2.155311   0.000000  ...   8.455318   \n",
       "\n",
       "            ArthCr089  ArthCt114  ArthCt124  ArthCr088  ArthCt112  ArthCp088  \\\n",
       "timepoint                                                                      \n",
       "t01        122.823875   8.980496   6.651848  21.939469   0.000000   0.161073   \n",
       "t02        130.786877   4.160837   0.244885  21.553952   3.428396   0.037675   \n",
       "t03         52.858647   5.848172   1.035596   6.500587   0.000000   2.156090   \n",
       "t04         80.358859   0.000000   3.078758  19.860159   4.717444   4.191593   \n",
       "t05         49.267266   2.607296   3.691827  10.404960   8.562370   1.142433   \n",
       "t06         29.321538  13.028191   7.357152   5.552243   1.294320   3.633794   \n",
       "t07         39.640506   4.836168   4.843796   7.371363   1.893213   3.312303   \n",
       "t08         76.691200  10.530133   7.631265   6.334308   3.134487   4.265254   \n",
       "t09         36.647809  11.555561  10.159008   6.338279   2.003623   2.928722   \n",
       "t10         51.960621  10.901817  18.273275   9.370636   0.543560   2.641585   \n",
       "t11         42.647855  10.125324   7.903202   6.330487   1.751637   3.533760   \n",
       "t12         52.389935   4.925738   4.856406   6.577449   1.839493   2.473787   \n",
       "t13         14.739778   5.188332   3.418423   3.251258   1.176304   3.502721   \n",
       "t14        115.132271   8.577471  18.219991  20.001145   5.330660   2.905501   \n",
       "t15        129.949292  11.787012  18.141093  17.139890   8.665054   4.959690   \n",
       "t16        140.333699   3.548782   1.720370  11.651617   1.471990   3.184776   \n",
       "t17        107.060234   6.964001  18.367253  16.866326  13.836056   4.532371   \n",
       "t18         82.273498  19.894355  26.296068  13.502918   4.309940   2.315590   \n",
       "t19        145.509469  26.657459  22.378787  26.349864  12.905727   3.168036   \n",
       "t20        127.163926  27.462534  29.092474  17.349198   8.045448   2.935488   \n",
       "t21        154.344280  45.056356  40.132013  22.015792   6.301920   2.044977   \n",
       "t22        143.946929  12.212214  18.661344  21.263880   3.229432   2.292980   \n",
       "t23         85.396672  12.445205  18.654071  12.588228   8.295015   2.934011   \n",
       "t24        100.952403   4.064760  23.047104   8.572817   0.647449   4.415334   \n",
       "t25         88.629797   2.070804   4.461611  23.691364   5.826791   1.913455   \n",
       "t26        117.951206   8.304278   8.022113  22.501514   9.979777   3.903073   \n",
       "t27        140.606941  18.876090  16.233629  27.525178   3.004944   3.044700   \n",
       "t28         68.205609   8.850996   6.219817   5.079004   5.674576   3.650114   \n",
       "\n",
       "           ArthCp083  ArthCp084  ArthCp085  \n",
       "timepoint                                   \n",
       "t01         0.267902   0.000000   0.965547  \n",
       "t02         0.238550   0.000000   1.325997  \n",
       "t03         0.197246   0.000000   0.972086  \n",
       "t04         0.176193   0.000000   2.531536  \n",
       "t05         0.176383   0.000000   1.769217  \n",
       "t06         0.327666   0.000000   1.578309  \n",
       "t07         0.351533   0.471103   2.210424  \n",
       "t08         0.406276   0.000000   2.934250  \n",
       "t09         0.324891   0.000000   1.937635  \n",
       "t10         0.214544   0.506434   2.206808  \n",
       "t11         0.216323   0.000000   1.640193  \n",
       "t12         0.187691   0.000000   1.629081  \n",
       "t13         0.232899   0.000000   1.762472  \n",
       "t14         0.375767   0.000000   1.965142  \n",
       "t15         0.781063   0.069186   1.396571  \n",
       "t16         0.361575   0.000000   2.089132  \n",
       "t17         0.776638   0.000000   2.136534  \n",
       "t18         0.647061   0.000000   1.472653  \n",
       "t19         0.928841   0.584784   1.767190  \n",
       "t20         0.567233   0.000000   1.487746  \n",
       "t21         0.848478   0.024524   1.624038  \n",
       "t22         1.036278   0.000000   2.224413  \n",
       "t23         0.855945   0.146556   1.518470  \n",
       "t24         0.291652   6.277642   3.292149  \n",
       "t25         0.567557   0.000000   2.206079  \n",
       "t26         0.706675   0.000000   1.846763  \n",
       "t27         0.888145   0.000000   1.641984  \n",
       "t28         0.330707   0.000000   3.669698  \n",
       "\n",
       "[28 rows x 27418 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# First, let's aggregate transcriptome data by timepoint (averaging across samples)\n",
    "long_term_transcriptome_w_tp = long_term_transcriptome.copy()\n",
    "long_term_transcriptome_w_tp[\"timepoint\"] = long_term_metadata[\"timepoint\"]\n",
    "\n",
    "# Average gene expression by timepoint\n",
    "transcriptome_by_timepoint = long_term_transcriptome_w_tp.groupby(\"timepoint\").mean()\n",
    "transcriptome_by_timepoint"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "7b56c62d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Available timepoints: ['t01', 't02', 't03', 't04', 't05', 't06', 't07', 't08', 't09', 't10', 't11', 't12', 't13', 't14', 't15', 't16', 't17', 't18', 't19', 't20', 't21', 't22', 't23', 't24', 't25', 't26', 't27', 't28']\n",
      "Number of timepoints: 28\n"
     ]
    }
   ],
   "source": [
    "# Get sorted unique timepoints\n",
    "timepoints = sorted(transcriptome_by_timepoint.index.unique())\n",
    "print(f\"Available timepoints: {timepoints}\")\n",
    "print(f\"Number of timepoints: {len(timepoints)}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "2d2c9c02",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Original transcriptome timepoints: ['t01', 't02', 't03', 't04', 't05', 't06', 't07', 't08', 't09', 't10', 't11', 't12', 't13', 't14', 't15', 't16', 't17', 't18', 't19', 't20', 't21', 't22', 't23', 't24', 't25', 't26', 't27', 't28']\n",
      "Forward shifted timepoints: ['t02', 't03', 't04', 't05', 't06', 't07', 't08', 't09', 't10', 't11', 't12', 't13', 't14', 't15', 't16', 't17', 't18', 't19', 't20', 't21', 't22', 't23', 't24', 't25', 't26', 't27', 't28']\n",
      "Backward shifted timepoints: ['t01', 't02', 't03', 't04', 't05', 't06', 't07', 't08', 't09', 't10', 't11', 't12', 't13', 't14', 't15', 't16', 't17', 't18', 't19', 't20', 't21', 't22', 't23', 't24', 't25', 't26', 't27']\n"
     ]
    }
   ],
   "source": [
    "# Create shifted transcriptome dataframes\n",
    "# Shift forward: transcriptome at t correlates with microbiome at t+1 (transcriptome leads)\n",
    "# Shift backward: transcriptome at t correlates with microbiome at t-1 (microbiome leads)\n",
    "\n",
    "# Create a mapping from original timepoint to shifted timepoint\n",
    "tp_to_idx = {tp: i for i, tp in enumerate(timepoints)}\n",
    "idx_to_tp = {i: tp for i, tp in enumerate(timepoints)}\n",
    "\n",
    "# Shift forward (+1): transcriptome data moved to next timepoint\n",
    "# This tests: does expression at time t correlate with microbiome at time t+1?\n",
    "transcriptome_shifted_forward = transcriptome_by_timepoint.copy()\n",
    "transcriptome_shifted_forward.index = [\n",
    "    idx_to_tp.get(tp_to_idx[tp] + 1, None) for tp in transcriptome_shifted_forward.index\n",
    "]\n",
    "transcriptome_shifted_forward = transcriptome_shifted_forward.dropna(subset=[])\n",
    "transcriptome_shifted_forward = transcriptome_shifted_forward[\n",
    "    transcriptome_shifted_forward.index.notna()\n",
    "]\n",
    "\n",
    "# Shift backward (-1): transcriptome data moved to previous timepoint\n",
    "# This tests: does expression at time t correlate with microbiome at time t-1?\n",
    "transcriptome_shifted_backward = transcriptome_by_timepoint.copy()\n",
    "transcriptome_shifted_backward.index = [\n",
    "    idx_to_tp.get(tp_to_idx[tp] - 1, None)\n",
    "    for tp in transcriptome_shifted_backward.index\n",
    "]\n",
    "transcriptome_shifted_backward = transcriptome_shifted_backward[\n",
    "    transcriptome_shifted_backward.index.notna()\n",
    "]\n",
    "\n",
    "print(f\"Original transcriptome timepoints: {list(transcriptome_by_timepoint.index)}\")\n",
    "print(f\"Forward shifted timepoints: {list(transcriptome_shifted_forward.index)}\")\n",
    "print(f\"Backward shifted timepoints: {list(transcriptome_shifted_backward.index)}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "a514b984",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Genus</th>\n",
       "      <th>Abditibacterium</th>\n",
       "      <th>Actinoallomurus</th>\n",
       "      <th>Actinocorallia</th>\n",
       "      <th>Actinomycetospora</th>\n",
       "      <th>Actinoplanes</th>\n",
       "      <th>Aeromicrobium</th>\n",
       "      <th>Aetherobacter</th>\n",
       "      <th>Agreia</th>\n",
       "      <th>Agrobacterium</th>\n",
       "      <th>Agrococcus</th>\n",
       "      <th>...</th>\n",
       "      <th>Telluria_573210</th>\n",
       "      <th>Terracoccus</th>\n",
       "      <th>UBA1952</th>\n",
       "      <th>UBA4416</th>\n",
       "      <th>UBA5216</th>\n",
       "      <th>Umezawaea</th>\n",
       "      <th>Variovorax</th>\n",
       "      <th>Virgisporangium</th>\n",
       "      <th>Williamsia_A</th>\n",
       "      <th>Xenophilus_587940</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>timepoint</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>t01</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.375000</td>\n",
       "      <td>0.137500</td>\n",
       "      <td>0.7875</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.675000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>2.362500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t02</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0625</td>\n",
       "      <td>0.587500</td>\n",
       "      <td>0.712500</td>\n",
       "      <td>0.3125</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.250000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>3.075000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.737500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t03</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0625</td>\n",
       "      <td>0.600000</td>\n",
       "      <td>0.450000</td>\n",
       "      <td>0.1500</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.350000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.425000</td>\n",
       "      <td>2.6500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>1.800000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0875</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t04</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.300000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.1625</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.387500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.912500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0625</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.950000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t05</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.075</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.337500</td>\n",
       "      <td>0.575000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.600000</td>\n",
       "      <td>0.1625</td>\n",
       "      <td>...</td>\n",
       "      <td>0.837500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.637500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t06</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.775000</td>\n",
       "      <td>1.262500</td>\n",
       "      <td>0.6125</td>\n",
       "      <td>0.062500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.175000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.775000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>2.612500</td>\n",
       "      <td>0.2375</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t07</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.362500</td>\n",
       "      <td>0.287500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.212500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>4.862500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0625</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.875000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t08</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.725000</td>\n",
       "      <td>0.312500</td>\n",
       "      <td>0.0625</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.125000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.700000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>2.012500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t09</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.212500</td>\n",
       "      <td>0.062500</td>\n",
       "      <td>0.1375</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.487500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.500000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.975000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t10</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.075000</td>\n",
       "      <td>0.150000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.175000</td>\n",
       "      <td>0.0625</td>\n",
       "      <td>0.200000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.225000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0625</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>2.800000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t11</th>\n",
       "      <td>0.075000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.212500</td>\n",
       "      <td>0.987500</td>\n",
       "      <td>0.1625</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.550000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.850000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>4.025000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t12</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.187500</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.0875</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.475000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.750000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0625</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>1.637500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t13</th>\n",
       "      <td>0.071429</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.100000</td>\n",
       "      <td>1.228571</td>\n",
       "      <td>0.1000</td>\n",
       "      <td>0.157143</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.185714</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.214286</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>3.028571</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t14</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.957143</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.600000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.085714</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.414286</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>2.085714</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t15</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.237500</td>\n",
       "      <td>0.175000</td>\n",
       "      <td>0.0625</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0875</td>\n",
       "      <td>0.337500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.625000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>3.012500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t16</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.375000</td>\n",
       "      <td>0.662500</td>\n",
       "      <td>0.2125</td>\n",
       "      <td>0.612500</td>\n",
       "      <td>0.0750</td>\n",
       "      <td>0.112500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.612500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>2.425000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t17</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.175000</td>\n",
       "      <td>0.137500</td>\n",
       "      <td>0.1000</td>\n",
       "      <td>0.250000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.612500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.987500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>2.037500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t18</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.187500</td>\n",
       "      <td>0.925000</td>\n",
       "      <td>0.0625</td>\n",
       "      <td>0.112500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.062500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>2.600000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>1.387500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t19</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.487500</td>\n",
       "      <td>0.737500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.900000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.725000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.112500</td>\n",
       "      <td>0.0625</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>2.925000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t20</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.800000</td>\n",
       "      <td>1.150000</td>\n",
       "      <td>0.1250</td>\n",
       "      <td>0.962500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.475000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.412500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.1375</td>\n",
       "      <td>2.650000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t21</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.650000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.2625</td>\n",
       "      <td>0.287500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.162500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>2.187500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t22</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.562500</td>\n",
       "      <td>0.3750</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.512500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.512500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>4.687500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t23</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.175000</td>\n",
       "      <td>0.750000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.150000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.337500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.337500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>3.300000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t24</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.062500</td>\n",
       "      <td>0.150000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.087500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.312500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.800000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>3.812500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t25</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.150000</td>\n",
       "      <td>0.512500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>2.512500</td>\n",
       "      <td>0.0625</td>\n",
       "      <td>0.550000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.962500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0625</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>3.887500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t26</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.562500</td>\n",
       "      <td>0.600000</td>\n",
       "      <td>0.1875</td>\n",
       "      <td>0.625000</td>\n",
       "      <td>0.1000</td>\n",
       "      <td>0.287500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.425000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>3.750000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t27</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.900000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.612500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.862500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.900000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>4.112500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t28</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.387500</td>\n",
       "      <td>0.812500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.737500</td>\n",
       "      <td>0.0750</td>\n",
       "      <td>0.612500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.912500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0875</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>4.487500</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>28 rows × 137 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Genus      Abditibacterium  Actinoallomurus  Actinocorallia  \\\n",
       "timepoint                                                     \n",
       "t01               0.000000            0.000          0.0000   \n",
       "t02               0.000000            0.000          0.0625   \n",
       "t03               0.000000            0.000          0.0625   \n",
       "t04               0.000000            0.000          0.0000   \n",
       "t05               0.000000            0.075          0.0000   \n",
       "t06               0.000000            0.000          0.0000   \n",
       "t07               0.000000            0.000          0.0000   \n",
       "t08               0.000000            0.000          0.0000   \n",
       "t09               0.000000            0.000          0.0000   \n",
       "t10               0.000000            0.000          0.0000   \n",
       "t11               0.075000            0.000          0.0000   \n",
       "t12               0.000000            0.000          0.0000   \n",
       "t13               0.071429            0.000          0.0000   \n",
       "t14               0.000000            0.000          0.0000   \n",
       "t15               0.000000            0.000          0.0000   \n",
       "t16               0.000000            0.000          0.0000   \n",
       "t17               0.000000            0.000          0.0000   \n",
       "t18               0.000000            0.000          0.0000   \n",
       "t19               0.000000            0.000          0.0000   \n",
       "t20               0.000000            0.000          0.0000   \n",
       "t21               0.000000            0.000          0.0000   \n",
       "t22               0.000000            0.000          0.0000   \n",
       "t23               0.000000            0.000          0.0000   \n",
       "t24               0.000000            0.000          0.0000   \n",
       "t25               0.000000            0.000          0.0000   \n",
       "t26               0.000000            0.000          0.0000   \n",
       "t27               0.000000            0.000          0.0000   \n",
       "t28               0.000000            0.000          0.0000   \n",
       "\n",
       "Genus      Actinomycetospora  Actinoplanes  Aeromicrobium  Aetherobacter  \\\n",
       "timepoint                                                                  \n",
       "t01                 0.375000      0.137500         0.7875       0.000000   \n",
       "t02                 0.587500      0.712500         0.3125       0.000000   \n",
       "t03                 0.600000      0.450000         0.1500       0.000000   \n",
       "t04                 0.300000      0.000000         0.1625       0.000000   \n",
       "t05                 0.337500      0.575000         0.0000       0.000000   \n",
       "t06                 0.775000      1.262500         0.6125       0.062500   \n",
       "t07                 0.362500      0.287500         0.0000       0.000000   \n",
       "t08                 0.725000      0.312500         0.0625       0.000000   \n",
       "t09                 0.212500      0.062500         0.1375       0.000000   \n",
       "t10                 0.075000      0.150000         0.0000       0.175000   \n",
       "t11                 0.212500      0.987500         0.1625       0.000000   \n",
       "t12                 0.187500      1.000000         0.0875       0.000000   \n",
       "t13                 0.100000      1.228571         0.1000       0.157143   \n",
       "t14                 0.957143      0.500000         0.0000       0.600000   \n",
       "t15                 0.237500      0.175000         0.0625       0.000000   \n",
       "t16                 0.375000      0.662500         0.2125       0.612500   \n",
       "t17                 0.175000      0.137500         0.1000       0.250000   \n",
       "t18                 0.187500      0.925000         0.0625       0.112500   \n",
       "t19                 0.487500      0.737500         0.0000       0.900000   \n",
       "t20                 0.800000      1.150000         0.1250       0.962500   \n",
       "t21                 0.650000      0.000000         0.2625       0.287500   \n",
       "t22                 0.000000      0.562500         0.3750       0.500000   \n",
       "t23                 0.175000      0.750000         0.0000       0.150000   \n",
       "t24                 0.062500      0.150000         0.0000       0.087500   \n",
       "t25                 0.150000      0.512500         0.0000       2.512500   \n",
       "t26                 0.562500      0.600000         0.1875       0.625000   \n",
       "t27                 0.000000      0.900000         0.0000       0.612500   \n",
       "t28                 0.387500      0.812500         0.0000       0.737500   \n",
       "\n",
       "Genus      Agreia  Agrobacterium  Agrococcus  ...  Telluria_573210  \\\n",
       "timepoint                                     ...                    \n",
       "t01        0.0000       0.000000      0.0000  ...         1.675000   \n",
       "t02        0.0000       0.250000      0.0000  ...         3.075000   \n",
       "t03        0.0000       0.350000      0.0000  ...         1.425000   \n",
       "t04        0.0000       0.387500      0.0000  ...         0.912500   \n",
       "t05        0.0000       0.600000      0.1625  ...         0.837500   \n",
       "t06        0.0000       0.175000      0.0000  ...         0.775000   \n",
       "t07        0.0000       0.212500      0.0000  ...         4.862500   \n",
       "t08        0.0000       0.125000      0.0000  ...         0.700000   \n",
       "t09        0.0000       0.487500      0.0000  ...         1.500000   \n",
       "t10        0.0625       0.200000      0.0000  ...         0.225000   \n",
       "t11        0.0000       0.550000      0.0000  ...         1.850000   \n",
       "t12        0.0000       0.475000      0.0000  ...         0.750000   \n",
       "t13        0.0000       0.185714      0.0000  ...         1.214286   \n",
       "t14        0.0000       0.085714      0.0000  ...         1.414286   \n",
       "t15        0.0875       0.337500      0.0000  ...         0.625000   \n",
       "t16        0.0750       0.112500      0.0000  ...         1.612500   \n",
       "t17        0.0000       0.612500      0.0000  ...         0.987500   \n",
       "t18        0.0000       0.062500      0.0000  ...         2.600000   \n",
       "t19        0.0000       0.725000      0.0000  ...         1.112500   \n",
       "t20        0.0000       0.475000      0.0000  ...         0.412500   \n",
       "t21        0.0000       0.000000      0.0000  ...         1.162500   \n",
       "t22        0.0000       0.512500      0.0000  ...         0.512500   \n",
       "t23        0.0000       0.337500      0.0000  ...         1.337500   \n",
       "t24        0.0000       0.312500      0.0000  ...         0.800000   \n",
       "t25        0.0625       0.550000      0.0000  ...         1.962500   \n",
       "t26        0.1000       0.287500      0.0000  ...         0.425000   \n",
       "t27        0.0000       0.862500      0.0000  ...         0.900000   \n",
       "t28        0.0750       0.612500      0.0000  ...         0.912500   \n",
       "\n",
       "Genus      Terracoccus  UBA1952  UBA4416  UBA5216  Umezawaea  Variovorax  \\\n",
       "timepoint                                                                  \n",
       "t01             0.0000   0.0000   0.0000   0.0000     0.0000    2.362500   \n",
       "t02             0.0000   0.0000   0.0000   0.0000     0.0000    0.737500   \n",
       "t03             2.6500   0.0000   0.0000   0.0000     0.0000    1.800000   \n",
       "t04             0.0000   0.0000   0.0000   0.0625     0.0000    0.950000   \n",
       "t05             0.0000   0.0000   0.0000   0.0000     0.0000    0.637500   \n",
       "t06             0.0000   0.0000   0.0000   0.0000     0.0000    2.612500   \n",
       "t07             0.0000   0.0625   0.0000   0.0000     0.0000    0.875000   \n",
       "t08             0.0000   0.0000   0.0000   0.0000     0.0000    2.012500   \n",
       "t09             0.0000   0.0000   0.0000   0.0000     0.0000    0.975000   \n",
       "t10             0.0000   0.0625   0.0000   0.0000     0.0000    2.800000   \n",
       "t11             0.0000   0.0000   0.0000   0.0000     0.0000    4.025000   \n",
       "t12             0.0000   0.0000   0.0000   0.0625     0.0000    1.637500   \n",
       "t13             0.0000   0.0000   0.0000   0.0000     0.0000    3.028571   \n",
       "t14             0.0000   0.0000   0.0000   0.0000     0.0000    2.085714   \n",
       "t15             0.0000   0.0000   0.0000   0.0000     0.0000    3.012500   \n",
       "t16             0.0000   0.0000   0.0000   0.0000     0.0000    2.425000   \n",
       "t17             0.0000   0.0000   0.0000   0.0000     0.0000    2.037500   \n",
       "t18             0.0000   0.0000   0.0000   0.0000     0.0000    1.387500   \n",
       "t19             0.0625   0.0000   0.0000   0.0000     0.0000    2.925000   \n",
       "t20             0.0000   0.0000   0.0000   0.0000     0.1375    2.650000   \n",
       "t21             0.0000   0.0000   0.0000   0.0000     0.0000    2.187500   \n",
       "t22             0.0000   0.0000   0.0000   0.0000     0.0000    4.687500   \n",
       "t23             0.0000   0.0000   0.0000   0.0000     0.0000    3.300000   \n",
       "t24             0.0000   0.0000   0.0000   0.0000     0.0000    3.812500   \n",
       "t25             0.0000   0.0625   0.0000   0.0000     0.0000    3.887500   \n",
       "t26             0.0000   0.0000   0.0000   0.0000     0.0000    3.750000   \n",
       "t27             0.0000   0.0000   0.0000   0.0000     0.0000    4.112500   \n",
       "t28             0.0000   0.0000   0.0875   0.0000     0.0000    4.487500   \n",
       "\n",
       "Genus      Virgisporangium  Williamsia_A  Xenophilus_587940  \n",
       "timepoint                                                    \n",
       "t01                 0.0000        0.0000              0.000  \n",
       "t02                 0.0000        0.0000              0.000  \n",
       "t03                 0.0000        0.0875              0.000  \n",
       "t04                 0.0000        0.0000              0.000  \n",
       "t05                 0.0000        0.0000              0.000  \n",
       "t06                 0.2375        0.0000              0.000  \n",
       "t07                 0.0000        0.0000              0.000  \n",
       "t08                 0.0000        0.0000              0.000  \n",
       "t09                 0.0000        0.0000              0.000  \n",
       "t10                 0.0000        0.0000              0.000  \n",
       "t11                 0.0000        0.0000              0.000  \n",
       "t12                 0.0000        0.0000              0.125  \n",
       "t13                 0.0000        0.0000              0.000  \n",
       "t14                 0.0000        0.0000              0.000  \n",
       "t15                 0.0000        0.0000              0.000  \n",
       "t16                 0.0000        0.0000              0.000  \n",
       "t17                 0.0000        0.0000              0.000  \n",
       "t18                 0.0000        0.0000              0.000  \n",
       "t19                 0.0000        0.0000              0.000  \n",
       "t20                 0.0000        0.0000              0.000  \n",
       "t21                 0.0000        0.0000              0.000  \n",
       "t22                 0.0000        0.0000              0.000  \n",
       "t23                 0.0000        0.0000              0.000  \n",
       "t24                 0.0000        0.0000              0.000  \n",
       "t25                 0.0000        0.0000              0.000  \n",
       "t26                 0.0000        0.0000              0.000  \n",
       "t27                 0.0000        0.0000              0.000  \n",
       "t28                 0.0000        0.0000              0.000  \n",
       "\n",
       "[28 rows x 137 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Pivot microbiome data to have timepoints as rows and genus as columns\n",
    "microbiome_by_timepoint = timepoint_average_abundance.pivot(\n",
    "    index=\"timepoint\", columns=\"Genus\", values=\"AbundR100\"\n",
    ")\n",
    "microbiome_by_timepoint = microbiome_by_timepoint.sort_index()\n",
    "microbiome_by_timepoint"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "bb14b94b",
   "metadata": {},
   "outputs": [],
   "source": [
    "microbiome_by_timepoint = microbiome_by_timepoint.loc[\n",
    "    :, (microbiome_by_timepoint > 0).sum() > minimum_samples_threshold_long\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "9380e32e",
   "metadata": {},
   "outputs": [],
   "source": [
    "def calculate_gene_genus_correlations(transcriptome_df, microbiome_df):\n",
    "    \"\"\"\n",
    "    Calculate correlations between all genes and all genera across shared timepoints.\n",
    "    Returns a DataFrame with genes as rows and genera as columns.\n",
    "    Uses vectorized Spearman correlation via pandas for speed.\n",
    "    \"\"\"\n",
    "    from scipy.stats import spearmanr\n",
    "\n",
    "    # Find shared timepoints\n",
    "    shared_timepoints = transcriptome_df.index.intersection(microbiome_df.index)\n",
    "\n",
    "    # Subset to shared timepoints\n",
    "    trans_subset = transcriptome_df.loc[shared_timepoints]\n",
    "    micro_subset = microbiome_df.loc[shared_timepoints]\n",
    "\n",
    "    print(f\"Calculating correlations across {len(shared_timepoints)} shared timepoints\")\n",
    "\n",
    "    # Vectorized approach: compute all correlations at once using scipy's spearmanr\n",
    "    # spearmanr can take two 2D arrays and return correlation matrix between all columns\n",
    "    trans_values = trans_subset.values  # shape: (n_timepoints, n_genes)\n",
    "    micro_values = micro_subset.values  # shape: (n_timepoints, n_genera)\n",
    "\n",
    "    # Concatenate and compute full correlation matrix, then extract the cross-correlation block\n",
    "    combined = np.hstack([trans_values, micro_values])\n",
    "    n_genes = trans_values.shape[1]\n",
    "    n_genera = micro_values.shape[1]\n",
    "\n",
    "    # spearmanr returns correlation matrix and p-value matrix\n",
    "    corr_full, pval_full = spearmanr(combined, axis=0)\n",
    "\n",
    "    # Extract the cross-correlation block (genes x genera)\n",
    "    correlation_matrix = pd.DataFrame(\n",
    "        corr_full[:n_genes, n_genes:],\n",
    "        index=trans_subset.columns,\n",
    "        columns=micro_subset.columns,\n",
    "    )\n",
    "    pvalue_matrix = pd.DataFrame(\n",
    "        pval_full[:n_genes, n_genes:],\n",
    "        index=trans_subset.columns,\n",
    "        columns=micro_subset.columns,\n",
    "    )\n",
    "\n",
    "    return correlation_matrix, pvalue_matrix, shared_timepoints"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "daec7ede",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "=== No shift (synchronous) ===\n",
      "Calculating correlations across 28 shared timepoints\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/michael/miniconda3/envs/Luke_terrace/lib/python3.13/site-packages/numpy/lib/_function_base_impl.py:3065: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[:, None]\n",
      "/Users/michael/miniconda3/envs/Luke_terrace/lib/python3.13/site-packages/numpy/lib/_function_base_impl.py:3066: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[None, :]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "=== Forward shift (transcriptome leads microbiome by 1 timepoint) ===\n",
      "Calculating correlations across 27 shared timepoints\n",
      "\n",
      "=== Backward shift (microbiome leads transcriptome by 1 timepoint) ===\n",
      "Calculating correlations across 27 shared timepoints\n"
     ]
    }
   ],
   "source": [
    "# Calculate correlations for all three scenarios\n",
    "print(\"=== No shift (synchronous) ===\")\n",
    "corr_no_shift, pval_no_shift, tp_no_shift = calculate_gene_genus_correlations(\n",
    "    transcriptome_by_timepoint, microbiome_by_timepoint\n",
    ")\n",
    "\n",
    "print(\"\\n=== Forward shift (transcriptome leads microbiome by 1 timepoint) ===\")\n",
    "corr_forward, pval_forward, tp_forward = calculate_gene_genus_correlations(\n",
    "    transcriptome_shifted_forward, microbiome_by_timepoint\n",
    ")\n",
    "\n",
    "print(\"\\n=== Backward shift (microbiome leads transcriptome by 1 timepoint) ===\")\n",
    "corr_backward, pval_backward, tp_backward = calculate_gene_genus_correlations(\n",
    "    transcriptome_shifted_backward, microbiome_by_timepoint\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "22549c8f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pickle\n",
    "import os\n",
    "\n",
    "# Create output directory if it doesn't exist\n",
    "output_dir = \"/Users/michael/Data/Luke_terrace_experiment/General_data/Junk_data/Time_shifted_correlation/v2\"\n",
    "os.makedirs(output_dir, exist_ok=True)\n",
    "\n",
    "# Save no shift correlations\n",
    "dbfile = open(f\"{output_dir}/corr_no_shift_remove_low_exp.p\", \"ab\")\n",
    "pickle.dump(corr_no_shift, dbfile)\n",
    "dbfile.close()\n",
    "\n",
    "dbfile = open(f\"{output_dir}/pval_no_shift_remove_low_exp.p\", \"ab\")\n",
    "pickle.dump(pval_no_shift, dbfile)\n",
    "dbfile.close()\n",
    "\n",
    "dbfile = open(f\"{output_dir}/tp_no_shift_remove_low_exp.p\", \"ab\")\n",
    "pickle.dump(tp_no_shift, dbfile)\n",
    "dbfile.close()\n",
    "\n",
    "# Save forward shift correlations\n",
    "dbfile = open(f\"{output_dir}/corr_forward_shift_remove_low_exp.p\", \"ab\")\n",
    "pickle.dump(corr_forward, dbfile)\n",
    "dbfile.close()\n",
    "\n",
    "dbfile = open(f\"{output_dir}/pval_forward_shift_remove_low_exp.p\", \"ab\")\n",
    "pickle.dump(pval_forward, dbfile)\n",
    "dbfile.close()\n",
    "\n",
    "dbfile = open(f\"{output_dir}/tp_forward_shift_remove_low_exp.p\", \"ab\")\n",
    "pickle.dump(tp_forward, dbfile)\n",
    "dbfile.close()\n",
    "\n",
    "# Save backward shift correlations\n",
    "dbfile = open(f\"{output_dir}/corr_backward_shift_remove_low_exp.p\", \"ab\")\n",
    "pickle.dump(corr_backward, dbfile)\n",
    "dbfile.close()\n",
    "\n",
    "dbfile = open(f\"{output_dir}/pval_backward_shift_remove_low_exp.p\", \"ab\")\n",
    "pickle.dump(pval_backward, dbfile)\n",
    "dbfile.close()\n",
    "\n",
    "dbfile = open(f\"{output_dir}/tp_backward_shift_remove_low_exp.p\", \"ab\")\n",
    "pickle.dump(tp_backward, dbfile)\n",
    "dbfile.close()\n",
    "\n",
    "# print(f\"Saved all correlation results to {output_dir}\")\n",
    "\n",
    "# # Read no shift correlations\n",
    "# dbfile = open(f\"{output_dir}/corr_no_shift.p\", \"rb\")\n",
    "# corr_no_shift = pickle.load(dbfile)\n",
    "# dbfile.close()\n",
    "\n",
    "# dbfile = open(f\"{output_dir}/pval_no_shift.p\", \"rb\")\n",
    "# pval_no_shift = pickle.load(dbfile)\n",
    "# dbfile.close()\n",
    "\n",
    "# dbfile = open(f\"{output_dir}/tp_no_shift.p\", \"rb\")\n",
    "# tp_no_shift = pickle.load(dbfile)\n",
    "# dbfile.close()\n",
    "\n",
    "# # Read forward shift correlations\n",
    "# dbfile = open(f\"{output_dir}/corr_forward_shift.p\", \"rb\")\n",
    "# corr_forward = pickle.load(dbfile)\n",
    "# dbfile.close()\n",
    "\n",
    "# dbfile = open(f\"{output_dir}/pval_forward_shift.p\", \"rb\")\n",
    "# pval_forward = pickle.load(dbfile)\n",
    "# dbfile.close()\n",
    "\n",
    "# dbfile = open(f\"{output_dir}/tp_forward_shift.p\", \"rb\")\n",
    "# tp_forward = pickle.load(dbfile)\n",
    "# dbfile.close()\n",
    "\n",
    "# # Read backward shift correlations\n",
    "# dbfile = open(f\"{output_dir}/corr_backward_shift.p\", \"rb\")\n",
    "# corr_backward = pickle.load(dbfile)\n",
    "# dbfile.close()\n",
    "\n",
    "# dbfile = open(f\"{output_dir}/pval_backward_shift.p\", \"rb\")\n",
    "# pval_backward = pickle.load(dbfile)\n",
    "# dbfile.close()\n",
    "\n",
    "# dbfile = open(f\"{output_dir}/tp_backward_shift.p\", \"rb\")\n",
    "# tp_backward = pickle.load(dbfile)\n",
    "# dbfile.close()\n",
    "\n",
    "# print(f\"Loaded all correlation results from {output_dir}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1ec3c77e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Compare correlation distributions across shifts\n",
    "fig, axes = plt.subplots(1, 3, figsize=(15, 5))\n",
    "\n",
    "# Flatten correlation matrices for distribution plots\n",
    "corr_no_shift_flat = corr_no_shift.values.flatten()\n",
    "corr_forward_flat = corr_forward.values.flatten()\n",
    "corr_backward_flat = corr_backward.values.flatten()\n",
    "\n",
    "# Remove NaN values\n",
    "corr_no_shift_flat = corr_no_shift_flat[~np.isnan(corr_no_shift_flat)]\n",
    "corr_forward_flat = corr_forward_flat[~np.isnan(corr_forward_flat)]\n",
    "corr_backward_flat = corr_backward_flat[~np.isnan(corr_backward_flat)]\n",
    "\n",
    "axes[0].hist(corr_no_shift_flat, bins=50, alpha=0.7, edgecolor=\"black\")\n",
    "axes[0].set_title(\"No Shift (Synchronous)\")\n",
    "axes[0].set_xlabel(\"Spearman Correlation\")\n",
    "axes[0].set_ylabel(\"Frequency\")\n",
    "axes[0].axvline(x=0, color=\"red\", linestyle=\"--\", alpha=0.5)\n",
    "\n",
    "axes[1].hist(corr_forward_flat, bins=50, alpha=0.7, edgecolor=\"black\", color=\"green\")\n",
    "axes[1].set_title(\"Forward Shift\\n(Transcriptome leads)\")\n",
    "axes[1].set_xlabel(\"Spearman Correlation\")\n",
    "axes[1].set_ylabel(\"Frequency\")\n",
    "axes[1].axvline(x=0, color=\"red\", linestyle=\"--\", alpha=0.5)\n",
    "\n",
    "axes[2].hist(corr_backward_flat, bins=50, alpha=0.7, edgecolor=\"black\", color=\"orange\")\n",
    "axes[2].set_title(\"Backward Shift\\n(Microbiome leads)\")\n",
    "axes[2].set_xlabel(\"Spearman Correlation\")\n",
    "axes[2].set_ylabel(\"Frequency\")\n",
    "axes[2].axvline(x=0, color=\"red\", linestyle=\"--\", alpha=0.5)\n",
    "\n",
    "plt.tight_layout()\n",
    "plt.show()\n",
    "\n",
    "print(f\"No shift - Mean abs correlation: {np.mean(np.abs(corr_no_shift_flat)):.4f}\")\n",
    "print(f\"Forward shift - Mean abs correlation: {np.mean(np.abs(corr_forward_flat)):.4f}\")\n",
    "print(\n",
    "    f\"Backward shift - Mean abs correlation: {np.mean(np.abs(corr_backward_flat)):.4f}\"\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c5b8abe5",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Find genes with strongest delayed correlations\n",
    "# Compare: which genes have better correlations with shifted vs non-shifted data?\n",
    "\n",
    "\n",
    "def find_best_shift_correlations(\n",
    "    corr_no_shift, corr_forward, corr_backward, pval_threshold=0.05\n",
    "):\n",
    "    \"\"\"\n",
    "    For each gene-genus pair, find which shift gives the strongest correlation.\n",
    "    \"\"\"\n",
    "    results = []\n",
    "\n",
    "    for gene in corr_no_shift.index:\n",
    "        for genus in corr_no_shift.columns:\n",
    "            r_sync = corr_no_shift.loc[gene, genus]\n",
    "            r_fwd = (\n",
    "                corr_forward.loc[gene, genus] if gene in corr_forward.index else np.nan\n",
    "            )\n",
    "            r_bwd = (\n",
    "                corr_backward.loc[gene, genus]\n",
    "                if gene in corr_backward.index\n",
    "                else np.nan\n",
    "            )\n",
    "\n",
    "            # Find the shift with max absolute correlation\n",
    "            correlations = {\"sync\": r_sync, \"forward\": r_fwd, \"backward\": r_bwd}\n",
    "            valid_corrs = {k: v for k, v in correlations.items() if not np.isnan(v)}\n",
    "\n",
    "            if valid_corrs:\n",
    "                best_shift = max(valid_corrs.keys(), key=lambda k: abs(valid_corrs[k]))\n",
    "                best_corr = valid_corrs[best_shift]\n",
    "\n",
    "                results.append(\n",
    "                    {\n",
    "                        \"gene\": gene,\n",
    "                        \"genus\": genus,\n",
    "                        \"best_shift\": best_shift,\n",
    "                        \"best_correlation\": best_corr,\n",
    "                        \"sync_correlation\": r_sync,\n",
    "                        \"forward_correlation\": r_fwd,\n",
    "                        \"backward_correlation\": r_bwd,\n",
    "                    }\n",
    "                )\n",
    "\n",
    "    return pd.DataFrame(results)\n",
    "\n",
    "\n",
    "shift_comparison = find_best_shift_correlations(\n",
    "    corr_no_shift, corr_forward, corr_backward\n",
    ")\n",
    "shift_comparison"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9f7782f7",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Summary: count how many gene-genus pairs have each shift as their best\n",
    "shift_counts = shift_comparison[\"best_shift\"].value_counts()\n",
    "print(\"Best shift distribution across all gene-genus pairs:\")\n",
    "print(shift_counts)\n",
    "print(f\"\\nTotal pairs analyzed: {len(shift_comparison)}\")\n",
    "\n",
    "# Plot\n",
    "fig, ax = plt.subplots(figsize=(8, 5))\n",
    "shift_counts.plot(kind=\"bar\", ax=ax, color=[\"blue\", \"green\", \"orange\"])\n",
    "ax.set_title(\"Distribution of Best Shift Across Gene-Genus Pairs\")\n",
    "ax.set_xlabel(\"Shift Type\")\n",
    "ax.set_ylabel(\"Number of Gene-Genus Pairs\")\n",
    "ax.set_xticklabels(\n",
    "    [\"Synchronous\", \"Transcriptome Leads\", \"Microbiome Leads\"], rotation=45\n",
    ")\n",
    "plt.tight_layout()\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c781af89",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Show top genes with strongest delayed correlations (forward or backward shift better than sync)\n",
    "# Filter for pairs where shifted correlation is notably stronger\n",
    "\n",
    "shift_comparison[\"improvement_forward\"] = (\n",
    "    shift_comparison[\"forward_correlation\"].abs()\n",
    "    - shift_comparison[\"sync_correlation\"].abs()\n",
    ")\n",
    "shift_comparison[\"improvement_backward\"] = (\n",
    "    shift_comparison[\"backward_correlation\"].abs()\n",
    "    - shift_comparison[\"sync_correlation\"].abs()\n",
    ")\n",
    "\n",
    "# Top genes where transcriptome leads (forward shift is better)\n",
    "top_forward = shift_comparison.nlargest(20, \"improvement_forward\")[\n",
    "    [\"gene\", \"genus\", \"sync_correlation\", \"forward_correlation\", \"improvement_forward\"]\n",
    "]\n",
    "print(\n",
    "    \"Top 20 gene-genus pairs where transcriptome LEADS microbiome (forward shift better):\"\n",
    ")\n",
    "display(top_forward)\n",
    "\n",
    "# Top genes where microbiome leads (backward shift is better)\n",
    "top_backward = shift_comparison.nlargest(20, \"improvement_backward\")[\n",
    "    [\n",
    "        \"gene\",\n",
    "        \"genus\",\n",
    "        \"sync_correlation\",\n",
    "        \"backward_correlation\",\n",
    "        \"improvement_backward\",\n",
    "    ]\n",
    "]\n",
    "print(\n",
    "    \"\\nTop 20 gene-genus pairs where microbiome LEADS transcriptome (backward shift better):\"\n",
    ")\n",
    "display(top_backward)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cdb6e30e",
   "metadata": {},
   "outputs": [],
   "source": [
    "shift_comparison.loc[shift_comparison[\"genus\"] == \"Sphingomonas_L_486704\"].sort_values(\n",
    "    by=\"improvement_forward\", ascending=False\n",
    ").head(30)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "334541b0",
   "metadata": {},
   "outputs": [],
   "source": [
    "sphingo_only = shift_comparison.loc[\n",
    "    shift_comparison[\"genus\"] == \"Sphingomonas_L_486704\"\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5cb45c52",
   "metadata": {},
   "outputs": [],
   "source": [
    "sphingo_only[\n",
    "    [\n",
    "        \"sync_correlation\",\n",
    "        \"forward_correlation\",\n",
    "        \"backward_correlation\",\n",
    "        \"improvement_forward\",\n",
    "        \"improvement_backward\",\n",
    "    ]\n",
    "].mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "acf3e04b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Heatmap comparison for a selected genus showing correlations across shifts\n",
    "# Select a few genera with interesting patterns\n",
    "top_genera = microbiome_by_timepoint.mean().nlargest(5).index.tolist()\n",
    "\n",
    "fig, axes = plt.subplots(1, 3, figsize=(18, 8))\n",
    "\n",
    "# For visualization, select top variable genes\n",
    "gene_variance = transcriptome_by_timepoint.var()\n",
    "top_variable_genes = gene_variance.nlargest(50).index\n",
    "\n",
    "for ax, (title, corr_mat) in zip(\n",
    "    axes,\n",
    "    [\n",
    "        (\"No Shift (Synchronous)\", corr_no_shift),\n",
    "        (\"Forward Shift (Transcriptome leads)\", corr_forward),\n",
    "        (\"Backward Shift (Microbiome leads)\", corr_backward),\n",
    "    ],\n",
    "):\n",
    "    subset = corr_mat.loc[top_variable_genes, top_genera]\n",
    "    sns.heatmap(\n",
    "        subset.astype(float),\n",
    "        ax=ax,\n",
    "        cmap=\"RdBu_r\",\n",
    "        center=0,\n",
    "        vmin=-1,\n",
    "        vmax=1,\n",
    "        cbar_kws={\"shrink\": 0.8},\n",
    "    )\n",
    "    ax.set_title(title)\n",
    "    ax.set_xlabel(\"Genus\")\n",
    "    ax.set_ylabel(\"Gene\")\n",
    "\n",
    "plt.tight_layout()\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3f499f0f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Store the correlation results for further analysis\n",
    "correlation_results = {\n",
    "    \"no_shift\": {\"correlations\": corr_no_shift, \"pvalues\": pval_no_shift},\n",
    "    \"forward_shift\": {\"correlations\": corr_forward, \"pvalues\": pval_forward},\n",
    "    \"backward_shift\": {\"correlations\": corr_backward, \"pvalues\": pval_backward},\n",
    "}\n",
    "\n",
    "# Summary statistics per genus for each shift\n",
    "print(\"Average absolute correlation per genus across all shifts:\\n\")\n",
    "summary_by_genus = pd.DataFrame(\n",
    "    {\n",
    "        \"Synchronous\": corr_no_shift.abs().mean(),\n",
    "        \"Forward (Trans leads)\": corr_forward.abs().mean(),\n",
    "        \"Backward (Micro leads)\": corr_backward.abs().mean(),\n",
    "    }\n",
    ")\n",
    "summary_by_genus[\"Best Shift\"] = summary_by_genus.idxmax(axis=1)\n",
    "display(summary_by_genus.sort_values(\"Synchronous\", ascending=False))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1ae571a8",
   "metadata": {},
   "outputs": [],
   "source": [
    "summary_by_genus.sort_values(\"Backward (Micro leads)\").head(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "47a78e71",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Circadian Experiment 1 - Time Shifted Correlation Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "80cabd0d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Prepare Circadian Experiment 1 microbiome data\n",
    "circadian_microbiome_1 = merged_microbiome.loc[\n",
    "    merged_microbiome[\"Experiment Type\"] == \"Circadian Experiment 1\"\n",
    "]\n",
    "\n",
    "# Aggregate genus abundance by sample\n",
    "circ1_genus_sum = (\n",
    "    circadian_microbiome_1[[\"Sample\", \"Genus\", \"AbundR100\"]]\n",
    "    .groupby([\"Sample\", \"Genus\"])\n",
    "    .sum()\n",
    "    .reset_index()\n",
    ")\n",
    "\n",
    "# Merge with timepoint info\n",
    "circ1_genus_sum_w_tp = circ1_genus_sum.merge(\n",
    "    metadata[[\"sampID\", \"timepoint\"]], left_on=\"Sample\", right_on=\"sampID\"\n",
    ")\n",
    "\n",
    "# Fill in missing genus for each sample\n",
    "all_circ1_genus = circ1_genus_sum_w_tp[\"Genus\"].unique()\n",
    "for sampID in circ1_genus_sum_w_tp[\"sampID\"].unique():\n",
    "    sampid_only_df = circ1_genus_sum_w_tp.loc[circ1_genus_sum_w_tp[\"sampID\"] == sampID]\n",
    "    present_genus_in_samp = sampid_only_df[\"Genus\"].unique()\n",
    "    difference_in_present_bacteria = list(\n",
    "        set(all_circ1_genus) - set(present_genus_in_samp)\n",
    "    )\n",
    "    current_time_point = sampid_only_df[\"timepoint\"].iloc[0]\n",
    "    empty_df_to_add = pd.DataFrame(\n",
    "        columns=[\"Sample\", \"Genus\", \"AbundR100\", \"sampID\", \"timepoint\"]\n",
    "    )\n",
    "    empty_df_to_add[\"Genus\"] = difference_in_present_bacteria\n",
    "    empty_df_to_add[\"Sample\"] = sampID\n",
    "    empty_df_to_add[\"AbundR100\"] = 0\n",
    "    empty_df_to_add[\"sampID\"] = sampID\n",
    "    empty_df_to_add[\"timepoint\"] = current_time_point\n",
    "    circ1_genus_sum_w_tp = pd.concat([circ1_genus_sum_w_tp, empty_df_to_add])\n",
    "\n",
    "# Calculate average abundance by timepoint\n",
    "circ1_timepoint_avg_abundance = (\n",
    "    circ1_genus_sum_w_tp[[\"Genus\", \"timepoint\", \"AbundR100\"]]\n",
    "    .groupby([\"Genus\", \"timepoint\"])\n",
    "    .mean()\n",
    "    .reset_index()\n",
    ")\n",
    "\n",
    "print(\n",
    "    f\"Circadian 1 - Unique timepoints: {sorted(circ1_timepoint_avg_abundance['timepoint'].unique())}\"\n",
    ")\n",
    "print(f\"Circadian 1 - Number of genera: {len(all_circ1_genus)}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4232b45e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Aggregate Circadian 1 transcriptome by timepoint\n",
    "circadian_transcriptome_1 = circadian_transcriptome_1.sort_index()\n",
    "circadian_metadata_1 = circadian_metadata_1.sort_index()\n",
    "\n",
    "circ1_transcriptome_w_tp = circadian_transcriptome_1.copy()\n",
    "circ1_transcriptome_w_tp[\"timepoint\"] = circadian_metadata_1[\"timepoint\"]\n",
    "circ1_transcriptome_by_tp = circ1_transcriptome_w_tp.groupby(\"timepoint\").mean()\n",
    "\n",
    "# Get sorted timepoints\n",
    "circ1_timepoints = sorted(circ1_transcriptome_by_tp.index.unique())\n",
    "print(f\"Circadian 1 transcriptome timepoints: {circ1_timepoints}\")\n",
    "\n",
    "# Create shifted transcriptome dataframes\n",
    "circ1_tp_to_idx = {tp: i for i, tp in enumerate(circ1_timepoints)}\n",
    "circ1_idx_to_tp = {i: tp for i, tp in enumerate(circ1_timepoints)}\n",
    "\n",
    "# Forward shift: transcriptome at t correlates with microbiome at t+1 (transcriptome leads)\n",
    "circ1_trans_shifted_forward = circ1_transcriptome_by_tp.copy()\n",
    "circ1_trans_shifted_forward.index = [\n",
    "    circ1_idx_to_tp.get(circ1_tp_to_idx[tp] + 1, None)\n",
    "    for tp in circ1_trans_shifted_forward.index\n",
    "]\n",
    "circ1_trans_shifted_forward = circ1_trans_shifted_forward[\n",
    "    circ1_trans_shifted_forward.index.notna()\n",
    "]\n",
    "\n",
    "# Backward shift: transcriptome at t correlates with microbiome at t-1 (microbiome leads)\n",
    "circ1_trans_shifted_backward = circ1_transcriptome_by_tp.copy()\n",
    "circ1_trans_shifted_backward.index = [\n",
    "    circ1_idx_to_tp.get(circ1_tp_to_idx[tp] - 1, None)\n",
    "    for tp in circ1_trans_shifted_backward.index\n",
    "]\n",
    "circ1_trans_shifted_backward = circ1_trans_shifted_backward[\n",
    "    circ1_trans_shifted_backward.index.notna()\n",
    "]\n",
    "\n",
    "print(f\"Forward shifted timepoints: {list(circ1_trans_shifted_forward.index)}\")\n",
    "print(f\"Backward shifted timepoints: {list(circ1_trans_shifted_backward.index)}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e4ba0f7d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Pivot microbiome data for Circadian 1\n",
    "circ1_microbiome_by_tp = circ1_timepoint_avg_abundance.pivot(\n",
    "    index=\"timepoint\", columns=\"Genus\", values=\"AbundR100\"\n",
    ")\n",
    "\n",
    "circ1_microbiome_by_tp = circ1_microbiome_by_tp.loc[\n",
    "    :, (circ1_microbiome_by_tp > 0).sum() > minimum_samples_threshold_circadian\n",
    "]\n",
    "\n",
    "\n",
    "circ1_microbiome_by_tp = circ1_microbiome_by_tp.sort_index()\n",
    "\n",
    "# Calculate correlations for Circadian 1\n",
    "print(\"=== Circadian 1: No shift (synchronous) ===\")\n",
    "circ1_corr_no_shift, circ1_pval_no_shift, circ1_tp_no_shift = (\n",
    "    calculate_gene_genus_correlations(circ1_transcriptome_by_tp, circ1_microbiome_by_tp)\n",
    ")\n",
    "\n",
    "print(\"\\n=== Circadian 1: Forward shift (transcriptome leads) ===\")\n",
    "circ1_corr_forward, circ1_pval_forward, circ1_tp_forward = (\n",
    "    calculate_gene_genus_correlations(\n",
    "        circ1_trans_shifted_forward, circ1_microbiome_by_tp\n",
    "    )\n",
    ")\n",
    "\n",
    "print(\"\\n=== Circadian 1: Backward shift (microbiome leads) ===\")\n",
    "circ1_corr_backward, circ1_pval_backward, circ1_tp_backward = (\n",
    "    calculate_gene_genus_correlations(\n",
    "        circ1_trans_shifted_backward, circ1_microbiome_by_tp\n",
    "    )\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fa318814",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Circadian 1: Compare correlation distributions\n",
    "fig, axes = plt.subplots(1, 3, figsize=(15, 5))\n",
    "\n",
    "circ1_corr_no_shift_flat = circ1_corr_no_shift.values.flatten()\n",
    "circ1_corr_forward_flat = circ1_corr_forward.values.flatten()\n",
    "circ1_corr_backward_flat = circ1_corr_backward.values.flatten()\n",
    "\n",
    "circ1_corr_no_shift_flat = circ1_corr_no_shift_flat[~np.isnan(circ1_corr_no_shift_flat)]\n",
    "circ1_corr_forward_flat = circ1_corr_forward_flat[~np.isnan(circ1_corr_forward_flat)]\n",
    "circ1_corr_backward_flat = circ1_corr_backward_flat[~np.isnan(circ1_corr_backward_flat)]\n",
    "\n",
    "axes[0].hist(circ1_corr_no_shift_flat, bins=50, alpha=0.7, edgecolor=\"black\")\n",
    "axes[0].set_title(\"Circadian 1: No Shift\")\n",
    "axes[0].set_xlabel(\"Spearman Correlation\")\n",
    "axes[0].axvline(x=0, color=\"red\", linestyle=\"--\", alpha=0.5)\n",
    "\n",
    "axes[1].hist(\n",
    "    circ1_corr_forward_flat, bins=50, alpha=0.7, edgecolor=\"black\", color=\"green\"\n",
    ")\n",
    "axes[1].set_title(\"Circadian 1: Forward Shift\\n(Transcriptome leads)\")\n",
    "axes[1].set_xlabel(\"Spearman Correlation\")\n",
    "axes[1].axvline(x=0, color=\"red\", linestyle=\"--\", alpha=0.5)\n",
    "\n",
    "axes[2].hist(\n",
    "    circ1_corr_backward_flat, bins=50, alpha=0.7, edgecolor=\"black\", color=\"orange\"\n",
    ")\n",
    "axes[2].set_title(\"Circadian 1: Backward Shift\\n(Microbiome leads)\")\n",
    "axes[2].set_xlabel(\"Spearman Correlation\")\n",
    "axes[2].axvline(x=0, color=\"red\", linestyle=\"--\", alpha=0.5)\n",
    "\n",
    "plt.suptitle(\"Circadian Experiment 1 - Correlation Distributions\", y=1.02)\n",
    "plt.tight_layout()\n",
    "plt.show()\n",
    "\n",
    "print(\n",
    "    f\"Circadian 1 - No shift mean abs correlation: {np.mean(np.abs(circ1_corr_no_shift_flat)):.4f}\"\n",
    ")\n",
    "print(\n",
    "    f\"Circadian 1 - Forward shift mean abs correlation: {np.mean(np.abs(circ1_corr_forward_flat)):.4f}\"\n",
    ")\n",
    "print(\n",
    "    f\"Circadian 1 - Backward shift mean abs correlation: {np.mean(np.abs(circ1_corr_backward_flat)):.4f}\"\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "58bec219",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save Circadian 1 results\n",
    "output_dir_circ1 = \"/Users/michael/Data/Luke_terrace_experiment/General_data/Junk_data/Time_shifted_correlation/Circadian_1\"\n",
    "os.makedirs(output_dir_circ1, exist_ok=True)\n",
    "\n",
    "# dbfile = open(f\"{output_dir_circ1}/corr_no_shift_remove_low_exp.p\", \"ab\")\n",
    "# pickle.dump(circ1_corr_no_shift, dbfile)\n",
    "# dbfile.close()\n",
    "\n",
    "# dbfile = open(f\"{output_dir_circ1}/pval_no_shift_remove_low_exp.p\", \"ab\")\n",
    "# pickle.dump(circ1_pval_no_shift, dbfile)\n",
    "# dbfile.close()\n",
    "\n",
    "# dbfile = open(f\"{output_dir_circ1}/tp_no_shift_remove_low_exp.p\", \"ab\")\n",
    "# pickle.dump(circ1_tp_no_shift, dbfile)\n",
    "# dbfile.close()\n",
    "\n",
    "# dbfile = open(f\"{output_dir_circ1}/corr_forward_shift_remove_low_exp.p\", \"ab\")\n",
    "# pickle.dump(circ1_corr_forward, dbfile)\n",
    "# dbfile.close()\n",
    "\n",
    "# dbfile = open(f\"{output_dir_circ1}/pval_forward_shift_remove_low_exp.p\", \"ab\")\n",
    "# pickle.dump(circ1_pval_forward, dbfile)\n",
    "# dbfile.close()\n",
    "\n",
    "# dbfile = open(f\"{output_dir_circ1}/tp_forward_shift_remove_low_exp.p\", \"ab\")\n",
    "# pickle.dump(circ1_tp_forward, dbfile)\n",
    "# dbfile.close()\n",
    "\n",
    "# dbfile = open(f\"{output_dir_circ1}/corr_backward_shift_remove_low_exp.p\", \"ab\")\n",
    "# pickle.dump(circ1_corr_backward, dbfile)\n",
    "# dbfile.close()\n",
    "\n",
    "# dbfile = open(f\"{output_dir_circ1}/pval_backward_shift_remove_low_exp.p\", \"ab\")\n",
    "# pickle.dump(circ1_pval_backward, dbfile)\n",
    "# dbfile.close()\n",
    "\n",
    "# dbfile = open(f\"{output_dir_circ1}/tp_backward_shift_remove_low_exp.p\", \"ab\")\n",
    "# pickle.dump(circ1_tp_backward, dbfile)\n",
    "# dbfile.close()\n",
    "\n",
    "print(f\"Saved Circadian 1 results to {output_dir_circ1}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "14c452a2",
   "metadata": {},
   "source": [
    "# Circadian Experiment 2 - Time Shifted Correlation Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e21b5e44",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Prepare Circadian Experiment 2 microbiome data\n",
    "circadian_microbiome_2 = merged_microbiome.loc[\n",
    "    merged_microbiome[\"Experiment Type\"] == \"Circadian Experiment 2\"\n",
    "]\n",
    "\n",
    "# Aggregate genus abundance by sample\n",
    "circ2_genus_sum = (\n",
    "    circadian_microbiome_2[[\"Sample\", \"Genus\", \"AbundR100\"]]\n",
    "    .groupby([\"Sample\", \"Genus\"])\n",
    "    .sum()\n",
    "    .reset_index()\n",
    ")\n",
    "\n",
    "# Merge with timepoint info\n",
    "circ2_genus_sum_w_tp = circ2_genus_sum.merge(\n",
    "    metadata[[\"sampID\", \"timepoint\"]], left_on=\"Sample\", right_on=\"sampID\"\n",
    ")\n",
    "\n",
    "# Fill in missing genus for each sample\n",
    "all_circ2_genus = circ2_genus_sum_w_tp[\"Genus\"].unique()\n",
    "for sampID in circ2_genus_sum_w_tp[\"sampID\"].unique():\n",
    "    sampid_only_df = circ2_genus_sum_w_tp.loc[circ2_genus_sum_w_tp[\"sampID\"] == sampID]\n",
    "    present_genus_in_samp = sampid_only_df[\"Genus\"].unique()\n",
    "    difference_in_present_bacteria = list(\n",
    "        set(all_circ2_genus) - set(present_genus_in_samp)\n",
    "    )\n",
    "    current_time_point = sampid_only_df[\"timepoint\"].iloc[0]\n",
    "    empty_df_to_add = pd.DataFrame(\n",
    "        columns=[\"Sample\", \"Genus\", \"AbundR100\", \"sampID\", \"timepoint\"]\n",
    "    )\n",
    "    empty_df_to_add[\"Genus\"] = difference_in_present_bacteria\n",
    "    empty_df_to_add[\"Sample\"] = sampID\n",
    "    empty_df_to_add[\"AbundR100\"] = 0\n",
    "    empty_df_to_add[\"sampID\"] = sampID\n",
    "    empty_df_to_add[\"timepoint\"] = current_time_point\n",
    "    circ2_genus_sum_w_tp = pd.concat([circ2_genus_sum_w_tp, empty_df_to_add])\n",
    "\n",
    "# Calculate average abundance by timepoint\n",
    "circ2_timepoint_avg_abundance = (\n",
    "    circ2_genus_sum_w_tp[[\"Genus\", \"timepoint\", \"AbundR100\"]]\n",
    "    .groupby([\"Genus\", \"timepoint\"])\n",
    "    .mean()\n",
    "    .reset_index()\n",
    ")\n",
    "\n",
    "print(\n",
    "    f\"Circadian 2 - Unique timepoints: {sorted(circ2_timepoint_avg_abundance['timepoint'].unique())}\"\n",
    ")\n",
    "print(f\"Circadian 2 - Number of genera: {len(all_circ2_genus)}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0a5e412c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Aggregate Circadian 2 transcriptome by timepoint\n",
    "circadian_transcriptome_2 = circadian_transcriptome_2.sort_index()\n",
    "circadian_metadata_2 = circadian_metadata_2.sort_index()\n",
    "\n",
    "circ2_transcriptome_w_tp = circadian_transcriptome_2.copy()\n",
    "circ2_transcriptome_w_tp[\"timepoint\"] = circadian_metadata_2[\"timepoint\"]\n",
    "circ2_transcriptome_by_tp = circ2_transcriptome_w_tp.groupby(\"timepoint\").mean()\n",
    "\n",
    "# Get sorted timepoints\n",
    "circ2_timepoints = sorted(circ2_transcriptome_by_tp.index.unique())\n",
    "print(f\"Circadian 2 transcriptome timepoints: {circ2_timepoints}\")\n",
    "\n",
    "# Create shifted transcriptome dataframes\n",
    "circ2_tp_to_idx = {tp: i for i, tp in enumerate(circ2_timepoints)}\n",
    "circ2_idx_to_tp = {i: tp for i, tp in enumerate(circ2_timepoints)}\n",
    "\n",
    "# Forward shift: transcriptome at t correlates with microbiome at t+1 (transcriptome leads)\n",
    "circ2_trans_shifted_forward = circ2_transcriptome_by_tp.copy()\n",
    "circ2_trans_shifted_forward.index = [\n",
    "    circ2_idx_to_tp.get(circ2_tp_to_idx[tp] + 1, None)\n",
    "    for tp in circ2_trans_shifted_forward.index\n",
    "]\n",
    "circ2_trans_shifted_forward = circ2_trans_shifted_forward[\n",
    "    circ2_trans_shifted_forward.index.notna()\n",
    "]\n",
    "\n",
    "# Backward shift: transcriptome at t correlates with microbiome at t-1 (microbiome leads)\n",
    "circ2_trans_shifted_backward = circ2_transcriptome_by_tp.copy()\n",
    "circ2_trans_shifted_backward.index = [\n",
    "    circ2_idx_to_tp.get(circ2_tp_to_idx[tp] - 1, None)\n",
    "    for tp in circ2_trans_shifted_backward.index\n",
    "]\n",
    "circ2_trans_shifted_backward = circ2_trans_shifted_backward[\n",
    "    circ2_trans_shifted_backward.index.notna()\n",
    "]\n",
    "\n",
    "print(f\"Forward shifted timepoints: {list(circ2_trans_shifted_forward.index)}\")\n",
    "print(f\"Backward shifted timepoints: {list(circ2_trans_shifted_backward.index)}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "44ac766e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Pivot microbiome data for Circadian 2\n",
    "circ2_microbiome_by_tp = circ2_timepoint_avg_abundance.pivot(\n",
    "    index=\"timepoint\", columns=\"Genus\", values=\"AbundR100\"\n",
    ")\n",
    "\n",
    "circ2_microbiome_by_tp = circ2_microbiome_by_tp.loc[\n",
    "    :, (circ2_microbiome_by_tp > 0).sum() > minimum_samples_threshold_circadian\n",
    "]\n",
    "circ2_microbiome_by_tp = circ2_microbiome_by_tp.sort_index()\n",
    "\n",
    "# Calculate correlations for Circadian 2\n",
    "print(\"=== Circadian 2: No shift (synchronous) ===\")\n",
    "circ2_corr_no_shift, circ2_pval_no_shift, circ2_tp_no_shift = (\n",
    "    calculate_gene_genus_correlations(circ2_transcriptome_by_tp, circ2_microbiome_by_tp)\n",
    ")\n",
    "\n",
    "print(\"\\n=== Circadian 2: Forward shift (transcriptome leads) ===\")\n",
    "circ2_corr_forward, circ2_pval_forward, circ2_tp_forward = (\n",
    "    calculate_gene_genus_correlations(\n",
    "        circ2_trans_shifted_forward, circ2_microbiome_by_tp\n",
    "    )\n",
    ")\n",
    "\n",
    "print(\"\\n=== Circadian 2: Backward shift (microbiome leads) ===\")\n",
    "circ2_corr_backward, circ2_pval_backward, circ2_tp_backward = (\n",
    "    calculate_gene_genus_correlations(\n",
    "        circ2_trans_shifted_backward, circ2_microbiome_by_tp\n",
    "    )\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "16bebb56",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Circadian 2: Compare correlation distributions\n",
    "fig, axes = plt.subplots(1, 3, figsize=(15, 5))\n",
    "\n",
    "circ2_corr_no_shift_flat = circ2_corr_no_shift.values.flatten()\n",
    "circ2_corr_forward_flat = circ2_corr_forward.values.flatten()\n",
    "circ2_corr_backward_flat = circ2_corr_backward.values.flatten()\n",
    "\n",
    "circ2_corr_no_shift_flat = circ2_corr_no_shift_flat[~np.isnan(circ2_corr_no_shift_flat)]\n",
    "circ2_corr_forward_flat = circ2_corr_forward_flat[~np.isnan(circ2_corr_forward_flat)]\n",
    "circ2_corr_backward_flat = circ2_corr_backward_flat[~np.isnan(circ2_corr_backward_flat)]\n",
    "\n",
    "axes[0].hist(circ2_corr_no_shift_flat, bins=50, alpha=0.7, edgecolor=\"black\")\n",
    "axes[0].set_title(\"Circadian 2: No Shift\")\n",
    "axes[0].set_xlabel(\"Spearman Correlation\")\n",
    "axes[0].axvline(x=0, color=\"red\", linestyle=\"--\", alpha=0.5)\n",
    "\n",
    "axes[1].hist(\n",
    "    circ2_corr_forward_flat, bins=50, alpha=0.7, edgecolor=\"black\", color=\"green\"\n",
    ")\n",
    "axes[1].set_title(\"Circadian 2: Forward Shift\\n(Transcriptome leads)\")\n",
    "axes[1].set_xlabel(\"Spearman Correlation\")\n",
    "axes[1].axvline(x=0, color=\"red\", linestyle=\"--\", alpha=0.5)\n",
    "\n",
    "axes[2].hist(\n",
    "    circ2_corr_backward_flat, bins=50, alpha=0.7, edgecolor=\"black\", color=\"orange\"\n",
    ")\n",
    "axes[2].set_title(\"Circadian 2: Backward Shift\\n(Microbiome leads)\")\n",
    "axes[2].set_xlabel(\"Spearman Correlation\")\n",
    "axes[2].axvline(x=0, color=\"red\", linestyle=\"--\", alpha=0.5)\n",
    "\n",
    "plt.suptitle(\"Circadian Experiment 2 - Correlation Distributions\", y=1.02)\n",
    "plt.tight_layout()\n",
    "plt.show()\n",
    "\n",
    "print(\n",
    "    f\"Circadian 2 - No shift mean abs correlation: {np.mean(np.abs(circ2_corr_no_shift_flat)):.4f}\"\n",
    ")\n",
    "print(\n",
    "    f\"Circadian 2 - Forward shift mean abs correlation: {np.mean(np.abs(circ2_corr_forward_flat)):.4f}\"\n",
    ")\n",
    "print(\n",
    "    f\"Circadian 2 - Backward shift mean abs correlation: {np.mean(np.abs(circ2_corr_backward_flat)):.4f}\"\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9917ebf4",
   "metadata": {},
   "outputs": [],
   "source": [
    "# # Save Circadian 2 results\n",
    "# output_dir_circ2 = \"/Users/michael/Data/Luke_terrace_experiment/General_data/Junk_data/Time_shifted_correlation/Circadian_2\"\n",
    "# os.makedirs(output_dir_circ2, exist_ok=True)\n",
    "\n",
    "# dbfile = open(f\"{output_dir_circ2}/corr_no_shift_remove_low_exp.p\", \"ab\")\n",
    "# pickle.dump(circ2_corr_no_shift, dbfile)\n",
    "# dbfile.close()\n",
    "\n",
    "# dbfile = open(f\"{output_dir_circ2}/pval_no_shift_remove_low_exp.p\", \"ab\")\n",
    "# pickle.dump(circ2_pval_no_shift, dbfile)\n",
    "# dbfile.close()\n",
    "\n",
    "# dbfile = open(f\"{output_dir_circ2}/tp_no_shift_remove_low_exp.p\", \"ab\")\n",
    "# pickle.dump(circ2_tp_no_shift, dbfile)\n",
    "# dbfile.close()\n",
    "\n",
    "# dbfile = open(f\"{output_dir_circ2}/corr_forward_shift_remove_low_exp.p\", \"ab\")\n",
    "# pickle.dump(circ2_corr_forward, dbfile)\n",
    "# dbfile.close()\n",
    "\n",
    "# dbfile = open(f\"{output_dir_circ2}/pval_forward_shift_remove_low_exp.p\", \"ab\")\n",
    "# pickle.dump(circ2_pval_forward, dbfile)\n",
    "# dbfile.close()\n",
    "\n",
    "# dbfile = open(f\"{output_dir_circ2}/tp_forward_shift_remove_low_exp.p\", \"ab\")\n",
    "# pickle.dump(circ2_tp_forward, dbfile)\n",
    "# dbfile.close()\n",
    "\n",
    "# dbfile = open(f\"{output_dir_circ2}/corr_backward_shift_remove_low_exp.p\", \"ab\")\n",
    "# pickle.dump(circ2_corr_backward, dbfile)\n",
    "# dbfile.close()\n",
    "\n",
    "# dbfile = open(f\"{output_dir_circ2}/pval_backward_shift_remove_low_exp.p\", \"ab\")\n",
    "# pickle.dump(circ2_pval_backward, dbfile)\n",
    "# dbfile.close()\n",
    "\n",
    "# dbfile = open(f\"{output_dir_circ2}/tp_backward_shift_remove_low_exp.p\", \"ab\")\n",
    "# pickle.dump(circ2_tp_backward, dbfile)\n",
    "# dbfile.close()\n",
    "\n",
    "print(f\"Saved Circadian 2 results to {output_dir_circ2}\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Luke_terrace",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
