{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "0656ec42",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import pickle\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "60f4ae72",
   "metadata": {},
   "outputs": [],
   "source": [
    "go_terms_unprocessed = pd.read_csv('/Users/michael/Data/Junk_data/mart_export (1).txt')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "4a62ce11",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/nk/6xkk9sgn1pz4ff1b36sfq3y40000gt/T/ipykernel_56869/2887750374.py:1: DtypeWarning: Columns (16) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  better_go_terms = pd.read_csv(\n"
     ]
    }
   ],
   "source": [
    "better_go_terms = pd.read_csv(\n",
    "    \"/Users/michael/Data/Reference_data/tair.gaf\",\n",
    "    sep=\"\\t\",\n",
    "    comment=\"!\",\n",
    "    names=[\n",
    "        \"Origin\",\n",
    "        \"GeneID\",\n",
    "        \"Gene Name\",\n",
    "        \"Relationship to GO term\",\n",
    "        \"GO term accession\",\n",
    "        \"Reference\",\n",
    "        \"Type of Evidence\",\n",
    "        \"External Group Reference\",\n",
    "        \" Letter Code\",\n",
    "        \"Description\",\n",
    "        \"External References\",\n",
    "        \"Gene Type\",\n",
    "        \"Species\",\n",
    "        \"Locus Code\",\n",
    "        \"Database\",\n",
    "        \"Last\",\n",
    "        \"Test\",\n",
    "    ],\n",
    "    index_col=False,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "fd637c82",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Origin</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Relationship to GO term</th>\n",
       "      <th>GO term accession</th>\n",
       "      <th>Reference</th>\n",
       "      <th>Type of Evidence</th>\n",
       "      <th>External Group Reference</th>\n",
       "      <th>Letter Code</th>\n",
       "      <th>Description</th>\n",
       "      <th>External References</th>\n",
       "      <th>Gene Type</th>\n",
       "      <th>Species</th>\n",
       "      <th>Locus Code</th>\n",
       "      <th>Database</th>\n",
       "      <th>Last</th>\n",
       "      <th>Test</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AGI_LocusCode</td>\n",
       "      <td>AT1G80420</td>\n",
       "      <td>ATXRCC1</td>\n",
       "      <td>involved_in</td>\n",
       "      <td>GO:0000012</td>\n",
       "      <td>TAIR:AnalysisReference:501756966</td>\n",
       "      <td>IEA</td>\n",
       "      <td>InterPro:IPR045080</td>\n",
       "      <td>P</td>\n",
       "      <td>AT1G80420</td>\n",
       "      <td>AT1G80420|ATXRCC1|XRCC1|homolog of X-ray repai...</td>\n",
       "      <td>protein</td>\n",
       "      <td>taxon:3702</td>\n",
       "      <td>20240419</td>\n",
       "      <td>InterPro</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TAIR:locus:2198973</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AGI_LocusCode</td>\n",
       "      <td>AT1G67120</td>\n",
       "      <td>MDN1</td>\n",
       "      <td>involved_in</td>\n",
       "      <td>GO:0000027</td>\n",
       "      <td>TAIR:AnalysisReference:501756966</td>\n",
       "      <td>IEA</td>\n",
       "      <td>InterPro:IPR012099</td>\n",
       "      <td>P</td>\n",
       "      <td>AT1G67120</td>\n",
       "      <td>AT1G67120|MDN1|AtMDN1|MIDASIN 1|F5A8.11|F5A8_11</td>\n",
       "      <td>protein</td>\n",
       "      <td>taxon:3702</td>\n",
       "      <td>20250907</td>\n",
       "      <td>InterPro</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TAIR:locus:2033661</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AGI_LocusCode</td>\n",
       "      <td>AT2G40430</td>\n",
       "      <td>SMO4</td>\n",
       "      <td>involved_in</td>\n",
       "      <td>GO:0000027</td>\n",
       "      <td>TAIR:Communication:501789215</td>\n",
       "      <td>ISS</td>\n",
       "      <td>UniProtKB:Q9NZM5</td>\n",
       "      <td>P</td>\n",
       "      <td>AT2G40430</td>\n",
       "      <td>AT2G40430|SMO4|SMALL ORGAN 4|T2P4.22</td>\n",
       "      <td>protein</td>\n",
       "      <td>taxon:3702</td>\n",
       "      <td>20210331</td>\n",
       "      <td>UniProt</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TAIR:locus:2061898</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AGI_LocusCode</td>\n",
       "      <td>AT1G54630</td>\n",
       "      <td>ACP3</td>\n",
       "      <td>enables</td>\n",
       "      <td>GO:0000036</td>\n",
       "      <td>TAIR:AnalysisReference:501756966</td>\n",
       "      <td>IEA</td>\n",
       "      <td>InterPro:IPR044813</td>\n",
       "      <td>F</td>\n",
       "      <td>AT1G54630</td>\n",
       "      <td>AT1G54630|ACP3|AtACP3|acyl carrier protein 3|T...</td>\n",
       "      <td>protein</td>\n",
       "      <td>taxon:3702</td>\n",
       "      <td>20240419</td>\n",
       "      <td>InterPro</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TAIR:locus:2199517</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AGI_LocusCode</td>\n",
       "      <td>AT4G25050</td>\n",
       "      <td>ACP4</td>\n",
       "      <td>enables</td>\n",
       "      <td>GO:0000036</td>\n",
       "      <td>TAIR:AnalysisReference:501756966</td>\n",
       "      <td>IEA</td>\n",
       "      <td>InterPro:IPR044813</td>\n",
       "      <td>F</td>\n",
       "      <td>AT4G25050</td>\n",
       "      <td>AT4G25050|ACP4|AtACP4|acyl carrier protein 4|F...</td>\n",
       "      <td>protein</td>\n",
       "      <td>taxon:3702</td>\n",
       "      <td>20240419</td>\n",
       "      <td>InterPro</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TAIR:locus:2117323</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>227206</th>\n",
       "      <td>TAIR</td>\n",
       "      <td>locus:2178148</td>\n",
       "      <td>LIP2</td>\n",
       "      <td>involved_in</td>\n",
       "      <td>GO:0006886</td>\n",
       "      <td>GO_REF:0000033</td>\n",
       "      <td>IBA</td>\n",
       "      <td>PANTHER:PTN001292747|SGD:S000001722|CGD:CAL000...</td>\n",
       "      <td>P</td>\n",
       "      <td>Small GTPase-like protein LIP2</td>\n",
       "      <td>UniProtKB:F4KFD8|PTN000633497</td>\n",
       "      <td>protein</td>\n",
       "      <td>taxon:3702</td>\n",
       "      <td>20250903</td>\n",
       "      <td>GO_Central</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>227207</th>\n",
       "      <td>TAIR</td>\n",
       "      <td>locus:2066107</td>\n",
       "      <td>GLR2.7</td>\n",
       "      <td>is_active_in</td>\n",
       "      <td>GO:0005886</td>\n",
       "      <td>GO_REF:0000033</td>\n",
       "      <td>IBA</td>\n",
       "      <td>PANTHER:PTN000437926|RGD:621705|MGI:MGI:95810|...</td>\n",
       "      <td>C</td>\n",
       "      <td>Glutamate receptor 2.7</td>\n",
       "      <td>UniProtKB:Q8LGN0|PTN000438602</td>\n",
       "      <td>protein</td>\n",
       "      <td>taxon:3702</td>\n",
       "      <td>20250903</td>\n",
       "      <td>GO_Central</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>227208</th>\n",
       "      <td>TAIR</td>\n",
       "      <td>locus:2153904</td>\n",
       "      <td>At5g24510</td>\n",
       "      <td>part_of</td>\n",
       "      <td>GO:0022625</td>\n",
       "      <td>GO_REF:0000033</td>\n",
       "      <td>IBA</td>\n",
       "      <td>PANTHER:PTN000474938|MGI:MGI:1927099|SGD:S0000...</td>\n",
       "      <td>C</td>\n",
       "      <td>60S acidic ribosomal protein family</td>\n",
       "      <td>UniProtKB:Q9FLV1|PTN000475055</td>\n",
       "      <td>protein</td>\n",
       "      <td>taxon:3702</td>\n",
       "      <td>20211218</td>\n",
       "      <td>GO_Central</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>227209</th>\n",
       "      <td>TAIR</td>\n",
       "      <td>locus:2174819</td>\n",
       "      <td>FTSHI4</td>\n",
       "      <td>is_active_in</td>\n",
       "      <td>GO:0009535</td>\n",
       "      <td>GO_REF:0000033</td>\n",
       "      <td>IBA</td>\n",
       "      <td>PANTHER:PTN004602537|TAIR:locus:2174819|TAIR:l...</td>\n",
       "      <td>C</td>\n",
       "      <td>Probable inactive ATP-dependent zinc metallopr...</td>\n",
       "      <td>UniProtKB:F4KF14|PTN000554248</td>\n",
       "      <td>protein</td>\n",
       "      <td>taxon:3702</td>\n",
       "      <td>20240307</td>\n",
       "      <td>GO_Central</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>227210</th>\n",
       "      <td>TAIR</td>\n",
       "      <td>locus:2161008</td>\n",
       "      <td>GAMT2</td>\n",
       "      <td>enables</td>\n",
       "      <td>GO:0008757</td>\n",
       "      <td>GO_REF:0000033</td>\n",
       "      <td>IBA</td>\n",
       "      <td>PANTHER:PTN001256450|TAIR:locus:2161680|TAIR:l...</td>\n",
       "      <td>F</td>\n",
       "      <td>Gibberellic acid methyltransferase 2</td>\n",
       "      <td>UniProtKB:Q5XF78|PTN000772549</td>\n",
       "      <td>protein</td>\n",
       "      <td>taxon:3702</td>\n",
       "      <td>20250904</td>\n",
       "      <td>GO_Central</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>227211 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               Origin         GeneID  Gene Name Relationship to GO term  \\\n",
       "0       AGI_LocusCode      AT1G80420    ATXRCC1             involved_in   \n",
       "1       AGI_LocusCode      AT1G67120       MDN1             involved_in   \n",
       "2       AGI_LocusCode      AT2G40430       SMO4             involved_in   \n",
       "3       AGI_LocusCode      AT1G54630       ACP3                 enables   \n",
       "4       AGI_LocusCode      AT4G25050       ACP4                 enables   \n",
       "...               ...            ...        ...                     ...   \n",
       "227206           TAIR  locus:2178148       LIP2             involved_in   \n",
       "227207           TAIR  locus:2066107     GLR2.7            is_active_in   \n",
       "227208           TAIR  locus:2153904  At5g24510                 part_of   \n",
       "227209           TAIR  locus:2174819     FTSHI4            is_active_in   \n",
       "227210           TAIR  locus:2161008      GAMT2                 enables   \n",
       "\n",
       "       GO term accession                         Reference Type of Evidence  \\\n",
       "0             GO:0000012  TAIR:AnalysisReference:501756966              IEA   \n",
       "1             GO:0000027  TAIR:AnalysisReference:501756966              IEA   \n",
       "2             GO:0000027      TAIR:Communication:501789215              ISS   \n",
       "3             GO:0000036  TAIR:AnalysisReference:501756966              IEA   \n",
       "4             GO:0000036  TAIR:AnalysisReference:501756966              IEA   \n",
       "...                  ...                               ...              ...   \n",
       "227206        GO:0006886                    GO_REF:0000033              IBA   \n",
       "227207        GO:0005886                    GO_REF:0000033              IBA   \n",
       "227208        GO:0022625                    GO_REF:0000033              IBA   \n",
       "227209        GO:0009535                    GO_REF:0000033              IBA   \n",
       "227210        GO:0008757                    GO_REF:0000033              IBA   \n",
       "\n",
       "                                 External Group Reference  Letter Code  \\\n",
       "0                                      InterPro:IPR045080            P   \n",
       "1                                      InterPro:IPR012099            P   \n",
       "2                                        UniProtKB:Q9NZM5            P   \n",
       "3                                      InterPro:IPR044813            F   \n",
       "4                                      InterPro:IPR044813            F   \n",
       "...                                                   ...          ...   \n",
       "227206  PANTHER:PTN001292747|SGD:S000001722|CGD:CAL000...            P   \n",
       "227207  PANTHER:PTN000437926|RGD:621705|MGI:MGI:95810|...            C   \n",
       "227208  PANTHER:PTN000474938|MGI:MGI:1927099|SGD:S0000...            C   \n",
       "227209  PANTHER:PTN004602537|TAIR:locus:2174819|TAIR:l...            C   \n",
       "227210  PANTHER:PTN001256450|TAIR:locus:2161680|TAIR:l...            F   \n",
       "\n",
       "                                              Description  \\\n",
       "0                                               AT1G80420   \n",
       "1                                               AT1G67120   \n",
       "2                                               AT2G40430   \n",
       "3                                               AT1G54630   \n",
       "4                                               AT4G25050   \n",
       "...                                                   ...   \n",
       "227206                     Small GTPase-like protein LIP2   \n",
       "227207                             Glutamate receptor 2.7   \n",
       "227208                60S acidic ribosomal protein family   \n",
       "227209  Probable inactive ATP-dependent zinc metallopr...   \n",
       "227210               Gibberellic acid methyltransferase 2   \n",
       "\n",
       "                                      External References Gene Type  \\\n",
       "0       AT1G80420|ATXRCC1|XRCC1|homolog of X-ray repai...   protein   \n",
       "1         AT1G67120|MDN1|AtMDN1|MIDASIN 1|F5A8.11|F5A8_11   protein   \n",
       "2                    AT2G40430|SMO4|SMALL ORGAN 4|T2P4.22   protein   \n",
       "3       AT1G54630|ACP3|AtACP3|acyl carrier protein 3|T...   protein   \n",
       "4       AT4G25050|ACP4|AtACP4|acyl carrier protein 4|F...   protein   \n",
       "...                                                   ...       ...   \n",
       "227206                      UniProtKB:F4KFD8|PTN000633497   protein   \n",
       "227207                      UniProtKB:Q8LGN0|PTN000438602   protein   \n",
       "227208                      UniProtKB:Q9FLV1|PTN000475055   protein   \n",
       "227209                      UniProtKB:F4KF14|PTN000554248   protein   \n",
       "227210                      UniProtKB:Q5XF78|PTN000772549   protein   \n",
       "\n",
       "           Species  Locus Code    Database  Last                Test  \n",
       "0       taxon:3702    20240419    InterPro   NaN  TAIR:locus:2198973  \n",
       "1       taxon:3702    20250907    InterPro   NaN  TAIR:locus:2033661  \n",
       "2       taxon:3702    20210331     UniProt   NaN  TAIR:locus:2061898  \n",
       "3       taxon:3702    20240419    InterPro   NaN  TAIR:locus:2199517  \n",
       "4       taxon:3702    20240419    InterPro   NaN  TAIR:locus:2117323  \n",
       "...            ...         ...         ...   ...                 ...  \n",
       "227206  taxon:3702    20250903  GO_Central   NaN                 NaN  \n",
       "227207  taxon:3702    20250903  GO_Central   NaN                 NaN  \n",
       "227208  taxon:3702    20211218  GO_Central   NaN                 NaN  \n",
       "227209  taxon:3702    20240307  GO_Central   NaN                 NaN  \n",
       "227210  taxon:3702    20250904  GO_Central   NaN                 NaN  \n",
       "\n",
       "[227211 rows x 17 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "better_go_terms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "4f3fff00",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneID</th>\n",
       "      <th>GO term accession</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AT1G80420</td>\n",
       "      <td>GO:0000012</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AT1G67120</td>\n",
       "      <td>GO:0000027</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AT2G40430</td>\n",
       "      <td>GO:0000027</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AT1G54630</td>\n",
       "      <td>GO:0000036</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AT4G25050</td>\n",
       "      <td>GO:0000036</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>186522</th>\n",
       "      <td>AT1G54370</td>\n",
       "      <td>GO:0010351</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>186523</th>\n",
       "      <td>AT5G27150</td>\n",
       "      <td>GO:0010351</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>186524</th>\n",
       "      <td>AT1G54370</td>\n",
       "      <td>GO:0015081</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>186525</th>\n",
       "      <td>AT5G27150</td>\n",
       "      <td>GO:0015081</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>186526</th>\n",
       "      <td>AT5G27150</td>\n",
       "      <td>GO:0035725</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>180281 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           GeneID GO term accession\n",
       "0       AT1G80420        GO:0000012\n",
       "1       AT1G67120        GO:0000027\n",
       "2       AT2G40430        GO:0000027\n",
       "3       AT1G54630        GO:0000036\n",
       "4       AT4G25050        GO:0000036\n",
       "...           ...               ...\n",
       "186522  AT1G54370        GO:0010351\n",
       "186523  AT5G27150        GO:0010351\n",
       "186524  AT1G54370        GO:0015081\n",
       "186525  AT5G27150        GO:0015081\n",
       "186526  AT5G27150        GO:0035725\n",
       "\n",
       "[180281 rows x 2 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "go_terms_unprocessed = better_go_terms[['GeneID','GO term accession']]\n",
    "go_terms_unprocessed = go_terms_unprocessed.loc[(go_terms_unprocessed['GeneID'].str.contains('AT')== True) & (go_terms_unprocessed['GeneID'].str.contains('Q')== False)]\n",
    "go_terms_unprocessed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "008a4b8a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# go_terms_unprocessed = go_terms_unprocessed.rename(columns = {'Gene stable ID':'GeneID',})\n",
    "# go_terms_unprocessed = go_terms_unprocessed.drop(columns = ['Transcript stable ID','GO domain','GO term evidence code',])\n",
    "# go_terms_unprocessed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "f10a8a23",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['GO:0000012', 'GO:0000027', 'GO:0000036', ..., 'GO:1901177',\n",
       "       'GO:0008092', 'GO:0035725'], shape=(6069,), dtype=object)"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "go_terms_unprocessed['GO term accession'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "b2d8c9b4",
   "metadata": {},
   "outputs": [],
   "source": [
    "unusable_samples = [\n",
    "    \"A2450525897_n01_LICRNA01_A06\",\n",
    "    \"A2450525897_n01_LICRNA01_A08\",\n",
    "    \"A2450525897_n01_LICRNA01_C11\",\n",
    "    \"A2450525897_n01_LICRNA01_D11\",\n",
    "    \"A2450525897_n01_LICRNA01_D12\",\n",
    "    \"A2449446903_n01_LICRNA02_F01\",\n",
    "    \"B2449500127_n01_LICRNA04_A04\",\n",
    "    \"B2449500127_n01_LICRNA04_A06\",\n",
    "    \"B2449500127_n01_LICRNA04_A07\",\n",
    "\n",
    "]\n",
    "borderline_unusable = ['A2450525897_n01_LICRNA01_F03', 'A2450525897_n01_LICRNA01_G07','A2449446903_n01_LICRNA02_A04','A2449446903_n01_LICRNA02_H10',\"B2449500127_n01_LICRNA04_H02\",\"B2449500127_n01_LICRNA04_H11\"]\n",
    "all_unsable = borderline_unusable+unusable_samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "3e84aefb",
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata = pd.read_csv('/Users/michael/Data/Luke_terrace_experiment/General_data/Plates_1_to_4_metadata_merged_luke.csv', index_col = 0,)\n",
    "metadata = metadata.drop(columns= ['arb.sort','sample-id','Ambiguous Unstranded','Ambiguous Forward','Multimapping','Unmapped Over Mapped'])\n",
    "metadata['Date and Time'] = metadata['date'] +' ' +  metadata['time']\n",
    "luke_time_data_format = \"%-m/%-d/%y %-H:%-M\"\n",
    "metadata['Date and Time'] = pd.to_datetime(metadata['Date and Time'], format = luke_time_data_format)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "23e3a939",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "526cab1c",
   "metadata": {},
   "outputs": [],
   "source": [
    "transcriptome = pd.read_csv('/Users/michael/Data/Luke_terrace_experiment/General_data/plate1_4_tpm_normalized.csv', index_col = 0)\n",
    "rows_to_drop_expression_data = ['A2450525897_n01_undetermined',\n",
    "                'A2449446903_n01_undetermined',\n",
    "                'B250508004_n01_undetermined',\n",
    "                'B2449500127_n01_undetermined']\n",
    "transcriptome = transcriptome.drop(index = rows_to_drop_expression_data)\n",
    "transcriptome = transcriptome.sort_index()\n",
    "trimmed_transcriptome = transcriptome.drop(index = all_unsable)\n",
    "trimmed_metadata = metadata.drop(index= all_unsable)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "1c1c4852",
   "metadata": {},
   "outputs": [],
   "source": [
    "go_terms_trimmed = go_terms_unprocessed.loc[go_terms_unprocessed['GeneID'].isin(transcriptome.T.index)]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "4caebfe5",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/nk/6xkk9sgn1pz4ff1b36sfq3y40000gt/T/ipykernel_56869/3147939452.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  go_terms_trimmed ['Present'] = 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>GO term accession</th>\n",
       "      <th>GO:0000001</th>\n",
       "      <th>GO:0000009</th>\n",
       "      <th>GO:0000012</th>\n",
       "      <th>GO:0000014</th>\n",
       "      <th>GO:0000023</th>\n",
       "      <th>GO:0000024</th>\n",
       "      <th>GO:0000025</th>\n",
       "      <th>GO:0000026</th>\n",
       "      <th>GO:0000027</th>\n",
       "      <th>GO:0000028</th>\n",
       "      <th>...</th>\n",
       "      <th>GO:2001071</th>\n",
       "      <th>GO:2001080</th>\n",
       "      <th>GO:2001141</th>\n",
       "      <th>GO:2001142</th>\n",
       "      <th>GO:2001143</th>\n",
       "      <th>GO:2001147</th>\n",
       "      <th>GO:2001227</th>\n",
       "      <th>GO:2001280</th>\n",
       "      <th>GO:2001289</th>\n",
       "      <th>GO:2001294</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GeneID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AT1G01010</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT1G01020</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT1G01030</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT1G01040</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT1G01046</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT5G67600</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT5G67610</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT5G67620</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT5G67630</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT5G67640</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>28537 rows × 6055 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "GO term accession  GO:0000001  GO:0000009  GO:0000012  GO:0000014  GO:0000023  \\\n",
       "GeneID                                                                          \n",
       "AT1G01010                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT1G01020                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT1G01030                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT1G01040                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT1G01046                 0.0         0.0         0.0         0.0         0.0   \n",
       "...                       ...         ...         ...         ...         ...   \n",
       "AT5G67600                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT5G67610                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT5G67620                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT5G67630                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT5G67640                 0.0         0.0         0.0         0.0         0.0   \n",
       "\n",
       "GO term accession  GO:0000024  GO:0000025  GO:0000026  GO:0000027  GO:0000028  \\\n",
       "GeneID                                                                          \n",
       "AT1G01010                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT1G01020                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT1G01030                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT1G01040                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT1G01046                 0.0         0.0         0.0         0.0         0.0   \n",
       "...                       ...         ...         ...         ...         ...   \n",
       "AT5G67600                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT5G67610                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT5G67620                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT5G67630                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT5G67640                 0.0         0.0         0.0         0.0         0.0   \n",
       "\n",
       "GO term accession  ...  GO:2001071  GO:2001080  GO:2001141  GO:2001142  \\\n",
       "GeneID             ...                                                   \n",
       "AT1G01010          ...         0.0         0.0         0.0         0.0   \n",
       "AT1G01020          ...         0.0         0.0         0.0         0.0   \n",
       "AT1G01030          ...         0.0         0.0         0.0         0.0   \n",
       "AT1G01040          ...         0.0         0.0         0.0         0.0   \n",
       "AT1G01046          ...         0.0         0.0         0.0         0.0   \n",
       "...                ...         ...         ...         ...         ...   \n",
       "AT5G67600          ...         0.0         0.0         0.0         0.0   \n",
       "AT5G67610          ...         0.0         0.0         0.0         0.0   \n",
       "AT5G67620          ...         0.0         0.0         0.0         0.0   \n",
       "AT5G67630          ...         0.0         0.0         0.0         0.0   \n",
       "AT5G67640          ...         0.0         0.0         0.0         0.0   \n",
       "\n",
       "GO term accession  GO:2001143  GO:2001147  GO:2001227  GO:2001280  GO:2001289  \\\n",
       "GeneID                                                                          \n",
       "AT1G01010                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT1G01020                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT1G01030                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT1G01040                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT1G01046                 0.0         0.0         0.0         0.0         0.0   \n",
       "...                       ...         ...         ...         ...         ...   \n",
       "AT5G67600                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT5G67610                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT5G67620                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT5G67630                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT5G67640                 0.0         0.0         0.0         0.0         0.0   \n",
       "\n",
       "GO term accession  GO:2001294  \n",
       "GeneID                         \n",
       "AT1G01010                 0.0  \n",
       "AT1G01020                 0.0  \n",
       "AT1G01030                 0.0  \n",
       "AT1G01040                 0.0  \n",
       "AT1G01046                 0.0  \n",
       "...                       ...  \n",
       "AT5G67600                 0.0  \n",
       "AT5G67610                 0.0  \n",
       "AT5G67620                 0.0  \n",
       "AT5G67630                 0.0  \n",
       "AT5G67640                 0.0  \n",
       "\n",
       "[28537 rows x 6055 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "go_terms_trimmed ['Present'] = 1\n",
    "go_terms_clean = go_terms_trimmed[['GeneID','GO term accession','Present']]\n",
    "go_terms_clean = go_terms_clean.drop_duplicates()\n",
    "\n",
    "matrixed_version = go_terms_clean.pivot(index='GeneID',\n",
    "                                        columns='GO term accession',\n",
    "                                        values='Present')\n",
    "\n",
    "\n",
    "\n",
    "filled_matrixed_version = matrixed_version.fillna(0)\n",
    "filled_matrixed_version"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "460d7c8d",
   "metadata": {},
   "outputs": [],
   "source": [
    "def generate_contigency_tables(go_annotations,gene_list, use_background_genes = False, background_genes =[]):\n",
    "    \"\"\"Takes list of Go Annotations as a binary matrix with Genes as the index and GO terms as columns, and a list of genes, and \n",
    "    generates the contigency tables on which to run a Fisher Exact Test for enrichment. \n",
    "    Args:\n",
    "        go_annotations (Panda DataFrame): Binary Matrix, with genes as index and GO Terms as colummns. Generated by Go_annotations function (may not\n",
    "        work for non plant species idk check the code for it its been a while)\n",
    "        gene_list (list): What is says on the tin\n",
    "    \"\"\"\n",
    "\n",
    "    if use_background_genes == True:\n",
    "        go_annotations = go_annotations.loc[go_annotations.index.intersection(background_genes)]\n",
    "        \n",
    "    gene_list = list(set(go_annotations.index) & set(gene_list))\n",
    "    background_distribution_num_genes_annotated_with_go_term = go_annotations.sum(axis =0)\n",
    "    background_dipstribution_num_genes_that_lack_go_term = go_annotations.shape[0]-background_distribution_num_genes_annotated_with_go_term\n",
    "    annotation_subset_to_genes_in_list = go_annotations.loc[gene_list]\n",
    "    go_subset_distribution_num_genes_annotatated_with_go_term = annotation_subset_to_genes_in_list.sum(axis =0)\n",
    "    go_subset_distribution_num_genes_that_lack_go_term = annotation_subset_to_genes_in_list.shape[0] - go_subset_distribution_num_genes_annotatated_with_go_term\n",
    "    \n",
    "    all_go_contingency_tables = []\n",
    "    i = 0\n",
    "    for term_of_go in background_distribution_num_genes_annotated_with_go_term:\n",
    "        contig_table = [[(term_of_go - go_subset_distribution_num_genes_annotatated_with_go_term[i]),go_subset_distribution_num_genes_annotatated_with_go_term[i]],[(background_dipstribution_num_genes_that_lack_go_term[i]-go_subset_distribution_num_genes_that_lack_go_term[i]),go_subset_distribution_num_genes_that_lack_go_term[i]]]\n",
    "        all_go_contingency_tables.append(contig_table)\n",
    "        i +=1\n",
    "    return all_go_contingency_tables\n",
    "\n",
    "\n",
    "def fishers_exact_on_contigency_tables(all_go_contigency_tables,original_GO_term_panda, return_odds_ratio = False):\n",
    "    \"\"\"Takes Table from generate_contigency_tables and runs fishers exact on all the tables, without correcting the P-values. Returns\n",
    "    either just a Panda with the p values, or a Panda of lists for odds ratios, P values, just like stats.fisher_exact. If you're wondering if \n",
    "    you want the Odds Ratios, you don't want them\n",
    "\n",
    "    Args:\n",
    "        all_go_contigency_tables (list): Contigency Tables formatted as list of lists from generate_contigency_tables\n",
    "        return_odds_ratio (bool): Boolean for if you want it to return both the odds ratios and the P values or just the P Values. Defaults to just the \n",
    "        P Values\n",
    "        original_GO_term_panda (Panda DataFrame):  Binary Matrix, with genes as index and GO Terms as colummns. Generated by Go_annotations function (may not\n",
    "        work for non plant species idk check the code for it its been a while)  Same as above\n",
    "    \"\"\"\n",
    "    import scipy.stats as stats\n",
    "    import pandas as pd\n",
    "\n",
    "    list_of_p_values_for_go_terms = []\n",
    "    list_of_odds_ratios_for_go_terms = []\n",
    "\n",
    "    for table in all_go_contigency_tables:\n",
    "        odds, p_value = stats.fisher_exact(table = table, alternative = 'less')\n",
    "        list_of_p_values_for_go_terms.append(p_value)\n",
    "        list_of_odds_ratios_for_go_terms.append(odds)\n",
    "\n",
    "    if return_odds_ratio == True:\n",
    "\n",
    "        go_terms_with_P_value_and_odds = pd.DataFrame(data = list(zip(list_of_p_values_for_go_terms, list_of_odds_ratios_for_go_terms)),index = original_GO_term_panda.columns, columns = ['P_value', 'Odds Ratios'])\n",
    "        return go_terms_with_P_value_and_odds\n",
    "\n",
    "    if return_odds_ratio == False:\n",
    "\n",
    "        go_terms_with_P_value = pd.DataFrame(data = list_of_p_values_for_go_terms,index = original_GO_term_panda.columns, columns = ['P_value'])\n",
    "        return go_terms_with_P_value\n",
    "\n",
    "        \n",
    "def multi_hypothesis_correct_fishers_exact(go_terms_with_P_value, alpha = 0.05, method = 'indep', is_sorted = False, panda_has_odds_ratio = False, Include_rejected_boolean_array = False):\n",
    "    \"\"\"Runs Multiple Hypothesis Testing via statsmodels.stats.multitest.fdrcorrection. Takes all values for that, as well as Panda of go terms from previous step\n",
    "\n",
    "    Args:\n",
    "        go_terms_with_P_value (Panda Dataframe): Pandas Dataframe from fishers_exact_on_contigency_tables\n",
    "        alpha (float): Familywise error rate, defaults to 0.05\n",
    "        method (str): Which method to use for FDR correction. {'i', 'indep', 'p', 'poscorr'} all refer to fdr_bh (Benjamini/Hochberg for independent or positively correlated tests).\n",
    "         {'n', 'negcorr'} both refer to fdr_by (Benjamini/Yekutieli for general or negatively correlated tests). Defaults to 'indep'.\n",
    "        is_sorted (bool): _f False (default), the p_values will be sorted, but the corrected pvalues are in the original order. If True, then it assumed that the pvalues are already sorted in ascending order.\n",
    "        panda_has_odds_ratio (bool): Did you ignore my previous Docstring in fishers_exact_on_contigency_tables and add the odds ratio? If so this fixes \n",
    "        your foolishness and drops that column\n",
    "        Include_rejected_boolean_array (bool): Makes function return a tuple of a boolean array of if a hypothesis was rejected, and the normal Panda of corrected_p_values. Don't do this just use your eyes\n",
    "    \"\"\"\n",
    "    import statsmodels.stats.multitest   \n",
    "    import pandas as pd\n",
    "\n",
    "    if panda_has_odds_ratio == True:\n",
    "        go_terms_with_P_value = go_terms_with_P_value[['P_value']]\n",
    "    \n",
    "    rejected, corrected_pvalues = statsmodels.stats.multitest.fdrcorrection(pvals = go_terms_with_P_value['P_value'].to_list(), alpha = alpha, method = method, is_sorted= is_sorted)\n",
    "    corrected_pvalues = corrected_pvalues.tolist()\n",
    "    go_enrichment_results = pd.DataFrame(corrected_pvalues,index = go_terms_with_P_value.index,columns=['P_value'])\n",
    "    \n",
    "    if Include_rejected_boolean_array == True:\n",
    "        return rejected, go_enrichment_results\n",
    "    \n",
    "    if Include_rejected_boolean_array == False:\n",
    "        return go_enrichment_results\n",
    "\n",
    "def test_using_kwargs(go_terms_with_P_value,panda_has_odds_ratio = False, Include_rejected_boolean_array = False, **kwargs):\n",
    "    \"\"\"Runs Multiple Hypothesis Testing via statsmodels.stats.multitest.fdrcorrection. Takes all values for that, as well as Panda of go terms from previous step\n",
    "\n",
    "    Args:\n",
    "        go_terms_with_P_value (Panda Dataframe): Pandas Dataframe from fishers_exact_on_contigency_tables\n",
    "        kwargs : Refer to statsmodels.stats.multitest.fdrcorrection for kwargs\n",
    "        alpha (float): Familywise error rate, defaults to 0.05\n",
    "        method (str): Which method to use for FDR correction. {'i', 'indep', 'p', 'poscorr'} all refer to fdr_bh (Benjamini/Hochberg for independent or positively correlated tests).\n",
    "         {'n', 'negcorr'} both refer to fdr_by (Benjamini/Yekutieli for general or negatively correlated tests). Defaults to 'indep'.\n",
    "        is_sorted (bool): _f False (default), the p_values will be sorted, but the corrected pvalues are in the original order. If True, then it assumed that the pvalues are already sorted in ascending order.\n",
    "        panda_has_odds_ratio (bool): Did you ignore my previous Docstring in fishers_exact_on_contigency_tables and add the odds ratio? If so this fixes \n",
    "        your foolishness and drops that column\n",
    "        Include_rejected_boolean_array (bool): Makes function return a tuple of a boolean array of if a hypothesis was rejected, and the normal Panda of corrected_p_values. Don't do this just use your eyes\n",
    "    \"\"\"\n",
    "    import statsmodels.stats.multitest   \n",
    "    import pandas as pd\n",
    "\n",
    "    if panda_has_odds_ratio == True:\n",
    "        go_terms_with_P_value = go_terms_with_P_value[['P_value']]\n",
    "    \n",
    "    rejected, corrected_pvalues = statsmodels.stats.multitest.fdrcorrection(pvals = go_terms_with_P_value['P_value'].to_list(), **kwargs)\n",
    "    go_enrichment_results = pd.DataFrame(corrected_pvalues,index = go_terms_with_P_value.index,columns=['P_value'])\n",
    "    \n",
    "    if Include_rejected_boolean_array == True:\n",
    "        return rejected, go_enrichment_results\n",
    "    \n",
    "    if Include_rejected_boolean_array == False:\n",
    "        return go_enrichment_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9fb61bea",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "478436f2",
   "metadata": {},
   "outputs": [],
   "source": [
    "long_term_transcriptome = trimmed_transcriptome.loc[trimmed_metadata['Experiment Type'] == 'Long Term']\n",
    "long_term_metadata = trimmed_metadata.loc[trimmed_metadata['Experiment Type'] == 'Long Term']\n",
    "\n",
    "groupby_timepoint = long_term_transcriptome.groupby(long_term_metadata['timepoint']).mean()\n",
    "short_transcriptome = groupby_timepoint.loc[:, (groupby_timepoint  > 1).mean() > .5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "0e672305",
   "metadata": {},
   "outputs": [],
   "source": [
    "dbfile_2 = open('/Users/michael/Data/Junk_data/gene_spearman_correlation_with_bacteria.p', 'rb')    \n",
    "full_spearman_correlation_matrix = pickle.load(dbfile_2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "b40e275c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AT1G01010</th>\n",
       "      <th>AT1G01020</th>\n",
       "      <th>AT1G01030</th>\n",
       "      <th>AT1G01040</th>\n",
       "      <th>AT1G01050</th>\n",
       "      <th>AT1G01060</th>\n",
       "      <th>AT1G01070</th>\n",
       "      <th>AT1G01080</th>\n",
       "      <th>AT1G01090</th>\n",
       "      <th>AT1G01100</th>\n",
       "      <th>...</th>\n",
       "      <th>ArthCt115</th>\n",
       "      <th>ArthCr091</th>\n",
       "      <th>ArthCr090</th>\n",
       "      <th>ArthCr089</th>\n",
       "      <th>ArthCt114</th>\n",
       "      <th>ArthCt124</th>\n",
       "      <th>ArthCr088</th>\n",
       "      <th>ArthCt112</th>\n",
       "      <th>ArthCp088</th>\n",
       "      <th>ArthCp085</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genus</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Actinomycetospora</th>\n",
       "      <td>-0.037245</td>\n",
       "      <td>0.359304</td>\n",
       "      <td>0.215254</td>\n",
       "      <td>0.235246</td>\n",
       "      <td>0.114200</td>\n",
       "      <td>0.003286</td>\n",
       "      <td>-0.002191</td>\n",
       "      <td>0.130631</td>\n",
       "      <td>0.230590</td>\n",
       "      <td>0.135835</td>\n",
       "      <td>...</td>\n",
       "      <td>0.036706</td>\n",
       "      <td>-0.097768</td>\n",
       "      <td>0.170889</td>\n",
       "      <td>0.095577</td>\n",
       "      <td>0.131179</td>\n",
       "      <td>-0.125428</td>\n",
       "      <td>-0.003286</td>\n",
       "      <td>0.156532</td>\n",
       "      <td>-0.035054</td>\n",
       "      <td>-0.252499</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Actinoplanes</th>\n",
       "      <td>-0.009582</td>\n",
       "      <td>0.145106</td>\n",
       "      <td>0.005476</td>\n",
       "      <td>-0.004107</td>\n",
       "      <td>0.088706</td>\n",
       "      <td>-0.125394</td>\n",
       "      <td>0.113895</td>\n",
       "      <td>-0.088159</td>\n",
       "      <td>-0.040246</td>\n",
       "      <td>-0.098015</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.140437</td>\n",
       "      <td>0.078303</td>\n",
       "      <td>-0.158248</td>\n",
       "      <td>-0.158248</td>\n",
       "      <td>0.205886</td>\n",
       "      <td>-0.073648</td>\n",
       "      <td>-0.223682</td>\n",
       "      <td>-0.026972</td>\n",
       "      <td>-0.015880</td>\n",
       "      <td>-0.301164</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Aeromicrobium</th>\n",
       "      <td>0.065863</td>\n",
       "      <td>0.026906</td>\n",
       "      <td>-0.052691</td>\n",
       "      <td>0.223375</td>\n",
       "      <td>-0.054372</td>\n",
       "      <td>0.034753</td>\n",
       "      <td>0.161155</td>\n",
       "      <td>0.287557</td>\n",
       "      <td>0.197590</td>\n",
       "      <td>0.440024</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.174832</td>\n",
       "      <td>-0.371358</td>\n",
       "      <td>-0.019058</td>\n",
       "      <td>0.158353</td>\n",
       "      <td>0.068106</td>\n",
       "      <td>-0.177691</td>\n",
       "      <td>0.044283</td>\n",
       "      <td>-0.233777</td>\n",
       "      <td>-0.132287</td>\n",
       "      <td>-0.365192</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Aetherobacter</th>\n",
       "      <td>-0.244510</td>\n",
       "      <td>-0.247898</td>\n",
       "      <td>-0.351801</td>\n",
       "      <td>-0.073692</td>\n",
       "      <td>-0.308037</td>\n",
       "      <td>-0.011858</td>\n",
       "      <td>0.066916</td>\n",
       "      <td>-0.609016</td>\n",
       "      <td>-0.611840</td>\n",
       "      <td>-0.662379</td>\n",
       "      <td>...</td>\n",
       "      <td>0.180109</td>\n",
       "      <td>0.345307</td>\n",
       "      <td>0.479703</td>\n",
       "      <td>0.476315</td>\n",
       "      <td>0.264274</td>\n",
       "      <td>0.387376</td>\n",
       "      <td>0.432834</td>\n",
       "      <td>0.389971</td>\n",
       "      <td>0.029364</td>\n",
       "      <td>0.239146</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Agrobacterium</th>\n",
       "      <td>0.209748</td>\n",
       "      <td>0.263417</td>\n",
       "      <td>-0.050931</td>\n",
       "      <td>-0.096112</td>\n",
       "      <td>0.039704</td>\n",
       "      <td>0.314075</td>\n",
       "      <td>0.283954</td>\n",
       "      <td>-0.158817</td>\n",
       "      <td>-0.076397</td>\n",
       "      <td>-0.107338</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.166947</td>\n",
       "      <td>0.105422</td>\n",
       "      <td>-0.040526</td>\n",
       "      <td>0.014513</td>\n",
       "      <td>-0.044359</td>\n",
       "      <td>0.004107</td>\n",
       "      <td>0.087623</td>\n",
       "      <td>0.378064</td>\n",
       "      <td>0.135542</td>\n",
       "      <td>0.220427</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Amnibacterium_382409</th>\n",
       "      <td>0.042205</td>\n",
       "      <td>0.203093</td>\n",
       "      <td>0.320076</td>\n",
       "      <td>-0.048153</td>\n",
       "      <td>0.375877</td>\n",
       "      <td>0.301948</td>\n",
       "      <td>0.101121</td>\n",
       "      <td>0.755436</td>\n",
       "      <td>0.744389</td>\n",
       "      <td>0.511555</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.306858</td>\n",
       "      <td>-0.491161</td>\n",
       "      <td>-0.612960</td>\n",
       "      <td>-0.580103</td>\n",
       "      <td>-0.426013</td>\n",
       "      <td>-0.684340</td>\n",
       "      <td>-0.478981</td>\n",
       "      <td>-0.279609</td>\n",
       "      <td>0.014729</td>\n",
       "      <td>0.002549</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Aquabacterium_A_592457</th>\n",
       "      <td>0.099716</td>\n",
       "      <td>-0.018531</td>\n",
       "      <td>0.142662</td>\n",
       "      <td>-0.066183</td>\n",
       "      <td>0.044122</td>\n",
       "      <td>-0.195314</td>\n",
       "      <td>-0.205904</td>\n",
       "      <td>0.315033</td>\n",
       "      <td>0.183843</td>\n",
       "      <td>0.280323</td>\n",
       "      <td>...</td>\n",
       "      <td>0.204536</td>\n",
       "      <td>0.134720</td>\n",
       "      <td>-0.143838</td>\n",
       "      <td>-0.158252</td>\n",
       "      <td>0.180313</td>\n",
       "      <td>-0.086774</td>\n",
       "      <td>-0.108247</td>\n",
       "      <td>-0.140181</td>\n",
       "      <td>-0.492698</td>\n",
       "      <td>-0.480638</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Aureimonas_A_501549</th>\n",
       "      <td>-0.163975</td>\n",
       "      <td>-0.291541</td>\n",
       "      <td>-0.025459</td>\n",
       "      <td>-0.212702</td>\n",
       "      <td>0.174651</td>\n",
       "      <td>0.069532</td>\n",
       "      <td>-0.116343</td>\n",
       "      <td>-0.080482</td>\n",
       "      <td>-0.050096</td>\n",
       "      <td>-0.103203</td>\n",
       "      <td>...</td>\n",
       "      <td>0.424012</td>\n",
       "      <td>-0.094717</td>\n",
       "      <td>0.160690</td>\n",
       "      <td>-0.063509</td>\n",
       "      <td>0.149466</td>\n",
       "      <td>0.002737</td>\n",
       "      <td>0.060224</td>\n",
       "      <td>-0.016975</td>\n",
       "      <td>0.096907</td>\n",
       "      <td>0.070627</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Blastococcus</th>\n",
       "      <td>0.254177</td>\n",
       "      <td>0.242126</td>\n",
       "      <td>0.131471</td>\n",
       "      <td>0.373596</td>\n",
       "      <td>0.401260</td>\n",
       "      <td>0.371131</td>\n",
       "      <td>0.169543</td>\n",
       "      <td>0.493016</td>\n",
       "      <td>0.603944</td>\n",
       "      <td>0.253355</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.242346</td>\n",
       "      <td>-0.327582</td>\n",
       "      <td>-0.376609</td>\n",
       "      <td>-0.209532</td>\n",
       "      <td>-0.591893</td>\n",
       "      <td>-0.696522</td>\n",
       "      <td>-0.066831</td>\n",
       "      <td>-0.170936</td>\n",
       "      <td>-0.227335</td>\n",
       "      <td>-0.012051</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Brevundimonas</th>\n",
       "      <td>0.051226</td>\n",
       "      <td>-0.328722</td>\n",
       "      <td>-0.305986</td>\n",
       "      <td>-0.102452</td>\n",
       "      <td>0.261334</td>\n",
       "      <td>-0.238871</td>\n",
       "      <td>-0.018080</td>\n",
       "      <td>-0.104643</td>\n",
       "      <td>-0.007944</td>\n",
       "      <td>0.133680</td>\n",
       "      <td>...</td>\n",
       "      <td>0.148796</td>\n",
       "      <td>0.144638</td>\n",
       "      <td>0.485139</td>\n",
       "      <td>0.096699</td>\n",
       "      <td>-0.081907</td>\n",
       "      <td>-0.072593</td>\n",
       "      <td>0.229010</td>\n",
       "      <td>-0.009178</td>\n",
       "      <td>-0.365977</td>\n",
       "      <td>-0.174771</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Caldora</th>\n",
       "      <td>0.033618</td>\n",
       "      <td>0.135312</td>\n",
       "      <td>0.115982</td>\n",
       "      <td>-0.231964</td>\n",
       "      <td>0.094691</td>\n",
       "      <td>-0.073679</td>\n",
       "      <td>-0.005323</td>\n",
       "      <td>0.012327</td>\n",
       "      <td>0.129989</td>\n",
       "      <td>-0.273706</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.202707</td>\n",
       "      <td>-0.219357</td>\n",
       "      <td>-0.387447</td>\n",
       "      <td>-0.419944</td>\n",
       "      <td>-0.122705</td>\n",
       "      <td>-0.117383</td>\n",
       "      <td>-0.451601</td>\n",
       "      <td>-0.221068</td>\n",
       "      <td>0.340382</td>\n",
       "      <td>0.228602</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Chryseobacterium_A_796612</th>\n",
       "      <td>0.250274</td>\n",
       "      <td>0.025739</td>\n",
       "      <td>0.210843</td>\n",
       "      <td>-0.300657</td>\n",
       "      <td>0.205367</td>\n",
       "      <td>-0.159912</td>\n",
       "      <td>-0.022453</td>\n",
       "      <td>0.273275</td>\n",
       "      <td>0.414567</td>\n",
       "      <td>0.546002</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.147907</td>\n",
       "      <td>-0.191128</td>\n",
       "      <td>-0.346112</td>\n",
       "      <td>-0.485761</td>\n",
       "      <td>-0.192771</td>\n",
       "      <td>-0.159912</td>\n",
       "      <td>-0.421139</td>\n",
       "      <td>-0.325346</td>\n",
       "      <td>-0.010405</td>\n",
       "      <td>0.214677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Curtobacterium</th>\n",
       "      <td>0.363779</td>\n",
       "      <td>0.324033</td>\n",
       "      <td>0.577374</td>\n",
       "      <td>0.378949</td>\n",
       "      <td>0.306436</td>\n",
       "      <td>0.151701</td>\n",
       "      <td>-0.143509</td>\n",
       "      <td>0.314324</td>\n",
       "      <td>0.713905</td>\n",
       "      <td>0.538538</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.146762</td>\n",
       "      <td>-0.341934</td>\n",
       "      <td>-0.329191</td>\n",
       "      <td>-0.268511</td>\n",
       "      <td>-0.370757</td>\n",
       "      <td>-0.452372</td>\n",
       "      <td>-0.083436</td>\n",
       "      <td>-0.331967</td>\n",
       "      <td>-0.365903</td>\n",
       "      <td>-0.169602</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Curvibacter_587305</th>\n",
       "      <td>-0.255278</td>\n",
       "      <td>0.296174</td>\n",
       "      <td>0.219814</td>\n",
       "      <td>-0.068692</td>\n",
       "      <td>0.172209</td>\n",
       "      <td>-0.084347</td>\n",
       "      <td>0.097127</td>\n",
       "      <td>0.252722</td>\n",
       "      <td>0.247610</td>\n",
       "      <td>0.298411</td>\n",
       "      <td>...</td>\n",
       "      <td>0.047008</td>\n",
       "      <td>-0.126201</td>\n",
       "      <td>-0.391704</td>\n",
       "      <td>-0.430363</td>\n",
       "      <td>0.101920</td>\n",
       "      <td>-0.038979</td>\n",
       "      <td>-0.350808</td>\n",
       "      <td>-0.253875</td>\n",
       "      <td>0.011182</td>\n",
       "      <td>-0.204159</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Duganella_570715</th>\n",
       "      <td>0.222654</td>\n",
       "      <td>0.131831</td>\n",
       "      <td>0.205590</td>\n",
       "      <td>0.116969</td>\n",
       "      <td>0.361916</td>\n",
       "      <td>-0.293936</td>\n",
       "      <td>0.154124</td>\n",
       "      <td>-0.028898</td>\n",
       "      <td>0.332742</td>\n",
       "      <td>0.097703</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.004438</td>\n",
       "      <td>-0.216599</td>\n",
       "      <td>-0.342926</td>\n",
       "      <td>-0.461821</td>\n",
       "      <td>-0.198985</td>\n",
       "      <td>-0.173389</td>\n",
       "      <td>-0.355861</td>\n",
       "      <td>-0.645344</td>\n",
       "      <td>-0.014862</td>\n",
       "      <td>0.058347</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Flavobacterium</th>\n",
       "      <td>-0.253421</td>\n",
       "      <td>-0.529283</td>\n",
       "      <td>-0.519431</td>\n",
       "      <td>-0.340449</td>\n",
       "      <td>-0.298851</td>\n",
       "      <td>-0.192118</td>\n",
       "      <td>-0.072250</td>\n",
       "      <td>-0.516147</td>\n",
       "      <td>-0.646415</td>\n",
       "      <td>-0.412698</td>\n",
       "      <td>...</td>\n",
       "      <td>0.144516</td>\n",
       "      <td>0.322934</td>\n",
       "      <td>0.306513</td>\n",
       "      <td>0.292830</td>\n",
       "      <td>0.283525</td>\n",
       "      <td>0.506842</td>\n",
       "      <td>0.245211</td>\n",
       "      <td>0.237580</td>\n",
       "      <td>0.105090</td>\n",
       "      <td>0.053640</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Friedmanniella</th>\n",
       "      <td>0.222161</td>\n",
       "      <td>0.446788</td>\n",
       "      <td>0.354746</td>\n",
       "      <td>0.192850</td>\n",
       "      <td>0.526503</td>\n",
       "      <td>0.333927</td>\n",
       "      <td>0.321326</td>\n",
       "      <td>0.660458</td>\n",
       "      <td>0.877414</td>\n",
       "      <td>0.670867</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.370195</td>\n",
       "      <td>-0.581564</td>\n",
       "      <td>-0.446240</td>\n",
       "      <td>-0.442953</td>\n",
       "      <td>-0.442405</td>\n",
       "      <td>-0.725106</td>\n",
       "      <td>-0.359951</td>\n",
       "      <td>-0.228493</td>\n",
       "      <td>-0.064923</td>\n",
       "      <td>-0.132311</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Frigoribacterium</th>\n",
       "      <td>0.239190</td>\n",
       "      <td>0.282978</td>\n",
       "      <td>-0.032293</td>\n",
       "      <td>0.290093</td>\n",
       "      <td>-0.394089</td>\n",
       "      <td>0.015873</td>\n",
       "      <td>-0.169130</td>\n",
       "      <td>0.180077</td>\n",
       "      <td>-0.091954</td>\n",
       "      <td>0.136836</td>\n",
       "      <td>...</td>\n",
       "      <td>0.023167</td>\n",
       "      <td>0.308703</td>\n",
       "      <td>0.033935</td>\n",
       "      <td>0.322386</td>\n",
       "      <td>0.094691</td>\n",
       "      <td>0.139026</td>\n",
       "      <td>0.294472</td>\n",
       "      <td>0.250718</td>\n",
       "      <td>-0.066229</td>\n",
       "      <td>0.241379</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Geodermatophilus</th>\n",
       "      <td>0.315303</td>\n",
       "      <td>0.293273</td>\n",
       "      <td>0.001377</td>\n",
       "      <td>0.218922</td>\n",
       "      <td>-0.007435</td>\n",
       "      <td>-0.144847</td>\n",
       "      <td>-0.241778</td>\n",
       "      <td>0.314752</td>\n",
       "      <td>0.315027</td>\n",
       "      <td>0.255547</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.230609</td>\n",
       "      <td>0.032769</td>\n",
       "      <td>-0.123092</td>\n",
       "      <td>-0.024784</td>\n",
       "      <td>-0.336231</td>\n",
       "      <td>-0.365145</td>\n",
       "      <td>0.003580</td>\n",
       "      <td>0.100387</td>\n",
       "      <td>-0.186703</td>\n",
       "      <td>-0.029465</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Herbaspirillum</th>\n",
       "      <td>-0.200828</td>\n",
       "      <td>0.097240</td>\n",
       "      <td>0.016736</td>\n",
       "      <td>0.117438</td>\n",
       "      <td>0.427626</td>\n",
       "      <td>0.393000</td>\n",
       "      <td>0.283064</td>\n",
       "      <td>0.303262</td>\n",
       "      <td>0.207176</td>\n",
       "      <td>-0.014716</td>\n",
       "      <td>...</td>\n",
       "      <td>0.039837</td>\n",
       "      <td>-0.285661</td>\n",
       "      <td>0.073002</td>\n",
       "      <td>-0.076176</td>\n",
       "      <td>0.032606</td>\n",
       "      <td>-0.182073</td>\n",
       "      <td>-0.034048</td>\n",
       "      <td>0.043576</td>\n",
       "      <td>-0.007214</td>\n",
       "      <td>-0.114553</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Hylemonella</th>\n",
       "      <td>0.212632</td>\n",
       "      <td>0.061104</td>\n",
       "      <td>-0.001370</td>\n",
       "      <td>0.214824</td>\n",
       "      <td>0.007398</td>\n",
       "      <td>-0.186875</td>\n",
       "      <td>-0.016715</td>\n",
       "      <td>0.220578</td>\n",
       "      <td>0.046856</td>\n",
       "      <td>0.264968</td>\n",
       "      <td>...</td>\n",
       "      <td>0.082288</td>\n",
       "      <td>0.103302</td>\n",
       "      <td>0.107138</td>\n",
       "      <td>0.013427</td>\n",
       "      <td>0.131799</td>\n",
       "      <td>-0.118098</td>\n",
       "      <td>0.057816</td>\n",
       "      <td>-0.174020</td>\n",
       "      <td>-0.496781</td>\n",
       "      <td>-0.416496</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Janthinobacterium_571130</th>\n",
       "      <td>-0.014017</td>\n",
       "      <td>-0.065962</td>\n",
       "      <td>-0.029683</td>\n",
       "      <td>-0.080803</td>\n",
       "      <td>0.283636</td>\n",
       "      <td>-0.090423</td>\n",
       "      <td>0.125053</td>\n",
       "      <td>0.219323</td>\n",
       "      <td>0.208879</td>\n",
       "      <td>0.142642</td>\n",
       "      <td>...</td>\n",
       "      <td>0.186956</td>\n",
       "      <td>-0.200909</td>\n",
       "      <td>-0.312494</td>\n",
       "      <td>-0.387251</td>\n",
       "      <td>-0.023911</td>\n",
       "      <td>-0.044249</td>\n",
       "      <td>-0.351797</td>\n",
       "      <td>-0.413692</td>\n",
       "      <td>0.229767</td>\n",
       "      <td>0.089598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Janthinobacterium_571526</th>\n",
       "      <td>0.045305</td>\n",
       "      <td>0.108457</td>\n",
       "      <td>0.182868</td>\n",
       "      <td>-0.016749</td>\n",
       "      <td>0.102417</td>\n",
       "      <td>0.012081</td>\n",
       "      <td>-0.108732</td>\n",
       "      <td>0.257552</td>\n",
       "      <td>0.383308</td>\n",
       "      <td>0.134542</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.057278</td>\n",
       "      <td>-0.263867</td>\n",
       "      <td>-0.432732</td>\n",
       "      <td>-0.316861</td>\n",
       "      <td>-0.220759</td>\n",
       "      <td>-0.264142</td>\n",
       "      <td>-0.227074</td>\n",
       "      <td>-0.411646</td>\n",
       "      <td>-0.024163</td>\n",
       "      <td>-0.171885</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Kineococcus</th>\n",
       "      <td>0.235391</td>\n",
       "      <td>0.433283</td>\n",
       "      <td>0.405365</td>\n",
       "      <td>0.203093</td>\n",
       "      <td>0.191323</td>\n",
       "      <td>0.186397</td>\n",
       "      <td>0.027645</td>\n",
       "      <td>0.252087</td>\n",
       "      <td>0.503900</td>\n",
       "      <td>0.550705</td>\n",
       "      <td>...</td>\n",
       "      <td>0.054063</td>\n",
       "      <td>-0.173806</td>\n",
       "      <td>-0.206377</td>\n",
       "      <td>-0.130012</td>\n",
       "      <td>0.029013</td>\n",
       "      <td>-0.213220</td>\n",
       "      <td>0.013138</td>\n",
       "      <td>-0.104572</td>\n",
       "      <td>-0.105378</td>\n",
       "      <td>-0.177638</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Kineosporia</th>\n",
       "      <td>-0.283087</td>\n",
       "      <td>-0.353630</td>\n",
       "      <td>-0.009122</td>\n",
       "      <td>-0.197948</td>\n",
       "      <td>-0.117370</td>\n",
       "      <td>0.138959</td>\n",
       "      <td>-0.221665</td>\n",
       "      <td>0.054732</td>\n",
       "      <td>0.062334</td>\n",
       "      <td>-0.144128</td>\n",
       "      <td>...</td>\n",
       "      <td>0.220013</td>\n",
       "      <td>0.018548</td>\n",
       "      <td>0.066895</td>\n",
       "      <td>0.101863</td>\n",
       "      <td>0.114633</td>\n",
       "      <td>0.106120</td>\n",
       "      <td>0.174231</td>\n",
       "      <td>-0.079220</td>\n",
       "      <td>-0.159635</td>\n",
       "      <td>-0.184569</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Klenkia</th>\n",
       "      <td>-0.151615</td>\n",
       "      <td>0.243021</td>\n",
       "      <td>-0.043240</td>\n",
       "      <td>0.013684</td>\n",
       "      <td>0.090859</td>\n",
       "      <td>0.032841</td>\n",
       "      <td>0.309250</td>\n",
       "      <td>-0.131910</td>\n",
       "      <td>-0.002189</td>\n",
       "      <td>-0.022441</td>\n",
       "      <td>...</td>\n",
       "      <td>0.057917</td>\n",
       "      <td>-0.028462</td>\n",
       "      <td>-0.102901</td>\n",
       "      <td>-0.059661</td>\n",
       "      <td>0.101806</td>\n",
       "      <td>-0.062397</td>\n",
       "      <td>-0.036125</td>\n",
       "      <td>0.008759</td>\n",
       "      <td>0.268747</td>\n",
       "      <td>0.303777</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Kordiimonas</th>\n",
       "      <td>-0.420966</td>\n",
       "      <td>-0.170795</td>\n",
       "      <td>-0.232927</td>\n",
       "      <td>-0.318051</td>\n",
       "      <td>-0.141782</td>\n",
       "      <td>-0.270699</td>\n",
       "      <td>-0.146709</td>\n",
       "      <td>-0.246339</td>\n",
       "      <td>-0.470234</td>\n",
       "      <td>-0.736280</td>\n",
       "      <td>...</td>\n",
       "      <td>0.278314</td>\n",
       "      <td>0.223074</td>\n",
       "      <td>0.192966</td>\n",
       "      <td>-0.027371</td>\n",
       "      <td>0.256193</td>\n",
       "      <td>0.339674</td>\n",
       "      <td>-0.070891</td>\n",
       "      <td>0.231317</td>\n",
       "      <td>0.257014</td>\n",
       "      <td>0.283564</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Marmoricola_A_392027</th>\n",
       "      <td>0.347275</td>\n",
       "      <td>0.111422</td>\n",
       "      <td>0.016968</td>\n",
       "      <td>0.120755</td>\n",
       "      <td>0.302876</td>\n",
       "      <td>0.391675</td>\n",
       "      <td>0.209836</td>\n",
       "      <td>0.525721</td>\n",
       "      <td>0.543537</td>\n",
       "      <td>0.299483</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.294395</td>\n",
       "      <td>-0.280252</td>\n",
       "      <td>-0.301179</td>\n",
       "      <td>-0.120189</td>\n",
       "      <td>-0.358022</td>\n",
       "      <td>-0.524307</td>\n",
       "      <td>-0.047227</td>\n",
       "      <td>-0.095033</td>\n",
       "      <td>0.059105</td>\n",
       "      <td>-0.032522</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Massilia_573596</th>\n",
       "      <td>-0.060216</td>\n",
       "      <td>-0.137129</td>\n",
       "      <td>-0.139592</td>\n",
       "      <td>-0.366224</td>\n",
       "      <td>-0.139866</td>\n",
       "      <td>-0.460380</td>\n",
       "      <td>-0.283564</td>\n",
       "      <td>0.080471</td>\n",
       "      <td>-0.139045</td>\n",
       "      <td>-0.113316</td>\n",
       "      <td>...</td>\n",
       "      <td>0.025101</td>\n",
       "      <td>0.191871</td>\n",
       "      <td>-0.256740</td>\n",
       "      <td>-0.310935</td>\n",
       "      <td>0.214589</td>\n",
       "      <td>0.277816</td>\n",
       "      <td>-0.366498</td>\n",
       "      <td>-0.054339</td>\n",
       "      <td>0.317230</td>\n",
       "      <td>0.133023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Massilia_574424</th>\n",
       "      <td>-0.375632</td>\n",
       "      <td>-0.381699</td>\n",
       "      <td>-0.312751</td>\n",
       "      <td>-0.025373</td>\n",
       "      <td>-0.373425</td>\n",
       "      <td>0.095976</td>\n",
       "      <td>0.002758</td>\n",
       "      <td>-0.172096</td>\n",
       "      <td>-0.436858</td>\n",
       "      <td>-0.275794</td>\n",
       "      <td>...</td>\n",
       "      <td>0.159533</td>\n",
       "      <td>0.116385</td>\n",
       "      <td>0.333159</td>\n",
       "      <td>0.409830</td>\n",
       "      <td>0.173750</td>\n",
       "      <td>0.392179</td>\n",
       "      <td>0.250421</td>\n",
       "      <td>0.254041</td>\n",
       "      <td>0.337572</td>\n",
       "      <td>0.236631</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Massilia_574544</th>\n",
       "      <td>-0.170271</td>\n",
       "      <td>-0.104572</td>\n",
       "      <td>-0.255133</td>\n",
       "      <td>-0.359978</td>\n",
       "      <td>-0.190255</td>\n",
       "      <td>-0.296195</td>\n",
       "      <td>-0.122913</td>\n",
       "      <td>0.047085</td>\n",
       "      <td>-0.244457</td>\n",
       "      <td>-0.249932</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.096554</td>\n",
       "      <td>0.183137</td>\n",
       "      <td>-0.113605</td>\n",
       "      <td>-0.169176</td>\n",
       "      <td>0.177936</td>\n",
       "      <td>0.313715</td>\n",
       "      <td>-0.252395</td>\n",
       "      <td>0.108145</td>\n",
       "      <td>0.432795</td>\n",
       "      <td>0.212428</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Methylobacterium</th>\n",
       "      <td>0.186174</td>\n",
       "      <td>0.340862</td>\n",
       "      <td>0.206708</td>\n",
       "      <td>0.232717</td>\n",
       "      <td>0.543190</td>\n",
       "      <td>0.409309</td>\n",
       "      <td>0.455852</td>\n",
       "      <td>0.329911</td>\n",
       "      <td>0.425462</td>\n",
       "      <td>-0.001916</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.099603</td>\n",
       "      <td>-0.377276</td>\n",
       "      <td>-0.173032</td>\n",
       "      <td>-0.201232</td>\n",
       "      <td>-0.144011</td>\n",
       "      <td>-0.452293</td>\n",
       "      <td>-0.047912</td>\n",
       "      <td>-0.205367</td>\n",
       "      <td>-0.146201</td>\n",
       "      <td>-0.239562</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Microbacterium_A_383321</th>\n",
       "      <td>0.272254</td>\n",
       "      <td>0.337442</td>\n",
       "      <td>0.531635</td>\n",
       "      <td>0.185977</td>\n",
       "      <td>0.178855</td>\n",
       "      <td>-0.075596</td>\n",
       "      <td>-0.183511</td>\n",
       "      <td>0.468091</td>\n",
       "      <td>0.581759</td>\n",
       "      <td>0.702274</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.005244</td>\n",
       "      <td>-0.245960</td>\n",
       "      <td>-0.213640</td>\n",
       "      <td>-0.321282</td>\n",
       "      <td>-0.120241</td>\n",
       "      <td>-0.274172</td>\n",
       "      <td>-0.204602</td>\n",
       "      <td>-0.289549</td>\n",
       "      <td>-0.317721</td>\n",
       "      <td>-0.238017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mycobacterium</th>\n",
       "      <td>-0.295913</td>\n",
       "      <td>0.073192</td>\n",
       "      <td>0.045173</td>\n",
       "      <td>-0.042314</td>\n",
       "      <td>-0.078052</td>\n",
       "      <td>0.110931</td>\n",
       "      <td>-0.126370</td>\n",
       "      <td>-0.299057</td>\n",
       "      <td>-0.082913</td>\n",
       "      <td>-0.287335</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.043218</td>\n",
       "      <td>0.053750</td>\n",
       "      <td>0.265035</td>\n",
       "      <td>0.212142</td>\n",
       "      <td>-0.020013</td>\n",
       "      <td>-0.052893</td>\n",
       "      <td>0.159821</td>\n",
       "      <td>0.406042</td>\n",
       "      <td>-0.001715</td>\n",
       "      <td>0.086915</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Nakamurella</th>\n",
       "      <td>-0.061627</td>\n",
       "      <td>-0.173925</td>\n",
       "      <td>0.036702</td>\n",
       "      <td>-0.306765</td>\n",
       "      <td>0.146261</td>\n",
       "      <td>0.158861</td>\n",
       "      <td>0.244865</td>\n",
       "      <td>-0.098603</td>\n",
       "      <td>-0.037798</td>\n",
       "      <td>0.071213</td>\n",
       "      <td>...</td>\n",
       "      <td>0.282369</td>\n",
       "      <td>0.084908</td>\n",
       "      <td>0.210353</td>\n",
       "      <td>0.001917</td>\n",
       "      <td>0.250616</td>\n",
       "      <td>0.245412</td>\n",
       "      <td>0.105177</td>\n",
       "      <td>0.224764</td>\n",
       "      <td>0.223774</td>\n",
       "      <td>0.198850</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Neorhizobium_500197</th>\n",
       "      <td>0.258347</td>\n",
       "      <td>0.335523</td>\n",
       "      <td>0.495348</td>\n",
       "      <td>0.223317</td>\n",
       "      <td>0.180624</td>\n",
       "      <td>0.173508</td>\n",
       "      <td>0.188834</td>\n",
       "      <td>0.183908</td>\n",
       "      <td>0.451560</td>\n",
       "      <td>0.389163</td>\n",
       "      <td>...</td>\n",
       "      <td>0.113627</td>\n",
       "      <td>-0.128626</td>\n",
       "      <td>-0.110016</td>\n",
       "      <td>-0.104543</td>\n",
       "      <td>0.036672</td>\n",
       "      <td>-0.175151</td>\n",
       "      <td>0.009305</td>\n",
       "      <td>-0.212673</td>\n",
       "      <td>-0.294472</td>\n",
       "      <td>-0.348659</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Nocardioides_A_392796</th>\n",
       "      <td>-0.124829</td>\n",
       "      <td>0.300849</td>\n",
       "      <td>0.158226</td>\n",
       "      <td>0.320285</td>\n",
       "      <td>0.148919</td>\n",
       "      <td>0.405420</td>\n",
       "      <td>0.165070</td>\n",
       "      <td>0.158774</td>\n",
       "      <td>0.175746</td>\n",
       "      <td>0.029838</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.070071</td>\n",
       "      <td>-0.236244</td>\n",
       "      <td>0.065152</td>\n",
       "      <td>0.154120</td>\n",
       "      <td>0.015330</td>\n",
       "      <td>-0.352039</td>\n",
       "      <td>0.058034</td>\n",
       "      <td>0.149076</td>\n",
       "      <td>0.015877</td>\n",
       "      <td>-0.184506</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Noviherbaspirillum_A_568104</th>\n",
       "      <td>-0.180267</td>\n",
       "      <td>0.121497</td>\n",
       "      <td>-0.205979</td>\n",
       "      <td>-0.205979</td>\n",
       "      <td>0.207957</td>\n",
       "      <td>0.082222</td>\n",
       "      <td>0.349232</td>\n",
       "      <td>0.148056</td>\n",
       "      <td>0.072615</td>\n",
       "      <td>-0.088156</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.229501</td>\n",
       "      <td>-0.061313</td>\n",
       "      <td>-0.136754</td>\n",
       "      <td>-0.166705</td>\n",
       "      <td>0.077701</td>\n",
       "      <td>-0.157098</td>\n",
       "      <td>-0.324650</td>\n",
       "      <td>-0.076582</td>\n",
       "      <td>0.260511</td>\n",
       "      <td>0.105109</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patulibacter</th>\n",
       "      <td>-0.091842</td>\n",
       "      <td>-0.093043</td>\n",
       "      <td>-0.195090</td>\n",
       "      <td>-0.099646</td>\n",
       "      <td>-0.493428</td>\n",
       "      <td>-0.531846</td>\n",
       "      <td>-0.296537</td>\n",
       "      <td>-0.592474</td>\n",
       "      <td>-0.669309</td>\n",
       "      <td>-0.262921</td>\n",
       "      <td>...</td>\n",
       "      <td>0.335736</td>\n",
       "      <td>0.803771</td>\n",
       "      <td>0.612283</td>\n",
       "      <td>0.490426</td>\n",
       "      <td>0.453810</td>\n",
       "      <td>0.680114</td>\n",
       "      <td>0.365569</td>\n",
       "      <td>0.449069</td>\n",
       "      <td>-0.130860</td>\n",
       "      <td>-0.050423</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Paucimonas</th>\n",
       "      <td>-0.067567</td>\n",
       "      <td>0.111596</td>\n",
       "      <td>0.014123</td>\n",
       "      <td>0.359155</td>\n",
       "      <td>-0.232606</td>\n",
       "      <td>-0.030183</td>\n",
       "      <td>-0.194669</td>\n",
       "      <td>-0.202977</td>\n",
       "      <td>-0.227622</td>\n",
       "      <td>-0.260021</td>\n",
       "      <td>...</td>\n",
       "      <td>0.151250</td>\n",
       "      <td>0.106057</td>\n",
       "      <td>0.142610</td>\n",
       "      <td>0.361648</td>\n",
       "      <td>0.138733</td>\n",
       "      <td>0.111319</td>\n",
       "      <td>0.159225</td>\n",
       "      <td>-0.153569</td>\n",
       "      <td>-0.400415</td>\n",
       "      <td>-0.013292</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pelomonas</th>\n",
       "      <td>-0.086216</td>\n",
       "      <td>0.237573</td>\n",
       "      <td>0.276530</td>\n",
       "      <td>0.346461</td>\n",
       "      <td>0.118467</td>\n",
       "      <td>0.136349</td>\n",
       "      <td>-0.061629</td>\n",
       "      <td>0.357318</td>\n",
       "      <td>0.338478</td>\n",
       "      <td>0.349015</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.056636</td>\n",
       "      <td>-0.393081</td>\n",
       "      <td>-0.090687</td>\n",
       "      <td>-0.039276</td>\n",
       "      <td>-0.098350</td>\n",
       "      <td>-0.375838</td>\n",
       "      <td>-0.134753</td>\n",
       "      <td>-0.315690</td>\n",
       "      <td>-0.055242</td>\n",
       "      <td>-0.225758</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pigmentiphaga</th>\n",
       "      <td>-0.311219</td>\n",
       "      <td>-0.390827</td>\n",
       "      <td>-0.126485</td>\n",
       "      <td>-0.290693</td>\n",
       "      <td>0.040220</td>\n",
       "      <td>-0.159770</td>\n",
       "      <td>-0.202487</td>\n",
       "      <td>-0.307891</td>\n",
       "      <td>-0.376681</td>\n",
       "      <td>-0.368637</td>\n",
       "      <td>...</td>\n",
       "      <td>0.500917</td>\n",
       "      <td>0.296796</td>\n",
       "      <td>0.208866</td>\n",
       "      <td>-0.107068</td>\n",
       "      <td>0.119550</td>\n",
       "      <td>0.308446</td>\n",
       "      <td>0.069067</td>\n",
       "      <td>0.178934</td>\n",
       "      <td>0.111506</td>\n",
       "      <td>-0.028570</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pseudoduganella_570666</th>\n",
       "      <td>0.228462</td>\n",
       "      <td>0.258869</td>\n",
       "      <td>0.066292</td>\n",
       "      <td>0.145460</td>\n",
       "      <td>0.416655</td>\n",
       "      <td>0.155321</td>\n",
       "      <td>0.229284</td>\n",
       "      <td>0.668676</td>\n",
       "      <td>0.407890</td>\n",
       "      <td>0.376387</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.149072</td>\n",
       "      <td>-0.545953</td>\n",
       "      <td>-0.372552</td>\n",
       "      <td>-0.339406</td>\n",
       "      <td>0.026846</td>\n",
       "      <td>-0.335297</td>\n",
       "      <td>-0.388166</td>\n",
       "      <td>-0.368767</td>\n",
       "      <td>-0.006301</td>\n",
       "      <td>-0.110396</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pseudoduganella_570691</th>\n",
       "      <td>-0.190953</td>\n",
       "      <td>0.112081</td>\n",
       "      <td>0.180096</td>\n",
       "      <td>0.041831</td>\n",
       "      <td>0.095476</td>\n",
       "      <td>0.139542</td>\n",
       "      <td>-0.209792</td>\n",
       "      <td>0.183928</td>\n",
       "      <td>0.230229</td>\n",
       "      <td>0.024588</td>\n",
       "      <td>...</td>\n",
       "      <td>0.197903</td>\n",
       "      <td>-0.204364</td>\n",
       "      <td>-0.079830</td>\n",
       "      <td>-0.088132</td>\n",
       "      <td>-0.084939</td>\n",
       "      <td>-0.188398</td>\n",
       "      <td>0.020436</td>\n",
       "      <td>-0.214612</td>\n",
       "      <td>-0.342629</td>\n",
       "      <td>-0.162533</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pseudomonas_B_650453</th>\n",
       "      <td>0.254959</td>\n",
       "      <td>0.293606</td>\n",
       "      <td>0.507610</td>\n",
       "      <td>0.126326</td>\n",
       "      <td>0.555487</td>\n",
       "      <td>0.149399</td>\n",
       "      <td>0.074988</td>\n",
       "      <td>0.493190</td>\n",
       "      <td>0.827174</td>\n",
       "      <td>0.633359</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.077894</td>\n",
       "      <td>-0.545681</td>\n",
       "      <td>-0.500112</td>\n",
       "      <td>-0.591251</td>\n",
       "      <td>-0.360519</td>\n",
       "      <td>-0.608556</td>\n",
       "      <td>-0.337446</td>\n",
       "      <td>-0.512295</td>\n",
       "      <td>-0.338022</td>\n",
       "      <td>-0.217465</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pseudomonas_E_647464</th>\n",
       "      <td>-0.215654</td>\n",
       "      <td>-0.281336</td>\n",
       "      <td>-0.449918</td>\n",
       "      <td>-0.302682</td>\n",
       "      <td>-0.535851</td>\n",
       "      <td>-0.435687</td>\n",
       "      <td>-0.328407</td>\n",
       "      <td>-0.030104</td>\n",
       "      <td>-0.441160</td>\n",
       "      <td>0.044335</td>\n",
       "      <td>...</td>\n",
       "      <td>0.024821</td>\n",
       "      <td>0.403394</td>\n",
       "      <td>0.076081</td>\n",
       "      <td>0.174056</td>\n",
       "      <td>0.218391</td>\n",
       "      <td>0.408867</td>\n",
       "      <td>-0.139573</td>\n",
       "      <td>0.010948</td>\n",
       "      <td>0.301587</td>\n",
       "      <td>0.170224</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pseudonocardia</th>\n",
       "      <td>0.144113</td>\n",
       "      <td>0.136300</td>\n",
       "      <td>0.085079</td>\n",
       "      <td>0.053247</td>\n",
       "      <td>-0.168422</td>\n",
       "      <td>0.112860</td>\n",
       "      <td>-0.409189</td>\n",
       "      <td>-0.074372</td>\n",
       "      <td>-0.047748</td>\n",
       "      <td>-0.213565</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.005249</td>\n",
       "      <td>0.284465</td>\n",
       "      <td>0.116911</td>\n",
       "      <td>0.199096</td>\n",
       "      <td>-0.065401</td>\n",
       "      <td>-0.028649</td>\n",
       "      <td>0.098969</td>\n",
       "      <td>0.329364</td>\n",
       "      <td>-0.217617</td>\n",
       "      <td>-0.048038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ramlibacter_588642</th>\n",
       "      <td>-0.019767</td>\n",
       "      <td>-0.356076</td>\n",
       "      <td>-0.515857</td>\n",
       "      <td>-0.199864</td>\n",
       "      <td>-0.076047</td>\n",
       "      <td>-0.434045</td>\n",
       "      <td>-0.031297</td>\n",
       "      <td>0.004667</td>\n",
       "      <td>-0.375294</td>\n",
       "      <td>0.017021</td>\n",
       "      <td>...</td>\n",
       "      <td>0.209713</td>\n",
       "      <td>0.227043</td>\n",
       "      <td>0.386275</td>\n",
       "      <td>0.136445</td>\n",
       "      <td>0.405218</td>\n",
       "      <td>0.278931</td>\n",
       "      <td>0.071105</td>\n",
       "      <td>0.048051</td>\n",
       "      <td>-0.097186</td>\n",
       "      <td>-0.414827</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Rathayibacter</th>\n",
       "      <td>0.166827</td>\n",
       "      <td>0.132311</td>\n",
       "      <td>-0.183811</td>\n",
       "      <td>0.125188</td>\n",
       "      <td>-0.031776</td>\n",
       "      <td>-0.248459</td>\n",
       "      <td>0.196959</td>\n",
       "      <td>-0.226270</td>\n",
       "      <td>-0.165183</td>\n",
       "      <td>0.116422</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.298315</td>\n",
       "      <td>0.111492</td>\n",
       "      <td>0.167374</td>\n",
       "      <td>0.164087</td>\n",
       "      <td>0.191481</td>\n",
       "      <td>0.111766</td>\n",
       "      <td>0.030000</td>\n",
       "      <td>-0.024380</td>\n",
       "      <td>0.052596</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Rhodoferax_C</th>\n",
       "      <td>-0.286262</td>\n",
       "      <td>-0.254516</td>\n",
       "      <td>-0.558292</td>\n",
       "      <td>-0.235359</td>\n",
       "      <td>0.063492</td>\n",
       "      <td>-0.023536</td>\n",
       "      <td>0.302682</td>\n",
       "      <td>-0.162014</td>\n",
       "      <td>-0.382047</td>\n",
       "      <td>-0.181719</td>\n",
       "      <td>...</td>\n",
       "      <td>0.167131</td>\n",
       "      <td>0.050903</td>\n",
       "      <td>0.178435</td>\n",
       "      <td>0.241927</td>\n",
       "      <td>0.420909</td>\n",
       "      <td>0.229338</td>\n",
       "      <td>0.007663</td>\n",
       "      <td>0.085398</td>\n",
       "      <td>0.239190</td>\n",
       "      <td>0.090312</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Rugamonas_570800</th>\n",
       "      <td>-0.085515</td>\n",
       "      <td>0.037550</td>\n",
       "      <td>-0.007948</td>\n",
       "      <td>-0.218994</td>\n",
       "      <td>-0.120872</td>\n",
       "      <td>-0.360971</td>\n",
       "      <td>-0.209401</td>\n",
       "      <td>0.148280</td>\n",
       "      <td>-0.018090</td>\n",
       "      <td>0.202823</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.116285</td>\n",
       "      <td>0.053721</td>\n",
       "      <td>-0.121146</td>\n",
       "      <td>-0.213239</td>\n",
       "      <td>0.035905</td>\n",
       "      <td>0.155132</td>\n",
       "      <td>-0.274634</td>\n",
       "      <td>-0.107593</td>\n",
       "      <td>0.217350</td>\n",
       "      <td>0.190764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Solirubrobacter</th>\n",
       "      <td>0.583341</td>\n",
       "      <td>0.450109</td>\n",
       "      <td>0.284469</td>\n",
       "      <td>0.512432</td>\n",
       "      <td>0.079219</td>\n",
       "      <td>0.271727</td>\n",
       "      <td>0.227686</td>\n",
       "      <td>0.410499</td>\n",
       "      <td>0.412992</td>\n",
       "      <td>0.275605</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.424289</td>\n",
       "      <td>-0.310229</td>\n",
       "      <td>-0.168410</td>\n",
       "      <td>0.009418</td>\n",
       "      <td>-0.274220</td>\n",
       "      <td>-0.441245</td>\n",
       "      <td>0.034624</td>\n",
       "      <td>-0.203477</td>\n",
       "      <td>-0.126030</td>\n",
       "      <td>-0.088083</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sphingomonas_L_486704</th>\n",
       "      <td>-0.092240</td>\n",
       "      <td>0.029287</td>\n",
       "      <td>0.249624</td>\n",
       "      <td>0.333105</td>\n",
       "      <td>0.031477</td>\n",
       "      <td>0.547420</td>\n",
       "      <td>0.257835</td>\n",
       "      <td>-0.053647</td>\n",
       "      <td>0.039688</td>\n",
       "      <td>-0.166416</td>\n",
       "      <td>...</td>\n",
       "      <td>0.015722</td>\n",
       "      <td>-0.194608</td>\n",
       "      <td>0.062680</td>\n",
       "      <td>0.242233</td>\n",
       "      <td>-0.158478</td>\n",
       "      <td>-0.119885</td>\n",
       "      <td>0.337485</td>\n",
       "      <td>0.109225</td>\n",
       "      <td>-0.029834</td>\n",
       "      <td>-0.006295</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Telluria_573210</th>\n",
       "      <td>0.066229</td>\n",
       "      <td>0.301587</td>\n",
       "      <td>0.188834</td>\n",
       "      <td>0.368911</td>\n",
       "      <td>0.319102</td>\n",
       "      <td>-0.063492</td>\n",
       "      <td>0.070060</td>\n",
       "      <td>0.029009</td>\n",
       "      <td>0.292830</td>\n",
       "      <td>0.146141</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.040266</td>\n",
       "      <td>-0.154351</td>\n",
       "      <td>-0.126437</td>\n",
       "      <td>-0.096880</td>\n",
       "      <td>-0.250684</td>\n",
       "      <td>-0.340449</td>\n",
       "      <td>0.006021</td>\n",
       "      <td>-0.173259</td>\n",
       "      <td>-0.357417</td>\n",
       "      <td>-0.221675</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Variovorax</th>\n",
       "      <td>-0.030651</td>\n",
       "      <td>-0.292830</td>\n",
       "      <td>-0.228243</td>\n",
       "      <td>-0.119869</td>\n",
       "      <td>-0.145047</td>\n",
       "      <td>0.064039</td>\n",
       "      <td>0.172961</td>\n",
       "      <td>-0.484401</td>\n",
       "      <td>-0.513410</td>\n",
       "      <td>-0.480569</td>\n",
       "      <td>...</td>\n",
       "      <td>0.073361</td>\n",
       "      <td>-0.000547</td>\n",
       "      <td>0.199234</td>\n",
       "      <td>0.255610</td>\n",
       "      <td>0.282978</td>\n",
       "      <td>0.328407</td>\n",
       "      <td>0.117679</td>\n",
       "      <td>0.029287</td>\n",
       "      <td>0.267652</td>\n",
       "      <td>0.186097</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>55 rows × 15091 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             AT1G01010  AT1G01020  AT1G01030  AT1G01040  \\\n",
       "Genus                                                                     \n",
       "Actinomycetospora            -0.037245   0.359304   0.215254   0.235246   \n",
       "Actinoplanes                 -0.009582   0.145106   0.005476  -0.004107   \n",
       "Aeromicrobium                 0.065863   0.026906  -0.052691   0.223375   \n",
       "Aetherobacter                -0.244510  -0.247898  -0.351801  -0.073692   \n",
       "Agrobacterium                 0.209748   0.263417  -0.050931  -0.096112   \n",
       "Amnibacterium_382409          0.042205   0.203093   0.320076  -0.048153   \n",
       "Aquabacterium_A_592457        0.099716  -0.018531   0.142662  -0.066183   \n",
       "Aureimonas_A_501549          -0.163975  -0.291541  -0.025459  -0.212702   \n",
       "Blastococcus                  0.254177   0.242126   0.131471   0.373596   \n",
       "Brevundimonas                 0.051226  -0.328722  -0.305986  -0.102452   \n",
       "Caldora                       0.033618   0.135312   0.115982  -0.231964   \n",
       "Chryseobacterium_A_796612     0.250274   0.025739   0.210843  -0.300657   \n",
       "Curtobacterium                0.363779   0.324033   0.577374   0.378949   \n",
       "Curvibacter_587305           -0.255278   0.296174   0.219814  -0.068692   \n",
       "Duganella_570715              0.222654   0.131831   0.205590   0.116969   \n",
       "Flavobacterium               -0.253421  -0.529283  -0.519431  -0.340449   \n",
       "Friedmanniella                0.222161   0.446788   0.354746   0.192850   \n",
       "Frigoribacterium              0.239190   0.282978  -0.032293   0.290093   \n",
       "Geodermatophilus              0.315303   0.293273   0.001377   0.218922   \n",
       "Herbaspirillum               -0.200828   0.097240   0.016736   0.117438   \n",
       "Hylemonella                   0.212632   0.061104  -0.001370   0.214824   \n",
       "Janthinobacterium_571130     -0.014017  -0.065962  -0.029683  -0.080803   \n",
       "Janthinobacterium_571526      0.045305   0.108457   0.182868  -0.016749   \n",
       "Kineococcus                   0.235391   0.433283   0.405365   0.203093   \n",
       "Kineosporia                  -0.283087  -0.353630  -0.009122  -0.197948   \n",
       "Klenkia                      -0.151615   0.243021  -0.043240   0.013684   \n",
       "Kordiimonas                  -0.420966  -0.170795  -0.232927  -0.318051   \n",
       "Marmoricola_A_392027          0.347275   0.111422   0.016968   0.120755   \n",
       "Massilia_573596              -0.060216  -0.137129  -0.139592  -0.366224   \n",
       "Massilia_574424              -0.375632  -0.381699  -0.312751  -0.025373   \n",
       "Massilia_574544              -0.170271  -0.104572  -0.255133  -0.359978   \n",
       "Methylobacterium              0.186174   0.340862   0.206708   0.232717   \n",
       "Microbacterium_A_383321       0.272254   0.337442   0.531635   0.185977   \n",
       "Mycobacterium                -0.295913   0.073192   0.045173  -0.042314   \n",
       "Nakamurella                  -0.061627  -0.173925   0.036702  -0.306765   \n",
       "Neorhizobium_500197           0.258347   0.335523   0.495348   0.223317   \n",
       "Nocardioides_A_392796        -0.124829   0.300849   0.158226   0.320285   \n",
       "Noviherbaspirillum_A_568104  -0.180267   0.121497  -0.205979  -0.205979   \n",
       "Patulibacter                 -0.091842  -0.093043  -0.195090  -0.099646   \n",
       "Paucimonas                   -0.067567   0.111596   0.014123   0.359155   \n",
       "Pelomonas                    -0.086216   0.237573   0.276530   0.346461   \n",
       "Pigmentiphaga                -0.311219  -0.390827  -0.126485  -0.290693   \n",
       "Pseudoduganella_570666        0.228462   0.258869   0.066292   0.145460   \n",
       "Pseudoduganella_570691       -0.190953   0.112081   0.180096   0.041831   \n",
       "Pseudomonas_B_650453          0.254959   0.293606   0.507610   0.126326   \n",
       "Pseudomonas_E_647464         -0.215654  -0.281336  -0.449918  -0.302682   \n",
       "Pseudonocardia                0.144113   0.136300   0.085079   0.053247   \n",
       "Ramlibacter_588642           -0.019767  -0.356076  -0.515857  -0.199864   \n",
       "Rathayibacter                 0.166827   0.132311  -0.183811   0.125188   \n",
       "Rhodoferax_C                 -0.286262  -0.254516  -0.558292  -0.235359   \n",
       "Rugamonas_570800             -0.085515   0.037550  -0.007948  -0.218994   \n",
       "Solirubrobacter               0.583341   0.450109   0.284469   0.512432   \n",
       "Sphingomonas_L_486704        -0.092240   0.029287   0.249624   0.333105   \n",
       "Telluria_573210               0.066229   0.301587   0.188834   0.368911   \n",
       "Variovorax                   -0.030651  -0.292830  -0.228243  -0.119869   \n",
       "\n",
       "                             AT1G01050  AT1G01060  AT1G01070  AT1G01080  \\\n",
       "Genus                                                                     \n",
       "Actinomycetospora             0.114200   0.003286  -0.002191   0.130631   \n",
       "Actinoplanes                  0.088706  -0.125394   0.113895  -0.088159   \n",
       "Aeromicrobium                -0.054372   0.034753   0.161155   0.287557   \n",
       "Aetherobacter                -0.308037  -0.011858   0.066916  -0.609016   \n",
       "Agrobacterium                 0.039704   0.314075   0.283954  -0.158817   \n",
       "Amnibacterium_382409          0.375877   0.301948   0.101121   0.755436   \n",
       "Aquabacterium_A_592457        0.044122  -0.195314  -0.205904   0.315033   \n",
       "Aureimonas_A_501549           0.174651   0.069532  -0.116343  -0.080482   \n",
       "Blastococcus                  0.401260   0.371131   0.169543   0.493016   \n",
       "Brevundimonas                 0.261334  -0.238871  -0.018080  -0.104643   \n",
       "Caldora                       0.094691  -0.073679  -0.005323   0.012327   \n",
       "Chryseobacterium_A_796612     0.205367  -0.159912  -0.022453   0.273275   \n",
       "Curtobacterium                0.306436   0.151701  -0.143509   0.314324   \n",
       "Curvibacter_587305            0.172209  -0.084347   0.097127   0.252722   \n",
       "Duganella_570715              0.361916  -0.293936   0.154124  -0.028898   \n",
       "Flavobacterium               -0.298851  -0.192118  -0.072250  -0.516147   \n",
       "Friedmanniella                0.526503   0.333927   0.321326   0.660458   \n",
       "Frigoribacterium             -0.394089   0.015873  -0.169130   0.180077   \n",
       "Geodermatophilus             -0.007435  -0.144847  -0.241778   0.314752   \n",
       "Herbaspirillum                0.427626   0.393000   0.283064   0.303262   \n",
       "Hylemonella                   0.007398  -0.186875  -0.016715   0.220578   \n",
       "Janthinobacterium_571130      0.283636  -0.090423   0.125053   0.219323   \n",
       "Janthinobacterium_571526      0.102417   0.012081  -0.108732   0.257552   \n",
       "Kineococcus                   0.191323   0.186397   0.027645   0.252087   \n",
       "Kineosporia                  -0.117370   0.138959  -0.221665   0.054732   \n",
       "Klenkia                       0.090859   0.032841   0.309250  -0.131910   \n",
       "Kordiimonas                  -0.141782  -0.270699  -0.146709  -0.246339   \n",
       "Marmoricola_A_392027          0.302876   0.391675   0.209836   0.525721   \n",
       "Massilia_573596              -0.139866  -0.460380  -0.283564   0.080471   \n",
       "Massilia_574424              -0.373425   0.095976   0.002758  -0.172096   \n",
       "Massilia_574544              -0.190255  -0.296195  -0.122913   0.047085   \n",
       "Methylobacterium              0.543190   0.409309   0.455852   0.329911   \n",
       "Microbacterium_A_383321       0.178855  -0.075596  -0.183511   0.468091   \n",
       "Mycobacterium                -0.078052   0.110931  -0.126370  -0.299057   \n",
       "Nakamurella                   0.146261   0.158861   0.244865  -0.098603   \n",
       "Neorhizobium_500197           0.180624   0.173508   0.188834   0.183908   \n",
       "Nocardioides_A_392796         0.148919   0.405420   0.165070   0.158774   \n",
       "Noviherbaspirillum_A_568104   0.207957   0.082222   0.349232   0.148056   \n",
       "Patulibacter                 -0.493428  -0.531846  -0.296537  -0.592474   \n",
       "Paucimonas                   -0.232606  -0.030183  -0.194669  -0.202977   \n",
       "Pelomonas                     0.118467   0.136349  -0.061629   0.357318   \n",
       "Pigmentiphaga                 0.040220  -0.159770  -0.202487  -0.307891   \n",
       "Pseudoduganella_570666        0.416655   0.155321   0.229284   0.668676   \n",
       "Pseudoduganella_570691        0.095476   0.139542  -0.209792   0.183928   \n",
       "Pseudomonas_B_650453          0.555487   0.149399   0.074988   0.493190   \n",
       "Pseudomonas_E_647464         -0.535851  -0.435687  -0.328407  -0.030104   \n",
       "Pseudonocardia               -0.168422   0.112860  -0.409189  -0.074372   \n",
       "Ramlibacter_588642           -0.076047  -0.434045  -0.031297   0.004667   \n",
       "Rathayibacter                -0.031776  -0.248459   0.196959  -0.226270   \n",
       "Rhodoferax_C                  0.063492  -0.023536   0.302682  -0.162014   \n",
       "Rugamonas_570800             -0.120872  -0.360971  -0.209401   0.148280   \n",
       "Solirubrobacter               0.079219   0.271727   0.227686   0.410499   \n",
       "Sphingomonas_L_486704         0.031477   0.547420   0.257835  -0.053647   \n",
       "Telluria_573210               0.319102  -0.063492   0.070060   0.029009   \n",
       "Variovorax                   -0.145047   0.064039   0.172961  -0.484401   \n",
       "\n",
       "                             AT1G01090  AT1G01100  ...  ArthCt115  ArthCr091  \\\n",
       "Genus                                              ...                         \n",
       "Actinomycetospora             0.230590   0.135835  ...   0.036706  -0.097768   \n",
       "Actinoplanes                 -0.040246  -0.098015  ...  -0.140437   0.078303   \n",
       "Aeromicrobium                 0.197590   0.440024  ...  -0.174832  -0.371358   \n",
       "Aetherobacter                -0.611840  -0.662379  ...   0.180109   0.345307   \n",
       "Agrobacterium                -0.076397  -0.107338  ...  -0.166947   0.105422   \n",
       "Amnibacterium_382409          0.744389   0.511555  ...  -0.306858  -0.491161   \n",
       "Aquabacterium_A_592457        0.183843   0.280323  ...   0.204536   0.134720   \n",
       "Aureimonas_A_501549          -0.050096  -0.103203  ...   0.424012  -0.094717   \n",
       "Blastococcus                  0.603944   0.253355  ...  -0.242346  -0.327582   \n",
       "Brevundimonas                -0.007944   0.133680  ...   0.148796   0.144638   \n",
       "Caldora                       0.129989  -0.273706  ...  -0.202707  -0.219357   \n",
       "Chryseobacterium_A_796612     0.414567   0.546002  ...  -0.147907  -0.191128   \n",
       "Curtobacterium                0.713905   0.538538  ...  -0.146762  -0.341934   \n",
       "Curvibacter_587305            0.247610   0.298411  ...   0.047008  -0.126201   \n",
       "Duganella_570715              0.332742   0.097703  ...  -0.004438  -0.216599   \n",
       "Flavobacterium               -0.646415  -0.412698  ...   0.144516   0.322934   \n",
       "Friedmanniella                0.877414   0.670867  ...  -0.370195  -0.581564   \n",
       "Frigoribacterium             -0.091954   0.136836  ...   0.023167   0.308703   \n",
       "Geodermatophilus              0.315027   0.255547  ...  -0.230609   0.032769   \n",
       "Herbaspirillum                0.207176  -0.014716  ...   0.039837  -0.285661   \n",
       "Hylemonella                   0.046856   0.264968  ...   0.082288   0.103302   \n",
       "Janthinobacterium_571130      0.208879   0.142642  ...   0.186956  -0.200909   \n",
       "Janthinobacterium_571526      0.383308   0.134542  ...  -0.057278  -0.263867   \n",
       "Kineococcus                   0.503900   0.550705  ...   0.054063  -0.173806   \n",
       "Kineosporia                   0.062334  -0.144128  ...   0.220013   0.018548   \n",
       "Klenkia                      -0.002189  -0.022441  ...   0.057917  -0.028462   \n",
       "Kordiimonas                  -0.470234  -0.736280  ...   0.278314   0.223074   \n",
       "Marmoricola_A_392027          0.543537   0.299483  ...  -0.294395  -0.280252   \n",
       "Massilia_573596              -0.139045  -0.113316  ...   0.025101   0.191871   \n",
       "Massilia_574424              -0.436858  -0.275794  ...   0.159533   0.116385   \n",
       "Massilia_574544              -0.244457  -0.249932  ...  -0.096554   0.183137   \n",
       "Methylobacterium              0.425462  -0.001916  ...  -0.099603  -0.377276   \n",
       "Microbacterium_A_383321       0.581759   0.702274  ...  -0.005244  -0.245960   \n",
       "Mycobacterium                -0.082913  -0.287335  ...  -0.043218   0.053750   \n",
       "Nakamurella                  -0.037798   0.071213  ...   0.282369   0.084908   \n",
       "Neorhizobium_500197           0.451560   0.389163  ...   0.113627  -0.128626   \n",
       "Nocardioides_A_392796         0.175746   0.029838  ...  -0.070071  -0.236244   \n",
       "Noviherbaspirillum_A_568104   0.072615  -0.088156  ...  -0.229501  -0.061313   \n",
       "Patulibacter                 -0.669309  -0.262921  ...   0.335736   0.803771   \n",
       "Paucimonas                   -0.227622  -0.260021  ...   0.151250   0.106057   \n",
       "Pelomonas                     0.338478   0.349015  ...  -0.056636  -0.393081   \n",
       "Pigmentiphaga                -0.376681  -0.368637  ...   0.500917   0.296796   \n",
       "Pseudoduganella_570666        0.407890   0.376387  ...  -0.149072  -0.545953   \n",
       "Pseudoduganella_570691        0.230229   0.024588  ...   0.197903  -0.204364   \n",
       "Pseudomonas_B_650453          0.827174   0.633359  ...  -0.077894  -0.545681   \n",
       "Pseudomonas_E_647464         -0.441160   0.044335  ...   0.024821   0.403394   \n",
       "Pseudonocardia               -0.047748  -0.213565  ...  -0.005249   0.284465   \n",
       "Ramlibacter_588642           -0.375294   0.017021  ...   0.209713   0.227043   \n",
       "Rathayibacter                -0.165183   0.116422  ...   0.000000   0.298315   \n",
       "Rhodoferax_C                 -0.382047  -0.181719  ...   0.167131   0.050903   \n",
       "Rugamonas_570800             -0.018090   0.202823  ...  -0.116285   0.053721   \n",
       "Solirubrobacter               0.412992   0.275605  ...  -0.424289  -0.310229   \n",
       "Sphingomonas_L_486704         0.039688  -0.166416  ...   0.015722  -0.194608   \n",
       "Telluria_573210               0.292830   0.146141  ...  -0.040266  -0.154351   \n",
       "Variovorax                   -0.513410  -0.480569  ...   0.073361  -0.000547   \n",
       "\n",
       "                             ArthCr090  ArthCr089  ArthCt114  ArthCt124  \\\n",
       "Genus                                                                     \n",
       "Actinomycetospora             0.170889   0.095577   0.131179  -0.125428   \n",
       "Actinoplanes                 -0.158248  -0.158248   0.205886  -0.073648   \n",
       "Aeromicrobium                -0.019058   0.158353   0.068106  -0.177691   \n",
       "Aetherobacter                 0.479703   0.476315   0.264274   0.387376   \n",
       "Agrobacterium                -0.040526   0.014513  -0.044359   0.004107   \n",
       "Amnibacterium_382409         -0.612960  -0.580103  -0.426013  -0.684340   \n",
       "Aquabacterium_A_592457       -0.143838  -0.158252   0.180313  -0.086774   \n",
       "Aureimonas_A_501549           0.160690  -0.063509   0.149466   0.002737   \n",
       "Blastococcus                 -0.376609  -0.209532  -0.591893  -0.696522   \n",
       "Brevundimonas                 0.485139   0.096699  -0.081907  -0.072593   \n",
       "Caldora                      -0.387447  -0.419944  -0.122705  -0.117383   \n",
       "Chryseobacterium_A_796612    -0.346112  -0.485761  -0.192771  -0.159912   \n",
       "Curtobacterium               -0.329191  -0.268511  -0.370757  -0.452372   \n",
       "Curvibacter_587305           -0.391704  -0.430363   0.101920  -0.038979   \n",
       "Duganella_570715             -0.342926  -0.461821  -0.198985  -0.173389   \n",
       "Flavobacterium                0.306513   0.292830   0.283525   0.506842   \n",
       "Friedmanniella               -0.446240  -0.442953  -0.442405  -0.725106   \n",
       "Frigoribacterium              0.033935   0.322386   0.094691   0.139026   \n",
       "Geodermatophilus             -0.123092  -0.024784  -0.336231  -0.365145   \n",
       "Herbaspirillum                0.073002  -0.076176   0.032606  -0.182073   \n",
       "Hylemonella                   0.107138   0.013427   0.131799  -0.118098   \n",
       "Janthinobacterium_571130     -0.312494  -0.387251  -0.023911  -0.044249   \n",
       "Janthinobacterium_571526     -0.432732  -0.316861  -0.220759  -0.264142   \n",
       "Kineococcus                  -0.206377  -0.130012   0.029013  -0.213220   \n",
       "Kineosporia                   0.066895   0.101863   0.114633   0.106120   \n",
       "Klenkia                      -0.102901  -0.059661   0.101806  -0.062397   \n",
       "Kordiimonas                   0.192966  -0.027371   0.256193   0.339674   \n",
       "Marmoricola_A_392027         -0.301179  -0.120189  -0.358022  -0.524307   \n",
       "Massilia_573596              -0.256740  -0.310935   0.214589   0.277816   \n",
       "Massilia_574424               0.333159   0.409830   0.173750   0.392179   \n",
       "Massilia_574544              -0.113605  -0.169176   0.177936   0.313715   \n",
       "Methylobacterium             -0.173032  -0.201232  -0.144011  -0.452293   \n",
       "Microbacterium_A_383321      -0.213640  -0.321282  -0.120241  -0.274172   \n",
       "Mycobacterium                 0.265035   0.212142  -0.020013  -0.052893   \n",
       "Nakamurella                   0.210353   0.001917   0.250616   0.245412   \n",
       "Neorhizobium_500197          -0.110016  -0.104543   0.036672  -0.175151   \n",
       "Nocardioides_A_392796         0.065152   0.154120   0.015330  -0.352039   \n",
       "Noviherbaspirillum_A_568104  -0.136754  -0.166705   0.077701  -0.157098   \n",
       "Patulibacter                  0.612283   0.490426   0.453810   0.680114   \n",
       "Paucimonas                    0.142610   0.361648   0.138733   0.111319   \n",
       "Pelomonas                    -0.090687  -0.039276  -0.098350  -0.375838   \n",
       "Pigmentiphaga                 0.208866  -0.107068   0.119550   0.308446   \n",
       "Pseudoduganella_570666       -0.372552  -0.339406   0.026846  -0.335297   \n",
       "Pseudoduganella_570691       -0.079830  -0.088132  -0.084939  -0.188398   \n",
       "Pseudomonas_B_650453         -0.500112  -0.591251  -0.360519  -0.608556   \n",
       "Pseudomonas_E_647464          0.076081   0.174056   0.218391   0.408867   \n",
       "Pseudonocardia                0.116911   0.199096  -0.065401  -0.028649   \n",
       "Ramlibacter_588642            0.386275   0.136445   0.405218   0.278931   \n",
       "Rathayibacter                 0.111492   0.167374   0.164087   0.191481   \n",
       "Rhodoferax_C                  0.178435   0.241927   0.420909   0.229338   \n",
       "Rugamonas_570800             -0.121146  -0.213239   0.035905   0.155132   \n",
       "Solirubrobacter              -0.168410   0.009418  -0.274220  -0.441245   \n",
       "Sphingomonas_L_486704         0.062680   0.242233  -0.158478  -0.119885   \n",
       "Telluria_573210              -0.126437  -0.096880  -0.250684  -0.340449   \n",
       "Variovorax                    0.199234   0.255610   0.282978   0.328407   \n",
       "\n",
       "                             ArthCr088  ArthCt112  ArthCp088  ArthCp085  \n",
       "Genus                                                                    \n",
       "Actinomycetospora            -0.003286   0.156532  -0.035054  -0.252499  \n",
       "Actinoplanes                 -0.223682  -0.026972  -0.015880  -0.301164  \n",
       "Aeromicrobium                 0.044283  -0.233777  -0.132287  -0.365192  \n",
       "Aetherobacter                 0.432834   0.389971   0.029364   0.239146  \n",
       "Agrobacterium                 0.087623   0.378064   0.135542   0.220427  \n",
       "Amnibacterium_382409         -0.478981  -0.279609   0.014729   0.002549  \n",
       "Aquabacterium_A_592457       -0.108247  -0.140181  -0.492698  -0.480638  \n",
       "Aureimonas_A_501549           0.060224  -0.016975   0.096907   0.070627  \n",
       "Blastococcus                 -0.066831  -0.170936  -0.227335  -0.012051  \n",
       "Brevundimonas                 0.229010  -0.009178  -0.365977  -0.174771  \n",
       "Caldora                      -0.451601  -0.221068   0.340382   0.228602  \n",
       "Chryseobacterium_A_796612    -0.421139  -0.325346  -0.010405   0.214677  \n",
       "Curtobacterium               -0.083436  -0.331967  -0.365903  -0.169602  \n",
       "Curvibacter_587305           -0.350808  -0.253875   0.011182  -0.204159  \n",
       "Duganella_570715             -0.355861  -0.645344  -0.014862   0.058347  \n",
       "Flavobacterium                0.245211   0.237580   0.105090   0.053640  \n",
       "Friedmanniella               -0.359951  -0.228493  -0.064923  -0.132311  \n",
       "Frigoribacterium              0.294472   0.250718  -0.066229   0.241379  \n",
       "Geodermatophilus              0.003580   0.100387  -0.186703  -0.029465  \n",
       "Herbaspirillum               -0.034048   0.043576  -0.007214  -0.114553  \n",
       "Hylemonella                   0.057816  -0.174020  -0.496781  -0.416496  \n",
       "Janthinobacterium_571130     -0.351797  -0.413692   0.229767   0.089598  \n",
       "Janthinobacterium_571526     -0.227074  -0.411646  -0.024163  -0.171885  \n",
       "Kineococcus                   0.013138  -0.104572  -0.105378  -0.177638  \n",
       "Kineosporia                   0.174231  -0.079220  -0.159635  -0.184569  \n",
       "Klenkia                      -0.036125   0.008759   0.268747   0.303777  \n",
       "Kordiimonas                  -0.070891   0.231317   0.257014   0.283564  \n",
       "Marmoricola_A_392027         -0.047227  -0.095033   0.059105  -0.032522  \n",
       "Massilia_573596              -0.366498  -0.054339   0.317230   0.133023  \n",
       "Massilia_574424               0.250421   0.254041   0.337572   0.236631  \n",
       "Massilia_574544              -0.252395   0.108145   0.432795   0.212428  \n",
       "Methylobacterium             -0.047912  -0.205367  -0.146201  -0.239562  \n",
       "Microbacterium_A_383321      -0.204602  -0.289549  -0.317721  -0.238017  \n",
       "Mycobacterium                 0.159821   0.406042  -0.001715   0.086915  \n",
       "Nakamurella                   0.105177   0.224764   0.223774   0.198850  \n",
       "Neorhizobium_500197           0.009305  -0.212673  -0.294472  -0.348659  \n",
       "Nocardioides_A_392796         0.058034   0.149076   0.015877  -0.184506  \n",
       "Noviherbaspirillum_A_568104  -0.324650  -0.076582   0.260511   0.105109  \n",
       "Patulibacter                  0.365569   0.449069  -0.130860  -0.050423  \n",
       "Paucimonas                    0.159225  -0.153569  -0.400415  -0.013292  \n",
       "Pelomonas                    -0.134753  -0.315690  -0.055242  -0.225758  \n",
       "Pigmentiphaga                 0.069067   0.178934   0.111506  -0.028570  \n",
       "Pseudoduganella_570666       -0.388166  -0.368767  -0.006301  -0.110396  \n",
       "Pseudoduganella_570691        0.020436  -0.214612  -0.342629  -0.162533  \n",
       "Pseudomonas_B_650453         -0.337446  -0.512295  -0.338022  -0.217465  \n",
       "Pseudomonas_E_647464         -0.139573   0.010948   0.301587   0.170224  \n",
       "Pseudonocardia                0.098969   0.329364  -0.217617  -0.048038  \n",
       "Ramlibacter_588642            0.071105   0.048051  -0.097186  -0.414827  \n",
       "Rathayibacter                 0.111766   0.030000  -0.024380   0.052596  \n",
       "Rhodoferax_C                  0.007663   0.085398   0.239190   0.090312  \n",
       "Rugamonas_570800             -0.274634  -0.107593   0.217350   0.190764  \n",
       "Solirubrobacter               0.034624  -0.203477  -0.126030  -0.088083  \n",
       "Sphingomonas_L_486704         0.337485   0.109225  -0.029834  -0.006295  \n",
       "Telluria_573210               0.006021  -0.173259  -0.357417  -0.221675  \n",
       "Variovorax                    0.117679   0.029287   0.267652   0.186097  \n",
       "\n",
       "[55 rows x 15091 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "full_spearman_correlation_matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e0e57ee1",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "c8089929",
   "metadata": {},
   "outputs": [],
   "source": [
    "top_200_spearman = np.array([full_spearman_correlation_matrix.T[c].nlargest(300).index.values for c in full_spearman_correlation_matrix.T])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "89ef551a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>...</th>\n",
       "      <th>290</th>\n",
       "      <th>291</th>\n",
       "      <th>292</th>\n",
       "      <th>293</th>\n",
       "      <th>294</th>\n",
       "      <th>295</th>\n",
       "      <th>296</th>\n",
       "      <th>297</th>\n",
       "      <th>298</th>\n",
       "      <th>299</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genus</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Actinomycetospora</th>\n",
       "      <td>AT4G28020</td>\n",
       "      <td>AT5G47610</td>\n",
       "      <td>AT1G05560</td>\n",
       "      <td>AT2G37050</td>\n",
       "      <td>AT1G48620</td>\n",
       "      <td>ArthCt091</td>\n",
       "      <td>AT5G03415</td>\n",
       "      <td>AT4G12970</td>\n",
       "      <td>AT3G54220</td>\n",
       "      <td>AT5G02010</td>\n",
       "      <td>...</td>\n",
       "      <td>AT5G58150</td>\n",
       "      <td>AT2G18330</td>\n",
       "      <td>AT2G21290</td>\n",
       "      <td>AT2G37630</td>\n",
       "      <td>AT4G24940</td>\n",
       "      <td>AT5G09250</td>\n",
       "      <td>AT5G65770</td>\n",
       "      <td>AT5G51170</td>\n",
       "      <td>AT3G53235</td>\n",
       "      <td>AT5G55530</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Actinoplanes</th>\n",
       "      <td>AT2G38995</td>\n",
       "      <td>AT1G01900</td>\n",
       "      <td>AT1G70890</td>\n",
       "      <td>AT5G17700</td>\n",
       "      <td>AT1G69523</td>\n",
       "      <td>AT3G62750</td>\n",
       "      <td>AT3G48185</td>\n",
       "      <td>AT4G22635</td>\n",
       "      <td>AT5G01495</td>\n",
       "      <td>AT4G33470</td>\n",
       "      <td>...</td>\n",
       "      <td>AT4G33260</td>\n",
       "      <td>AT5G08640</td>\n",
       "      <td>AT5G02060</td>\n",
       "      <td>AT1G08570</td>\n",
       "      <td>AT1G44575</td>\n",
       "      <td>AT1G57770</td>\n",
       "      <td>AT1G10500</td>\n",
       "      <td>AT4G33790</td>\n",
       "      <td>AT1G47240</td>\n",
       "      <td>AT4G24990</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Aeromicrobium</th>\n",
       "      <td>AT2G04430</td>\n",
       "      <td>AT1G12440</td>\n",
       "      <td>AT2G38860</td>\n",
       "      <td>AT1G35260</td>\n",
       "      <td>AT1G08213</td>\n",
       "      <td>AT3G09150</td>\n",
       "      <td>AT3G02060</td>\n",
       "      <td>AT1G13145</td>\n",
       "      <td>AT2G37130</td>\n",
       "      <td>AT3G11591</td>\n",
       "      <td>...</td>\n",
       "      <td>AT4G29140</td>\n",
       "      <td>AT1G76120</td>\n",
       "      <td>AT5G47930</td>\n",
       "      <td>AT1G31120</td>\n",
       "      <td>AT1G70850</td>\n",
       "      <td>AT2G39080</td>\n",
       "      <td>AT3G49100</td>\n",
       "      <td>AT5G44785</td>\n",
       "      <td>AT1G26460</td>\n",
       "      <td>AT3G07230</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Aetherobacter</th>\n",
       "      <td>ArthCp002</td>\n",
       "      <td>ArthCp030</td>\n",
       "      <td>AT1G28010</td>\n",
       "      <td>AT5G24150</td>\n",
       "      <td>ArthCp007</td>\n",
       "      <td>ArthCp053</td>\n",
       "      <td>AT3G28500</td>\n",
       "      <td>ArthCp017</td>\n",
       "      <td>ArthCp015</td>\n",
       "      <td>ArthCp042</td>\n",
       "      <td>...</td>\n",
       "      <td>AT2G34920</td>\n",
       "      <td>AT3G23480</td>\n",
       "      <td>AT1G23550</td>\n",
       "      <td>AT4G38850</td>\n",
       "      <td>AT3G06080</td>\n",
       "      <td>AT5G52550</td>\n",
       "      <td>AT4G17970</td>\n",
       "      <td>AT5G12230</td>\n",
       "      <td>AT5G40380</td>\n",
       "      <td>AT2G15340</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Agrobacterium</th>\n",
       "      <td>AT5G46500</td>\n",
       "      <td>AT3G45443</td>\n",
       "      <td>AT3G12110</td>\n",
       "      <td>AT1G02960</td>\n",
       "      <td>AT2G32250</td>\n",
       "      <td>AT1G16820</td>\n",
       "      <td>AT2G10606</td>\n",
       "      <td>AT5G48340</td>\n",
       "      <td>AT2G40670</td>\n",
       "      <td>AT4G03845</td>\n",
       "      <td>...</td>\n",
       "      <td>AT5G03530</td>\n",
       "      <td>AT5G50340</td>\n",
       "      <td>AT2G47240</td>\n",
       "      <td>AT4G40042</td>\n",
       "      <td>AT5G46150</td>\n",
       "      <td>AT3G29760</td>\n",
       "      <td>AT5G45250</td>\n",
       "      <td>AT1G05997</td>\n",
       "      <td>AT5G14100</td>\n",
       "      <td>AT2G20190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Amnibacterium_382409</th>\n",
       "      <td>AT2G22230</td>\n",
       "      <td>AT5G16390</td>\n",
       "      <td>AT1G53520</td>\n",
       "      <td>AT5G52920</td>\n",
       "      <td>AT1G73620</td>\n",
       "      <td>AT5G19370</td>\n",
       "      <td>AT1G33170</td>\n",
       "      <td>AT4G35335</td>\n",
       "      <td>AT3G18080</td>\n",
       "      <td>AT3G51820</td>\n",
       "      <td>...</td>\n",
       "      <td>AT2G45270</td>\n",
       "      <td>AT5G63310</td>\n",
       "      <td>AT3G14220</td>\n",
       "      <td>AT5G05960</td>\n",
       "      <td>AT5G44520</td>\n",
       "      <td>AT2G24570</td>\n",
       "      <td>AT2G43030</td>\n",
       "      <td>AT5G64050</td>\n",
       "      <td>AT5G66530</td>\n",
       "      <td>AT1G21050</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Aquabacterium_A_592457</th>\n",
       "      <td>AT3G46630</td>\n",
       "      <td>AT5G08415</td>\n",
       "      <td>AT2G42710</td>\n",
       "      <td>AT4G29440</td>\n",
       "      <td>AT2G43290</td>\n",
       "      <td>AT2G24970</td>\n",
       "      <td>AT3G13445</td>\n",
       "      <td>AT1G52270</td>\n",
       "      <td>AT5G03370</td>\n",
       "      <td>AT2G39900</td>\n",
       "      <td>...</td>\n",
       "      <td>AT3G23700</td>\n",
       "      <td>AT2G40085</td>\n",
       "      <td>AT1G66070</td>\n",
       "      <td>AT4G00880</td>\n",
       "      <td>AT5G42740</td>\n",
       "      <td>AT1G12090</td>\n",
       "      <td>AT1G71850</td>\n",
       "      <td>AT4G09500</td>\n",
       "      <td>AT3G20320</td>\n",
       "      <td>AT1G70480</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Aureimonas_A_501549</th>\n",
       "      <td>AT3G21670</td>\n",
       "      <td>AT2G27420</td>\n",
       "      <td>AT5G42760</td>\n",
       "      <td>AT3G24460</td>\n",
       "      <td>AT1G64780</td>\n",
       "      <td>AT3G27170</td>\n",
       "      <td>AT4G27360</td>\n",
       "      <td>AT1G19490</td>\n",
       "      <td>AT3G51375</td>\n",
       "      <td>AT3G47430</td>\n",
       "      <td>...</td>\n",
       "      <td>AT5G24850</td>\n",
       "      <td>AT3G04140</td>\n",
       "      <td>AT5G20070</td>\n",
       "      <td>AT3G57020</td>\n",
       "      <td>AT2G30040</td>\n",
       "      <td>AT5G17600</td>\n",
       "      <td>AT3G52110</td>\n",
       "      <td>AT1G70580</td>\n",
       "      <td>AT3G19710</td>\n",
       "      <td>AT5G07460</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Blastococcus</th>\n",
       "      <td>AT2G21410</td>\n",
       "      <td>AT5G05620</td>\n",
       "      <td>AT4G15800</td>\n",
       "      <td>AT3G50800</td>\n",
       "      <td>AT5G17520</td>\n",
       "      <td>AT3G46550</td>\n",
       "      <td>AT1G29890</td>\n",
       "      <td>AT3G09035</td>\n",
       "      <td>AT3G06550</td>\n",
       "      <td>AT5G22050</td>\n",
       "      <td>...</td>\n",
       "      <td>AT5G46620</td>\n",
       "      <td>AT2G30410</td>\n",
       "      <td>AT1G28570</td>\n",
       "      <td>AT3G47833</td>\n",
       "      <td>AT5G05980</td>\n",
       "      <td>AT5G59160</td>\n",
       "      <td>AT1G08040</td>\n",
       "      <td>AT4G32530</td>\n",
       "      <td>AT2G42500</td>\n",
       "      <td>AT2G46910</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Brevundimonas</th>\n",
       "      <td>AT5G63690</td>\n",
       "      <td>AT5G05140</td>\n",
       "      <td>AT2G29440</td>\n",
       "      <td>AT5G62430</td>\n",
       "      <td>AT2G31840</td>\n",
       "      <td>AT5G58710</td>\n",
       "      <td>AT5G48470</td>\n",
       "      <td>AT4G07705</td>\n",
       "      <td>AT4G08390</td>\n",
       "      <td>AT5G09270</td>\n",
       "      <td>...</td>\n",
       "      <td>AT2G39250</td>\n",
       "      <td>AT3G09890</td>\n",
       "      <td>AT1G19430</td>\n",
       "      <td>AT2G21280</td>\n",
       "      <td>AT3G21870</td>\n",
       "      <td>AT1G03860</td>\n",
       "      <td>AT4G18170</td>\n",
       "      <td>AT1G56460</td>\n",
       "      <td>AT5G23010</td>\n",
       "      <td>AT2G29310</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Caldora</th>\n",
       "      <td>AT5G02575</td>\n",
       "      <td>AT5G02565</td>\n",
       "      <td>AT1G77490</td>\n",
       "      <td>AT3G27200</td>\n",
       "      <td>AT4G07835</td>\n",
       "      <td>AT1G58070</td>\n",
       "      <td>AT4G33960</td>\n",
       "      <td>AT5G35170</td>\n",
       "      <td>AT1G69320</td>\n",
       "      <td>AT3G49110</td>\n",
       "      <td>...</td>\n",
       "      <td>AT1G73470</td>\n",
       "      <td>AT1G64660</td>\n",
       "      <td>AT3G44070</td>\n",
       "      <td>AT5G52570</td>\n",
       "      <td>AT1G55152</td>\n",
       "      <td>AT5G62560</td>\n",
       "      <td>AT1G79790</td>\n",
       "      <td>AT5G51260</td>\n",
       "      <td>AT1G49240</td>\n",
       "      <td>AT1G02405</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Chryseobacterium_A_796612</th>\n",
       "      <td>AT4G17520</td>\n",
       "      <td>AT3G28460</td>\n",
       "      <td>AT3G63160</td>\n",
       "      <td>AT3G61110</td>\n",
       "      <td>AT3G59540</td>\n",
       "      <td>AT4G00100</td>\n",
       "      <td>AT4G24770</td>\n",
       "      <td>AT1G56070</td>\n",
       "      <td>AT1G31500</td>\n",
       "      <td>AT1G74970</td>\n",
       "      <td>...</td>\n",
       "      <td>AT3G15140</td>\n",
       "      <td>AT4G26670</td>\n",
       "      <td>AT1G69620</td>\n",
       "      <td>AT5G15960</td>\n",
       "      <td>AT1G29900</td>\n",
       "      <td>AT1G35680</td>\n",
       "      <td>AT4G10450</td>\n",
       "      <td>AT5G66060</td>\n",
       "      <td>AT3G53020</td>\n",
       "      <td>AT4G13850</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Curtobacterium</th>\n",
       "      <td>AT1G55010</td>\n",
       "      <td>AT3G14750</td>\n",
       "      <td>AT3G20170</td>\n",
       "      <td>AT5G01870</td>\n",
       "      <td>AT1G80020</td>\n",
       "      <td>AT1G60989</td>\n",
       "      <td>AT1G71310</td>\n",
       "      <td>AT4G13235</td>\n",
       "      <td>AT2G14580</td>\n",
       "      <td>AT4G22505</td>\n",
       "      <td>...</td>\n",
       "      <td>AT4G20020</td>\n",
       "      <td>AT5G18310</td>\n",
       "      <td>AT5G27720</td>\n",
       "      <td>AT2G33220</td>\n",
       "      <td>AT3G58030</td>\n",
       "      <td>AT3G58600</td>\n",
       "      <td>AT4G16440</td>\n",
       "      <td>AT4G27070</td>\n",
       "      <td>AT1G70190</td>\n",
       "      <td>AT3G22330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Curvibacter_587305</th>\n",
       "      <td>AT1G34245</td>\n",
       "      <td>AT5G16140</td>\n",
       "      <td>AT1G78790</td>\n",
       "      <td>AT2G05440</td>\n",
       "      <td>AT3G24110</td>\n",
       "      <td>AT5G53080</td>\n",
       "      <td>AT2G47960</td>\n",
       "      <td>AT5G56910</td>\n",
       "      <td>AT1G79850</td>\n",
       "      <td>AT3G60530</td>\n",
       "      <td>...</td>\n",
       "      <td>AT2G02540</td>\n",
       "      <td>AT5G08260</td>\n",
       "      <td>AT5G58780</td>\n",
       "      <td>AT3G48820</td>\n",
       "      <td>AT5G11230</td>\n",
       "      <td>AT3G15530</td>\n",
       "      <td>AT2G33330</td>\n",
       "      <td>AT4G01985</td>\n",
       "      <td>AT4G32120</td>\n",
       "      <td>AT1G09900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Duganella_570715</th>\n",
       "      <td>AT5G06600</td>\n",
       "      <td>AT3G16565</td>\n",
       "      <td>AT3G61970</td>\n",
       "      <td>AT5G62900</td>\n",
       "      <td>AT1G08830</td>\n",
       "      <td>AT5G38510</td>\n",
       "      <td>AT5G55610</td>\n",
       "      <td>AT3G05990</td>\n",
       "      <td>AT4G39730</td>\n",
       "      <td>AT5G27520</td>\n",
       "      <td>...</td>\n",
       "      <td>AT4G27830</td>\n",
       "      <td>AT1G69640</td>\n",
       "      <td>AT4G39680</td>\n",
       "      <td>AT3G22190</td>\n",
       "      <td>AT1G53035</td>\n",
       "      <td>AT3G62980</td>\n",
       "      <td>AT4G10810</td>\n",
       "      <td>AT5G44250</td>\n",
       "      <td>AT5G10730</td>\n",
       "      <td>AT5G45300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Flavobacterium</th>\n",
       "      <td>AT1G04907</td>\n",
       "      <td>AT1G29071</td>\n",
       "      <td>AT1G70700</td>\n",
       "      <td>AT5G06845</td>\n",
       "      <td>AT2G01480</td>\n",
       "      <td>AT2G05655</td>\n",
       "      <td>AT1G49700</td>\n",
       "      <td>AT3G16410</td>\n",
       "      <td>AT1G80540</td>\n",
       "      <td>AT3G27660</td>\n",
       "      <td>...</td>\n",
       "      <td>AT3G01970</td>\n",
       "      <td>AT1G35260</td>\n",
       "      <td>AT5G61340</td>\n",
       "      <td>ArthCr091</td>\n",
       "      <td>AT2G42570</td>\n",
       "      <td>AT4G08125</td>\n",
       "      <td>AT3G13525</td>\n",
       "      <td>AT5G44130</td>\n",
       "      <td>AT2G15340</td>\n",
       "      <td>ArthCp019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Friedmanniella</th>\n",
       "      <td>AT3G06430</td>\n",
       "      <td>AT3G06050</td>\n",
       "      <td>AT4G39900</td>\n",
       "      <td>AT1G67700</td>\n",
       "      <td>AT2G41770</td>\n",
       "      <td>AT5G19770</td>\n",
       "      <td>AT3G03070</td>\n",
       "      <td>AT2G37860</td>\n",
       "      <td>AT1G56340</td>\n",
       "      <td>AT1G01090</td>\n",
       "      <td>...</td>\n",
       "      <td>AT3G63190</td>\n",
       "      <td>AT5G14320</td>\n",
       "      <td>AT5G42190</td>\n",
       "      <td>AT2G42130</td>\n",
       "      <td>AT3G46940</td>\n",
       "      <td>AT4G03120</td>\n",
       "      <td>AT5G16050</td>\n",
       "      <td>AT5G62790</td>\n",
       "      <td>AT1G80080</td>\n",
       "      <td>AT2G42690</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Frigoribacterium</th>\n",
       "      <td>AT1G61910</td>\n",
       "      <td>AT1G66660</td>\n",
       "      <td>AT3G57790</td>\n",
       "      <td>AT2G42650</td>\n",
       "      <td>AT1G29750</td>\n",
       "      <td>AT1G80550</td>\n",
       "      <td>AT3G26850</td>\n",
       "      <td>AT3G08990</td>\n",
       "      <td>AT2G41990</td>\n",
       "      <td>AT2G35340</td>\n",
       "      <td>...</td>\n",
       "      <td>AT2G41590</td>\n",
       "      <td>AT5G64030</td>\n",
       "      <td>AT1G30835</td>\n",
       "      <td>AT3G50370</td>\n",
       "      <td>AT1G16670</td>\n",
       "      <td>AT1G32190</td>\n",
       "      <td>AT5G17930</td>\n",
       "      <td>AT3G63052</td>\n",
       "      <td>AT4G27730</td>\n",
       "      <td>AT1G10890</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Geodermatophilus</th>\n",
       "      <td>AT1G49150</td>\n",
       "      <td>AT5G66100</td>\n",
       "      <td>AT3G13225</td>\n",
       "      <td>AT1G67890</td>\n",
       "      <td>AT1G02370</td>\n",
       "      <td>AT2G01190</td>\n",
       "      <td>AT5G10600</td>\n",
       "      <td>AT1G78850</td>\n",
       "      <td>AT3G01430</td>\n",
       "      <td>AT1G15130</td>\n",
       "      <td>...</td>\n",
       "      <td>AT5G57480</td>\n",
       "      <td>AT5G59140</td>\n",
       "      <td>AT5G01270</td>\n",
       "      <td>AT3G19930</td>\n",
       "      <td>AT3G15351</td>\n",
       "      <td>AT1G18390</td>\n",
       "      <td>AT2G35510</td>\n",
       "      <td>AT5G10940</td>\n",
       "      <td>AT5G43560</td>\n",
       "      <td>AT5G65180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Herbaspirillum</th>\n",
       "      <td>AT1G61000</td>\n",
       "      <td>AT5G38980</td>\n",
       "      <td>AT3G01710</td>\n",
       "      <td>AT5G03555</td>\n",
       "      <td>AT5G25830</td>\n",
       "      <td>AT5G33280</td>\n",
       "      <td>AT4G15430</td>\n",
       "      <td>AT2G28620</td>\n",
       "      <td>AT3G02380</td>\n",
       "      <td>AT3G09050</td>\n",
       "      <td>...</td>\n",
       "      <td>AT1G77122</td>\n",
       "      <td>AT2G27920</td>\n",
       "      <td>AT1G19840</td>\n",
       "      <td>AT1G62500</td>\n",
       "      <td>AT2G43750</td>\n",
       "      <td>AT3G10080</td>\n",
       "      <td>AT1G15980</td>\n",
       "      <td>AT2G32530</td>\n",
       "      <td>AT1G23090</td>\n",
       "      <td>AT5G51970</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Hylemonella</th>\n",
       "      <td>AT4G10760</td>\n",
       "      <td>AT3G55510</td>\n",
       "      <td>AT1G18340</td>\n",
       "      <td>AT5G60730</td>\n",
       "      <td>AT2G30600</td>\n",
       "      <td>AT5G60990</td>\n",
       "      <td>AT3G27550</td>\n",
       "      <td>AT5G38890</td>\n",
       "      <td>AT3G49890</td>\n",
       "      <td>AT3G10270</td>\n",
       "      <td>...</td>\n",
       "      <td>AT2G36885</td>\n",
       "      <td>AT3G07300</td>\n",
       "      <td>AT3G14800</td>\n",
       "      <td>AT4G18905</td>\n",
       "      <td>AT5G42800</td>\n",
       "      <td>AT2G24500</td>\n",
       "      <td>AT3G57490</td>\n",
       "      <td>AT1G29880</td>\n",
       "      <td>AT2G41790</td>\n",
       "      <td>AT5G06000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Janthinobacterium_571130</th>\n",
       "      <td>AT1G27340</td>\n",
       "      <td>AT3G09455</td>\n",
       "      <td>AT1G08830</td>\n",
       "      <td>AT2G28190</td>\n",
       "      <td>AT1G13440</td>\n",
       "      <td>AT5G12860</td>\n",
       "      <td>AT4G13940</td>\n",
       "      <td>AT5G51010</td>\n",
       "      <td>AT3G52930</td>\n",
       "      <td>AT1G32930</td>\n",
       "      <td>...</td>\n",
       "      <td>AT2G44525</td>\n",
       "      <td>AT4G35830</td>\n",
       "      <td>AT4G35630</td>\n",
       "      <td>AT5G26330</td>\n",
       "      <td>AT1G80490</td>\n",
       "      <td>AT1G20190</td>\n",
       "      <td>AT1G64850</td>\n",
       "      <td>AT1G79600</td>\n",
       "      <td>AT3G12640</td>\n",
       "      <td>AT4G01840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Janthinobacterium_571526</th>\n",
       "      <td>AT2G30870</td>\n",
       "      <td>AT2G43150</td>\n",
       "      <td>AT1G52910</td>\n",
       "      <td>AT1G08830</td>\n",
       "      <td>AT3G12500</td>\n",
       "      <td>AT3G01480</td>\n",
       "      <td>AT5G07440</td>\n",
       "      <td>AT5G19250</td>\n",
       "      <td>AT1G75690</td>\n",
       "      <td>AT5G66040</td>\n",
       "      <td>...</td>\n",
       "      <td>AT3G01920</td>\n",
       "      <td>AT4G21105</td>\n",
       "      <td>AT3G14310</td>\n",
       "      <td>AT3G52730</td>\n",
       "      <td>AT3G57750</td>\n",
       "      <td>AT5G16710</td>\n",
       "      <td>AT4G01060</td>\n",
       "      <td>AT1G60610</td>\n",
       "      <td>AT3G57070</td>\n",
       "      <td>AT2G18328</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Kineococcus</th>\n",
       "      <td>AT2G46900</td>\n",
       "      <td>AT1G55350</td>\n",
       "      <td>AT1G16690</td>\n",
       "      <td>AT5G07180</td>\n",
       "      <td>AT3G55390</td>\n",
       "      <td>AT1G73400</td>\n",
       "      <td>AT4G38980</td>\n",
       "      <td>AT5G11230</td>\n",
       "      <td>AT5G05960</td>\n",
       "      <td>AT4G22505</td>\n",
       "      <td>...</td>\n",
       "      <td>AT4G31200</td>\n",
       "      <td>AT1G79010</td>\n",
       "      <td>AT5G49510</td>\n",
       "      <td>AT1G16610</td>\n",
       "      <td>AT1G18450</td>\n",
       "      <td>AT1G29900</td>\n",
       "      <td>AT4G11160</td>\n",
       "      <td>AT1G15730</td>\n",
       "      <td>AT5G56370</td>\n",
       "      <td>AT3G42150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Kineosporia</th>\n",
       "      <td>AT1G09180</td>\n",
       "      <td>ArthCp010</td>\n",
       "      <td>AT3G21760</td>\n",
       "      <td>AT2G27420</td>\n",
       "      <td>AT1G52910</td>\n",
       "      <td>AT4G28290</td>\n",
       "      <td>AT1G07500</td>\n",
       "      <td>AT1G52342</td>\n",
       "      <td>AT5G02940</td>\n",
       "      <td>AT1G65870</td>\n",
       "      <td>...</td>\n",
       "      <td>AT5G58990</td>\n",
       "      <td>AT2G33250</td>\n",
       "      <td>AT2G21370</td>\n",
       "      <td>AT2G26550</td>\n",
       "      <td>AT1G77122</td>\n",
       "      <td>AT2G03980</td>\n",
       "      <td>AT2G45360</td>\n",
       "      <td>AT3G61990</td>\n",
       "      <td>AT3G15356</td>\n",
       "      <td>AT3G50800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Klenkia</th>\n",
       "      <td>AT2G28570</td>\n",
       "      <td>AT1G70830</td>\n",
       "      <td>AT3G22750</td>\n",
       "      <td>AT5G54250</td>\n",
       "      <td>AT4G24540</td>\n",
       "      <td>AT5G56890</td>\n",
       "      <td>AT3G22942</td>\n",
       "      <td>AT1G74800</td>\n",
       "      <td>AT1G04295</td>\n",
       "      <td>AT2G47920</td>\n",
       "      <td>...</td>\n",
       "      <td>AT3G57410</td>\n",
       "      <td>AT1G65890</td>\n",
       "      <td>AT2G38980</td>\n",
       "      <td>AT4G38640</td>\n",
       "      <td>AT1G15760</td>\n",
       "      <td>AT3G46590</td>\n",
       "      <td>AT5G53440</td>\n",
       "      <td>AT2G33360</td>\n",
       "      <td>AT5G19650</td>\n",
       "      <td>AT2G34420</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Kordiimonas</th>\n",
       "      <td>ArthCp021</td>\n",
       "      <td>AT5G04455</td>\n",
       "      <td>ArthCp035</td>\n",
       "      <td>ArthCp060</td>\n",
       "      <td>ArthCp015</td>\n",
       "      <td>ArthCp062</td>\n",
       "      <td>AT5G24150</td>\n",
       "      <td>ArthCp039</td>\n",
       "      <td>ArthCp048</td>\n",
       "      <td>AT5G08155</td>\n",
       "      <td>...</td>\n",
       "      <td>AT5G51130</td>\n",
       "      <td>AT4G27350</td>\n",
       "      <td>AT4G32340</td>\n",
       "      <td>AT2G33390</td>\n",
       "      <td>ArthCr091</td>\n",
       "      <td>AT1G10960</td>\n",
       "      <td>AT5G22580</td>\n",
       "      <td>AT2G31141</td>\n",
       "      <td>AT4G28025</td>\n",
       "      <td>AT5G44260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Marmoricola_A_392027</th>\n",
       "      <td>AT1G03160</td>\n",
       "      <td>AT4G05190</td>\n",
       "      <td>AT1G50200</td>\n",
       "      <td>AT2G37080</td>\n",
       "      <td>AT5G26670</td>\n",
       "      <td>AT3G12700</td>\n",
       "      <td>AT1G12250</td>\n",
       "      <td>AT1G78915</td>\n",
       "      <td>AT3G60900</td>\n",
       "      <td>AT3G01390</td>\n",
       "      <td>...</td>\n",
       "      <td>AT2G21410</td>\n",
       "      <td>AT3G09300</td>\n",
       "      <td>AT1G50430</td>\n",
       "      <td>AT3G27360</td>\n",
       "      <td>AT3G29180</td>\n",
       "      <td>AT4G01610</td>\n",
       "      <td>AT1G09340</td>\n",
       "      <td>AT3G54180</td>\n",
       "      <td>AT5G04530</td>\n",
       "      <td>AT5G47750</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Massilia_573596</th>\n",
       "      <td>AT5G04085</td>\n",
       "      <td>AT3G06325</td>\n",
       "      <td>AT5G45307</td>\n",
       "      <td>AT2G04965</td>\n",
       "      <td>ArthCp031</td>\n",
       "      <td>AT5G15780</td>\n",
       "      <td>AT1G49500</td>\n",
       "      <td>AT5G04305</td>\n",
       "      <td>AT2G03020</td>\n",
       "      <td>AT3G57645</td>\n",
       "      <td>...</td>\n",
       "      <td>AT5G47330</td>\n",
       "      <td>AT4G38580</td>\n",
       "      <td>AT5G42130</td>\n",
       "      <td>AT5G42940</td>\n",
       "      <td>AT5G66567</td>\n",
       "      <td>AT3G44860</td>\n",
       "      <td>AT5G14230</td>\n",
       "      <td>AT1G09350</td>\n",
       "      <td>AT1G23150</td>\n",
       "      <td>AT1G28050</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Massilia_574424</th>\n",
       "      <td>ArthCp048</td>\n",
       "      <td>ArthCp033</td>\n",
       "      <td>ArthCp045</td>\n",
       "      <td>ArthCp080</td>\n",
       "      <td>ArthCp076</td>\n",
       "      <td>ArthCp077</td>\n",
       "      <td>ArthCp004</td>\n",
       "      <td>ArthCp034</td>\n",
       "      <td>ArthCp023</td>\n",
       "      <td>ArthCp079</td>\n",
       "      <td>...</td>\n",
       "      <td>AT2G04066</td>\n",
       "      <td>AT1G08900</td>\n",
       "      <td>AT4G03995</td>\n",
       "      <td>AT1G04217</td>\n",
       "      <td>AT3G56825</td>\n",
       "      <td>AT5G54250</td>\n",
       "      <td>AT1G19450</td>\n",
       "      <td>AT2G15080</td>\n",
       "      <td>AT5G09830</td>\n",
       "      <td>AT4G19370</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Massilia_574544</th>\n",
       "      <td>AT3G06325</td>\n",
       "      <td>ArthCp031</td>\n",
       "      <td>AT5G24570</td>\n",
       "      <td>AT1G04193</td>\n",
       "      <td>AT5G57380</td>\n",
       "      <td>AT4G04595</td>\n",
       "      <td>AT5G14780</td>\n",
       "      <td>AT4G06375</td>\n",
       "      <td>AT1G49500</td>\n",
       "      <td>AT5G39110</td>\n",
       "      <td>...</td>\n",
       "      <td>AT5G47330</td>\n",
       "      <td>AT3G02445</td>\n",
       "      <td>AT1G69890</td>\n",
       "      <td>AT3G58540</td>\n",
       "      <td>AT2G01300</td>\n",
       "      <td>AT4G11360</td>\n",
       "      <td>AT1G16010</td>\n",
       "      <td>AT3G50980</td>\n",
       "      <td>AT5G66564</td>\n",
       "      <td>AT1G18330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Methylobacterium</th>\n",
       "      <td>AT1G52510</td>\n",
       "      <td>AT1G79870</td>\n",
       "      <td>AT2G40830</td>\n",
       "      <td>AT4G20760</td>\n",
       "      <td>AT4G37930</td>\n",
       "      <td>AT4G39970</td>\n",
       "      <td>AT3G44880</td>\n",
       "      <td>AT1G55480</td>\n",
       "      <td>AT2G04850</td>\n",
       "      <td>AT5G58260</td>\n",
       "      <td>...</td>\n",
       "      <td>AT4G01070</td>\n",
       "      <td>AT2G44360</td>\n",
       "      <td>AT3G07880</td>\n",
       "      <td>AT3G60340</td>\n",
       "      <td>AT5G53850</td>\n",
       "      <td>AT2G14560</td>\n",
       "      <td>AT5G53330</td>\n",
       "      <td>AT5G05180</td>\n",
       "      <td>AT1G06130</td>\n",
       "      <td>AT4G32830</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Microbacterium_A_383321</th>\n",
       "      <td>AT5G44590</td>\n",
       "      <td>AT4G31730</td>\n",
       "      <td>AT5G56910</td>\n",
       "      <td>AT1G55010</td>\n",
       "      <td>AT4G33250</td>\n",
       "      <td>AT5G01870</td>\n",
       "      <td>AT1G20220</td>\n",
       "      <td>AT2G05510</td>\n",
       "      <td>AT4G19810</td>\n",
       "      <td>AT2G40590</td>\n",
       "      <td>...</td>\n",
       "      <td>AT5G13780</td>\n",
       "      <td>AT2G39990</td>\n",
       "      <td>AT5G14030</td>\n",
       "      <td>AT5G35620</td>\n",
       "      <td>AT1G14320</td>\n",
       "      <td>AT3G01790</td>\n",
       "      <td>AT4G27090</td>\n",
       "      <td>AT5G10360</td>\n",
       "      <td>AT5G67500</td>\n",
       "      <td>AT5G58420</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mycobacterium</th>\n",
       "      <td>ArthCt099</td>\n",
       "      <td>AT3G05685</td>\n",
       "      <td>AT5G10946</td>\n",
       "      <td>ArthCp049</td>\n",
       "      <td>AT3G48310</td>\n",
       "      <td>AT2G23430</td>\n",
       "      <td>ArthCp005</td>\n",
       "      <td>AT1G78290</td>\n",
       "      <td>AT3G02895</td>\n",
       "      <td>ArthCp040</td>\n",
       "      <td>...</td>\n",
       "      <td>AT2G31350</td>\n",
       "      <td>AT2G39580</td>\n",
       "      <td>AT3G02290</td>\n",
       "      <td>AT1G60900</td>\n",
       "      <td>AT3G18770</td>\n",
       "      <td>AT3G47420</td>\n",
       "      <td>ArthCp069</td>\n",
       "      <td>AT1G61380</td>\n",
       "      <td>ArthCp059</td>\n",
       "      <td>AT2G47490</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Nakamurella</th>\n",
       "      <td>AT2G24540</td>\n",
       "      <td>AT5G16600</td>\n",
       "      <td>AT2G41250</td>\n",
       "      <td>AT3G09450</td>\n",
       "      <td>AT2G19620</td>\n",
       "      <td>AT1G78230</td>\n",
       "      <td>AT3G24460</td>\n",
       "      <td>AT3G27170</td>\n",
       "      <td>ArthCt096</td>\n",
       "      <td>AT3G21670</td>\n",
       "      <td>...</td>\n",
       "      <td>AT4G04610</td>\n",
       "      <td>AT4G09350</td>\n",
       "      <td>AT4G22490</td>\n",
       "      <td>AT2G34850</td>\n",
       "      <td>AT5G17600</td>\n",
       "      <td>AT4G26960</td>\n",
       "      <td>AT1G53920</td>\n",
       "      <td>AT1G05210</td>\n",
       "      <td>AT2G44290</td>\n",
       "      <td>AT1G27340</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Neorhizobium_500197</th>\n",
       "      <td>AT3G52660</td>\n",
       "      <td>AT2G22821</td>\n",
       "      <td>AT1G61450</td>\n",
       "      <td>AT5G67260</td>\n",
       "      <td>AT4G38430</td>\n",
       "      <td>AT1G14340</td>\n",
       "      <td>AT3G06660</td>\n",
       "      <td>AT3G15536</td>\n",
       "      <td>AT1G64500</td>\n",
       "      <td>AT3G20670</td>\n",
       "      <td>...</td>\n",
       "      <td>AT1G53090</td>\n",
       "      <td>AT3G11490</td>\n",
       "      <td>AT3G13235</td>\n",
       "      <td>AT4G31875</td>\n",
       "      <td>AT4G37830</td>\n",
       "      <td>AT5G48070</td>\n",
       "      <td>AT1G17440</td>\n",
       "      <td>AT1G22900</td>\n",
       "      <td>AT2G03870</td>\n",
       "      <td>AT2G29550</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Nocardioides_A_392796</th>\n",
       "      <td>AT4G15430</td>\n",
       "      <td>AT5G03555</td>\n",
       "      <td>AT5G04815</td>\n",
       "      <td>AT1G29170</td>\n",
       "      <td>AT5G01370</td>\n",
       "      <td>AT1G77122</td>\n",
       "      <td>AT3G54390</td>\n",
       "      <td>AT3G10130</td>\n",
       "      <td>AT1G07200</td>\n",
       "      <td>AT2G26170</td>\n",
       "      <td>...</td>\n",
       "      <td>AT3G18890</td>\n",
       "      <td>AT4G26255</td>\n",
       "      <td>AT5G62480</td>\n",
       "      <td>AT1G06110</td>\n",
       "      <td>AT1G02180</td>\n",
       "      <td>AT1G04350</td>\n",
       "      <td>AT1G70470</td>\n",
       "      <td>AT2G17033</td>\n",
       "      <td>AT1G54490</td>\n",
       "      <td>AT2G45960</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Noviherbaspirillum_A_568104</th>\n",
       "      <td>AT1G05470</td>\n",
       "      <td>AT2G21300</td>\n",
       "      <td>AT4G14480</td>\n",
       "      <td>AT4G00230</td>\n",
       "      <td>AT4G32060</td>\n",
       "      <td>AT5G54200</td>\n",
       "      <td>AT4G28780</td>\n",
       "      <td>AT1G04660</td>\n",
       "      <td>AT4G25440</td>\n",
       "      <td>AT5G04815</td>\n",
       "      <td>...</td>\n",
       "      <td>AT2G26760</td>\n",
       "      <td>AT3G44610</td>\n",
       "      <td>AT3G60740</td>\n",
       "      <td>AT5G63860</td>\n",
       "      <td>AT3G02550</td>\n",
       "      <td>AT5G53130</td>\n",
       "      <td>AT1G27840</td>\n",
       "      <td>AT1G63700</td>\n",
       "      <td>AT3G53670</td>\n",
       "      <td>AT3G14850</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patulibacter</th>\n",
       "      <td>ArthCr091</td>\n",
       "      <td>ArthCp056</td>\n",
       "      <td>AT2G42865</td>\n",
       "      <td>AT3G46590</td>\n",
       "      <td>AT2G08740</td>\n",
       "      <td>AT1G16810</td>\n",
       "      <td>AT5G51130</td>\n",
       "      <td>ArthCt124</td>\n",
       "      <td>AT5G59430</td>\n",
       "      <td>ArthCp014</td>\n",
       "      <td>...</td>\n",
       "      <td>AT4G04475</td>\n",
       "      <td>AT3G24070</td>\n",
       "      <td>AT2G13665</td>\n",
       "      <td>AT2G23760</td>\n",
       "      <td>AT1G11060</td>\n",
       "      <td>AT1G07593</td>\n",
       "      <td>AT5G21940</td>\n",
       "      <td>AT1G26973</td>\n",
       "      <td>AT5G49530</td>\n",
       "      <td>ArthCp057</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Paucimonas</th>\n",
       "      <td>AT2G28130</td>\n",
       "      <td>AT5G13970</td>\n",
       "      <td>AT4G22350</td>\n",
       "      <td>AT1G80930</td>\n",
       "      <td>AT2G04795</td>\n",
       "      <td>AT5G13950</td>\n",
       "      <td>AT2G39660</td>\n",
       "      <td>AT1G65280</td>\n",
       "      <td>AT4G09980</td>\n",
       "      <td>AT1G15430</td>\n",
       "      <td>...</td>\n",
       "      <td>AT4G34680</td>\n",
       "      <td>AT1G02065</td>\n",
       "      <td>AT1G05785</td>\n",
       "      <td>AT1G06080</td>\n",
       "      <td>AT5G07270</td>\n",
       "      <td>AT1G16590</td>\n",
       "      <td>AT3G24070</td>\n",
       "      <td>AT1G67195</td>\n",
       "      <td>AT3G27520</td>\n",
       "      <td>AT5G18650</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pelomonas</th>\n",
       "      <td>AT5G43180</td>\n",
       "      <td>AT2G39560</td>\n",
       "      <td>AT1G02690</td>\n",
       "      <td>AT1G79985</td>\n",
       "      <td>AT1G28710</td>\n",
       "      <td>AT2G43750</td>\n",
       "      <td>AT4G18460</td>\n",
       "      <td>AT4G31730</td>\n",
       "      <td>AT1G35560</td>\n",
       "      <td>AT4G24660</td>\n",
       "      <td>...</td>\n",
       "      <td>AT5G38940</td>\n",
       "      <td>AT2G05220</td>\n",
       "      <td>AT5G53180</td>\n",
       "      <td>AT3G04920</td>\n",
       "      <td>AT2G24020</td>\n",
       "      <td>AT2G37660</td>\n",
       "      <td>AT4G25100</td>\n",
       "      <td>AT5G57390</td>\n",
       "      <td>AT3G11580</td>\n",
       "      <td>AT4G19970</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pigmentiphaga</th>\n",
       "      <td>AT5G24105</td>\n",
       "      <td>AT5G24150</td>\n",
       "      <td>AT1G65060</td>\n",
       "      <td>AT4G17340</td>\n",
       "      <td>AT2G20724</td>\n",
       "      <td>AT2G33855</td>\n",
       "      <td>AT1G65490</td>\n",
       "      <td>ArthCt115</td>\n",
       "      <td>AT1G02820</td>\n",
       "      <td>AT5G65870</td>\n",
       "      <td>...</td>\n",
       "      <td>AT3G44735</td>\n",
       "      <td>AT1G02705</td>\n",
       "      <td>AT1G20070</td>\n",
       "      <td>AT3G52050</td>\n",
       "      <td>AT5G07920</td>\n",
       "      <td>AT3G14395</td>\n",
       "      <td>AT5G38420</td>\n",
       "      <td>AT2G33380</td>\n",
       "      <td>AT4G19450</td>\n",
       "      <td>AT1G62660</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pseudoduganella_570666</th>\n",
       "      <td>AT2G40660</td>\n",
       "      <td>AT5G01470</td>\n",
       "      <td>AT5G45550</td>\n",
       "      <td>AT1G79720</td>\n",
       "      <td>AT3G14920</td>\n",
       "      <td>AT4G14890</td>\n",
       "      <td>AT4G29830</td>\n",
       "      <td>AT3G53850</td>\n",
       "      <td>AT5G20060</td>\n",
       "      <td>AT4G13980</td>\n",
       "      <td>...</td>\n",
       "      <td>AT4G27870</td>\n",
       "      <td>AT5G60800</td>\n",
       "      <td>AT4G14680</td>\n",
       "      <td>AT1G20950</td>\n",
       "      <td>AT1G30580</td>\n",
       "      <td>AT3G22300</td>\n",
       "      <td>AT5G14260</td>\n",
       "      <td>AT4G01590</td>\n",
       "      <td>AT1G36390</td>\n",
       "      <td>AT3G60370</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pseudoduganella_570691</th>\n",
       "      <td>AT5G20790</td>\n",
       "      <td>AT1G72610</td>\n",
       "      <td>AT1G61000</td>\n",
       "      <td>AT5G21482</td>\n",
       "      <td>AT2G29690</td>\n",
       "      <td>AT3G08600</td>\n",
       "      <td>AT3G19680</td>\n",
       "      <td>AT4G33720</td>\n",
       "      <td>AT1G18440</td>\n",
       "      <td>AT4G28190</td>\n",
       "      <td>...</td>\n",
       "      <td>AT2G29660</td>\n",
       "      <td>AT5G19855</td>\n",
       "      <td>AT1G13640</td>\n",
       "      <td>AT4G12730</td>\n",
       "      <td>AT5G43870</td>\n",
       "      <td>AT1G30290</td>\n",
       "      <td>AT3G48040</td>\n",
       "      <td>AT1G53730</td>\n",
       "      <td>AT5G13860</td>\n",
       "      <td>AT3G03270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pseudomonas_B_650453</th>\n",
       "      <td>AT5G13120</td>\n",
       "      <td>AT3G52960</td>\n",
       "      <td>AT5G55710</td>\n",
       "      <td>AT3G26060</td>\n",
       "      <td>AT4G13590</td>\n",
       "      <td>AT1G50320</td>\n",
       "      <td>AT1G60989</td>\n",
       "      <td>AT4G29480</td>\n",
       "      <td>ArthCp024</td>\n",
       "      <td>AT4G30950</td>\n",
       "      <td>...</td>\n",
       "      <td>AT5G51020</td>\n",
       "      <td>AT1G76200</td>\n",
       "      <td>AT3G56910</td>\n",
       "      <td>AT1G03190</td>\n",
       "      <td>AT3G08610</td>\n",
       "      <td>AT3G44100</td>\n",
       "      <td>AT3G55390</td>\n",
       "      <td>AT4G11190</td>\n",
       "      <td>AT4G21210</td>\n",
       "      <td>AT4G28950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pseudomonas_E_647464</th>\n",
       "      <td>AT5G44286</td>\n",
       "      <td>AT4G06810</td>\n",
       "      <td>AT3G09855</td>\n",
       "      <td>AT5G18600</td>\n",
       "      <td>AT5G53460</td>\n",
       "      <td>AT4G20690</td>\n",
       "      <td>AT1G49500</td>\n",
       "      <td>AT3G01715</td>\n",
       "      <td>Arthct120</td>\n",
       "      <td>AT1G60840</td>\n",
       "      <td>...</td>\n",
       "      <td>AT1G64640</td>\n",
       "      <td>AT4G15660</td>\n",
       "      <td>AT2G39681</td>\n",
       "      <td>AT3G08520</td>\n",
       "      <td>AT5G43720</td>\n",
       "      <td>AT5G08865</td>\n",
       "      <td>AT5G06550</td>\n",
       "      <td>AT1G49560</td>\n",
       "      <td>AT3G27865</td>\n",
       "      <td>AT5G22555</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pseudonocardia</th>\n",
       "      <td>AT1G49840</td>\n",
       "      <td>AT3G16350</td>\n",
       "      <td>AT3G15540</td>\n",
       "      <td>unassigned_gene_139</td>\n",
       "      <td>AT3G57157</td>\n",
       "      <td>AT1G64572</td>\n",
       "      <td>AT2G36800</td>\n",
       "      <td>ArthCp006</td>\n",
       "      <td>ArthCp074</td>\n",
       "      <td>AT1G31935</td>\n",
       "      <td>...</td>\n",
       "      <td>AT1G11303</td>\n",
       "      <td>AT1G50140</td>\n",
       "      <td>AT2G01590</td>\n",
       "      <td>AT3G22370</td>\n",
       "      <td>AT5G19950</td>\n",
       "      <td>AT1G47970</td>\n",
       "      <td>AT4G22780</td>\n",
       "      <td>AT5G30360</td>\n",
       "      <td>AT5G38110</td>\n",
       "      <td>AT1G53460</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ramlibacter_588642</th>\n",
       "      <td>AT3G01715</td>\n",
       "      <td>AT2G21640</td>\n",
       "      <td>AT5G02255</td>\n",
       "      <td>AT4G12500</td>\n",
       "      <td>AT5G09115</td>\n",
       "      <td>AT3G09162</td>\n",
       "      <td>AT4G06810</td>\n",
       "      <td>AT4G12490</td>\n",
       "      <td>AT1G61275</td>\n",
       "      <td>AT4G30230</td>\n",
       "      <td>...</td>\n",
       "      <td>AT3G13450</td>\n",
       "      <td>AT3G61520</td>\n",
       "      <td>AT1G64380</td>\n",
       "      <td>AT2G38210</td>\n",
       "      <td>AT5G09065</td>\n",
       "      <td>AT3G23620</td>\n",
       "      <td>AT3G06345</td>\n",
       "      <td>AT4G19610</td>\n",
       "      <td>AT5G04105</td>\n",
       "      <td>AT3G05936</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Rathayibacter</th>\n",
       "      <td>AT1G76820</td>\n",
       "      <td>AT1G31870</td>\n",
       "      <td>AT3G21805</td>\n",
       "      <td>AT1G07910</td>\n",
       "      <td>AT4G08593</td>\n",
       "      <td>AT1G74800</td>\n",
       "      <td>AT3G06345</td>\n",
       "      <td>AT1G79900</td>\n",
       "      <td>AT1G08600</td>\n",
       "      <td>AT1G04053</td>\n",
       "      <td>...</td>\n",
       "      <td>AT5G02810</td>\n",
       "      <td>AT3G16565</td>\n",
       "      <td>AT2G05915</td>\n",
       "      <td>AT2G40260</td>\n",
       "      <td>AT5G17550</td>\n",
       "      <td>AT5G45560</td>\n",
       "      <td>AT2G31280</td>\n",
       "      <td>AT3G18770</td>\n",
       "      <td>AT1G26620</td>\n",
       "      <td>AT3G19510</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Rhodoferax_C</th>\n",
       "      <td>AT3G03190</td>\n",
       "      <td>AT2G34920</td>\n",
       "      <td>AT3G62860</td>\n",
       "      <td>AT3G08445</td>\n",
       "      <td>AT5G40210</td>\n",
       "      <td>AT1G47370</td>\n",
       "      <td>AT1G62800</td>\n",
       "      <td>AT1G77590</td>\n",
       "      <td>AT5G23065</td>\n",
       "      <td>AT5G20670</td>\n",
       "      <td>...</td>\n",
       "      <td>AT3G48490</td>\n",
       "      <td>AT1G17560</td>\n",
       "      <td>AT1G20190</td>\n",
       "      <td>ArthCt099</td>\n",
       "      <td>AT1G63710</td>\n",
       "      <td>AT1G67750</td>\n",
       "      <td>ArthCp015</td>\n",
       "      <td>AT1G69523</td>\n",
       "      <td>AT1G76410</td>\n",
       "      <td>AT4G04925</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Rugamonas_570800</th>\n",
       "      <td>AT5G23040</td>\n",
       "      <td>AT3G10745</td>\n",
       "      <td>AT5G05600</td>\n",
       "      <td>AT2G05520</td>\n",
       "      <td>AT4G38470</td>\n",
       "      <td>AT5G47550</td>\n",
       "      <td>AT2G22880</td>\n",
       "      <td>AT5G64750</td>\n",
       "      <td>AT1G74370</td>\n",
       "      <td>AT1G07457</td>\n",
       "      <td>...</td>\n",
       "      <td>AT2G34140</td>\n",
       "      <td>AT4G26220</td>\n",
       "      <td>AT1G03610</td>\n",
       "      <td>AT2G24060</td>\n",
       "      <td>AT3G13180</td>\n",
       "      <td>AT5G13330</td>\n",
       "      <td>AT5G66470</td>\n",
       "      <td>AT1G04635</td>\n",
       "      <td>AT2G38980</td>\n",
       "      <td>AT3G28210</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Solirubrobacter</th>\n",
       "      <td>AT2G31810</td>\n",
       "      <td>AT5G57940</td>\n",
       "      <td>AT1G74710</td>\n",
       "      <td>AT1G26090</td>\n",
       "      <td>AT3G27930</td>\n",
       "      <td>AT3G61860</td>\n",
       "      <td>AT3G57870</td>\n",
       "      <td>AT3G45190</td>\n",
       "      <td>AT1G22200</td>\n",
       "      <td>AT5G26920</td>\n",
       "      <td>...</td>\n",
       "      <td>AT1G51720</td>\n",
       "      <td>AT2G46560</td>\n",
       "      <td>AT5G49300</td>\n",
       "      <td>AT2G29990</td>\n",
       "      <td>AT5G16260</td>\n",
       "      <td>AT5G37070</td>\n",
       "      <td>AT1G09520</td>\n",
       "      <td>AT2G30740</td>\n",
       "      <td>AT3G10250</td>\n",
       "      <td>AT4G20850</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sphingomonas_L_486704</th>\n",
       "      <td>AT2G08600</td>\n",
       "      <td>AT3G49580</td>\n",
       "      <td>AT1G19840</td>\n",
       "      <td>AT5G16530</td>\n",
       "      <td>AT4G35640</td>\n",
       "      <td>AT1G62500</td>\n",
       "      <td>AT1G80760</td>\n",
       "      <td>AT1G52342</td>\n",
       "      <td>AT5G24660</td>\n",
       "      <td>AT1G62660</td>\n",
       "      <td>...</td>\n",
       "      <td>AT3G22890</td>\n",
       "      <td>AT1G29440</td>\n",
       "      <td>AT3G52360</td>\n",
       "      <td>AT5G54130</td>\n",
       "      <td>ArthCp016</td>\n",
       "      <td>AT4G14010</td>\n",
       "      <td>AT3G05900</td>\n",
       "      <td>AT1G78865</td>\n",
       "      <td>ArthCp032</td>\n",
       "      <td>AT3G49360</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Telluria_573210</th>\n",
       "      <td>AT2G25920</td>\n",
       "      <td>AT2G43080</td>\n",
       "      <td>AT4G24880</td>\n",
       "      <td>AT3G53120</td>\n",
       "      <td>AT5G10610</td>\n",
       "      <td>AT4G26240</td>\n",
       "      <td>AT5G56750</td>\n",
       "      <td>AT2G34860</td>\n",
       "      <td>AT2G01090</td>\n",
       "      <td>AT4G38290</td>\n",
       "      <td>...</td>\n",
       "      <td>AT1G68840</td>\n",
       "      <td>AT3G18820</td>\n",
       "      <td>AT3G44320</td>\n",
       "      <td>AT3G53980</td>\n",
       "      <td>AT5G02750</td>\n",
       "      <td>AT5G43100</td>\n",
       "      <td>AT1G15230</td>\n",
       "      <td>AT1G33490</td>\n",
       "      <td>AT2G44450</td>\n",
       "      <td>AT3G48320</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Variovorax</th>\n",
       "      <td>AT5G25140</td>\n",
       "      <td>AT3G28500</td>\n",
       "      <td>AT2G20870</td>\n",
       "      <td>AT5G24150</td>\n",
       "      <td>AT5G24200</td>\n",
       "      <td>AT1G20070</td>\n",
       "      <td>AT1G47370</td>\n",
       "      <td>AT1G68725</td>\n",
       "      <td>AT4G33790</td>\n",
       "      <td>AT1G12570</td>\n",
       "      <td>...</td>\n",
       "      <td>AT5G24155</td>\n",
       "      <td>AT1G19950</td>\n",
       "      <td>AT1G53220</td>\n",
       "      <td>AT3G60540</td>\n",
       "      <td>AT4G35770</td>\n",
       "      <td>AT2G40900</td>\n",
       "      <td>AT4G11211</td>\n",
       "      <td>ArthCp077</td>\n",
       "      <td>AT2G21560</td>\n",
       "      <td>AT4G21650</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>55 rows × 300 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   0          1          2    \\\n",
       "Genus                                                          \n",
       "Actinomycetospora            AT4G28020  AT5G47610  AT1G05560   \n",
       "Actinoplanes                 AT2G38995  AT1G01900  AT1G70890   \n",
       "Aeromicrobium                AT2G04430  AT1G12440  AT2G38860   \n",
       "Aetherobacter                ArthCp002  ArthCp030  AT1G28010   \n",
       "Agrobacterium                AT5G46500  AT3G45443  AT3G12110   \n",
       "Amnibacterium_382409         AT2G22230  AT5G16390  AT1G53520   \n",
       "Aquabacterium_A_592457       AT3G46630  AT5G08415  AT2G42710   \n",
       "Aureimonas_A_501549          AT3G21670  AT2G27420  AT5G42760   \n",
       "Blastococcus                 AT2G21410  AT5G05620  AT4G15800   \n",
       "Brevundimonas                AT5G63690  AT5G05140  AT2G29440   \n",
       "Caldora                      AT5G02575  AT5G02565  AT1G77490   \n",
       "Chryseobacterium_A_796612    AT4G17520  AT3G28460  AT3G63160   \n",
       "Curtobacterium               AT1G55010  AT3G14750  AT3G20170   \n",
       "Curvibacter_587305           AT1G34245  AT5G16140  AT1G78790   \n",
       "Duganella_570715             AT5G06600  AT3G16565  AT3G61970   \n",
       "Flavobacterium               AT1G04907  AT1G29071  AT1G70700   \n",
       "Friedmanniella               AT3G06430  AT3G06050  AT4G39900   \n",
       "Frigoribacterium             AT1G61910  AT1G66660  AT3G57790   \n",
       "Geodermatophilus             AT1G49150  AT5G66100  AT3G13225   \n",
       "Herbaspirillum               AT1G61000  AT5G38980  AT3G01710   \n",
       "Hylemonella                  AT4G10760  AT3G55510  AT1G18340   \n",
       "Janthinobacterium_571130     AT1G27340  AT3G09455  AT1G08830   \n",
       "Janthinobacterium_571526     AT2G30870  AT2G43150  AT1G52910   \n",
       "Kineococcus                  AT2G46900  AT1G55350  AT1G16690   \n",
       "Kineosporia                  AT1G09180  ArthCp010  AT3G21760   \n",
       "Klenkia                      AT2G28570  AT1G70830  AT3G22750   \n",
       "Kordiimonas                  ArthCp021  AT5G04455  ArthCp035   \n",
       "Marmoricola_A_392027         AT1G03160  AT4G05190  AT1G50200   \n",
       "Massilia_573596              AT5G04085  AT3G06325  AT5G45307   \n",
       "Massilia_574424              ArthCp048  ArthCp033  ArthCp045   \n",
       "Massilia_574544              AT3G06325  ArthCp031  AT5G24570   \n",
       "Methylobacterium             AT1G52510  AT1G79870  AT2G40830   \n",
       "Microbacterium_A_383321      AT5G44590  AT4G31730  AT5G56910   \n",
       "Mycobacterium                ArthCt099  AT3G05685  AT5G10946   \n",
       "Nakamurella                  AT2G24540  AT5G16600  AT2G41250   \n",
       "Neorhizobium_500197          AT3G52660  AT2G22821  AT1G61450   \n",
       "Nocardioides_A_392796        AT4G15430  AT5G03555  AT5G04815   \n",
       "Noviherbaspirillum_A_568104  AT1G05470  AT2G21300  AT4G14480   \n",
       "Patulibacter                 ArthCr091  ArthCp056  AT2G42865   \n",
       "Paucimonas                   AT2G28130  AT5G13970  AT4G22350   \n",
       "Pelomonas                    AT5G43180  AT2G39560  AT1G02690   \n",
       "Pigmentiphaga                AT5G24105  AT5G24150  AT1G65060   \n",
       "Pseudoduganella_570666       AT2G40660  AT5G01470  AT5G45550   \n",
       "Pseudoduganella_570691       AT5G20790  AT1G72610  AT1G61000   \n",
       "Pseudomonas_B_650453         AT5G13120  AT3G52960  AT5G55710   \n",
       "Pseudomonas_E_647464         AT5G44286  AT4G06810  AT3G09855   \n",
       "Pseudonocardia               AT1G49840  AT3G16350  AT3G15540   \n",
       "Ramlibacter_588642           AT3G01715  AT2G21640  AT5G02255   \n",
       "Rathayibacter                AT1G76820  AT1G31870  AT3G21805   \n",
       "Rhodoferax_C                 AT3G03190  AT2G34920  AT3G62860   \n",
       "Rugamonas_570800             AT5G23040  AT3G10745  AT5G05600   \n",
       "Solirubrobacter              AT2G31810  AT5G57940  AT1G74710   \n",
       "Sphingomonas_L_486704        AT2G08600  AT3G49580  AT1G19840   \n",
       "Telluria_573210              AT2G25920  AT2G43080  AT4G24880   \n",
       "Variovorax                   AT5G25140  AT3G28500  AT2G20870   \n",
       "\n",
       "                                             3          4          5    \\\n",
       "Genus                                                                    \n",
       "Actinomycetospora                      AT2G37050  AT1G48620  ArthCt091   \n",
       "Actinoplanes                           AT5G17700  AT1G69523  AT3G62750   \n",
       "Aeromicrobium                          AT1G35260  AT1G08213  AT3G09150   \n",
       "Aetherobacter                          AT5G24150  ArthCp007  ArthCp053   \n",
       "Agrobacterium                          AT1G02960  AT2G32250  AT1G16820   \n",
       "Amnibacterium_382409                   AT5G52920  AT1G73620  AT5G19370   \n",
       "Aquabacterium_A_592457                 AT4G29440  AT2G43290  AT2G24970   \n",
       "Aureimonas_A_501549                    AT3G24460  AT1G64780  AT3G27170   \n",
       "Blastococcus                           AT3G50800  AT5G17520  AT3G46550   \n",
       "Brevundimonas                          AT5G62430  AT2G31840  AT5G58710   \n",
       "Caldora                                AT3G27200  AT4G07835  AT1G58070   \n",
       "Chryseobacterium_A_796612              AT3G61110  AT3G59540  AT4G00100   \n",
       "Curtobacterium                         AT5G01870  AT1G80020  AT1G60989   \n",
       "Curvibacter_587305                     AT2G05440  AT3G24110  AT5G53080   \n",
       "Duganella_570715                       AT5G62900  AT1G08830  AT5G38510   \n",
       "Flavobacterium                         AT5G06845  AT2G01480  AT2G05655   \n",
       "Friedmanniella                         AT1G67700  AT2G41770  AT5G19770   \n",
       "Frigoribacterium                       AT2G42650  AT1G29750  AT1G80550   \n",
       "Geodermatophilus                       AT1G67890  AT1G02370  AT2G01190   \n",
       "Herbaspirillum                         AT5G03555  AT5G25830  AT5G33280   \n",
       "Hylemonella                            AT5G60730  AT2G30600  AT5G60990   \n",
       "Janthinobacterium_571130               AT2G28190  AT1G13440  AT5G12860   \n",
       "Janthinobacterium_571526               AT1G08830  AT3G12500  AT3G01480   \n",
       "Kineococcus                            AT5G07180  AT3G55390  AT1G73400   \n",
       "Kineosporia                            AT2G27420  AT1G52910  AT4G28290   \n",
       "Klenkia                                AT5G54250  AT4G24540  AT5G56890   \n",
       "Kordiimonas                            ArthCp060  ArthCp015  ArthCp062   \n",
       "Marmoricola_A_392027                   AT2G37080  AT5G26670  AT3G12700   \n",
       "Massilia_573596                        AT2G04965  ArthCp031  AT5G15780   \n",
       "Massilia_574424                        ArthCp080  ArthCp076  ArthCp077   \n",
       "Massilia_574544                        AT1G04193  AT5G57380  AT4G04595   \n",
       "Methylobacterium                       AT4G20760  AT4G37930  AT4G39970   \n",
       "Microbacterium_A_383321                AT1G55010  AT4G33250  AT5G01870   \n",
       "Mycobacterium                          ArthCp049  AT3G48310  AT2G23430   \n",
       "Nakamurella                            AT3G09450  AT2G19620  AT1G78230   \n",
       "Neorhizobium_500197                    AT5G67260  AT4G38430  AT1G14340   \n",
       "Nocardioides_A_392796                  AT1G29170  AT5G01370  AT1G77122   \n",
       "Noviherbaspirillum_A_568104            AT4G00230  AT4G32060  AT5G54200   \n",
       "Patulibacter                           AT3G46590  AT2G08740  AT1G16810   \n",
       "Paucimonas                             AT1G80930  AT2G04795  AT5G13950   \n",
       "Pelomonas                              AT1G79985  AT1G28710  AT2G43750   \n",
       "Pigmentiphaga                          AT4G17340  AT2G20724  AT2G33855   \n",
       "Pseudoduganella_570666                 AT1G79720  AT3G14920  AT4G14890   \n",
       "Pseudoduganella_570691                 AT5G21482  AT2G29690  AT3G08600   \n",
       "Pseudomonas_B_650453                   AT3G26060  AT4G13590  AT1G50320   \n",
       "Pseudomonas_E_647464                   AT5G18600  AT5G53460  AT4G20690   \n",
       "Pseudonocardia               unassigned_gene_139  AT3G57157  AT1G64572   \n",
       "Ramlibacter_588642                     AT4G12500  AT5G09115  AT3G09162   \n",
       "Rathayibacter                          AT1G07910  AT4G08593  AT1G74800   \n",
       "Rhodoferax_C                           AT3G08445  AT5G40210  AT1G47370   \n",
       "Rugamonas_570800                       AT2G05520  AT4G38470  AT5G47550   \n",
       "Solirubrobacter                        AT1G26090  AT3G27930  AT3G61860   \n",
       "Sphingomonas_L_486704                  AT5G16530  AT4G35640  AT1G62500   \n",
       "Telluria_573210                        AT3G53120  AT5G10610  AT4G26240   \n",
       "Variovorax                             AT5G24150  AT5G24200  AT1G20070   \n",
       "\n",
       "                                   6          7          8          9    ...  \\\n",
       "Genus                                                                    ...   \n",
       "Actinomycetospora            AT5G03415  AT4G12970  AT3G54220  AT5G02010  ...   \n",
       "Actinoplanes                 AT3G48185  AT4G22635  AT5G01495  AT4G33470  ...   \n",
       "Aeromicrobium                AT3G02060  AT1G13145  AT2G37130  AT3G11591  ...   \n",
       "Aetherobacter                AT3G28500  ArthCp017  ArthCp015  ArthCp042  ...   \n",
       "Agrobacterium                AT2G10606  AT5G48340  AT2G40670  AT4G03845  ...   \n",
       "Amnibacterium_382409         AT1G33170  AT4G35335  AT3G18080  AT3G51820  ...   \n",
       "Aquabacterium_A_592457       AT3G13445  AT1G52270  AT5G03370  AT2G39900  ...   \n",
       "Aureimonas_A_501549          AT4G27360  AT1G19490  AT3G51375  AT3G47430  ...   \n",
       "Blastococcus                 AT1G29890  AT3G09035  AT3G06550  AT5G22050  ...   \n",
       "Brevundimonas                AT5G48470  AT4G07705  AT4G08390  AT5G09270  ...   \n",
       "Caldora                      AT4G33960  AT5G35170  AT1G69320  AT3G49110  ...   \n",
       "Chryseobacterium_A_796612    AT4G24770  AT1G56070  AT1G31500  AT1G74970  ...   \n",
       "Curtobacterium               AT1G71310  AT4G13235  AT2G14580  AT4G22505  ...   \n",
       "Curvibacter_587305           AT2G47960  AT5G56910  AT1G79850  AT3G60530  ...   \n",
       "Duganella_570715             AT5G55610  AT3G05990  AT4G39730  AT5G27520  ...   \n",
       "Flavobacterium               AT1G49700  AT3G16410  AT1G80540  AT3G27660  ...   \n",
       "Friedmanniella               AT3G03070  AT2G37860  AT1G56340  AT1G01090  ...   \n",
       "Frigoribacterium             AT3G26850  AT3G08990  AT2G41990  AT2G35340  ...   \n",
       "Geodermatophilus             AT5G10600  AT1G78850  AT3G01430  AT1G15130  ...   \n",
       "Herbaspirillum               AT4G15430  AT2G28620  AT3G02380  AT3G09050  ...   \n",
       "Hylemonella                  AT3G27550  AT5G38890  AT3G49890  AT3G10270  ...   \n",
       "Janthinobacterium_571130     AT4G13940  AT5G51010  AT3G52930  AT1G32930  ...   \n",
       "Janthinobacterium_571526     AT5G07440  AT5G19250  AT1G75690  AT5G66040  ...   \n",
       "Kineococcus                  AT4G38980  AT5G11230  AT5G05960  AT4G22505  ...   \n",
       "Kineosporia                  AT1G07500  AT1G52342  AT5G02940  AT1G65870  ...   \n",
       "Klenkia                      AT3G22942  AT1G74800  AT1G04295  AT2G47920  ...   \n",
       "Kordiimonas                  AT5G24150  ArthCp039  ArthCp048  AT5G08155  ...   \n",
       "Marmoricola_A_392027         AT1G12250  AT1G78915  AT3G60900  AT3G01390  ...   \n",
       "Massilia_573596              AT1G49500  AT5G04305  AT2G03020  AT3G57645  ...   \n",
       "Massilia_574424              ArthCp004  ArthCp034  ArthCp023  ArthCp079  ...   \n",
       "Massilia_574544              AT5G14780  AT4G06375  AT1G49500  AT5G39110  ...   \n",
       "Methylobacterium             AT3G44880  AT1G55480  AT2G04850  AT5G58260  ...   \n",
       "Microbacterium_A_383321      AT1G20220  AT2G05510  AT4G19810  AT2G40590  ...   \n",
       "Mycobacterium                ArthCp005  AT1G78290  AT3G02895  ArthCp040  ...   \n",
       "Nakamurella                  AT3G24460  AT3G27170  ArthCt096  AT3G21670  ...   \n",
       "Neorhizobium_500197          AT3G06660  AT3G15536  AT1G64500  AT3G20670  ...   \n",
       "Nocardioides_A_392796        AT3G54390  AT3G10130  AT1G07200  AT2G26170  ...   \n",
       "Noviherbaspirillum_A_568104  AT4G28780  AT1G04660  AT4G25440  AT5G04815  ...   \n",
       "Patulibacter                 AT5G51130  ArthCt124  AT5G59430  ArthCp014  ...   \n",
       "Paucimonas                   AT2G39660  AT1G65280  AT4G09980  AT1G15430  ...   \n",
       "Pelomonas                    AT4G18460  AT4G31730  AT1G35560  AT4G24660  ...   \n",
       "Pigmentiphaga                AT1G65490  ArthCt115  AT1G02820  AT5G65870  ...   \n",
       "Pseudoduganella_570666       AT4G29830  AT3G53850  AT5G20060  AT4G13980  ...   \n",
       "Pseudoduganella_570691       AT3G19680  AT4G33720  AT1G18440  AT4G28190  ...   \n",
       "Pseudomonas_B_650453         AT1G60989  AT4G29480  ArthCp024  AT4G30950  ...   \n",
       "Pseudomonas_E_647464         AT1G49500  AT3G01715  Arthct120  AT1G60840  ...   \n",
       "Pseudonocardia               AT2G36800  ArthCp006  ArthCp074  AT1G31935  ...   \n",
       "Ramlibacter_588642           AT4G06810  AT4G12490  AT1G61275  AT4G30230  ...   \n",
       "Rathayibacter                AT3G06345  AT1G79900  AT1G08600  AT1G04053  ...   \n",
       "Rhodoferax_C                 AT1G62800  AT1G77590  AT5G23065  AT5G20670  ...   \n",
       "Rugamonas_570800             AT2G22880  AT5G64750  AT1G74370  AT1G07457  ...   \n",
       "Solirubrobacter              AT3G57870  AT3G45190  AT1G22200  AT5G26920  ...   \n",
       "Sphingomonas_L_486704        AT1G80760  AT1G52342  AT5G24660  AT1G62660  ...   \n",
       "Telluria_573210              AT5G56750  AT2G34860  AT2G01090  AT4G38290  ...   \n",
       "Variovorax                   AT1G47370  AT1G68725  AT4G33790  AT1G12570  ...   \n",
       "\n",
       "                                   290        291        292        293  \\\n",
       "Genus                                                                     \n",
       "Actinomycetospora            AT5G58150  AT2G18330  AT2G21290  AT2G37630   \n",
       "Actinoplanes                 AT4G33260  AT5G08640  AT5G02060  AT1G08570   \n",
       "Aeromicrobium                AT4G29140  AT1G76120  AT5G47930  AT1G31120   \n",
       "Aetherobacter                AT2G34920  AT3G23480  AT1G23550  AT4G38850   \n",
       "Agrobacterium                AT5G03530  AT5G50340  AT2G47240  AT4G40042   \n",
       "Amnibacterium_382409         AT2G45270  AT5G63310  AT3G14220  AT5G05960   \n",
       "Aquabacterium_A_592457       AT3G23700  AT2G40085  AT1G66070  AT4G00880   \n",
       "Aureimonas_A_501549          AT5G24850  AT3G04140  AT5G20070  AT3G57020   \n",
       "Blastococcus                 AT5G46620  AT2G30410  AT1G28570  AT3G47833   \n",
       "Brevundimonas                AT2G39250  AT3G09890  AT1G19430  AT2G21280   \n",
       "Caldora                      AT1G73470  AT1G64660  AT3G44070  AT5G52570   \n",
       "Chryseobacterium_A_796612    AT3G15140  AT4G26670  AT1G69620  AT5G15960   \n",
       "Curtobacterium               AT4G20020  AT5G18310  AT5G27720  AT2G33220   \n",
       "Curvibacter_587305           AT2G02540  AT5G08260  AT5G58780  AT3G48820   \n",
       "Duganella_570715             AT4G27830  AT1G69640  AT4G39680  AT3G22190   \n",
       "Flavobacterium               AT3G01970  AT1G35260  AT5G61340  ArthCr091   \n",
       "Friedmanniella               AT3G63190  AT5G14320  AT5G42190  AT2G42130   \n",
       "Frigoribacterium             AT2G41590  AT5G64030  AT1G30835  AT3G50370   \n",
       "Geodermatophilus             AT5G57480  AT5G59140  AT5G01270  AT3G19930   \n",
       "Herbaspirillum               AT1G77122  AT2G27920  AT1G19840  AT1G62500   \n",
       "Hylemonella                  AT2G36885  AT3G07300  AT3G14800  AT4G18905   \n",
       "Janthinobacterium_571130     AT2G44525  AT4G35830  AT4G35630  AT5G26330   \n",
       "Janthinobacterium_571526     AT3G01920  AT4G21105  AT3G14310  AT3G52730   \n",
       "Kineococcus                  AT4G31200  AT1G79010  AT5G49510  AT1G16610   \n",
       "Kineosporia                  AT5G58990  AT2G33250  AT2G21370  AT2G26550   \n",
       "Klenkia                      AT3G57410  AT1G65890  AT2G38980  AT4G38640   \n",
       "Kordiimonas                  AT5G51130  AT4G27350  AT4G32340  AT2G33390   \n",
       "Marmoricola_A_392027         AT2G21410  AT3G09300  AT1G50430  AT3G27360   \n",
       "Massilia_573596              AT5G47330  AT4G38580  AT5G42130  AT5G42940   \n",
       "Massilia_574424              AT2G04066  AT1G08900  AT4G03995  AT1G04217   \n",
       "Massilia_574544              AT5G47330  AT3G02445  AT1G69890  AT3G58540   \n",
       "Methylobacterium             AT4G01070  AT2G44360  AT3G07880  AT3G60340   \n",
       "Microbacterium_A_383321      AT5G13780  AT2G39990  AT5G14030  AT5G35620   \n",
       "Mycobacterium                AT2G31350  AT2G39580  AT3G02290  AT1G60900   \n",
       "Nakamurella                  AT4G04610  AT4G09350  AT4G22490  AT2G34850   \n",
       "Neorhizobium_500197          AT1G53090  AT3G11490  AT3G13235  AT4G31875   \n",
       "Nocardioides_A_392796        AT3G18890  AT4G26255  AT5G62480  AT1G06110   \n",
       "Noviherbaspirillum_A_568104  AT2G26760  AT3G44610  AT3G60740  AT5G63860   \n",
       "Patulibacter                 AT4G04475  AT3G24070  AT2G13665  AT2G23760   \n",
       "Paucimonas                   AT4G34680  AT1G02065  AT1G05785  AT1G06080   \n",
       "Pelomonas                    AT5G38940  AT2G05220  AT5G53180  AT3G04920   \n",
       "Pigmentiphaga                AT3G44735  AT1G02705  AT1G20070  AT3G52050   \n",
       "Pseudoduganella_570666       AT4G27870  AT5G60800  AT4G14680  AT1G20950   \n",
       "Pseudoduganella_570691       AT2G29660  AT5G19855  AT1G13640  AT4G12730   \n",
       "Pseudomonas_B_650453         AT5G51020  AT1G76200  AT3G56910  AT1G03190   \n",
       "Pseudomonas_E_647464         AT1G64640  AT4G15660  AT2G39681  AT3G08520   \n",
       "Pseudonocardia               AT1G11303  AT1G50140  AT2G01590  AT3G22370   \n",
       "Ramlibacter_588642           AT3G13450  AT3G61520  AT1G64380  AT2G38210   \n",
       "Rathayibacter                AT5G02810  AT3G16565  AT2G05915  AT2G40260   \n",
       "Rhodoferax_C                 AT3G48490  AT1G17560  AT1G20190  ArthCt099   \n",
       "Rugamonas_570800             AT2G34140  AT4G26220  AT1G03610  AT2G24060   \n",
       "Solirubrobacter              AT1G51720  AT2G46560  AT5G49300  AT2G29990   \n",
       "Sphingomonas_L_486704        AT3G22890  AT1G29440  AT3G52360  AT5G54130   \n",
       "Telluria_573210              AT1G68840  AT3G18820  AT3G44320  AT3G53980   \n",
       "Variovorax                   AT5G24155  AT1G19950  AT1G53220  AT3G60540   \n",
       "\n",
       "                                   294        295        296        297  \\\n",
       "Genus                                                                     \n",
       "Actinomycetospora            AT4G24940  AT5G09250  AT5G65770  AT5G51170   \n",
       "Actinoplanes                 AT1G44575  AT1G57770  AT1G10500  AT4G33790   \n",
       "Aeromicrobium                AT1G70850  AT2G39080  AT3G49100  AT5G44785   \n",
       "Aetherobacter                AT3G06080  AT5G52550  AT4G17970  AT5G12230   \n",
       "Agrobacterium                AT5G46150  AT3G29760  AT5G45250  AT1G05997   \n",
       "Amnibacterium_382409         AT5G44520  AT2G24570  AT2G43030  AT5G64050   \n",
       "Aquabacterium_A_592457       AT5G42740  AT1G12090  AT1G71850  AT4G09500   \n",
       "Aureimonas_A_501549          AT2G30040  AT5G17600  AT3G52110  AT1G70580   \n",
       "Blastococcus                 AT5G05980  AT5G59160  AT1G08040  AT4G32530   \n",
       "Brevundimonas                AT3G21870  AT1G03860  AT4G18170  AT1G56460   \n",
       "Caldora                      AT1G55152  AT5G62560  AT1G79790  AT5G51260   \n",
       "Chryseobacterium_A_796612    AT1G29900  AT1G35680  AT4G10450  AT5G66060   \n",
       "Curtobacterium               AT3G58030  AT3G58600  AT4G16440  AT4G27070   \n",
       "Curvibacter_587305           AT5G11230  AT3G15530  AT2G33330  AT4G01985   \n",
       "Duganella_570715             AT1G53035  AT3G62980  AT4G10810  AT5G44250   \n",
       "Flavobacterium               AT2G42570  AT4G08125  AT3G13525  AT5G44130   \n",
       "Friedmanniella               AT3G46940  AT4G03120  AT5G16050  AT5G62790   \n",
       "Frigoribacterium             AT1G16670  AT1G32190  AT5G17930  AT3G63052   \n",
       "Geodermatophilus             AT3G15351  AT1G18390  AT2G35510  AT5G10940   \n",
       "Herbaspirillum               AT2G43750  AT3G10080  AT1G15980  AT2G32530   \n",
       "Hylemonella                  AT5G42800  AT2G24500  AT3G57490  AT1G29880   \n",
       "Janthinobacterium_571130     AT1G80490  AT1G20190  AT1G64850  AT1G79600   \n",
       "Janthinobacterium_571526     AT3G57750  AT5G16710  AT4G01060  AT1G60610   \n",
       "Kineococcus                  AT1G18450  AT1G29900  AT4G11160  AT1G15730   \n",
       "Kineosporia                  AT1G77122  AT2G03980  AT2G45360  AT3G61990   \n",
       "Klenkia                      AT1G15760  AT3G46590  AT5G53440  AT2G33360   \n",
       "Kordiimonas                  ArthCr091  AT1G10960  AT5G22580  AT2G31141   \n",
       "Marmoricola_A_392027         AT3G29180  AT4G01610  AT1G09340  AT3G54180   \n",
       "Massilia_573596              AT5G66567  AT3G44860  AT5G14230  AT1G09350   \n",
       "Massilia_574424              AT3G56825  AT5G54250  AT1G19450  AT2G15080   \n",
       "Massilia_574544              AT2G01300  AT4G11360  AT1G16010  AT3G50980   \n",
       "Methylobacterium             AT5G53850  AT2G14560  AT5G53330  AT5G05180   \n",
       "Microbacterium_A_383321      AT1G14320  AT3G01790  AT4G27090  AT5G10360   \n",
       "Mycobacterium                AT3G18770  AT3G47420  ArthCp069  AT1G61380   \n",
       "Nakamurella                  AT5G17600  AT4G26960  AT1G53920  AT1G05210   \n",
       "Neorhizobium_500197          AT4G37830  AT5G48070  AT1G17440  AT1G22900   \n",
       "Nocardioides_A_392796        AT1G02180  AT1G04350  AT1G70470  AT2G17033   \n",
       "Noviherbaspirillum_A_568104  AT3G02550  AT5G53130  AT1G27840  AT1G63700   \n",
       "Patulibacter                 AT1G11060  AT1G07593  AT5G21940  AT1G26973   \n",
       "Paucimonas                   AT5G07270  AT1G16590  AT3G24070  AT1G67195   \n",
       "Pelomonas                    AT2G24020  AT2G37660  AT4G25100  AT5G57390   \n",
       "Pigmentiphaga                AT5G07920  AT3G14395  AT5G38420  AT2G33380   \n",
       "Pseudoduganella_570666       AT1G30580  AT3G22300  AT5G14260  AT4G01590   \n",
       "Pseudoduganella_570691       AT5G43870  AT1G30290  AT3G48040  AT1G53730   \n",
       "Pseudomonas_B_650453         AT3G08610  AT3G44100  AT3G55390  AT4G11190   \n",
       "Pseudomonas_E_647464         AT5G43720  AT5G08865  AT5G06550  AT1G49560   \n",
       "Pseudonocardia               AT5G19950  AT1G47970  AT4G22780  AT5G30360   \n",
       "Ramlibacter_588642           AT5G09065  AT3G23620  AT3G06345  AT4G19610   \n",
       "Rathayibacter                AT5G17550  AT5G45560  AT2G31280  AT3G18770   \n",
       "Rhodoferax_C                 AT1G63710  AT1G67750  ArthCp015  AT1G69523   \n",
       "Rugamonas_570800             AT3G13180  AT5G13330  AT5G66470  AT1G04635   \n",
       "Solirubrobacter              AT5G16260  AT5G37070  AT1G09520  AT2G30740   \n",
       "Sphingomonas_L_486704        ArthCp016  AT4G14010  AT3G05900  AT1G78865   \n",
       "Telluria_573210              AT5G02750  AT5G43100  AT1G15230  AT1G33490   \n",
       "Variovorax                   AT4G35770  AT2G40900  AT4G11211  ArthCp077   \n",
       "\n",
       "                                   298        299  \n",
       "Genus                                              \n",
       "Actinomycetospora            AT3G53235  AT5G55530  \n",
       "Actinoplanes                 AT1G47240  AT4G24990  \n",
       "Aeromicrobium                AT1G26460  AT3G07230  \n",
       "Aetherobacter                AT5G40380  AT2G15340  \n",
       "Agrobacterium                AT5G14100  AT2G20190  \n",
       "Amnibacterium_382409         AT5G66530  AT1G21050  \n",
       "Aquabacterium_A_592457       AT3G20320  AT1G70480  \n",
       "Aureimonas_A_501549          AT3G19710  AT5G07460  \n",
       "Blastococcus                 AT2G42500  AT2G46910  \n",
       "Brevundimonas                AT5G23010  AT2G29310  \n",
       "Caldora                      AT1G49240  AT1G02405  \n",
       "Chryseobacterium_A_796612    AT3G53020  AT4G13850  \n",
       "Curtobacterium               AT1G70190  AT3G22330  \n",
       "Curvibacter_587305           AT4G32120  AT1G09900  \n",
       "Duganella_570715             AT5G10730  AT5G45300  \n",
       "Flavobacterium               AT2G15340  ArthCp019  \n",
       "Friedmanniella               AT1G80080  AT2G42690  \n",
       "Frigoribacterium             AT4G27730  AT1G10890  \n",
       "Geodermatophilus             AT5G43560  AT5G65180  \n",
       "Herbaspirillum               AT1G23090  AT5G51970  \n",
       "Hylemonella                  AT2G41790  AT5G06000  \n",
       "Janthinobacterium_571130     AT3G12640  AT4G01840  \n",
       "Janthinobacterium_571526     AT3G57070  AT2G18328  \n",
       "Kineococcus                  AT5G56370  AT3G42150  \n",
       "Kineosporia                  AT3G15356  AT3G50800  \n",
       "Klenkia                      AT5G19650  AT2G34420  \n",
       "Kordiimonas                  AT4G28025  AT5G44260  \n",
       "Marmoricola_A_392027         AT5G04530  AT5G47750  \n",
       "Massilia_573596              AT1G23150  AT1G28050  \n",
       "Massilia_574424              AT5G09830  AT4G19370  \n",
       "Massilia_574544              AT5G66564  AT1G18330  \n",
       "Methylobacterium             AT1G06130  AT4G32830  \n",
       "Microbacterium_A_383321      AT5G67500  AT5G58420  \n",
       "Mycobacterium                ArthCp059  AT2G47490  \n",
       "Nakamurella                  AT2G44290  AT1G27340  \n",
       "Neorhizobium_500197          AT2G03870  AT2G29550  \n",
       "Nocardioides_A_392796        AT1G54490  AT2G45960  \n",
       "Noviherbaspirillum_A_568104  AT3G53670  AT3G14850  \n",
       "Patulibacter                 AT5G49530  ArthCp057  \n",
       "Paucimonas                   AT3G27520  AT5G18650  \n",
       "Pelomonas                    AT3G11580  AT4G19970  \n",
       "Pigmentiphaga                AT4G19450  AT1G62660  \n",
       "Pseudoduganella_570666       AT1G36390  AT3G60370  \n",
       "Pseudoduganella_570691       AT5G13860  AT3G03270  \n",
       "Pseudomonas_B_650453         AT4G21210  AT4G28950  \n",
       "Pseudomonas_E_647464         AT3G27865  AT5G22555  \n",
       "Pseudonocardia               AT5G38110  AT1G53460  \n",
       "Ramlibacter_588642           AT5G04105  AT3G05936  \n",
       "Rathayibacter                AT1G26620  AT3G19510  \n",
       "Rhodoferax_C                 AT1G76410  AT4G04925  \n",
       "Rugamonas_570800             AT2G38980  AT3G28210  \n",
       "Solirubrobacter              AT3G10250  AT4G20850  \n",
       "Sphingomonas_L_486704        ArthCp032  AT3G49360  \n",
       "Telluria_573210              AT2G44450  AT3G48320  \n",
       "Variovorax                   AT2G21560  AT4G21650  \n",
       "\n",
       "[55 rows x 300 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_top200 = pd.DataFrame(index = full_spearman_correlation_matrix.index, data = top_200_spearman)\n",
    "df_top200"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "88e0a7ac",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0     AT4G17520\n",
       "1     AT3G28460\n",
       "2     AT3G63160\n",
       "3     AT3G61110\n",
       "4     AT3G59540\n",
       "5     AT4G00100\n",
       "6     AT4G24770\n",
       "7     AT1G56070\n",
       "8     AT1G31500\n",
       "9     AT1G74970\n",
       "10    AT5G23310\n",
       "11    AT5G60390\n",
       "12    AT5G05000\n",
       "13    AT4G18520\n",
       "14    AT3G45060\n",
       "15    AT5G54600\n",
       "16    AT5G46160\n",
       "17    AT1G08005\n",
       "18    AT1G21750\n",
       "19    AT2G27130\n",
       "20    AT5G64150\n",
       "21    AT3G08580\n",
       "22    AT2G47840\n",
       "23    AT2G37220\n",
       "24    AT3G05590\n",
       "25    AT5G64050\n",
       "26    AT2G05510\n",
       "27    AT2G40085\n",
       "28    AT2G05440\n",
       "29    AT5G48760\n",
       "30    AT4G00810\n",
       "31    AT3G53870\n",
       "32    AT4G38100\n",
       "33    AT3G05870\n",
       "34    AT5G09510\n",
       "35    AT1G77510\n",
       "36    AT4G05090\n",
       "37    AT1G80480\n",
       "38    AT5G66470\n",
       "39    AT4G34200\n",
       "Name: Chryseobacterium_A_796612, dtype: object"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_top200.loc['Chryseobacterium_A_796612'].head(40)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dceafd26",
   "metadata": {},
   "outputs": [],
   "source": [
    "#df_top200.to_csv('/Users/michael/Data/Luke_terrace_experiment/Output_for_Luke/top_200_spearman_correlated_genes_per_bacteria.csv',)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "id": "18737161",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['AT4G28020', 'AT5G47610', 'AT1G05560', 'AT2G37050', 'AT1G48620',\n",
       "       'ArthCt091', 'AT5G03415', 'AT4G12970', 'AT3G54220', 'AT5G02010',\n",
       "       'AT5G60190', 'AT5G53900', 'AT2G33310', 'AT5G56460', 'AT5G53440',\n",
       "       'AT5G57130', 'AT1G17510', 'AT1G77480', 'AT1G04210', 'AT1G05720',\n",
       "       'AT4G15840', 'AT1G60200', 'AT5G23580', 'AT2G19080', 'AT4G39235',\n",
       "       'AT5G01370', 'AT4G31160', 'AT1G70180', 'AT1G13640', 'AT5G04290',\n",
       "       'AT5G14460', 'AT2G23520', 'AT2G30140', 'AT1G54490', 'AT3G11580',\n",
       "       'AT1G42550', 'AT3G54190', 'AT5G43180', 'AT4G37020', 'ArthCp006',\n",
       "       'AT4G24660', 'AT3G14160', 'AT3G10130', 'AT5G46840', 'AT1G32240',\n",
       "       'AT2G34070', 'AT3G04600', 'AT5G12900', 'AT3G48800', 'AT3G20930',\n",
       "       'AT1G64450', 'AT1G47970', 'AT3G21690', 'AT3G48150', 'AT5G57390',\n",
       "       'AT1G53050', 'AT3G10050', 'AT5G53800', 'AT2G42080', 'AT1G31935',\n",
       "       'AT1G68720', 'AT1G32370', 'AT2G30170', 'AT1G06110', 'AT4G21705',\n",
       "       'AT2G08650', 'AT5G47690', 'AT4G00720', 'AT2G47350', 'AT2G37680',\n",
       "       'AT2G20400', 'AT3G15970', 'AT2G26170', 'AT3G47120', 'AT4G14010',\n",
       "       'AT5G58530', 'AT4G00110', 'AT1G60690', 'AT5G05550', 'AT2G46250',\n",
       "       'AT4G37590', 'AT1G20890', 'AT5G44500', 'AT4G19600', 'AT4G31730',\n",
       "       'AT3G54720', 'AT5G55280', 'AT1G71350', 'AT3G18240', 'AT2G41945',\n",
       "       'AT3G59640', 'AT2G34780', 'AT1G76880', 'AT3G18230', 'AT2G19270',\n",
       "       'AT3G10210', 'AT3G03610', 'AT2G40980', 'AT1G27510', 'AT2G26570',\n",
       "       'AT1G15230', 'AT1G14687', 'AT5G50230', 'AT1G62790', 'AT3G58470',\n",
       "       'AT5G58930', 'AT2G34730', 'AT2G32020', 'AT1G30160', 'AT3G07510',\n",
       "       'AT3G59990', 'AT4G21660', 'AT3G06340', 'AT3G11220', 'AT3G01142',\n",
       "       'AT2G30620', 'AT3G52660', 'AT4G28640', 'AT1G54210', 'AT2G45770',\n",
       "       'AT2G26280', 'AT1G15660', 'AT3G21290', 'AT4G17170', 'AT5G15920',\n",
       "       'AT5G59460', 'AT5G49880', 'AT5G43450', 'AT3G13360', 'AT3G52525',\n",
       "       'AT5G37710', 'AT2G01290', 'AT5G24490', 'AT3G13300', 'AT1G18440',\n",
       "       'AT4G26630', 'AT2G43750', 'AT5G38980', 'AT2G18250', 'AT3G56580',\n",
       "       'AT4G16310', 'AT3G53920', 'AT5G55660', 'AT5G53210', 'AT3G50690',\n",
       "       'AT5G20520', 'AT3G52280', 'AT3G52090', 'AT3G06960', 'AT5G58570',\n",
       "       'AT4G24810', 'AT2G36630', 'AT1G19850', 'AT1G47270', 'AT4G25680',\n",
       "       'AT2G47000', 'AT3G57830', 'AT4G15950', 'AT2G47640', 'AT1G80790',\n",
       "       'AT4G29940', 'AT3G01770', 'AT3G06480', 'AT3G61880', 'AT1G10320',\n",
       "       'AT2G46450', 'AT2G43770', 'AT1G16740', 'AT5G66820', 'AT4G00180',\n",
       "       'AT1G20575', 'AT4G31200', 'AT3G20910', 'AT5G60030', 'AT2G35900',\n",
       "       'AT1G14890', 'AT1G24290', 'AT3G08720', 'AT1G05380', 'AT3G25855',\n",
       "       'AT5G01445', 'AT1G22610', 'AT5G53890', 'AT5G24690', 'AT2G44300',\n",
       "       'AT3G02400', 'AT4G28100', 'AT5G13890', 'AT5G15802', 'AT5G16440',\n",
       "       'AT3G26690', 'AT3G02640', 'AT3G18390', 'AT5G53486', 'AT5G03555',\n",
       "       'AT5G12230', 'AT5G23430', 'AT2G32415', 'AT3G04485', 'AT4G37550',\n",
       "       'AT1G32490', 'AT5G19030', 'AT1G71460', 'AT4G12735', 'AT4G34135',\n",
       "       'AT2G27230', 'AT5G63550', 'AT4G02730', 'AT1G07705', 'AT5G44370',\n",
       "       'AT2G15230', 'AT2G16070', 'AT4G24330', 'AT1G07570', 'AT5G41860',\n",
       "       'AT3G20430', 'AT5G41240', 'AT1G20960', 'AT3G26744', 'AT5G09260',\n",
       "       'AT5G26760', 'AT5G27840', 'AT3G15260', 'AT5G07960', 'AT5G48600',\n",
       "       'AT1G61660', 'AT3G17590', 'AT1G03040', 'AT1G73590', 'AT2G45340',\n",
       "       'AT5G56590', 'AT3G20630', 'AT1G09770', 'AT1G30680', 'AT3G49601',\n",
       "       'AT5G26940', 'AT5G54540', 'AT2G43720', 'AT2G27800', 'AT3G48310',\n",
       "       'AT5G61150', 'AT1G55170', 'AT2G26190', 'AT5G08420', 'AT5G45480',\n",
       "       'AT5G66030', 'AT1G75990', 'AT3G21140', 'AT1G79090', 'AT3G26340',\n",
       "       'AT4G14350', 'AT5G23920', 'AT5G59830', 'AT1G09760', 'AT1G11592',\n",
       "       'AT5G55125', 'AT2G23080', 'AT2G38450', 'AT5G46190', 'AT5G44390',\n",
       "       'AT2G32910', 'AT3G02920', 'AT5G66360', 'AT1G18040', 'AT1G79985',\n",
       "       'AT4G35730', 'AT1G26170', 'AT4G16800', 'AT4G02390', 'AT5G05520',\n",
       "       'AT1G65280', 'AT5G44800', 'AT3G17160', 'AT5G38840', 'AT2G13540',\n",
       "       'AT3G54420', 'AT4G39850', 'AT5G51840', 'AT1G61000', 'AT1G60995',\n",
       "       'AT5G23100', 'AT1G01940', 'AT2G01913', 'AT5G24240', 'AT5G04130',\n",
       "       'AT1G79940', 'AT4G30250', 'AT1G09570', 'AT5G14440', 'AT3G04020',\n",
       "       'AT5G58150', 'AT2G18330', 'AT2G21290', 'AT2G37630', 'AT4G24940',\n",
       "       'AT5G09250', 'AT5G65770', 'AT5G51170', 'AT3G53235', 'AT5G55530'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "top_200_spearman[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "c1ed8582",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['AT1G01010', 'AT1G01020', 'AT1G01030', 'AT1G01040', 'AT1G01050',\n",
       "       'AT1G01060', 'AT1G01070', 'AT1G01080', 'AT1G01090', 'AT1G01100',\n",
       "       ...\n",
       "       'ArthCt115', 'ArthCr091', 'ArthCr090', 'ArthCr089', 'ArthCt114',\n",
       "       'ArthCt124', 'ArthCr088', 'ArthCt112', 'ArthCp088', 'ArthCp085'],\n",
       "      dtype='object', length=15091)"
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "short_transcriptome.T.index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9a5eca0c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "d30dd735",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/nk/6xkk9sgn1pz4ff1b36sfq3y40000gt/T/ipykernel_63247/379467233.py:23: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`\n",
      "  contig_table = [[(term_of_go - go_subset_distribution_num_genes_annotatated_with_go_term[i]),go_subset_distribution_num_genes_annotatated_with_go_term[i]],[(background_dipstribution_num_genes_that_lack_go_term[i]-go_subset_distribution_num_genes_that_lack_go_term[i]),go_subset_distribution_num_genes_that_lack_go_term[i]]]\n"
     ]
    }
   ],
   "source": [
    "bacteria_one = generate_contigency_tables(go_annotations=filled_matrixed_version, gene_list=df_top200.loc['Sphingomonas_L_486704'].to_list(), use_background_genes = True, background_genes = short_transcriptome.T.index)\n",
    "#bacteria_one = generate_contigency_tables(go_annotations=filled_matrixed_version, gene_list=df_top200.loc['Sphingomonas_L_486704'].to_list(),)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "id": "3f39651c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>GO term accession</th>\n",
       "      <th>GO:0000001</th>\n",
       "      <th>GO:0000009</th>\n",
       "      <th>GO:0000012</th>\n",
       "      <th>GO:0000014</th>\n",
       "      <th>GO:0000023</th>\n",
       "      <th>GO:0000024</th>\n",
       "      <th>GO:0000025</th>\n",
       "      <th>GO:0000026</th>\n",
       "      <th>GO:0000027</th>\n",
       "      <th>GO:0000028</th>\n",
       "      <th>...</th>\n",
       "      <th>GO:2001071</th>\n",
       "      <th>GO:2001080</th>\n",
       "      <th>GO:2001141</th>\n",
       "      <th>GO:2001142</th>\n",
       "      <th>GO:2001143</th>\n",
       "      <th>GO:2001147</th>\n",
       "      <th>GO:2001227</th>\n",
       "      <th>GO:2001280</th>\n",
       "      <th>GO:2001289</th>\n",
       "      <th>GO:2001294</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GeneID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AT1G01010</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT1G01020</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT1G01030</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT1G01040</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT1G01046</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT5G67600</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT5G67610</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT5G67620</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT5G67630</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT5G67640</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>28537 rows × 6055 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "GO term accession  GO:0000001  GO:0000009  GO:0000012  GO:0000014  GO:0000023  \\\n",
       "GeneID                                                                          \n",
       "AT1G01010                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT1G01020                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT1G01030                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT1G01040                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT1G01046                 0.0         0.0         0.0         0.0         0.0   \n",
       "...                       ...         ...         ...         ...         ...   \n",
       "AT5G67600                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT5G67610                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT5G67620                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT5G67630                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT5G67640                 0.0         0.0         0.0         0.0         0.0   \n",
       "\n",
       "GO term accession  GO:0000024  GO:0000025  GO:0000026  GO:0000027  GO:0000028  \\\n",
       "GeneID                                                                          \n",
       "AT1G01010                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT1G01020                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT1G01030                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT1G01040                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT1G01046                 0.0         0.0         0.0         0.0         0.0   \n",
       "...                       ...         ...         ...         ...         ...   \n",
       "AT5G67600                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT5G67610                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT5G67620                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT5G67630                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT5G67640                 0.0         0.0         0.0         0.0         0.0   \n",
       "\n",
       "GO term accession  ...  GO:2001071  GO:2001080  GO:2001141  GO:2001142  \\\n",
       "GeneID             ...                                                   \n",
       "AT1G01010          ...         0.0         0.0         0.0         0.0   \n",
       "AT1G01020          ...         0.0         0.0         0.0         0.0   \n",
       "AT1G01030          ...         0.0         0.0         0.0         0.0   \n",
       "AT1G01040          ...         0.0         0.0         0.0         0.0   \n",
       "AT1G01046          ...         0.0         0.0         0.0         0.0   \n",
       "...                ...         ...         ...         ...         ...   \n",
       "AT5G67600          ...         0.0         0.0         0.0         0.0   \n",
       "AT5G67610          ...         0.0         0.0         0.0         0.0   \n",
       "AT5G67620          ...         0.0         0.0         0.0         0.0   \n",
       "AT5G67630          ...         0.0         0.0         0.0         0.0   \n",
       "AT5G67640          ...         0.0         0.0         0.0         0.0   \n",
       "\n",
       "GO term accession  GO:2001143  GO:2001147  GO:2001227  GO:2001280  GO:2001289  \\\n",
       "GeneID                                                                          \n",
       "AT1G01010                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT1G01020                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT1G01030                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT1G01040                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT1G01046                 0.0         0.0         0.0         0.0         0.0   \n",
       "...                       ...         ...         ...         ...         ...   \n",
       "AT5G67600                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT5G67610                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT5G67620                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT5G67630                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT5G67640                 0.0         0.0         0.0         0.0         0.0   \n",
       "\n",
       "GO term accession  GO:2001294  \n",
       "GeneID                         \n",
       "AT1G01010                 0.0  \n",
       "AT1G01020                 0.0  \n",
       "AT1G01030                 0.0  \n",
       "AT1G01040                 0.0  \n",
       "AT1G01046                 0.0  \n",
       "...                       ...  \n",
       "AT5G67600                 0.0  \n",
       "AT5G67610                 0.0  \n",
       "AT5G67620                 0.0  \n",
       "AT5G67630                 0.0  \n",
       "AT5G67640                 0.0  \n",
       "\n",
       "[28537 rows x 6055 columns]"
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filled_matrixed_version"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "88641524",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>GO term accession</th>\n",
       "      <th>GO:0000001</th>\n",
       "      <th>GO:0000009</th>\n",
       "      <th>GO:0000012</th>\n",
       "      <th>GO:0000014</th>\n",
       "      <th>GO:0000023</th>\n",
       "      <th>GO:0000024</th>\n",
       "      <th>GO:0000025</th>\n",
       "      <th>GO:0000026</th>\n",
       "      <th>GO:0000027</th>\n",
       "      <th>GO:0000028</th>\n",
       "      <th>...</th>\n",
       "      <th>GO:2001071</th>\n",
       "      <th>GO:2001080</th>\n",
       "      <th>GO:2001141</th>\n",
       "      <th>GO:2001142</th>\n",
       "      <th>GO:2001143</th>\n",
       "      <th>GO:2001147</th>\n",
       "      <th>GO:2001227</th>\n",
       "      <th>GO:2001280</th>\n",
       "      <th>GO:2001289</th>\n",
       "      <th>GO:2001294</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AT1G01010</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT1G01020</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT1G01030</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT1G01040</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT1G01050</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT5G67600</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT5G67610</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT5G67620</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT5G67630</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AT5G67640</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>14601 rows × 6055 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "GO term accession  GO:0000001  GO:0000009  GO:0000012  GO:0000014  GO:0000023  \\\n",
       "AT1G01010                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT1G01020                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT1G01030                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT1G01040                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT1G01050                 0.0         0.0         0.0         0.0         0.0   \n",
       "...                       ...         ...         ...         ...         ...   \n",
       "AT5G67600                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT5G67610                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT5G67620                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT5G67630                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT5G67640                 0.0         0.0         0.0         0.0         0.0   \n",
       "\n",
       "GO term accession  GO:0000024  GO:0000025  GO:0000026  GO:0000027  GO:0000028  \\\n",
       "AT1G01010                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT1G01020                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT1G01030                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT1G01040                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT1G01050                 0.0         0.0         0.0         0.0         0.0   \n",
       "...                       ...         ...         ...         ...         ...   \n",
       "AT5G67600                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT5G67610                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT5G67620                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT5G67630                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT5G67640                 0.0         0.0         0.0         0.0         0.0   \n",
       "\n",
       "GO term accession  ...  GO:2001071  GO:2001080  GO:2001141  GO:2001142  \\\n",
       "AT1G01010          ...         0.0         0.0         0.0         0.0   \n",
       "AT1G01020          ...         0.0         0.0         0.0         0.0   \n",
       "AT1G01030          ...         0.0         0.0         0.0         0.0   \n",
       "AT1G01040          ...         0.0         0.0         0.0         0.0   \n",
       "AT1G01050          ...         0.0         0.0         0.0         0.0   \n",
       "...                ...         ...         ...         ...         ...   \n",
       "AT5G67600          ...         0.0         0.0         0.0         0.0   \n",
       "AT5G67610          ...         0.0         0.0         0.0         0.0   \n",
       "AT5G67620          ...         0.0         0.0         0.0         0.0   \n",
       "AT5G67630          ...         0.0         0.0         0.0         0.0   \n",
       "AT5G67640          ...         0.0         0.0         0.0         0.0   \n",
       "\n",
       "GO term accession  GO:2001143  GO:2001147  GO:2001227  GO:2001280  GO:2001289  \\\n",
       "AT1G01010                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT1G01020                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT1G01030                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT1G01040                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT1G01050                 0.0         0.0         0.0         0.0         0.0   \n",
       "...                       ...         ...         ...         ...         ...   \n",
       "AT5G67600                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT5G67610                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT5G67620                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT5G67630                 0.0         0.0         0.0         0.0         0.0   \n",
       "AT5G67640                 0.0         0.0         0.0         0.0         0.0   \n",
       "\n",
       "GO term accession  GO:2001294  \n",
       "AT1G01010                 0.0  \n",
       "AT1G01020                 0.0  \n",
       "AT1G01030                 0.0  \n",
       "AT1G01040                 0.0  \n",
       "AT1G01050                 0.0  \n",
       "...                       ...  \n",
       "AT5G67600                 0.0  \n",
       "AT5G67610                 0.0  \n",
       "AT5G67620                 0.0  \n",
       "AT5G67630                 0.0  \n",
       "AT5G67640                 0.0  \n",
       "\n",
       "[14601 rows x 6055 columns]"
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filled_matrixed_version.loc[filled_matrixed_version.index.intersection(short_transcriptome.T.index)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "id": "cf635b9a",
   "metadata": {},
   "outputs": [],
   "source": [
    "bacteria_one_fishers_results = fishers_exact_on_contigency_tables(bacteria_one, filled_matrixed_version)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "id": "50d479f5",
   "metadata": {},
   "outputs": [],
   "source": [
    "corrected_bacteria_one_reults = multi_hypothesis_correct_fishers_exact(bacteria_one_fishers_results)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "id": "46038c35",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>P_value</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO term accession</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GO:0015250</th>\n",
       "      <td>1.738355e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0006833</th>\n",
       "      <td>2.534916e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0010438</th>\n",
       "      <td>2.246746e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0055085</th>\n",
       "      <td>1.750075e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0010431</th>\n",
       "      <td>1.773608e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2001227</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2001280</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2001289</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2001294</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000001</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6055 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                        P_value\n",
       "GO term accession              \n",
       "GO:0015250         1.738355e-10\n",
       "GO:0006833         2.534916e-10\n",
       "GO:0010438         2.246746e-06\n",
       "GO:0055085         1.750075e-05\n",
       "GO:0010431         1.773608e-04\n",
       "...                         ...\n",
       "GO:2001227         1.000000e+00\n",
       "GO:2001280         1.000000e+00\n",
       "GO:2001289         1.000000e+00\n",
       "GO:2001294         1.000000e+00\n",
       "GO:0000001         1.000000e+00\n",
       "\n",
       "[6055 rows x 1 columns]"
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bacteria_one_fishers_results.sort_values(by = 'P_value', ascending = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "id": "1e72d826",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>P_value</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO term accession</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GO:0015250</th>\n",
       "      <td>7.674459e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0006833</th>\n",
       "      <td>7.674459e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0010438</th>\n",
       "      <td>4.534683e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0055085</th>\n",
       "      <td>2.649176e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0010431</th>\n",
       "      <td>2.147839e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0042754</th>\n",
       "      <td>3.424299e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0015171</th>\n",
       "      <td>3.424299e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0061928</th>\n",
       "      <td>7.096125e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0003985</th>\n",
       "      <td>7.096125e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000234</th>\n",
       "      <td>7.096125e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2001141</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2001080</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2001071</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2001070</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2001066</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2001057</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2001011</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2001009</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2001008</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2001007</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2001006</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2000904</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2000786</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2000785</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2000769</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2000762</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000041</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000038</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000036</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000035</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000033</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000030</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000028</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000027</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000026</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000025</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000024</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000023</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000014</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000012</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000009</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:2001142</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000146</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000145</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000139</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000138</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000137</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000132</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000127</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GO:0000123</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        P_value\n",
       "GO term accession              \n",
       "GO:0015250         7.674459e-07\n",
       "GO:0006833         7.674459e-07\n",
       "GO:0010438         4.534683e-03\n",
       "GO:0055085         2.649176e-02\n",
       "GO:0010431         2.147839e-01\n",
       "GO:0042754         3.424299e-01\n",
       "GO:0015171         3.424299e-01\n",
       "GO:0061928         7.096125e-01\n",
       "GO:0003985         7.096125e-01\n",
       "GO:0000234         7.096125e-01\n",
       "GO:2001141         1.000000e+00\n",
       "GO:2001080         1.000000e+00\n",
       "GO:2001071         1.000000e+00\n",
       "GO:2001070         1.000000e+00\n",
       "GO:2001066         1.000000e+00\n",
       "GO:2001057         1.000000e+00\n",
       "GO:2001011         1.000000e+00\n",
       "GO:2001009         1.000000e+00\n",
       "GO:2001008         1.000000e+00\n",
       "GO:2001007         1.000000e+00\n",
       "GO:2001006         1.000000e+00\n",
       "GO:2000904         1.000000e+00\n",
       "GO:2000786         1.000000e+00\n",
       "GO:2000785         1.000000e+00\n",
       "GO:2000769         1.000000e+00\n",
       "GO:2000762         1.000000e+00\n",
       "GO:0000041         1.000000e+00\n",
       "GO:0000038         1.000000e+00\n",
       "GO:0000036         1.000000e+00\n",
       "GO:0000035         1.000000e+00\n",
       "GO:0000033         1.000000e+00\n",
       "GO:0000030         1.000000e+00\n",
       "GO:0000028         1.000000e+00\n",
       "GO:0000027         1.000000e+00\n",
       "GO:0000026         1.000000e+00\n",
       "GO:0000025         1.000000e+00\n",
       "GO:0000024         1.000000e+00\n",
       "GO:0000023         1.000000e+00\n",
       "GO:0000014         1.000000e+00\n",
       "GO:0000012         1.000000e+00\n",
       "GO:0000009         1.000000e+00\n",
       "GO:2001142         1.000000e+00\n",
       "GO:0000146         1.000000e+00\n",
       "GO:0000145         1.000000e+00\n",
       "GO:0000139         1.000000e+00\n",
       "GO:0000138         1.000000e+00\n",
       "GO:0000137         1.000000e+00\n",
       "GO:0000132         1.000000e+00\n",
       "GO:0000127         1.000000e+00\n",
       "GO:0000123         1.000000e+00"
      ]
     },
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "corrected_bacteria_one_reults.sort_values(by = 'P_value', ascending = True).head(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7945e25f",
   "metadata": {},
   "outputs": [],
   "source": [
    "shannon_diversity_correlation = pd.read_csv('/Users/michael/Data/Luke_terrace_experiment/General_data/Junk_data/gene_spearman_correlation_with_shannon_diversity.csv')\n",
    "shannon_diversity_correlation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "ae1ac511",
   "metadata": {},
   "outputs": [],
   "source": [
    "dbfile_1 = open('/Users/michael/Data/Junk_data/gene_pearson_correlation_with_bacteria.p', 'rb')    \n",
    "full_correlation_matrix = pickle.load(dbfile_1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "beef2b13",
   "metadata": {},
   "outputs": [],
   "source": [
    "top_200_pearson = np.array([full_correlation_matrix.T[c].nlargest(200).index.values for c in full_correlation_matrix.T])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ac254419",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_top200_pearson = pd.DataFrame(index = full_correlation_matrix.index, data = top_200_pearson)\n",
    "df_top200"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "40721cf4",
   "metadata": {},
   "outputs": [],
   "source": [
    "#df_top200_pearson.to_csv('/Users/michael/Data/Luke_terrace_experiment/Output_for_Luke/top_200_pearson_correlated_genes_per_bacteria.csv',)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Luke_terrace",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
